---
title: 'Gemma API R Wrapper'
author: 'Javier Castillo-Arnemann, Jordan Sicherman'
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Gemma API R Wrapper}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
# Prevent certificate issues for GitHub actions
options(gemma.SSL = FALSE)

knitr::opts_chunk$set(
  comment = '#>',
  collapse = FALSE
  )
```

```{r setup, message = FALSE}
library(gemmaAPI)
library(Biobase)
library(dplyr)
library(data.table)
library(memoise)
library(microbenchmark)
```

## About Gemma
[Gemma](https://gemma.ubc.msl.ca/) is a web site, database and a set of tools for the meta-analysis, re-use and sharing of genomics data, currently primarily targeted at the analysis of gene expression profiles. Gemma contains data from thousands of public studies, referencing thousands of published papers. Every dataset in Gemma has passed a rigorous curation process that re-annotates the expression platform at the sequence level, which allows for more consistent cross-platform comparisons and meta-analyses.

For detailed information on the curation process, check this [page](https://pavlidislab.github.io/Gemma/curation.html) or the latest [publication](https://academic.oup.com/database/article/doi/10.1093/database/baab006/6143045).

## Using the Gemma API R Wrapper
The main goal of this wrapper is to give easy access to Gemma's curated datasets for downstream analyses or meta-analyses combining multiple datasets. In this example, we want to find datasets that are associated with bipolar disorder, and we are only interested in human data. In addition, we'll subset our results to datasets that have been batch corrected. To do this, we use the [`searchDatasets()`](https://pavlidislab.github.io/Gemma-API/reference/searchDatasets.html) endpoint:

```{r search}
searchDatasets('bipolar', limit = 100) %>%
  filter(taxon.Name == 'human', geeq.batchCorrected == TRUE) %>%
  select(ee.ShortName, ee.Name, ee.Samples)
```

We are left with two datasets, and we'll pick [GSE35974](https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?id=5939) since it has the larger number of samples.
Now that we have the ID for our experiment, we can fetch the data associated with it.

To get the basic information of a dataset, we use the [`getDatasets()`](https://pavlidislab.github.io/Gemma-API/reference/getDatasets.html) endpoint.

```{r dataset}
dataset <- getDatasets('GSE35974')
colnames(dataset)
dataset$ee.Description
```

To access the actual expression data, you can use [`getBioc()`](https://pavlidislab.github.io/Gemma-API/reference/getBioc.html).
It is a high-level wrapper that combines various
endpoints to return an annotated [`SummarizedExperiment`](https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html) or [`ExpressionSet`](https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf) of the queried dataset that is compatible with other Bioconductor packages for downstream analyses. They include the expression matrix along with
the experimental design, and ensure the sample names match between both when transforming/subsetting data. 

Alternatively, you can use [`getDatasetData()`](https://pavlidislab.github.io/Gemma-API/reference/getDatasetData.html) to get the expression matrix directly, and [`getDatasetDesign()`](https://pavlidislab.github.io/Gemma-API/reference/getDatasetDesign.html) to get the matrix with the experimental design. 

Since this dataset also contains patients with schizophrenia and depression, and we're only interested in bipolar patients, we can remove those samples.
We're also keeping the `reference_subject_role`s, which refers to the control samples in Gemma datasets.
```{r bioconductor, eval = FALSE}
# Create ExpressionSet
eset <- getBioc('ExpressionSet', 'GSE35974')
eset

# Check the levels of the DiseaseState cofactor
eset$DiseaseState %>% unique()

# Subset patients with bipolar disorder and controls.
bipolar <- eset[ , eset$DiseaseState == 'Bipolar_Disorder' | eset$DiseaseState == 'reference_subject_role']
bipolar
```

```{r echo = FALSE}
## This code pre-loads the eset to save time
 
# Load ExpressionSet
eset <- readRDS('eset.rds')
eset

# Check the levels of the DiseaseState cofactor
eset$DiseaseState %>% unique()

# Subset patients with bipolar disorder and controls.
bipolar <- eset[ , eset$DiseaseState == 'Bipolar_Disorder' | eset$DiseaseState == 'reference_subject_role']
bipolar
```

Let's check the expression for every sample to make sure they look OK:
```{r base-boxplot}
# Get Expression matrix
bipExpr <- exprs(bipolar)

boxplot(bipExpr, pch='.', xaxt='n', xlab='Sample', ylab='Expression')
```

We can quickly see three samples among bipolar patients and controls are empty. This is because during the curation process Gemma flags potential outlier samples automatically, which are then manually reviewed. Samples confirmed to be outliers are represented as missing data to ensure transparency.

You can also use [`getTidyDataset()`](https://pavlidislab.github.io/Gemma-API/reference/getTidyDataset.html) to fetch data in a long, 'tidy' format that combines the expression matrix with the metadata. This is less memory-efficient than the two alternatives above, but allows quick and easy visualization and exploration of the data with [`ggplot2`](https://ggplot2.tidyverse.org/).

```{r tidy, eval = FALSE}
tidyData <- getTidyDataset('GSE35974')
```


### Differential Expression Example (TODO)
From this point on, you can use [`limma`](https://bioconductor.org/packages/release/bioc/html/limma.html) or other alternatives in Bioconductor to compare the gene expression of bipolar patients with healthy controls and determine which genes showed the most significant changes.

### Longer queries
Many endpoints also allow you to query multiple objects at the same time. For example, getting information on 3 datasets at the same time.

```{r double-query}
getDatasets(datasets = c('GSE35974', 'GSE12649', 'GSE99349')) %>%
  glimpse()
```

You'll be able to tell which endpoints accept multiple inputs from the function examples or simply because the the parameter name is plural in the documentation. The rest of the endpoints only support a single identifier:

```{r error, error = TRUE}
getDatasetAnnotations(c('GSE35974', 'GSE12649'))
```

In these cases, you will have to loop over all the identifiers you wish to query and send separate requests.

```{r loop}
lapply(c('GSE35974', 'GSE12649'), function(dataset) {
  getDatasetAnnotations(dataset) %>% mutate(shortName = dataset)
}) %>% rbindlist
```

### Memoise data
To speed up results, you can remember past results so future queries can proceed virtually instantly. This is enabled through the [`memoise`](https://github.com/r-lib/memoise) package. To enable memoisation, simply set `memoised = T`.
```{r memoised, eval=FALSE}
knitr::kable(summary(microbenchmark(getDatasetData('GSE35974', memoised = TRUE), times = 1), unit = 'ms'))
knitr::kable(summary(microbenchmark(getDatasetData('GSE35974', memoised = TRUE), times = 1), unit = 'ms'))
```

If you're done with your fetching and want to ensure no space is being used for cached results, or if you just want to ensure you're getting up-to-date data from Gemma, you can clear the cache using `forgetGemmaMemoised`.

```{r clear-cache}
forgetGemmaMemoised()
```

## Output options

### Raw data
By default, Gemma API does some parsing on the raw API results to make it easier to work with inside of R. In the process, it drops some typically unused values. If you wish to fetch everything, use `raw = T`. Instead of a `data.table`, you'll be served (typically) a list that represents the underlying JSON response.
```{r raw}
getDatasets('GSE35974', raw = TRUE) %>%
  glimpse()
```

### File outputs
Sometimes, you may wish to save results to a file for future inspection. You can do this simply by providing a filename to `file`. The extension for this file will be one of three options:

1.  `.json`, if you requested results with `raw = TRUE`
2.  `.csv` if the results have no nested `data.table`s
3.  `.rds` otherwise

You can also specify whether or not the new fetched results are allowed to overwrite an existing file by specifying the `overwrite` parameter.

### Changing defaults
We've seen how to change `memoised = T`, `overwrite = T` and `raw = T`... It's possible that you want to always use the functions these ways without specifying the option every time. You can do this by simply changing the default, which is visible in the function definition. For example, if you want to memoise results by default, you can use:

```{r defaults}
options(gemma.memoise = TRUE)
```
