---
title: 'Accessing curated gene expression data with gemma.R'
author:
- name: Javier Castillo-Arnemann
  affiliation: Michael Smith Laboratories, University of British Columbia, Vancouver, Canada
- name: Jordan Sicherman
  affiliation: Michael Smith Laboratories, University of British Columbia, Vancouver, Canada
package: gemma.R
output:
    BiocStyle::html_document
header-includes:
    - \DeclareUnicodeCharacter{2009}{ } 
vignette: >
    %\VignetteIndexEntry{Accessing curated gene expression data with gemma.R}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
# Prevent certificate issues for GitHub actions
options(gemma.SSL = FALSE,gemma.memoised = TRUE)

knitr::opts_chunk$set(
    comment = ""
)
```

```{r setup, message = FALSE}
library(gemma.R)
library(dplyr)
library(ggplot2)
library(ggrepel)
library(SummarizedExperiment)
```
```{r, include = FALSE}
forgetGemmaMemoised() # to make sure local tests don't succeed because of history
```

# About Gemma
[Gemma](https://gemma.msl.ubc.ca/) is a web site, database and a set of tools for the meta-analysis, re-use and sharing of genomics data, currently primarily targeted at the analysis of gene expression profiles. Gemma contains data from thousands of public studies, referencing thousands of published papers. Every dataset in Gemma has passed a rigorous curation process that re-annotates the expression platform at the sequence level, which allows for more consistent cross-platform comparisons and meta-analyses.

For detailed information on the curation process, read this [page](https://pavlidislab.github.io/Gemma/curation.html) or the latest [publication](https://academic.oup.com/database/article/doi/10.1093/database/baab006/6143045).

# Installation instructions


## Bioconductor (not available yet)
You can install `gemma.R` through [Bioconductor](http://bioconductor.org/) with the following code:

```{r 'install', eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("gemma.R")
```

## Development

The current development version of `gemma.R` can be installed with:

```{r, eval = FALSE}
if (!requireNamespace("devtools", quietly = TRUE)) {
    install.packages("devtools")
}
devtools::install_github("PavlidisLab/gemma.R")
```

# Downloading expression data
The main goal of this wrapper is to enable easy access to Gemma's curated datasets for downstream analyses or meta-analyses combining multiple datasets. In this example, we want to find datasets that are associated with bipolar disorder, and we are only interested in human data. In addition, we'll subset our results to datasets that have been batch corrected.

```{r search}
searchDatasets("bipolar", limit = 100, taxon = "human") %>%
    filter(geeq.batchCorrected == TRUE) %>%
    select(ee.ShortName, ee.Name, ee.ID, ee.Accession, ee.Samples)
```

We are left with two datasets. For simplicity, we'll pick [GSE46416](https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?id=8997) since it has the smaller number of samples. Now that we have the ID for our experiment, we can fetch the data associated with it.

```{r dataset}
getDatasetsInfo("GSE46416") %>%
    select(ee.ShortName, ee.Name, ee.ID, ee.Description)
```

To access the expression data in a convenient form, you can use [`getDataset()`](https://pavlidislab.github.io/gemma.R/reference/getDataset.html).
It is a high-level wrapper that combines various
endpoint calls to return an annotated [`SummarizedExperiment`](https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html) or [`ExpressionSet`](https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf) of the queried dataset that is compatible with other Bioconductor packages for downstream analyses. They include the expression matrix along with
the experimental design, and ensure the sample names match between both when transforming/subsetting data. 

```{r load-expression, eval = TRUE}
dat <- getDataset("GSE46416")
```

```{r include = FALSE}
## Pre-loads the summarized experiment to save time and pass vignette build
# dat <- readRDS("se.rds")
```

To show how subsetting works, we'll keep the manic phase data and the `reference_subject_role`s, which refers to the control samples in Gemma datasets.
```{r}
# Check the levels of the disease factor
dat$disease %>% unique()

# Subset patients during manic phase and controls
manic <- dat[, dat$disease == "bipolar_disorder_|_manic_phase_|" | 
        dat$disease == "reference_subject_role"]
manic
```

Let's check the expression for every sample to make sure they look OK:
```{r boxplot, fig.cap="Gene expression distributions of bipolar patients during manic phase and controls."}
# Get Expression matrix
manicExpr <- assay(manic, "counts")
boxplot(manicExpr, pch = ".", xaxt = "n", xlab = "Sample", ylab = "Expression")
```

You can also use [`getDatasetExpression()`](https://pavlidislab.github.io/gemma.R/reference/getDatasetExpression.html) to only get the expression matrix, and [`getDatasetDesign()`](https://pavlidislab.github.io/gemma.R/reference/getDatasetDesign.html) to get the experimental design matrix. Finally, you can use [`getDatasetTidy()`](https://pavlidislab.github.io/gemma.R/reference/getDatasetTidy.html) to get a tibble that combines the expression and design matrices in [tidy format](https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html). This format is less memory-efficient than [`getDataset()`](https://pavlidislab.github.io/gemma.R/reference/getDataset.html), but allows easy visualization and exploration with [ggplot2](https://ggplot2.tidyverse.org/index.html) and the rest of the [tidyverse](https://www.tidyverse.org/).

## Differential expression analyses
Gemma also contains precomputed differential expression analyses for most of its datasets, and some datasets contain more than one analysis to account for different factors and their interactions. These tables are stored as resultSets, and you can access them using [`getDatasetDE()`](https://pavlidislab.github.io/gemma.R/reference/getDatasetDE.html). From here on, we can explore and visualize the data to find the most differentially-expressed genes:

```{r diffExpr, fig.cap="Differentially-expressed genes in bipolar patients during manic phase versus controls.", fig.wide=TRUE, warning = FALSE}
de <- getDatasetDE("GSE46416")[[1]]

# Classify probes for plotting
de$diffexpr <- "No"
de$diffexpr[de$`contrast_bipolar disorder, manic phase_logFoldChange` > 1.0 & 
        de$`contrast_bipolar disorder, manic phase_pvalue` < 0.05] <- "Up"
de$diffexpr[de$`contrast_bipolar disorder, manic phase_logFoldChange` < -1.0 & 
        de$`contrast_bipolar disorder, manic phase_pvalue` < 0.05] <- "Down"

# Upregulated probes
filter(de, diffexpr == "Up") %>%
    arrange(`contrast_bipolar disorder, manic phase_pvalue`) %>%
    select(Probe, GeneSymbol, `contrast_bipolar disorder, manic phase_pvalue`, 
        `contrast_bipolar disorder, manic phase_logFoldChange`) %>%
    head(10)

# Downregulated probes
filter(de, diffexpr == "Down") %>%
    arrange(`contrast_bipolar disorder, manic phase_pvalue`) %>%
    select(Probe, GeneSymbol, `contrast_bipolar disorder, manic phase_pvalue`, 
        `contrast_bipolar disorder, manic phase_logFoldChange`) %>%
    head(10)

# Add gene symbols as labels to DE genes
de$delabel <- ""
de$delabel[de$diffexpr != "No"] <- de$GeneSymbol[de$diffexpr != "No"]

# Volcano plot for bipolar patients vs controls
ggplot(
    data = de,
    aes(
        x = `contrast_bipolar disorder, manic phase_logFoldChange`,
        y = -log10(`contrast_bipolar disorder, manic phase_pvalue`),
        color = diffexpr,
        label = delabel
    )
) +
    geom_point() +
    geom_hline(yintercept = -log10(0.05), col = "gray45", linetype = "dashed") +
    geom_vline(xintercept = c(-1.0, 1.0), col = "gray45", linetype = "dashed") +
    labs(x = "log2(FoldChange)", y = "-log10(p-value)") +
    scale_color_manual(values = c("blue", "black", "red")) +
    geom_text_repel(show.legend = FALSE) +
    theme_minimal()
```


## Larger queries
The `*Info()` endpoints accept multiple identifiers in a single function call. 
For example, getting information on 2 datasets at the same time.

```{r triple-query}
getDatasetsInfo(datasets = c("GSE35974", "GSE46416")) %>%
    select(ee.ShortName, ee.Name, ee.ID, ee.Accession, ee.Samples, taxon.Name)
```

To query large amounts of data, the API has a pagination system which uses
the `limit` and `offset` parameters. To avoid overloading the server, calls are limited
to a maximum of 100 entries, so the offset allows you to get the next batch of entries
in the next call(s). For simplicity, this example shows how pagination works with 5 entries per query.
```{r}
lapply(c(0, 5, 10), function(offset){
    getPlatformsInfo(limit = 5, offset = offset) %>%
        select(platform.ID, platform.ShortName, taxon.Name)
})
```

The rest of the endpoints only support a single identifier:
```{r error, error = TRUE}
getDatasetAnnotations(c("GSE35974", "GSE46416"))
```

In these cases, you will have to loop over all the identifiers you wish to query and send separate requests.

```{r loop}
lapply(c("GSE35974", "GSE12649"), function(dataset) {
    getDatasetAnnotations(dataset) %>% 
        mutate(ee.shortName = dataset) %>%
        select(ee.shortName, class.Type, class.Name, term.Name)
})
```

# Output options

## Raw data
By default, Gemma API does some parsing on the raw API results to make it easier to work with inside of R. In the process, it drops some typically unused values. If you wish to fetch everything, use `raw = TRUE`. Instead of a data table, you'll usually be served a list that represents the underlying JSON response.

```{r}
getGeneLocation("DYRK1A")

getGeneLocation("DYRK1A", raw = TRUE)
```

## File outputs
Sometimes, you may wish to save results to a file for future inspection. You can do this simply by providing a filename to the `file` parameter. The extension for this file will be one of three options:

1.  `.json`, if you requested results with `raw=TRUE`
2.  `.csv` if the results have no nested data tables
3.  `.rds` otherwise

You can also specify whether or not the new fetched results are allowed to overwrite an existing file by specifying the `overwrite = TRUE` parameter.

## Memoise data
To speed up results, you can remember past results so future queries can proceed virtually instantly. This is enabled through the [`memoise`](https://github.com/r-lib/memoise) package. To enable memoisation, simply set `memoised = TRUE` in the function call whenever you want to refer to the cache, both to save data for future calls and use the saved data for repeated calls.

By default this will create a cache in your local filesystem. 

You can see
the path to your cache by using `rappdirs::user_cache_dir(appname = "gemmaR")`. If you want to use a different directory, use
`options(gemma.cache = 'path')`.

If you're done with your fetching and want to ensure no space is being used for cached results, or if you just want to ensure you're getting up-to-date data from Gemma, you can clear the cache using [`forgetGemmaMemoised()`](https://pavlidislab.github.io/gemma.R/reference/forgetGemmaMemoised.html).

## Changing defaults
We've seen how to change `raw = TRUE`, `overwrite = TRUE` and `memoised = TRUE` in individual function calls. It's possible that you want to always use the functions these ways without specifying the option every time. You can do this by simply changing the default, which is visible in the function definition. See below for examples.

```{r defaults, eval = FALSE}
options(gemma.memoised = TRUE) # always refer to cache
options(gemma.overwrite = TRUE) # always overwrite when saving files
options(gemma.raw = TRUE) # always recieve results as-is from Gemma
```


```{r,include = FALSE}
options(gemma.memoised = FALSE)

```

# Session info
```{r}
sessionInfo()
```

