<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve all datasets — get_datasets • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve all datasets — get_datasets"><meta property="og:description" content="Retrieve all datasets"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve all datasets</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>get_datasets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieve all datasets</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_datasets</span><span class="op">(</span></span>
<span>  query <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  filter <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  taxa <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  uris <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  offset <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  sort <span class="op">=</span> <span class="st">"+id"</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>The search query. Either plain text ('traumatic'), or an ontology
term URI ('http://purl.obolibrary.org/obo/UBERON_0002048'). Datasets that
contain the given string in their short of full name will also be matched.</p></dd>


<dt>filter</dt>
<dd><p>Filter results by matching expression. Use <code><a href="filter_properties.html">filter_properties</a></code>
function to get a list of all available parameters. These properties can be
combined using "and" "or" clauses and may contain common operators such as "=", "&lt;" or "in".
(e.g. "taxon.commonName = human", "taxon.commonName in (human,mouse), "id &lt; 1000")</p></dd>


<dt>taxa</dt>
<dd><p>A vector of taxon common names (e.g. human, mouse, rat). Providing multiple
species will return results for all species. These are appended
to the filter and equivalent to filtering for <code>taxon.commonName</code> property</p></dd>


<dt>uris</dt>
<dd><p>A vector of ontology term URIs. Providing multiple terms will
return results containing any of the terms and their children. These are</p></dd>


<dt>offset</dt>
<dd><p>The offset of the first retrieved result.</p></dd>


<dt>limit</dt>
<dd><p>Optional, defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt>sort</dt>
<dd><p>Order results by the given property and direction. The '+' sign
indicate ascending order whereas the '-' indicate descending.</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data table with information about the queried dataset(s). A list if
<code>raw = TRUE</code>. Returns an empty list if no datasets matched. A successful
response may contain 'Geeq' information, which aims to provide a unified
metric to measure experiments by the quality of their data, and their
suitability for use in Gemma. You can
read more about the geeq properties <a href="https://pavlidislab.github.io/Gemma/geeq.html" class="external-link">here</a>.</p>


<p>The fields of the output data.table are:</p>


<ul><li><p><code>experiment.ShortName</code>: Shortname given to the dataset within Gemma. Often corresponds to accession ID</p></li>
<li><p><code>experiment.Name</code>: Full title of the dataset</p></li>
<li><p><code>experiment.ID</code>: Internal ID of the dataset.</p></li>
<li><p><code>experiment.Description</code>: Description of the dataset</p></li>
<li><p><code>experiment.Troubled</code>: Did an automatic process within gemma or a curator mark the dataset as "troubled"</p></li>
<li><p><code>experiment.Accession</code>: Accession ID of the dataset in the external database it was taken from</p></li>
<li><p><code>experiment.Database</code>: The name of the database where the dataset was taken from</p></li>
<li><p><code>experiment.URI</code>: URI of the original database</p></li>
<li><p><code>experiment.SampleCount</code>: Number of samples in the dataset</p></li>
<li><p><code>experiment.batchEffect</code>: A text field describing whether the dataset has batch effects</p></li>
<li><p><code>geeq.batchCorrected</code>: Whether batch correction has been performed on the dataset.</p></li>
<li><p><code>geeq.batchConfound</code>: 0 if batch info isn't available, -1 if batch counfoud is detected, 1 if batch information is available and no batch confound found</p></li>
<li><p><code>geeq.batchEffect</code>: -1 if batch p value &lt; 0.0001, 1 if batch p value &gt; 0.1, 0 if otherwise and when there is no batch information is available or when the data is confounded with batches.</p></li>
<li><p><code>geeq.rawData</code>: -1 if no raw data available, 1 if raw data was available. When available, Gemma reprocesses raw data to get expression values and batches</p></li>
<li><p><code>geeq.qScore</code>: Data quality score given to the dataset by Gemma.</p></li>
<li><p><code>geeq.sScore</code>: Suitability score given to the dataset by Gemma. Refers to factors like batches, platforms and other aspects of experimental design</p></li>
<li><p><code>taxon.Name</code>: Name of the species</p></li>
<li><p><code>taxon.Scientific</code>: Scientific name for the taxon</p></li>
<li><p><code>taxon.ID</code>: Internal identifier given to the species by Gemma</p></li>
<li><p><code>taxon.NCBI</code>: NCBI ID of the taxon</p></li>
<li><p><code>taxon.Database.Name</code>: Underlying database used in Gemma for the taxon</p></li>
<li><p><code>taxon.Database.ID</code>: ID of the underyling database used in Gemma for the taxon</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">get_datasets</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ShortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:              GSE2018</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:              GSE2872</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:              GSE4523</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:              GSE4036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:              GSE4034</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:              GSE2866</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:              GSE3253</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:              GSE2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:              GSE2426</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:              GSE2161</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:              GSE2871</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:              GSE2869</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:              GSE2868</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:              GSE2867</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:              GSE3489</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:              GSE1997</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:              GSE1837</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:              GSE2178</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:              GSE2031</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:              GSE1611</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                              experiment.Name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                              Human Lung Transplant - BAL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                    d'mel-affy-rat-168311</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                   Expression Studies of Melanotransferrin in Mouse Brain</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                  perro-affy-human-186940</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                  palme-affy-mouse-198967</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                Donarum-3R01NS040270-03S1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: Exaggerated neuroinflammation and sickness behavior in aged mice after activation of the peripheral innate immune system</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                 Hippocampus of HuD overexpressor mice. Perrone-Bizzozero-5R01NS030255-12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                  Pre-Neoplastic Stage of Medulloblastoma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:         Identification of genes that are dysregulated in the telencephalon of Dlx1/2 mutants. Rubenstein-2R01MH049428-11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                      giza-affy-rat-84719</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                Rosenfeld-5R01NS034934-14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                 Tosini-1R01NS043459-01A1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                   Zoghbi-5R01NS027699-14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                   Patterns of gene dysregulation in the frontal cortex of patients with HIV encephalitis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                               Hippocampal development in fetal alcohol exposed rats. Perrone-Bizzozero-5R01NS030255-12-3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                              Comparison of  SNc and VTA dopaminergic neurons. Greene-5P50NS038399-050001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                            Mouse Expression DB 2001 CHMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               Identification of genes and quantitative trait loci (QTL) that control hematopoietic stem cell functioning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                          Transcriptome of Ts1Cje and euploids cerebellum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:             3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:             5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:            11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:            12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:            13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:            15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:            16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:            17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:            18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:            19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:            21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:            24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:            25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:            26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         experiment.Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Bronchoalveolar lavage samples collected from lung transplant recipients.  Numeric portion of sample name is an arbitrary patient ID and AxBx number indicates the perivascular (A) and bronchiolar (B) scores from biopsies collected on the same day as the BAL fluid was collected.  Several patients have more than one sample in this series and can be determined by patient number followed by a lower case letter.  Acute rejection state is determined by the combined A and B score - specifically, a combined AB score of 2 or greater is considered an acute rejection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Melanotransferrin (MTf) or melanoma tumor antigen p97 is an iron (Fe)-binding transferrin homolog expressed highly on melanomas and at lower levels on normal tissues. It has been suggested that MTf is involved in a variety of processes such as Fe metabolism and cellular differentiation. Considering the crucial role of Fe in many metabolic pathways e.g., DNA synthesis, it is important to understand the function of MTf. To define the roles of MTf, a MTf knockout (MTf -/-) mouse model was developed. Examination of the MTf -/- mice demonstrated no phenotypic differences compared to wild-type littermates. However, microarray analysis showed differential expression of molecules involved in proliferation such as Mef2a, Tcf4, Gls and Apod in MTf -/- mice compared to MTf +/+ littermates, suggesting a role for MTf in proliferation and tumorigenesis.\nDate GSE4523 Last Updated: Jun 13 2006\nContributors:  Louise L Dunn Yohan Suryo Rahmanto Eric O Sekyere Des R Richardson\nIncludes GDS1964.\n Update date: Jun 14 2006.\n Dataset description GDS1964: Analysis of brain from melanotransferrin (MTf) knockout mutants. MTf or melanoma tumor antigen p97 is a membrane bound iron binding transferrin homolog highly expressed in melanomas and at lower levels in normal tissues. Results provide insight into the function of MTf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Our laboratory has developed the first mouse model overexpressing a RNA-binding protein, the ELAV-like protein HuD, in the CNS under the control of the CaMKinII alpha promoter. Initial behavioral characterization of the mice revealed that they had significant learning deficits together with abnormalities in prepulse inhibition (PPI). At the molecular level, we found that the expression of the growth-associated protein GAP-43, one of the targets of HuD, was increased in the hippocampus of HuD transgenic mice. To characterize these mice further and to evaluate the utility of these animals in understanding human diseases, we propose to use DNA microarray methods. To test our hypothesis we propose 3 specific aims: 1) To characterize the pattern of gene expression in the hippocampus of HuD overexpressor mice 2) To compare the pattern of gene expression in our mouse model with that in the hippocampus of rats prenatally exposed to alcohol (FAS model) and 3) To compare the pattern of gene expression in our mouse model with that shown in post-mortem tissues of patients with schizophrenia. In our previous protocols we examined the pattern of gene expression in our HuD transgenic mice and in rats prenaltally exposed to alcohol. A report by another group (Hakak et al, 2001) showed that three of the HuD targets were upregulated in the prefrontal cortex of patients with schizophrenia. To evaluate whether other target of HuD may be affected in this illness, in the current protocol, we want to compare the pattern of expression in our transgenic mice with in tissue from patients with schizophrenia Based on the behavioral and molecular properties of our HuD transgenic mice we hypothesize that these animals may be good models for the studying the basis of learning disabilities and of diseases that show deficits in PPI such as fetal alcohol syndrome and schizophrenia. All 28 samples are derived from cerebellar tissues for patients with schizophrenia and matched controls.  The specimens were obtained from the Maryland Brain Collection according to NIH guidelines for confidentially and privacy. The protocol used in these studies was reviewed by our HRRC which found that our studies do not fall within the category of protocols monitored by the IRB (see attached letter form the HRRC). Specimens from 14 patients with a diagnosis of schizophrenia performed according to DSM-IV criteria and 14 sex-, age- and PMI-matched controls was included in this study. No differences were found between patients and control subjects in the average age (45±12 versus 43±10 years, p=0.86) or PMI (12±5 versus 16±6 hours, p=0.11). We will provide 28 samples containing 5 ug of RNA each in DEPC water (see validation of the quality of the RNA below). In addition, we include in our study animals treated with haloperidol as control for medication. These samples will be submitted in a separate protocol.\nDate GSE4036 Last Updated: Jan 13 2006\nContributors:  Nora I Perrone-Bizzozero\nIncludes GDS1917.\n Update date: Jun 14 2006.\n Dataset description GDS1917: Analysis of cortical samples corresponding to the crus I/VIIa area of the cerebellum from schizophrenia patients. A study indicates that targets of the RNA-binding ELAV-like protein HuD are overexpressed in the prefrontal cortex of patients with schizophrenia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Fear conditioning (FC) is a behavioral paradigm that measures an animal's ability to learn fear related information.  FC is measured by pairing a mild foot-shock with the surroundings in which the shock was recieved.  Upon being placed back in the context, mice exhibit freezing behavior, which is a species-specific response to fear.  We have previously used selective breeding to produce lines of mice with high or low levels of freezing behavior.  This experiment is a replication of a previous experiment that produced lines of mice with high or low levels of freezing behavior.  These lines derive from different progenitor mouse strains.  We are able to identify alleles that govern the genetic variability for FC by using chromosomal markers in these selected lines.  Using microarrays, we will identify differences in gene expression in two key brain regions: amygdala and hippocampus.  Gene expression differences and data regarding chromosomal regions involved in the behavior will be compared to identify particular genes that are both differentially expressed and whose expression is governed by alleles that fall into critical chromosomal regions. We will compare gene expresion in the amygdala and hippocampus (brain regions known to be relevant to fear behavior) from the these two lines of mice and to the those in the previous experiment.  Bayesian statistics will be used in an effort to identify gene expression that affects fear behavior. We hypothesize that selection has acted in part by changing the frequency of alleles that cause differential expression of key genes in the amygdala and hippocampus of our selected lines.  Slective breeding changes the frequency of trait relevand (FC) alleles.  A relevant allele is expected to increas in one selected line and decrease in the oppositely selected line.  Some trait relevant alleles are expected to cause changes in the level of expression at particular genes. Amygdala and hippocampus will be rapidly dissected out of experimentally naïve mice from each line. Naïve mice will be used for expression studies since the behavior of the mice in the FC test can be reliably anticipated due to their lineage. We have practiced these procedures, and can accurately and reproducibly remove these regions in less than 5 minutes. Different mice will be used to collect each brain region, since the dissection of hippocampus disrupts the removal of amygdala. We will collect enough samples from each region to accommodate a total of 6 microarrays per brain region, per line, thus we will use a total of 24 microarrays. We anticipate that a single brain region will be sufficient to for a microarray. However, we propose to utilize three samples per microarray, because this will reduce variability due to environmental factors and due to slight variability in our dissection procedures. Once this tissue is removed, we will isolate RNA for shipment to the Microarray consortium. We will also collect spleens from each subject as a source of genomic DNA, in order to permit direct comparison of genotype and expression phenotypes. Once we have the results of the microarray analysis, we use WebQTL.org to identify the chromosomal locations of alleles that are know to influence the expression of genes for which we have found differential expression. We will then superimpose this information on trait relevant chromosomal regions identified from our selected lines. This will allow us to rapidly identify genes which may account for genetic variability in FC due to differential expression. Such genes will then be subjected to further study.\r\nDate GSE4034 Last Updated: Jan 13 2006\r\n\r\nContributors:  Abraham A Palmer\r\n\r\nIncludes GDS1901.\r\n Update date: Jun 09 2006.\r\n Dataset description GDS1901: Analysis of the amygdalae and hippocampi of strains exhibiting low or high freezing behavior in response to fear.  Previous work identified alleles that contribute to the variability in freezing behavior, and the chromosomal location of these alleles.\r</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Succinate semialdehyde dehydrogenase (SSADH) deficiency is a rare autosomal recessive disorder effecting approximately 350 people around the world. Patients suffering from SSADH deficiency experience language acquisition failure, memory deficiencies, autism, increased aggressive behaviors, and seizures. There is a chemical buildup of both gamma-aminobutyric acid (GABA) and gamma-hydroxybutyric acid (GHB) in the neurological system of these patients. The Aldh5a1-/- knock out mouse model of SSADH deficiency shows the same chemical imbalances as the human disease, with additional fatal tonic-clonic seizures at three weeks of age.  The elucidation of seizure causing pathways will facilitate treatment of seizure phenotypes in diseases with related epilepsy. Gene expression patterns within the hippocampus, cerebellum, and cortex of SSADH deficient mice (Aldh5a1-/- mice) will be compared to wild type mice at a time point immediately prior to fatal seizures. We hypothesis that the SSADH deficient mice experience a dysfunction of glutamate/GABA/ glutamine neurotransmitter cycle linked to astroglial metabolism and/or uptake of neuronally-released glutamate. The increased levels of GHB and GABA lead to down regulation of GABA-B-Receptor leading to seizures. The SSADH deficient phenotype may also be caused by ongoing oxidative damage and the pathological role of succinic semialdehyde. SSADH deficient mice (Aldh5a1-/- knock out) exhibit fatal seizures around three weeks of age. Mutant and wild type mice will be sacrificed between 17 and 19 days of life, and brain sections will be extracted and frozen (using a standard protocol). Hippocampus, cerebellum, and cortex from three mutant mice and three wild type mice will individually be expression profiled on the Affymetrix platform, giving a total of eighteen arrays. Comparative analysis will then be carried out, evaluating the transcript differences between mutant and wild type mice in each brain region.\nDate GSE2866 Last Updated: Jun 08 2006\nContributors:  E A Donarum\nIncludes GDS1745.\n Update date: Jun 08 2006.\n Dataset description GDS1745: Analysis of brain hippocampi, cerebella, and cortices of succinate semialdehyde dehydrogenase (SSADH)-deficient mutants at 3 weeks of age, when fatal seizures occur. Results indicate that SSADH deficiency results in the dysregulation of genes involved in myelin structure and compaction.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Acute cognitive impairment (i.e., delirium) is common in elderly emergency department patients and frequently results from infections that are unrelated to the central nervous system. Since activation of the peripheral innate immune system induces brain microglia to produce inflammatory cytokines that are responsible for behavioral deficits, we investigated if aging exacerbated neuroinflammation and sickness behavior after peripheral injection of lipopolysaccharide (LPS). Microarray analysis revealed a transcriptional profile indicating the presence of primed or activated microglia and increased inflammation in the aged brain. Furthermore, aged mice had a unique gene expression profile in the brain after an intraperitoneal injection of LPS, and the LPS-induced elevation in the brain inflammatory cytokines and oxidative stress was both exaggerated and prolonged compared with adults. Aged mice were anorectic longer and lost more weight than adults after peripheral LPS administration. Moreover, reductions in both locomotor and social behavior remained 24 h later in aged mice, when adults had fully recovered, and the exaggerated neuroinflammatory response in aged mice was not reliably paralleled by increased circulating cytokines in the periphery. Taken together, these data establish that activation of the peripheral innate immune system leads to exacerbated neuroinflammation in the aged as compared with adult mice. This dysregulated link between the peripheral and central innate immune system is likely to be involved in the severe behavioral deficits that frequently occur in older adults with systemic infections.\nDate GSE3253 Last Updated: Oct 28 2005\nContributors:  Johnathan P Godbout Jing Chen Amy F Richwine Brian M Berg Keith K Kelley Rodney W Johnson Jayne Abraham\nIncludes GDS1311.\n Update date: Nov 04 2005.\n Dataset description GDS1311: Analysis of aged brain after activation of the peripheral innate immune system by intraperitoneal injection of E. coli lipopolysaccharide (LPS).  Results provide insight into molecular events underlying severe behavioral deficits (delirium) common in older adults with systemic infections.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Post-transcriptional mechanisms play an important role in the control of gene expression. RNA-binding proteins are key players in the post-transcriptional control of many neural genes and they participate in multiple processes, from RNA splicing and mRNA transport to mRNA stability and translation. Our laboratory has developed the first mouse model overexpressing a RNA-binding protein, the ELAV-like protein HuD, in the CNS under the control of the CaMKinII alpha promoter. Initial behavioral characterization of the mice revealed that they had significant learning deficits together with abnormalities in prepulse inhibition (PPI). At the molecular level, we found that the expression of the growth-associated protein GAP-43, one of the targets of HuD, was increased in the hippocampus of HuD transgenic mice. To characterize these mice further and to evaluate the utility of these animals in understanding human diseases, we propose to use DNA microarray methods. To test our hypothesis we propose 2 specific aims: 1)To characterize the pattern of gene expression in the hippocampus of HuD overexpressor mice 2)To compare the pattern of gene expression in our mouse model with that in the hippocampus of rats prenatally exposed to alcohol (FAS model) and in post-mortem tissues of patients with schizophrenia Based on the behavioral and molecular properties of our HuD transgenic mice we hypothesize that these animals may be good models for the studying the basis of learning disabilities and of diseases that show deficits in PPI such as fetal alcohol syndrome and schizophrenia. All mice are in C57BL/6 background and are male approximately 60 days old. Initial studies were performed in animals that were not subjected to any experimental manipulation. Animals were bred and sacrificed according to our approved animal protocol.  The brain was rapidly dissected on ice and we isolated the hippocampus, which has the highest expression of the transgene. After dissection both hippocampi were frozen in liquid nitrogen, pooled and stored at -80C until analysis. RNA samples were isolated using RNAeasy Qiagen columns. For our first experiment, we want to examine the pattern of gene expression in the hippocampus of 3 transgenic mice and 3 non-transgenic littermates. RNAs from the 6 hippocampi were of high quality as revealed by the integrity of the 28S and 18S rRNA. We will provide 6 samples containing 10 ug of RNA each in DEPC water at a concentration of about 0.5 ug/ul. Three of the samples (#1, #2 and # 3) are from transgenic mice and three from their non-transgenic littermates (#4, #5 and #6).\nDate GSE2005 Last Updated: May 29 2005\nContributors:  Nora I Perrone-Bizzozero\nIncludes GDS1111.\n Update date: May 09 2005.\n Dataset description GDS1111: Analysis of hippocampus of 60 day old C57BL/6 males overexpressing the RNA binding protein HuD. HuD overexpressor mice exhibit learning deficits and abnormalities in prepulse inhibition.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: SUMMARY  Medulloblastoma is the most common malignant brain tumor in children. It is thought to result from transformation of granule cell precursors (GCPs) in the developing cerebellum, but little is known about the early stages of the disease. Here we identify a pre-neoplastic stage of medulloblastoma in patched heterozygous mice, a model of the human disease. We show that pre-neoplastic cells are present in the majority of patched mutants, although only 16% of these mice develop tumors. Pre-neoplastic cells, like tumor cells, exhibit activation of the Sonic hedgehog pathway and constitutive proliferation. Importantly, they also lack expression of the wild-type patched allele, suggesting that loss of patched is an early event in tumorigenesis. While pre-neoplastic cells resemble GCPs and tumor cells in many respects, they have a distinct molecular signature. Genes that mark the pre-neoplastic stage include regulators of migration, apoptosis and differentiation, processes critical for normal development but previously unrecognized for their role in medulloblastoma. The identification and molecular characterization of pre-neoplastic cells provides insight into the early steps in medulloblastoma formation, and may yield important markers for early detection and therapy of this disease.  ANIMALS  Patched heterozygous mice (Goodrich et al., 1997) were obtained from Matthew Scott?s lab at Stanford, and maintained by breeding with 129X1/SvJ mice from Jackson Laboratories. Math1-GFP transgenic mice (Lumpkin et al., 2003) were provided by Jane Johnson at UT Southwestern Medical Center. Math1-GFP/patched+/- mice were generated by crossing patched heterozygotes with Math1-GFP mice, and then backcrossing to Math1-GFP mice three times before further analysis. All mice were maintained in the Cancer Center Isolation Facility at Duke University Medical Center.  ISOLATION OF CELLS  Granule cell precursors (GCPs) were isolated from 7-day-old (P7) patched+/- mice; preneoplastic cells were obtained from 6 week-old patched mutants; and tumor cells were obtained from 10-25-week old patched mutants displaying physical and behavioral signs of medulloblastoma. Cells were isolated from each source using a protocol described in (Wechsler- Reya and Scott, 1999). Briefly, cerebella were digested in solution containing 10 U/ml papain Oliver et al. Pre-Neoplastic Stage of Medulloblastoma - 8 - (Worthington, Lakewood, NJ) and 250 U/ml DNase (Sigma) and triturated to obtain a cell suspension. This suspension was centrifuged through a step gradient of 35% and 65% Percoll (Amersham Biosciences, Piscataway, NJ), and cells were harvested from the 35%-65% interface. Cells were resuspended in serum-free culture medium consisting of Neurobasal containing B27 supplement, sodium pyruvate, L-glutamine, and penicillin/streptomycin (all from Invitrogen, Carlsbad, California) and counted on a hemacytometer. Cells used for RNA isolation were centrifuged and flash frozen in liquid nitrogen. For proliferation assays or immunostaining, cells were plated on poly-D-lysine (PDL)-coated tissue culture vessels and incubated in serum-free culture medium.  MICROARRAY HYBRIDIZATION AND ANALYSIS  RNA from GCPs, pre-neoplastic cells and tumor cells (isolated as described above, but not FACS-sorted) and from normal adult cerebellum (not dissociated) was converted to cDNA using the Superscript Choice cDNA kit (Invitrogen) and a T7-dT(24) primer (Genset/Proligo, Boulder, CO). cRNA was generated using a T7-transcription/labeling kit from Enzo Life Sciences and hybridized to Affymetrix U74Av2 chips (Affymetrix, Santa Clara, CA). Chips were scanned, and hybridization data were acquired using Affymetrix Suite 5.0 software. Affymetrix CEL files were normalized and quantified using Bioconductor software with the gcRMA model to quantify gene expression levels (Gentleman and Carey, 2002). Unsupervised principal components analysis (PCA) was used to identify the relationships among normal adult cerebellum, GCPs, pre-neoplastic cells and tumor cells based on expression profiles. To identify genes that were differentially expressed among GCPs, pre-neoplastic cells and tumor cells, supervised analysis was carried out. A gene-by-gene ANOVA model with three groups (GCP, pre-neoplastic, tumor) was used to fit the log2-transformed intensities. To correct for multiple comparisons, the nominal p-value was adjusted using the false discovery rate (Benjamini and Hochberg, 1995). Genes were considered to be differentially expressed if they satisfied all of the following criteria: a difference in expression greater than 1.9-fold between any two groups, a maximum absolute intensity difference larger than 32 units, and an adjusted pvalue &lt; 0.01. There were 118 genes that met these criteria. The identities of differentially expressed genes were verified by integrating data from the Affymetrix and Unigene databases. Gene functions were determined using information from Gene Ontology, Unigene, LocusLink and PubMed databases. Clustering was performed with Cluster and Treeview (Eisen et al., 1998). All statistical analysis was performed using R-1.7 software (Dalgaard, 2002). Results were visualized with Spotfire 6.0 (Somerville, MA).  CORRESPONDING AUTHOR  Robert J. Wechsler-Reya* Dept. of Pharmacology &amp; Cancer Biology, Duke University Medical Center Box 3813, Durham, NC 27710 Phone: 919-613-8754. Fax: 919-668-3556. Email: rw.reya@duke.edu.\nDate GSE2426 Last Updated: Oct 28 2005\nContributors:  R J Wechsler-Reya Trang T Huynh Simon M Lin Jonathan F Wells Tracy A Read Anriada Mehmeti Jessica D Kessler Robert J Wechsler-Reya Trudy G Oliver\nIncludes GDS1110.\n Update date: May 09 2005.\n Dataset description GDS1110: Analysis of granule cell precursors, pre-neoplastic cells, and tumor cells obtained from 7 day old, 6 week old, and 10 to 25 week old patched heterozygous animals, respectively. Results provide insight into the mechanisms of pathogenesis of medulloblastoma at the early stage.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         The Dlx homeodomain transcription factors are implicated in regulating the function of inhibitory neurons; therefore understanding their functions will provide insights into disorders such as epilepsy, mental retardation, autism and cerebral palsy. Identify genes that are dysregulated in the telencephalon of Dlx1/2 mutants. I am sending separate samples of the Basal Ganglia (BG) and cortex (Ctx). Comparing gene expression in the embryonic basal ganglia and cortex in wild type and dlx1/2 mutant mice will provide information regarding the types of genes that are downstream of Dlx1/2 function. Perform gene array analyses in duplicate or triplicate; compare expression profile of total RNA from wild type and dlx1/2 telencephalons. I am sending separate samples of the Basal Ganglia (BG) and cortex. I want to perform gene expression comparison between: 1) wild type and dlx1/2 basal ganglia (BG) 2) wild typee and dlx1/2 cortex (ctx) Age of all mice has been indicated as 14 days to satisfy system requirements, however all mice are embryonic day 14.5.\nDate GSE2161 Last Updated: May 29 2005\nContributors:  John L Rubenstein\nIncludes GDS1084.\n Update date: May 08 2005.\n Dataset description GDS1084: Expression profiling of cortex and basal ganglia from embryonic telencephalon of homozygous and heterozygous Dlx1/2 mutants. Mutants examined at embryonic day 14.5. Dlx homeobox transcription factors are implicated in regulating the function of inhibitory neurons.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Traumatic brain injury (TBI) induces a complex cascade of molecular and physiological effects.  This study proposes to investigate the gene expression profile in cortex and hippocampus over early time points, following two different injury severities.  These results will complement prior knowledge of both metabolic and neuroplastic changes after TBI, as well as serve as a starting point to investigate additional gene families whose expression is altered after TBI. To characterize the profile of gene expression following a diffuse traumatic brain injury of varying severity in adult rats. Distinct patterns of gene expression following traumatic brain injury will occur in a time- and injury-dependent fashion.  In particular, changes in expression of enzymes involved in energy metabolism and neuroplasticity will be detected. Adult rats will be subjected to mild and severe lateral fluid percussion injury OR sham surgery without injury.  At various post-injury timepoints (0.5, 4 and 24 hours), animals will be sacrificed, brain regions (parietal cortex and hippocampus, ipsilateral and contralateral to injury) will be dissected and RNA isolated.  RNA will be used to synthesize cRNA probes for microarray hybridization.  RNA from 2 matched animals will be pooled onto a single chip (U34A rat, Affymetrix).  Comparisons will be made between sham and injured animals, with brain region, injury severity, and post-injury time point as the experimental variables.\nDate GSE2871 Last Updated: Jul 06 2005\nContributors:  C C Giza\nIncludes GDS1795.\n Update date: Jun 09 2006.\n Dataset description GDS1795: Analysis of cortices and hippocampi of animals 4 and 24 hours after mild or severe traumatic brain injury (TBI). TBI induced by lateral fluid percussion. Gene expression in tissues ipsi- and contralateral to the injury examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Using targeted gene deletion, we have firmly established that the Class IV POU domain transcription factor Brn-3.2 controls a developmental program regulating axon pathfinding in the mouse visual system. We have isolated and identified downstream gene targets of Brn-3.2, using Representational Difference Analysis (RDA), on cDNA populations derived from wildtype and Brn-3.2-/- retina at the appropriate embryonic stage. One of these candidate genes, the Homeodomain Interacting Protein Kinase 2 (HIPK2), is postulated to be a transcriptional coregulator, based on its in vitro interactions with repressor homeodomain proteins of the NK class, as well as other components of repressor complexes. HIPK2 has also been shown to be involved in post-translational modification of two major transcriptional regulators, p53 and CtBP. Expression of a dominant-negative form of HIPK2 in sensory neurons affects the innervation patterns of their target tissues, suggesting an axon pathfinding defect. The aim of this project is to identify targets for the Homeodomain Interacting Protein Kinase 2 (HIPK2), that we isolated as a downstream gene target of the class IV POU domain transcription factor Brn-3.2, and to investigate their function in the Brn-3.2 dependent pathway regulating axon pathfinding. We hypothesize that the genes regulated by HIPK2 will play a critical role in axon pathfinding and that the results of this study will provide novel insights into the molecular mechanisms of axon pathfinding, and possibly neural plasticity and regeneration, and therefore, be of great interest to the field of neurobiology in general. In order to uncover potential biological role(s) of HIPK2 in neural development, and particularly axon pathfinding, we generated transgenic mouse models for in vivo studies. Our preliminary observations in transgenic mice, indicate that a form of the molecule expected to act as a dominant-negative and designed to be expressed in sensory neurons, affects the innervation patterns of their target tissues, suggesting an axon pathfinding defect. We plan to take advantage of this model to identify genes regulated by HIPK2, and which are likely to be involved in axon pathfinding. To this end, we will compare gene expression profiles in sensory neurons isolated wild-type and transgenic mice. In our experimental paradigm, the trigeminal ganglion represents the ideal sensory structure in which to perform such an analysis: 1) High levels of transgene can be expressed in the trigeminal ganglion, as assessed by the expression of LacZ from the bicistronic construct which contains IRES-LacZ downstream of a dominant negative form of HIPK2.  2) The trigeminal ganglion can be dissected at embryonic day 13.5, when the phenotype is apparent, with ease and free of contamination from surrounding tissues.  3) The amounts of RNA that can be isolated from a single ganglion are in the range of 200-300 ng, which should be sufficient for microarray analysis following linear amplification of RNA. One series will be carried out with transgenic embryos and a control series will be carried out with wildtype littermates. Animals will be prepared and sacrificed by a standard protocol. Tissue will be rapidly dissected from E13.5 trigeminal ganglion, frozen in liquid nitrogen, and stored at -80C until RNA is extracted. RNA will be prepared using RNeasy Micro Kit. We will be providing 4 tissue samples for each of the wildtype and transgenic animals to mitigate any expression differences resulting from mouse to mouse variation.\nDate GSE2869 Last Updated: Jul 06 2005\nContributors:  M Rosenfeld\nIncludes GDS1793.\n Update date: Jun 13 2006.\n Dataset description GDS1793: Analysis of the trigeminal ganglia of transgenics expressing a dominant-negative form of homeodomain interacting protein kinase 2 (HIPK2). HIPK2 is a downstream target of Brn-3.2, a class IV POU domain transcription factor that controls a developmental program regulating axon pathfinding.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 It is well known that many genes are expressed in the retina where they show a clear daily pattern of expression.  Although several studies have investigated gene expression in the retina, no study ? so far - has investigated the daily and the circadian pattern of gene expression using microarray. In the present study we propose to investigate the daily and circadian pattern of expression in the retina. To determine gene expression in the retina in Light:Dark cycles and in constant condition (constant dim light) in the rat. We hypothesize that (a) different genes are expressed in different retinal layers and (b) the pattern of expression of the same genes can differ among the different layers. Wistar rats (3 for each time point) will be killed in the middle of the day (ZT6) and in the middle of the night (ZT18). Eyeballs will be collected and the retina will be removed and immediately frozen and then stored at -80 oC. The same experiment will be repeated on animals that have been maintained in constant dim light for 3. Retinas (3 for each time point) will be obtained from animal killed at CT6 (middle subjective day) and CT 18 (middle subjectice night) and stored at ? 80 oC. The samples will be then shipped to NINDS-NIMH array facility for analysis.\nDate GSE2868 Last Updated: Oct 28 2005\nContributors:  G Tosini\nIncludes GDS1759.\n Update date: Jun 16 2006.\n Dataset description GDS1759: Analysis of retinas under a light-dark cycle at Zeitgeber times (ZT) 6 and 18 or in constant dim light at circadian times (CT) 6 and 18. Gene expression in retinas compared to that in pineal glands.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        A number of human neurodegenerative diseases result from the expansion of a glutamine repeat within the disease-causing protein. Spinocerebellar ataxia type1 is one such disease, caused by expansion of a polyglutamine tract in the novel protein ataxin-1. To faithfully model SCA1 in the mouse, we generated  knock-in mice carrying 154 CAG repeats in the mouse Sca1 locus. These mice reproduced many aspects of the human disease. Despite ubiquitous expression of the mutant protein, they developed slowly progressive selective neurodegeneration , which is most distinct in Purkinje cells and spinal cord. Alterations in gene expression have been proposed to be involved in the pathogenesis of polyglutamine disease including SCA1, but the changes of gene expression in an authentic disease model have not been characterized. We believe that knowledge of these genes will give insight into the pathophysiology of SCA1 and may ultimately be relevant to the treatment of SCA1. We will determine gene expression patterns in the cerebellum and forebrain at three different time points. We propose that the genes whose expression is affected  by mutant ataxin-1 expression are effectors for neuronal dysfunction and neuronal degeneration in the knock-in mice. We also hypothesize the difference in the expression levels of these genes accounts for selective vulnerability of neurons. We will examine three time points: 4 weeks, 11 weeks, and 20 weeks of age. We have shown that the mice developed motor incoordination as revealed  by rotating rod test as early as 5 weeks of age but it was not until 10 weeks that they start showing clear  neurological  phenotype such as clasping.  At each point, we will compare the pattern between the mutant and wild-type littermate. Animals will be prepared and sacrificed by a standard procedure. Cerebellum and spinal cord are the most affected areas whereas forebrain shows less neurodegeneration. Tissue will be rapidly dissected from cerebellum and forebrain, frozen in liquid nitrogen, and stored at ?80 C until analysis. Tissues will be sent to the centers (as opposed to RNA). We will be providing 9 tissue samples from three litters for each time point to mitigate any expression differences resulting from subtle differences of procedure or environment where the mice grow.\nDate GSE2867 Last Updated: Jul 06 2005\nContributors:  H Zoghbi\nIncludes GDS1756.\n Update date: May 02 2006.\n Dataset description GDS1756: Analysis of cerebellum and forebrain tissue of knock-in mice carrying 154 CAG repeats in the spinocerebellar ataxia type 1 (SCA1) locus at 4 and 12 weeks of age. Results provide insight into the pathophysiology of the neurodegenerative disease SCA1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             The neurodegenerative process in HIV encephalitis (HIVE) is associated with extensive damage to the dendritic and synaptic structure that often leads to cognitive impairment. Several mechanisms might be at play, including release of neurotoxins, oxidative stress and decreased activity of neurotrophic factors. Furthermore, HIV-mediated dysregulation of genes involved in neuronal maintenance might play an important role. For this purpose, cRNA was prepared from the brains of 17 AIDS patients for analysis with the Affymetrix Human U95Av2 GeneChip and analyzed with the GeneSpring Expression Analysis Software. Out of 12,625 genes analyzed, 74 were downregulated and 59 were upregulated compared to controls. Initial alternative analysis of RNA was performed by ribonuclease protection assay (RPA). In cases with HIVE, downregulated genes included neuronal molecules involved in synaptic plasticity and transmission (ion channels, synaptogyrin, synapsin II), cell cycle (p35, p39, CDC-L2, CDC42, PAK1) and signaling molecules (PI3K, Ras-Raf-MEK1), transcription factors and cytoskeletal components (MAP-1B, MAP-2, tubulin, adducin-2). Upregulated genes included those involved in neuroimmune (IgG, MHC, ?2microglobulin) and anti-viral responses (interferon-inducible molecules), transcription (STAT1, OLIG2, Pax-6) and signaling modulation (MEK3, EphB1) of the cytoskeleton (myosin, aduccin-3, radixin, dystrobrevin). Taken together, this study suggests that HIV proteins released from infected macrophages might not only induce a neuroinflammatory response, but also may promote neurodegeneration by interfering with neuronal transcription of genes involved in regulating signaling and cytoskeletal molecules important in maintaining synapto-dendritic functioning and integrity.\nDate GSE3489 Last Updated: Mar 07 2006\nContributors:  Eleanor S Roberts Dianne Langford Anthony Adame Edward Rockenstein Leslie Crews Howard S Fox Eliezer Masiliah Ian Everall\nIncludes GDS1726.\n Update date: May 30 2006.\n Dataset description GDS1726: Analysis of brain frontal cortex of HIV-seropositive patients with HIV encephalitis (HIVE). HIVE affects &gt;40% of AIDS patients, promoting neurodegeneration and cognitive impairment. Results suggest HIV-mediated dysregulation of genes involved in neuronal maintenance might play an important role.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Study the effect of fetal alcohol exposure (FAE) on hippocampal development Compare the pattern of gene expression in the hippocampus of FAE and control rats fed either an isocaloric diet or a normal diet, at post-natal day 5 of development. FAE will delay the maturation of the hippocampus Rats were fed one of three diet, a liquid diet with 5% ethanol (FAE group), an isocaloric liquid diet (Isocalorc group) or nomal lab chao (normal group).\nDate GSE1997 Last Updated: Feb 02 2006\nContributors:  Nora I Perrone-Bizzozero\nIncludes GDS1660.\n Update date: Jun 08 2006.\n Dataset description GDS1660: Analysis of hippocampi of 5-day-old animals subjected to fetal alcohol exposure (FAE) with 5% ETOH. FAE affects the development and function of the hippocampus.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     The cardinal clinical features of Parkinson's disease (PD) (rigidity, rest tremor, bradykinesia, and postural instability) result from selective loss of midbrain dopaminergic neurons.  More specifically, dopaminergic neurons in the substantia nigra pars compacta (SNc) are much more susceptible to damage than the adjacent dopaminergic neurons in the ventral tegmental area (VTA).  This dichotomy is not only seen in human Parkinsons disease, but also in many animal models of PD, including administration of the mitochondrial toxin rotenone to rats, which replicates many of the behavioral and neuropathological features of PD.  The factors underlying this selective vulnerability are unknown, but could be related to differences in neuronal circuitry, differences in glial support, or intrinsic differences between the neuronal populations of the two regions.  Elucidation of these factors may lead to a greater understanding of the pathogenesis and treatment of Parkinson's disease. We will determine gene expression profiles of untreated rat SNc and VTA dopaminergic neurons using laser capture microscopy to obtain region-specific neuronal mRNA. There are intrinsic differences in gene expression between dopaminergic neurons in the rat SNc and VTA that result in greater susceptibility of SNc neurons to degeneration in experimental parkinsonism.  These differences may be related to dopamine metabolism, oxidative metabolism and stress, protein aggregation, or other unforseen pathways. We will compare gene expression profiles between SNc and VTA dopaminergic neurons in normal rats.  No treatment or time points will be studied in this experiment.  Animals will be anesthetized, sacrificed by decapitation, and brains frozen on dry ice.  Frozen sections will be collected onto glass microscope slides and rapidly immunostained for tyrosine hydroxylase to identify dopaminergic neurons.  SNc and VTA neurons (approx. 200 per sample) will be isolated using laser capture microscopy.  Total RNA will be extracted and poly-A RNA will be amplified using a modified Eberwine protocol.  aRNA will be sent to the centers for labeling and hybridization to Affymetrix rat U34A arrays.  We have confirmed with the center that our aRNA protocol is compatible with the centers amplification protocols; in fact, it is essentially identical.  We will be providing a two-round amplification product to the center for labeling and hybridization.  We recognize that using RNA after three rounds of amplification  may decrease sensitivity for low copy number transcripts, but favor this approach versus pooling our samples (which are inherently paired) at this point.  We have discussed this point in detail with the center. SNc and VTA samples from eight animals (16 samples total) will be provided to mitigate differences specific to individual animals.  With the assisatnce of the center, paired t-tests will be used to determine differential expression between the two regions.  Permutational t-test analysis and/or Benjamini and Hochberg analysis of expression ratios will be used to protect against multiple comparisons.  Selected differentially expressed genes will be validated on separate tissue samples using quantitative RT-PCR or in situ hybridization.\nDate GSE1837 Last Updated: Feb 02 2006\nContributors:  James G Greene\nIncludes GDS1641.\n Update date: Jun 08 2006.\n Dataset description GDS1641: Analysis of dopaminergic neurons of the substantia nigra pars compacta (SNc) and the ventral tegmental area (VTA) from normal animals. SNc dopaminergic neurons are more susceptible to degeneration in Parkinson's disease than VTA dopaminergic neurons.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Expression profiles of different mouse tissue samples profiles.\nDate GSE2178 Last Updated: May 29 2005\nContributors:  Michael Mucenski Bruce J Aronow Belinda Peach Mitchell Cohen Amy Moseley Susan E Waltz John Maggio Sandra Degen Steve Potter Thomas Doetschman Chris Erwin Joanna A Groden James Lessard Linda Parysek R Hirsch MaryBeth Genter Jianhua Zhang Kathleen Anderson Jonathan D Katz John Dedman Michael Lehman Jorge A Bezerra Michael D Bates Ming Xu Dan Wiginton Jeff A Molkentin\nIncludes GDS1322.\n Update date: Nov 09 2005.\n Dataset description GDS1322: Gene expression profiles across a wide variety of tissue samples. The tissues examined include nervous system, skeletal muscle, male and female reproductive organs, gastrointestinal tract, as well as organs in a developmental context including lung, liver, kidney, and heart.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     We have combined large-scale mRNA expression and gene mapping methods to identify genes and loci that control hematopoietic stem cell (HSC) functioning. mRNA expression levels were measured in purified HSC isolated from a panel of densely genotyped recombinant inbred mouse strains. Quantitative trait loci (QTLs) associated with variation in expression of thousands of transcripts were mapped. Comparison of the physical transcript position with the location of the controlling QTL identified polymorphic cis-acting stem cell genes. In addition, multiple trans-acting control loci were highlighted that modify expression of large numbers of genes. These groups of co-regulated transcripts identify pathways that specify variation in stem cells. We illustrate this concept with the identification of strong candidate genes involved with HSC turnover.  We compared expression QTLs in HSC and brain from the same animals, and document both shared and tissue-specific QTLs. Our data are accessible through WebQTL, a web-based interface that allows custom genetic linkage analysis and identification of co-regulated transcripts.\nDate GSE2031 Last Updated: Jun 29 2005\nContributors:  Bert Dontje Kenneth F Manly Sue Sutton Michael Cooke Rudi Alberts Gerald de Haan Jintao Wang Edo Vellenga Ellen Weersing Elissa Chesler Leonid Bystrykh Andrew I Su Tim Wiltshire Mathew T Pletcher Robert W Williams Ritsert C Jansen Lu Lu\nIncludes GDS1077.\n Update date: Mar 16 2005.\n Dataset description GDS1077: Expression profiling of Lin- Sca-1+ c-kit+ hematopoietic stem cells (HSC) from 22 different BXD recombinant inbred (RI) strains. Each RI strain is homozygous for alleles at about 98% of loci. Results combined with QTL mapping to identify candidate genes for the control of HSC function.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Transcriptome analysis of Ts1Cje (mouse model of Down syndrome) and euploids murine cerebellum during postnatal development.\nDate GSE1611 Last Updated: Mar 27 2006\nContributors:  Randal Moldrich Isabelle Rivals Pierre-Marie Sinet Stylianos E Antonarakis Robert Lyle Geoffroy Golfier Luce Dauphinot Laurence Ettwiller Kiyoko Toyama Charles J Epstein Léon Personnaz Maire-Claude Potier Minh Tran Dang Jean Rossier\nIncludes GDS994.\n Update date: Jan 11 2005.\n Dataset description GDS994: Expression profiling of brain cerebella from Ts1Cje males at postnatal days 0, 15, and 30.  RNA pooled from 3 males at each developmental stage. Results provide insight into the molecular changes contributing to the pathogenesis of Down syndrome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.Troubled experiment.Accession experiment.Database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE              GSE2018                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE              GSE2872                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE              GSE4523                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE              GSE4036                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE              GSE4034                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE              GSE2866                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE              GSE3253                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE              GSE2005                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE              GSE2426                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE              GSE2161                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE              GSE2871                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE              GSE2869                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE              GSE2868                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE              GSE2867                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE              GSE3489                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE              GSE1997                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE              GSE1837                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE              GSE2178                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE              GSE2031                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE              GSE1611                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.SampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: http://www.ncbi.nlm.nih.gov/geo/                     34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: http://www.ncbi.nlm.nih.gov/geo/                     12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: http://www.ncbi.nlm.nih.gov/geo/                      6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: http://www.ncbi.nlm.nih.gov/geo/                     28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: http://www.ncbi.nlm.nih.gov/geo/                     18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: http://www.ncbi.nlm.nih.gov/geo/                     12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: http://www.ncbi.nlm.nih.gov/geo/                      6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: http://www.ncbi.nlm.nih.gov/geo/                     14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: http://www.ncbi.nlm.nih.gov/geo/                     47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13: http://www.ncbi.nlm.nih.gov/geo/                     16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14: http://www.ncbi.nlm.nih.gov/geo/                     20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: http://www.ncbi.nlm.nih.gov/geo/                     28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: http://www.ncbi.nlm.nih.gov/geo/                      5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: http://www.ncbi.nlm.nih.gov/geo/                     15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: http://www.ncbi.nlm.nih.gov/geo/                    161</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: http://www.ncbi.nlm.nih.gov/geo/                     44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: http://www.ncbi.nlm.nih.gov/geo/                     12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.LastUpdated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    2023-09-25 18:33:39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    2023-09-25 18:37:44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    2023-10-03 19:39:05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    2023-09-25 17:16:22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    2023-09-25 17:24:46</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    2023-09-25 17:26:29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    2023-10-03 19:39:06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    2023-09-18 20:30:42</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    2023-09-18 20:32:07</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    2023-09-18 20:53:05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    2023-09-18 20:53:57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    2023-09-18 21:11:02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    2023-09-18 21:12:09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    2023-09-18 21:15:13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    2023-10-03 19:39:07</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    2023-09-18 21:16:34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    2022-08-16 16:43:25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    2023-10-03 19:39:08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    2022-08-30 17:24:27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    2022-08-12 14:03:02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: This data set may have a batch artifact (PC 2), p=0.00049622</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: This data set may have a batch artifact (PC 2), p=0.00064920</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: This data set may have a batch artifact (PC 2), p=9.1909e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: This data set may have a batch artifact (PC 3), p=0.00011938</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                TRUE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                TRUE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.Name  taxon.Scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:   0.9976341      0.8750      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:   0.8552303      0.5000        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:   0.2739433      0.5875      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:   0.9971246      0.7500      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:   0.5703423      1.0000      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:   0.4273933      0.6875      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:   0.2661131      0.6250      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:   0.8565510      0.6500      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:   0.8558619      0.6875      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:   0.8557322      0.8375      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:   0.5688754      0.6875        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:   0.8558936      0.5875      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:   0.7119963      0.5625        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:   0.8469065      0.6250      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:   0.4098798      0.5000      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:   0.7101798      0.1875        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:   0.9979230      0.3125        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:  -0.2259327      0.0625      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:   0.1386041      0.8125      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:   0.8565754      0.6875      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.Database.Name taxon.Database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                mm10                81</span>
<span class="r-in"><span><span class="fu">get_datasets</span><span class="op">(</span>taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mouse"</span>, <span class="st">"human"</span><span class="op">)</span>, uris <span class="op">=</span> <span class="st">"http://purl.obolibrary.org/obo/UBERON_0002048"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ShortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:              GSE2018</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:              GSE2178</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               GSE476</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               GSE478</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               GSE484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               GSE485</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:              GSE1037</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            GSE1133.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               GSE479</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                GSE97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               GSE446</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:   bhattacharjee-lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:     ramaswamy-cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:       staunton-nci60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:              GSE2361</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               GSE740</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                GSE96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:              GSE4345</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:              GSE6858</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:              GSE4772</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                      experiment.Name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                      Human Lung Transplant - BAL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                    Mouse Expression DB 2001 CHMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                         Ozone effect on airways hyperpermability</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                              Alveoli loss during caloric restriction time course</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                      Alveoli septation inhibition and protection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                               Genetic basis of sensitivity to pulmonary fibrosis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                      Lung cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                        tissue-specific pattern of mRNA expression - Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                               Alveolar septation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                  Large-scale analysis of the mouse transcriptome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                          Lungs from ovalbumen sensitized and challenged C3H mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                               bhattacharjee-lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                 ramaswamy-cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                   staunton-nci60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                      Expression Proflies of Human Normal tissues</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                Rosetta_Merck_Splicing_Experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                  Large-scale analysis of the human transcriptome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:   In Vivo Regulation of Human Skeletal Muscle Gene Expression by Thyroid Hormone</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                  Expression data from experimental murine asthma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: Expression data from pre natal and post natal CCAM samples and relative controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:            24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:            55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:            56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:            57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:            58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:           141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:           166</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:           171</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:           181</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:           197</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:           204</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:           219</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:           225</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:           305</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:           306</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:           398</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:           453</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:           502</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:           532</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          experiment.Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Bronchoalveolar lavage samples collected from lung transplant recipients.  Numeric portion of sample name is an arbitrary patient ID and AxBx number indicates the perivascular (A) and bronchiolar (B) scores from biopsies collected on the same day as the BAL fluid was collected.  Several patients have more than one sample in this series and can be determined by patient number followed by a lower case letter.  Acute rejection state is determined by the combined A and B score - specifically, a combined AB score of 2 or greater is considered an acute rejection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Expression profiles of different mouse tissue samples profiles.\nDate GSE2178 Last Updated: May 29 2005\nContributors:  Michael Mucenski Bruce J Aronow Belinda Peach Mitchell Cohen Amy Moseley Susan E Waltz John Maggio Sandra Degen Steve Potter Thomas Doetschman Chris Erwin Joanna A Groden James Lessard Linda Parysek R Hirsch MaryBeth Genter Jianhua Zhang Kathleen Anderson Jonathan D Katz John Dedman Michael Lehman Jorge A Bezerra Michael D Bates Ming Xu Dan Wiginton Jeff A Molkentin\nIncludes GDS1322.\n Update date: Nov 09 2005.\n Dataset description GDS1322: Gene expression profiles across a wide variety of tissue samples. The tissues examined include nervous system, skeletal muscle, male and female reproductive organs, gastrointestinal tract, as well as organs in a developmental context including lung, liver, kidney, and heart.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Ozone is a very common environmental pollutant and has been associated with the exacerbation of cardiopulmonary diseases like asthma. In this study the molecular mechanisms underlying the effect of ozone on airways hyperpermability were investigated. Two strains of mice, HeJ (Tlr4 mutant) and OuJ (wildtype) were exposed continuously to air or 0.3ppm of ozone. Lungs were removed and RNA was collected to generate expression profiles.\nDate GSE476 Last Updated: Jun 15 2005\nContributors:  Stephan Kleeberger\nIncludes GDS239.\n Update date: Mar 22 2004.\n Dataset description GDS239: Ozone is an environmental pollutant associated with  exacerbation of cardiopulmonary diseases like asthma. HeJ (Tlr4 mutant) and OuJ (wild type) exposed continuously to air or 0.3 ppm of ozone, and lungs  examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Pulmonary alveoli are complex architectural units thought to undergo endogenous or pharmacologically induced programs of regeneration and degeneration. To study the molecular mechanism of alveoli loss mice were calorie restricted at different timepoints. Lungs were harvested and processed for RNA extraction.\nDate GSE478 Last Updated: Jul 14 2006\nContributors:  Donald   Massaro Gloria D Massaro\nIncludes GDS241.\n Update date: Jun 27 2003.\n Dataset description GDS241: Temporal study of lung alveoli destruction induced by caloric restriction. Alveoli number and surface area are linked to oxygen consumption, and calorie restriction lowers oxygen consumption.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     It has been shown that dexamethasone (Dex) impairs the normal lung septation that occurs in the early postnatal period. Treatment with retinoic acid (ATRA) abrogates the effects of Dex. To understand the molecular basis for the Dex indiced inhibition of the formation of the alveoli and the ability of ATRA to prevent the inhibition of septation, gene expression was analyzed in 4-day old mice treated with diluent (control), Dex-treated and ATRA+Dex-treated.\nDate GSE484 Last Updated: Aug 24 2005\nContributors:  Linda Clerch\nIncludes GDS250.\n Update date: Jun 27 2003.\n Dataset description GDS250: Analysis of how dexamethasone (DEX) inhibits normal lung septation and alveoli formation, and mechanism by which retinoic acid (ATRA) protects against DEX. Three day old mice treated with ATRA or control. On day 4, half of each group administered DEX.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          The murine pulmonary response is strain specific.  Susceptible strains such as C57BL6/J experience an inflammatory response followed by progressive lung disease.  Resistant strains such as Balb/c do not experience significant degrees of inflammation or fibrosis.  This study aims to determine the genetic basis of sensitivity differences between strains.\nDate GSE485 Last Updated: Jul 27 2006\nContributors:  David   Moller Y W Chen\nIncludes GDS251.\n Update date: Jun 27 2003.\n Dataset description GDS251: Determination of genetic basis of sensitivity to pulmonary fibrosis induced by bleomycin by comparing susceptible (C57BL6/J) and resistant (Balb/c) strains. C57BL6/J experience inflammatory response then progressive lung disease, Balb/c do not.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: Two prognostically significant subtypes of high-grade lung neuroendocrine tumors independent of small-cell and large-cell neuroendocrine carcinomas identified by gene expression profiles.  BACKGROUND: Classification of high-grade neuroendocrine tumors (HGNT) of the lung currently recognises large-cell neuroendocrine carcinoma (LCNEC) and small-cell lung carcinoma (SCLC) as distinct groups. However, a similarity in histology for these two carcinomas and uncertain clinical course have led to suggestions that a single HGNT classification would be more appropriate. Gene expression profiling, which can reproduce histopathological classification, and often defines new subclasses with prognostic significance, can be used to resolve HGNT classification. METHODS: We used cDNA microarrays with 40386 elements to analyze the gene expression profiles of 38 surgically resected samples of lung neuroendocrine tumors and 11 SCLC cell lines. Samples of large-cell carcinoma, adenocarcinoma, and normal lung were also included to give a total of 105 samples analyzed. The data were subjected to filtering to yield informative genes before unsupervised hierarchical clustering that identified relatedness of tumor samples. FINDINGS: Distinct groups for carcinoids, large-cell carcinoma, adenocarcinoma, and normal lung were readily identified. However, we were unable to distinguish LCNEC from SCLC by gene expression profiling. Three independent rounds of unsupervised hierarchical clustering consistently divided SCLC samples into two main groups with LCNEC samples largely integrated with these groups. Furthermore, patients in one of the groups identified by clustering had a significantly better clinical outcome than the other (83% vs 12% survived for 5 years; p=0.0094. None of the highly proliferative SCLC cell lines subsequently analyzed clustered with this good-prognosis group. INTERPRETATION: Our findings show that HGNT of the lung can be classified into two groups independent of SCLC and LCNEC. To this end, we have identified many genes, some of which encode well-characterized markers of cancer that distinguish the HGNT groups. These results have implications for the diagnosis, classification, and treatment of lung neuroendocrine tumors, and provide important insights into their underlying biology.\nDate GSE1037 Last Updated: May 29 2005\nContributors:  Virtanen Carl Ishikawa Yuichi Jones Michael Yuichi Ishikawa Daisuke Ishikawa Carl Virtanen Michael H Jones\nIncludes GDS619.\n Update date: Jul 12 2004.\n Dataset description GDS619: Molecular classification of lung high-grade neuroendocrine tumor (HGNT) groups. Carcinoids, large-cell carcinoma, adenocarcinoma, small-cell lung carcinoma cell lines, and normal lung examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The tissue-specific pattern of mRNA expression can indicate important clues about gene function. High-density oligonucleotide arrays offer the opportunity to examine patterns of gene expression on a genome scale. Toward this end, we have designed custom arrays that interrogate the expression of the vast majority of protein-encoding human and mouse genes and have used them to profile a panel of 79 human and 61 mouse tissues. The resulting data set provides the expression patterns for thousands of predicted genes, as well as known and poorly characterized genes, from mice and humans. We have explored this data set for global trends in gene expression, evaluated commonly used lines of evidence in gene prediction methodologies, and investigated patterns indicative of chromosomal organization of transcription. We describe hundreds of regions of correlated transcription and show that some are subject to both tissue and parental allele-specific expression, suggesting a link between spatial expression and imprinting.\nDate GSE1133.2 Last Updated: \nContributors:  John B Hogenesch John R Walker Gabriel Kreiman Tim Wiltshire David Block Mimi Hayakawa Hilmar Lapp Richard Soden Michael P Cooke Serge Batalov Keith A Ching Andrew I Su Jie Zhang\nFrom GSE1133\nIncludes GDS592.\n Update date: May 13 2004.\n Dataset description GDS592: Gene atlas of mouse protein-encoding transcriptome. Gene expression profiles from 61 physiologically normal tissues. Samples generated from adult (10-12 week old) C57BL/6 mice (4 male, 3 female) by dissection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The Mg-delta strain of fibrillin-1 deficient mice display an early defect in alveolar septation.  This study attempts to identify expression patterns in the mutant mouse lung compared to their wild-type littermates that suggest pathways that are critical for normal septation.  This experiment focuses on postnatal days 1 and 5.  Other time points will be added at a later date.\nDate GSE479 Last Updated: Jul 27 2006\nContributors:  Enid R Neptune\nIncludes GDS243.\n Update date: Jul 27 2006.\n Dataset description GDS243: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema.\nIncludes GDS244.\n Update date: Jul 27 2006.\n Dataset description GDS244: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema.\nIncludes GDS242.\n Update date: Jul 27 2006.\n Dataset description GDS242: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema. [Switched to use MG-U74A/B/C by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       High-throughput gene expression profiling has become an important tool for investigating transcriptional activity in a variety of biological samples. To date, the vast majority of these experiments have focused on specific biological processes and perturbations. Here, we have generated and analyzed gene expression from a set of samples spanning a broad range of biological conditions. Specifically, we profiled gene expression from 91 human and mouse samples across a diverse array of tissues, organs, and cell lines. Because these samples predominantly come from the normal physiological state in the human and mouse, this dataset represents a preliminary, but substantial, description of the normal mammalian transcriptome. We have used this dataset to illustrate methods of mining these data, and to reveal insights into molecular and physiological gene function, mechanisms of transcriptional regulation, disease etiology, and comparative genomics. Finally, to allow the scientific community to use this resource, we have built a free and publicly accessible website (http://expression.gnf.org) that integrates data visualization and curation of current gene annotations.\nDate GSE97 Last Updated: May 29 2005\nContributors:  M P Cooke J R Walker R G Vega K A Ching L M Sapinoso A I Su T Wiltshire A Patapoutian P G Schultz A P Orth A Moqrich Y Hakak J B Hogenesch G M Hampton\nIncludes GDS182.\n Update date: May 13 2004.\n Dataset description GDS182: Gene expression profiles from a diverse array of tissues, organs, and cell lines, from the normal physiological state. Represents a preliminary description of the normal mammalian transcriptome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Profile of gene expression in lungs of ovalbumen sensitized and challenged C3H mice.\nDate GSE446 Last Updated: May 29 2005\nIncludes GDS347.\n Update date: Mar 22 2004.\n Dataset description GDS347: Lungs of ovalbumin sensitized/challenged C3H mice compared with controls. C3H are ovalbumin resistant. Attempt to identify genes involved in antigen-induced airway hyperresponsiveness.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Classification of human lung carcinomas by mRNA expression profiling reveals distinct adenocarcinoma subclasses.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Multiclass cancer diagnosis using tumor gene expression signatures.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Chemosensitivity prediction by transcriptional profiling.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       We performed expression profiling of 36 types of normal human tissues and identified 2,503 tissue-specific genes. We then systematically studied the expression of these genes in cancers by re-analyzing a large collection of published DNA microarray datasets. Our study shows that integration of each gene's breadth of expression (BOE) in normal tissues is important for biological interpretation of the expression profiles of cancers in terms of tumor differentiation, cell lineage and metastasis. Twenty five total RNA specimens were purchased from Clontech (Palo Alto, CA), Ambion (Austin, TX) and Strategene (La Jolla, CA). We tried to cover as many tissue types as possible by using pooled RNA samples. In order to define breadth-of-expression (BOE) accurately at a reasonable cost, we tried to cover as many tissue types as possible by using pooled RNA samples. Each specimen represents a human organ. We used RNA samples pooled from 2 to 84 donors to avoid differences at the individual level.  Detailed sample information and Affymetrix .CEL files are available at http://www.genome.rcast.u-tokyo.ac.jp/normal/  Publication:Ge X et al., Interpreting expression profiles of cancers by genome-wide survey of breadth of expression in normal tissues.  Genomics. 2005 Aug;86(2):127-141. PMID: 15950434\nLast Updated (by provider): May 11 2006\nContributors:  Xijin Ge SanMing Wang Siego Ihara Hiroyuki Aburatani Yutaka Midorikawa Shogo Yamamoto Shuichi Tsutsumi\nIncludes GDS1096.\n Update date: Apr 08 2005.\n Dataset description GDS1096: Expression profiling of 36 types of normal tissue. Each RNA tissue sample pooled from several donors.  Results identify tissue specific genes and provide baselines for interpreting gene expression in cancer.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                This series represents 52 tissues hybridized across 5 different chip patterns. Probes were placed at every exon-exon junction in each transcript.\nLast Updated (by provider): May 29 2005\nContributors:  Christopher D Armour Eric E Schadt Daniel D Shoemaker Patrick M Loerch Philip Garrett-Engele Jason M Johnson Roland Stoughton Ralph Santos Zhengyan Kan John Castle\nIncludes GDS830.\n Update date: Nov 10 2004.\n Dataset description GDS830: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS829.\n Update date: Nov 10 2004.\n Dataset description GDS829: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions.  mRNA from 52 tissues and cell lines examined.  Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS833.\n Update date: Nov 10 2004.\n Dataset description GDS833: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS831.\n Update date: Nov 10 2004.\n Dataset description GDS831: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS832.\n Update date: Nov 10 2004.\n Dataset description GDS832: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants. [Switched to use Rosetta_Merged by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    High-throughput gene expression profiling has become an important tool for investigating transcriptional activity in a variety of biological samples. To date, the vast majority of these experiments have focused on specific biological processes and perturbations. Here, we have generated and analyzed gene expression from a set of samples spanning a broad range of biological conditions. Specifically, we profiled gene expression from 91 human and mouse samples across a diverse array of tissues, organs, and cell lines. Because these samples predominantly come from the normal physiological state in the human and mouse, this dataset represents a preliminary, but substantial, description of the normal mammalian transcriptome. We have used this dataset to illustrate methods of mining these data, and to reveal insights into molecular and physiological gene function, mechanisms of transcriptional regulation, disease etiology, and comparative genomics. Finally, to allow the scientific community to use this resource, we have built a free and publicly accessible website (http://expression.gnf.org) that integrates data visualization and curation of current gene annotations.\nLast Updated (by provider): May 29 2005\nContributors:  A Moqrich R G Vega P G Schultz M P Cooke A I Su T Wiltshire A P Orth K A Ching L M Sapinoso J B Hogenesch J R Walker Y Hakak A Patapoutian G M Hampton\nIncludes GDS181.\n Update date: May 13 2004.\n Dataset description GDS181: Gene expression profiles from a diverse array of tissues, organs, and cell lines, from the normal physiological state. Represents a preliminary description of the normal mammalian transcriptome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  This SuperSeries is composed of the following subset Series: GSE3461: Gene expression in miscellaneous human tissues and cell lines GSE3462: Triiodothyronine Treatment: Effects on vastus lateralis skeletal muscle  Abstract: Thyroid hormones are key regulators of metabolism that modulate transcription via nuclear receptors. Hyperthyroidism is associated with increased metabolic rate, protein breakdown, and weight loss. Although the molecular actions of thyroid hormones have been studied thoroughly, their pleiotropic effects are mediated by complex changes in expression of an unknown number of target genes. Here, we measured patterns of skeletal muscle gene expression in five healthy men treated for 14 days with 75 µg of triiodothyronine, using 24,000 cDNA element microarrays. To analyze the data, we used a new statistical method that identifies significant changes in expression and estimates the false discovery rate. The 381 up-regulated genes were involved in a wide range of cellular functions including transcriptional control, mRNA maturation, protein turnover, signal transduction, cellular trafficking, and energy metabolism. Only two genes were down-regulated. Most of the genes are novel targets of thyroid hormone. Cluster analysis of triiodothyronine-regulated gene expression among 19 different human tissues or cell lines revealed sets of coregulated genes that serve similar biologic functions. These results define molecular signatures that help to understand the physiology and pathophysiology of thyroid hormone action.\nLast Updated (by provider): Dec 12 2006\n [Switched to use SMHu24k by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Experimental asthma was induced in BALB/c mice by sensitization and challenge with the allergen ovalbumin. Control groups received PBS. To investigate the innate immune component of experimental asthma, we also analyzed recombinase activating gene (RAG) deficient mice following exposure to ovalbumin and control PBS\nLast Updated (by provider): Apr 23 2007\nContributors:  Xin Lu Patricia W Finn Vipul V Jain David L Perkins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               This study aims at giving an insight on gene expression in CCAM.\nLast Updated (by provider): May 05 2006\nContributors:  Agnes Paquet Sam Hawgood Amy Wagner</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.Troubled experiment.Accession experiment.Database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE              GSE2018                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE              GSE2178                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE               GSE476                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE               GSE478                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE               GSE484                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE               GSE485                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE              GSE1037                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE              GSE1133                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE               GSE479                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                GSE97                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE               GSE446                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE              GSE2361                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE               GSE740                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                GSE96                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE              GSE4345                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE              GSE6858                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE              GSE4772                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.SampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: http://www.ncbi.nlm.nih.gov/geo/                     34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: http://www.ncbi.nlm.nih.gov/geo/                    161</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: http://www.ncbi.nlm.nih.gov/geo/                     91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: http://www.ncbi.nlm.nih.gov/geo/                    122</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: http://www.ncbi.nlm.nih.gov/geo/                     90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: http://www.ncbi.nlm.nih.gov/geo/                      5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                             &lt;NA&gt;                    203</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                             &lt;NA&gt;                    280</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                             &lt;NA&gt;                     60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: http://www.ncbi.nlm.nih.gov/geo/                     36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: http://www.ncbi.nlm.nih.gov/geo/                    264</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: http://www.ncbi.nlm.nih.gov/geo/                     85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: http://www.ncbi.nlm.nih.gov/geo/                     16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: http://www.ncbi.nlm.nih.gov/geo/                     14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.LastUpdated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    2023-09-25 18:33:39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    2023-10-03 19:39:08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    2022-08-25 17:22:56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    2022-09-04 16:11:24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    2022-08-12 14:55:30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    2022-08-25 17:24:31</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    2022-08-22 19:49:36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    2023-04-20 07:08:24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    2022-08-16 16:41:01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    2023-09-26 07:33:45</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    2023-05-31 23:03:37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    2023-09-07 21:43:44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    2023-09-07 21:47:00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    2023-09-07 21:49:19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    2021-11-03 18:44:27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    2022-05-12 07:30:04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    2022-09-05 02:52:57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    2023-05-09 20:52:44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    2022-08-17 05:35:24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    2023-09-07 22:32:46</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: This data set may have a batch artifact (PC 1), p=0.00069053</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:  This data set may have a batch artifact (PC 1), p=0.0054886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: This data set may have a batch artifact (PC 2), p=3.1342e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  This data set may have a batch artifact (PC 1), p=0.0024570</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: This data set may have a batch artifact (PC 3), p=0.00022690</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE                  1                1           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                  1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                  1                1           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE                  1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                TRUE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.Name taxon.Scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:  0.99763407      0.8750      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: -0.22593272      0.0625      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:  0.85574622      0.6500      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:  0.42626507      0.8125      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:  0.85510163      0.6500      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:  0.42565509      0.5625      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: -0.08397851      0.1875      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:  0.84599960      0.2500      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:  0.42857143      0.1875      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:  0.55722466      0.3125      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: -0.03621544     -0.3750      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:  0.39301864      0.2500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:  0.23585518      0.0000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:  0.24928653      0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  0.54414795      0.8750      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: -0.14285714     -0.2500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:  0.28369987      0.7500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: -0.13451104      0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:  0.85445543      0.8750      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:  0.28025816      0.3750      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.Database.Name taxon.Database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                hg38                87</span>
<span class="r-in"><span><span class="co"># filter below is equivalent to the call above</span></span></span>
<span class="r-in"><span><span class="fu">get_datasets</span><span class="op">(</span>filter <span class="op">=</span> <span class="st">"taxon.commonName in (mouse,human) and allCharacteristics.valueUri = http://purl.obolibrary.org/obo/UBERON_0002048"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ShortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:              GSE2018</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:              GSE2178</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               GSE476</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               GSE478</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               GSE484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               GSE485</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:              GSE1037</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            GSE1133.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               GSE479</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                GSE97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               GSE446</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:   bhattacharjee-lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:     ramaswamy-cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:       staunton-nci60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:              GSE2361</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               GSE740</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                GSE96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:              GSE4345</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:              GSE6858</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:              GSE4772</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                      experiment.Name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                      Human Lung Transplant - BAL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                    Mouse Expression DB 2001 CHMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                         Ozone effect on airways hyperpermability</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                              Alveoli loss during caloric restriction time course</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                      Alveoli septation inhibition and protection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                               Genetic basis of sensitivity to pulmonary fibrosis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                      Lung cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                        tissue-specific pattern of mRNA expression - Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                               Alveolar septation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                  Large-scale analysis of the mouse transcriptome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                          Lungs from ovalbumen sensitized and challenged C3H mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                               bhattacharjee-lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                 ramaswamy-cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                   staunton-nci60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                      Expression Proflies of Human Normal tissues</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                Rosetta_Merck_Splicing_Experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                  Large-scale analysis of the human transcriptome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:   In Vivo Regulation of Human Skeletal Muscle Gene Expression by Thyroid Hormone</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                  Expression data from experimental murine asthma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: Expression data from pre natal and post natal CCAM samples and relative controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:            24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:            55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:            56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:            57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:            58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:           141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:           166</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:           171</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:           181</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:           197</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:           204</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:           219</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:           225</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:           305</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:           306</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:           398</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:           453</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:           502</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:           532</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          experiment.Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Bronchoalveolar lavage samples collected from lung transplant recipients.  Numeric portion of sample name is an arbitrary patient ID and AxBx number indicates the perivascular (A) and bronchiolar (B) scores from biopsies collected on the same day as the BAL fluid was collected.  Several patients have more than one sample in this series and can be determined by patient number followed by a lower case letter.  Acute rejection state is determined by the combined A and B score - specifically, a combined AB score of 2 or greater is considered an acute rejection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Expression profiles of different mouse tissue samples profiles.\nDate GSE2178 Last Updated: May 29 2005\nContributors:  Michael Mucenski Bruce J Aronow Belinda Peach Mitchell Cohen Amy Moseley Susan E Waltz John Maggio Sandra Degen Steve Potter Thomas Doetschman Chris Erwin Joanna A Groden James Lessard Linda Parysek R Hirsch MaryBeth Genter Jianhua Zhang Kathleen Anderson Jonathan D Katz John Dedman Michael Lehman Jorge A Bezerra Michael D Bates Ming Xu Dan Wiginton Jeff A Molkentin\nIncludes GDS1322.\n Update date: Nov 09 2005.\n Dataset description GDS1322: Gene expression profiles across a wide variety of tissue samples. The tissues examined include nervous system, skeletal muscle, male and female reproductive organs, gastrointestinal tract, as well as organs in a developmental context including lung, liver, kidney, and heart.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Ozone is a very common environmental pollutant and has been associated with the exacerbation of cardiopulmonary diseases like asthma. In this study the molecular mechanisms underlying the effect of ozone on airways hyperpermability were investigated. Two strains of mice, HeJ (Tlr4 mutant) and OuJ (wildtype) were exposed continuously to air or 0.3ppm of ozone. Lungs were removed and RNA was collected to generate expression profiles.\nDate GSE476 Last Updated: Jun 15 2005\nContributors:  Stephan Kleeberger\nIncludes GDS239.\n Update date: Mar 22 2004.\n Dataset description GDS239: Ozone is an environmental pollutant associated with  exacerbation of cardiopulmonary diseases like asthma. HeJ (Tlr4 mutant) and OuJ (wild type) exposed continuously to air or 0.3 ppm of ozone, and lungs  examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Pulmonary alveoli are complex architectural units thought to undergo endogenous or pharmacologically induced programs of regeneration and degeneration. To study the molecular mechanism of alveoli loss mice were calorie restricted at different timepoints. Lungs were harvested and processed for RNA extraction.\nDate GSE478 Last Updated: Jul 14 2006\nContributors:  Donald   Massaro Gloria D Massaro\nIncludes GDS241.\n Update date: Jun 27 2003.\n Dataset description GDS241: Temporal study of lung alveoli destruction induced by caloric restriction. Alveoli number and surface area are linked to oxygen consumption, and calorie restriction lowers oxygen consumption.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     It has been shown that dexamethasone (Dex) impairs the normal lung septation that occurs in the early postnatal period. Treatment with retinoic acid (ATRA) abrogates the effects of Dex. To understand the molecular basis for the Dex indiced inhibition of the formation of the alveoli and the ability of ATRA to prevent the inhibition of septation, gene expression was analyzed in 4-day old mice treated with diluent (control), Dex-treated and ATRA+Dex-treated.\nDate GSE484 Last Updated: Aug 24 2005\nContributors:  Linda Clerch\nIncludes GDS250.\n Update date: Jun 27 2003.\n Dataset description GDS250: Analysis of how dexamethasone (DEX) inhibits normal lung septation and alveoli formation, and mechanism by which retinoic acid (ATRA) protects against DEX. Three day old mice treated with ATRA or control. On day 4, half of each group administered DEX.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          The murine pulmonary response is strain specific.  Susceptible strains such as C57BL6/J experience an inflammatory response followed by progressive lung disease.  Resistant strains such as Balb/c do not experience significant degrees of inflammation or fibrosis.  This study aims to determine the genetic basis of sensitivity differences between strains.\nDate GSE485 Last Updated: Jul 27 2006\nContributors:  David   Moller Y W Chen\nIncludes GDS251.\n Update date: Jun 27 2003.\n Dataset description GDS251: Determination of genetic basis of sensitivity to pulmonary fibrosis induced by bleomycin by comparing susceptible (C57BL6/J) and resistant (Balb/c) strains. C57BL6/J experience inflammatory response then progressive lung disease, Balb/c do not.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: Two prognostically significant subtypes of high-grade lung neuroendocrine tumors independent of small-cell and large-cell neuroendocrine carcinomas identified by gene expression profiles.  BACKGROUND: Classification of high-grade neuroendocrine tumors (HGNT) of the lung currently recognises large-cell neuroendocrine carcinoma (LCNEC) and small-cell lung carcinoma (SCLC) as distinct groups. However, a similarity in histology for these two carcinomas and uncertain clinical course have led to suggestions that a single HGNT classification would be more appropriate. Gene expression profiling, which can reproduce histopathological classification, and often defines new subclasses with prognostic significance, can be used to resolve HGNT classification. METHODS: We used cDNA microarrays with 40386 elements to analyze the gene expression profiles of 38 surgically resected samples of lung neuroendocrine tumors and 11 SCLC cell lines. Samples of large-cell carcinoma, adenocarcinoma, and normal lung were also included to give a total of 105 samples analyzed. The data were subjected to filtering to yield informative genes before unsupervised hierarchical clustering that identified relatedness of tumor samples. FINDINGS: Distinct groups for carcinoids, large-cell carcinoma, adenocarcinoma, and normal lung were readily identified. However, we were unable to distinguish LCNEC from SCLC by gene expression profiling. Three independent rounds of unsupervised hierarchical clustering consistently divided SCLC samples into two main groups with LCNEC samples largely integrated with these groups. Furthermore, patients in one of the groups identified by clustering had a significantly better clinical outcome than the other (83% vs 12% survived for 5 years; p=0.0094. None of the highly proliferative SCLC cell lines subsequently analyzed clustered with this good-prognosis group. INTERPRETATION: Our findings show that HGNT of the lung can be classified into two groups independent of SCLC and LCNEC. To this end, we have identified many genes, some of which encode well-characterized markers of cancer that distinguish the HGNT groups. These results have implications for the diagnosis, classification, and treatment of lung neuroendocrine tumors, and provide important insights into their underlying biology.\nDate GSE1037 Last Updated: May 29 2005\nContributors:  Virtanen Carl Ishikawa Yuichi Jones Michael Yuichi Ishikawa Daisuke Ishikawa Carl Virtanen Michael H Jones\nIncludes GDS619.\n Update date: Jul 12 2004.\n Dataset description GDS619: Molecular classification of lung high-grade neuroendocrine tumor (HGNT) groups. Carcinoids, large-cell carcinoma, adenocarcinoma, small-cell lung carcinoma cell lines, and normal lung examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The tissue-specific pattern of mRNA expression can indicate important clues about gene function. High-density oligonucleotide arrays offer the opportunity to examine patterns of gene expression on a genome scale. Toward this end, we have designed custom arrays that interrogate the expression of the vast majority of protein-encoding human and mouse genes and have used them to profile a panel of 79 human and 61 mouse tissues. The resulting data set provides the expression patterns for thousands of predicted genes, as well as known and poorly characterized genes, from mice and humans. We have explored this data set for global trends in gene expression, evaluated commonly used lines of evidence in gene prediction methodologies, and investigated patterns indicative of chromosomal organization of transcription. We describe hundreds of regions of correlated transcription and show that some are subject to both tissue and parental allele-specific expression, suggesting a link between spatial expression and imprinting.\nDate GSE1133.2 Last Updated: \nContributors:  John B Hogenesch John R Walker Gabriel Kreiman Tim Wiltshire David Block Mimi Hayakawa Hilmar Lapp Richard Soden Michael P Cooke Serge Batalov Keith A Ching Andrew I Su Jie Zhang\nFrom GSE1133\nIncludes GDS592.\n Update date: May 13 2004.\n Dataset description GDS592: Gene atlas of mouse protein-encoding transcriptome. Gene expression profiles from 61 physiologically normal tissues. Samples generated from adult (10-12 week old) C57BL/6 mice (4 male, 3 female) by dissection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The Mg-delta strain of fibrillin-1 deficient mice display an early defect in alveolar septation.  This study attempts to identify expression patterns in the mutant mouse lung compared to their wild-type littermates that suggest pathways that are critical for normal septation.  This experiment focuses on postnatal days 1 and 5.  Other time points will be added at a later date.\nDate GSE479 Last Updated: Jul 27 2006\nContributors:  Enid R Neptune\nIncludes GDS243.\n Update date: Jul 27 2006.\n Dataset description GDS243: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema.\nIncludes GDS244.\n Update date: Jul 27 2006.\n Dataset description GDS244: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema.\nIncludes GDS242.\n Update date: Jul 27 2006.\n Dataset description GDS242: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema. [Switched to use MG-U74A/B/C by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       High-throughput gene expression profiling has become an important tool for investigating transcriptional activity in a variety of biological samples. To date, the vast majority of these experiments have focused on specific biological processes and perturbations. Here, we have generated and analyzed gene expression from a set of samples spanning a broad range of biological conditions. Specifically, we profiled gene expression from 91 human and mouse samples across a diverse array of tissues, organs, and cell lines. Because these samples predominantly come from the normal physiological state in the human and mouse, this dataset represents a preliminary, but substantial, description of the normal mammalian transcriptome. We have used this dataset to illustrate methods of mining these data, and to reveal insights into molecular and physiological gene function, mechanisms of transcriptional regulation, disease etiology, and comparative genomics. Finally, to allow the scientific community to use this resource, we have built a free and publicly accessible website (http://expression.gnf.org) that integrates data visualization and curation of current gene annotations.\nDate GSE97 Last Updated: May 29 2005\nContributors:  M P Cooke J R Walker R G Vega K A Ching L M Sapinoso A I Su T Wiltshire A Patapoutian P G Schultz A P Orth A Moqrich Y Hakak J B Hogenesch G M Hampton\nIncludes GDS182.\n Update date: May 13 2004.\n Dataset description GDS182: Gene expression profiles from a diverse array of tissues, organs, and cell lines, from the normal physiological state. Represents a preliminary description of the normal mammalian transcriptome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Profile of gene expression in lungs of ovalbumen sensitized and challenged C3H mice.\nDate GSE446 Last Updated: May 29 2005\nIncludes GDS347.\n Update date: Mar 22 2004.\n Dataset description GDS347: Lungs of ovalbumin sensitized/challenged C3H mice compared with controls. C3H are ovalbumin resistant. Attempt to identify genes involved in antigen-induced airway hyperresponsiveness.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Classification of human lung carcinomas by mRNA expression profiling reveals distinct adenocarcinoma subclasses.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Multiclass cancer diagnosis using tumor gene expression signatures.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Chemosensitivity prediction by transcriptional profiling.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       We performed expression profiling of 36 types of normal human tissues and identified 2,503 tissue-specific genes. We then systematically studied the expression of these genes in cancers by re-analyzing a large collection of published DNA microarray datasets. Our study shows that integration of each gene's breadth of expression (BOE) in normal tissues is important for biological interpretation of the expression profiles of cancers in terms of tumor differentiation, cell lineage and metastasis. Twenty five total RNA specimens were purchased from Clontech (Palo Alto, CA), Ambion (Austin, TX) and Strategene (La Jolla, CA). We tried to cover as many tissue types as possible by using pooled RNA samples. In order to define breadth-of-expression (BOE) accurately at a reasonable cost, we tried to cover as many tissue types as possible by using pooled RNA samples. Each specimen represents a human organ. We used RNA samples pooled from 2 to 84 donors to avoid differences at the individual level.  Detailed sample information and Affymetrix .CEL files are available at http://www.genome.rcast.u-tokyo.ac.jp/normal/  Publication:Ge X et al., Interpreting expression profiles of cancers by genome-wide survey of breadth of expression in normal tissues.  Genomics. 2005 Aug;86(2):127-141. PMID: 15950434\nLast Updated (by provider): May 11 2006\nContributors:  Xijin Ge SanMing Wang Siego Ihara Hiroyuki Aburatani Yutaka Midorikawa Shogo Yamamoto Shuichi Tsutsumi\nIncludes GDS1096.\n Update date: Apr 08 2005.\n Dataset description GDS1096: Expression profiling of 36 types of normal tissue. Each RNA tissue sample pooled from several donors.  Results identify tissue specific genes and provide baselines for interpreting gene expression in cancer.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                This series represents 52 tissues hybridized across 5 different chip patterns. Probes were placed at every exon-exon junction in each transcript.\nLast Updated (by provider): May 29 2005\nContributors:  Christopher D Armour Eric E Schadt Daniel D Shoemaker Patrick M Loerch Philip Garrett-Engele Jason M Johnson Roland Stoughton Ralph Santos Zhengyan Kan John Castle\nIncludes GDS830.\n Update date: Nov 10 2004.\n Dataset description GDS830: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS829.\n Update date: Nov 10 2004.\n Dataset description GDS829: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions.  mRNA from 52 tissues and cell lines examined.  Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS833.\n Update date: Nov 10 2004.\n Dataset description GDS833: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS831.\n Update date: Nov 10 2004.\n Dataset description GDS831: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS832.\n Update date: Nov 10 2004.\n Dataset description GDS832: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants. [Switched to use Rosetta_Merged by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    High-throughput gene expression profiling has become an important tool for investigating transcriptional activity in a variety of biological samples. To date, the vast majority of these experiments have focused on specific biological processes and perturbations. Here, we have generated and analyzed gene expression from a set of samples spanning a broad range of biological conditions. Specifically, we profiled gene expression from 91 human and mouse samples across a diverse array of tissues, organs, and cell lines. Because these samples predominantly come from the normal physiological state in the human and mouse, this dataset represents a preliminary, but substantial, description of the normal mammalian transcriptome. We have used this dataset to illustrate methods of mining these data, and to reveal insights into molecular and physiological gene function, mechanisms of transcriptional regulation, disease etiology, and comparative genomics. Finally, to allow the scientific community to use this resource, we have built a free and publicly accessible website (http://expression.gnf.org) that integrates data visualization and curation of current gene annotations.\nLast Updated (by provider): May 29 2005\nContributors:  A Moqrich R G Vega P G Schultz M P Cooke A I Su T Wiltshire A P Orth K A Ching L M Sapinoso J B Hogenesch J R Walker Y Hakak A Patapoutian G M Hampton\nIncludes GDS181.\n Update date: May 13 2004.\n Dataset description GDS181: Gene expression profiles from a diverse array of tissues, organs, and cell lines, from the normal physiological state. Represents a preliminary description of the normal mammalian transcriptome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  This SuperSeries is composed of the following subset Series: GSE3461: Gene expression in miscellaneous human tissues and cell lines GSE3462: Triiodothyronine Treatment: Effects on vastus lateralis skeletal muscle  Abstract: Thyroid hormones are key regulators of metabolism that modulate transcription via nuclear receptors. Hyperthyroidism is associated with increased metabolic rate, protein breakdown, and weight loss. Although the molecular actions of thyroid hormones have been studied thoroughly, their pleiotropic effects are mediated by complex changes in expression of an unknown number of target genes. Here, we measured patterns of skeletal muscle gene expression in five healthy men treated for 14 days with 75 µg of triiodothyronine, using 24,000 cDNA element microarrays. To analyze the data, we used a new statistical method that identifies significant changes in expression and estimates the false discovery rate. The 381 up-regulated genes were involved in a wide range of cellular functions including transcriptional control, mRNA maturation, protein turnover, signal transduction, cellular trafficking, and energy metabolism. Only two genes were down-regulated. Most of the genes are novel targets of thyroid hormone. Cluster analysis of triiodothyronine-regulated gene expression among 19 different human tissues or cell lines revealed sets of coregulated genes that serve similar biologic functions. These results define molecular signatures that help to understand the physiology and pathophysiology of thyroid hormone action.\nLast Updated (by provider): Dec 12 2006\n [Switched to use SMHu24k by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Experimental asthma was induced in BALB/c mice by sensitization and challenge with the allergen ovalbumin. Control groups received PBS. To investigate the innate immune component of experimental asthma, we also analyzed recombinase activating gene (RAG) deficient mice following exposure to ovalbumin and control PBS\nLast Updated (by provider): Apr 23 2007\nContributors:  Xin Lu Patricia W Finn Vipul V Jain David L Perkins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               This study aims at giving an insight on gene expression in CCAM.\nLast Updated (by provider): May 05 2006\nContributors:  Agnes Paquet Sam Hawgood Amy Wagner</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.Troubled experiment.Accession experiment.Database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE              GSE2018                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE              GSE2178                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE               GSE476                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE               GSE478                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE               GSE484                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE               GSE485                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE              GSE1037                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE              GSE1133                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE               GSE479                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                GSE97                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE               GSE446                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE              GSE2361                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE               GSE740                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                GSE96                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE              GSE4345                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE              GSE6858                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE              GSE4772                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.SampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: http://www.ncbi.nlm.nih.gov/geo/                     34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: http://www.ncbi.nlm.nih.gov/geo/                    161</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: http://www.ncbi.nlm.nih.gov/geo/                     91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: http://www.ncbi.nlm.nih.gov/geo/                    122</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: http://www.ncbi.nlm.nih.gov/geo/                     90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: http://www.ncbi.nlm.nih.gov/geo/                      5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                             &lt;NA&gt;                    203</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                             &lt;NA&gt;                    280</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                             &lt;NA&gt;                     60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: http://www.ncbi.nlm.nih.gov/geo/                     36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: http://www.ncbi.nlm.nih.gov/geo/                    264</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: http://www.ncbi.nlm.nih.gov/geo/                     85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: http://www.ncbi.nlm.nih.gov/geo/                     16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: http://www.ncbi.nlm.nih.gov/geo/                     14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.LastUpdated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    2023-09-25 18:33:39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    2023-10-03 19:39:08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    2022-08-25 17:22:56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    2022-09-04 16:11:24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    2022-08-12 14:55:30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    2022-08-25 17:24:31</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    2022-08-22 19:49:36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    2023-04-20 07:08:24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    2022-08-16 16:41:01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    2023-09-26 07:33:45</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    2023-05-31 23:03:37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    2023-09-07 21:43:44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    2023-09-07 21:47:00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    2023-09-07 21:49:19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    2021-11-03 18:44:27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    2022-05-12 07:30:04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    2022-09-05 02:52:57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    2023-05-09 20:52:44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    2022-08-17 05:35:24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    2023-09-07 22:32:46</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: This data set may have a batch artifact (PC 1), p=0.00069053</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:  This data set may have a batch artifact (PC 1), p=0.0054886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: This data set may have a batch artifact (PC 2), p=3.1342e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  This data set may have a batch artifact (PC 1), p=0.0024570</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: This data set may have a batch artifact (PC 3), p=0.00022690</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE                  1                1           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                  1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                  1                1           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE                  1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                TRUE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.Name taxon.Scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:  0.99763407      0.8750      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: -0.22593272      0.0625      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:  0.85574622      0.6500      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:  0.42626507      0.8125      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:  0.85510163      0.6500      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:  0.42565509      0.5625      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: -0.08397851      0.1875      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:  0.84599960      0.2500      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:  0.42857143      0.1875      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:  0.55722466      0.3125      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: -0.03621544     -0.3750      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:  0.39301864      0.2500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:  0.23585518      0.0000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:  0.24928653      0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  0.54414795      0.8750      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: -0.14285714     -0.2500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:  0.28369987      0.7500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: -0.13451104      0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:  0.85445543      0.8750      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:  0.28025816      0.3750      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.Database.Name taxon.Database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                hg38                87</span>
<span class="r-in"><span><span class="fu">get_datasets</span><span class="op">(</span>query <span class="op">=</span> <span class="st">"lung"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ShortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             GSE31552</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             GSE37768</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:             GSE66303</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               GSE476</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               GSE485</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:              GSE7271</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             GSE11056</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:             GSE17248</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:             GSE17249</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:             GSE18707</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:             GSE19605</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:             GSE25612</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:              GSE3141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:             GSE17770</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:             GSE30219</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:             GSE41041</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:             GSE29042</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:             GSE35485</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:              GSE3593</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:             GSE50081</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                             experiment.Name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                              Expression Data from human Lung tissue of Patients with Non Small Cell Lung Cancer (NSCLC)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                              Expression data in lung tissue from moderate COPD patients, healthy smokers and nonsmokers</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                         Transcriptional responses to i.v. administered I-131 in various mouse normal tissues underlie diurnal variation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                Ozone effect on airways hyperpermability</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                      Genetic basis of sensitivity to pulmonary fibrosis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                         Sex differences in gene expression profiles during hantavirus infection of rats</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                         Expression data from mouse lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                              Inflammatory gene expression in response to Francisella tularensis exposure in Balb/c mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                  Inflammatory gene expression in response to sublethal phosgene exposure in Balb/c mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                          Expression data from mouse lung treated with MACA, BSA or HBSS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                             Specificity for the Nature of Inflammation in Lung Cancer Promotion in Mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                        Circadian regulation in rat Lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                     Lung Cancer Dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                             Expression Profiling Identifies Genes Involved in Emphysema</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                              "Off-context" gene expression in lung cancer identifies a group of metastatic-prone tumors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                   Pulmonary retention of titanium dioxide nanoparticles and biological outcomes following low dose instillation in mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                    Expression profiling of mice exposed to multiwalled carbon nanotubes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                          Transcriptional Programs Controlling Perinatal Lung Maturation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                  Genomic approach to refine prognosis for adjuvant therapy in early stage non-small cell lung carcinoma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: Validation of a histology-independent prognostic gene signature for early stage, non-small cell lung cancer including stage IA patients</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:          5342</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          5453</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:         15782</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:            55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:            58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:          1602</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:          1642</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:          2879</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:          2880</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:          2922</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:          2985</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:          3253</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:          4079</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:          4909</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:          6542</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:          7180</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:          7410</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:          7633</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:          7651</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:          8647</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        experiment.Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: Lung cancers are a heterogeneous group of diseases with respect to biology and clinical behavior. Currently, diagnosis and classification are based on histological morphology and immunohistological methods for discrimination between two main histologic groups: small cell lung cancer (SCLC) and non-small cell lung cancer which account for 20% and 80% of lung carcinomas, respectively. NSCLCs, which are divided into the three major subtypes adenocarcinoma, squamous cell carcinoma and dedifferentiated large cell carcinoma, show different characteristics such as the expression of certain keratins or production of mucin and lack of neuroedocrine differentiation. The molecular pathogenesis of lung cancer involves the accumulation of genetic und epigenetic alterations including the activation of proto-oncogenes and inactivation of tumor suppressor genes which are different for lung cancer subgroups. The development of microarray technologies opened up the possibility to quantify the expression of a large number of genes simultaneously in a given sample. There are several recent reports on expression profiling on lung cancers but the analysis interpretation of the results might be difficult because of the heterogeneity of cellular components. The methods used for sample selection and processing can have a strong influence on the expression values obtained through microarray profiling. Laser capture microdissection (LCM) provides higher specificity in the selection of target cells compared to traditional bulk tissue selection methods, but at an increased processing cost. Here we describe the use of an expression microarray study on NSCLC samples and surrounding tissue, comparing macroscopic lung tumor and tissue samples (“grind and bind”), versus tumor and alveolar compartment cells laser capture microdissected (LCM) from the same macroscopic lung samples. In this study, an initial set of 16 pairs of macroscopic tumor and non-tumor samples (9 pairs squamous-cell carcinoma, 7 pairs adenocarcinoma) was selected for bulk/macro sampling. Of these 16 pairs, 12 pairs were reanalyzed using laser capture microdissection (LCM) for sampling the cells (7 pairs squamous, 5 pairs adenocarcinomas). For macroscopic samples, 50 to 80 µg of tissue was used to isolate total RNA. Gene expression profile was determined using Affymetrix Human Genome Gene 1.0 ST genechip.  For the LCM samples, from representative slides histologically confirmed and mapped by a pathologist, approximately 1000 cells/sample were collected by LCM. cDNA was amplified using Nugen WT-Ovation One-Direct amplification system. In order to validate Nugen amplification bias of WT-Ovation One-Direct amplification system, the total RNA samples of 10 pairs of macroscopic tumor and non-tumor samples were amplified with this amplification system, and their cDNA was used to microarray.\nLast Updated (by provider): Jun 04 2012\nContributors:  S D Spivack T Wang N Mullapudi M Shi S Keller J Locker J Lin G Marquardt</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Chronic obstructive pulmonary disease (COPD) is a progressive and irreversible chronic inflammatory lung disease. The abnormal inflammatory response of the lung, mainly to cigarette smoke, causes multiple cellular and structural changes affecting all of its compartments, which leads to disease progression. The molecular mechanisms underlying these pathological changes are still not fully understood The aim of this study was to identify genes and molecular pathways potentially involved in the pathogenesis of COPD\nLast Updated (by provider): Jun 18 2012\nContributors:  Ricardo Bastos Susana G Kalko</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Circadian rhythm study on transcriptional responses to i.v. administered 90 kBq iodine-131 after 24h in mouse kidney cortex and medulla, liver, lungs, spleen, and thyroid. Female BALB/c nude mice (n=4/group) were i.v. injected with 90 kBq 131I at three different times of day, i.e. at 9.00 am, at 12.00 pm, or at 3.00 pm, and killed under anesthesia after 24h following treatment for excision of organs. For each time of day, a control group (n=3-4/group) was i.v. injected with physiol. saline. The kidneys, liver, lungs, spleen, and thyroid were excised, flash-frozen, and stored at -80°C until extraction of total RNA. Total RNA was extracted from homogenized tissue samples and subjected to expression analysis using RNA microarray technology.\nAt time of import, last updated (by provider) on: Jun 14 2018\n\nContributors: ; [Khalil Helou, Toshima Z Parris, Eva Forssell-Aronsson, Britta Langen, Nils Rudqvist]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Ozone is a very common environmental pollutant and has been associated with the exacerbation of cardiopulmonary diseases like asthma. In this study the molecular mechanisms underlying the effect of ozone on airways hyperpermability were investigated. Two strains of mice, HeJ (Tlr4 mutant) and OuJ (wildtype) were exposed continuously to air or 0.3ppm of ozone. Lungs were removed and RNA was collected to generate expression profiles.\nDate GSE476 Last Updated: Jun 15 2005\nContributors:  Stephan Kleeberger\nIncludes GDS239.\n Update date: Mar 22 2004.\n Dataset description GDS239: Ozone is an environmental pollutant associated with  exacerbation of cardiopulmonary diseases like asthma. HeJ (Tlr4 mutant) and OuJ (wild type) exposed continuously to air or 0.3 ppm of ozone, and lungs  examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        The murine pulmonary response is strain specific.  Susceptible strains such as C57BL6/J experience an inflammatory response followed by progressive lung disease.  Resistant strains such as Balb/c do not experience significant degrees of inflammation or fibrosis.  This study aims to determine the genetic basis of sensitivity differences between strains.\nDate GSE485 Last Updated: Jul 27 2006\nContributors:  David   Moller Y W Chen\nIncludes GDS251.\n Update date: Jun 27 2003.\n Dataset description GDS251: Determination of genetic basis of sensitivity to pulmonary fibrosis induced by bleomycin by comparing susceptible (C57BL6/J) and resistant (Balb/c) strains. C57BL6/J experience inflammatory response then progressive lung disease, Balb/c do not.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Gene expression profiles were examined in whole lung tissue collected from male and female Long-Evans rats at different time points after inoculation with Seoul virus (i.e., the species-specific hantavirus that infects Norway rats) We used microarray analyses to analyze global gene expression differences between the sexes during hantavirus infection\nLast Updated (by provider): Jan 29 2008\nContributors:  Sabra L Klein</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Exposure to high levels of arsenic in drinking water is associated with several types of cancers including lung, bladder and skin, as well as vascular disease and diabetes. Drinking water standards are based primarily on epidemiology and extrapolation from higher dose experiments, rather than measurements of phenotypic changes associated with chronic exposure to levels of arsenic similar to the current standard of 10ppb, and little is known about the difference between arsenic in food as opposed to arsenic in water. Measurement of phenotypic changes at low doses may be confounded by the effect of laboratory diet, in part because of trace amounts of arsenic in standard laboratory chows, but also because of broad metabolic changes in response to the chow itself. Finally, this series contrasts 8hr, 1mg/kg injected arsenic with the various chronic exposures, and also contrasts the acute effects of arsenic, dexamethasone or their combination. Male C57BL/6 mice were fed on two commercially available laboratory diets (LRD-5001 and AIN-76A) were chronically exposed, through drinking water or food, to environmentally relevant concentrations of sodium arsenite, or acutely exposed to dexamethasone.\nLast Updated (by provider): Mar 15 2009\nContributors:  Joshua W Hamilton Michael A Ihnat</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Francisella tularensis may enter the body thorugh the lungs and cause fatal infection. In this study the inflammatory response to the virulent strain of Francisella (Schu4) was mapped over a 96h time-course using a custom microarray.\nLast Updated (by provider): Mar 01 2011\nContributors:  Jonathan David Gareth D Griffiths Roman Lukaszewski</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Phosgene is a lung damaging chemical warfare agent which can cause death by the inhalation route. In this study the inflammatory response to phosgene was mapped over a 96h time-course using a custom microarray in the Balb/c mouse.\nLast Updated (by provider): Mar 01 2011\nContributors:  Jonathan David Gareth D Griffiths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          We hypothesize that there is a set of differentially-expressed gene “biomarkers” that distinguishes the immune response associated with the induction of allergic sensitization from non-allergic immune responsiveness following a single exposure. The goal of this preliminary study was to identify potential biomarkers that differentiate the response to allergens vs non-allergens following an acute exposure in naïve individuals. We evaluated the gene transcriptional responses of mouse lungs following a single exposure to agents of high or low allergenicity or vehicle control.\nLast Updated (by provider): Dec 22 2009\nContributors:  Cherie M Pucheu-Haston Marsha D Ward</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      A large amount of epidemiologic data supports a role for chronic inflammation in epithelial carcinogenesis. In the lung, several studies have found that smokers with chronic obstructive pulmonary disease (COPD), an inflammatory disease of the airways and alveoli, have an increased risk of lung cancer (1.3 to 4.9 fold) compared to smokers without COPD. We have also shown that COPD-like airway inflammation induced by an aerosolized lysate of non-typeable Hemophilus influenzae (NTHi) promotes lung cancer in a Clara cell-targeted K-ras mutant mouse model (CC-LR) of lung cancer. In contrast, existing epidemiologic data suggest that allergic inflammation of the airways may be protective against lung cancer. We tested this association in a mouse model of allergic airway inflammation. CC-LR mice were sensitized to ovalbumin by intraperitoneal injection weekly for two weeks, then challenged for 30 min to an aerosol of ovalbumin in 0.9% saline weekly for eight weeks. This resulted in eosinophilic lung inflammation associated with increased levels of T helper 2 (Th2) cytokines and mucous metaplasia of airway epithelium, similar to what is seen in asthma patients. However, consistent with epidemiologic data, this type of inflammation did not result in any significant differences in lung surface tumor number (22 ± 3 in OVA exposed vs 26 ± 6 in control mice). We conclude that asthma-like (Th2) inflammation does not promote lung carcinogenesis in a Ras-initiated background, and demonstrate a clear specificity for the nature of inflammation in lung cancer promotion. These findings will assist in determination of the essential cells and signaling events in lung cancer promotion by inflammation. \nLast Updated (by provider): May 18 2011\nContributors:  Seyed J Moghaddam Scott E Evans</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Circadian rhythms are oscillations with a periodicity of 24 hours that are controlled by an endogenous clock and are observed in virtually all aspects of mammalian function from expression of genes to complex physiological processes. The master clock is present in the suprachiasmatic nucleus (SCN) in the anterior part of the hypothalamus and controls peripheral clocks present in other parts of the body . Although much is known about the mechanism of the central clock in the SCN, the regulation of clocks present in peripheral tissues is still unclear.  This study is designed to examine fluctuations in gene expression in lungs within the 24 hour circadian cycle in normal animals. The objectives of this study is  to identify and analyze circadian oscillation in gene expression in lungs, and to identify the role of circadian regulation in coordinating the functioning of this dynamic organ.\nLast Updated (by provider): Aug 10 2011\nContributors:  Richard R Almon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Signatures of Oncogenic Pathway Deregulation in Human Cancers.  The ability to define cancer subtypes, recurrence of disease, and response to specific therapies using DNA microarray-based gene expression signatures has been demonstrated in multiple studies. Such data is also of substantial importance to the analysis of cellular signaling pathways central to the oncogenic process. With this focus, we have developed a series of gene expression signatures that reliably reflect the activation status of several oncogenic pathways. When evaluated in several large collections of human cancers, these gene expression signatures identify patterns of pathway deregulation in tumors, and clinically relevant associations with disease outcomes. Combining signature-based predictions across several pathways identifies coordinated patterns of pathway deregulation that distinguish between specific cancers and tumor sub-types. Clustering tumors based on pathway signatures further defines prognosis in respective patient subsets, demonstrating that patterns of oncogenic pathway deregulation underlie the development of the oncogenic phenotype and reflect the biology and outcome of specific cancers. Furthermore, predictions of pathway deregulation in cancer cell lines are shown to coincide with sensitivity to therapeutic agents that target components of the pathway, underscoring the potential for such pathway prediction to guide the use of targeted therapeutics.\nLast Updated (by provider): Nov 12 2011\nContributors:  Joseph R Nevins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Chronic obstructive pulmonary disease (COPD) is a major public health problem. The aim of this study was to identify genes involved in emphysema severity in COPD patients. Gene expression profiling was performed on total RNA extracted from non-tumor lung tissue from 30 smokers with emphysema. Class comparison analysis based on gas transfer measurement was performed to identify differentially expressed genes. Genes were then selected for technical validation by quantitative reverse transcriptase-PCR (qRT-PCR) if also represented on microarray platforms used in previously published emphysema studies. Genes technically validated advanced to tests of biological replication by qRT-PCR using an independent test set of 62 lung samples. Class comparison identified 98 differentially expressed genes (p&lt;0.01). Fifty-one of those genes had been previously evaluated in differentiation between normal and severe emphysema lung. qRT-PCR confirmed the direction of change in expression in 29 of the 51 genes and 11 of those validated, remaining significant at p&lt;0.05. Biological replication in an independent cohort confirmed the altered expression of eight genes, with seven genes differentially expressed by greater than 1.3 fold, identifying these as candidate determinants of emphysema severity. Gene expression profiling of lung from emphysema patients identified seven candidate genes associated with emphysema severity including COL6A3, SERPINF1, ZNHIT6, NEDD4, CDKN2A, NRN1 and GSTM3.\nLast Updated (by provider): Mar 21 2012\nContributors:  Kwun M Fong Ian A Yang Nick K Hayward Rayleen V Bowman Jill E Larsen Santiyagu M Savarimuthu Francis Sandra J Pavey</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              An unexplored consequence of epigenetic alterations associated with cancer is the ectopic expression of tissue-restricted genes. Here, a new strategy was developed to decipher genome-wide expression data in search for these “off-context” gene activations, which consisted first, in identifying a large number of tissue-specific genes normally epigenetically silenced in most somatic cells and second, in using them as cancer biomarkers on an “on/off” basis. Applying this concept to analyze whole-genome transcriptome data in lung cancer, we discovered a specific group of 26 genes whose expression was a strong and independent predictor of poor prognosis in our cohort of 293 lung tumours, as well as in two independent external populations. In addition, these 26 classifying genes enabled us to isolate a homogenous group of metastatic-prone highly aggressive tumours, whose characteristic gene expression profile revealed a high proliferative potential combined to a significant decrease in immune and signaling functions. This work illustrates a new approach for a personalized management of cancer, with applications to any cancer type.\nLast Updated (by provider): Jun 15 2013\nContributors:  Alexandra Debernardi Sylvie Lantuejoul Aurore LeBescont Baptiste Jacquiau Christian Brambilla Elisabeth Brambilla Pierre-Yves Brichon Sophie Rousseaux Aurelien Vesin Saadi Khochbin Hélène Nagy-Mignotte Jean-François Timsit Denis Moro-Sibilot Mary Callanan David G Beer Anne-Laure Vitte</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Pulmonary exposure to high doses of nanoparticles (NP) leads to well characterized lung toxicity in addition to long-term NP retention. However, pulmonary NP accumulation and toxicity following low dose exposures are not well described. In the present study we sought to: (1) investigate particle retention in mouse lungs following intratracheal instillation of varying doses of nano-sized titanium dioxide (nano-TiO2) and (2) determine the effects of long-term particle accumulation on pulmonary systems. Female C57BL/6 mice were exposed to rutile nano-TiO2 (primary size of 20.6 nm and surface area of 107.7 m2/g) via single intratracheal instillations of 18, 54 and 162 µg/mouse and sampled 1, 3 and 28 days post-exposure. The deposition of nano-TiO2 in the lungs was assessed using Nanoscale Hyperspectral Microscope. DNA microarrays, pathway-specific real-time RT-PCR (qPCR) and gene-specific qPCR arrays, and tissue protein analyses were employed to characterize pulmonary responses. Hyperspectral mapping showed dose-dependent retention of nano-TiO2 in the lungs up to 28 days post-exposure time. Retention did not correlate with the extent of inflammatory neutrophil influx into the lungs. DNA microarray analysis showed altered expression of approximately 3000 genes across all treatment groups (±1.3 fold; p&lt;0.1). Several inflammatory mediators changed in a dose- and time-dependent manner at both the mRNA and protein levels. Although the low dose exposure failed to induce observable inflammation, significant changes in the expression of genes and proteins associated with inflammation were observed. Moreover, diminished (or absent) neutrophil influx in the low and medium dose groups was correlated with negative regulation of genes associated with ion homeostasis and muscle regulation. Gene expression changes for several inflammatory mediators have previously been noted in mice exposed to the same nano-TiO2 via inhalation. Our results suggest that retention of nano-TiO2 in the absence of inflammation and effective clearance can perturb calcium and ion homeostasis, and affect smooth muscle activities over time.\nLast Updated (by provider): Nov 03 2013\nContributors:  H Wallin Mainul Husain U Vogel A T Saber Charles Guo Carole Yauk N R Jacobsen Sabina Halappanavar Andrew Williams K A Jensen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Mice were aspirated with multiwalled carbon nanotubes in four doses ranging from 10 µg to 80 µg, plus control, and sacrificed after 1, 7, 28, and 56 days. RNA was extracted from the lungs and expression profilling by microarray was performed.\nLast Updated (by provider): May 13 2013\nContributors:  Goran Boskovic Mike Wolfarth Yong Qian Maricica Pacurari Nancy L Guo James Denvir Vincent Castranova Dale Porter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 In this study, time dependent genome wide lung mRNA profiling changes were assessed using C57BL/6J and A/J mice. Through comprehensive bioinformatics and functional genomics analyses, we identified both temporal and strain dependent gene expression patterns, systemically mapped key regulators, bioprocesses, and transcriptional networks controlling lung maturation, providing the basis for new therapeutic strategies to enhance lung function in preterm infants. \nLast Updated (by provider): Nov 25 2013\nContributors:  Steve A Murray Jeffrey A Whitsett Susan E Wert Valerie Besnard Leah R Donahue Yanhua Wang Yan Xu Machiko Ikegami</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Abstract from paper - Potti A, et al\nLast Updated (by provider): Aug 29 2013\nContributors:  Andrea Bild David Har Sayan Mukherjee Anil Potti Michael Kelley Mike West Jason Koontz Geoffrey Ginsburg Holly Dressman Rebecca Prince</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Background: Patients with early stage non-small cell lung carcinoma (NSCLC) may benefit from treatments based on more accurate prognosis. A 15-gene prognostic classifier for NSCLC was identified from mRNA expression profiling of tumor samples from the NCIC CTG JBR.10 trial. Here, we assessed its value in an independent set of cases. Methods: Expression profiling was performed on RNA from frozen, resected tumor tissues corresponding to 181 Stage I and II NSCLC cases collected at University Health Network (UHN181). Kaplan-Meier methodology was used to estimate three year overall survival probabilities and the prognostic effect of the classifier was assessed using log-rank testing. Cox proportional hazards model evaluated the signature's effect adjusting for clinical prognostic factors. Results: Expression data of the 15-gene classifier stratified UHN181 cases into high and low-risk subgroups with significantly different overall survival (HR=1.92, 95% CI: 1.15-3.23, p=0.012). Its strength as a prognostic classifier was superior to stage alone (HR=1.52, 95% CI: 0.90-2.55, p-value=0.11). In subgroup analysis, this classifier predicted survival in 127 Stage I patients (HR=2.17, 95% CI: 1.12-4.20, p=0.018) and the smaller subgroup of 48 Stage IA patients (HR=5.61, 95% CI: 1.19-26.45, p=0.014. The signature was prognostic for both adenocarcinoma and squamous cell carcinoma cases (HR= 1.76, p-value=0.058; HR= 4.19, p-value=0.045, respectively). Conclusions: The prognostic accuracy of a 15-gene classifier was validated in an independent cohort of 181 early stage NSCLC samples including Stage IA cases and in different NSCLC histologic subtypes.\nLast Updated (by provider): Jun 06 2014\nContributors:  Ni Liu Dan Strumpf Igor Jurisica Ming-Sound Tsao Chang-Qi Zhu Frances A Shepherd Melania Pintilie</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.Troubled experiment.Accession experiment.Database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE             GSE31552                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE             GSE37768                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE             GSE66303                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE               GSE476                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE               GSE485                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE              GSE7271                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE             GSE11056                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE             GSE17248                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE             GSE17249                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE             GSE18707                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE             GSE19605                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE             GSE25612                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE              GSE3141                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE             GSE17770                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE             GSE30219                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE             GSE41041                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE             GSE29042                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE             GSE35485                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE              GSE3593                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE             GSE50081                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.SampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: http://www.ncbi.nlm.nih.gov/geo/                    131</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: http://www.ncbi.nlm.nih.gov/geo/                     38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: http://www.ncbi.nlm.nih.gov/geo/                    136</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: http://www.ncbi.nlm.nih.gov/geo/                     55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: http://www.ncbi.nlm.nih.gov/geo/                     35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: http://www.ncbi.nlm.nih.gov/geo/                     35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: http://www.ncbi.nlm.nih.gov/geo/                     21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12: http://www.ncbi.nlm.nih.gov/geo/                     54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13: http://www.ncbi.nlm.nih.gov/geo/                    111</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14: http://www.ncbi.nlm.nih.gov/geo/                     30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: http://www.ncbi.nlm.nih.gov/geo/                    307</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: http://www.ncbi.nlm.nih.gov/geo/                     65</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: http://www.ncbi.nlm.nih.gov/geo/                    160</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: http://www.ncbi.nlm.nih.gov/geo/                     37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: http://www.ncbi.nlm.nih.gov/geo/                    198</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: http://www.ncbi.nlm.nih.gov/geo/                    181</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.LastUpdated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    2022-08-27 07:33:34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    2022-08-25 07:31:16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    2022-08-25 07:31:29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    2022-08-25 17:22:56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    2022-08-25 17:24:31</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    2022-08-18 14:36:00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    2022-09-06 21:02:07</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    2023-05-09 01:38:52</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    2023-05-09 01:39:08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    2022-08-28 22:45:28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    2023-04-22 06:22:18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    2022-08-10 06:51:57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    2023-09-14 18:30:05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    2020-03-28 03:50:22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    2023-09-21 21:37:11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    2022-08-24 07:32:17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    2022-08-21 07:33:32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    2022-09-09 07:33:16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    2023-09-16 07:32:40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    2020-03-29 15:03:17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: This data set may have a batch artifact (PC 2), p=1.1725e-17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:  This data set may have a batch artifact (PC 1), p=0.0054886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: This data set may have a batch artifact (PC 1), p=0.00045464</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12: This data set may have a batch artifact (PC 1), p=2.9829e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                         &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: This data set may have a batch artifact (PC 2), p=2.7222e-18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: This data set may have a batch artifact (PC 1), p=2.3959e-24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:     This data set may have a batch artifact (PC 1), p=0.0000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                TRUE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                TRUE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                TRUE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE                 -1                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE                  1                1           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                 -1                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                TRUE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      geeq.qScore geeq.sScore taxon.Name  taxon.Scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:  0.267308827      1.0000      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:  0.997089314      0.7500      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: -0.002979611      0.6875      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:  0.855746221      0.6500      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:  0.425655088      0.5625      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:  0.426196843      0.7500        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:  0.570005092      1.0000      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: -0.332502595      0.0000      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: -0.112016784      0.2500      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:  0.855836865      0.8750      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: -0.037810170      0.4375      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:  0.427544601      0.8750        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13: -0.038658882      0.7500      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14: -0.359990156      0.1250      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  0.411906834      0.7500      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:  0.996249876      0.5000      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:  0.559280922      0.5000      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:  0.426649920      1.0000      mouse      Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:  0.263532369      0.6250      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:  0.987609971      1.0000      human      Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.Database.Name taxon.Database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                hg38                87</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

