<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve all datasets — get_datasets • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve all datasets — get_datasets"><meta property="og:description" content="Retrieve all datasets"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve all datasets</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>get_datasets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieve all datasets</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_datasets</span><span class="op">(</span></span>
<span>  query <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  filter <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  taxa <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  uris <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  offset <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  sort <span class="op">=</span> <span class="st">"+id"</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>The search query. Either plain text ('traumatic'), or an ontology
term URI ('http://purl.obolibrary.org/obo/UBERON_0002048'). Datasets that
contain the given string in their short of full name will also be matched.</p></dd>


<dt>filter</dt>
<dd><p>Filter results by matching expression. Use <code><a href="filter_properties.html">filter_properties</a></code>
function to get a list of all available parameters. These properties can be
combined using "and" "or" clauses and may contain common operators such as "=", "&lt;" or "in".
(e.g. "taxon.commonName = human", "taxon.commonName in (human,mouse), "id &lt; 1000")</p></dd>


<dt>taxa</dt>
<dd><p>A vector of taxon common names (e.g. human, mouse, rat). Providing multiple
species will return results for all species. These are appended
to the filter and equivalent to filtering for <code>taxon.commonName</code> property</p></dd>


<dt>uris</dt>
<dd><p>A vector of ontology term URIs. Providing multiple terms will
return results containing any of the terms and their children. These are</p></dd>


<dt>offset</dt>
<dd><p>The offset of the first retrieved result.</p></dd>


<dt>limit</dt>
<dd><p>Optional, defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt>sort</dt>
<dd><p>Order results by the given property and direction. The '+' sign
indicate ascending order whereas the '-' indicate descending.</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data table with information about the queried dataset(s). A list if
<code>raw = TRUE</code>. Returns an empty list if no datasets matched. A successful
response may contain 'Geeq' information, which aims to provide a unified
metric to measure experiments by the quality of their data, and their
suitability for use in Gemma. You can
read more about the geeq properties <a href="https://pavlidislab.github.io/Gemma/geeq.html" class="external-link">here</a>.</p>


<p>The fields of the output data.table are:</p>


<ul><li><p><code>experiment.shortName</code>: Shortname given to the dataset within Gemma. Often corresponds to accession ID</p></li>
<li><p><code>experiment.name</code>: Full title of the dataset</p></li>
<li><p><code>experiment.ID</code>: Internal ID of the dataset.</p></li>
<li><p><code>experiment.description</code>: Description of the dataset</p></li>
<li><p><code>experiment.troubled</code>: Did an automatic process within gemma or a curator mark the dataset as "troubled"</p></li>
<li><p><code>experiment.accession</code>: Accession ID of the dataset in the external database it was taken from</p></li>
<li><p><code>experiment.database</code>: The name of the database where the dataset was taken from</p></li>
<li><p><code>experiment.URI</code>: URI of the original database</p></li>
<li><p><code>experiment.sampleCount</code>: Number of samples in the dataset</p></li>
<li><p><code>experiment.batchEffect</code>: A text field describing whether the dataset has batch effects</p></li>
<li><p><code>geeq.batchCorrected</code>: Whether batch correction has been performed on the dataset.</p></li>
<li><p><code>geeq.batchConfound</code>: 0 if batch info isn't available, -1 if batch counfoud is detected, 1 if batch information is available and no batch confound found</p></li>
<li><p><code>geeq.batchEffect</code>: -1 if batch p value &lt; 0.0001, 1 if batch p value &gt; 0.1, 0 if otherwise and when there is no batch information is available or when the data is confounded with batches.</p></li>
<li><p><code>geeq.rawData</code>: -1 if no raw data available, 1 if raw data was available. When available, Gemma reprocesses raw data to get expression values and batches</p></li>
<li><p><code>geeq.qScore</code>: Data quality score given to the dataset by Gemma.</p></li>
<li><p><code>geeq.sScore</code>: Suitability score given to the dataset by Gemma. Refers to factors like batches, platforms and other aspects of experimental design</p></li>
<li><p><code>taxon.Name</code>: Name of the species</p></li>
<li><p><code>taxon.Scientific</code>: Scientific name for the taxon</p></li>
<li><p><code>taxon.ID</code>: Internal identifier given to the species by Gemma</p></li>
<li><p><code>taxon.NCBI</code>: NCBI ID of the taxon</p></li>
<li><p><code>taxon.Database.Name</code>: Underlying database used in Gemma for the taxon</p></li>
<li><p><code>taxon.Database.ID</code>: ID of the underyling database used in Gemma for the taxon</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">get_datasets</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:              GSE2018</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:              GSE2872</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:              GSE4523</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:              GSE4036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:              GSE4034</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:              GSE2866</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:              GSE3253</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:              GSE2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:              GSE2426</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:              GSE2161</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:              GSE2871</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:              GSE2869</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:              GSE2868</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:              GSE2867</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:              GSE3489</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:              GSE1997</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:              GSE1837</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:              GSE2178</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:              GSE2031</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:              GSE1611</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                              experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                       &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                              Human Lung Transplant - BAL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                    d'mel-affy-rat-168311</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                   Expression Studies of Melanotransferrin in Mouse Brain</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                  perro-affy-human-186940</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                  palme-affy-mouse-198967</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                Donarum-3R01NS040270-03S1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: Exaggerated neuroinflammation and sickness behavior in aged mice after activation of the peripheral innate immune system</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                 Hippocampus of HuD overexpressor mice. Perrone-Bizzozero-5R01NS030255-12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                  Pre-Neoplastic Stage of Medulloblastoma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:         Identification of genes that are dysregulated in the telencephalon of Dlx1/2 mutants. Rubenstein-2R01MH049428-11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                      giza-affy-rat-84719</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                Rosenfeld-5R01NS034934-14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                 Tosini-1R01NS043459-01A1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                   Zoghbi-5R01NS027699-14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                   Patterns of gene dysregulation in the frontal cortex of patients with HIV encephalitis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                               Hippocampal development in fetal alcohol exposed rats. Perrone-Bizzozero-5R01NS030255-12-3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                              Comparison of  SNc and VTA dopaminergic neurons. Greene-5P50NS038399-050001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                            Mouse Expression DB 2001 CHMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               Identification of genes and quantitative trait loci (QTL) that control hematopoietic stem cell functioning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                          Transcriptome of Ts1Cje and euploids cerebellum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                              experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:             3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:             5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:            11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:            12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:            13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:            15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:            16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:            17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:            18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:            19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:            21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:            24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:            25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:            26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Bronchoalveolar lavage samples collected from lung transplant recipients.  Numeric portion of sample name is an arbitrary patient ID and AxBx number indicates the perivascular (A) and bronchiolar (B) scores from biopsies collected on the same day as the BAL fluid was collected.  Several patients have more than one sample in this series and can be determined by patient number followed by a lower case letter.  Acute rejection state is determined by the combined A and B score - specifically, a combined AB score of 2 or greater is considered an acute rejection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Melanotransferrin (MTf) or melanoma tumor antigen p97 is an iron (Fe)-binding transferrin homolog expressed highly on melanomas and at lower levels on normal tissues. It has been suggested that MTf is involved in a variety of processes such as Fe metabolism and cellular differentiation. Considering the crucial role of Fe in many metabolic pathways e.g., DNA synthesis, it is important to understand the function of MTf. To define the roles of MTf, a MTf knockout (MTf -/-) mouse model was developed. Examination of the MTf -/- mice demonstrated no phenotypic differences compared to wild-type littermates. However, microarray analysis showed differential expression of molecules involved in proliferation such as Mef2a, Tcf4, Gls and Apod in MTf -/- mice compared to MTf +/+ littermates, suggesting a role for MTf in proliferation and tumorigenesis.\nDate GSE4523 Last Updated: Jun 13 2006\nContributors:  Louise L Dunn Yohan Suryo Rahmanto Eric O Sekyere Des R Richardson\nIncludes GDS1964.\n Update date: Jun 14 2006.\n Dataset description GDS1964: Analysis of brain from melanotransferrin (MTf) knockout mutants. MTf or melanoma tumor antigen p97 is a membrane bound iron binding transferrin homolog highly expressed in melanomas and at lower levels in normal tissues. Results provide insight into the function of MTf.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Our laboratory has developed the first mouse model overexpressing a RNA-binding protein, the ELAV-like protein HuD, in the CNS under the control of the CaMKinII alpha promoter. Initial behavioral characterization of the mice revealed that they had significant learning deficits together with abnormalities in prepulse inhibition (PPI). At the molecular level, we found that the expression of the growth-associated protein GAP-43, one of the targets of HuD, was increased in the hippocampus of HuD transgenic mice. To characterize these mice further and to evaluate the utility of these animals in understanding human diseases, we propose to use DNA microarray methods. To test our hypothesis we propose 3 specific aims: 1) To characterize the pattern of gene expression in the hippocampus of HuD overexpressor mice 2) To compare the pattern of gene expression in our mouse model with that in the hippocampus of rats prenatally exposed to alcohol (FAS model) and 3) To compare the pattern of gene expression in our mouse model with that shown in post-mortem tissues of patients with schizophrenia. In our previous protocols we examined the pattern of gene expression in our HuD transgenic mice and in rats prenaltally exposed to alcohol. A report by another group (Hakak et al, 2001) showed that three of the HuD targets were upregulated in the prefrontal cortex of patients with schizophrenia. To evaluate whether other target of HuD may be affected in this illness, in the current protocol, we want to compare the pattern of expression in our transgenic mice with in tissue from patients with schizophrenia Based on the behavioral and molecular properties of our HuD transgenic mice we hypothesize that these animals may be good models for the studying the basis of learning disabilities and of diseases that show deficits in PPI such as fetal alcohol syndrome and schizophrenia. All 28 samples are derived from cerebellar tissues for patients with schizophrenia and matched controls.  The specimens were obtained from the Maryland Brain Collection according to NIH guidelines for confidentially and privacy. The protocol used in these studies was reviewed by our HRRC which found that our studies do not fall within the category of protocols monitored by the IRB (see attached letter form the HRRC). Specimens from 14 patients with a diagnosis of schizophrenia performed according to DSM-IV criteria and 14 sex-, age- and PMI-matched controls was included in this study. No differences were found between patients and control subjects in the average age (45±12 versus 43±10 years, p=0.86) or PMI (12±5 versus 16±6 hours, p=0.11). We will provide 28 samples containing 5 ug of RNA each in DEPC water (see validation of the quality of the RNA below). In addition, we include in our study animals treated with haloperidol as control for medication. These samples will be submitted in a separate protocol.\nDate GSE4036 Last Updated: Jan 13 2006\nContributors:  Nora I Perrone-Bizzozero\nIncludes GDS1917.\n Update date: Jun 14 2006.\n Dataset description GDS1917: Analysis of cortical samples corresponding to the crus I/VIIa area of the cerebellum from schizophrenia patients. A study indicates that targets of the RNA-binding ELAV-like protein HuD are overexpressed in the prefrontal cortex of patients with schizophrenia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Fear conditioning (FC) is a behavioral paradigm that measures an animal's ability to learn fear related information.  FC is measured by pairing a mild foot-shock with the surroundings in which the shock was recieved.  Upon being placed back in the context, mice exhibit freezing behavior, which is a species-specific response to fear.  We have previously used selective breeding to produce lines of mice with high or low levels of freezing behavior.  This experiment is a replication of a previous experiment that produced lines of mice with high or low levels of freezing behavior.  These lines derive from different progenitor mouse strains.  We are able to identify alleles that govern the genetic variability for FC by using chromosomal markers in these selected lines.  Using microarrays, we will identify differences in gene expression in two key brain regions: amygdala and hippocampus.  Gene expression differences and data regarding chromosomal regions involved in the behavior will be compared to identify particular genes that are both differentially expressed and whose expression is governed by alleles that fall into critical chromosomal regions. We will compare gene expresion in the amygdala and hippocampus (brain regions known to be relevant to fear behavior) from the these two lines of mice and to the those in the previous experiment.  Bayesian statistics will be used in an effort to identify gene expression that affects fear behavior. We hypothesize that selection has acted in part by changing the frequency of alleles that cause differential expression of key genes in the amygdala and hippocampus of our selected lines.  Slective breeding changes the frequency of trait relevand (FC) alleles.  A relevant allele is expected to increas in one selected line and decrease in the oppositely selected line.  Some trait relevant alleles are expected to cause changes in the level of expression at particular genes. Amygdala and hippocampus will be rapidly dissected out of experimentally naïve mice from each line. Naïve mice will be used for expression studies since the behavior of the mice in the FC test can be reliably anticipated due to their lineage. We have practiced these procedures, and can accurately and reproducibly remove these regions in less than 5 minutes. Different mice will be used to collect each brain region, since the dissection of hippocampus disrupts the removal of amygdala. We will collect enough samples from each region to accommodate a total of 6 microarrays per brain region, per line, thus we will use a total of 24 microarrays. We anticipate that a single brain region will be sufficient to for a microarray. However, we propose to utilize three samples per microarray, because this will reduce variability due to environmental factors and due to slight variability in our dissection procedures. Once this tissue is removed, we will isolate RNA for shipment to the Microarray consortium. We will also collect spleens from each subject as a source of genomic DNA, in order to permit direct comparison of genotype and expression phenotypes. Once we have the results of the microarray analysis, we use WebQTL.org to identify the chromosomal locations of alleles that are know to influence the expression of genes for which we have found differential expression. We will then superimpose this information on trait relevant chromosomal regions identified from our selected lines. This will allow us to rapidly identify genes which may account for genetic variability in FC due to differential expression. Such genes will then be subjected to further study.\r\nDate GSE4034 Last Updated: Jan 13 2006\r\n\r\nContributors:  Abraham A Palmer\r\n\r\nIncludes GDS1901.\r\n Update date: Jun 09 2006.\r\n Dataset description GDS1901: Analysis of the amygdalae and hippocampi of strains exhibiting low or high freezing behavior in response to fear.  Previous work identified alleles that contribute to the variability in freezing behavior, and the chromosomal location of these alleles.\r</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Succinate semialdehyde dehydrogenase (SSADH) deficiency is a rare autosomal recessive disorder effecting approximately 350 people around the world. Patients suffering from SSADH deficiency experience language acquisition failure, memory deficiencies, autism, increased aggressive behaviors, and seizures. There is a chemical buildup of both gamma-aminobutyric acid (GABA) and gamma-hydroxybutyric acid (GHB) in the neurological system of these patients. The Aldh5a1-/- knock out mouse model of SSADH deficiency shows the same chemical imbalances as the human disease, with additional fatal tonic-clonic seizures at three weeks of age.  The elucidation of seizure causing pathways will facilitate treatment of seizure phenotypes in diseases with related epilepsy. Gene expression patterns within the hippocampus, cerebellum, and cortex of SSADH deficient mice (Aldh5a1-/- mice) will be compared to wild type mice at a time point immediately prior to fatal seizures. We hypothesis that the SSADH deficient mice experience a dysfunction of glutamate/GABA/ glutamine neurotransmitter cycle linked to astroglial metabolism and/or uptake of neuronally-released glutamate. The increased levels of GHB and GABA lead to down regulation of GABA-B-Receptor leading to seizures. The SSADH deficient phenotype may also be caused by ongoing oxidative damage and the pathological role of succinic semialdehyde. SSADH deficient mice (Aldh5a1-/- knock out) exhibit fatal seizures around three weeks of age. Mutant and wild type mice will be sacrificed between 17 and 19 days of life, and brain sections will be extracted and frozen (using a standard protocol). Hippocampus, cerebellum, and cortex from three mutant mice and three wild type mice will individually be expression profiled on the Affymetrix platform, giving a total of eighteen arrays. Comparative analysis will then be carried out, evaluating the transcript differences between mutant and wild type mice in each brain region.\nDate GSE2866 Last Updated: Jun 08 2006\nContributors:  E A Donarum\nIncludes GDS1745.\n Update date: Jun 08 2006.\n Dataset description GDS1745: Analysis of brain hippocampi, cerebella, and cortices of succinate semialdehyde dehydrogenase (SSADH)-deficient mutants at 3 weeks of age, when fatal seizures occur. Results indicate that SSADH deficiency results in the dysregulation of genes involved in myelin structure and compaction.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Acute cognitive impairment (i.e., delirium) is common in elderly emergency department patients and frequently results from infections that are unrelated to the central nervous system. Since activation of the peripheral innate immune system induces brain microglia to produce inflammatory cytokines that are responsible for behavioral deficits, we investigated if aging exacerbated neuroinflammation and sickness behavior after peripheral injection of lipopolysaccharide (LPS). Microarray analysis revealed a transcriptional profile indicating the presence of primed or activated microglia and increased inflammation in the aged brain. Furthermore, aged mice had a unique gene expression profile in the brain after an intraperitoneal injection of LPS, and the LPS-induced elevation in the brain inflammatory cytokines and oxidative stress was both exaggerated and prolonged compared with adults. Aged mice were anorectic longer and lost more weight than adults after peripheral LPS administration. Moreover, reductions in both locomotor and social behavior remained 24 h later in aged mice, when adults had fully recovered, and the exaggerated neuroinflammatory response in aged mice was not reliably paralleled by increased circulating cytokines in the periphery. Taken together, these data establish that activation of the peripheral innate immune system leads to exacerbated neuroinflammation in the aged as compared with adult mice. This dysregulated link between the peripheral and central innate immune system is likely to be involved in the severe behavioral deficits that frequently occur in older adults with systemic infections.\nDate GSE3253 Last Updated: Oct 28 2005\nContributors:  Johnathan P Godbout Jing Chen Amy F Richwine Brian M Berg Keith K Kelley Rodney W Johnson Jayne Abraham\nIncludes GDS1311.\n Update date: Nov 04 2005.\n Dataset description GDS1311: Analysis of aged brain after activation of the peripheral innate immune system by intraperitoneal injection of E. coli lipopolysaccharide (LPS).  Results provide insight into molecular events underlying severe behavioral deficits (delirium) common in older adults with systemic infections.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Post-transcriptional mechanisms play an important role in the control of gene expression. RNA-binding proteins are key players in the post-transcriptional control of many neural genes and they participate in multiple processes, from RNA splicing and mRNA transport to mRNA stability and translation. Our laboratory has developed the first mouse model overexpressing a RNA-binding protein, the ELAV-like protein HuD, in the CNS under the control of the CaMKinII alpha promoter. Initial behavioral characterization of the mice revealed that they had significant learning deficits together with abnormalities in prepulse inhibition (PPI). At the molecular level, we found that the expression of the growth-associated protein GAP-43, one of the targets of HuD, was increased in the hippocampus of HuD transgenic mice. To characterize these mice further and to evaluate the utility of these animals in understanding human diseases, we propose to use DNA microarray methods. To test our hypothesis we propose 2 specific aims: 1)To characterize the pattern of gene expression in the hippocampus of HuD overexpressor mice 2)To compare the pattern of gene expression in our mouse model with that in the hippocampus of rats prenatally exposed to alcohol (FAS model) and in post-mortem tissues of patients with schizophrenia Based on the behavioral and molecular properties of our HuD transgenic mice we hypothesize that these animals may be good models for the studying the basis of learning disabilities and of diseases that show deficits in PPI such as fetal alcohol syndrome and schizophrenia. All mice are in C57BL/6 background and are male approximately 60 days old. Initial studies were performed in animals that were not subjected to any experimental manipulation. Animals were bred and sacrificed according to our approved animal protocol.  The brain was rapidly dissected on ice and we isolated the hippocampus, which has the highest expression of the transgene. After dissection both hippocampi were frozen in liquid nitrogen, pooled and stored at -80C until analysis. RNA samples were isolated using RNAeasy Qiagen columns. For our first experiment, we want to examine the pattern of gene expression in the hippocampus of 3 transgenic mice and 3 non-transgenic littermates. RNAs from the 6 hippocampi were of high quality as revealed by the integrity of the 28S and 18S rRNA. We will provide 6 samples containing 10 ug of RNA each in DEPC water at a concentration of about 0.5 ug/ul. Three of the samples (#1, #2 and # 3) are from transgenic mice and three from their non-transgenic littermates (#4, #5 and #6).\nDate GSE2005 Last Updated: May 29 2005\nContributors:  Nora I Perrone-Bizzozero\nIncludes GDS1111.\n Update date: May 09 2005.\n Dataset description GDS1111: Analysis of hippocampus of 60 day old C57BL/6 males overexpressing the RNA binding protein HuD. HuD overexpressor mice exhibit learning deficits and abnormalities in prepulse inhibition.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: SUMMARY  Medulloblastoma is the most common malignant brain tumor in children. It is thought to result from transformation of granule cell precursors (GCPs) in the developing cerebellum, but little is known about the early stages of the disease. Here we identify a pre-neoplastic stage of medulloblastoma in patched heterozygous mice, a model of the human disease. We show that pre-neoplastic cells are present in the majority of patched mutants, although only 16% of these mice develop tumors. Pre-neoplastic cells, like tumor cells, exhibit activation of the Sonic hedgehog pathway and constitutive proliferation. Importantly, they also lack expression of the wild-type patched allele, suggesting that loss of patched is an early event in tumorigenesis. While pre-neoplastic cells resemble GCPs and tumor cells in many respects, they have a distinct molecular signature. Genes that mark the pre-neoplastic stage include regulators of migration, apoptosis and differentiation, processes critical for normal development but previously unrecognized for their role in medulloblastoma. The identification and molecular characterization of pre-neoplastic cells provides insight into the early steps in medulloblastoma formation, and may yield important markers for early detection and therapy of this disease.  ANIMALS  Patched heterozygous mice (Goodrich et al., 1997) were obtained from Matthew Scott?s lab at Stanford, and maintained by breeding with 129X1/SvJ mice from Jackson Laboratories. Math1-GFP transgenic mice (Lumpkin et al., 2003) were provided by Jane Johnson at UT Southwestern Medical Center. Math1-GFP/patched+/- mice were generated by crossing patched heterozygotes with Math1-GFP mice, and then backcrossing to Math1-GFP mice three times before further analysis. All mice were maintained in the Cancer Center Isolation Facility at Duke University Medical Center.  ISOLATION OF CELLS  Granule cell precursors (GCPs) were isolated from 7-day-old (P7) patched+/- mice; preneoplastic cells were obtained from 6 week-old patched mutants; and tumor cells were obtained from 10-25-week old patched mutants displaying physical and behavioral signs of medulloblastoma. Cells were isolated from each source using a protocol described in (Wechsler- Reya and Scott, 1999). Briefly, cerebella were digested in solution containing 10 U/ml papain Oliver et al. Pre-Neoplastic Stage of Medulloblastoma - 8 - (Worthington, Lakewood, NJ) and 250 U/ml DNase (Sigma) and triturated to obtain a cell suspension. This suspension was centrifuged through a step gradient of 35% and 65% Percoll (Amersham Biosciences, Piscataway, NJ), and cells were harvested from the 35%-65% interface. Cells were resuspended in serum-free culture medium consisting of Neurobasal containing B27 supplement, sodium pyruvate, L-glutamine, and penicillin/streptomycin (all from Invitrogen, Carlsbad, California) and counted on a hemacytometer. Cells used for RNA isolation were centrifuged and flash frozen in liquid nitrogen. For proliferation assays or immunostaining, cells were plated on poly-D-lysine (PDL)-coated tissue culture vessels and incubated in serum-free culture medium.  MICROARRAY HYBRIDIZATION AND ANALYSIS  RNA from GCPs, pre-neoplastic cells and tumor cells (isolated as described above, but not FACS-sorted) and from normal adult cerebellum (not dissociated) was converted to cDNA using the Superscript Choice cDNA kit (Invitrogen) and a T7-dT(24) primer (Genset/Proligo, Boulder, CO). cRNA was generated using a T7-transcription/labeling kit from Enzo Life Sciences and hybridized to Affymetrix U74Av2 chips (Affymetrix, Santa Clara, CA). Chips were scanned, and hybridization data were acquired using Affymetrix Suite 5.0 software. Affymetrix CEL files were normalized and quantified using Bioconductor software with the gcRMA model to quantify gene expression levels (Gentleman and Carey, 2002). Unsupervised principal components analysis (PCA) was used to identify the relationships among normal adult cerebellum, GCPs, pre-neoplastic cells and tumor cells based on expression profiles. To identify genes that were differentially expressed among GCPs, pre-neoplastic cells and tumor cells, supervised analysis was carried out. A gene-by-gene ANOVA model with three groups (GCP, pre-neoplastic, tumor) was used to fit the log2-transformed intensities. To correct for multiple comparisons, the nominal p-value was adjusted using the false discovery rate (Benjamini and Hochberg, 1995). Genes were considered to be differentially expressed if they satisfied all of the following criteria: a difference in expression greater than 1.9-fold between any two groups, a maximum absolute intensity difference larger than 32 units, and an adjusted pvalue &lt; 0.01. There were 118 genes that met these criteria. The identities of differentially expressed genes were verified by integrating data from the Affymetrix and Unigene databases. Gene functions were determined using information from Gene Ontology, Unigene, LocusLink and PubMed databases. Clustering was performed with Cluster and Treeview (Eisen et al., 1998). All statistical analysis was performed using R-1.7 software (Dalgaard, 2002). Results were visualized with Spotfire 6.0 (Somerville, MA).  CORRESPONDING AUTHOR  Robert J. Wechsler-Reya* Dept. of Pharmacology &amp; Cancer Biology, Duke University Medical Center Box 3813, Durham, NC 27710 Phone: 919-613-8754. Fax: 919-668-3556. Email: rw.reya@duke.edu.\nDate GSE2426 Last Updated: Oct 28 2005\nContributors:  R J Wechsler-Reya Trang T Huynh Simon M Lin Jonathan F Wells Tracy A Read Anriada Mehmeti Jessica D Kessler Robert J Wechsler-Reya Trudy G Oliver\nIncludes GDS1110.\n Update date: May 09 2005.\n Dataset description GDS1110: Analysis of granule cell precursors, pre-neoplastic cells, and tumor cells obtained from 7 day old, 6 week old, and 10 to 25 week old patched heterozygous animals, respectively. Results provide insight into the mechanisms of pathogenesis of medulloblastoma at the early stage.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         The Dlx homeodomain transcription factors are implicated in regulating the function of inhibitory neurons; therefore understanding their functions will provide insights into disorders such as epilepsy, mental retardation, autism and cerebral palsy. Identify genes that are dysregulated in the telencephalon of Dlx1/2 mutants. I am sending separate samples of the Basal Ganglia (BG) and cortex (Ctx). Comparing gene expression in the embryonic basal ganglia and cortex in wild type and dlx1/2 mutant mice will provide information regarding the types of genes that are downstream of Dlx1/2 function. Perform gene array analyses in duplicate or triplicate; compare expression profile of total RNA from wild type and dlx1/2 telencephalons. I am sending separate samples of the Basal Ganglia (BG) and cortex. I want to perform gene expression comparison between: 1) wild type and dlx1/2 basal ganglia (BG) 2) wild typee and dlx1/2 cortex (ctx) Age of all mice has been indicated as 14 days to satisfy system requirements, however all mice are embryonic day 14.5.\nDate GSE2161 Last Updated: May 29 2005\nContributors:  John L Rubenstein\nIncludes GDS1084.\n Update date: May 08 2005.\n Dataset description GDS1084: Expression profiling of cortex and basal ganglia from embryonic telencephalon of homozygous and heterozygous Dlx1/2 mutants. Mutants examined at embryonic day 14.5. Dlx homeobox transcription factors are implicated in regulating the function of inhibitory neurons.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Traumatic brain injury (TBI) induces a complex cascade of molecular and physiological effects.  This study proposes to investigate the gene expression profile in cortex and hippocampus over early time points, following two different injury severities.  These results will complement prior knowledge of both metabolic and neuroplastic changes after TBI, as well as serve as a starting point to investigate additional gene families whose expression is altered after TBI. To characterize the profile of gene expression following a diffuse traumatic brain injury of varying severity in adult rats. Distinct patterns of gene expression following traumatic brain injury will occur in a time- and injury-dependent fashion.  In particular, changes in expression of enzymes involved in energy metabolism and neuroplasticity will be detected. Adult rats will be subjected to mild and severe lateral fluid percussion injury OR sham surgery without injury.  At various post-injury timepoints (0.5, 4 and 24 hours), animals will be sacrificed, brain regions (parietal cortex and hippocampus, ipsilateral and contralateral to injury) will be dissected and RNA isolated.  RNA will be used to synthesize cRNA probes for microarray hybridization.  RNA from 2 matched animals will be pooled onto a single chip (U34A rat, Affymetrix).  Comparisons will be made between sham and injured animals, with brain region, injury severity, and post-injury time point as the experimental variables.\nDate GSE2871 Last Updated: Jul 06 2005\nContributors:  C C Giza\nIncludes GDS1795.\n Update date: Jun 09 2006.\n Dataset description GDS1795: Analysis of cortices and hippocampi of animals 4 and 24 hours after mild or severe traumatic brain injury (TBI). TBI induced by lateral fluid percussion. Gene expression in tissues ipsi- and contralateral to the injury examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Using targeted gene deletion, we have firmly established that the Class IV POU domain transcription factor Brn-3.2 controls a developmental program regulating axon pathfinding in the mouse visual system. We have isolated and identified downstream gene targets of Brn-3.2, using Representational Difference Analysis (RDA), on cDNA populations derived from wildtype and Brn-3.2-/- retina at the appropriate embryonic stage. One of these candidate genes, the Homeodomain Interacting Protein Kinase 2 (HIPK2), is postulated to be a transcriptional coregulator, based on its in vitro interactions with repressor homeodomain proteins of the NK class, as well as other components of repressor complexes. HIPK2 has also been shown to be involved in post-translational modification of two major transcriptional regulators, p53 and CtBP. Expression of a dominant-negative form of HIPK2 in sensory neurons affects the innervation patterns of their target tissues, suggesting an axon pathfinding defect. The aim of this project is to identify targets for the Homeodomain Interacting Protein Kinase 2 (HIPK2), that we isolated as a downstream gene target of the class IV POU domain transcription factor Brn-3.2, and to investigate their function in the Brn-3.2 dependent pathway regulating axon pathfinding. We hypothesize that the genes regulated by HIPK2 will play a critical role in axon pathfinding and that the results of this study will provide novel insights into the molecular mechanisms of axon pathfinding, and possibly neural plasticity and regeneration, and therefore, be of great interest to the field of neurobiology in general. In order to uncover potential biological role(s) of HIPK2 in neural development, and particularly axon pathfinding, we generated transgenic mouse models for in vivo studies. Our preliminary observations in transgenic mice, indicate that a form of the molecule expected to act as a dominant-negative and designed to be expressed in sensory neurons, affects the innervation patterns of their target tissues, suggesting an axon pathfinding defect. We plan to take advantage of this model to identify genes regulated by HIPK2, and which are likely to be involved in axon pathfinding. To this end, we will compare gene expression profiles in sensory neurons isolated wild-type and transgenic mice. In our experimental paradigm, the trigeminal ganglion represents the ideal sensory structure in which to perform such an analysis: 1) High levels of transgene can be expressed in the trigeminal ganglion, as assessed by the expression of LacZ from the bicistronic construct which contains IRES-LacZ downstream of a dominant negative form of HIPK2.  2) The trigeminal ganglion can be dissected at embryonic day 13.5, when the phenotype is apparent, with ease and free of contamination from surrounding tissues.  3) The amounts of RNA that can be isolated from a single ganglion are in the range of 200-300 ng, which should be sufficient for microarray analysis following linear amplification of RNA. One series will be carried out with transgenic embryos and a control series will be carried out with wildtype littermates. Animals will be prepared and sacrificed by a standard protocol. Tissue will be rapidly dissected from E13.5 trigeminal ganglion, frozen in liquid nitrogen, and stored at -80C until RNA is extracted. RNA will be prepared using RNeasy Micro Kit. We will be providing 4 tissue samples for each of the wildtype and transgenic animals to mitigate any expression differences resulting from mouse to mouse variation.\nDate GSE2869 Last Updated: Jul 06 2005\nContributors:  M Rosenfeld\nIncludes GDS1793.\n Update date: Jun 13 2006.\n Dataset description GDS1793: Analysis of the trigeminal ganglia of transgenics expressing a dominant-negative form of homeodomain interacting protein kinase 2 (HIPK2). HIPK2 is a downstream target of Brn-3.2, a class IV POU domain transcription factor that controls a developmental program regulating axon pathfinding.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 It is well known that many genes are expressed in the retina where they show a clear daily pattern of expression.  Although several studies have investigated gene expression in the retina, no study ? so far - has investigated the daily and the circadian pattern of gene expression using microarray. In the present study we propose to investigate the daily and circadian pattern of expression in the retina. To determine gene expression in the retina in Light:Dark cycles and in constant condition (constant dim light) in the rat. We hypothesize that (a) different genes are expressed in different retinal layers and (b) the pattern of expression of the same genes can differ among the different layers. Wistar rats (3 for each time point) will be killed in the middle of the day (ZT6) and in the middle of the night (ZT18). Eyeballs will be collected and the retina will be removed and immediately frozen and then stored at -80 oC. The same experiment will be repeated on animals that have been maintained in constant dim light for 3. Retinas (3 for each time point) will be obtained from animal killed at CT6 (middle subjective day) and CT 18 (middle subjectice night) and stored at ? 80 oC. The samples will be then shipped to NINDS-NIMH array facility for analysis.\nDate GSE2868 Last Updated: Oct 28 2005\nContributors:  G Tosini\nIncludes GDS1759.\n Update date: Jun 16 2006.\n Dataset description GDS1759: Analysis of retinas under a light-dark cycle at Zeitgeber times (ZT) 6 and 18 or in constant dim light at circadian times (CT) 6 and 18. Gene expression in retinas compared to that in pineal glands.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        A number of human neurodegenerative diseases result from the expansion of a glutamine repeat within the disease-causing protein. Spinocerebellar ataxia type1 is one such disease, caused by expansion of a polyglutamine tract in the novel protein ataxin-1. To faithfully model SCA1 in the mouse, we generated  knock-in mice carrying 154 CAG repeats in the mouse Sca1 locus. These mice reproduced many aspects of the human disease. Despite ubiquitous expression of the mutant protein, they developed slowly progressive selective neurodegeneration , which is most distinct in Purkinje cells and spinal cord. Alterations in gene expression have been proposed to be involved in the pathogenesis of polyglutamine disease including SCA1, but the changes of gene expression in an authentic disease model have not been characterized. We believe that knowledge of these genes will give insight into the pathophysiology of SCA1 and may ultimately be relevant to the treatment of SCA1. We will determine gene expression patterns in the cerebellum and forebrain at three different time points. We propose that the genes whose expression is affected  by mutant ataxin-1 expression are effectors for neuronal dysfunction and neuronal degeneration in the knock-in mice. We also hypothesize the difference in the expression levels of these genes accounts for selective vulnerability of neurons. We will examine three time points: 4 weeks, 11 weeks, and 20 weeks of age. We have shown that the mice developed motor incoordination as revealed  by rotating rod test as early as 5 weeks of age but it was not until 10 weeks that they start showing clear  neurological  phenotype such as clasping.  At each point, we will compare the pattern between the mutant and wild-type littermate. Animals will be prepared and sacrificed by a standard procedure. Cerebellum and spinal cord are the most affected areas whereas forebrain shows less neurodegeneration. Tissue will be rapidly dissected from cerebellum and forebrain, frozen in liquid nitrogen, and stored at ?80 C until analysis. Tissues will be sent to the centers (as opposed to RNA). We will be providing 9 tissue samples from three litters for each time point to mitigate any expression differences resulting from subtle differences of procedure or environment where the mice grow.\nDate GSE2867 Last Updated: Jul 06 2005\nContributors:  H Zoghbi\nIncludes GDS1756.\n Update date: May 02 2006.\n Dataset description GDS1756: Analysis of cerebellum and forebrain tissue of knock-in mice carrying 154 CAG repeats in the spinocerebellar ataxia type 1 (SCA1) locus at 4 and 12 weeks of age. Results provide insight into the pathophysiology of the neurodegenerative disease SCA1.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             The neurodegenerative process in HIV encephalitis (HIVE) is associated with extensive damage to the dendritic and synaptic structure that often leads to cognitive impairment. Several mechanisms might be at play, including release of neurotoxins, oxidative stress and decreased activity of neurotrophic factors. Furthermore, HIV-mediated dysregulation of genes involved in neuronal maintenance might play an important role. For this purpose, cRNA was prepared from the brains of 17 AIDS patients for analysis with the Affymetrix Human U95Av2 GeneChip and analyzed with the GeneSpring Expression Analysis Software. Out of 12,625 genes analyzed, 74 were downregulated and 59 were upregulated compared to controls. Initial alternative analysis of RNA was performed by ribonuclease protection assay (RPA). In cases with HIVE, downregulated genes included neuronal molecules involved in synaptic plasticity and transmission (ion channels, synaptogyrin, synapsin II), cell cycle (p35, p39, CDC-L2, CDC42, PAK1) and signaling molecules (PI3K, Ras-Raf-MEK1), transcription factors and cytoskeletal components (MAP-1B, MAP-2, tubulin, adducin-2). Upregulated genes included those involved in neuroimmune (IgG, MHC, ?2microglobulin) and anti-viral responses (interferon-inducible molecules), transcription (STAT1, OLIG2, Pax-6) and signaling modulation (MEK3, EphB1) of the cytoskeleton (myosin, aduccin-3, radixin, dystrobrevin). Taken together, this study suggests that HIV proteins released from infected macrophages might not only induce a neuroinflammatory response, but also may promote neurodegeneration by interfering with neuronal transcription of genes involved in regulating signaling and cytoskeletal molecules important in maintaining synapto-dendritic functioning and integrity.\nDate GSE3489 Last Updated: Mar 07 2006\nContributors:  Eleanor S Roberts Dianne Langford Anthony Adame Edward Rockenstein Leslie Crews Howard S Fox Eliezer Masiliah Ian Everall\nIncludes GDS1726.\n Update date: May 30 2006.\n Dataset description GDS1726: Analysis of brain frontal cortex of HIV-seropositive patients with HIV encephalitis (HIVE). HIVE affects &gt;40% of AIDS patients, promoting neurodegeneration and cognitive impairment. Results suggest HIV-mediated dysregulation of genes involved in neuronal maintenance might play an important role.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Study the effect of fetal alcohol exposure (FAE) on hippocampal development Compare the pattern of gene expression in the hippocampus of FAE and control rats fed either an isocaloric diet or a normal diet, at post-natal day 5 of development. FAE will delay the maturation of the hippocampus Rats were fed one of three diet, a liquid diet with 5% ethanol (FAE group), an isocaloric liquid diet (Isocalorc group) or nomal lab chao (normal group).\nDate GSE1997 Last Updated: Feb 02 2006\nContributors:  Nora I Perrone-Bizzozero\nIncludes GDS1660.\n Update date: Jun 08 2006.\n Dataset description GDS1660: Analysis of hippocampi of 5-day-old animals subjected to fetal alcohol exposure (FAE) with 5% ETOH. FAE affects the development and function of the hippocampus.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     The cardinal clinical features of Parkinson's disease (PD) (rigidity, rest tremor, bradykinesia, and postural instability) result from selective loss of midbrain dopaminergic neurons.  More specifically, dopaminergic neurons in the substantia nigra pars compacta (SNc) are much more susceptible to damage than the adjacent dopaminergic neurons in the ventral tegmental area (VTA).  This dichotomy is not only seen in human Parkinsons disease, but also in many animal models of PD, including administration of the mitochondrial toxin rotenone to rats, which replicates many of the behavioral and neuropathological features of PD.  The factors underlying this selective vulnerability are unknown, but could be related to differences in neuronal circuitry, differences in glial support, or intrinsic differences between the neuronal populations of the two regions.  Elucidation of these factors may lead to a greater understanding of the pathogenesis and treatment of Parkinson's disease. We will determine gene expression profiles of untreated rat SNc and VTA dopaminergic neurons using laser capture microscopy to obtain region-specific neuronal mRNA. There are intrinsic differences in gene expression between dopaminergic neurons in the rat SNc and VTA that result in greater susceptibility of SNc neurons to degeneration in experimental parkinsonism.  These differences may be related to dopamine metabolism, oxidative metabolism and stress, protein aggregation, or other unforseen pathways. We will compare gene expression profiles between SNc and VTA dopaminergic neurons in normal rats.  No treatment or time points will be studied in this experiment.  Animals will be anesthetized, sacrificed by decapitation, and brains frozen on dry ice.  Frozen sections will be collected onto glass microscope slides and rapidly immunostained for tyrosine hydroxylase to identify dopaminergic neurons.  SNc and VTA neurons (approx. 200 per sample) will be isolated using laser capture microscopy.  Total RNA will be extracted and poly-A RNA will be amplified using a modified Eberwine protocol.  aRNA will be sent to the centers for labeling and hybridization to Affymetrix rat U34A arrays.  We have confirmed with the center that our aRNA protocol is compatible with the centers amplification protocols; in fact, it is essentially identical.  We will be providing a two-round amplification product to the center for labeling and hybridization.  We recognize that using RNA after three rounds of amplification  may decrease sensitivity for low copy number transcripts, but favor this approach versus pooling our samples (which are inherently paired) at this point.  We have discussed this point in detail with the center. SNc and VTA samples from eight animals (16 samples total) will be provided to mitigate differences specific to individual animals.  With the assisatnce of the center, paired t-tests will be used to determine differential expression between the two regions.  Permutational t-test analysis and/or Benjamini and Hochberg analysis of expression ratios will be used to protect against multiple comparisons.  Selected differentially expressed genes will be validated on separate tissue samples using quantitative RT-PCR or in situ hybridization.\nDate GSE1837 Last Updated: Feb 02 2006\nContributors:  James G Greene\nIncludes GDS1641.\n Update date: Jun 08 2006.\n Dataset description GDS1641: Analysis of dopaminergic neurons of the substantia nigra pars compacta (SNc) and the ventral tegmental area (VTA) from normal animals. SNc dopaminergic neurons are more susceptible to degeneration in Parkinson's disease than VTA dopaminergic neurons.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Expression profiles of different mouse tissue samples profiles.\nDate GSE2178 Last Updated: May 29 2005\nContributors:  Michael Mucenski Bruce J Aronow Belinda Peach Mitchell Cohen Amy Moseley Susan E Waltz John Maggio Sandra Degen Steve Potter Thomas Doetschman Chris Erwin Joanna A Groden James Lessard Linda Parysek R Hirsch MaryBeth Genter Jianhua Zhang Kathleen Anderson Jonathan D Katz John Dedman Michael Lehman Jorge A Bezerra Michael D Bates Ming Xu Dan Wiginton Jeff A Molkentin\nIncludes GDS1322.\n Update date: Nov 09 2005.\n Dataset description GDS1322: Gene expression profiles across a wide variety of tissue samples. The tissues examined include nervous system, skeletal muscle, male and female reproductive organs, gastrointestinal tract, as well as organs in a developmental context including lung, liver, kidney, and heart.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     We have combined large-scale mRNA expression and gene mapping methods to identify genes and loci that control hematopoietic stem cell (HSC) functioning. mRNA expression levels were measured in purified HSC isolated from a panel of densely genotyped recombinant inbred mouse strains. Quantitative trait loci (QTLs) associated with variation in expression of thousands of transcripts were mapped. Comparison of the physical transcript position with the location of the controlling QTL identified polymorphic cis-acting stem cell genes. In addition, multiple trans-acting control loci were highlighted that modify expression of large numbers of genes. These groups of co-regulated transcripts identify pathways that specify variation in stem cells. We illustrate this concept with the identification of strong candidate genes involved with HSC turnover.  We compared expression QTLs in HSC and brain from the same animals, and document both shared and tissue-specific QTLs. Our data are accessible through WebQTL, a web-based interface that allows custom genetic linkage analysis and identification of co-regulated transcripts.\nDate GSE2031 Last Updated: Jun 29 2005\nContributors:  Bert Dontje Kenneth F Manly Sue Sutton Michael Cooke Rudi Alberts Gerald de Haan Jintao Wang Edo Vellenga Ellen Weersing Elissa Chesler Leonid Bystrykh Andrew I Su Tim Wiltshire Mathew T Pletcher Robert W Williams Ritsert C Jansen Lu Lu\nIncludes GDS1077.\n Update date: Mar 16 2005.\n Dataset description GDS1077: Expression profiling of Lin- Sca-1+ c-kit+ hematopoietic stem cells (HSC) from 22 different BXD recombinant inbred (RI) strains. Each RI strain is homozygous for alleles at about 98% of loci. Results combined with QTL mapping to identify candidate genes for the control of HSC function.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Transcriptome analysis of Ts1Cje (mouse model of Down syndrome) and euploids murine cerebellum during postnatal development.\nDate GSE1611 Last Updated: Mar 27 2006\nContributors:  Randal Moldrich Isabelle Rivals Pierre-Marie Sinet Stylianos E Antonarakis Robert Lyle Geoffroy Golfier Luce Dauphinot Laurence Ettwiller Kiyoko Toyama Charles J Epstein Léon Personnaz Maire-Claude Potier Minh Tran Dang Jean Rossier\nIncludes GDS994.\n Update date: Jan 11 2005.\n Dataset description GDS994: Expression profiling of brain cerebella from Ts1Cje males at postnatal days 0, 15, and 30.  RNA pooled from 3 males at each developmental stage. Results provide insight into the molecular changes contributing to the pathogenesis of Down syndrome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE              GSE2018                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE              GSE2872                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE              GSE4523                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE              GSE4036                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE              GSE4034                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE              GSE2866                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE              GSE3253                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE              GSE2005                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE              GSE2426                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE              GSE2161                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE              GSE2871                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE              GSE2869                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE              GSE2868                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE              GSE2867                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE              GSE3489                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE              GSE1997                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE              GSE1837                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE              GSE2178                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE              GSE2031                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE              GSE1611                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.sampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               &lt;char&gt;                  &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: http://www.ncbi.nlm.nih.gov/geo/                     34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: http://www.ncbi.nlm.nih.gov/geo/                     12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: http://www.ncbi.nlm.nih.gov/geo/                      6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: http://www.ncbi.nlm.nih.gov/geo/                     28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: http://www.ncbi.nlm.nih.gov/geo/                     18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: http://www.ncbi.nlm.nih.gov/geo/                     12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: http://www.ncbi.nlm.nih.gov/geo/                      6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: http://www.ncbi.nlm.nih.gov/geo/                     14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: http://www.ncbi.nlm.nih.gov/geo/                     47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13: http://www.ncbi.nlm.nih.gov/geo/                     16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14: http://www.ncbi.nlm.nih.gov/geo/                     20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: http://www.ncbi.nlm.nih.gov/geo/                     28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: http://www.ncbi.nlm.nih.gov/geo/                      5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: http://www.ncbi.nlm.nih.gov/geo/                     15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: http://www.ncbi.nlm.nih.gov/geo/                    161</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: http://www.ncbi.nlm.nih.gov/geo/                     44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: http://www.ncbi.nlm.nih.gov/geo/                     12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.sampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.lastUpdated         experiment.batchEffect geeq.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     &lt;POSc&gt;                         &lt;char&gt;              &lt;lgcl&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    2024-02-10 08:31:41        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    2024-01-02 18:57:36           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    2024-01-25 08:33:05 PROBLEMATIC_BATCH_INFO_FAILURE               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    2023-12-19 13:29:05           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    2023-12-19 13:24:49                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    2023-12-18 19:37:20                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    2024-01-25 08:33:10 PROBLEMATIC_BATCH_INFO_FAILURE               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    2023-12-17 22:04:32        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    2023-12-18 12:59:41        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    2023-12-18 05:24:05           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    2024-01-25 08:33:12           BATCH_EFFECT_FAILURE               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    2023-12-18 19:41:16           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    2023-12-18 19:39:30           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    2023-12-18 19:37:54           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    2023-12-19 05:05:46                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    2023-12-17 21:44:22           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    2023-12-17 14:25:42        BATCH_CORRECTED_SUCCESS                TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    2023-12-18 06:10:33                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    2023-12-17 23:37:28                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    2023-12-17 03:23:31        BATCH_CORRECTED_SUCCESS                TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.lastUpdated         experiment.batchEffect geeq.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;num&gt;            &lt;num&gt;        &lt;num&gt;       &lt;num&gt;       &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                  1                1            1   0.9976341      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                  1                1            1   0.8552303      0.5000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                  0                0           -1   0.2739433      0.5875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                  1                1            1   0.9971246      0.7500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                 -1                0            1   0.5703423      1.0000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                 -1                0            1   0.4273933      0.6875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                  0                0           -1   0.2661131      0.3750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                  1                1            1   0.8565510      0.6500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                  1                1            1   0.8558619      0.6875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                  1                1            1   0.8557322      0.8375</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                 -1                0            1   0.5688754      0.6875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                  1                1            1   0.8558936      0.5875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  1                1            1   0.7119963      0.5625</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                  1                1            1   0.8469065      0.6250</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                  0                0           -1   0.4098798      0.5000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                  1                1            1   0.7101798      0.1875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                  1                1            1   0.9979230      0.3125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                  0                0           -1  -0.2259327      0.0625</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                 -1                0            1   0.1386041      0.8125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                  1                1            1   0.8565754      0.6875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.name  taxon.scientific taxon.ID taxon.NCBI taxon.database.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;char&gt;            &lt;char&gt;    &lt;int&gt;      &lt;int&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:      human      Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:        rat Rattus norvegicus        3      10116                 rn6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:      human      Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:        rat Rattus norvegicus        3      10116                 rn6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:        rat Rattus norvegicus        3      10116                 rn6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:      human      Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:        rat Rattus norvegicus        3      10116                 rn6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:        rat Rattus norvegicus        3      10116                 rn6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:      mouse      Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.name  taxon.scientific taxon.ID taxon.NCBI taxon.database.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.ID</span>
<span class="r-in"><span><span class="fu">get_datasets</span><span class="op">(</span>taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mouse"</span>, <span class="st">"human"</span><span class="op">)</span>, uris <span class="op">=</span> <span class="st">"http://purl.obolibrary.org/obo/UBERON_0002048"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:              GSE2018</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:              GSE2178</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               GSE476</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               GSE478</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               GSE484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               GSE485</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:              GSE1037</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            GSE1133.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               GSE479</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                GSE97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               GSE446</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:   bhattacharjee-lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:     ramaswamy-cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:       staunton-nci60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:              GSE2361</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               GSE740</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                GSE96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:              GSE4345</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:              GSE6858</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:              GSE4772</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                      experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                               &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                      Human Lung Transplant - BAL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                    Mouse Expression DB 2001 CHMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                         Ozone effect on airways hyperpermability</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                              Alveoli loss during caloric restriction time course</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                      Alveoli septation inhibition and protection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                               Genetic basis of sensitivity to pulmonary fibrosis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                      Lung cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                        tissue-specific pattern of mRNA expression - Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                               Alveolar septation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                  Large-scale analysis of the mouse transcriptome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                          Lungs from ovalbumen sensitized and challenged C3H mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                               bhattacharjee-lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                 ramaswamy-cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                   staunton-nci60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                      Expression Proflies of Human Normal tissues</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                Rosetta_Merck_Splicing_Experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                  Large-scale analysis of the human transcriptome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:   In Vivo Regulation of Human Skeletal Muscle Gene Expression by Thyroid Hormone</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                  Expression data from experimental murine asthma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: Expression data from pre natal and post natal CCAM samples and relative controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                      experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:            24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:            55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:            56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:            57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:            58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:           141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:           166</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:           171</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:           181</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:           197</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:           204</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:           219</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:           225</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:           305</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:           306</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:           398</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:           453</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:           502</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:           532</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Bronchoalveolar lavage samples collected from lung transplant recipients.  Numeric portion of sample name is an arbitrary patient ID and AxBx number indicates the perivascular (A) and bronchiolar (B) scores from biopsies collected on the same day as the BAL fluid was collected.  Several patients have more than one sample in this series and can be determined by patient number followed by a lower case letter.  Acute rejection state is determined by the combined A and B score - specifically, a combined AB score of 2 or greater is considered an acute rejection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Expression profiles of different mouse tissue samples profiles.\nDate GSE2178 Last Updated: May 29 2005\nContributors:  Michael Mucenski Bruce J Aronow Belinda Peach Mitchell Cohen Amy Moseley Susan E Waltz John Maggio Sandra Degen Steve Potter Thomas Doetschman Chris Erwin Joanna A Groden James Lessard Linda Parysek R Hirsch MaryBeth Genter Jianhua Zhang Kathleen Anderson Jonathan D Katz John Dedman Michael Lehman Jorge A Bezerra Michael D Bates Ming Xu Dan Wiginton Jeff A Molkentin\nIncludes GDS1322.\n Update date: Nov 09 2005.\n Dataset description GDS1322: Gene expression profiles across a wide variety of tissue samples. The tissues examined include nervous system, skeletal muscle, male and female reproductive organs, gastrointestinal tract, as well as organs in a developmental context including lung, liver, kidney, and heart.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Ozone is a very common environmental pollutant and has been associated with the exacerbation of cardiopulmonary diseases like asthma. In this study the molecular mechanisms underlying the effect of ozone on airways hyperpermability were investigated. Two strains of mice, HeJ (Tlr4 mutant) and OuJ (wildtype) were exposed continuously to air or 0.3ppm of ozone. Lungs were removed and RNA was collected to generate expression profiles.\nDate GSE476 Last Updated: Jun 15 2005\nContributors:  Stephan Kleeberger\nIncludes GDS239.\n Update date: Mar 22 2004.\n Dataset description GDS239: Ozone is an environmental pollutant associated with  exacerbation of cardiopulmonary diseases like asthma. HeJ (Tlr4 mutant) and OuJ (wild type) exposed continuously to air or 0.3 ppm of ozone, and lungs  examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Pulmonary alveoli are complex architectural units thought to undergo endogenous or pharmacologically induced programs of regeneration and degeneration. To study the molecular mechanism of alveoli loss mice were calorie restricted at different timepoints. Lungs were harvested and processed for RNA extraction.\nDate GSE478 Last Updated: Jul 14 2006\nContributors:  Donald   Massaro Gloria D Massaro\nIncludes GDS241.\n Update date: Jun 27 2003.\n Dataset description GDS241: Temporal study of lung alveoli destruction induced by caloric restriction. Alveoli number and surface area are linked to oxygen consumption, and calorie restriction lowers oxygen consumption.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     It has been shown that dexamethasone (Dex) impairs the normal lung septation that occurs in the early postnatal period. Treatment with retinoic acid (ATRA) abrogates the effects of Dex. To understand the molecular basis for the Dex indiced inhibition of the formation of the alveoli and the ability of ATRA to prevent the inhibition of septation, gene expression was analyzed in 4-day old mice treated with diluent (control), Dex-treated and ATRA+Dex-treated.\nDate GSE484 Last Updated: Aug 24 2005\nContributors:  Linda Clerch\nIncludes GDS250.\n Update date: Jun 27 2003.\n Dataset description GDS250: Analysis of how dexamethasone (DEX) inhibits normal lung septation and alveoli formation, and mechanism by which retinoic acid (ATRA) protects against DEX. Three day old mice treated with ATRA or control. On day 4, half of each group administered DEX.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          The murine pulmonary response is strain specific.  Susceptible strains such as C57BL6/J experience an inflammatory response followed by progressive lung disease.  Resistant strains such as Balb/c do not experience significant degrees of inflammation or fibrosis.  This study aims to determine the genetic basis of sensitivity differences between strains.\nDate GSE485 Last Updated: Jul 27 2006\nContributors:  David   Moller Y W Chen\nIncludes GDS251.\n Update date: Jun 27 2003.\n Dataset description GDS251: Determination of genetic basis of sensitivity to pulmonary fibrosis induced by bleomycin by comparing susceptible (C57BL6/J) and resistant (Balb/c) strains. C57BL6/J experience inflammatory response then progressive lung disease, Balb/c do not.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: Two prognostically significant subtypes of high-grade lung neuroendocrine tumors independent of small-cell and large-cell neuroendocrine carcinomas identified by gene expression profiles.  BACKGROUND: Classification of high-grade neuroendocrine tumors (HGNT) of the lung currently recognises large-cell neuroendocrine carcinoma (LCNEC) and small-cell lung carcinoma (SCLC) as distinct groups. However, a similarity in histology for these two carcinomas and uncertain clinical course have led to suggestions that a single HGNT classification would be more appropriate. Gene expression profiling, which can reproduce histopathological classification, and often defines new subclasses with prognostic significance, can be used to resolve HGNT classification. METHODS: We used cDNA microarrays with 40386 elements to analyze the gene expression profiles of 38 surgically resected samples of lung neuroendocrine tumors and 11 SCLC cell lines. Samples of large-cell carcinoma, adenocarcinoma, and normal lung were also included to give a total of 105 samples analyzed. The data were subjected to filtering to yield informative genes before unsupervised hierarchical clustering that identified relatedness of tumor samples. FINDINGS: Distinct groups for carcinoids, large-cell carcinoma, adenocarcinoma, and normal lung were readily identified. However, we were unable to distinguish LCNEC from SCLC by gene expression profiling. Three independent rounds of unsupervised hierarchical clustering consistently divided SCLC samples into two main groups with LCNEC samples largely integrated with these groups. Furthermore, patients in one of the groups identified by clustering had a significantly better clinical outcome than the other (83% vs 12% survived for 5 years; p=0.0094. None of the highly proliferative SCLC cell lines subsequently analyzed clustered with this good-prognosis group. INTERPRETATION: Our findings show that HGNT of the lung can be classified into two groups independent of SCLC and LCNEC. To this end, we have identified many genes, some of which encode well-characterized markers of cancer that distinguish the HGNT groups. These results have implications for the diagnosis, classification, and treatment of lung neuroendocrine tumors, and provide important insights into their underlying biology.\nDate GSE1037 Last Updated: May 29 2005\nContributors:  Virtanen Carl Ishikawa Yuichi Jones Michael Yuichi Ishikawa Daisuke Ishikawa Carl Virtanen Michael H Jones\nIncludes GDS619.\n Update date: Jul 12 2004.\n Dataset description GDS619: Molecular classification of lung high-grade neuroendocrine tumor (HGNT) groups. Carcinoids, large-cell carcinoma, adenocarcinoma, small-cell lung carcinoma cell lines, and normal lung examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The tissue-specific pattern of mRNA expression can indicate important clues about gene function. High-density oligonucleotide arrays offer the opportunity to examine patterns of gene expression on a genome scale. Toward this end, we have designed custom arrays that interrogate the expression of the vast majority of protein-encoding human and mouse genes and have used them to profile a panel of 79 human and 61 mouse tissues. The resulting data set provides the expression patterns for thousands of predicted genes, as well as known and poorly characterized genes, from mice and humans. We have explored this data set for global trends in gene expression, evaluated commonly used lines of evidence in gene prediction methodologies, and investigated patterns indicative of chromosomal organization of transcription. We describe hundreds of regions of correlated transcription and show that some are subject to both tissue and parental allele-specific expression, suggesting a link between spatial expression and imprinting.\nDate GSE1133.2 Last Updated: \nContributors:  John B Hogenesch John R Walker Gabriel Kreiman Tim Wiltshire David Block Mimi Hayakawa Hilmar Lapp Richard Soden Michael P Cooke Serge Batalov Keith A Ching Andrew I Su Jie Zhang\nFrom GSE1133\nIncludes GDS592.\n Update date: May 13 2004.\n Dataset description GDS592: Gene atlas of mouse protein-encoding transcriptome. Gene expression profiles from 61 physiologically normal tissues. Samples generated from adult (10-12 week old) C57BL/6 mice (4 male, 3 female) by dissection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The Mg-delta strain of fibrillin-1 deficient mice display an early defect in alveolar septation.  This study attempts to identify expression patterns in the mutant mouse lung compared to their wild-type littermates that suggest pathways that are critical for normal septation.  This experiment focuses on postnatal days 1 and 5.  Other time points will be added at a later date.\nDate GSE479 Last Updated: Jul 27 2006\nContributors:  Enid R Neptune\nIncludes GDS243.\n Update date: Jul 27 2006.\n Dataset description GDS243: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema.\nIncludes GDS244.\n Update date: Jul 27 2006.\n Dataset description GDS244: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema.\nIncludes GDS242.\n Update date: Jul 27 2006.\n Dataset description GDS242: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema. [Switched to use MG-U74A/B/C by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       High-throughput gene expression profiling has become an important tool for investigating transcriptional activity in a variety of biological samples. To date, the vast majority of these experiments have focused on specific biological processes and perturbations. Here, we have generated and analyzed gene expression from a set of samples spanning a broad range of biological conditions. Specifically, we profiled gene expression from 91 human and mouse samples across a diverse array of tissues, organs, and cell lines. Because these samples predominantly come from the normal physiological state in the human and mouse, this dataset represents a preliminary, but substantial, description of the normal mammalian transcriptome. We have used this dataset to illustrate methods of mining these data, and to reveal insights into molecular and physiological gene function, mechanisms of transcriptional regulation, disease etiology, and comparative genomics. Finally, to allow the scientific community to use this resource, we have built a free and publicly accessible website (http://expression.gnf.org) that integrates data visualization and curation of current gene annotations.\nDate GSE97 Last Updated: May 29 2005\nContributors:  M P Cooke J R Walker R G Vega K A Ching L M Sapinoso A I Su T Wiltshire A Patapoutian P G Schultz A P Orth A Moqrich Y Hakak J B Hogenesch G M Hampton\nIncludes GDS182.\n Update date: May 13 2004.\n Dataset description GDS182: Gene expression profiles from a diverse array of tissues, organs, and cell lines, from the normal physiological state. Represents a preliminary description of the normal mammalian transcriptome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Profile of gene expression in lungs of ovalbumen sensitized and challenged C3H mice.\nDate GSE446 Last Updated: May 29 2005\nIncludes GDS347.\n Update date: Mar 22 2004.\n Dataset description GDS347: Lungs of ovalbumin sensitized/challenged C3H mice compared with controls. C3H are ovalbumin resistant. Attempt to identify genes involved in antigen-induced airway hyperresponsiveness.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Classification of human lung carcinomas by mRNA expression profiling reveals distinct adenocarcinoma subclasses.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Multiclass cancer diagnosis using tumor gene expression signatures.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Chemosensitivity prediction by transcriptional profiling.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       We performed expression profiling of 36 types of normal human tissues and identified 2,503 tissue-specific genes. We then systematically studied the expression of these genes in cancers by re-analyzing a large collection of published DNA microarray datasets. Our study shows that integration of each gene's breadth of expression (BOE) in normal tissues is important for biological interpretation of the expression profiles of cancers in terms of tumor differentiation, cell lineage and metastasis. Twenty five total RNA specimens were purchased from Clontech (Palo Alto, CA), Ambion (Austin, TX) and Strategene (La Jolla, CA). We tried to cover as many tissue types as possible by using pooled RNA samples. In order to define breadth-of-expression (BOE) accurately at a reasonable cost, we tried to cover as many tissue types as possible by using pooled RNA samples. Each specimen represents a human organ. We used RNA samples pooled from 2 to 84 donors to avoid differences at the individual level.  Detailed sample information and Affymetrix .CEL files are available at http://www.genome.rcast.u-tokyo.ac.jp/normal/  Publication:Ge X et al., Interpreting expression profiles of cancers by genome-wide survey of breadth of expression in normal tissues.  Genomics. 2005 Aug;86(2):127-141. PMID: 15950434\nLast Updated (by provider): May 11 2006\nContributors:  Xijin Ge SanMing Wang Siego Ihara Hiroyuki Aburatani Yutaka Midorikawa Shogo Yamamoto Shuichi Tsutsumi\nIncludes GDS1096.\n Update date: Apr 08 2005.\n Dataset description GDS1096: Expression profiling of 36 types of normal tissue. Each RNA tissue sample pooled from several donors.  Results identify tissue specific genes and provide baselines for interpreting gene expression in cancer.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                This series represents 52 tissues hybridized across 5 different chip patterns. Probes were placed at every exon-exon junction in each transcript.\nLast Updated (by provider): May 29 2005\nContributors:  Christopher D Armour Eric E Schadt Daniel D Shoemaker Patrick M Loerch Philip Garrett-Engele Jason M Johnson Roland Stoughton Ralph Santos Zhengyan Kan John Castle\nIncludes GDS830.\n Update date: Nov 10 2004.\n Dataset description GDS830: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS829.\n Update date: Nov 10 2004.\n Dataset description GDS829: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions.  mRNA from 52 tissues and cell lines examined.  Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS833.\n Update date: Nov 10 2004.\n Dataset description GDS833: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS831.\n Update date: Nov 10 2004.\n Dataset description GDS831: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS832.\n Update date: Nov 10 2004.\n Dataset description GDS832: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants. [Switched to use Rosetta_Merged by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    High-throughput gene expression profiling has become an important tool for investigating transcriptional activity in a variety of biological samples. To date, the vast majority of these experiments have focused on specific biological processes and perturbations. Here, we have generated and analyzed gene expression from a set of samples spanning a broad range of biological conditions. Specifically, we profiled gene expression from 91 human and mouse samples across a diverse array of tissues, organs, and cell lines. Because these samples predominantly come from the normal physiological state in the human and mouse, this dataset represents a preliminary, but substantial, description of the normal mammalian transcriptome. We have used this dataset to illustrate methods of mining these data, and to reveal insights into molecular and physiological gene function, mechanisms of transcriptional regulation, disease etiology, and comparative genomics. Finally, to allow the scientific community to use this resource, we have built a free and publicly accessible website (http://expression.gnf.org) that integrates data visualization and curation of current gene annotations.\nLast Updated (by provider): May 29 2005\nContributors:  A Moqrich R G Vega P G Schultz M P Cooke A I Su T Wiltshire A P Orth K A Ching L M Sapinoso J B Hogenesch J R Walker Y Hakak A Patapoutian G M Hampton\nIncludes GDS181.\n Update date: May 13 2004.\n Dataset description GDS181: Gene expression profiles from a diverse array of tissues, organs, and cell lines, from the normal physiological state. Represents a preliminary description of the normal mammalian transcriptome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  This SuperSeries is composed of the following subset Series: GSE3461: Gene expression in miscellaneous human tissues and cell lines GSE3462: Triiodothyronine Treatment: Effects on vastus lateralis skeletal muscle  Abstract: Thyroid hormones are key regulators of metabolism that modulate transcription via nuclear receptors. Hyperthyroidism is associated with increased metabolic rate, protein breakdown, and weight loss. Although the molecular actions of thyroid hormones have been studied thoroughly, their pleiotropic effects are mediated by complex changes in expression of an unknown number of target genes. Here, we measured patterns of skeletal muscle gene expression in five healthy men treated for 14 days with 75 µg of triiodothyronine, using 24,000 cDNA element microarrays. To analyze the data, we used a new statistical method that identifies significant changes in expression and estimates the false discovery rate. The 381 up-regulated genes were involved in a wide range of cellular functions including transcriptional control, mRNA maturation, protein turnover, signal transduction, cellular trafficking, and energy metabolism. Only two genes were down-regulated. Most of the genes are novel targets of thyroid hormone. Cluster analysis of triiodothyronine-regulated gene expression among 19 different human tissues or cell lines revealed sets of coregulated genes that serve similar biologic functions. These results define molecular signatures that help to understand the physiology and pathophysiology of thyroid hormone action.\nLast Updated (by provider): Dec 12 2006\n [Switched to use SMHu24k by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Experimental asthma was induced in BALB/c mice by sensitization and challenge with the allergen ovalbumin. Control groups received PBS. To investigate the innate immune component of experimental asthma, we also analyzed recombinase activating gene (RAG) deficient mice following exposure to ovalbumin and control PBS\nLast Updated (by provider): Apr 23 2007\nContributors:  Xin Lu Patricia W Finn Vipul V Jain David L Perkins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               This study aims at giving an insight on gene expression in CCAM.\nLast Updated (by provider): May 05 2006\nContributors:  Agnes Paquet Sam Hawgood Amy Wagner</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE              GSE2018                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE              GSE2178                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE               GSE476                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE               GSE478                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE               GSE484                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE               GSE485                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE              GSE1037                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE              GSE1133                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE               GSE479                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                GSE97                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE               GSE446                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE              GSE2361                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE               GSE740                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                GSE96                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE              GSE4345                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE              GSE6858                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE              GSE4772                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.sampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               &lt;char&gt;                  &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: http://www.ncbi.nlm.nih.gov/geo/                     34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: http://www.ncbi.nlm.nih.gov/geo/                    161</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: http://www.ncbi.nlm.nih.gov/geo/                     91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: http://www.ncbi.nlm.nih.gov/geo/                    122</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: http://www.ncbi.nlm.nih.gov/geo/                     90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: http://www.ncbi.nlm.nih.gov/geo/                      5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                             &lt;NA&gt;                    203</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                             &lt;NA&gt;                    280</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                             &lt;NA&gt;                     60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: http://www.ncbi.nlm.nih.gov/geo/                     36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: http://www.ncbi.nlm.nih.gov/geo/                     54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: http://www.ncbi.nlm.nih.gov/geo/                     85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: http://www.ncbi.nlm.nih.gov/geo/                     16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: http://www.ncbi.nlm.nih.gov/geo/                     14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.sampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.lastUpdated         experiment.batchEffect geeq.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     &lt;POSc&gt;                         &lt;char&gt;              &lt;lgcl&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    2024-02-10 08:31:41        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    2023-12-18 06:10:33                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    2023-12-19 23:41:59           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    2023-12-19 23:54:57                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    2023-12-20 01:12:41           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    2023-12-20 01:32:02                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    2024-01-25 08:33:32 PROBLEMATIC_BATCH_INFO_FAILURE               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    2023-12-19 19:15:53        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    2023-12-20 00:06:15        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    2023-12-21 13:05:22        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    2023-12-19 19:22:01                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    2023-09-07 21:43:44                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    2023-09-07 21:47:00                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    2023-09-07 21:49:19                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    2023-12-18 11:53:39                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    2023-12-21 00:01:48                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    2023-12-21 12:41:55                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    2024-01-25 00:30:18                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    2023-12-20 20:39:44        BATCH_CORRECTED_SUCCESS                TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    2023-12-19 23:48:33                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.lastUpdated         experiment.batchEffect geeq.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;num&gt;            &lt;num&gt;        &lt;num&gt;       &lt;num&gt;       &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                  1                1            1  0.99763407      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                  0                0           -1 -0.22593272      0.0625</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                  1                1            1  0.85574622      0.6500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                 -1                0            1  0.42626507      0.8125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                  1                1            1  0.85510163      0.6500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                 -1                0            1  0.42565509      0.5625</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                  0                0           -1 -0.08397851      0.1875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                  1                1           -1  0.84599960      0.2500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                  1                0            1  0.42857143      0.1875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                  1                1           -1  0.55722466      0.3125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                  0                0           -1 -0.03621544     -0.3750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                  0                0           -1  0.39301864      0.2500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  0                0           -1  0.23585518      0.0000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                  0                0           -1  0.24928653      0.1250</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                  1                0            1  0.54414795      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                  0                0           -1 -0.14285714     -0.2500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                 -1                0            1  0.28369987      0.7500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                  0                0           -1 -0.13451104      0.1250</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                  1                1            1  0.85445543      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                  0                0           -1  0.28025816      0.3750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.name taxon.scientific taxon.ID taxon.NCBI taxon.database.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;char&gt;           &lt;char&gt;    &lt;int&gt;      &lt;int&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.name taxon.scientific taxon.ID taxon.NCBI taxon.database.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.ID</span>
<span class="r-in"><span><span class="co"># filter below is equivalent to the call above</span></span></span>
<span class="r-in"><span><span class="fu">get_datasets</span><span class="op">(</span>filter <span class="op">=</span> <span class="st">"taxon.commonName in (mouse,human) and allCharacteristics.valueUri = http://purl.obolibrary.org/obo/UBERON_0002048"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:              GSE2018</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:              GSE2178</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               GSE476</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               GSE478</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               GSE484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               GSE485</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:              GSE1037</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            GSE1133.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               GSE479</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                GSE97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               GSE446</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:   bhattacharjee-lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:     ramaswamy-cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:       staunton-nci60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:              GSE2361</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               GSE740</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                GSE96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:              GSE4345</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:              GSE6858</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:              GSE4772</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                      experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                               &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                      Human Lung Transplant - BAL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                    Mouse Expression DB 2001 CHMC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                         Ozone effect on airways hyperpermability</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                              Alveoli loss during caloric restriction time course</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                      Alveoli septation inhibition and protection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                               Genetic basis of sensitivity to pulmonary fibrosis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                      Lung cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                        tissue-specific pattern of mRNA expression - Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                               Alveolar septation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                  Large-scale analysis of the mouse transcriptome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                          Lungs from ovalbumen sensitized and challenged C3H mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                               bhattacharjee-lung</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                 ramaswamy-cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                   staunton-nci60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                      Expression Proflies of Human Normal tissues</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                Rosetta_Merck_Splicing_Experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                  Large-scale analysis of the human transcriptome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:   In Vivo Regulation of Human Skeletal Muscle Gene Expression by Thyroid Hormone</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                  Expression data from experimental murine asthma</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: Expression data from pre natal and post natal CCAM samples and relative controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                      experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:            24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:            55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:            56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:            57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:            58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:           141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:           166</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:           171</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:           181</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:           197</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:           204</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:           219</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:           225</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:           305</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:           306</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:           398</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:           453</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:           502</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:           532</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Bronchoalveolar lavage samples collected from lung transplant recipients.  Numeric portion of sample name is an arbitrary patient ID and AxBx number indicates the perivascular (A) and bronchiolar (B) scores from biopsies collected on the same day as the BAL fluid was collected.  Several patients have more than one sample in this series and can be determined by patient number followed by a lower case letter.  Acute rejection state is determined by the combined A and B score - specifically, a combined AB score of 2 or greater is considered an acute rejection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Expression profiles of different mouse tissue samples profiles.\nDate GSE2178 Last Updated: May 29 2005\nContributors:  Michael Mucenski Bruce J Aronow Belinda Peach Mitchell Cohen Amy Moseley Susan E Waltz John Maggio Sandra Degen Steve Potter Thomas Doetschman Chris Erwin Joanna A Groden James Lessard Linda Parysek R Hirsch MaryBeth Genter Jianhua Zhang Kathleen Anderson Jonathan D Katz John Dedman Michael Lehman Jorge A Bezerra Michael D Bates Ming Xu Dan Wiginton Jeff A Molkentin\nIncludes GDS1322.\n Update date: Nov 09 2005.\n Dataset description GDS1322: Gene expression profiles across a wide variety of tissue samples. The tissues examined include nervous system, skeletal muscle, male and female reproductive organs, gastrointestinal tract, as well as organs in a developmental context including lung, liver, kidney, and heart.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Ozone is a very common environmental pollutant and has been associated with the exacerbation of cardiopulmonary diseases like asthma. In this study the molecular mechanisms underlying the effect of ozone on airways hyperpermability were investigated. Two strains of mice, HeJ (Tlr4 mutant) and OuJ (wildtype) were exposed continuously to air or 0.3ppm of ozone. Lungs were removed and RNA was collected to generate expression profiles.\nDate GSE476 Last Updated: Jun 15 2005\nContributors:  Stephan Kleeberger\nIncludes GDS239.\n Update date: Mar 22 2004.\n Dataset description GDS239: Ozone is an environmental pollutant associated with  exacerbation of cardiopulmonary diseases like asthma. HeJ (Tlr4 mutant) and OuJ (wild type) exposed continuously to air or 0.3 ppm of ozone, and lungs  examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Pulmonary alveoli are complex architectural units thought to undergo endogenous or pharmacologically induced programs of regeneration and degeneration. To study the molecular mechanism of alveoli loss mice were calorie restricted at different timepoints. Lungs were harvested and processed for RNA extraction.\nDate GSE478 Last Updated: Jul 14 2006\nContributors:  Donald   Massaro Gloria D Massaro\nIncludes GDS241.\n Update date: Jun 27 2003.\n Dataset description GDS241: Temporal study of lung alveoli destruction induced by caloric restriction. Alveoli number and surface area are linked to oxygen consumption, and calorie restriction lowers oxygen consumption.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     It has been shown that dexamethasone (Dex) impairs the normal lung septation that occurs in the early postnatal period. Treatment with retinoic acid (ATRA) abrogates the effects of Dex. To understand the molecular basis for the Dex indiced inhibition of the formation of the alveoli and the ability of ATRA to prevent the inhibition of septation, gene expression was analyzed in 4-day old mice treated with diluent (control), Dex-treated and ATRA+Dex-treated.\nDate GSE484 Last Updated: Aug 24 2005\nContributors:  Linda Clerch\nIncludes GDS250.\n Update date: Jun 27 2003.\n Dataset description GDS250: Analysis of how dexamethasone (DEX) inhibits normal lung septation and alveoli formation, and mechanism by which retinoic acid (ATRA) protects against DEX. Three day old mice treated with ATRA or control. On day 4, half of each group administered DEX.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          The murine pulmonary response is strain specific.  Susceptible strains such as C57BL6/J experience an inflammatory response followed by progressive lung disease.  Resistant strains such as Balb/c do not experience significant degrees of inflammation or fibrosis.  This study aims to determine the genetic basis of sensitivity differences between strains.\nDate GSE485 Last Updated: Jul 27 2006\nContributors:  David   Moller Y W Chen\nIncludes GDS251.\n Update date: Jun 27 2003.\n Dataset description GDS251: Determination of genetic basis of sensitivity to pulmonary fibrosis induced by bleomycin by comparing susceptible (C57BL6/J) and resistant (Balb/c) strains. C57BL6/J experience inflammatory response then progressive lung disease, Balb/c do not.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: Two prognostically significant subtypes of high-grade lung neuroendocrine tumors independent of small-cell and large-cell neuroendocrine carcinomas identified by gene expression profiles.  BACKGROUND: Classification of high-grade neuroendocrine tumors (HGNT) of the lung currently recognises large-cell neuroendocrine carcinoma (LCNEC) and small-cell lung carcinoma (SCLC) as distinct groups. However, a similarity in histology for these two carcinomas and uncertain clinical course have led to suggestions that a single HGNT classification would be more appropriate. Gene expression profiling, which can reproduce histopathological classification, and often defines new subclasses with prognostic significance, can be used to resolve HGNT classification. METHODS: We used cDNA microarrays with 40386 elements to analyze the gene expression profiles of 38 surgically resected samples of lung neuroendocrine tumors and 11 SCLC cell lines. Samples of large-cell carcinoma, adenocarcinoma, and normal lung were also included to give a total of 105 samples analyzed. The data were subjected to filtering to yield informative genes before unsupervised hierarchical clustering that identified relatedness of tumor samples. FINDINGS: Distinct groups for carcinoids, large-cell carcinoma, adenocarcinoma, and normal lung were readily identified. However, we were unable to distinguish LCNEC from SCLC by gene expression profiling. Three independent rounds of unsupervised hierarchical clustering consistently divided SCLC samples into two main groups with LCNEC samples largely integrated with these groups. Furthermore, patients in one of the groups identified by clustering had a significantly better clinical outcome than the other (83% vs 12% survived for 5 years; p=0.0094. None of the highly proliferative SCLC cell lines subsequently analyzed clustered with this good-prognosis group. INTERPRETATION: Our findings show that HGNT of the lung can be classified into two groups independent of SCLC and LCNEC. To this end, we have identified many genes, some of which encode well-characterized markers of cancer that distinguish the HGNT groups. These results have implications for the diagnosis, classification, and treatment of lung neuroendocrine tumors, and provide important insights into their underlying biology.\nDate GSE1037 Last Updated: May 29 2005\nContributors:  Virtanen Carl Ishikawa Yuichi Jones Michael Yuichi Ishikawa Daisuke Ishikawa Carl Virtanen Michael H Jones\nIncludes GDS619.\n Update date: Jul 12 2004.\n Dataset description GDS619: Molecular classification of lung high-grade neuroendocrine tumor (HGNT) groups. Carcinoids, large-cell carcinoma, adenocarcinoma, small-cell lung carcinoma cell lines, and normal lung examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The tissue-specific pattern of mRNA expression can indicate important clues about gene function. High-density oligonucleotide arrays offer the opportunity to examine patterns of gene expression on a genome scale. Toward this end, we have designed custom arrays that interrogate the expression of the vast majority of protein-encoding human and mouse genes and have used them to profile a panel of 79 human and 61 mouse tissues. The resulting data set provides the expression patterns for thousands of predicted genes, as well as known and poorly characterized genes, from mice and humans. We have explored this data set for global trends in gene expression, evaluated commonly used lines of evidence in gene prediction methodologies, and investigated patterns indicative of chromosomal organization of transcription. We describe hundreds of regions of correlated transcription and show that some are subject to both tissue and parental allele-specific expression, suggesting a link between spatial expression and imprinting.\nDate GSE1133.2 Last Updated: \nContributors:  John B Hogenesch John R Walker Gabriel Kreiman Tim Wiltshire David Block Mimi Hayakawa Hilmar Lapp Richard Soden Michael P Cooke Serge Batalov Keith A Ching Andrew I Su Jie Zhang\nFrom GSE1133\nIncludes GDS592.\n Update date: May 13 2004.\n Dataset description GDS592: Gene atlas of mouse protein-encoding transcriptome. Gene expression profiles from 61 physiologically normal tissues. Samples generated from adult (10-12 week old) C57BL/6 mice (4 male, 3 female) by dissection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The Mg-delta strain of fibrillin-1 deficient mice display an early defect in alveolar septation.  This study attempts to identify expression patterns in the mutant mouse lung compared to their wild-type littermates that suggest pathways that are critical for normal septation.  This experiment focuses on postnatal days 1 and 5.  Other time points will be added at a later date.\nDate GSE479 Last Updated: Jul 27 2006\nContributors:  Enid R Neptune\nIncludes GDS243.\n Update date: Jul 27 2006.\n Dataset description GDS243: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema.\nIncludes GDS244.\n Update date: Jul 27 2006.\n Dataset description GDS244: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema.\nIncludes GDS242.\n Update date: Jul 27 2006.\n Dataset description GDS242: Study of alveolar septation using the Mg-delta strain of fibrillin-1 deficient mice which display an early defect in alveolar septation, and serve as a model for Marfan syndrome and emphysema. [Switched to use MG-U74A/B/C by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       High-throughput gene expression profiling has become an important tool for investigating transcriptional activity in a variety of biological samples. To date, the vast majority of these experiments have focused on specific biological processes and perturbations. Here, we have generated and analyzed gene expression from a set of samples spanning a broad range of biological conditions. Specifically, we profiled gene expression from 91 human and mouse samples across a diverse array of tissues, organs, and cell lines. Because these samples predominantly come from the normal physiological state in the human and mouse, this dataset represents a preliminary, but substantial, description of the normal mammalian transcriptome. We have used this dataset to illustrate methods of mining these data, and to reveal insights into molecular and physiological gene function, mechanisms of transcriptional regulation, disease etiology, and comparative genomics. Finally, to allow the scientific community to use this resource, we have built a free and publicly accessible website (http://expression.gnf.org) that integrates data visualization and curation of current gene annotations.\nDate GSE97 Last Updated: May 29 2005\nContributors:  M P Cooke J R Walker R G Vega K A Ching L M Sapinoso A I Su T Wiltshire A Patapoutian P G Schultz A P Orth A Moqrich Y Hakak J B Hogenesch G M Hampton\nIncludes GDS182.\n Update date: May 13 2004.\n Dataset description GDS182: Gene expression profiles from a diverse array of tissues, organs, and cell lines, from the normal physiological state. Represents a preliminary description of the normal mammalian transcriptome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Profile of gene expression in lungs of ovalbumen sensitized and challenged C3H mice.\nDate GSE446 Last Updated: May 29 2005\nIncludes GDS347.\n Update date: Mar 22 2004.\n Dataset description GDS347: Lungs of ovalbumin sensitized/challenged C3H mice compared with controls. C3H are ovalbumin resistant. Attempt to identify genes involved in antigen-induced airway hyperresponsiveness.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Classification of human lung carcinomas by mRNA expression profiling reveals distinct adenocarcinoma subclasses.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Multiclass cancer diagnosis using tumor gene expression signatures.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Chemosensitivity prediction by transcriptional profiling.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       We performed expression profiling of 36 types of normal human tissues and identified 2,503 tissue-specific genes. We then systematically studied the expression of these genes in cancers by re-analyzing a large collection of published DNA microarray datasets. Our study shows that integration of each gene's breadth of expression (BOE) in normal tissues is important for biological interpretation of the expression profiles of cancers in terms of tumor differentiation, cell lineage and metastasis. Twenty five total RNA specimens were purchased from Clontech (Palo Alto, CA), Ambion (Austin, TX) and Strategene (La Jolla, CA). We tried to cover as many tissue types as possible by using pooled RNA samples. In order to define breadth-of-expression (BOE) accurately at a reasonable cost, we tried to cover as many tissue types as possible by using pooled RNA samples. Each specimen represents a human organ. We used RNA samples pooled from 2 to 84 donors to avoid differences at the individual level.  Detailed sample information and Affymetrix .CEL files are available at http://www.genome.rcast.u-tokyo.ac.jp/normal/  Publication:Ge X et al., Interpreting expression profiles of cancers by genome-wide survey of breadth of expression in normal tissues.  Genomics. 2005 Aug;86(2):127-141. PMID: 15950434\nLast Updated (by provider): May 11 2006\nContributors:  Xijin Ge SanMing Wang Siego Ihara Hiroyuki Aburatani Yutaka Midorikawa Shogo Yamamoto Shuichi Tsutsumi\nIncludes GDS1096.\n Update date: Apr 08 2005.\n Dataset description GDS1096: Expression profiling of 36 types of normal tissue. Each RNA tissue sample pooled from several donors.  Results identify tissue specific genes and provide baselines for interpreting gene expression in cancer.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                This series represents 52 tissues hybridized across 5 different chip patterns. Probes were placed at every exon-exon junction in each transcript.\nLast Updated (by provider): May 29 2005\nContributors:  Christopher D Armour Eric E Schadt Daniel D Shoemaker Patrick M Loerch Philip Garrett-Engele Jason M Johnson Roland Stoughton Ralph Santos Zhengyan Kan John Castle\nIncludes GDS830.\n Update date: Nov 10 2004.\n Dataset description GDS830: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS829.\n Update date: Nov 10 2004.\n Dataset description GDS829: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions.  mRNA from 52 tissues and cell lines examined.  Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS833.\n Update date: Nov 10 2004.\n Dataset description GDS833: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS831.\n Update date: Nov 10 2004.\n Dataset description GDS831: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants.\nIncludes GDS832.\n Update date: Nov 10 2004.\n Dataset description GDS832: Monitoring of mRNA splice variants for more than 10,000 multi-exon genes using arrays with oligonucleotide probes positioned at exon-exon junctions. mRNA from 52 tissues and cell lines examined. Results provide tissue distribution of splice variants and identify novel splice variants. [Switched to use Rosetta_Merged by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    High-throughput gene expression profiling has become an important tool for investigating transcriptional activity in a variety of biological samples. To date, the vast majority of these experiments have focused on specific biological processes and perturbations. Here, we have generated and analyzed gene expression from a set of samples spanning a broad range of biological conditions. Specifically, we profiled gene expression from 91 human and mouse samples across a diverse array of tissues, organs, and cell lines. Because these samples predominantly come from the normal physiological state in the human and mouse, this dataset represents a preliminary, but substantial, description of the normal mammalian transcriptome. We have used this dataset to illustrate methods of mining these data, and to reveal insights into molecular and physiological gene function, mechanisms of transcriptional regulation, disease etiology, and comparative genomics. Finally, to allow the scientific community to use this resource, we have built a free and publicly accessible website (http://expression.gnf.org) that integrates data visualization and curation of current gene annotations.\nLast Updated (by provider): May 29 2005\nContributors:  A Moqrich R G Vega P G Schultz M P Cooke A I Su T Wiltshire A P Orth K A Ching L M Sapinoso J B Hogenesch J R Walker Y Hakak A Patapoutian G M Hampton\nIncludes GDS181.\n Update date: May 13 2004.\n Dataset description GDS181: Gene expression profiles from a diverse array of tissues, organs, and cell lines, from the normal physiological state. Represents a preliminary description of the normal mammalian transcriptome.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  This SuperSeries is composed of the following subset Series: GSE3461: Gene expression in miscellaneous human tissues and cell lines GSE3462: Triiodothyronine Treatment: Effects on vastus lateralis skeletal muscle  Abstract: Thyroid hormones are key regulators of metabolism that modulate transcription via nuclear receptors. Hyperthyroidism is associated with increased metabolic rate, protein breakdown, and weight loss. Although the molecular actions of thyroid hormones have been studied thoroughly, their pleiotropic effects are mediated by complex changes in expression of an unknown number of target genes. Here, we measured patterns of skeletal muscle gene expression in five healthy men treated for 14 days with 75 µg of triiodothyronine, using 24,000 cDNA element microarrays. To analyze the data, we used a new statistical method that identifies significant changes in expression and estimates the false discovery rate. The 381 up-regulated genes were involved in a wide range of cellular functions including transcriptional control, mRNA maturation, protein turnover, signal transduction, cellular trafficking, and energy metabolism. Only two genes were down-regulated. Most of the genes are novel targets of thyroid hormone. Cluster analysis of triiodothyronine-regulated gene expression among 19 different human tissues or cell lines revealed sets of coregulated genes that serve similar biologic functions. These results define molecular signatures that help to understand the physiology and pathophysiology of thyroid hormone action.\nLast Updated (by provider): Dec 12 2006\n [Switched to use SMHu24k by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Experimental asthma was induced in BALB/c mice by sensitization and challenge with the allergen ovalbumin. Control groups received PBS. To investigate the innate immune component of experimental asthma, we also analyzed recombinase activating gene (RAG) deficient mice following exposure to ovalbumin and control PBS\nLast Updated (by provider): Apr 23 2007\nContributors:  Xin Lu Patricia W Finn Vipul V Jain David L Perkins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               This study aims at giving an insight on gene expression in CCAM.\nLast Updated (by provider): May 05 2006\nContributors:  Agnes Paquet Sam Hawgood Amy Wagner</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE              GSE2018                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE              GSE2178                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE               GSE476                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE               GSE478                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE               GSE484                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE               GSE485                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE              GSE1037                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE              GSE1133                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE               GSE479                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                GSE97                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE               GSE446                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE              GSE2361                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE               GSE740                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                GSE96                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE              GSE4345                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE              GSE6858                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE              GSE4772                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.sampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               &lt;char&gt;                  &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: http://www.ncbi.nlm.nih.gov/geo/                     34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: http://www.ncbi.nlm.nih.gov/geo/                    161</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: http://www.ncbi.nlm.nih.gov/geo/                     91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: http://www.ncbi.nlm.nih.gov/geo/                    122</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: http://www.ncbi.nlm.nih.gov/geo/                     90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: http://www.ncbi.nlm.nih.gov/geo/                      5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                             &lt;NA&gt;                    203</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                             &lt;NA&gt;                    280</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                             &lt;NA&gt;                     60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: http://www.ncbi.nlm.nih.gov/geo/                     36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: http://www.ncbi.nlm.nih.gov/geo/                     54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: http://www.ncbi.nlm.nih.gov/geo/                     85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: http://www.ncbi.nlm.nih.gov/geo/                     16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: http://www.ncbi.nlm.nih.gov/geo/                     14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.sampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.lastUpdated         experiment.batchEffect geeq.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     &lt;POSc&gt;                         &lt;char&gt;              &lt;lgcl&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    2024-02-10 08:31:41        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    2023-12-18 06:10:33                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    2023-12-19 23:41:59           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    2023-12-19 23:54:57                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    2023-12-20 01:12:41           SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    2023-12-20 01:32:02                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    2024-01-25 08:33:32 PROBLEMATIC_BATCH_INFO_FAILURE               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    2023-12-19 19:15:53        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    2023-12-20 00:06:15        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    2023-12-21 13:05:22        NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    2023-12-19 19:22:01                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    2023-09-07 21:43:44                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    2023-09-07 21:47:00                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    2023-09-07 21:49:19                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    2023-12-18 11:53:39                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    2023-12-21 00:01:48                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    2023-12-21 12:41:55                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    2024-01-25 00:30:18                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    2023-12-20 20:39:44        BATCH_CORRECTED_SUCCESS                TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    2023-12-19 23:48:33                  NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.lastUpdated         experiment.batchEffect geeq.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;num&gt;            &lt;num&gt;        &lt;num&gt;       &lt;num&gt;       &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                  1                1            1  0.99763407      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                  0                0           -1 -0.22593272      0.0625</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                  1                1            1  0.85574622      0.6500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                 -1                0            1  0.42626507      0.8125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                  1                1            1  0.85510163      0.6500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                 -1                0            1  0.42565509      0.5625</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                  0                0           -1 -0.08397851      0.1875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                  1                1           -1  0.84599960      0.2500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                  1                0            1  0.42857143      0.1875</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                  1                1           -1  0.55722466      0.3125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                  0                0           -1 -0.03621544     -0.3750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                  0                0           -1  0.39301864      0.2500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  0                0           -1  0.23585518      0.0000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                  0                0           -1  0.24928653      0.1250</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                  1                0            1  0.54414795      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                  0                0           -1 -0.14285714     -0.2500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                 -1                0            1  0.28369987      0.7500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                  0                0           -1 -0.13451104      0.1250</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                  1                1            1  0.85445543      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                  0                0           -1  0.28025816      0.3750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.name taxon.scientific taxon.ID taxon.NCBI taxon.database.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;char&gt;           &lt;char&gt;    &lt;int&gt;      &lt;int&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.name taxon.scientific taxon.ID taxon.NCBI taxon.database.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.ID</span>
<span class="r-in"><span><span class="fu">get_datasets</span><span class="op">(</span>query <span class="op">=</span> <span class="st">"lung"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             GSE12205</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             GSE21059</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:             GSE27335</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             GSE27964</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:             GSE31625</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             GSE38530</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             GSE48397</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:             GSE40839</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:             GSE37703</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:             GSE12236</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:             GSE30242</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:             GSE47542</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:            GSE9077.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:            GSE9077.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:            GSE9077.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:             GSE83896</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:             GSE10006</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:             GSE26864</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:             GSE30079</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:             GSE29006</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                      experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                               &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                     Effect of PTP inhibition on changes in gene expression after Cr(VI) exposure</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                            IMR90 radiation bystander time-course experiment 0.5Gy alpha particle</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                               Genomic differences distinguish the myofibroblast phenotype of distal lung from airway fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                    Age-dependent decline in lung regeneration with loss of clonogenicity and myofibroblastic differentiation of lung fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: Gene Expression Patterns that Predict Sensitivity to Epidermal Growth Factor Receptor Tyrosine Kinase Inhibitors in Lung Cancer Cell Lines and Human Lung Tumors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                Gene expression data in over expressed Human Fetal Lung Fibroblasts with IPF downregulated miRNAs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                        Expression data from (mouse) normal lung fibroblasts and carcinoma-associated fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                             Expression data from fibroblasts cultured from normal and fibrotic human lung tissue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                        Differential analysis of HOXA1 in adult cells at isoform resolution by RNA-Seq [Illumina]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                          Whole Genome Exon Arrays Identify Differential Expression of Alternatively Spliced, Cancer-related Genes in Lung Cancer</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                               Time course effect of the glucocorticoid mometasone furoate on the proliferation of primary human lung fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                IMR90 lung fibroblasts treated with IL-4 or IL-13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                       Expression profiles of immortal lung fibroblasts - GPL1928</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                       Expression profiles of immortal lung fibroblasts - GPL4133</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                        Expression profiles of immortal lung fibroblasts - GPL570</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                           Small molecules increase direct neural conversion of human fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                            Decreased Expression of Intelectin 1 in The Human Airway Epithelium of Smokers Compared to Nonsmokers</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                           Gene expression in mouse lung epithelial cells treated with synthetic TLR ligands to induce resistance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                  Genetic toxicology and toxicogenomic analysis of 3 cigarette smoke condensates in vitro reveals few differences between high and low tar brands</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                        mRNA-seq of Human Airway Epithelial Cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                      experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:           970</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          3055</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:          3315</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:          3332</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:          5344</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:          6497</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:          6686</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:          6801</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:          7070</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:          7208</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:          7437</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:          8584</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:          9267</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:          9268</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:          9269</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:         16019</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:          1149</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:          3300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:          5296</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:          6246</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Although the consequences of genotoxic injury include cell cycle arrest and apoptosis, cell survival responses after genotoxic injury can produce intrinsic death-resistance and contribute to the development of a transformed phenotype. Protein tyrosine phosphatases (PTPs) are integral components of key survival pathways, and are responsible for their inactivation, while PTP inhibition is are often associated with enhanced cell proliferation. Our aim was to elucidate signaling events that modulate cell survival after genotoxin exposure. Diploid human lung fibroblasts (HLF) were treated with Cr(VI) (as Na2CrO4), a well known human respiratory carcinogen that induces a wide spectrum of DNA damage, in the presence and absence of a broad-range PTP inhibitor, sodium orthovanadate. Notably, PTP inhibition abrogated Cr(VI)-induced clonogenic lethality. The enhanced survival of Cr(VI)-exposed cells after PTP inhibition was predominantly due to a bypass of cell cycle arrest and was not due to decreased Cr uptake as evidenced by unchanged Cr-DNA adduct burden. Additionally, the bypass of Cr-induced growth arrest by PTP inhibition, was accompanied by a decrease in Cr(VI)-induced expression of cell cycle inhibiting genes, and an increase in the Cr(VI)-induced expression of cell cycle promoting genes. Importantly, PTP inhibition resulted in an increase in forward mutations at the HPRT locus, supporting the hypothesis that PTP inhibition in the presence of DNA damage may lead to genomic instability, via bypass of cell cycle checkpoints.\nLast Updated (by provider): Dec 02 2008\nContributors:  Dongsoon Bae Travis J O'Brien Tura C Camilli Susan Ceryak Gina Chun Kristen Wright Steven R Patierno Madhu Lal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: The radiation bystander effect is an important component of the overall biological response of tissues and organisms to ionizing radiation. Little is known about the contribution of genome level changes in neighboring bystander cells to tissue and organ stress after irradiation. The timing of these changes is critical in the physiological context and these questions can only be answered by studying signaling and global transcriptomics in a chronological way. Here, we present a strategy to identify different biologically important signaling modules that act in concert in the radiation and bystander responses. We used time series gene expression analysis of normal human fibroblast cells measured at 0.5 hour, 1 hour, 2 hours, 4 hours, 6 hours and 24 hours after exposure to radiation coupled with a novel clustering method targeted to short time series, Feature Based Partitioning around medoids Algorithm (FBPA), to look for genes that were potentially co-regulated. This method uses biologically meaningful features of the expression profile and dimension augmentation to address the analysis of sparse data sets such as ours. We applied FBPA and Short Time series Expression Miner (STEM) to the same datasets and present the results of our comparisons using computational metrics as well as biological enrichment.  Enrichment showed that gene expression in irradiated cells fell into broad categories of signal transduction, cell cycle/cell death and inflammation/immunity; but only FBPA clustered functions well. In bystander cells, the gene expression response was also broadly categorized into functions associated with cell communication and motility, signal transduction and inflammation; but neither STEM nor FBPA separated biological functions as well as in irradiated samples. Network analysis revealed that p53 and NF-kappaB were central players in gene expression in both irradiated and bystander gene clusters. Analysis of individual clusters also suggested new regulators of gene expression in the radiation and bystander response that may act at the epigenetic level such as histone deacetylases (HDAC1 and HDAC2) and methylases (KDM5B) that can act as strong transcription repressors.  Based on these results, we propose a novel time series clustering method, FBPA, as a powerful approach that can be applied to sparse data sets (including genomic profiling data), where the choice of features selected for clustering and stringent statistical outcome analysis can augment our knowledge of the underlying cellular mechanisms in biological processes.\nLast Updated (by provider): Aug 02 2011\nContributors:  Shanaz A Ghandhi Sally A Amundson</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     ABSTRACT  Primary human distal lung/parenchymal fibroblasts (DLF) exhibit a different phenotype from airway fibroblasts (AF), including the expression of high levels of a-smooth muscle actin (a-SMA). The scope of the differences and the mechanisms driving them are unknown. To determine whether distinct fibroblast characteristics and function based on lung region are predicted by a broad range of genomic differences in AF vs DLF. Matched human fibroblast pairs isolated from proximal and distal lung in 18 asthmatic and 4 normal subjects were studied. Microarray analysis was performed on 12 matched fibroblast pairs (8 asthmatic and 4 normal subjects) and validated by quantitative real-time PCR (qRT-PCR). The functional impact of these molecular differences on AF and DLF was then revealed using computational approaches. Microarray data demonstrated 474 transcripts upregulated in AF, and 611 transcripts upregulated in DLF, when the asthmatic and normal fibroblasts were combined for all the analysis. Further gene ontology (GO) and network analysis identified distinct pathway activation patterns between AF and DLF, including identification of the SMAD3 and MAPK8 signaling pathways. These results demonstrated that marked molecular and functional differences exist between these two lung regional fibroblast populations. These striking differences identify multiple potential mechanisms by which AF and DLF differ in their responses to injury, regeneration and remodeling in the lungs.\nLast Updated (by provider): Jul 22 2011\nContributors:  Jadaranka Milosevic Kazuhisa Konishi Wei Wu Xiuxia Zhou Sally E Wenzel Haizhen Hu Naftali Kaminski</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              While aging leads to a reduction in the capacity for regeneration after pneumonectomy (PNX) in most mammals, this biological phenomenon has not been characterized over the lifetime of mice. We measured the age-specific (3, 9, 24 month) effects of PNX on physiology, morphometry, cell proliferation and apoptosis, global gene expression, and lung fibroblast phenotype and clonogenicity in female C57BL6 mice. The data show that only 3 month old mice were fully capable of restoring lung volumes by day 7 and total alveolar surface area by 21 days. By 9 months, the rate of regeneration was slower (with incomplete regeneration by 21 days), and by 24 months there was no regrowth 21 days post-PNX. The early decline in regeneration rate was not associated with changes in alveolar epithelial cell type II (AECII) proliferation or apoptosis rate. However, significant apoptosis and lack of cell proliferation was evident after PNX in both total cells and AECII cells in 24 mo mice. Analysis of gene expression at several time points (1, 3 and 7 days) post-PNX in 9 versus 3 month mice was consistent with a myofibroblast signature (increased Tnc, Lox1, Col3A1, Eln and Tnfrsf12a) and more alpha smooth muscle actin (?SMA) positive myofibroblasts were present after PNX in 9 month than 3 month mice. Isolated lung fibroblasts showed a significant age-dependent loss of clonogenicity. Moreover, lung fibroblasts isolated from 9 and 17 month mice exhibited higher ?SMA, Col3A1, Fn1 and S100A expression, and lower expression of the survival gene Mdk consistent with terminal differentiation. These data show that concomitant loss of clonogenicity and progressive myofibroblastic differentiation contributes to the age-dependent decline in the rate of lung regeneration.\nLast Updated: Aug 30 2011\nContributors:  Paxson JA, Gruntman A, Parkin CD, Mazan MR, Davis A, Ingenito EP, Hoffman AM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Global gene expression data were generated from cultured non small cell lung cancer cell lines (NSCLC), normalized using MAS 5.0, filtered and used to predict response of cells to EGFR inhibition\nLast Updated (by provider): May 16 2012\nContributors:  Justin M Balko Esther P Black</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Idiopathic pulmonary fibrosis (IPF) is an untreatable fibrotic lung disease characterized by fibroblast proliferation and epithelial mesenchymal transition.miRNA let-7d and mir30b were found to be signifcantly down regulated in IPF. Compared to control we over expressed these miRNAs in Human Fetal Lung Fibroblast cell line\nLast Updated (by provider): Jun 06 2013\nContributors:  Luai Huleihel Naftali Kaminski</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Cancer-associated fibroblasts (CAFs) have been reported to support tumor progression by a variety of mechanisms. However, their role in the progression of non-small cell lung cancer (NSCLC) remains poorly defined. In addition, the extent to which specific proteins secreted by CAFs contribute directly to tumor growth is unclear. To study the role of CAFs in NSCLC, a cross-species functional characterization of mouse and human lung CAFs was performed, including gene expression analysis comparing normal mouse lung fibroblasts (NFs) and mouse lung CAFs to seek for differentially-expressed secreted proteins. Gene expression microarrays were used to identify transcriptomic changes between NFs and CAFs that may contribute to their different tumor-enhancing capacity.\nLast Updated (by provider): Jul 10 2013\nContributors:  Silvestre Vicent Alejandro Sweet-Cordero</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Pulmonary fibrosis develops as a consequence of environmentally induced lung injury and/or an inherent disease susceptibility causing fibroblast activation, proliferation and extracellular matrix deposition. The study was undertaken to characterise global gene expression in pulmonary fibroblasts to better understand the mechanisms underlying the fibrotic fibroblast phenotype.\nLast Updated (by provider): Aug 05 2013\nContributors:  Xu Shi-Wen David J Abraham Carmel J Stock Michael E Dusmet Gisela E Lindahl George Bou-Gharios Elisabetta A Renzoni Christopher P Denton Andrew G Nicholson Athol U Wells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          High-throughput RNA sequencing (RNA-Seq) has enabled accurate gene discovery and expression estimation, but robust differential analysis of gene and transcript abundances has proven difficult. We present a new algorithm, implemented in the freely available tool Cuffdiff, which integrates transcript-level expression estimation with a method to control for variability across replicate samples. Cuffdiff robustly identifies differentially regulated isoforms and genes and reveals differential splicing or promoter-preference changes. We demonstrate the accuracy of our approach through differential analysis of lung fibroblasts in response to loss of the developmental transcription factor HOXA1 and uncover a critical role for this gene in the maintenance of adult cells. We show that HOXA1 is required for lung fibroblast and HeLa cell cycle progression, and loss of HOXA1 results in significant expression level changes in thousands of individual transcripts, along with isoform switching events in key regulators of the cell cycle.\nLast Updated (by provider): Sep 30 2013\nContributors:  Cole Trapnell John Rinn Martin Sauvageau Lior Pachter David Hendrickson\n [Switched to use human_illumina_rnaseq by Gemma]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Alternative processing of pre-mRNA transcripts is a major source of protein diversity in eukaryotes and has been implicated in several disease processes including cancer. In this study we have performed a genome wide analysis of alternative splicing events with the GeneChip Human Exon 1.0 ST Array from Affymetrix in lung adenocarcinoma. We found that ~13.3% of the 17800 core Refseq genes appear to have alternative transcripts that are differentially expressed in lung adenocarcinoma versus normal. According to their known functions the largest subset of these genes (30.8%) is believed to be cancer related. Detailed analysis was performed for several genes using PCR, quantitative RT-PCR and DNA sequencing. We found overexpression of ERG variant 2 but not variant 1 in lung tumors and overexpression of CEACAM1 variant 1 but not variant 2 in lung tumors but not in breast or colon tumors. We also identified a novel, overexpressed variant of CDH3 and verified the overexpression of a novel variant of P16. These findings demonstrate how analysis of alternative pre-mRNA processing can shed additional light on differences between tumors and normal tissues as well as between different tumor types. Such studies may lead to the development of additional tools for tumor diagnosis, prognosis and therapy. \nLast Updated (by provider): May 07 2013\nContributors:  Rodney J Landreneau Liqiang Xi Tony E Godfrey James D Luketich Andrew Feber Maoxin Wu Vanita Gupta Virginia R Litle Andrew D Bergemann Arjun Pennathur</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          The effect of mometasone furoate on the proliferation of 2 patients of human lung fibroblasts was investigated at 8 different time points\nLast Updated (by provider): Aug 29 2013\nContributors:  Florent Baty Martin Brutsche</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       IL-4 and IL-13 are cytokines involved in type 2 immune responses involved in atopic asthma and other diseases. They have a partially overlapping set of cognate receptors whose roles are incompletely understood. Here we explore the effects of these cytokines on lung fibroblasts in order to assess the global similarities and contrasts in the genes whose expression their ligation modulates.\nLast Updated (by provider): May 07 2014\nContributors:  Joseph R Arron Sanjay Chandriani Alexander R Abbas</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Activation of telomerase often endows cancer cells, but rarely normal somatic cells, with immortality. Especially, fetal lung fibroblasts are known to be hardly immortalized by TERT overexpression. We here established an immortal non-transformed lung fibroblast cell line only by TERT transfection, as well as an immortal transformed cell line by transfection of TERT and SV40 early antigens. Comparing the expression profiles of these cell lines with those of mortal cell strains with elongated lifespan after TERT transfection, 51 genes, including 19 upregulated and 32 downregulated, were explored to be the candidates responsible for regulation of cellular proliferation of lung fibroblasts. These included the genes previously reported to be involved in cellular proliferation, transformation, or self-renewal capacity, and those highly expressed in lung tissues obtained from patients with idiopathic pulmonary fibrosis or hypersensitivity pneumonitis. This set of lung fibrobrast cell lines/strains of identical genetic background with different proliferative capacity, mortal and immortal non-transformed fibroblasts may become useful model cells for research on lung fibroblast growth regulation and the candidate genes explored in this study may provide promising biomarkers or molecular targets of pulmonary fibrosis.\nLast Updated (by provider): \nContributors:  Keiko Hiyama Eiso Hiyama Masahiko Nishiyama Keiji Tanimoto</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Activation of telomerase often endows cancer cells, but rarely normal somatic cells, with immortality. Especially, fetal lung fibroblasts are known to be hardly immortalized by TERT overexpression. We here established an immortal non-transformed lung fibroblast cell line only by TERT transfection, as well as an immortal transformed cell line by transfection of TERT and SV40 early antigens. Comparing the expression profiles of these cell lines with those of mortal cell strains with elongated lifespan after TERT transfection, 51 genes, including 19 upregulated and 32 downregulated, were explored to be the candidates responsible for regulation of cellular proliferation of lung fibroblasts. These included the genes previously reported to be involved in cellular proliferation, transformation, or self-renewal capacity, and those highly expressed in lung tissues obtained from patients with idiopathic pulmonary fibrosis or hypersensitivity pneumonitis. This set of lung fibrobrast cell lines/strains of identical genetic background with different proliferative capacity, mortal and immortal non-transformed fibroblasts may become useful model cells for research on lung fibroblast growth regulation and the candidate genes explored in this study may provide promising biomarkers or molecular targets of pulmonary fibrosis.\nLast Updated (by provider): \nContributors:  Keiko Hiyama Eiso Hiyama Masahiko Nishiyama Keiji Tanimoto</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Activation of telomerase often endows cancer cells, but rarely normal somatic cells, with immortality. Especially, fetal lung fibroblasts are known to be hardly immortalized by TERT overexpression. We here established an immortal non-transformed lung fibroblast cell line only by TERT transfection, as well as an immortal transformed cell line by transfection of TERT and SV40 early antigens. Comparing the expression profiles of these cell lines with those of mortal cell strains with elongated lifespan after TERT transfection, 51 genes, including 19 upregulated and 32 downregulated, were explored to be the candidates responsible for regulation of cellular proliferation of lung fibroblasts. These included the genes previously reported to be involved in cellular proliferation, transformation, or self-renewal capacity, and those highly expressed in lung tissues obtained from patients with idiopathic pulmonary fibrosis or hypersensitivity pneumonitis. This set of lung fibrobrast cell lines/strains of identical genetic background with different proliferative capacity, mortal and immortal non-transformed fibroblasts may become useful model cells for research on lung fibroblast growth regulation and the candidate genes explored in this study may provide promising biomarkers or molecular targets of pulmonary fibrosis.\nLast Updated (by provider): \nContributors:  Keiko Hiyama Eiso Hiyama Masahiko Nishiyama Keiji Tanimoto</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Time course micro array experiment  to identify transcriptional changes in response to exposure of hFLs to different combinations of small molecules during direct neuronal reprogramming\nAt time of import, last updated (by provider) on: Mar 25 2019\n\nContributors: ; [Ulrich Pfisterer, Shamit Soneji, Malin Parmar, Stefan Lang]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                   Lectins are proteins present on cell surfaces or as shed extracellular proteins that function in innate immune defense as phagocytic receptors to recognize specific bacterial cell wall components. Based on the knowledge that cigarette smoking is associated with increased risk of bacterial infection, we hypothesized that cigarette smoking may modulate the expression of lectin genes in the airway epithelium. Affymetrix HG U133 Plus 2.0 microarrays were used to survey expression of lectin genes in large (3rd to 4th order bronchi) airway epithelium from 9 normal nonsmokers and 20 phenotypic normal smokers and small (10th to 12th order bronchi) airway epithelium from 13 normal nonsmokers and 20 phenotypic normal smokers. From the 72 lectin genes that were surveyed, there were no changes (&gt;2-fold change, p&lt;0.05) in gene expression in either large or small airway epithelium among normal smokers compared to nonsmokers except for a striking down regulation in both large and small airway epithelium of normal smokers of intelectin 1, a recently described lectin that participates in the innate immune response by recognizing and binding to galactofuranosyl residues in the cell walls of bacteria (large airway epithelium, p&lt;0.003; small airway epithelium, p&lt;0.002). TaqMan RT-PCR confirmed the observation that intelectin 1 was down-regulated in both large (p&lt;0.05) and small airway epithelium (p&lt;0.02) of normal smokers compared to normal nonsmokers. Immunohistochemistry assessment of biopsies of the large airway epithelium of normal nonsmokers demonstrated intelectin 1 was expressed in secretory cells, with qualitatively decreased expression in biopsies from normal smokers. Western analysis confirmed the decreased expression of intelectin 1 in airway epithelium of normal smokers compared to normal nonsmokers (p&lt;0.02). Finally, compared to normal nonsmokers, intelectin 1 expression was decreased in small airway epithelium of smokers with early COPD (n= 13, p&lt;0.001) and smokers with established COPD (n= 14, p&lt;0.001), in a fashion similar to that of normal smokers. In the context that intelectin 1 is an epithelial molecule that likely plays a role in defense against bacteria, the down regulation of expression of intelectin 1 in response to cigarette smoking may contribute to the increase in susceptibility to infections observed in smokers, including those with COPD.\nLast Updated (by provider): Mar 15 2009\nContributors:  Bishnu P De Holly Vanni Ben-Gary Harvey Ronald G Crystal Brendan Carolan</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Infectious pneumonias exact an unacceptable mortality burden worldwide.  Efforts to protect populations from pneumonia have historically focused on antibiotic development and vaccine-enhanced adaptive immunity.  However, we have recently reported that the lungs’ innate defenses can be therapeutically induced by inhalation of a bacterial lysate that protects mice against otherwise lethal pneumonia.  Here, we tested in mice the hypothesis that Toll-like receptors (TLRs) are required for this antimicrobial phenomenon, and found that resistance could not be induced in the absence of the TLR signaling adaptor protein MyD88.  We then attempted to recapitulate the protection afforded by the bacterial lysate by stimulating the lung epithelium with aerosolized synthetic TLR ligands.  While most single or combination treatments yielded no protection, simultaneous treatment with ligands for TLR2/6 and TLR9 conferred robust, synergistic protection against virulent Gram-positive and Gram-negative pathogens.   Protection was associated with rapid pathogen killing in the lungs, and pathogen killing could be induced from lung epithelial cells in isolation.  Taken together, these data demonstrate the requirement for TLRs in inducible resistance against pneumonia, reveal a remarkable, unanticipated synergistic interaction of TLR2/6 and TLR9, reinforce the emerging evidence supporting the antimicrobial capacity of the lung epithelium, and may provide the basis for a novel clinical therapeutic that can protect patients against pneumonia during periods of peak vulnerability.\nLast Updated (by provider): May 18 2011</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          This experiment investigated the response of mouse pulmonary epithelial cells in culture (FE1 cell line) to 3 different brands of cigarette smoke condensate. The three brands represent a full flavour, regular and king-size light cigarettes. These vary in some key chemical consistuents. Brand #1 has the highest tar content. The cells were exposed to two doses: one at approximately 50% cytotoxicity, and the other at approximately 20%. Both doses were clastogenic. The major pathways affected included xenobiotic metabolism, cell cycle, apoptosis, oxidative stress and inflammation. Although brand #1 exhibited the most numbers of genes that were differentially expressed, there did not appear to be any major differences in the expression profiles between the brands. Although some genes did not reach statistical significance, in general there was consistency in the direction of fold changes for genes across all brands.\nLast Updated (by provider): Apr 08 2012\nContributors:  Julie Buick Rebecca Maertens Paul A White Carole Yauk Sabina Halappanavar Andrew Williams</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           mRNA expression was profiled from pooled bronchial airway epithelial cell brushings (n=3 patients/pool) obtained during bronchoscopy from healthy never (NS) and current smokers (S) and smokers with (C) and without (NC) lung cancer\nLast Updated (by provider): Apr 11 2013\nContributors:  Jennifer Beane</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE             GSE12205                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE             GSE21059                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE             GSE27335                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE             GSE27964                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE             GSE31625                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE             GSE38530                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE             GSE48397                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE             GSE40839                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE             GSE37703                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE             GSE12236                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE             GSE30242                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE             GSE47542                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE              GSE9077                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE              GSE9077                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE              GSE9077                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE             GSE83896                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE             GSE10006                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE             GSE26864                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE             GSE30079                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE             GSE29006                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.sampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               &lt;char&gt;                  &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: http://www.ncbi.nlm.nih.gov/geo/                     12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: http://www.ncbi.nlm.nih.gov/geo/                     72</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: http://www.ncbi.nlm.nih.gov/geo/                     36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: http://www.ncbi.nlm.nih.gov/geo/                     48</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: http://www.ncbi.nlm.nih.gov/geo/                     20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: http://www.ncbi.nlm.nih.gov/geo/                     10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: http://www.ncbi.nlm.nih.gov/geo/                     21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: http://www.ncbi.nlm.nih.gov/geo/                     12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: http://www.ncbi.nlm.nih.gov/geo/                     40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13: http://www.ncbi.nlm.nih.gov/geo/                      7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14: http://www.ncbi.nlm.nih.gov/geo/                      4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: http://www.ncbi.nlm.nih.gov/geo/                      6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: http://www.ncbi.nlm.nih.gov/geo/                     30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: http://www.ncbi.nlm.nih.gov/geo/                     87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: http://www.ncbi.nlm.nih.gov/geo/                     22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: http://www.ncbi.nlm.nih.gov/geo/                     66</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: http://www.ncbi.nlm.nih.gov/geo/                      4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.sampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.lastUpdated  experiment.batchEffect geeq.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     &lt;POSc&gt;                  &lt;char&gt;              &lt;lgcl&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    2023-12-16 07:28:10 BATCH_CORRECTED_SUCCESS                TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    2023-12-18 02:13:17           NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    2023-12-18 17:43:52 NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    2023-12-18 18:37:58           NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    2023-12-19 00:29:29 BATCH_CORRECTED_SUCCESS                TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    2023-12-19 11:01:23 NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    2023-12-20 01:11:43    SINGLE_BATCH_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    2023-12-19 14:19:36 BATCH_CORRECTED_SUCCESS                TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    2023-12-20 08:47:35 BATCH_CORRECTED_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    2023-12-16 11:40:21           NO_BATCH_INFO                TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    2023-12-18 22:19:55           NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    2024-01-03 22:57:49 NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    2023-12-21 10:21:37           NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    2023-12-21 10:21:15           NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    2023-12-22 08:58:48           NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    2023-12-21 06:04:57 NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    2023-12-15 20:33:01           NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    2023-12-18 16:53:31           NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    2023-12-18 21:54:08           NO_BATCH_INFO               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    2023-12-18 20:10:23 NO_BATCH_EFFECT_SUCCESS               FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.lastUpdated  experiment.batchEffect geeq.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;num&gt;            &lt;num&gt;        &lt;num&gt;       &lt;num&gt;       &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                  1                1            1  0.85582592      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                 -1                0           -1  0.42483397      0.5000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                  1                1           -1  0.85060640      0.7500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                  0                0           -1  0.28459247      0.7500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                  1                1            1  0.70752488      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                  1                1           -1  0.99746711      0.7500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                  1                1            1  0.99514636      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                  1                1            1  0.85425178      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                  1                1            1  0.99875195      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                  1                0            1  0.83735677      1.0000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                 -1                0            1  0.28517914      0.8750</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                  1                1           -1  0.84367763      0.5000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  0                0           -1  0.13234039     -0.0375</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                  0                0           -1  0.13195099      0.5000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                  0                0            1  0.13797564      0.8375</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                  1                1            1  0.56855414      1.0000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                 -1                0            1  0.56606943      1.0000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                  0                0           -1  0.41962800      0.4375</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                  0                0           -1 -0.04787478      0.5000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                  1                1            1  0.70516800      0.7500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.name taxon.scientific taxon.ID taxon.NCBI taxon.database.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;char&gt;           &lt;char&gt;    &lt;int&gt;      &lt;int&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:      mouse     Mus musculus        2      10090                mm10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:      human     Homo sapiens        1       9606                hg38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.name taxon.scientific taxon.ID taxon.NCBI taxon.database.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.ID</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

