<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve the probes associated to a genes across all platforms — get_gene_probes • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve the probes associated to a genes across all platforms — get_gene_probes"><meta property="og:description" content="Retrieve the probes associated to a genes across all platforms"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve the probes associated to a genes across all platforms</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>get_gene_probes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieve the probes associated to a genes across all platforms</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_gene_probes</span><span class="op">(</span></span>
<span>  <span class="va">gene</span>,</span>
<span>  offset <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>gene</dt>
<dd><p>An ensembl gene identifier which typically starts with ensg or an ncbi gene identifier or an official gene symbol approved by hgnc</p></dd>


<dt>offset</dt>
<dd><p>The offset of the first retrieved result.</p></dd>


<dt>limit</dt>
<dd><p>Optional, defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data table with information about the probes representing a gene across
all platrofms. A list if <code>raw = TRUE</code>.
A <code>404 error</code> if the given identifier does not map to any genes.</p>


<p>The fields of the output data.table are:</p>


<ul><li><p><code>element.name</code>: Name of the element. Typically the probeset name</p></li>
<li><p><code>element.description</code>: A free text field providing optional information about the element</p></li>
<li><p><code>platform.ShortName</code>: Shortname of the platform given by Gemma. Typically the GPL identifier.</p></li>
<li><p><code>platform.Name</code>: Full name of the platform</p></li>
<li><p><code>platform.ID</code>: Id number of the platform given by Gemma</p></li>
<li><p><code>platform.taxon</code>: Species the platform was designed for</p></li>
<li><p><code>platform.taxonID</code>: Id number of the species given by Gemma</p></li>
<li><p><code>platform.type</code>: Type of the platform.</p></li>
<li><p><code>platform.description</code>: Free text field describing the platform.</p></li>
<li><p><code>platform.troubled</code>: Whether the platform is marked as troubled by a Gemma curator.</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">get_gene_probes</span><span class="op">(</span><span class="st">"DYRK1A"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         element.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:       1374773_at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:       1370183_at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: rc_AI104012_g_at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:   rc_AI104012_at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:            19904</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             1639</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:            19904</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:             1639</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:       1374773_at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:       1370183_at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:       R000990_01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:        GE1269392</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:          GE15753</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:   rc_AI073261_at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:            43666</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:              541</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    1374773_at_10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:     1374773_at_9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:     1374773_at_8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:     1374773_at_7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         element.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                 element.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                           Transcribed sequences</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                               dual-specificity tyrosine-(Y)-phosphorylation regulated kinase 1A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                               dual-specificity tyrosine-(Y)-phosphorylation regulated kinase 1A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                               dual-specificity tyrosine-(Y)-phosphorylation regulated kinase 1A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                          Dyrk1a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                          Dyrk1a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                               Transcribed locus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                               dual-specificity tyrosine-(Y)-phosphorylation regulated kinase 1A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12: Multiple external sequence references: NM_012791.1, BF554476.1;  [Renamed by Gemma from 311063]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                   [Renamed by Gemma from 58029]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                            &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                          Dyrk1a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                 element.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     platform.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:            GPL1355</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:            GPL1355</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:              GPL85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:              GPL85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:             GPL890</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             GPL890</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:            GPL2882</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            GPL2882</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:             GPL341</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:             GPL341</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:            GPL4528</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:            GPL2896</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:            GPL2896</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:              GPL87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:            GPL2877</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:            GPL2877</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:            GPL3240</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:            GPL3240</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:            GPL3240</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:            GPL3240</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     platform.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                platform.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                       &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                             Affymetrix GeneChip Rat Genome 230 2.0 Array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                             Affymetrix GeneChip Rat Genome 230 2.0 Array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                     Affymetrix GeneChip Rat Genome U34 Array Set RG-U34A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                     Affymetrix GeneChip Rat Genome U34 Array Set RG-U34A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                             Agilent-011868 Rat Oligo Microarray (G4130A)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                             Agilent-011868 Rat Oligo Microarray (G4130A)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                          Agilent-013328 Rat Oligo Microarray (V2) G4130B</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                          Agilent-013328 Rat Oligo Microarray (V2) G4130B</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                 Affymetrix GeneChip Rat Expression Set 230 Array RAE230A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                 Affymetrix GeneChip Rat Expression Set 230 Array RAE230A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                      CapitalBio Rat 27k long oligo array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12: GE Healthcare/Amersham Biosciences CodeLink?   Rat Whole Genome Bioarray</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13: GE Healthcare/Amersham Biosciences CodeLink?   Rat Whole Genome Bioarray</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                     Affymetrix GeneChip Rat Genome U34 Array Set RG-U34C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                        Agilent-013162 Whole Rat Genome Microarray G4131A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                        Agilent-013162 Whole Rat Genome Microarray G4131A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:       Affymetrix GeneChip Rat Expression Set 230 Array RAE230A PM probes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:       Affymetrix GeneChip Rat Expression Set 230 Array RAE230A PM probes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:       Affymetrix GeneChip Rat Expression Set 230 Array RAE230A PM probes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:       Affymetrix GeneChip Rat Expression Set 230 Array RAE230A PM probes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                platform.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     platform.ID platform.type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;int&gt;        &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:           2      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:           2      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:           6      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:           6      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:           9      TWOCOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:           9      TWOCOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:         160      TWOCOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:         160      TWOCOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:         164      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:         164      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:         165      TWOCOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:         169      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:         169      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:         192      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:         240      DUALMODE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:         240      DUALMODE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:         265      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:         265      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:         265      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:         265      ONECOLOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     platform.ID platform.type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             platform.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The RG-U34 set includes 3 arrays with a total of 26379 entries and was indexed 29-Jan-2002. The set includes ~7,000 known genes and &gt;17,000 EST clusters. The A array contains probes derived from all full-length or annotated genes as well as thousands of EST clusters. The B and C arrays contain only EST clusters.  Keywords = high density oligonucleotide array\nFrom GPL85\nLast Updated: Mar 09 2006</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The RG-U34 set includes 3 arrays with a total of 26379 entries and was indexed 29-Jan-2002. The set includes ~7,000 known genes and &gt;17,000 EST clusters. The A array contains probes derived from all full-length or annotated genes as well as thousands of EST clusters. The B and C arrays contain only EST clusters.  Keywords = high density oligonucleotide array\nFrom GPL85\nLast Updated: Mar 09 2006</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         This microarray contains representative sequences derived from rat RefSeq, high-quality public rat mRNA accessions, homologs to human and mouse RefSeq, EST clusters determined by a leading biopharmaceutical company and the Incyte Zooseq database as well as toxicology markers determined by the National Institute of Environmental Health Sciences (NIEHS) and others. Many of the toxicology markers used in the design of this microarray were released to the public domain for the first time and will be used on this unique microarray product.  Arrays of this design have barcodes that begin with 16011868 or 2511868.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. \nFrom GPL890\nLast Updated: Sep 22 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         This microarray contains representative sequences derived from rat RefSeq, high-quality public rat mRNA accessions, homologs to human and mouse RefSeq, EST clusters determined by a leading biopharmaceutical company and the Incyte Zooseq database as well as toxicology markers determined by the National Institute of Environmental Health Sciences (NIEHS) and others. Many of the toxicology markers used in the design of this microarray were released to the public domain for the first time and will be used on this unique microarray product.  Arrays of this design have barcodes that begin with 16011868 or 2511868.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. \nFrom GPL890\nLast Updated: Sep 22 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:   This microarray contains representative sequences derived from rat RefSeq, high-quality public rat mRNA accessions, homologs to human and mouse RefSeq, EST clusters determined by a leading biopharmaceutical company and the Incyte Zooseq database as well as toxicology markers determined by the National Institute of Environmental Health Sciences (NIEHS) and others. Many of the toxicology markers used in the design of this microarray were released to the public domain for the first time and will be used on this unique microarray product.  Arrays of this design have barcodes that begin with 16013328 or 2513328.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions.   This microarray contains representative sequences derived from rat RefSeq, high-quality public rat mRNA accessions, homologs to human and mouse RefSeq, EST clusters determined by a leading biopharmaceutical company and the Incyte Zooseq database as well as toxicology markers determined by the National Institute of Environmental Health Sciences (NIEHS) and others. Many of the toxicology markers used in the design of this microarray were released to the public domain for the first time and will be used on this unique microarray product.  Arrays of this design have barcodes that begin with 16013328 or 2513328.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:   This microarray contains representative sequences derived from rat RefSeq, high-quality public rat mRNA accessions, homologs to human and mouse RefSeq, EST clusters determined by a leading biopharmaceutical company and the Incyte Zooseq database as well as toxicology markers determined by the National Institute of Environmental Health Sciences (NIEHS) and others. Many of the toxicology markers used in the design of this microarray were released to the public domain for the first time and will be used on this unique microarray product.  Arrays of this design have barcodes that begin with 16013328 or 2513328.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions.   This microarray contains representative sequences derived from rat RefSeq, high-quality public rat mRNA accessions, homologs to human and mouse RefSeq, EST clusters determined by a leading biopharmaceutical company and the Incyte Zooseq database as well as toxicology markers determined by the National Institute of Environmental Health Sciences (NIEHS) and others. Many of the toxicology markers used in the design of this microarray were released to the public domain for the first time and will be used on this unique microarray product.  Arrays of this design have barcodes that begin with 16013328 or 2513328.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Array A of GeneChip Rat Expression Set 230 Has 15923 entries and was indexed 09-Apr-2003 Sequences used in the design of the array were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from Build 99 of UniGene (June 2002) and refined by analysis and comparison with a number of other publicly available databases including the Baylor College of Medicine Human Genome Sequencing Center's preliminary rat genome assembly (June 2002). In addition, sequences were analyzed for untrimmed low-quality sequence information, correct orientation, false clustering, alternative splicing and alternative polyadenylation. \nFrom GPL341\nLast Updated: Jul 28 2006</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Array A of GeneChip Rat Expression Set 230 Has 15923 entries and was indexed 09-Apr-2003 Sequences used in the design of the array were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from Build 99 of UniGene (June 2002) and refined by analysis and comparison with a number of other publicly available databases including the Baylor College of Medicine Human Genome Sequencing Center's preliminary rat genome assembly (June 2002). In addition, sequences were analyzed for untrimmed low-quality sequence information, correct orientation, false clustering, alternative splicing and alternative polyadenylation. \nFrom GPL341\nLast Updated: Jul 28 2006</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                This set includes 27k oligonucleotides, mostly 70-mers, designed based upon the Ensembl Rat Database Version v19.3b.2 and Rat Genome Project.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                CodeLink Rat Whole Genome Bioarray is designed to interrogate approximately 34 000 transcripts representing most of the known and predictive genes of the rat genome, as it is described today in the public domain. Each transcript is represented by a 30-mer probe which is designed to conserved exons across the transcripts of targeted genes. The probe sequences, representing well annotated, full length and partial rat gene sequences, were designed based on sequences selected from the NCBI UniGene build #129, RefSeq database (April 1, 2004 release) and dbEST database (March 20, 2004 release).  Well-annotated mRNA or coding sequences were chosen to ensure usefulness for a large range of applications in basic research, biotechnology, and drug development. Each sequence was carefully screened to ensure high-quality, specific probe design, and to reduce redundancy of gene targets. All of the probes designed to these sequences were functionally tested, and over 95% were functionally validated against 10 rat tissues to ensure best representation of the gene and biologically relevant results. For flexibility and data preservation, CodeLink Rat Whole Genome Bioarrays are compatible with lower density CodeLink rat bioarrays. This means that existing data from CodeLink UniSet? Rat 10K Bioarrays can be continuously used in future studies along with newly generated data from CodeLink Rat Whole Genome Bioarrays.  CodeLink users can submit data to GEO by exporting tab-delimited text files from the CodeLink software. The "Feature_id" header will need to be renamed to "ID_REF", and the "Normalized_intensity" header to "VALUE" before uploading the exported tables.  In addition, users are encouraged to include additional raw data columns in their data uploads. These additional columns do not need to be renamed. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                CodeLink Rat Whole Genome Bioarray is designed to interrogate approximately 34 000 transcripts representing most of the known and predictive genes of the rat genome, as it is described today in the public domain. Each transcript is represented by a 30-mer probe which is designed to conserved exons across the transcripts of targeted genes. The probe sequences, representing well annotated, full length and partial rat gene sequences, were designed based on sequences selected from the NCBI UniGene build #129, RefSeq database (April 1, 2004 release) and dbEST database (March 20, 2004 release).  Well-annotated mRNA or coding sequences were chosen to ensure usefulness for a large range of applications in basic research, biotechnology, and drug development. Each sequence was carefully screened to ensure high-quality, specific probe design, and to reduce redundancy of gene targets. All of the probes designed to these sequences were functionally tested, and over 95% were functionally validated against 10 rat tissues to ensure best representation of the gene and biologically relevant results. For flexibility and data preservation, CodeLink Rat Whole Genome Bioarrays are compatible with lower density CodeLink rat bioarrays. This means that existing data from CodeLink UniSet? Rat 10K Bioarrays can be continuously used in future studies along with newly generated data from CodeLink Rat Whole Genome Bioarrays.  CodeLink users can submit data to GEO by exporting tab-delimited text files from the CodeLink software. The "Feature_id" header will need to be renamed to "ID_REF", and the "Normalized_intensity" header to "VALUE" before uploading the exported tables.  In addition, users are encouraged to include additional raw data columns in their data uploads. These additional columns do not need to be renamed. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The RG-U34 set includes 3 arrays with a total of 26379 entries and was indexed 29-Jan-2002. The set includes ~7,000 known genes and &gt;17,000 EST clusters. The A array contains probes derived from all full-length or annotated genes as well as thousands of EST clusters. The B and C arrays contain only EST clusters.  Keywords = high density oligonucleotide array\nFrom GPL87\nLast Updated: Jul 28 2006</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                         With a focus on well known rat genes and homologues to human and  mouse genes with useful annotation, Agilent's Whole Rat Genome Oligo Microarray provides researchers with a new tool for modeling human biology in the rat model organism. For researchers, this means they now have access to a microarray made up of relevant content that has been empirically validated by Agilent.  Arrays of this design have barcodes that begin with 16013162 or 2513162.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions.   With a focus on well known rat genes and homologues to human and  mouse genes with useful annotation, Agilent's Whole Rat Genome Oligo Microarray provides researchers with a new tool for modeling human biology in the rat model organism. For researchers, this means they now have access to a microarray made up of relevant content that has been empirically validated by Agilent.  Arrays of this design have barcodes that begin with 16013162 or 2513162.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                         With a focus on well known rat genes and homologues to human and  mouse genes with useful annotation, Agilent's Whole Rat Genome Oligo Microarray provides researchers with a new tool for modeling human biology in the rat model organism. For researchers, this means they now have access to a microarray made up of relevant content that has been empirically validated by Agilent.  Arrays of this design have barcodes that begin with 16013162 or 2513162.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions.   With a focus on well known rat genes and homologues to human and  mouse genes with useful annotation, Agilent's Whole Rat Genome Oligo Microarray provides researchers with a new tool for modeling human biology in the rat model organism. For researchers, this means they now have access to a microarray made up of relevant content that has been empirically validated by Agilent.  Arrays of this design have barcodes that begin with 16013162 or 2513162.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             platform.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     platform.troubled taxon.name  taxon.scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                &lt;lgcl&gt;     &lt;char&gt;            &lt;char&gt;    &lt;int&gt;      &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:             FALSE        rat Rattus norvegicus        3      10116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     platform.troubled taxon.name  taxon.scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.name taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;char&gt;             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.name taxon.database.ID</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

