<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve differential expression results — get_differential_expression_values • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve differential expression results — get_differential_expression_values"><meta property="og:description" content="Retrieves the differential expression result set(s) associated with the dataset.
To get more information about the contrasts in individual resultSets and
annotation terms associated them, use get_dataset_differential_expression_analyses()"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve differential expression results</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/convenience.R" class="external-link"><code>R/convenience.R</code></a></small>
    <div class="hidden name"><code>get_differential_expression_values.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieves the differential expression result set(s) associated with the dataset.
To get more information about the contrasts in individual resultSets and
annotation terms associated them, use <code><a href="get_dataset_differential_expression_analyses.html">get_dataset_differential_expression_analyses()</a></code></p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_differential_expression_values</span><span class="op">(</span></span>
<span>  dataset <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  resultSets <span class="op">=</span> <span class="cn">NA_integer_</span>,</span>
<span>  readableContrasts <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dataset</dt>
<dd><p>A dataset identifier.</p></dd>


<dt>resultSets</dt>
<dd><p>resultSet identifiers. If a dataset is not provided, all
result sets will be downloaded. If it is provided it will only be used
to ensure all result sets belong to the dataset.</p></dd>


<dt>readableContrasts</dt>
<dd><p>If <code>FALSE</code> (default), the returned columns will
use internal constrasts IDs as names. Details about the contrasts can be accessed
using <code><a href="get_dataset_differential_expression_analyses.html">get_dataset_differential_expression_analyses</a></code>. If TRUE IDs will
be replaced with human readable contrast information.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A list of data tables with differential expression
values per result set.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>In Gemma each result set corresponds to
the estimated effects associated with a single factor in the design, and each can have multiple contrasts (for each level compared to baseline).
Thus a dataset with a 2x3 factorial design will have two result sets, one of which will have one contrast, and one having two contrasts.</p>
<p>The methodology for differential expression is explained in <a href="https://doi.org/10.1093/database/baab006" class="external-link">Curation of over 10000 transcriptomic studies to enable data reuse</a>.
Briefly, differential expression analysis is performed on the dataset based on the annotated
experimental design with up two three potentially nested factors.
Gemma attempts to automatically assign baseline conditions for each factor.
In the absence of a clear control condition, a baseline is arbitrarily selected.
A generalized linear model with empirical Bayes shrinkage of t-statistics is fit to the data
for each platform element (probe/gene) using an implementation of the limma algorithm. For RNA-seq data,
we use weighted regression, applying the
voom algorithm to compute weights from the mean–variance relationship of the data.
Contrasts of each condition are then computed compared to the selected baseline.
In some situations, Gemma will split the data into subsets for analysis.
A typical such situation is when a ‘batch’ factor is present and confounded with another factor,
the subsets being determined by the levels of the confounding factor.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">get_differential_expression_values</span><span class="op">(</span><span class="st">"GSE2018"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $`573187`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              Probe          NCBIid         GeneSymbol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;char&gt;          &lt;char&gt;             &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1:   219118_at           51303             FKBP11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     2: 203308_x_at            3257               HPS1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     3:   210005_at            2618               GART</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     4: 221791_s_at 51372|112268293         TMA7|TMA7B</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5: 202803_s_at            3689              ITGB2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ---                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22279:   212384_at                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22280: 203578_s_at            9057             SLC7A6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22281:   203871_at 26168|100533955 SENP3|SENP3-EIF4A1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22282:   218582_at           54708            MARCHF5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22283:   205896_at            6583            SLC22A4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                          GeneName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                            &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1:                                                                                                   FKBP prolyl isomerase 11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     2:                                                                HPS1 biogenesis of lysosomal organelles complex 3 subunit 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     3: phosphoribosylglycinamide formyltransferase, phosphoribosylglycinamide synthetase, phosphoribosylaminoimidazole synthetase</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     4:                                    translation machinery associated 7 homolog|translation machinery associated 7 homolog B</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5:                                                                                                    integrin subunit beta 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ---                                                                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22279:                                                                                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22280:                                                                                           solute carrier family 7 member 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22281:                                                         SUMO specific peptidase 3|SENP3-EIF4A1 readthrough (NMD candidate)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22282:                                                                                  membrane associated ring-CH-type finger 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22283:                                                                                          solute carrier family 22 member 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           pvalue corrected_pvalue   rank contrast_1_log2fc contrast_1_tstat</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;num&gt;            &lt;num&gt;  &lt;num&gt;             &lt;num&gt;            &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1: 6.184e-06           0.0004 0.0154            0.2811           5.3405</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     2: 2.900e-02           0.1150 0.2521           -0.1410          -2.2800</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     3: 1.496e-01           0.3291 0.4546            0.1121           1.4742</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     4: 3.670e-02           0.1336 0.2748            0.1944           2.1744</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5: 4.798e-01           0.6676 0.7187           -0.0649          -0.7145</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ---                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22279: 2.310e-02           0.0993 0.2330            0.1177           2.3784</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22280: 2.737e-01           0.4766 0.5743            0.0462           1.1125</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22281: 4.720e-02           0.1572 0.3004           -0.1823          -2.0590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22282: 7.510e-01           0.8626 0.8706            0.0198           0.3199</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22283: 5.120e-02           0.1659 0.3086            0.2398           2.0211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        contrast_1_pvalue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1:         6.184e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     2:         2.900e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     3:         1.496e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     4:         3.670e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5:         4.798e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ---                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22279:         2.310e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22280:         2.737e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22281:         4.720e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22282:         7.510e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22283:         5.120e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

