<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve the datasets for a given taxon — get_taxon_datasets • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve the datasets for a given taxon — get_taxon_datasets"><meta property="og:description" content="This function is deprecated in favor of get_datasets"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.1.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve the datasets for a given taxon</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/devel/R/deprecation.R" class="external-link"><code>R/deprecation.R</code></a></small>
    <div class="hidden name"><code>get_taxon_datasets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function is deprecated in favor of <code><a href="get_datasets.html">get_datasets</a></code></p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_taxon_datasets</span><span class="op">(</span></span>
<span>  <span class="va">taxon</span>,</span>
<span>  offset <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  sort <span class="op">=</span> <span class="st">"+id"</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-taxon">taxon<a class="anchor" aria-label="anchor" href="#arg-taxon"></a></dt>
<dd><p>can either be Taxon ID, Taxon NCBI ID, or one of its string identifiers: scientific name, common name.
It is recommended to use Taxon ID for efficiency.
Please note, that not all taxa have all the possible identifiers available.
Use the <code><a href="get_taxa_by_ids.html">get_taxa_by_ids</a></code> function to retrieve the necessary information. For convenience, below is a list of officially supported taxa:</p><table class="table table"><tr><td><strong>ID</strong></td><td><strong>Comm.name</strong></td><td><strong>Scient.name</strong></td><td><strong>NcbiID</strong></td></tr><tr><td>1</td><td>human</td><td>Homo sapiens</td><td>9606</td></tr><tr><td>2</td><td>mouse</td><td>Mus musculus</td><td>10090</td></tr><tr><td>3</td><td>rat</td><td>Rattus norvegicus</td><td>10116</td></tr><tr><td>11</td><td>yeast</td><td>Saccharomyces cerevisiae</td><td>4932</td></tr><tr><td>12</td><td>zebrafish</td><td>Danio rerio</td><td>7955</td></tr><tr><td>13</td><td>fly</td><td>Drosophila melanogaster</td><td>7227</td></tr><tr><td>14</td><td>worm</td><td>Caenorhabditis elegans</td><td>6239</td></tr></table></dd>


<dt id="arg-offset">offset<a class="anchor" aria-label="anchor" href="#arg-offset"></a></dt>
<dd><p>The offset of the first retrieved result.</p></dd>


<dt id="arg-limit">limit<a class="anchor" aria-label="anchor" href="#arg-limit"></a></dt>
<dd><p>Optional, defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt id="arg-sort">sort<a class="anchor" aria-label="anchor" href="#arg-sort"></a></dt>
<dd><p>Order results by the given property and direction. The '+' sign indicate ascending order whereas the '-' indicate descending.</p></dd>


<dt id="arg-raw">raw<a class="anchor" aria-label="anchor" href="#arg-raw"></a></dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt id="arg-memoised">memoised<a class="anchor" aria-label="anchor" href="#arg-memoised"></a></dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt id="arg-file">file<a class="anchor" aria-label="anchor" href="#arg-file"></a></dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be a JSON file. Otherwise,
it will be a RDS file.</p></dd>


<dt id="arg-overwrite">overwrite<a class="anchor" aria-label="anchor" href="#arg-overwrite"></a></dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Kept for compatibility. Ignored</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A data table with information about the queried dataset(s). A list if
<code>raw = TRUE</code>. Returns an empty list if no datasets matched.</p>
<p>The fields of the output data.table are:</p>
<ul><li><p><code>experiment.shortName</code>: Shortname given to the dataset within Gemma. Often corresponds to accession ID</p></li>
<li><p><code>experiment.name</code>: Full title of the dataset</p></li>
<li><p><code>experiment.ID</code>: Internal ID of the dataset.</p></li>
<li><p><code>experiment.description</code>: Description of the dataset</p></li>
<li><p><code>experiment.troubled</code>: Did an automatic process within gemma or a curator mark the dataset as "troubled"</p></li>
<li><p><code>experiment.accession</code>: Accession ID of the dataset in the external database it was taken from</p></li>
<li><p><code>experiment.database</code>: The name of the database where the dataset was taken from</p></li>
<li><p><code>experiment.URI</code>: URI of the original database</p></li>
<li><p><code>experiment.sampleCount</code>: Number of samples in the dataset</p></li>
<li><p><code>experiment.batchEffectText</code>: A text field describing whether the dataset has batch effects</p></li>
<li><p><code>experiment.batchCorrected</code>: Whether batch correction has been performed on the dataset.</p></li>
<li><p><code>experiment.batchConfound</code>: 0 if batch info isn't available, -1 if batch counfoud is detected, 1 if batch information is available and no batch confound found</p></li>
<li><p><code>experiment.batchEffect</code>: -1 if batch p value &lt; 0.0001, 1 if batch p value &gt; 0.1, 0 if otherwise and when there is no batch information is available or when the data is confounded with batches.</p></li>
<li><p><code>experiment.rawData</code>: -1 if no raw data available, 1 if raw data was available. When available, Gemma reprocesses raw data to get expression values and batches</p></li>
<li><p><code>geeq.qScore</code>: Data quality score given to the dataset by Gemma.</p></li>
<li><p><code>geeq.sScore</code>: Suitability score given to the dataset by Gemma. Refers to factors like batches, platforms and other aspects of experimental design</p></li>
<li><p><code>taxon.name</code>: Name of the species</p></li>
<li><p><code>taxon.scientific</code>: Scientific name for the taxon</p></li>
<li><p><code>taxon.ID</code>: Internal identifier given to the species by Gemma</p></li>
<li><p><code>taxon.NCBI</code>: NCBI ID of the taxon</p></li>
<li><p><code>taxon.database.name</code>: Underlying database used in Gemma for the taxon</p></li>
<li><p><code>taxon.database.ID</code>: ID of the underyling database used in Gemma for the taxon</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">get_taxon_datasets</span><span class="op">(</span><span class="st">"human"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>get_taxon_datasets is deprecated. please use get_datasets instead</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:              GSE2018</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:              GSE4036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:              GSE3489</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:              GSE1923</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               GSE361</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               GSE492</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               GSE713</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               GSE833</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               GSE420</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               GSE430</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               GSE675</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               GSE712</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               GSE701</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               GSE685</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               GSE593</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               GSE837</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               GSE590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               GSE755</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               GSE994</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               GSE846</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                            experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                     &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                            Human Lung Transplant - BAL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                perro-affy-human-186940</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: Patterns of gene dysregulation in the frontal cortex of patients with HIV encephalitis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: Identification of PDGF-dependent patterns of gene expression in U87 glioblastoma cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                   Mammary epithelial cell transduction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                               Effect of prostaglandin analogs on aqueous humor outflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                        UV radiation-induced DNA damage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                          Amyotrophic lateral sclerosis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                       Aortic stiffness</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:  Effect of left ventricular assist device support on congestive heart failure patients</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                     Time course analysis of response to HCMV infection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                        CVB3-infected HeLa cells (multiple time points)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                 Transcriptional response of lymphoblastoid cells to ionizing radiation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                DACH1 inhibits TGF-beta signaling through binding Smad4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                 Uterine Fibroid and Normal Myometrial Expression Profiles- U133 Arrays</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                           angiogenesis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                              USF1 haplotype comparison</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                         MRI lytic and no lytic lesions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:           Effects of cigarette smoke on the human airway epithelial cell transcriptome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                             Conversion</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                            experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:            18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:            50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:            98</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:           100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:           102</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:           106</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:           110</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:           111</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:           113</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:           114</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:           116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:           117</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:           118</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:           120</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:           124</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:           126</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:           128</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:           129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Bronchoalveolar lavage samples collected from lung transplant recipients.  Numeric portion of sample name is an arbitrary patient ID and AxBx number indicates the perivascular (A) and bronchiolar (B) scores from biopsies collected on the same day as the BAL fluid was collected.  Several patients have more than one sample in this series and can be determined by patient number followed by a lower case letter.  Acute rejection state is determined by the combined A and B score - specifically, a combined AB score of 2 or greater is considered an acute rejection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: Our laboratory has developed the first mouse model overexpressing a RNA-binding protein, the ELAV-like protein HuD, in the CNS under the control of the CaMKinII alpha promoter. Initial behavioral characterization of the mice revealed that they had significant learning deficits together with abnormalities in prepulse inhibition (PPI). At the molecular level, we found that the expression of the growth-associated protein GAP-43, one of the targets of HuD, was increased in the hippocampus of HuD transgenic mice. To characterize these mice further and to evaluate the utility of these animals in understanding human diseases, we propose to use DNA microarray methods. To test our hypothesis we propose 3 specific aims: 1) To characterize the pattern of gene expression in the hippocampus of HuD overexpressor mice 2) To compare the pattern of gene expression in our mouse model with that in the hippocampus of rats prenatally exposed to alcohol (FAS model) and 3) To compare the pattern of gene expression in our mouse model with that shown in post-mortem tissues of patients with schizophrenia. In our previous protocols we examined the pattern of gene expression in our HuD transgenic mice and in rats prenaltally exposed to alcohol. A report by another group (Hakak et al, 2001) showed that three of the HuD targets were upregulated in the prefrontal cortex of patients with schizophrenia. To evaluate whether other target of HuD may be affected in this illness, in the current protocol, we want to compare the pattern of expression in our transgenic mice with in tissue from patients with schizophrenia Based on the behavioral and molecular properties of our HuD transgenic mice we hypothesize that these animals may be good models for the studying the basis of learning disabilities and of diseases that show deficits in PPI such as fetal alcohol syndrome and schizophrenia. All 28 samples are derived from cerebellar tissues for patients with schizophrenia and matched controls.  The specimens were obtained from the Maryland Brain Collection according to NIH guidelines for confidentially and privacy. The protocol used in these studies was reviewed by our HRRC which found that our studies do not fall within the category of protocols monitored by the IRB (see attached letter form the HRRC). Specimens from 14 patients with a diagnosis of schizophrenia performed according to DSM-IV criteria and 14 sex-, age- and PMI-matched controls was included in this study. No differences were found between patients and control subjects in the average age (45±12 versus 43±10 years, p=0.86) or PMI (12±5 versus 16±6 hours, p=0.11). We will provide 28 samples containing 5 ug of RNA each in DEPC water (see validation of the quality of the RNA below). In addition, we include in our study animals treated with haloperidol as control for medication. These samples will be submitted in a separate protocol.\nDate GSE4036 Last Updated: Jan 13 2006\nContributors:  Nora I Perrone-Bizzozero\nIncludes GDS1917.\n Update date: Jun 14 2006.\n Dataset description GDS1917: Analysis of cortical samples corresponding to the crus I/VIIa area of the cerebellum from schizophrenia patients. A study indicates that targets of the RNA-binding ELAV-like protein HuD are overexpressed in the prefrontal cortex of patients with schizophrenia.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              The neurodegenerative process in HIV encephalitis (HIVE) is associated with extensive damage to the dendritic and synaptic structure that often leads to cognitive impairment. Several mechanisms might be at play, including release of neurotoxins, oxidative stress and decreased activity of neurotrophic factors. Furthermore, HIV-mediated dysregulation of genes involved in neuronal maintenance might play an important role. For this purpose, cRNA was prepared from the brains of 17 AIDS patients for analysis with the Affymetrix Human U95Av2 GeneChip and analyzed with the GeneSpring Expression Analysis Software. Out of 12,625 genes analyzed, 74 were downregulated and 59 were upregulated compared to controls. Initial alternative analysis of RNA was performed by ribonuclease protection assay (RPA). In cases with HIVE, downregulated genes included neuronal molecules involved in synaptic plasticity and transmission (ion channels, synaptogyrin, synapsin II), cell cycle (p35, p39, CDC-L2, CDC42, PAK1) and signaling molecules (PI3K, Ras-Raf-MEK1), transcription factors and cytoskeletal components (MAP-1B, MAP-2, tubulin, adducin-2). Upregulated genes included those involved in neuroimmune (IgG, MHC, ?2microglobulin) and anti-viral responses (interferon-inducible molecules), transcription (STAT1, OLIG2, Pax-6) and signaling modulation (MEK3, EphB1) of the cytoskeleton (myosin, aduccin-3, radixin, dystrobrevin). Taken together, this study suggests that HIV proteins released from infected macrophages might not only induce a neuroinflammatory response, but also may promote neurodegeneration by interfering with neuronal transcription of genes involved in regulating signaling and cytoskeletal molecules important in maintaining synapto-dendritic functioning and integrity.\nDate GSE3489 Last Updated: Mar 07 2006\nContributors:  Eleanor S Roberts Dianne Langford Anthony Adame Edward Rockenstein Leslie Crews Howard S Fox Eliezer Masiliah Ian Everall\nIncludes GDS1726.\n Update date: May 30 2006.\n Dataset description GDS1726: Analysis of brain frontal cortex of HIV-seropositive patients with HIV encephalitis (HIVE). HIVE affects &gt;40% of AIDS patients, promoting neurodegeneration and cognitive impairment. Results suggest HIV-mediated dysregulation of genes involved in neuronal maintenance might play an important role.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Overall study: Identification of PDGF-dependent patterns of gene expression in U87 glioblastoma cells.  RNA was obtained from triplicate dishes of 5 different groups of U87 cells, each (total 15) analyzed with one U95 microarray chip.  Three different comparisons were made: 1) Clone 3.1 (34580-34582) vs. clone 3.3 (34583-34585) vs. parent U87 (34592-34594).  Purpose: demonstrate that the gene expression profiles between these 3 cell lines are not different, so they could be pooled as a single untreated group.  2)  Pooled control group (34580-34585, 34592-34594) vs. clone 8.1 (34586-34588).  Purpose:  identify genes specifically controlled by autocrine PDGF activity.  3)  Clone 8.1 (34586-34588) vs. clone 8.1 treated with PDGF (34589-34591)  Purpose:  Identify genes specifically induced by exogenous PDGF.\nDate GSE1923 Last Updated: Dec 19 2005\nContributors:  David M Kaetzel Catherine Nutt Xuejun Peng Piam Shanehsaz Deqin Ma David N Louis\nIncludes GDS1730.\n Update date: May 02 2006.\n Dataset description GDS1730: Analysis of U87 glioblastoma (GBM) cell clones overexpressing a dominant-negative form of platelet-derived growth factor (PDGF) A subunit to inactive the PDGF autocrine signaling loop. The autocrine PDGF loop is a hallmark of GBM. Results identify genes regulated by oncogenic PDGF signaling.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Analysis of gene expression in mammary epithelial cells transduced with either hTERT, empty LXSN vector or empty BABE vector.\nDate GSE361 Last Updated: May 29 2005\nIncludes GDS337.\n Update date: May 21 2003.\n Dataset description GDS337: Effect of overexpression of the telomerase catalytic subunit (TERT) in mammary epithelial cells (HMEC).  Findings imply that ectopic telomerase expression modulates growth-controlling genes and enhances cell proliferation.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     The purpose of this study is to discover genes that might increase aqueous humor outflow when human ciliary muscle or human trabecular meshwork cells are treated with the prostaglandin analogues latanoprost free acid or prostaglandin F2alpha. Five tissue donors were pooled on each chip.\nDate GSE492 Last Updated: Jul 27 2006\nContributors:  Paul   Russell\nIncludes GDS359.\n Update date: Apr 14 2004.\n Dataset description GDS359: Glaucoma study investigating molecular basis of increase in aqueous humor outflow when human ciliary muscle or human trabecular meshwork cells are treated with prostaglandin analogs latanoprost free acid or prostaglandin F2alpha.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           DNA damage caused by UV radiation initiates cellular recovery mechanisms, which involve activation of DNA damage response pathways, cell cycle arrest and apoptosis. To assess cellular transcriptional responses to UVC-induced DNA damage we compared time course responses of human skin fibroblasts to low and high doses of UVC radiation known to induce a transient cellular replicative arrest or apoptosis, respectively. UVC radiation elicited &gt;3-fold changes in 460 out of 12,000 transcripts and 89% of these represented downregulated transcripts. Only 5% of the regulated genes were common to both low and high doses of radiation. Cells inflicted with a low dose of UVC exhibited transcription profiles demonstrating transient regulation followed by recovery, whereas the responses were persistent after the high dose. A detailed clustering analysis and functional classification of the targets implied regulation of biologically divergent responses and suggested involvement of transcriptional and translational machinery, inflammatory, anti-proliferative and anti-angiogenic responses. The data support the notion that UVC radiation induces prominent, dose-dependent downregulation of transcription. However, the data strongly suggest that transcriptional repression is also target gene selective. Furthermore, the results demonstrate that dose-dependent induction of cell cycle arrest and apoptosis by UVC radiation are transcriptionally highly distinct responses.\nDate GSE713 Last Updated: Jun 15 2005\nContributors:  Massimiliano Gentile Marikki Laiho Leena Latonen\nIncludes GDS400.\n Update date: Oct 08 2003.\n Dataset description GDS400: Temporal analysis of differences in WS1 human skin fibroblast gene expression response to low (10 J/m2; induces transient cellular replicative arrest) or high (50 J/m2; induces apoptosis) doses of short wavelength UV radiation (UVC; 254 nm).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Identification of amyotrophic lateral sclerosis (ALS) associated genes. Post mortem spinal cord grey matter from sporadic and familial ALS patients compared with controls.\nDate GSE833 Last Updated: Jun 28 2005\nContributors:  Steven R Gullans Robert H Brown Fernando Dangond\nIncludes GDS412.\n Update date: Nov 24 2003.\n Dataset description GDS412: Identification of amyotrophic lateral sclerosis (ALS) associated genes. Post mortem spinal cord grey matter from sporadic and familial ALS patients compared with controls. Attempt to identify mechanisms by which ALS destroys motor neurons.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        BACKGROUND: Previous genomic studies with human tissues have compared differential gene expression between 2 conditions (ie, normal versus diseased) to identify altered gene expression in a binary manner; however, a potentially more informative approach is to correlate the levels of gene expression with quantitative physiological parameters. METHODS AND RESULTS: In this study, we have used this approach to examine genes whose expression correlates with arterial stiffness in human aortic specimens. Our data identify 2 distinct groups of genes, those associated with cell signaling and those associated with the mechanical regulation of vascular structure (cytoskeletal-cell membrane-extracellular matrix). Although previous studies have concentrated on the contribution of the latter group toward arterial stiffness, our data suggest that changes in expression of signaling molecules play an equally important role. Alterations in the profiles of signaling molecules could be involved in the regulation of cell cytoskeletal organization, cell-matrix interactions, or the contractile state of the cell. CONCLUSIONS: Although the influence of smooth muscle contraction/relaxation on arterial stiffness could be controversial, our provocative data would suggest that further studies on this subject are indicated.\nDate GSE420 Last Updated: May 29 2005\nContributors:  Richard Pratt Stephane Laurent\nIncludes GDS461.\n Update date: Nov 20 2003.\n Dataset description GDS461: Examination of molecular basis of aortic stiffness which predicts pulse pressure, coronary disease, stroke and cardiovascular mortality. Biopsies from patients with increased aortic stiffness compared with patients with distensible aorta.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Molecular analysis of the effect left ventricular assist device (LVAD) support has on congestive heart failure patients.\nDate GSE430 Last Updated: Jun 15 2005\nContributors:  Robert J Bache Yingjie Chen Li Yunfang Soon Park Leslie W Miller Jennifer Hall Xinqiang Han Emil Missov\nIncludes GDS462.\n Update date: Nov 20 2003.\n Dataset description GDS462: Molecular analysis of effect left ventricular assist device (LVAD) support in left ventricular myocardium tissue of patients with idiopathic dilated cardiomyopathy and end-stage heart failure. Results suggest LVAD support may improve endothelial function.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          The effect of human cytomegalovirus infection on cellular mRNA accumulation was analyzed by gene chip technology over a 48h time course.\nDate GSE675 Last Updated: May 29 2005\nContributors:  Thomas Shenk Edward P Browne Bret Wing David Coleman\nIncludes GDS476.\n Update date: Nov 25 2003.\n Dataset description GDS476: Expression profiles of foreskin fibroblasts at 12 time points beginning 30 minutes after infection by human cytomegalovirus (HCMV) and continuing until 48 hours after infection.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       HeLa cells were serum starved and preincubated with DMSO (vehicle) and infected with CVB3.  Following infection, virus was removed and fresh media containing 10% fetal bovine serum was added for the remainder of the infectious process.  At 0, 30 minutes, 1, 3, 5, 7 and 9 hours following CVB3 infection, RNA was isolated, processed and hybridized to GeneChip®s.\nDate GSE712 Last Updated: Oct 28 2005\nContributors:  Decheng Yang Jingchun Zhang Bobby Yanagawa Nana Rezai Timothy J Triche Bruce McManus Zsuzsanna Hollander Ji Yuan Honglin Luo Raymond T Ng\nIncludes GDS477.\n Update date: Nov 28 2003.\n Dataset description GDS477: Temporal analaysis of an in vitro model of coxsackievirus B3 (CVB3) infection. HeLa cells infected with either CVB3 or control PBS and samples examined at 0, 0.5, 1, 3, 5, 7 and 9 hours following treatment.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Time series for gene expression changes following 3 Gy and 10 Gy of ionizing radiation exposure.\nDate GSE701 Last Updated: May 29 2005\nContributors:  Vivian G Cheung Kuang-Yu Jen\nIncludes GDS479.\n Update date: Nov 28 2003.\n Dataset description GDS479: Temporal analysis of effect of 3 Gy and 10 Gy ionizing radiation (IR) exposure on lymphoblastoid cells. Various time points up to 24 hours examined.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           The vertebrate homologues of Drosophila dachsund, DACH1 and DACH2, have been implicated as important regulatory genes in development. DACH1 plays a role in retinal and pituitary precursor cell proliferation and DACH2 plays a specific role in myogenesis. DACH proteins contain a domain (DS-domain) that is conserved with the proto-oncogenes Ski and Sno. Since the Ski/Sno proto-oncogenes repress AP-1 and SMAD signaling, we hypothesized that DACH1 might play a similar cellular function. Herein, DACH1 was found to be expressed in breast cancer cell lines and to inhibit TGF-beta induced apoptosis. DACH1 repressed TGF-beta induction of AP-1 and Smad signaling in gene reporter assays and repressed endogenous TGF-beta responsive genes by microarray analyses. DACH1 bound to endogenous NCoR and Smad4 in cultured cells and DACH1 co-localized with NCoR in nuclear dot-like structures. NCoR enhanced DACH1 repression and the repression of TGF-beta-induced AP-1 or Smad-signaling by DACH1 required the DACH1 DS domain. The DS-domain of DACH was sufficient for NCoR-binding at a Smad4-binding site. Smad4 was required for DACH1 repression of Smad signaling. In Smad4 null HTB-134 cells, DACH1 inhibited the activation of SBE-4 reporter activity induced by Smad2 or Smad3 only in the presence of Smad4. DACH1 participates in the negative regulation of TGF-beta signaling by interacting with NCoR and Smad4.\nDate GSE685 Last Updated: Jun 15 2005\nContributors:  Kveta Cveklova Maria A Davoli Richard G Pestell Chenguang Wang Kongming Wu Mark D'Amico Robert G Russell Anping Li Zbynek Kozmik Ying Yang Michael P Lisanti Ales Cvekl\nIncludes GDS483.\n Update date: Dec 01 2003.\n Dataset description GDS483: Analysis of effect of DACH1 in breast cancer cell line MDA-MB-231. DACH1 induced by ponasterone A treatment for 0, 18 or 36 hours.  DACH1 may regulate aberrant TGF beta signals that have role in breast cancer progression.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Our study seeks to identify genes differentially expressed between uterine leiomyomas and normal myometrial tissue. This series consists of samples derived from normal myometrium and uterine leiomyomas obtained from fibroid afflicted patients.Total RNA was extracted from samples, converted to biotin-labeled cRNA, hybridized to oligonucleotide arrays, and followed by model based expression analysis.  In order to select differentially expressed genes of interest, dChip model-based expression analysis was employed. Significant genes were identified, utilizing the dChip software, as having an average fold change of &gt; +1.5 or &lt; -1.5 between leiomyoma and normal myometrium and p &lt; 0.05. Under these conditions the 226 genes in the following list were identified.\nDate GSE593 Last Updated: May 29 2005\nContributors:  Dawn B Milliken Ryan R Davis Jeffrey P Gregg Paul J Hoffman\nIncludes GDS484.\n Update date: Dec 01 2003.\n Dataset description GDS484: Comparison of normal myometrium and uterine leiomyomas obtained from fibroid afflicted patients.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  HUVECs (human umbilical cord vein endothelial cells) are treated with the angiogenic factors VEGF-A (vascular endothelial growth factor-A) and PlGF (placental growth factor) in low or high serum media.\nDate GSE837 Last Updated: May 29 2005\nIncludes GDS495.\n Update date: Mar 01 2004.\n Dataset description GDS495: Temporal analysis of human umbilical cord vein endothelial cell (HUVEC) isolates treated with angiogenic factors vascular endothelial growth factor-A (VEGF-A) and placental growth factor (PlGF) in low or high serum media.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Comparison of gene expression for individuals affected with FCHL exhibiting the USF1 susceptibility haplotype and FCHL affected indiviuals carrying the protective haplotype.\nDate GSE590 Last Updated: Oct 28 2005\nContributors:  Massimiliano Gentile Leena Peltonen Päivi Pajukanta Rita M Cantor Janet S Sinsheimer Aldons J Lusis Heidi E Lilja\nIncludes GDS513.\n Update date: Mar 18 2004.\n Dataset description GDS513: Comparison of subcutaneous adipose tissue from individuals with familial combined hyperlipidemia (FCHL) exhibiting either upstream transcription factor 1 (USF1) susceptibility or protective haplotype. USF1 regulates glucose and lipid metabolism genes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  This series represents samples of multiple myeloma patients with and without bone lytic lesion by MRI.\nDate GSE755 Last Updated: May 29 2005\nContributors:  John D Shaughnessy Erik Rasmussen Yupo Ma Bart Barlogie Fenghuang Zhan Ronald Walker Erming Tian\nIncludes GDS531.\n Update date: Mar 29 2004.\n Dataset description GDS531: Comparison of gene expression in bone marrow plasma cells of multiple myeloma patients with and without bone lesions. Osteolytic lesions increase in multiple myeloma patients.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 A number of studies have shown that cigarette smoking produces a field defect, such that genetic mutations induced by smoking occur throughout the lung and its intra and extra-pulmonary airways. Based on this concept, we have begun this study, which has as its goal the definition of the normal airway transcriptome, an analysis of how that transcriptome is affected by cigarette smoke, and to explore the reversibility of altered gene expression when smoking has been discontinued. We have obtained brushings from intra-pulmonary airways (the right upper lobe carina) and scrapings from the buccal mucosa, from normal smoking and non-smoking volunteers (including 34 Current Smokers, 23 Never Smokers and 18 Former Smokers). RNA was isolated from these samples and gene expression profiles from intra-pulmonary airway epithelial cells were analyzed using Affymetrix U133A human gene expression arrays. All microarray data from the experiments described above have been stored, preprocessed and analyzed in a relational MySQL database that is accessible through our website at http://pulm.bumc.bu.edu/aged.\nDate GSE994 Last Updated: May 29 2005\nContributors:  Jennifer Beane Frank Schembri John Palma Gang Liu Jerome Brody Avrum Spira Xuemei Yang Vishal Shah\nIncludes GDS534.\n Update date: May 02 2004.\n Dataset description GDS534: Analysis of cigarette smoking-induced changes in  bronchial epithelia, and reversibility of effects when smoking is discontinued. May provide insight to molecular events leading to chronic obstructive pulmonary disease (COPD) and lung cancer.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     AR overexpression converts antagonists to weak agonists.\nDate GSE846 Last Updated: May 29 2005\nContributors:  Derek Welsbie Charles Sawyers Charlie Chen\nIncludes GDS536.\n Update date: Mar 30 2004.\n Dataset description GDS536: Examination of antagonist to agonist conversion in androgen receptor-expressing hormone-sensitive LNCaP prostate cancer cells. Cells challenged with increasing doses of R1881, or bicalutamide.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE              GSE2018                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE              GSE4036                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE              GSE3489                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE              GSE1923                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE               GSE361                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE               GSE492                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE               GSE713                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE               GSE833                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE               GSE420                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE               GSE430                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE               GSE675                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE               GSE712                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE               GSE701                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE               GSE685                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE               GSE593                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE               GSE837                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE               GSE590                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE               GSE755                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE               GSE994                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE               GSE846                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                 experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                         &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE2018</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE4036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE3489</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE1923</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE361</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE492</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE713</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE833</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE420</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE430</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE675</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE712</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE701</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE685</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE593</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE837</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE755</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE994</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE846</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                 experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      &lt;int&gt;                 &lt;POSc&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                     34    2024-02-10 08:31:41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                     28    2023-12-19 13:29:05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                     28    2023-12-19 05:05:46</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                     15    2023-12-17 18:06:41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                     20    2023-12-19 06:49:29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                     12    2023-12-20 02:28:26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                     20    2023-12-20 22:21:41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                     11    2023-12-21 05:48:43</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                     10    2023-12-19 15:53:04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                     14    2023-12-19 17:18:40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                     12    2024-07-24 07:30:20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                     13    2023-12-20 22:15:09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                     12    2023-12-20 21:33:32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                      9    2023-09-07 21:30:56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                     10    2023-12-20 12:42:48</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                     33    2023-12-21 06:01:54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                     10    2023-12-20 12:28:23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                    173    2023-12-21 01:05:02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                     75    2023-12-21 14:29:34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                     17    2023-12-21 07:05:17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         experiment.batchEffectText experiment.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             &lt;char&gt;                    &lt;lgcl&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:        NO_BATCH_EFFECT_SUCCESS                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:           SINGLE_BATCH_SUCCESS                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:        BATCH_CORRECTED_SUCCESS                      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:        BATCH_CORRECTED_SUCCESS                      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: PROBLEMATIC_BATCH_INFO_FAILURE                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:        BATCH_CORRECTED_SUCCESS                      TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                  NO_BATCH_INFO                     FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         experiment.batchEffectText experiment.batchCorrected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.batchConfound experiment.batchEffect experiment.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        &lt;num&gt;                  &lt;num&gt;              &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                        1                      1                  1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                        1                      1                  1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                        1                      1                  1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                        1                      1                  1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                        1                      1                  1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                        0                      0                 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.batchConfound experiment.batchEffect experiment.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.name taxon.scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;num&gt;       &lt;num&gt;     &lt;char&gt;           &lt;char&gt;    &lt;int&gt;      &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:   0.9976341      0.8750      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:   0.9971246      0.7500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:   0.4098798      0.5000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:   0.5706032      0.6250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:   0.4063306      0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:   0.8561848      0.3750      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:   0.2669028      0.5000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:  -0.0616038      0.0000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:   0.2512567      0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:   0.4099263      0.5000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:   0.1303742      0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:   0.1302309      0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:   0.1283844      0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:   0.2789924      0.4625      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:   0.4050721      0.2500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:  -0.1569077      0.5000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:   0.2736367      0.5000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:   0.4021395      0.2500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:   0.9969304      0.8750      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:   0.1346832      0.5000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.name taxon.scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.name taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;char&gt;             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.name taxon.database.ID</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

