<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve all experiments using a given platform — get_platform_datasets • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve all experiments using a given platform — get_platform_datasets"><meta property="og:description" content="Retrieve all experiments using a given platform"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.1.12</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve all experiments using a given platform</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/devel/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>get_platform_datasets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieve all experiments using a given platform</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_platform_datasets</span><span class="op">(</span></span>
<span>  <span class="va">platform</span>,</span>
<span>  offset <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-platform">platform<a class="anchor" aria-label="anchor" href="#arg-platform"></a></dt>
<dd><p>A platform numerical identifier or a platform short name</p></dd>


<dt id="arg-offset">offset<a class="anchor" aria-label="anchor" href="#arg-offset"></a></dt>
<dd><p>The offset of the first retrieved result.</p></dd>


<dt id="arg-limit">limit<a class="anchor" aria-label="anchor" href="#arg-limit"></a></dt>
<dd><p>Defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt id="arg-raw">raw<a class="anchor" aria-label="anchor" href="#arg-raw"></a></dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt id="arg-memoised">memoised<a class="anchor" aria-label="anchor" href="#arg-memoised"></a></dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt id="arg-file">file<a class="anchor" aria-label="anchor" href="#arg-file"></a></dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt id="arg-overwrite">overwrite<a class="anchor" aria-label="anchor" href="#arg-overwrite"></a></dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A data table with information about the queried dataset(s). A list if
<code>raw = TRUE</code>. Returns an empty list if no datasets matched.</p>
<p>The fields of the output data.table are:</p>
<ul><li><p><code>experiment.shortName</code>: Shortname given to the dataset within Gemma. Often corresponds to accession ID</p></li>
<li><p><code>experiment.name</code>: Full title of the dataset</p></li>
<li><p><code>experiment.ID</code>: Internal ID of the dataset.</p></li>
<li><p><code>experiment.description</code>: Description of the dataset</p></li>
<li><p><code>experiment.troubled</code>: Did an automatic process within gemma or a curator mark the dataset as "troubled"</p></li>
<li><p><code>experiment.accession</code>: Accession ID of the dataset in the external database it was taken from</p></li>
<li><p><code>experiment.database</code>: The name of the database where the dataset was taken from</p></li>
<li><p><code>experiment.URI</code>: URI of the original database</p></li>
<li><p><code>experiment.sampleCount</code>: Number of samples in the dataset</p></li>
<li><p><code>experiment.batchEffectText</code>: A text field describing whether the dataset has batch effects</p></li>
<li><p><code>experiment.batchCorrected</code>: Whether batch correction has been performed on the dataset.</p></li>
<li><p><code>experiment.batchConfound</code>: 0 if batch info isn't available, -1 if batch counfoud is detected, 1 if batch information is available and no batch confound found</p></li>
<li><p><code>experiment.batchEffect</code>: -1 if batch p value &lt; 0.0001, 1 if batch p value &gt; 0.1, 0 if otherwise and when there is no batch information is available or when the data is confounded with batches.</p></li>
<li><p><code>experiment.rawData</code>: -1 if no raw data available, 1 if raw data was available. When available, Gemma reprocesses raw data to get expression values and batches</p></li>
<li><p><code>geeq.qScore</code>: Data quality score given to the dataset by Gemma.</p></li>
<li><p><code>geeq.sScore</code>: Suitability score given to the dataset by Gemma. Refers to factors like batches, platforms and other aspects of experimental design</p></li>
<li><p><code>taxon.name</code>: Name of the species</p></li>
<li><p><code>taxon.scientific</code>: Scientific name for the taxon</p></li>
<li><p><code>taxon.ID</code>: Internal identifier given to the species by Gemma</p></li>
<li><p><code>taxon.NCBI</code>: NCBI ID of the taxon</p></li>
<li><p><code>taxon.database.name</code>: Underlying database used in Gemma for the taxon</p></li>
<li><p><code>taxon.database.ID</code>: ID of the underyling database used in Gemma for the taxon</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">get_platform_datasets</span><span class="op">(</span><span class="st">"GPL1355"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:             GSE40348</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:             GSE14903</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:             GSE40421</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:              GSE2872</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:             GSE74676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:             GSE19366</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                      experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                               &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:                                                                                                                    Hepatotoxicity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:                                    The relationship between brain mRNA levels and behaviour among inbred Lewis rats: Experiment 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:                                                                 Generation of oligodendroglial cells by direct lineage conversion</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:                                                                                                             d'mel-affy-rat-168311</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5: Toxicogenomic Characterization of Molecular Mechanisms Contributing to Chlorpyrifos Neurotoxicity in Adult Male Rats [microarray]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:                    Genomic-Derived Markers for Early Detection of Calcineurin Inhibitor Immunosuppressant–Mediated Nephrotoxicity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:          5633</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:         12033</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:         12289</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:             2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:         13570</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:          5124</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1: Summary from subseries GSE40336:  This study provides an evaluation of changes in gene expression associated with acetominophen treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses (500 uM and 5 mM) of acetominophen and 1% DMSO vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nSummary from subseries GSE40338:  This study provides an evaluation of changes in gene expression associated with sodium valproate treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses (500 uM and 10 mM) of sodium valproate and water vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nSummary from subseries GSE40337:  This study provides an evaluation of changes in gene expression associated with dioctyl phthalate treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses (250 uM and 1 mM) of dioctyl phthalate and 1% DMSO vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nSummary from subseries GSE40339:  This study provides an evaluation of changes in gene expression associated with phenobarbital treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses (300 uM and 3 mM) of phenobarbital and water vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nSummary from subseries GSE40347:  This study provides an evaluation of changes in gene expression associated with methapyrilene treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses (3 uM and 100 uM) of methaphyriline and 1% DMSO vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nSummary from subseries GSE40346:  This study provides an evaluation of changes in gene expression associated with WY-14643 treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses 8 uM and 200 uM) of WY-14643 and 1% DMSO vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nSummary from subseries GSE40340:  This study provides an evaluation of changes in gene expression associated with beta-naphthaflavone treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses 1 uM and 100 mM) of beta-naphthaflavone and 1% DMSO vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nSummary from subseries GSE40341:  This study provides an evaluation of changes in gene expression associated with chlorpromazine HCl treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses (0.8 uM and 20 uM) of chlorpromazine HCl and 1% DMSO vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nSummary from subseries GSE40342:  This study provides an evaluation of changes in gene expression associated with diisononyl phthalate treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses (25 mM and 100 mM) of diisononyl phthalate and 1% DMSO vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nSummary from subseries GSE40344:  This study provides an evaluation of changes in gene expression associated with clofibrate treatment of rat hepatocytes in vitro. Primary rat hepatocytes were treated for 24 and 48 hours with two doses (60 uM and 1 mM) of clobibrate and 1% DMSO vehicle control.  Five replicates of each treatment were performed.  Cells were then extracted and RNA processed for microarray analysis.\nLast Updated (by provider): Aug 28 2012\nContributors:  Jay P Tiesman</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Behaviour is derived from complex molecular interactions within the brain, in response to specific environmental stimuli. In some rare cases, the direct causes of behaviour have been attributed to the interactions of a single or small group of gene transcripts and proteins. We conducted two experiments with the hope of defining some of the molecular interactors for four separate behaviours: sugar feeding, locomotor activity in a novel field, and acoustic startle reflex, and prepulse inhibition of the acoustic startle reflex, which have been linked to prefrontal cortex dopaminergic function or as predictors of sensorimotor gating in diseases such as schizophrenia. Rats with high and low response phenotypes were selected to determine the differences between these two extremes of behaviour. From our analyses, transcripts of Homer1, a neuronal scaffolding protein which interacts with group1 metabotropic glutamate receptors, were found to be significantly correlated with array data in both experiments, and with behaviour data across three separate tests in the second experiment, indicating that this gene's transcripts and probably downstream protein interactions have a significant correlation with behaviour phenotype in the inbred Lewis rat. Future areas of pursuit for this data should involve modification of the expression of Homer1 in an isolated fashion to determine a pharmacological threshold for differences in behaviour. \nLast Updated (by provider): Jul 31 2017\nContributors:  Susan Rotzinger Arturas Petronis Albert H Wong Franco Vaccarino Laura A Feldcamp Mawahib O Semeralul Paul C Boutros</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       We report the generation of induced oligodendrocyte precursor cells (iOPCs) by direct lineage conversion. Forced expression of the three transcription factors Sox10, Olig2 and Zfp536 was sufficient to convert mouse and rat fibroblasts into iOPCs with morphologies and gene expression signatures that resemble OPCs.\nLast Updated (by provider): Jul 31 2017\nContributors:  Samuele Marro Nan Yang Marius Wernig</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Chlorpyrifos (CPF) is an organophosphorus pesticide (OP), and one of the most widely used pesticides in the world. Metabolites of CPF and other OPs continue to be identified in the majority of human samples, even in countries such as the United States where OP use is declining (Arcury et al., 2010). The effects of repeated occupational and environmental exposures to OPs are poorly understood, although human and animal studies consistently identify neurotoxicity as the primary endpoint of concern.  Thus, occupational exposures to sublethal doses of CPF are consistently associated with problems in cognitive abilities, such as learning and memory but the biological mechanism(s) underlying this association remain speculative.  To identify potential mechanisms of CPF neurotoxicity, we employed a rat model that simulated documented CPF exposures in Egyptian agricultural workers. We quantified mRNA expression profiles in the CA1 region of the hippocampus of adult male Long Evans (LE) rats administered CPF at 3 or 10 mg/kg/d (s.c.)  for 21 days. Despite significant inhibition of cholinesterase activity by the end of the 21 d exposure period, the CPF-exposed rats displayed minimal signs of cholinergic toxicity. Distinct hippocampal mRNA and miRNA signatures were associated with CPF exposure. Toxicogenomics-based evidence identified increased expression of neuropeptide genes in the hippocampi of CPF-exposed rats, which have been shown to activate receptor-mediated signaling pathways involved in cell survival. The analysis of small non-coding RNA profiles suggested the possibility that miR132/212-mediated homeostatic regulatory pathways may also be activated by repeated exposures to CPF. These findings identify potential molecular effects that may contribute to neurobehavioral deficits.\nLast Updated (by provider): Jul 31 2017\nContributors:  John A Lewis Jonathan D Stallings Naissan Hussainzada Pamela J Lein Young S Lee David A Jackson</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            The use of calcineurin inhibitor (CI) immunosuppressants has significantly improved the early allograft survival rate in organ transplantation. However, CI therapy has been associated with chronic nephrotoxicity, which limits their long-term utility. In order to understand the mechanisms of the toxicity, we analyzed the gene expression changes that underlie the development of CI immunosuppressant-mediated nephrotoxicity, in male Sprague-Dawley (SD) rats dosed daily with cyclosporine (CsA), FK506 or rapamycin (Rapa) for 1 to 28 days. We identified a group of genes, whose expression in rat kidney is quantitatively correlated with CI-induced kidney injury as observed in changes in blood urea nitrogen (BUN) levels and kidney histopathology. These genes include both up-regulated genes, such as Ren1 and Klks3, and down-regulated genes, such as Calb1, Egf, NCC, and kidney specific Wnk1 (KS-Wnk1). Using the down-regulated genes alone we successfully predicted CI immunosuppressant-mediated kidney injury in rats following 7 days of treatment. Among these genes are two mechanism-related genes, NCC and KS-Wnk1, both of which are involved in the sodium transport in the distal nephrons. The down-regulation of both genes at the mRNA and protein level in rat kidney following CI treatment was confirmed by quantitative RT-PCR and immunohistochemical staining, respectively. We hypothesize that decreased expression of NCC may cause reduced sodium chloride reabsorption in the distal tubules, and contribute to the prolonged activation of the Renin-Angiotensin-System (RAS), a demonstrated contributor to the development of CI-induced nephrotoxicity in both animal models and clinical settings. Therefore, NCC and KS-Wnk1 could potentially be used as biomarkers for early detection and prevention of CI-related nephrotoxicity in clinical practice.\nLast Updated (by provider): Mar 21 2012\nContributors:  James T Auman Richard S Paules Jeff Chou Qihong Huang Supriya Jayadev Yuxia Cui</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:               FALSE             GSE40348                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:               FALSE             GSE14903                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:               FALSE             GSE40421                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:               FALSE              GSE2872                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:               FALSE             GSE74676                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:               FALSE             GSE19366                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                 experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                         &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE40348</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE14903</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE40421</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE2872</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE19366</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    experiment.sampleCount experiment.lastUpdated experiment.batchEffectText</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                     &lt;int&gt;                 &lt;POSc&gt;                     &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:                    300    2023-12-19 13:26:42       BATCH_EFFECT_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:                     10    2023-12-17 12:18:16       SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:                      9    2024-03-18 16:21:27       BATCH_EFFECT_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:                     12    2024-01-02 18:57:36       SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:                     60    2023-12-21 00:32:14    BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:                    120    2023-12-17 18:53:19       BATCH_EFFECT_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    experiment.batchCorrected experiment.batchConfound experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       &lt;lgcl&gt;                    &lt;num&gt;                  &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:                     FALSE                       -1                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:                     FALSE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:                     FALSE                        1                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:                     FALSE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:                      TRUE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:                     FALSE                       -1                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    experiment.rawData geeq.qScore geeq.sScore taxon.name  taxon.scientific</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;num&gt;       &lt;num&gt;       &lt;num&gt;     &lt;char&gt;            &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:                  1   0.2845997      0.6250        rat Rattus norvegicus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:                  1   0.8563497      0.5000        rat Rattus norvegicus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:                  1   0.5511185      0.7125        rat Rattus norvegicus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:                  1   0.8552303      0.5000        rat Rattus norvegicus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:                  1   0.9987732      0.6250        rat Rattus norvegicus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:                  1   0.4268694      0.8750        rat Rattus norvegicus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    taxon.ID taxon.NCBI taxon.database.name taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;int&gt;      &lt;int&gt;              &lt;char&gt;             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:        3      10116                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:        3      10116                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:        3      10116                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:        3      10116                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:        3      10116                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:        3      10116                 rn6                86</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

