<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve annotations and surface level stats for a dataset's differential analyses — get_dataset_differential_expression_analyses • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve annotations and surface level stats for a dataset's differential analyses — get_dataset_differential_expression_analyses"><meta property="og:description" content="Retrieve annotations and surface level stats for a dataset's differential analyses"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve annotations and surface level stats for a dataset's differential analyses</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>get_dataset_differential_expression_analyses.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieve annotations and surface level stats for a dataset's differential analyses</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_dataset_differential_expression_analyses</span><span class="op">(</span></span>
<span>  <span class="va">dataset</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dataset</dt>
<dd><p>A numerical dataset identifier or a dataset short name</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data table with information about the differential expression
analysis of the queried dataset. Note that this funciton does not return
differential expression values themselves. Use <code><a href="get_differential_expression_values.html">get_differential_expression_values</a></code></p>


<p>to get differential expression values (see examples).</p>


<p>The fields of the output data.table are:</p>


<ul><li><p><code>result.ID</code>: Result set ID of the differential expression analysis.
May represent multiple factors in a single model.</p></li>
<li><p><code>contrast.ID</code>: Id of the specific contrast factor. Together with the result.ID
they uniquely represent a given contrast.</p></li>
<li><p><code>experiment.ID</code>: Id of the source experiment</p></li>
<li><p><code>factor.category</code>: Category for the contrast</p></li>
<li><p><code>factor.category.URI</code>: URI for the contrast category</p></li>
<li><p><code>factor.ID</code>: ID of the factor</p></li>
<li><p><code>baseline.factors</code>: Characteristics of the baseline. This field is a data.table</p></li>
<li><p><code>experimental.factors</code>: Characteristics of the experimental group. This field is a data.table</p></li>
<li><p><code>isSubset</code>: TRUE if the result set belong to a subset, FALSE if not. Subsets are created when performing differential expression to avoid unhelpful comparisons.</p></li>
<li><p><code>subsetFactor</code>: Characteristics of the subset. This field is a data.table</p></li>
<li><p><code>probes.analyzed</code>: Number of probesets represented in the contrast</p></li>
<li><p><code>genes.analyzed</code>: Number of genes represented in the contrast</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">get_dataset_differential_expression_analyses</span><span class="op">(</span><span class="st">"GSE2872"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="get_differential_expression_values.html">get_differential_expression_values</a></span><span class="op">(</span>resultSet <span class="op">=</span> <span class="va">result</span><span class="op">$</span><span class="va">result.ID</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $`570867`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               Probe NCBIid GeneSymbol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;char&gt; &lt;char&gt;     &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1:   1368058_at  64196       Safb</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     2:   1386270_at                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     3:   1368851_at  24356       Ets1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     4:   1395939_at                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5:   1381999_at 296580    Camsap1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ---                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31069: 1393379_a_at                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31070:   1377975_at                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31071:   1382534_at                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31072:   1389727_at 314848     Lrrc10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31073:   1392428_at 360604    Spata20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                  GeneName pvalue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                    &lt;char&gt;  &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1:                       scaffold attachment factor B 0.9716</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     2:                                                    0.1520</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     3:         ETS proto-oncogene 1, transcription factor 0.6364</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     4:                                                    0.6581</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5: calmodulin regulated spectrin-associated protein 1 0.5268</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ---                                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31069:                                                    0.4792</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31070:                                                    0.3199</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31071:                                                    0.5406</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31072:                  leucine-rich repeat-containing 10 0.5045</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31073:                      spermatogenesis associated 20 0.6297</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        corrected_pvalue   rank contrast_134441_log2fc contrast_134441_tstat</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;num&gt;  &lt;num&gt;                  &lt;num&gt;                 &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1:           0.9999 0.9655                 0.0434                0.3027</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     2:           0.9999 0.0911                -0.0748               -0.9062</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     3:           0.9999 0.5707                -0.0850               -0.5670</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     4:           0.9999 0.5957                 0.0411                0.4143</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5:           0.9999 0.4488                 0.0158                0.0797</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ---                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31069:           0.9999 0.3961                -0.0235               -0.1884</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31070:           0.9999 0.2388                 0.0924                0.6052</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31071:           0.9999 0.4637                -0.0054               -0.0561</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31072:           0.9999 0.4249                -0.0593               -0.5642</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31073:           0.9999 0.5629                 0.0828                0.5923</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        contrast_134441_pvalue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1:                 0.7676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     2:                 0.3839</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     3:                 0.5819</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     4:                 0.6865</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     5:                 0.9379</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ---                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31069:                 0.8539</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31070:                 0.5572</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31071:                 0.9562</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31072:                 0.5838</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31073:                 0.5654</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.8.</p>
</div>

      </footer></div>

  


  

  </body></html>

