<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Dataset samples — getDatasetSamples • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Dataset samples — getDatasetSamples"><meta property="og:description" content="Retrieves samples for the given dataset"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.26</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Dataset samples</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>getDatasetSamples.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieves samples for the given dataset</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getDatasetSamples</span><span class="op">(</span></span>
<span>  <span class="va">dataset</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dataset</dt>
<dd><p>Required, part of the URL path.
Can either be the dataset ID or its short name (e.g. <code>GSE1234</code>).
Retrieval by ID is more efficient.
Only datasets that user has access to will be available</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable parsing. Raw results usually contain additional fields and flags that are omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the same inputs and use the result saved in cache if a result is already saved. Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always used. Use <code><a href="forgetGemmaMemoised.html">forgetGemmaMemoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write results to a file. If <code>raw == TRUE</code>, the output will be a JSON file. Otherwise, it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data table with information about the samples of the queried dataset. A <code>404 error</code> if the given identifier does not map to any object.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">getDatasetSamples</span><span class="op">(</span><span class="st">"GSE2018"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   bioMaterial.Name  sample.Name sample.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:  GSE2018_bioMaterial_4|GSM36426   BAL_1_A0B0        31</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2: GSE2018_bioMaterial_15|GSM36437 BAL_27a_A0B0        20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3: GSE2018_bioMaterial_26|GSM36448  BAL_43_A0B0         9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:  GSE2018_bioMaterial_6|GSM36428  BAL_13_A0B0        29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5: GSE2018_bioMaterial_17|GSM36439  BAL_29_A1B0        18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6: GSE2018_bioMaterial_28|GSM36450 BAL_44b_A0B0         7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                 sample.Correspondence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:                                Corresponding to Samples:  GSM36426</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:                                Corresponding to Samples:  GSM36437</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3: Corresponding to samples:  ial corresponding to samples:  GSM36448</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:                                Corresponding to Samples:  GSM36428</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:                                Corresponding to Samples:  GSM36439</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6: Corresponding to samples:  ial corresponding to samples:  GSM36450</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                       sample.Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:                                                             Bronchoalveolar lavage cells from a patient with an ISHLT biopsy score of A=0 and B=0 (Non-acute rejection).\nSource GEO sample is GSM36426\nLast updated (according to GEO): May 27 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:  Bronchoalveolar lavage cells from a patient with an ISHLT biopsy score of A=0 and B=0 (Non-acute rejection). One of two samples from the same patient, different visits\nSource GEO sample is GSM36437\nLast updated (according to GEO): May 27 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:                                                             Bronchoalveolar lavage cells from a patient with an ISHLT biopsy score of A=0 and B=0 (Non-acute rejection).\nSource GEO sample is GSM36448\nLast updated (according to GEO): May 27 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:                                                             Bronchoalveolar lavage cells from a patient with an ISHLT biopsy score of A=0 and B=0 (Non-acute rejection).\nSource GEO sample is GSM36428\nLast updated (according to GEO): May 27 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:                                                             Bronchoalveolar lavage cells from a patient with an ISHLT biopsy score of A=1 and B=0 (Non-acute rejection).\nSource GEO sample is GSM36439\nLast updated (according to GEO): May 27 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:  Bronchoalveolar lavage cells from a patient with an ISHLT biopsy score of A=0 and B=0 (Non-acute rejection). One of two samples from the same patient, different visits\nSource GEO sample is GSM36450\nLast updated (according to GEO): May 27 2005</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample.Outlier sample.Accession sample.Processed sample.Characteristics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:          FALSE         GSM36426     1.036695e+12      &lt;data.table[2x4]&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:          FALSE         GSM36437     1.036693e+12      &lt;data.table[2x4]&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:          FALSE         GSM36448     1.035573e+12      &lt;data.table[2x4]&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:          FALSE         GSM36428     1.022102e+12      &lt;data.table[2x4]&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:          FALSE         GSM36439     1.033757e+12      &lt;data.table[2x4]&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:          FALSE         GSM36450     1.018980e+12      &lt;data.table[2x4]&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    sample.FactorValues array.ShortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:   &lt;data.table[2x4]&gt;           GPL96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:   &lt;data.table[2x4]&gt;           GPL96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:   &lt;data.table[2x4]&gt;           GPL96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:   &lt;data.table[2x4]&gt;           GPL96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:   &lt;data.table[2x4]&gt;           GPL96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:   &lt;data.table[2x4]&gt;           GPL96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                  array.Name array.Taxon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1: Affymetrix GeneChip Human Genome U133 Array Set HG-U133A          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2: Affymetrix GeneChip Human Genome U133 Array Set HG-U133A          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3: Affymetrix GeneChip Human Genome U133 Array Set HG-U133A          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4: Affymetrix GeneChip Human Genome U133 Array Set HG-U133A          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5: Affymetrix GeneChip Human Genome U133 Array Set HG-U133A          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6: Affymetrix GeneChip Human Genome U133 Array Set HG-U133A          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    array.TaxonID array.Type array.Description array.Troubled</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:            NA         NA                NA          FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:            NA         NA                NA          FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:            NA         NA                NA          FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:            NA         NA                NA          FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:            NA         NA                NA          FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:            NA         NA                NA          FALSE</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

