<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Genes at location — getGenesAtLocation • gemmaAPI</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Genes at location — getGenesAtLocation" />
<meta property="og:description" content="Finds genes overlapping a given region." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemmaAPI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gemmaAPI.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/async.html">Asynchronous Requests (advanced)</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/PavlidisLab/Gemma-API/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Genes at location</h1>
    <small class="dont-index">Source: <a href='https://github.com/PavlidisLab/Gemma-API/blob/master/R/allEndpoints.R'><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>getGenesAtLocation.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Finds genes overlapping a given region.</p>
    </div>

    <pre class="usage"><span class='fu'>getGenesAtLocation</span><span class='op'>(</span>
  taxon <span class='op'>=</span> <span class='cn'>NA_character_</span>,
  chromosome <span class='op'>=</span> <span class='cn'>NA_character_</span>,
  strand <span class='op'>=</span> <span class='st'>"+"</span>,
  start <span class='op'>=</span> <span class='cn'>NA_integer_</span>,
  size <span class='op'>=</span> <span class='cn'>NA_integer_</span>,
  raw <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.raw"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  async <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.async"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  memoised <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.memoise"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  file <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.file"</span>, <span class='cn'>NA_character_</span><span class='op'>)</span>,
  overwrite <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.overwrite"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>taxon</th>
      <td><p>Not required, part of the URL path.
can either be Taxon ID, Taxon NCBI ID, or one of its string identifiers:
scientific name, common name
It is recommended to use Taxon ID for efficiency.
Please note, that not all taxa have all the possible identifiers
available.
Use the 'All Taxa' endpoint to retrieve the necessary information. For
convenience, below is a list of officially supported taxa:
ID   Comm.name   Scient.name                NcbiID</p>
<p>1    human       Homo sapiens               9606
2    mouse       Mus musculus               10090
3    rat         Rattus norvegicus          10116
11   yeast       Saccharomyces cerevisiae   4932
12   zebrafish   Danio rerio                7955
13   fly         Drosophila melanogaster    7227
14   worm        Caenorhabditis elegans     6239</p></td>
    </tr>
    <tr>
      <th>chromosome</th>
      <td><p>Required, defaults to <code>empty</code>.
The chromosome of the query location. Eg: <code>3</code>, <code>21</code>, <code>X</code></p></td>
    </tr>
    <tr>
      <th>strand</th>
      <td><p>Optional, defaults to <code><a href='https://rdrr.io/r/base/Arithmetic.html'>+</a></code>.
Can either be <code><a href='https://rdrr.io/r/base/Arithmetic.html'>+</a></code> or <code><a href='https://rdrr.io/r/base/Arithmetic.html'>-</a></code>.
This is a
WIP parameter and does currently not do anything
When
using in scripts, remember to URL-encode the '+' plus character (see
the compiled URL below).</p></td>
    </tr>
    <tr>
      <th>start</th>
      <td><p>Required, defaults to <code>empty</code>.
Number of the start nucleotide of the desired region.</p></td>
    </tr>
    <tr>
      <th>size</th>
      <td><p>Required, defaults to <code>empty</code>.
Amount of nucleotides in the desired region (i.e. the length of the
region).</p></td>
    </tr>
    <tr>
      <th>raw</th>
      <td><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing.</p></td>
    </tr>
    <tr>
      <th>async</th>
      <td><p><code>TRUE</code> to run the API query on a separate worker, or <code>FALSE</code> to run
synchronously. See the <code>async</code> package for details.</p></td>
    </tr>
    <tr>
      <th>memoised</th>
      <td><p>Whether or not to cache results so future requests for the same data
will be faster. Use <code>forgetGemmaMemoised</code> to clear the cache.</p></td>
    </tr>
    <tr>
      <th>file</th>
      <td><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be a JSON file.
Otherwise, it will be a RDS file.</p></td>
    </tr>
    <tr>
      <th>overwrite</th>
      <td><p>Whether or not to overwrite if a file exists at the specified filename.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An array of genes (gene value objects) that overlap the given region.
A <code>404 error</code> if the given identifier does not map to any object.
A <code>400 error</code> if required parameters are missing.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>getGenesAtLocation</span><span class='op'>(</span><span class='st'>"human"</span>, chromosome <span class='op'>=</span> <span class='fl'>21</span>, strand <span class='op'>=</span> <span class='st'>"+"</span>, start <span class='op'>=</span> <span class='fl'>2</span>, size <span class='op'>=</span> <span class='fl'>20000</span><span class='op'>)</span>
</div><div class='output co'>#&gt;     gene.Symbol    gene.Ensembl gene.NCBI
#&gt; 1:    LOC728312            &lt;NA&gt;    728312
#&gt; 2:        PRED4            &lt;NA&gt;    400858
#&gt; 3:    LOC642398            &lt;NA&gt;    642398
#&gt; 4: LOC100129305            &lt;NA&gt; 100129305
#&gt; 5: LOC100134011            &lt;NA&gt; 100134011
#&gt; 6: LOC100134305            &lt;NA&gt; 100134305
#&gt; 7:      COL18A1 ENSG00000182871     80781
#&gt; 8: LOC105379618            &lt;NA&gt; 105379618
#&gt;                                      gene.Name gene.Aliases
#&gt; 1:                         similar to tektin 4           NA
#&gt; 2:                            similar to PRED4           NA
#&gt; 3:                      hypothetical LOC642398           NA
#&gt; 4: similar to Uncharacterized protein C21orf94           NA
#&gt; 5:          similar to PR domain containing 15           NA
#&gt; 6:          similar to PR domain containing 15           NA
#&gt; 7:           collagen type XVIII alpha 1 chain           NA
#&gt; 8:                uncharacterized LOC105379618           NA
#&gt;                                         gene.Description gene.GO
#&gt; 1: Imported from NCBI gene; Nomenclature status: UNKNOWN       0
#&gt; 2: Imported from NCBI gene; Nomenclature status: UNKNOWN       0
#&gt; 3: Imported from NCBI gene; Nomenclature status: UNKNOWN       0
#&gt; 4: Imported from NCBI gene; Nomenclature status: UNKNOWN       0
#&gt; 5: Imported from NCBI gene; Nomenclature status: UNKNOWN       0
#&gt; 6: Imported from NCBI gene; Nomenclature status: UNKNOWN       0
#&gt; 7: Imported from NCBI gene; Nomenclature status: INTERIM       0
#&gt; 8: Imported from NCBI gene; Nomenclature status: UNKNOWN       0
#&gt;    gene.Homologues gene.MFX.Rank taxon.Name taxon.ID taxon.Scientific
#&gt; 1:              NA             0      human        1     Homo sapiens
#&gt; 2:              NA             0      human        1     Homo sapiens
#&gt; 3:              NA             0      human        1     Homo sapiens
#&gt; 4:              NA             0      human        1     Homo sapiens
#&gt; 5:              NA             0      human        1     Homo sapiens
#&gt; 6:              NA             0      human        1     Homo sapiens
#&gt; 7:              NA             0      human        1     Homo sapiens
#&gt; 8:              NA             0      human        1     Homo sapiens
#&gt;    phenotypes
#&gt; 1:         NA
#&gt; 2:         NA
#&gt; 3:         NA
#&gt; 4:         NA
#&gt; 5:         NA
#&gt; 6:         NA
#&gt; 7:         NA
#&gt; 8:         NA</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jordan Sicherman, Ogan Mancarci, Javier Castillo-Arnemann.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


