<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Gene probes — getGeneProbes • gemmaAPI</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Gene probes — getGeneProbes" />
<meta property="og:description" content="Retrieves the probes (composite sequences) with this gene" />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemmaAPI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gemmaAPI.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/PavlidisLab/Gemma-API/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Gene probes</h1>
    <small class="dont-index">Source: <a href='https://github.com/PavlidisLab/Gemma-API/blob/master/R/allEndpoints.R'><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>getGeneProbes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieves the probes (composite sequences) with this gene</p>
    </div>

    <pre class="usage"><span class='fu'>getGeneProbes</span><span class='op'>(</span>
  gene <span class='op'>=</span> <span class='cn'>NA_character_</span>,
  offset <span class='op'>=</span> <span class='fl'>0L</span>,
  limit <span class='op'>=</span> <span class='fl'>20L</span>,
  raw <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.raw"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  async <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.async"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  memoised <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.memoise"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  file <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.file"</span>, <span class='cn'>NA_character_</span><span class='op'>)</span>,
  overwrite <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.overwrite"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>gene</th>
      <td><p>Required, part of the URL path.
Can either be the NCBI ID (<code>1859</code>), Ensembl ID (<code>ENSG00000157540</code>) or
official symbol (<code>DYRK1A</code>) of the gene.
NCBI ID is the most efficient (and guaranteed to be unique) identifier.
Official
symbol represents a gene homologue for a random taxon, unless used in a
specific taxon (see Taxon Endpoints).</p></td>
    </tr>
    <tr>
      <th>offset</th>
      <td><p>Optional, defaults to <code>0</code>.
Skips the specified amount of objects when retrieving them from the
database.</p></td>
    </tr>
    <tr>
      <th>limit</th>
      <td><p>Optional, defaults to <code>20</code>.
Limits the result to specified amount of objects. Use 0 for no limit.</p></td>
    </tr>
    <tr>
      <th>raw</th>
      <td><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing.</p></td>
    </tr>
    <tr>
      <th>async</th>
      <td><p><code>TRUE</code> to run the API query on a separate worker, or <code>FALSE</code> to run
synchronously. See the <code>async</code> package for details.</p></td>
    </tr>
    <tr>
      <th>memoised</th>
      <td><p>Whether or not to cache results so future requests for the same data
will be faster. Use <code>forgetGemmaMemoised</code> to clear the cache.</p></td>
    </tr>
    <tr>
      <th>file</th>
      <td><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be a JSON file.
Otherwise, it will be a RDS file.</p></td>
    </tr>
    <tr>
      <th>overwrite</th>
      <td><p>Whether or not to overwrite if a file exists at the specified filename.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An array of probes (composite sequence value objects) that are mapped to
this gene. Note, that it is possible for probes to map to multiple
genes.
A <code>404 error</code> if the given identifier does not map to any object.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>getGeneProbes</span><span class='op'>(</span><span class='st'>"DYRK1A"</span><span class='op'>)</span>
</div><div class='output co'>#&gt;          mapping.Name mapping.Summary mapping.Description array.ShortName
#&gt;  1:             26311              NA              Dyrk1a        GPL14745
#&gt;  2:             20686              NA              Dyrk1a        GPL14745
#&gt;  3:              1646              NA              Dyrk1a        GPL14745
#&gt;  4:             44226              NA              Dyrk1a        GPL14745
#&gt;  5:             34715              NA              Dyrk1a        GPL14745
#&gt;  6:             23687              NA              Dyrk1a        GPL14745
#&gt;  7:             33197              NA              Dyrk1a        GPL14745
#&gt;  8:             37065              NA              Dyrk1a        GPL14745
#&gt;  9:             38256              NA              Dyrk1a        GPL14745
#&gt; 10:             42251              NA              Dyrk1a        GPL14745
#&gt; 11:           7087018              NA                &lt;NA&gt;         GPL6543
#&gt; 12:          10750437              NA                            GPL11534
#&gt; 13: TC1100000329.rn.1              NA                            GPL22388
#&gt; 14:             23876              NA              Dyrk1a        GPL15011
#&gt; 15:             41351              NA                            GPL15011
#&gt; 16:              9388              NA                            GPL15011
#&gt; 17:             15610              NA                            GPL15011
#&gt; 18:             24083              NA                            GPL11280
#&gt; 19:             26644              NA                            GPL11280
#&gt; 20:              2949              NA                            GPL11280
#&gt;                                                                           array.Name
#&gt;  1:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt;  2:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt;  3:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt;  4:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt;  5:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt;  6:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt;  7:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt;  8:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt;  9:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt; 10:                              Agilent-028282 Whole Rat Genome Microarray 4x44K v3
#&gt; 11:       [RaEx-1_0-st] Affymetrix Rat Exon 1.0 ST Array [transcript (gene) version]
#&gt; 12:     [RaGene-1_1-st] Affymetrix Rat Gene 1.1 ST Array [transcript (gene) version]
#&gt; 13: [RTA-1_0] Affymetrix Rat Transcriptome Array 1.0 [transcript (gene) CSV version]
#&gt; 14:                                              Agilent-016475 Patkany genome 4x44K
#&gt; 15:                                              Agilent-016475 Patkany genome 4x44K
#&gt; 16:                                              Agilent-016475 Patkany genome 4x44K
#&gt; 17:                                              Agilent-016475 Patkany genome 4x44K
#&gt; 18:        Agilent-024196 Whole Rat Genome Microarray 4x44K (Feature Number version)
#&gt; 19:        Agilent-024196 Whole Rat Genome Microarray 4x44K (Feature Number version)
#&gt; 20:        Agilent-024196 Whole Rat Genome Microarray 4x44K (Feature Number version)
#&gt;     array.Taxon array.TaxonID array.Type
#&gt;  1:         rat             3   ONECOLOR
#&gt;  2:         rat             3   ONECOLOR
#&gt;  3:         rat             3   ONECOLOR
#&gt;  4:         rat             3   ONECOLOR
#&gt;  5:         rat             3   ONECOLOR
#&gt;  6:         rat             3   ONECOLOR
#&gt;  7:         rat             3   ONECOLOR
#&gt;  8:         rat             3   ONECOLOR
#&gt;  9:         rat             3   ONECOLOR
#&gt; 10:         rat             3   ONECOLOR
#&gt; 11:         rat             3   ONECOLOR
#&gt; 12:         rat             3   ONECOLOR
#&gt; 13:         rat             3   ONECOLOR
#&gt; 14:         rat             3   ONECOLOR
#&gt; 15:         rat             3   ONECOLOR
#&gt; 16:         rat             3   ONECOLOR
#&gt; 17:         rat             3   ONECOLOR
#&gt; 18:         rat             3   ONECOLOR
#&gt; 19:         rat             3   ONECOLOR
#&gt; 20:         rat             3   ONECOLOR
#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          array.Description
#&gt;  1:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt;  2:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt;  3:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt;  4:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt;  5:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt;  6:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt;  7:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt;  8:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt;  9:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt; 10:   Whole Rat Genome Microarray 4x44K v3  Arrays of this design have barcodes that begin with 16028282 or 2528282.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt; 11:                                                                                                                                                                                                                    Affymetrix submissions are typically submitted to GEO using the GEOarchive method described at http://www.ncbi.nlm.nih.gov/projects/geo/info/geo_affy.html  #%create_date=Mon Mar 16 20:01:52 2009 PDT #%chip_type=RaEx-1_0-st-v1 #%lib_set_name=RaEx-1_0-st #%lib_set_version=v1 #%genome-species=Rattus norvegicus #%genome-version=rn4 #%genome-version-ucsc=rn4 #%genome-version-ncbi=HGSC v3.4 #%genome-version-create_date=2004-11 #%netaffx-annotation-date=2009-03-16 #%netaffx-annotation-netaffx-build=28
#&gt; 12:                                      Affymetrix submissions are typically submitted to GEO using the GEOarchive method described at http://www.ncbi.nlm.nih.gov/projects/geo/info/geo_affy.html  RaGene-1_1-st-v1.na31.rn4.transcript.csv #%create_date=Fri Sep 17 11:36:51 2010 PDT #%chip_type=RaGene-1_1-st-v1 #%lib_set_name=RaGene-1_1-st #%lib_set_version=v1 #%genome-species=Rattus norvegicus #%genome-version=rn4 #%genome-version-ucsc=rn4 #%genome-version-ncbi=Rnor3.4 #%genome-version-create_date=2004-11-00 #%genome-lifted_from-species=Rattus norvegicus #%genome-lifted_from-version-ucsc=rn4 #%genome-lifted_from-version-ncbi=Rnor3.4 #%netaffx-annotation-date=2009-11-23 #%netaffx-annotation-netaffx-build=31 
#&gt; 13:                                                                       Annotation was derived from the Affymetrix annotation file RTA-1_0.na36.rn6.transcript.csv:  #%create_date=Wed Jun  8 16:48:57 2016 PDT #%chip_type=RTA-1_0 #%lib_set_name=RTA-1_0 #%lib_set_version=r1 #%genome-species=Rattus norvegicus #%genome-version=rn6 #%genome-version-ucsc=rn6 #%genome-version-ncbi=Rnor_6.0 #%genome-version-create_date=2014-07-30 #%genome-lifted-method=psl-map.pl #%genome-lifted_from-species=Rattus norvegicus #%genome-lifted_from-version-ucsc=rn6 #%genome-lifted_from-version-ncbi=Rnor_6.0 #%netaffx-annotation-date=2016-03-15 #%netaffx-annotation-netaffx-build=36 #%netaffx-annotation-data-type=transcript_cluster 
#&gt; 14:                                         Arrays of this design have barcodes that begin with 16016475 or 2516475.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt; 15:                                         Arrays of this design have barcodes that begin with 16016475 or 2516475.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt; 16:                                         Arrays of this design have barcodes that begin with 16016475 or 2516475.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt; 17:                                         Arrays of this design have barcodes that begin with 16016475 or 2516475.  Orientation: Features are numbered numbered Left-to-Right, Top-to-Bottom as scanned by an Agilent scanner (barcode on the left, DNA on the back surface, scanned through the glass), matching the FeatureNum output from Agilent's Feature Extraction software.  The ID column represents the Agilent Feature Extraction feature number.  Rows and columns are numbered as scanned by an Axon Scanner (barcode on the bottom, DNA on the front surface).  To match data scanned on an Axon scanner, use the RefNumber column contained in the Agilent-provided GAL file as the ID_REF column in sample submissions. 
#&gt; 18:                                                                                                                                                                                                                                                                                             With a focus on well known rat genes and homologues to human and mouse genes with useful annotation, Agilent's Whole Rat Genome Oligo – 4X44K microarray provides researchers with a new tool for modeling human biology in the rat model organism. For researchers, this means they now have access to a microarray made up of relevant content that has been empirically validated by Agilent.\nFrom GPL11280\nLast Updated: Dec 06 2010
#&gt; 19:                                                                                                                                                                                                                                                                                             With a focus on well known rat genes and homologues to human and mouse genes with useful annotation, Agilent's Whole Rat Genome Oligo – 4X44K microarray provides researchers with a new tool for modeling human biology in the rat model organism. For researchers, this means they now have access to a microarray made up of relevant content that has been empirically validated by Agilent.\nFrom GPL11280\nLast Updated: Dec 06 2010
#&gt; 20:                                                                                                                                                                                                                                                                                             With a focus on well known rat genes and homologues to human and mouse genes with useful annotation, Agilent's Whole Rat Genome Oligo – 4X44K microarray provides researchers with a new tool for modeling human biology in the rat model organism. For researchers, this means they now have access to a microarray made up of relevant content that has been empirically validated by Agilent.\nFrom GPL11280\nLast Updated: Dec 06 2010
#&gt;     array.Troubled
#&gt;  1:          FALSE
#&gt;  2:          FALSE
#&gt;  3:          FALSE
#&gt;  4:          FALSE
#&gt;  5:          FALSE
#&gt;  6:          FALSE
#&gt;  7:          FALSE
#&gt;  8:          FALSE
#&gt;  9:          FALSE
#&gt; 10:          FALSE
#&gt; 11:          FALSE
#&gt; 12:          FALSE
#&gt; 13:          FALSE
#&gt; 14:          FALSE
#&gt; 15:          FALSE
#&gt; 16:          FALSE
#&gt; 17:          FALSE
#&gt; 18:          FALSE
#&gt; 19:          FALSE
#&gt; 20:          FALSE</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jordan Sicherman, Ogan Mancarci, Javier Castillo-Arnemann.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


