<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Platform datasets — getPlatformDatasets • gemmaAPI</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Platform datasets — getPlatformDatasets" />
<meta property="og:description" content="Retrieves experiments in the given platform" />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemmaAPI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gemmaAPI.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/PavlidisLab/Gemma-API/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Platform datasets</h1>
    <small class="dont-index">Source: <a href='https://github.com/PavlidisLab/Gemma-API/blob/master/R/allEndpoints.R'><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>getPlatformDatasets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieves experiments in the given platform</p>
    </div>

    <pre class="usage"><span class='fu'>getPlatformDatasets</span><span class='op'>(</span>
  platform <span class='op'>=</span> <span class='cn'>NA_character_</span>,
  offset <span class='op'>=</span> <span class='fl'>0L</span>,
  limit <span class='op'>=</span> <span class='fl'>20L</span>,
  raw <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.raw"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  async <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.async"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  memoised <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.memoise"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  file <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.file"</span>, <span class='cn'>NA_character_</span><span class='op'>)</span>,
  overwrite <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.overwrite"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>platform</th>
      <td><p>Required, part of the URL path.
Can either be the platform ID or its short name (e.g: <code>GPL1355</code>)
Retrieval by ID is more efficient.
Only platforms that user has access to will be available.</p></td>
    </tr>
    <tr>
      <th>offset</th>
      <td><p>Optional, defaults to <code>0</code>.
Skips the specified amount of objects when retrieving them from the
database.</p></td>
    </tr>
    <tr>
      <th>limit</th>
      <td><p>Optional, defaults to <code>20</code>.
Limits the result to specified amount of objects. Use 0 for no limit.</p></td>
    </tr>
    <tr>
      <th>raw</th>
      <td><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing.</p></td>
    </tr>
    <tr>
      <th>async</th>
      <td><p><code>TRUE</code> to run the API query on a separate worker, or <code>FALSE</code> to run
synchronously. See the <code>async</code> package for details.</p></td>
    </tr>
    <tr>
      <th>memoised</th>
      <td><p>Whether or not to cache results so future requests for the same data
will be faster. Use <code>forgetGemmaMemoised</code> to clear the cache.</p></td>
    </tr>
    <tr>
      <th>file</th>
      <td><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be a JSON file.
Otherwise, it will be a RDS file.</p></td>
    </tr>
    <tr>
      <th>overwrite</th>
      <td><p>Whether or not to overwrite if a file exists at the specified filename.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An array of datasets (expression experiment value objects) that are on
the given platform.
A <code>404 error</code> if the given identifier does not map to any object.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>getPlatformDatasets</span><span class='op'>(</span><span class='st'>"GPL1355"</span><span class='op'>)</span>
</div><div class='output co'>#&gt;     ee.ShortName               ee.Name ee.ID
#&gt;  1:      GSE2872 d'mel-affy-rat-168311     2
#&gt;  2:      GSE2872 d'mel-affy-rat-168311     2
#&gt;  3:      GSE2872 d'mel-affy-rat-168311     2
#&gt;  4:      GSE2872 d'mel-affy-rat-168311     2
#&gt;  5:      GSE2872 d'mel-affy-rat-168311     2
#&gt;  6:      GSE2872 d'mel-affy-rat-168311     2
#&gt;  7:      GSE2872 d'mel-affy-rat-168311     2
#&gt;  8:      GSE2872 d'mel-affy-rat-168311     2
#&gt;  9:      GSE2872 d'mel-affy-rat-168311     2
#&gt; 10:      GSE2872 d'mel-affy-rat-168311     2
#&gt; 11:      GSE2872 d'mel-affy-rat-168311     2
#&gt; 12:      GSE2872 d'mel-affy-rat-168311     2
#&gt; 13:      GSE2870  Ogle-5P01NS037520-05   199
#&gt; 14:      GSE2870  Ogle-5P01NS037520-05   199
#&gt; 15:      GSE2870  Ogle-5P01NS037520-05   199
#&gt; 16:      GSE2870  Ogle-5P01NS037520-05   199
#&gt; 17:      GSE2870  Ogle-5P01NS037520-05   199
#&gt; 18:      GSE2870  Ogle-5P01NS037520-05   199
#&gt; 19:      GSE2870  Ogle-5P01NS037520-05   199
#&gt; 20:      GSE2870  Ogle-5P01NS037520-05   199
#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            ee.Description
#&gt;  1:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt;  2:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt;  3:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt;  4:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt;  5:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt;  6:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt;  7:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt;  8:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt;  9:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt; 10:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt; 11:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt; 12:                                                                                                                                              Neurological diseases disrupt the quality of the lives of patients and often lead to their death prematurely.  A common feature of most neurological diseases is the degeneration of neurons, which results from an inappropriate activation of apoptosis. Drugs that inhibit neuronal apoptosis could thus be candidates for therapeutic intervention in neurodegenerative disorders.  We have identified (and recently reported) a chemical called GW5074 that inhibits apoptosis in a variety of cell culture paradigms of neuronal apoptosis.  Additionally, we have found that GW5074 reduces neurodegeneration and improves behavioral outcome in a mouse model of Huntington's disease.  Although GW5074 is a c-Raf inhibitor, we know very little about the molecular mechanisms underlying its neuroprotective action.  Identifying genes that are regulated by GW5074 in neurons will shed insight into this issue. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited.  The specific aim is to identify genes that are differentially expressed in neurons treated with GW5074. We believe that neuroprotection by GW5074 involves the regulation of several genes.  Some of these genes are likely to be induced whereas the expression of other genes might be inhibited. Cultures of cerebellar granule neurons undergo apoptosis when switched from medium containing levated levels of potassium (high K+ or HK) to medium containing low potassium (LK).  Although cell death begins at about 18 h, we have found that the by 6 h after LK treatment these neurons are irreversibly committed to cell death.  We will treat cerebellar granule neuron cultures with LK medium (which induces them to undergo apoptosis) or with GW5074 (1 uM).  We will extract RNA at two time-points after treatment: 3 h and 6 h. Analysis at the two different time-points will show us whether changes in expression of specific genes is transient or sustained or whether the changes occurs early or relatively late in the process.  Neuronal cultures will be prepared from 1 week old Wistar rats.  The cultures will be maintained in culture for 1 week before treatment.  Following treatment the cells will be lysed and total RNA isolated.  The RNA will be stored at -80oC and shipped to the microarray facility for analysis.  The experiment will be done in triplicate.  Thus, for each time-point (3 or 6h treatment) we will be hope to provide 3 sets of samples (each set coming from a different culture preparation and containing lysates from cells treated with LK or GW5074).  Having samples from 3 independent cultures will mitigate any expression differences resulting from subtle variations in culture quality or in the preparation or quality of RNA.\nDate GSE2872 Last Updated: Jul 06 2005\nContributors:  S R D'Mello\nIncludes GDS1837.\n Update date: Jun 09 2006.\n Dataset description GDS1837: Analysis of cultured cerebellar granule neurons in low potassium medium 3 and 6 hours following treatment with the c-Raf inhibitor GW5074. GW5074 blocks low potassium induced cell death. Results provide insight into the neuroprotective action of GW5074.
#&gt; 13: Prolonged stress has adverse consequences for neurons in the hippocampus (HIPP). The amygdala (AMY) is a brain region that is similar to HIPP across many measures, suggesting that chronic stress might modulate AMY and HIPP function in similar ways. However, studies addressing this issue have produced surprising results. For example, a regimen of chronic stress shown to produce atrophy in HIPP neurons caused dendritic branching in AMY neurons. These and other data have revealed that excessive stress induces fundamentally opposing processes in the AMY and HIPP, such that AMY function is facilitated by levels of stress that produce deleterious effects in HIPP. The mechanisms that contribute to the opposing responses of these brain regions to chronic stress are unknown. Such knowledge may suggest novel directions for pharmacological interventions to protect the HIPP from stress-mediated damage. We will determine gene expression patterns in the hippocampus and amygdala under basal and stressful conditions, with a particular interest in those genes that are differentially regulated across the two regions. We hypothesize that genes that are oppositely regulated in the amygdala and hippocampus after stress mediate the opposing functional consequences of chronic stress in these two brain regions.  Moreover, genes which are differentially expressed in these regions under basal conditions may underlie the distinct responses of these regions to stress. Rats will receive either 30sec of handling (Control group; n=15) or 3hr of immobilization stress (Stress group; n=15) each day for 14 consecutive days.  This stress paradigm has previously been shown to produce opposing effects on dendritic morphology and region-dependent behaviors for  the hippocampus versus amygdala.  Twenty-four hours after the last stress or handling session, the rats will be overanaesthetized, and the brain will be removed and placed in ice-cold buffer.  Tissue will be rapidly dissected from the basolateral complex of the amygdala and the CA3 region of the hippocampus of both hemispheres, flash frozen in liquid nitrogen, and stored at -80C until further processing.  The tissue from each brain region will be pooled across groups, yielding four samples (Control-Hippocampus, Control-Amygdala, Stress-Hippocampus, and Stress-Amygdala).  Total RNA will be extracted using standard methods, and sent to the centers.  Each sample will be run in triplicate, utilizing a total of 12 Affymetrix Rat Genome U34A gene chips.  A triplicate array assay of pooled samples has previously been shown to both substantially increase the sensitivity of detecting genes of interest and reduce the variability associated with individual microarrays.  In addition, the large number of rats used in each group will reduce the variability in gene expression associated with individual responses to chronic stress, as well as variability due to slight anatomical differences in the tissue extracted from each rat.\nLast Updated (by provider): Jul 06 2005\nContributors:  W Ogle\nIncludes GDS1794.\n Update date: Jun 08 2006.\n Dataset description GDS1794: Analysis of the amygdalae (Amy) and hippocampi (Hipp) of animals subjected to chronic stress induced by daily handling sessions for 14 days. Studies indicate that chronic stress produce a response in Hipp that is opposite of that in Amy.
#&gt; 14: Prolonged stress has adverse consequences for neurons in the hippocampus (HIPP). The amygdala (AMY) is a brain region that is similar to HIPP across many measures, suggesting that chronic stress might modulate AMY and HIPP function in similar ways. However, studies addressing this issue have produced surprising results. For example, a regimen of chronic stress shown to produce atrophy in HIPP neurons caused dendritic branching in AMY neurons. These and other data have revealed that excessive stress induces fundamentally opposing processes in the AMY and HIPP, such that AMY function is facilitated by levels of stress that produce deleterious effects in HIPP. The mechanisms that contribute to the opposing responses of these brain regions to chronic stress are unknown. Such knowledge may suggest novel directions for pharmacological interventions to protect the HIPP from stress-mediated damage. We will determine gene expression patterns in the hippocampus and amygdala under basal and stressful conditions, with a particular interest in those genes that are differentially regulated across the two regions. We hypothesize that genes that are oppositely regulated in the amygdala and hippocampus after stress mediate the opposing functional consequences of chronic stress in these two brain regions.  Moreover, genes which are differentially expressed in these regions under basal conditions may underlie the distinct responses of these regions to stress. Rats will receive either 30sec of handling (Control group; n=15) or 3hr of immobilization stress (Stress group; n=15) each day for 14 consecutive days.  This stress paradigm has previously been shown to produce opposing effects on dendritic morphology and region-dependent behaviors for  the hippocampus versus amygdala.  Twenty-four hours after the last stress or handling session, the rats will be overanaesthetized, and the brain will be removed and placed in ice-cold buffer.  Tissue will be rapidly dissected from the basolateral complex of the amygdala and the CA3 region of the hippocampus of both hemispheres, flash frozen in liquid nitrogen, and stored at -80C until further processing.  The tissue from each brain region will be pooled across groups, yielding four samples (Control-Hippocampus, Control-Amygdala, Stress-Hippocampus, and Stress-Amygdala).  Total RNA will be extracted using standard methods, and sent to the centers.  Each sample will be run in triplicate, utilizing a total of 12 Affymetrix Rat Genome U34A gene chips.  A triplicate array assay of pooled samples has previously been shown to both substantially increase the sensitivity of detecting genes of interest and reduce the variability associated with individual microarrays.  In addition, the large number of rats used in each group will reduce the variability in gene expression associated with individual responses to chronic stress, as well as variability due to slight anatomical differences in the tissue extracted from each rat.\nLast Updated (by provider): Jul 06 2005\nContributors:  W Ogle\nIncludes GDS1794.\n Update date: Jun 08 2006.\n Dataset description GDS1794: Analysis of the amygdalae (Amy) and hippocampi (Hipp) of animals subjected to chronic stress induced by daily handling sessions for 14 days. Studies indicate that chronic stress produce a response in Hipp that is opposite of that in Amy.
#&gt; 15: Prolonged stress has adverse consequences for neurons in the hippocampus (HIPP). The amygdala (AMY) is a brain region that is similar to HIPP across many measures, suggesting that chronic stress might modulate AMY and HIPP function in similar ways. However, studies addressing this issue have produced surprising results. For example, a regimen of chronic stress shown to produce atrophy in HIPP neurons caused dendritic branching in AMY neurons. These and other data have revealed that excessive stress induces fundamentally opposing processes in the AMY and HIPP, such that AMY function is facilitated by levels of stress that produce deleterious effects in HIPP. The mechanisms that contribute to the opposing responses of these brain regions to chronic stress are unknown. Such knowledge may suggest novel directions for pharmacological interventions to protect the HIPP from stress-mediated damage. We will determine gene expression patterns in the hippocampus and amygdala under basal and stressful conditions, with a particular interest in those genes that are differentially regulated across the two regions. We hypothesize that genes that are oppositely regulated in the amygdala and hippocampus after stress mediate the opposing functional consequences of chronic stress in these two brain regions.  Moreover, genes which are differentially expressed in these regions under basal conditions may underlie the distinct responses of these regions to stress. Rats will receive either 30sec of handling (Control group; n=15) or 3hr of immobilization stress (Stress group; n=15) each day for 14 consecutive days.  This stress paradigm has previously been shown to produce opposing effects on dendritic morphology and region-dependent behaviors for  the hippocampus versus amygdala.  Twenty-four hours after the last stress or handling session, the rats will be overanaesthetized, and the brain will be removed and placed in ice-cold buffer.  Tissue will be rapidly dissected from the basolateral complex of the amygdala and the CA3 region of the hippocampus of both hemispheres, flash frozen in liquid nitrogen, and stored at -80C until further processing.  The tissue from each brain region will be pooled across groups, yielding four samples (Control-Hippocampus, Control-Amygdala, Stress-Hippocampus, and Stress-Amygdala).  Total RNA will be extracted using standard methods, and sent to the centers.  Each sample will be run in triplicate, utilizing a total of 12 Affymetrix Rat Genome U34A gene chips.  A triplicate array assay of pooled samples has previously been shown to both substantially increase the sensitivity of detecting genes of interest and reduce the variability associated with individual microarrays.  In addition, the large number of rats used in each group will reduce the variability in gene expression associated with individual responses to chronic stress, as well as variability due to slight anatomical differences in the tissue extracted from each rat.\nLast Updated (by provider): Jul 06 2005\nContributors:  W Ogle\nIncludes GDS1794.\n Update date: Jun 08 2006.\n Dataset description GDS1794: Analysis of the amygdalae (Amy) and hippocampi (Hipp) of animals subjected to chronic stress induced by daily handling sessions for 14 days. Studies indicate that chronic stress produce a response in Hipp that is opposite of that in Amy.
#&gt; 16: Prolonged stress has adverse consequences for neurons in the hippocampus (HIPP). The amygdala (AMY) is a brain region that is similar to HIPP across many measures, suggesting that chronic stress might modulate AMY and HIPP function in similar ways. However, studies addressing this issue have produced surprising results. For example, a regimen of chronic stress shown to produce atrophy in HIPP neurons caused dendritic branching in AMY neurons. These and other data have revealed that excessive stress induces fundamentally opposing processes in the AMY and HIPP, such that AMY function is facilitated by levels of stress that produce deleterious effects in HIPP. The mechanisms that contribute to the opposing responses of these brain regions to chronic stress are unknown. Such knowledge may suggest novel directions for pharmacological interventions to protect the HIPP from stress-mediated damage. We will determine gene expression patterns in the hippocampus and amygdala under basal and stressful conditions, with a particular interest in those genes that are differentially regulated across the two regions. We hypothesize that genes that are oppositely regulated in the amygdala and hippocampus after stress mediate the opposing functional consequences of chronic stress in these two brain regions.  Moreover, genes which are differentially expressed in these regions under basal conditions may underlie the distinct responses of these regions to stress. Rats will receive either 30sec of handling (Control group; n=15) or 3hr of immobilization stress (Stress group; n=15) each day for 14 consecutive days.  This stress paradigm has previously been shown to produce opposing effects on dendritic morphology and region-dependent behaviors for  the hippocampus versus amygdala.  Twenty-four hours after the last stress or handling session, the rats will be overanaesthetized, and the brain will be removed and placed in ice-cold buffer.  Tissue will be rapidly dissected from the basolateral complex of the amygdala and the CA3 region of the hippocampus of both hemispheres, flash frozen in liquid nitrogen, and stored at -80C until further processing.  The tissue from each brain region will be pooled across groups, yielding four samples (Control-Hippocampus, Control-Amygdala, Stress-Hippocampus, and Stress-Amygdala).  Total RNA will be extracted using standard methods, and sent to the centers.  Each sample will be run in triplicate, utilizing a total of 12 Affymetrix Rat Genome U34A gene chips.  A triplicate array assay of pooled samples has previously been shown to both substantially increase the sensitivity of detecting genes of interest and reduce the variability associated with individual microarrays.  In addition, the large number of rats used in each group will reduce the variability in gene expression associated with individual responses to chronic stress, as well as variability due to slight anatomical differences in the tissue extracted from each rat.\nLast Updated (by provider): Jul 06 2005\nContributors:  W Ogle\nIncludes GDS1794.\n Update date: Jun 08 2006.\n Dataset description GDS1794: Analysis of the amygdalae (Amy) and hippocampi (Hipp) of animals subjected to chronic stress induced by daily handling sessions for 14 days. Studies indicate that chronic stress produce a response in Hipp that is opposite of that in Amy.
#&gt; 17: Prolonged stress has adverse consequences for neurons in the hippocampus (HIPP). The amygdala (AMY) is a brain region that is similar to HIPP across many measures, suggesting that chronic stress might modulate AMY and HIPP function in similar ways. However, studies addressing this issue have produced surprising results. For example, a regimen of chronic stress shown to produce atrophy in HIPP neurons caused dendritic branching in AMY neurons. These and other data have revealed that excessive stress induces fundamentally opposing processes in the AMY and HIPP, such that AMY function is facilitated by levels of stress that produce deleterious effects in HIPP. The mechanisms that contribute to the opposing responses of these brain regions to chronic stress are unknown. Such knowledge may suggest novel directions for pharmacological interventions to protect the HIPP from stress-mediated damage. We will determine gene expression patterns in the hippocampus and amygdala under basal and stressful conditions, with a particular interest in those genes that are differentially regulated across the two regions. We hypothesize that genes that are oppositely regulated in the amygdala and hippocampus after stress mediate the opposing functional consequences of chronic stress in these two brain regions.  Moreover, genes which are differentially expressed in these regions under basal conditions may underlie the distinct responses of these regions to stress. Rats will receive either 30sec of handling (Control group; n=15) or 3hr of immobilization stress (Stress group; n=15) each day for 14 consecutive days.  This stress paradigm has previously been shown to produce opposing effects on dendritic morphology and region-dependent behaviors for  the hippocampus versus amygdala.  Twenty-four hours after the last stress or handling session, the rats will be overanaesthetized, and the brain will be removed and placed in ice-cold buffer.  Tissue will be rapidly dissected from the basolateral complex of the amygdala and the CA3 region of the hippocampus of both hemispheres, flash frozen in liquid nitrogen, and stored at -80C until further processing.  The tissue from each brain region will be pooled across groups, yielding four samples (Control-Hippocampus, Control-Amygdala, Stress-Hippocampus, and Stress-Amygdala).  Total RNA will be extracted using standard methods, and sent to the centers.  Each sample will be run in triplicate, utilizing a total of 12 Affymetrix Rat Genome U34A gene chips.  A triplicate array assay of pooled samples has previously been shown to both substantially increase the sensitivity of detecting genes of interest and reduce the variability associated with individual microarrays.  In addition, the large number of rats used in each group will reduce the variability in gene expression associated with individual responses to chronic stress, as well as variability due to slight anatomical differences in the tissue extracted from each rat.\nLast Updated (by provider): Jul 06 2005\nContributors:  W Ogle\nIncludes GDS1794.\n Update date: Jun 08 2006.\n Dataset description GDS1794: Analysis of the amygdalae (Amy) and hippocampi (Hipp) of animals subjected to chronic stress induced by daily handling sessions for 14 days. Studies indicate that chronic stress produce a response in Hipp that is opposite of that in Amy.
#&gt; 18: Prolonged stress has adverse consequences for neurons in the hippocampus (HIPP). The amygdala (AMY) is a brain region that is similar to HIPP across many measures, suggesting that chronic stress might modulate AMY and HIPP function in similar ways. However, studies addressing this issue have produced surprising results. For example, a regimen of chronic stress shown to produce atrophy in HIPP neurons caused dendritic branching in AMY neurons. These and other data have revealed that excessive stress induces fundamentally opposing processes in the AMY and HIPP, such that AMY function is facilitated by levels of stress that produce deleterious effects in HIPP. The mechanisms that contribute to the opposing responses of these brain regions to chronic stress are unknown. Such knowledge may suggest novel directions for pharmacological interventions to protect the HIPP from stress-mediated damage. We will determine gene expression patterns in the hippocampus and amygdala under basal and stressful conditions, with a particular interest in those genes that are differentially regulated across the two regions. We hypothesize that genes that are oppositely regulated in the amygdala and hippocampus after stress mediate the opposing functional consequences of chronic stress in these two brain regions.  Moreover, genes which are differentially expressed in these regions under basal conditions may underlie the distinct responses of these regions to stress. Rats will receive either 30sec of handling (Control group; n=15) or 3hr of immobilization stress (Stress group; n=15) each day for 14 consecutive days.  This stress paradigm has previously been shown to produce opposing effects on dendritic morphology and region-dependent behaviors for  the hippocampus versus amygdala.  Twenty-four hours after the last stress or handling session, the rats will be overanaesthetized, and the brain will be removed and placed in ice-cold buffer.  Tissue will be rapidly dissected from the basolateral complex of the amygdala and the CA3 region of the hippocampus of both hemispheres, flash frozen in liquid nitrogen, and stored at -80C until further processing.  The tissue from each brain region will be pooled across groups, yielding four samples (Control-Hippocampus, Control-Amygdala, Stress-Hippocampus, and Stress-Amygdala).  Total RNA will be extracted using standard methods, and sent to the centers.  Each sample will be run in triplicate, utilizing a total of 12 Affymetrix Rat Genome U34A gene chips.  A triplicate array assay of pooled samples has previously been shown to both substantially increase the sensitivity of detecting genes of interest and reduce the variability associated with individual microarrays.  In addition, the large number of rats used in each group will reduce the variability in gene expression associated with individual responses to chronic stress, as well as variability due to slight anatomical differences in the tissue extracted from each rat.\nLast Updated (by provider): Jul 06 2005\nContributors:  W Ogle\nIncludes GDS1794.\n Update date: Jun 08 2006.\n Dataset description GDS1794: Analysis of the amygdalae (Amy) and hippocampi (Hipp) of animals subjected to chronic stress induced by daily handling sessions for 14 days. Studies indicate that chronic stress produce a response in Hipp that is opposite of that in Amy.
#&gt; 19: Prolonged stress has adverse consequences for neurons in the hippocampus (HIPP). The amygdala (AMY) is a brain region that is similar to HIPP across many measures, suggesting that chronic stress might modulate AMY and HIPP function in similar ways. However, studies addressing this issue have produced surprising results. For example, a regimen of chronic stress shown to produce atrophy in HIPP neurons caused dendritic branching in AMY neurons. These and other data have revealed that excessive stress induces fundamentally opposing processes in the AMY and HIPP, such that AMY function is facilitated by levels of stress that produce deleterious effects in HIPP. The mechanisms that contribute to the opposing responses of these brain regions to chronic stress are unknown. Such knowledge may suggest novel directions for pharmacological interventions to protect the HIPP from stress-mediated damage. We will determine gene expression patterns in the hippocampus and amygdala under basal and stressful conditions, with a particular interest in those genes that are differentially regulated across the two regions. We hypothesize that genes that are oppositely regulated in the amygdala and hippocampus after stress mediate the opposing functional consequences of chronic stress in these two brain regions.  Moreover, genes which are differentially expressed in these regions under basal conditions may underlie the distinct responses of these regions to stress. Rats will receive either 30sec of handling (Control group; n=15) or 3hr of immobilization stress (Stress group; n=15) each day for 14 consecutive days.  This stress paradigm has previously been shown to produce opposing effects on dendritic morphology and region-dependent behaviors for  the hippocampus versus amygdala.  Twenty-four hours after the last stress or handling session, the rats will be overanaesthetized, and the brain will be removed and placed in ice-cold buffer.  Tissue will be rapidly dissected from the basolateral complex of the amygdala and the CA3 region of the hippocampus of both hemispheres, flash frozen in liquid nitrogen, and stored at -80C until further processing.  The tissue from each brain region will be pooled across groups, yielding four samples (Control-Hippocampus, Control-Amygdala, Stress-Hippocampus, and Stress-Amygdala).  Total RNA will be extracted using standard methods, and sent to the centers.  Each sample will be run in triplicate, utilizing a total of 12 Affymetrix Rat Genome U34A gene chips.  A triplicate array assay of pooled samples has previously been shown to both substantially increase the sensitivity of detecting genes of interest and reduce the variability associated with individual microarrays.  In addition, the large number of rats used in each group will reduce the variability in gene expression associated with individual responses to chronic stress, as well as variability due to slight anatomical differences in the tissue extracted from each rat.\nLast Updated (by provider): Jul 06 2005\nContributors:  W Ogle\nIncludes GDS1794.\n Update date: Jun 08 2006.\n Dataset description GDS1794: Analysis of the amygdalae (Amy) and hippocampi (Hipp) of animals subjected to chronic stress induced by daily handling sessions for 14 days. Studies indicate that chronic stress produce a response in Hipp that is opposite of that in Amy.
#&gt; 20: Prolonged stress has adverse consequences for neurons in the hippocampus (HIPP). The amygdala (AMY) is a brain region that is similar to HIPP across many measures, suggesting that chronic stress might modulate AMY and HIPP function in similar ways. However, studies addressing this issue have produced surprising results. For example, a regimen of chronic stress shown to produce atrophy in HIPP neurons caused dendritic branching in AMY neurons. These and other data have revealed that excessive stress induces fundamentally opposing processes in the AMY and HIPP, such that AMY function is facilitated by levels of stress that produce deleterious effects in HIPP. The mechanisms that contribute to the opposing responses of these brain regions to chronic stress are unknown. Such knowledge may suggest novel directions for pharmacological interventions to protect the HIPP from stress-mediated damage. We will determine gene expression patterns in the hippocampus and amygdala under basal and stressful conditions, with a particular interest in those genes that are differentially regulated across the two regions. We hypothesize that genes that are oppositely regulated in the amygdala and hippocampus after stress mediate the opposing functional consequences of chronic stress in these two brain regions.  Moreover, genes which are differentially expressed in these regions under basal conditions may underlie the distinct responses of these regions to stress. Rats will receive either 30sec of handling (Control group; n=15) or 3hr of immobilization stress (Stress group; n=15) each day for 14 consecutive days.  This stress paradigm has previously been shown to produce opposing effects on dendritic morphology and region-dependent behaviors for  the hippocampus versus amygdala.  Twenty-four hours after the last stress or handling session, the rats will be overanaesthetized, and the brain will be removed and placed in ice-cold buffer.  Tissue will be rapidly dissected from the basolateral complex of the amygdala and the CA3 region of the hippocampus of both hemispheres, flash frozen in liquid nitrogen, and stored at -80C until further processing.  The tissue from each brain region will be pooled across groups, yielding four samples (Control-Hippocampus, Control-Amygdala, Stress-Hippocampus, and Stress-Amygdala).  Total RNA will be extracted using standard methods, and sent to the centers.  Each sample will be run in triplicate, utilizing a total of 12 Affymetrix Rat Genome U34A gene chips.  A triplicate array assay of pooled samples has previously been shown to both substantially increase the sensitivity of detecting genes of interest and reduce the variability associated with individual microarrays.  In addition, the large number of rats used in each group will reduce the variability in gene expression associated with individual responses to chronic stress, as well as variability due to slight anatomical differences in the tissue extracted from each rat.\nLast Updated (by provider): Jul 06 2005\nContributors:  W Ogle\nIncludes GDS1794.\n Update date: Jun 08 2006.\n Dataset description GDS1794: Analysis of the amygdalae (Amy) and hippocampi (Hipp) of animals subjected to chronic stress induced by daily handling sessions for 14 days. Studies indicate that chronic stress produce a response in Hipp that is opposite of that in Amy.
#&gt;     ee.Public ee.Troubled ee.Accession ee.Database
#&gt;  1:      TRUE       FALSE      GSE2872         GEO
#&gt;  2:      TRUE       FALSE      GSE2872         GEO
#&gt;  3:      TRUE       FALSE      GSE2872         GEO
#&gt;  4:      TRUE       FALSE      GSE2872         GEO
#&gt;  5:      TRUE       FALSE      GSE2872         GEO
#&gt;  6:      TRUE       FALSE      GSE2872         GEO
#&gt;  7:      TRUE       FALSE      GSE2872         GEO
#&gt;  8:      TRUE       FALSE      GSE2872         GEO
#&gt;  9:      TRUE       FALSE      GSE2872         GEO
#&gt; 10:      TRUE       FALSE      GSE2872         GEO
#&gt; 11:      TRUE       FALSE      GSE2872         GEO
#&gt; 12:      TRUE       FALSE      GSE2872         GEO
#&gt; 13:      TRUE       FALSE      GSE2870         GEO
#&gt; 14:      TRUE       FALSE      GSE2870         GEO
#&gt; 15:      TRUE       FALSE      GSE2870         GEO
#&gt; 16:      TRUE       FALSE      GSE2870         GEO
#&gt; 17:      TRUE       FALSE      GSE2870         GEO
#&gt; 18:      TRUE       FALSE      GSE2870         GEO
#&gt; 19:      TRUE       FALSE      GSE2870         GEO
#&gt; 20:      TRUE       FALSE      GSE2870         GEO
#&gt;                               ee.URL ee.Samples      ee.LastUpdated ee.Batch.P
#&gt;  1: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt;  2: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt;  3: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt;  4: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt;  5: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt;  6: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt;  7: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt;  8: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt;  9: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt; 10: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt; 11: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt; 12: http://www.ncbi.nlm.nih.gov/geo/         12 2020-03-27 03:41:10         NA
#&gt; 13: http://www.ncbi.nlm.nih.gov/geo/         18 2020-03-27 04:20:38         NA
#&gt; 14: http://www.ncbi.nlm.nih.gov/geo/         18 2020-03-27 04:20:38         NA
#&gt; 15: http://www.ncbi.nlm.nih.gov/geo/         18 2020-03-27 04:20:38         NA
#&gt; 16: http://www.ncbi.nlm.nih.gov/geo/         18 2020-03-27 04:20:38         NA
#&gt; 17: http://www.ncbi.nlm.nih.gov/geo/         18 2020-03-27 04:20:38         NA
#&gt; 18: http://www.ncbi.nlm.nih.gov/geo/         18 2020-03-27 04:20:38         NA
#&gt; 19: http://www.ncbi.nlm.nih.gov/geo/         18 2020-03-27 04:20:38         NA
#&gt; 20: http://www.ncbi.nlm.nih.gov/geo/         18 2020-03-27 04:20:38         NA
#&gt;     ee.Batch.PC geeq.batchCorrected geeq.qScore geeq.sScore taxon.Name taxon.ID
#&gt;  1:          NA               FALSE   0.8552303         0.5        rat        3
#&gt;  2:          NA               FALSE   0.8552303         0.5        rat        3
#&gt;  3:          NA               FALSE   0.8552303         0.5        rat        3
#&gt;  4:          NA               FALSE   0.8552303         0.5        rat        3
#&gt;  5:          NA               FALSE   0.8552303         0.5        rat        3
#&gt;  6:          NA               FALSE   0.8552303         0.5        rat        3
#&gt;  7:          NA               FALSE   0.8552303         0.5        rat        3
#&gt;  8:          NA               FALSE   0.8552303         0.5        rat        3
#&gt;  9:          NA               FALSE   0.8552303         0.5        rat        3
#&gt; 10:          NA               FALSE   0.8552303         0.5        rat        3
#&gt; 11:          NA               FALSE   0.8552303         0.5        rat        3
#&gt; 12:          NA               FALSE   0.8552303         0.5        rat        3
#&gt; 13:          NA               FALSE   0.5708228         0.5        rat        3
#&gt; 14:          NA               FALSE   0.5708228         0.5        rat        3
#&gt; 15:          NA               FALSE   0.5708228         0.5        rat        3
#&gt; 16:          NA               FALSE   0.5708228         0.5        rat        3
#&gt; 17:          NA               FALSE   0.5708228         0.5        rat        3
#&gt; 18:          NA               FALSE   0.5708228         0.5        rat        3
#&gt; 19:          NA               FALSE   0.5708228         0.5        rat        3
#&gt; 20:          NA               FALSE   0.5708228         0.5        rat        3
#&gt;     technology.Type
#&gt;  1:        ONECOLOR
#&gt;  2:        ONECOLOR
#&gt;  3:        ONECOLOR
#&gt;  4:        ONECOLOR
#&gt;  5:        ONECOLOR
#&gt;  6:        ONECOLOR
#&gt;  7:        ONECOLOR
#&gt;  8:        ONECOLOR
#&gt;  9:        ONECOLOR
#&gt; 10:        ONECOLOR
#&gt; 11:        ONECOLOR
#&gt; 12:        ONECOLOR
#&gt; 13:        ONECOLOR
#&gt; 14:        ONECOLOR
#&gt; 15:        ONECOLOR
#&gt; 16:        ONECOLOR
#&gt; 17:        ONECOLOR
#&gt; 18:        ONECOLOR
#&gt; 19:        ONECOLOR
#&gt; 20:        ONECOLOR</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jordan Sicherman, Ogan Mancarci, Javier Castillo-Arnemann.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


