<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Platform elements — getPlatformElements • gemmaAPI</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Platform elements — getPlatformElements" />
<meta property="og:description" content="Retrieves the composite sequences (elements) for the given platform" />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemmaAPI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gemmaAPI.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Platform elements</h1>
    
    <div class="hidden name"><code>getPlatformElements.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieves the composite sequences (elements) for the given platform</p>
    </div>

    <pre class="usage"><span class='fu'>getPlatformElements</span><span class='op'>(</span>
  platform <span class='op'>=</span> <span class='cn'>NA_character_</span>,
  element <span class='op'>=</span> <span class='cn'>NA_character_</span>,
  offset <span class='op'>=</span> <span class='fl'>0L</span>,
  limit <span class='op'>=</span> <span class='fl'>20L</span>,
  raw <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.raw"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  async <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.async"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  memoised <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.memoise"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>,
  file <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.file"</span>, <span class='cn'>NA_character_</span><span class='op'>)</span>,
  overwrite <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/options.html'>getOption</a></span><span class='op'>(</span><span class='st'>"gemma.overwrite"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>platform</th>
      <td><p>Required, part of the URL path.
Can either be the platform ID or its short name (e.g: <code>GPL1355</code>)
Retrieval by ID is more efficient.
Only platforms that user has access to will be available.</p></td>
    </tr>
    <tr>
      <th>element</th>
      <td><p>Optional, defaults to <code>empty</code>.
Limits the result to entities with given identifiers.
A list of identifiers, separated by commas (e.g:
<code>AFFX_Rat_beta-actin_M_at, AFFX_Rat_Hexokinase_M_at</code>).
Can either be probes name or IDs.
Do not combine different identifiers in one query.
When
using in scripts, remember to URL-encode any forward slashes in the
probe name (see the compiled URL below).</p></td>
    </tr>
    <tr>
      <th>offset</th>
      <td><p>Optional, defaults to <code>0</code>.
Skips the specified amount of objects when retrieving them from the
database.</p></td>
    </tr>
    <tr>
      <th>limit</th>
      <td><p>Optional, defaults to <code>20</code>.
Limits the result to specified amount of objects. Use 0 for no limit.</p></td>
    </tr>
    <tr>
      <th>raw</th>
      <td><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing.</p></td>
    </tr>
    <tr>
      <th>async</th>
      <td><p><code>TRUE</code> to run the API query on a separate worker, or <code>FALSE</code> to run
synchronously. See the <code>async</code> package for details.</p></td>
    </tr>
    <tr>
      <th>memoised</th>
      <td><p>Whether or not to cache results so future requests for the same data
will be faster. Use <code>forgetGemmaMemoised</code> to clear the cache.</p></td>
    </tr>
    <tr>
      <th>file</th>
      <td><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be a JSON file.
Otherwise, it will be a RDS file.</p></td>
    </tr>
    <tr>
      <th>overwrite</th>
      <td><p>Whether or not to overwrite if a file exists at the specified filename.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An array of elements (composite sequence value objects) of the given
platform.
Empty collection, if no elements matched the <code>elements</code> parameter.
A <code>404 error</code> if the given identifier does not map to any object.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>getPlatformElements</span><span class='op'>(</span><span class='st'>"GPL1355"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: Item 3 has 2 rows but longest item has 7; recycled with remainder.</span></div><div class='output co'>#&gt;                 mapping.Name     mapping.Summary
#&gt;  1: AFFX_Rat_Hexokinase_M_at   &lt;data.table[1x3]&gt;
#&gt;  2: AFFX_Rat_Hexokinase_5_at   &lt;data.table[1x3]&gt;
#&gt;  3: AFFX_Rat_Hexokinase_3_at   &lt;data.table[1x3]&gt;
#&gt;  4:      AFFX_Rat_GAPDH_M_at &lt;data.table[194x3]&gt;
#&gt;  5:      AFFX_Rat_GAPDH_5_at &lt;data.table[141x3]&gt;
#&gt;  6:      AFFX_Rat_GAPDH_3_at &lt;data.table[186x3]&gt;
#&gt;  7: AFFX_Rat_beta-actin_M_at   &lt;data.table[1x3]&gt;
#&gt;  8: AFFX_Rat_beta-actin_5_at   &lt;data.table[1x3]&gt;
#&gt;  9: AFFX_Rat_beta-actin_3_at   &lt;data.table[7x3]&gt;
#&gt; 10:      AFFX_rat_5S_rRNA_at  &lt;data.table[43x3]&gt;
#&gt; 11:     AFFX_ratb2/X14115_at   &lt;data.table[2x3]&gt;
#&gt; 12:     AFFX_ratb1/X12957_at   &lt;data.table[1x3]&gt;
#&gt; 13:          AFFX-TrpnX-M_at   &lt;data.table[1x3]&gt;
#&gt; 14:          AFFX-TrpnX-5_at   &lt;data.table[1x3]&gt;
#&gt; 15:          AFFX-TrpnX-3_at   &lt;data.table[1x3]&gt;
#&gt; 16:           AFFX-ThrX-M_at   &lt;data.table[1x3]&gt;
#&gt; 17:           AFFX-ThrX-5_at   &lt;data.table[1x3]&gt;
#&gt; 18:           AFFX-ThrX-3_at   &lt;data.table[1x3]&gt;
#&gt; 19:      AFFX-r2-P1-cre-5_at   &lt;data.table[1x3]&gt;
#&gt; 20:      AFFX-r2-P1-cre-3_at   &lt;data.table[1x3]&gt;
#&gt;                           mapping.Description array.ShortName
#&gt;  1:                              hexokinase 1         GPL1355
#&gt;  2:                              hexokinase 1         GPL1355
#&gt;  3:                              hexokinase 1         GPL1355
#&gt;  4:  glyceraldehyde-3-phosphate dehydrogenase         GPL1355
#&gt;  5:  glyceraldehyde-3-phosphate dehydrogenase         GPL1355
#&gt;  6:  glyceraldehyde-3-phosphate dehydrogenase         GPL1355
#&gt;  7:                               actin, beta         GPL1355
#&gt;  8:                               actin, beta         GPL1355
#&gt;  9:                               actin, beta         GPL1355
#&gt; 10:                                      &lt;NA&gt;         GPL1355
#&gt; 11:                                      &lt;NA&gt;         GPL1355
#&gt; 12:                                      &lt;NA&gt;         GPL1355
#&gt; 13:                                      &lt;NA&gt;         GPL1355
#&gt; 14:                                      &lt;NA&gt;         GPL1355
#&gt; 15:                                      &lt;NA&gt;         GPL1355
#&gt; 16:                                      &lt;NA&gt;         GPL1355
#&gt; 17:                                      &lt;NA&gt;         GPL1355
#&gt; 18:                                      &lt;NA&gt;         GPL1355
#&gt; 19:                                      &lt;NA&gt;         GPL1355
#&gt; 20:                                      &lt;NA&gt;         GPL1355
#&gt;                                       array.Name array.Taxon array.TaxonID
#&gt;  1: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt;  2: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt;  3: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt;  4: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt;  5: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt;  6: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt;  7: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt;  8: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt;  9: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 10: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 11: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 12: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 13: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 14: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 15: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 16: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 17: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 18: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 19: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt; 20: Affymetrix GeneChip Rat Genome 230 2.0 Array         rat             3
#&gt;     array.Type
#&gt;  1:   ONECOLOR
#&gt;  2:   ONECOLOR
#&gt;  3:   ONECOLOR
#&gt;  4:   ONECOLOR
#&gt;  5:   ONECOLOR
#&gt;  6:   ONECOLOR
#&gt;  7:   ONECOLOR
#&gt;  8:   ONECOLOR
#&gt;  9:   ONECOLOR
#&gt; 10:   ONECOLOR
#&gt; 11:   ONECOLOR
#&gt; 12:   ONECOLOR
#&gt; 13:   ONECOLOR
#&gt; 14:   ONECOLOR
#&gt; 15:   ONECOLOR
#&gt; 16:   ONECOLOR
#&gt; 17:   ONECOLOR
#&gt; 18:   ONECOLOR
#&gt; 19:   ONECOLOR
#&gt; 20:   ONECOLOR
#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            array.Description
#&gt;  1:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt;  2:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt;  3:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt;  4:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt;  5:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt;  6:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt;  7:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt;  8:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt;  9:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 10:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 11:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 12:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 13:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 14:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 15:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 16:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 17:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 18:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 19:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt; 20:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
#&gt;     array.Troubled
#&gt;  1:          FALSE
#&gt;  2:          FALSE
#&gt;  3:          FALSE
#&gt;  4:          FALSE
#&gt;  5:          FALSE
#&gt;  6:          FALSE
#&gt;  7:          FALSE
#&gt;  8:          FALSE
#&gt;  9:          FALSE
#&gt; 10:          FALSE
#&gt; 11:          FALSE
#&gt; 12:          FALSE
#&gt; 13:          FALSE
#&gt; 14:          FALSE
#&gt; 15:          FALSE
#&gt; 16:          FALSE
#&gt; 17:          FALSE
#&gt; 18:          FALSE
#&gt; 19:          FALSE
#&gt; 20:          FALSE</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jordan Sicherman, Ogan Mancarci, Javier Castillo-Arnemann.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


