<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Search everything in Gemma. — search_gemma • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Search everything in Gemma. — search_gemma"><meta property="og:description" content="Search everything in Gemma."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Search everything in Gemma.</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>search_gemma.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Search everything in Gemma.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">search_gemma</span><span class="op">(</span></span>
<span>  <span class="va">query</span>,</span>
<span>  taxon <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  platform <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  resultType <span class="op">=</span> <span class="st">"experiment"</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.attributes"</span>, <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>The search query. Either plain text ('traumatic'), or an ontology term URI ('http://purl.obolibrary.org/obo/UBERON_0002048'). Datasets that contain the given string in their short of full name will also be matched ('GSE201', 'Bronchoalveolar lavage samples'.</p></dd>


<dt>taxon</dt>
<dd><p>A numerical taxon identifier or an ncbi taxon identifier or a taxon identifier that matches either its scientific or common name</p></dd>


<dt>platform</dt>
<dd><p>A platform numerical identifier or a platform short name</p></dd>


<dt>limit</dt>
<dd><p>Optional, defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt>resultType</dt>
<dd><p>The kind of results that should be included in the output. Can be experiment, gene, platform or a long object type name, documented in the API documentation.</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be a JSON file. Otherwise,
it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>


<dt>attributes</dt>
<dd><p>If <code>TRUE</code> additional information from the call will be added
into the output object's attributes such as offset and available elements.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>If <code>raw = FALSE</code> and resultType is experiment, gene or platform,
a data.table containing the search results. If it is any other type, a list
of results. A list with additional details about the search if <code>raw = TRUE</code></p>


    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">search_gemma</span><span class="op">(</span><span class="st">"bipolar"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ShortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:            GSE197966</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             GSE33085</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:            GSE205422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             GSE29318</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:          GSE179921.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:            GSE157509</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:       stanley_vawter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:             GSE11767</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:   McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:            GSE134497</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:            GSE117877</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:            GSE127771</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:             GSE66433</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:              byne-cc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:            GSE169212</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:             GSE45484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:             GSE67645</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:            GSE160761</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:             GSE10843</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:            GSE126942</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                       experiment.Name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                  Transcriptional effects of bipolar disorder drugs on NT2-N cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                Transcriptome analysis of adult retina cell types.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                     Network-based integrative analysis of lithium response in bipolar disorder using transcriptomic and GWAS data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                             Expression profile of FAC-sorted murine retinal cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                           Split part 2 of: TCF7L2 lncRNA: A Link between Bipolar Disorder and Body Mass Index through Glucocorticoid Signaling [RNA-Seq] [collection of material = Experiment 1 ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                   Increased IL-6 and altered inflammatory response in bipolar disorder patient-derived astrocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                           Stanley array collection DLPFC - Vawter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                   Gene expression profiling of fibroblasts and lymphoblastoid cells derived from four individuals</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                      Total RNA sequecing for human induced pluripotent derived cerebral organoids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                  A candidate causal variant underlying both enhanced cognitive performance and increased risk of bipolar disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                         The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells [P7]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                          Effects of the microRNA 137 and its connection to psychiatric disorders.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                              Corpus Callosum data from Stanley collection samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                    Transcriptomic and epigenetic characterization of PVT neurons in bipolar disorder model mouse.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                        Gene-expression differences in peripheral blood between lithium responders and non-responders in the “Lithium Treatment -Moderate dose Use Study” (LiTMUS)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: Transcriptome dynamics of developing photoreceptors in 3-D retina cultures recapitulates temporal sequence of human cone and rod differentiation revealing cell surface markers and gene networks</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                            RNA sequencing  in human  iPSCs derived from  bipolar  patients  to identify important therapeutic molecular targets of Valproate(VPA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                    mRNA Cancer Cell Line Profiles</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                              The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:         25070</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          8887</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:         25972</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:         12310</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:         21159</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:         23425</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:           836</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:          1879</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:           670</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:         16450</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:         20617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:         19455</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:         24427</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:          4354</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:         19187</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:          6145</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:         11634</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:         17943</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:          7196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:         19454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           experiment.Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Human neuronal-like cells (NT2-N) were treated with either lamotrigine (50 µM), lithium (2.5 mM), quetiapine (50 µM), valproate (0.5 mM) or vehicle control for 24 hours. Genome wide mRNA expression was quantified by RNA-sequencing. Results offer insights on the mechanism(s) of action of bipolar disorder drugs at the transcriptional level.\nAt time of import, last updated (by provider) on: Apr 27 2022\n\nContributors: ; [Srisaiyini Kidnapillai, Chiara Bortolasci, Laura Gray, Trang Truong, Bruna Panizzutti, Mark Richardson, Craig Smith, Olivia Dean, Zoe Liu, Briana Spolding, Michael Berk, Jee H Kim, Ken Walder]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                        Brain circuits are assembled from a large variety of morphologically and functionally diverse cell types. It is not known how the intermingled cell types of individual brain regions differ in their expressed genomes. Here we describe an atlas of cell type transcriptomes of the adult retina. We found that each adult cell type expresses a specific set of genes, including a unique set of transcription factors, forming a “barcode” for cell identity. Cell type transcriptomes carry enough information to categorize cells into corresponding morphological classes and types. Surprisingly, several barcode genes are eye disease-associated genes that we demonstrate to be specifically expressed not only in photoreceptors but also in particular retinal circuit elements such as inhibitory neurons as well as in retinal microglia. Our data suggest that distinct cell types of individual brain regions are characterized by marked differences in their expressed genomes. We assembled a library of 22 transgenic mouse lines in which each line had a group of retinal cells marked with fluorescent proteins. We built up the library with the purpose of having some mouse lines in which single retinal cell types and others in which combinations of types from a single class are labeled. The library had mouse lines with labeled cells representing each of the six retinal cell classes. Retinal cells were characterized by physiological recording and immunohistochemical staining. We isolated 200 fluorescent protein-labeled retinal cells (“cell groups”) from at least three different mice of each mouse line by fluorescence-activated cell sorting. The transcripts of each cell group of these biological triplicates were independently amplified in batches. Each batch contained an internal control cell group from the Arc-line.\nLast Updated (by provider): Jun 10 2014\nContributors:  Botond Roska Erik Cabuy Sandra Siegert</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Lithium is the gold standard treatment for bipolar disorder. The goal of this study was to identify gene expression networks associated with lithium response. RNAseq data was obtained from IPSC derived neurons from lithium responders and non-responders. Focal adhesion was the network most associated with response.\nAt time of import, last updated (by provider) on: Jun 09 2022\n\nContributors: ; [Vipavee Niemsiri, Fred Gage, John Kelsoe]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Microarray experiments were performed using FAC-sorted young photoreceptors to analyze their transcriptome in comparison to remaining retinal cells at same developmental stage and retinal progenitors.\nLast Updated (by provider): Apr 02 2018\nContributors:  Marius Ader Kai Postel</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: This experiment was created by Gemma splitting another: \nExpressionExperiment Id=20933 Name=TCF7L2 lncRNA: A Link between Bipolar Disorder and Body Mass Index through Glucocorticoid Signaling [RNA-Seq] (GSE179921) Bipolar disorder (BD) and obesity are highly comorbid. We previously performed a genome-wide association study (GWAS) for BD risk accounting for the  effect of body mass index (BMI) which identified a genome-wide significant single-nucleotide polymorphism (SNP) in the gene encoding the transcription factor 7 like 2 (TCF7L2). However, the molecular function of TCF7L2 in the central nervous system (CNS) and its possible role in BD and BMI interaction remained unclear. In the present study, we demonstrated by studying human induced pluripotent stem cell (hiPSC)-derived astrocytes, cells which highly express TCF7L2 in the CNS, that the BD-BMI GWAS risk SNP is associated with glucocorticoid-dependent repression of the expression of a previously uncharacterized TCF7L2 transcript variant. That transcript is a long non-coding RNA (lncRNA-TCF7L2) that is highly expressed in the CNS but not in peripheral tissues such as the liver and pancreas which are involved in metabolism.  In astrocytes, knock-down of the lncRNA-TCF7L2 resulted in decreased expression of the parent gene, TCF7L2, as well as alterations in the expression of a series of genes involved in insulin signaling and diabetes.  We also studied the function of TCF7L2 in hiPSC-derived astrocytes by integrating RNA sequencing data after TCF7L2 knock-down with TCF7L2 chromatin-immunoprecipitation sequencing (ChIP-seq) data. Those studies showed that TCF7L2 directly regulated a series of BD-risk genes. In summary, these results support the existence of a CNS-based mechanism underlying BD-BMI genetic risk, a mechanism based on a glucocorticoid-dependent expression quantitative trait locus that regulates the expression of a novel TCF7L2 non-coding transcript.\nAt time of import, last updated (by provider) on: Sep 20 2021\n\nContributors: ; [Mark A Frye, Thanh L Nguyen, Tamas Ordog, Brandon Coombes, Richard M Weinshilboum, Huaizhi Huang, Zhenqing Ye, Liewei Wang, Huanyao Gao, Daniel Kim, Jeong-Heon Lee, Brenna Sharp, Duan Liu, Joanna Biernacka]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       The goals of this study are to examine responses to inflammation in astrocytes from  induced pluripotent stem cells derived from healthy controls and bipolar disorder patients. We examine the transcriptomic inflmmatory signature of generated astrocytes following Il1Beta exposure in BD vs. control Results: BD-patient astrocytes show a unique inflammatory response with differentially regulated genes.\nAt time of import, last updated (by provider) on: Mar 19 2021\n\nContributors: ; [Maxim N Shokhirev, Fred Gage, Krishna Vadodaria, Carol Marchetto]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          98 samples of individuals from 3 different diagnostic groups: bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Fibroblasts and lymphoblastoid cells (LCLs) are the most widely used cells in genetic, genomic, and transcriptomic studies in relation to human diseases. Examining the gene expression patterns in these two cell types will provide valuable information regarding the validity of using them to study gene expression related to various human diseases. Fibroblasts and LCLs from four members of the Old Order Amish family 884 were purchased from Coriell cell repositories (Coriell Institute for Medical Research, Camden, NJ). We used microarrays to profile the patterns of gene expression in these eight cell lines. By employing the PennCNV algorithm to the Illumina HumanHap550 SNP genotyping data, we detected 13 Copy Number Variants (CNV) that exist in these four individuals. CNV-expression association analysis revealed that seven of these 13 CNVs were associated with the expression of genes within or near (&lt;2Mb sweep) these CNVs at a nominal regression P value of 0.05.\nLast Updated (by provider): Feb 20 2009\nContributors:  Maja Bucan Shuzhang Yang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Hippocampus of schizophrenic, bipolar, and control subjects. Analyzed from CEL files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Total RNA sequecing for human induced pluripotent derived cerebral organoids from healthy controls and Bipolar disorder\nAt time of import, last updated (by provider) on: Apr 01 2020\n\nContributors: ; [Annie Kathuria, Rakesh Karmacharya]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Bipolar disorder is a highly heritable mental illness, but the relevant genetic variants and molecular mechanisms are largely unknown. Recent GWAS’s have identified an intergenic region associated with both enhanced cognitive performance and bipolar disorder. This region contains dozens of putative fetal brain-specific enhancers and is located ~0.7 Mb upstream of the neuronal transcription factor POU3F2. We identified a candidate causal variant, rs77910749, that falls within a highly conserved putative enhancer, LC1. This human-specific variant is a single-base deletion in a PAX6 binding site and is predicted to be functional. We hypothesized that rs77910749 alters LC1 activity and hence POU3F2 expression during neurodevelopment. Indeed, transgenic reporter mice demonstrated LC1 activity in the developing cerebral cortex and amygdala. Furthermore, ex vivo reporter assays in embryonic mouse brain and human iPSC-derived cerebral organoids revealed increased enhancer activity conferred by the variant. To probe the in vivo function of LC1, we deleted the orthologous mouse region, which resulted in amygdala-specific changes in Pou3f2 expression. Lastly, ‘humanized’ rs77910749 knock-in mice displayed behavioral defects in sensory gating, an amygdala-dependent endophenotype seen in patients with bipolar disorder. Our study suggests a molecular mechanism underlying the long-speculated link between higher cognition and neuropsychiatric disease.\nAt time of import, last updated (by provider) on: Jul 28 2021\n\nContributors: ; [Susan Q Shen, Joseph C Corbo]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 MicroRNAs have been implicated in the pathology not only of cancer, but also of psychiatric diseases, such as bipolar disorder and schizophrenia. As several psychiatric disorders share the same risk genes, we hypothesized that this microRNA could also be associated with attention-deficit/hyperactivity disorder (ADHD) and that this association to psychiatric disorders might be due to the variable number of tandem repeats (VNTR) polymorphism within the internal miR-137 (Imir137) promoter (PMID 18316599; PMID 25154622). To further understand the role of the microRNA 137 in the brain a knock-down of miR-137 expression in SH-SY5Y neuroblastoma cells was performed followed by expression analysis using a microarray.\nAt time of import, last updated (by provider) on: Aug 08 2019\n\nContributors: ; [Lena Weißflog, Andreas Reif, Stefanie Berger, Heike Weber, Claus J Scholz]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             We have previosuly shown that our Polg(D181A) show spontaneous depressive episodes as a result of mtDNA mutations, but we do not know the cellular mechanisms that link mtDNA mutations to behavioural changes. We hypothesized that mtDNA mutation-induced mitochondrial dysfunction in PVT causes a dysregulation of epigenetics, causing a transcriptional response which ffects neuronal function and ultimately causes the depressive phenotype. We assessed this using a combination of RNA-seq, H3K27Ac ChIP-seq, and ATAC-seq and compared our H3K27Ac results to other brain regions.\nAt time of import, last updated (by provider) on: Jun 01 2021\n\nContributors: ; [Tadafumi Kato, Emilie K Bagge]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Analysis of gene-expression changes in treatment responders vs non-responders to two different treatments among subjectrs participating in LiTMUS. Results provide information on pathways that may be involved in the clinical response to Lithium in patients with bipolar disorder.\nLast Updated (by provider): Apr 01 2013\nContributors:  Robert Beech</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  To define molecular mechanisms underlying rod and cone differentiation, we generated H9 human embryonic stem cell line carrying a GFP reporter that is controlled by the promoter of cone-rod homeobox (CRX) gene, the first known marker of post-mitotic photoreceptor precursors. CRXp-GFP reporter in H9 line replicates endogenous CRX expression when induced to form self-organizing 3-D retina-like tissue. We define temporal transcriptome dynamics of developing photoreceptors during the establishment of cone and rod cell fate. Our studies provide an essential framework for delineating molecules and cellular pathways that guide human photoreceptor development and should assist in chemical screening and cell-based therapies of retinal degeneration.\nLast Updated (by provider): Nov 14 2017\nContributors:  Kohei Homma Rossukon Kaewkhaw Anand Swaroop Koray D Kaya Jizhong Zou Mahendra Rao Matthew Brooks Vijender Chaitankar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Valproate(VPA) has been used in the treatment of bipolar disorder since the 1990s.  However, the therapeutic targetsof VPA have remained elusive. Here we  used RNA sequencing  in human  iPSCs derived from  bipolar  patients  to further identify important molecular targets. Human iPSCs were homogenized and total RNA was isolated using the RNeasy Plus Micro Kit (Qiagen, Hilden, Germany). RNA quantity and quality were assessed using fluorometry (Qubit RNA Broad Range Assay Kit and Fluorometer; Invitrogen, Carlsbad, CA) and chromatography (Bioanalyzer and RNA 6000 Nano Kit; Agilent, Santa Clara, CA), respectively. Libraries were prepared using TruSeq Stranded mRNA (PolyA+) kit (Illumina, San Diego, CA) and sequenced by Illumina NextSeq 500. The read length was 75bp with 30-40M reads per sample. FastQC (v0.11.3) was performed to assess data quality. TopHat2 (v2.0.9) aligned the reads to the mouse reference genome (Mus musculus UCSC mm10) and to the Ensembl human reference genome (GRCh38.p13) using default parameters. Alignments were then converted to expression count data using HTseq (v0.6.1) with default union mode.\nAt time of import, last updated (by provider) on: Dec 31 2020\n\nContributors: ; [George Tseng, Colleen McClung, Wei Zong, Ryan Logan]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      mRNA cancer cell line profiles\nLast Updated (by provider): Oct 25 2013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.Troubled experiment.Accession experiment.Database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE            GSE197966                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE             GSE33085                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE            GSE205422                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE             GSE29318                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE            GSE179921                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE            GSE157509                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE             GSE11767                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE            GSE134497                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE            GSE117877                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE            GSE127771                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE             GSE66433                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE            GSE169212                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE             GSE45484                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE             GSE67645                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE            GSE160761                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE             GSE10843                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE            GSE126942                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       experiment.URI experiment.SampleCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: http://www.ncbi.nlm.nih.gov/geo/                     78</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: http://www.ncbi.nlm.nih.gov/geo/                     34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: http://www.ncbi.nlm.nih.gov/geo/                      9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: http://www.ncbi.nlm.nih.gov/geo/                      4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: http://www.ncbi.nlm.nih.gov/geo/                     33</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                             &lt;NA&gt;                     98</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: http://www.ncbi.nlm.nih.gov/geo/                      8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                             &lt;NA&gt;                     23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: http://www.ncbi.nlm.nih.gov/geo/                     22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: http://www.ncbi.nlm.nih.gov/geo/                     12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12: http://www.ncbi.nlm.nih.gov/geo/                      6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13: http://www.ncbi.nlm.nih.gov/geo/                      6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                             &lt;NA&gt;                     93</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15: http://www.ncbi.nlm.nih.gov/geo/                     20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16: http://www.ncbi.nlm.nih.gov/geo/                    120</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17: http://www.ncbi.nlm.nih.gov/geo/                     23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18: http://www.ncbi.nlm.nih.gov/geo/                     24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: http://www.ncbi.nlm.nih.gov/geo/                    206</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20: http://www.ncbi.nlm.nih.gov/geo/                      6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.LastUpdated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:    2022-10-04 17:22:42</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:    2022-09-01 07:31:34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:    2022-11-28 19:26:04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:    2022-08-17 07:30:12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:    2022-08-09 14:38:17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:    2022-08-17 17:48:14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:    2023-06-08 22:47:42</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:    2022-08-14 15:47:04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:    2020-06-11 16:52:03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:    2022-08-05 07:31:26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:    2022-08-19 19:30:37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:    2022-08-10 01:09:22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:    2022-08-25 00:29:10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:    2022-08-30 23:36:54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:    2022-08-05 20:25:19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:    2022-09-02 07:33:16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:    2022-08-22 07:31:30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:    2021-08-26 07:32:49</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:    2020-03-28 19:12:15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:    2022-08-13 17:05:54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2: This data set may have a batch artifact (PC 1), p=8.2617e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                         &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:  This data set may have a batch artifact (PC 3), p=0.0065166</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: This data set may have a batch artifact (PC 1), p=4.7909e-27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                  0                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                TRUE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE                  0                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                  0                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE                  0                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                  0                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                  1                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE                  1                1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE                  0                0           -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                  0                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE                  0                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE                 -1                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE                  0                0            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       geeq.qScore geeq.sScore taxon.Name taxon.Scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:  0.2824440021      1.0000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:  0.1321881258      1.0000      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:  0.9948397950      0.7500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:  0.8510058562      0.8375      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:  0.8542963453      0.7500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:  0.4225533696      1.0000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:  0.1342491265     -0.1250      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:  0.8545254725      0.7125      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:  0.2819105453      0.2500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:  0.4175250260      0.7500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:  0.2824959983      0.6250      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:  0.2820713713      0.8375      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:            NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:  0.4224650073      0.4375      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  0.8543105102      0.7500      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:  0.4254126991      0.7500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:  0.2817262472      1.0000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:  0.4170825388      0.7500      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: -0.0006619385      1.0000      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:  0.2804561409      0.8375      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.Database.Name taxon.Database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                mm10                81</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

