<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Search everything in Gemma — search_gemma • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Search everything in Gemma — search_gemma"><meta property="og:description" content="Search everything in Gemma"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Search everything in Gemma</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>search_gemma.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Search everything in Gemma</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">search_gemma</span><span class="op">(</span></span>
<span>  <span class="va">query</span>,</span>
<span>  taxon <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  platform <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  resultType <span class="op">=</span> <span class="st">"experiment"</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>The search query. Either plain text ('traumatic'), or an ontology
term URI ('http://purl.obolibrary.org/obo/UBERON_0002048'). Datasets that
contain the given string in their short of full name will also be matched.
Can be multiple identifiers separated by commas.</p></dd>


<dt>taxon</dt>
<dd><p>A numerical taxon identifier or an ncbi taxon identifier or a taxon identifier that matches either its scientific or common name</p></dd>


<dt>platform</dt>
<dd><p>A platform numerical identifier or a platform short name</p></dd>


<dt>limit</dt>
<dd><p>Defaults to 100 with a maximum value of 2000.
Limits the number of returned results. Note
that this function does not support pagination.</p></dd>


<dt>resultType</dt>
<dd><p>The kind of results that should be included in the output. Can be experiment, gene, platform or a long object type name, documented in the API documentation.</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>If <code>raw = FALSE</code> and resultType is experiment, gene or platform,
a data.table containing the search results. If it is any other type, a list
of results. A list with additional details about the search if <code>raw = TRUE</code></p>


    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">search_gemma</span><span class="op">(</span><span class="st">"bipolar"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:            GSE157509</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             GSE66196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:            GSE210064</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             GSE23848</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:             GSE35077</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             GSE98838</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             GSE33085</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            GSE117877</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:          GSE179921.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:   McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:              byne-cc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:            GSE202537</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:            GSE160761</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:            GSE134497</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:             GSE45484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:            GSE169212</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:            GSE197966</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:            GSE205422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:            GSE246593</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:             GSE66433</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:              GSE7036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:             GSE67645</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23:     stanley_feinberg</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24:             GSE46416</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25:             GSE58933</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26:       stanley_altarC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27:       stanley_sklarB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28:             GSE12654</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29:             GSE53987</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30:            GSE116820</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31:           GSE92538.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32:            GSE124326</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33:       stanley_dobrin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34:       stanley_altarB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35:             GSE80655</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36:             GSE46449</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37:           GSE92538.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38:             GSE78936</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39:       stanley_sklarA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40:             GSE39653</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41:             GSE74358</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42:             GSE80336</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43:             GSE35977</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44:             GSE62191</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45:             GSE11767</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46:         stanley_chen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47:             GSE12679</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48:             GSE87610</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49:       stanley_altarA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50:              GSE5389</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51:            GSE208338</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52:             GSE81396</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53:              GSE5388</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54:        stanley_young</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55:         stanley_kato</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56:           McLean_PFC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57:         stanley_bahn</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58:             GSE60190</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59:       stanley_vawter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60:            GSE234795</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61:             GSE18312</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62:            GSE112523</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63:             GSE35974</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64:             GSE12649</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65:             GSE29318</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66:            GSE126942</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67:            GSE127771</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68:             GSE97534</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69:             GSE10843</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70:            GSE215308</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71:          GSE169459.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72:          GSE169459.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73:             GSE29417</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74:            GSE109713</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75:             GSE93577</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76:             GSE53061</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77:            GSE153638</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78:            GSE160874</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79:             GSE85417</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80:            GSE103457</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81:            GSE150945</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82:            GSE151923</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83:            GSE229218</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84:             GSE18794</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85:             GSE55119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86:             GSE54426</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87:            GSE150258</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88:            GSE112510</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89:            GSE120423</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90:             GSE54112</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91:            GSE8641.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92:            GSE8641.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93:            GSE8641.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94:             GSE78877</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95:            GSE137318</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96:            GSE184930</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97:            GSE156533</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98:            GSE152474</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                         experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                  &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                     Increased IL-6 and altered inflammatory response in bipolar disorder patient-derived astrocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                  Bipolar disorder and lithium-induced gene expression in two peripheral cell models</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                 Gene expression alterations in the postmortem hippocampus from older patients with bipolar disorder – a hypothesis generating study</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                   Peripheral blood gene-expression in depressed subjects with bipolar disorder vs healthy controls.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                              A gene expression database for retinal neuron subtypes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                          Combining RNA-Seq and Somatic CRISPR Mutagenesis to Study Mouse Neural Development in vivo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                  Transcriptome analysis of adult retina cell types.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                    A candidate causal variant underlying both enhanced cognitive performance and increased risk of bipolar disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                             Split part 2 of: TCF7L2 lncRNA: A Link between Bipolar Disorder and Body Mass Index through Glucocorticoid Signaling [RNA-Seq] [collection of material = Experiment 1 ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                  McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                Corpus Callosum data from Stanley collection samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                      Diurnal alterations in gene expression across striatal subregions in psychosis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                              RNA sequencing  in human  iPSCs derived from  bipolar  patients  to identify important therapeutic molecular targets of Valproate(VPA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                        Total RNA sequecing for human induced pluripotent derived cerebral organoids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                          Gene-expression differences in peripheral blood between lithium responders and non-responders in the “Lithium Treatment -Moderate dose Use Study” (LiTMUS)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                      Transcriptomic and epigenetic characterization of PVT neurons in bipolar disorder model mouse.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                    Transcriptional effects of bipolar disorder drugs on NT2-N cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                       Network-based integrative analysis of lithium response in bipolar disorder using transcriptomic and GWAS data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                           Transition of allele-specific DNA hydroxymethylation at regulatory loci is associated with phenotypic variation in monozygotic twins discordant for psychiatric disorders</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                            Effects of the microRNA 137 and its connection to psychiatric disorders.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:                                                                                                                           Expression profiling in monozygotic twins discordant for bipolar disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:   Transcriptome dynamics of developing photoreceptors in 3-D retina cultures recapitulates temporal sequence of human cone and rod differentiation revealing cell surface markers and gene networks</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23:                                                                                                                                                 Stanley consortium collection Cerebellum - Feinberg</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24:                                                                                                                                     State- and trait-specific gene expression in euthymia and mania</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25:                                                                                                                         Hyper-excitability of Neurons generated from Patients with Bipolar Disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26:                                                                                                                                                       Stanley consortium collection DLPFC - Altar C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27:                                                                                                                                                   Stanley consortium collection Cerebellum - SklarB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28:                                                                                                                                                 Gene expression from human prefrontal cortex (BA10)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29:                                                                                                  Microarray profiling of PFC, HPC and STR from subjects with schizophrenia, bipolar, MDD or control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30:                                                                                                                    Expression data of Glutarmatergic neuron and GABAergic neruon induced from iPSCs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31:                       Inference of cell-type composition from human brain transcriptomic datasets illuminates the effects of age, manner of death, dissection, and psychiatric diagnosis - GPL10526</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32:                                                                                                                Whole blood transcriptome analysis in bipolar disorder reveals strong lithium effect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33:                                                                                                                                                             Stanley array collection DLPFC - Dobrin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34:                                                                                                                                                       Stanley consortium collection DLPFC - Altar B</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35:                                                                                                                                                   RNA-sequencing of human post-mortem brain tissues</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36:                                                                                                               Expression data from Patients with Bipolar (BP) Disorder and Matched Control Subjects</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37:                       Inference of cell-type composition from human brain transcriptomic datasets illuminates the effects of age, manner of death, dissection, and psychiatric diagnosis - GPL17027</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38:                                                                            Systematically characterizing dysfunctional long intergenic non-coding RNAs in multiple brain regions of major psychosis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39:                                                                                                                                                        Stanley consortium collection DLPFC - SklarA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40:                                                                                                                                        Differential Gene Expression in Patients with Mood Disorders</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41:                                                                        Transcriptomic Comparison of Neuronal Development Stages using Induced Pluripotent Stem Cells from Bipolar Disorder Patients</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42:                                                                                                                            Transcriptome profiling of the human dorsal striatum in bipolar disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43:                                                                                                                                                Expression data from the human parietal cortex brain</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44:                                                                                                      Gene expression profiles of patients with schizophrenia, bipolar disorder and healthy controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45:                                                                                                     Gene expression profiling of fibroblasts and lymphoblastoid cells derived from four individuals</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46:                                                                                                                                                          Stanley consortium collection DLPFC - Chen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47:                                                                                           Laser capture microdissection of endothelial and neuronal cells from human dorsolateral prefrontal cortex</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48:                                                Gene expression of L3 and L5 pyramidal neurons in the DLPFC comparing schizophrenia from bipolar major depressive disorders and unaffected subjects.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49:                                                                                                                                                            Stanley array collection DLPFC - Altar A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50:                                                                                       Adult postmortem brain tissue (orbitofrontal cortex) from subjects with bipolar disorder and healthy controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51:                                                                                     Expression data from postmortem human dorsolateral prefrontal cortex - psychiatric disorders &amp; healthy controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52:                                                                                       Total RNAseq of human putamen and caudate nucleus tissues in healthy control and Bipolar Disorder individuals</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53:                                                                             Adult postmortem brain tissue (dorsolateral prefrontal cortex) from subjects with bipolar disorder and healthy controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54:                                                                                                                                                              Stanley array collection DLPFC - Young</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55:                                                                                                                                                               Stanley array collection DLPFC - Kato</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56:                                                                                                                                                                                          McLean_PFC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57:                                                                                                                                                               Stanley array collection DLPFC - Bahn</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58:                                                                                                                                           Genetic Neuropathology of Obsessive Psychiatric Syndromes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59:                                                                                                                                                             Stanley array collection DLPFC - Vawter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60:                                                                                                                                      REST and Impaired Neural Stress Resistance in Bipolar Disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61:                                                                                                                                       Gene Expression in Blood in Scizophrenia and Bipolar Disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62:                                                                                                  DNA methylation in neurons from post-mortem brains in schizophrenia and bipolar disorder (RNA-Seq)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63:                                                                                                                                                     Expression data from the human cerebellum brain</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64:                                                                                                                                                 Gene expression from human prefrontal cortex (BA46)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65:                                                                                                                                               Expression profile of FAC-sorted murine retinal cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66:                                                The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67:                                           The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells [P7]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68:                                                                                                                                                    RNA sequencing of developing cone photoreceptors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69:                                                                                                                                                                      mRNA Cancer Cell Line Profiles</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70:                                                                                                     MUC1-C IS A MASTER REGULATOR OF MICA/B NKG2D LIGAND AND EXOSOME SECRETION IN HUMAN CANCER CELLS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71:                                                                                            Molecular regulation of melatonin biosynthesis pathway in unipolar and bipolar depression - Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72:                                                                                            Molecular regulation of melatonin biosynthesis pathway in unipolar and bipolar depression - Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73:                                                                                                                   A New Mouse Model for Mania Shares Genetic Correlates with Human Bipolar Disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74:                                                                        Retinal Cell Type Epigenetic Memory Predicts Reprogramming Efficiency and Retinogenesis in 3D Organoid Cultures [RNA-Seq_Mm]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75:                                                                                             Gene expression of L3 parvalbumin neurons in the DLPFC comparing schizophrenia with unaffected subjects</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76:                                                                                                           Genome-wide analysis of gene expression in wild-type and Sp4 hypomorphic mouse cerebellum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77:                                                                                                                        RNA-seq analysis of rat neurons treated with shRNA-mediated Trank1 knockdown</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78:                                                                                               RNA sequencing in 12-week old orbital frontal cortex in circHomer1/Homer1b knockdown and control mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79:                                                                                                                 Transcriptome of iPSC-derived Cerebral Organoids with Heterozygous Knockout in CHD8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80:                                                                                   Quantitative Analysis of Wildtype and Neurog2CKO Heterozygote and Mutant Retinal Transcriptomes by RNA Sequencing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81:                                                                                                          Genome-wide gene expression prolifing post GNL3 knockdown in human neural progenitor cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82:                                                                                                                          Transcriptional profiling of cortex from 6-month-old wildtype C57BL/6 mice</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83: An allelic series of spontaneous Rorb mutant mice exhibit a gait phenotype, changes in retina morphology and behavior, and gene expression signatures associated with the unfolded protein response</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84:                                                                                          Mycobacterium tuberculosis Chaperonin 60.1 has Bipolar Effects on Human peripheral blood-derived Monocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85:                                                                                                                                                         Expression data from mouse ventral midbrain</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86:                                                        Medial prefrontal cortex: genes linked to bipolar disorder and schizophrenia have altered expression in the highly social maternal phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87:                                                                                                  Differential susceptibility of retinal neurons to the loss of mitochondrial biogenesis factor Nrf1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88:                                                                                   Nitrated meat products are associated with mania in humans and altered behavior and brain gene expression in rats</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89:                                                                                          Brain transcriptome profiling in wildtype mice and mice with Igf2 enhancer deletion (Igf2enh-/-) [RNA-seq]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90:                                                                                         ZNF804A transcriptome networks in differentiating human neurons derived from induced pluripotent stem cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91:                                                                                        Rnf41 is associated with anxiety like behavior, major depression and beta carboline induced seizure - GPL340</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92:                                                                                        Rnf41 is associated with anxiety like behavior, major depression and beta carboline induced seizure - GPL339</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93:                                                                                       Rnf41 is associated with anxiety like behavior, major depression and beta carboline induced seizure - GPL4055</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94:                                                                                                     Gene expression differences between wildtype and Atrx conditional knockout mouse retina tissues</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95:                                                                                                             Cis-Regulatory Accessibility Directs Muller Glial Development and Regenerative Capacity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96:                                                                                                                                                Lithium treatment and human hippocampal neurogenesis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97:                                                                                                                     Transcriptomic profiling in 3-month-old P23H Retinitis Pigmentosa mouse retinas</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98:                                                                                                                        Transcriptomic profiling in juvenile P23H Retinitis Pigmentosa mouse retinas</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                         experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:         23425</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:         24428</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:         28041</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:          1958</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:          7608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:         13214</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:          8887</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:         20617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:         21159</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:           670</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:          4354</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:         25749</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:         17943</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:         16450</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:          6145</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:         19187</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:         25070</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:         25972</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:         31703</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:         24427</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:           660</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:         11634</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23:           831</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24:          8997</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25:         12503</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26:           840</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27:           835</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28:          1204</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29:          8359</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30:         14828</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31:         18742</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32:         23636</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33:           843</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34:           839</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35:         13014</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36:          6583</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37:         18741</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38:         12983</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39:           834</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40:          5724</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41:         12701</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42:         10627</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43:          5949</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44:          8995</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45:          1879</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46:           842</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47:           946</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48:         10878</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49:           838</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50:           326</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51:         26880</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52:         13338</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53:           594</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54:           837</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55:           833</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56:           672</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57:           841</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58:          8994</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59:           836</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60:         31677</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61:          5804</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62:         16773</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63:          5939</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64:          1205</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65:         12310</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66:         19454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67:         19455</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68:         13694</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69:          7196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70:         33397</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71:         24344</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72:         24345</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73:          4657</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74:         12097</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75:         14141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76:         12419</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77:         17318</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78:         24621</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79:         11666</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80:         11819</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81:         17343</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82:         26363</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83:         27738</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84:          2259</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85:         14457</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86:         12434</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87:         26953</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88:         14188</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89:         15231</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90:          9777</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91:          1629</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92:          1628</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93:          1630</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94:         12982</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95:         24907</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96:         20334</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97:         17673</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98:         17330</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          The goals of this study are to examine responses to inflammation in astrocytes from  induced pluripotent stem cells derived from healthy controls and bipolar disorder patients. We examine the transcriptomic inflmmatory signature of generated astrocytes following Il1Beta exposure in BD vs. control Results: BD-patient astrocytes show a unique inflammatory response with differentially regulated genes.\nAt time of import, last updated (by provider) on: Mar 19 2021\n\nContributors: ; [Maxim N Shokhirev, Fred Gage, Krishna Vadodaria, Carol Marchetto]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Bipolar disorder is a severe and heritable psychiatric disorder and affects up to 1% of the population worldwide. Lithium is recommended as first-line treatment for the maintenance treatment of bipolar-affective disorder in current guidelines, its molecular modes of action are however poorly understood. Cell models derived from bipolar patients could prove useful to gain more insight in the molecular mechanisms of bipolar disorder and the common pharmacological treatments. As primary neuronal cell lines cannot be easily derived from patients, peripheral cell models should be evaluated in their usefulness to study pathomechanisms and the mode of action of medication as well as in regard to develop biomarkers for diagnosis and treatment response.\nAt time of import, last updated (by provider) on: Mar 25 2019\n\nContributors: ; [Sarah Kittel-Schneider, Max Hilscher, Andreas Reif, Claus J Scholz]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Gene expression of samples from the postmortem hippocampus of older bipolar disorder subjects and controls. Gene expression data was generated using the SurePrint G3 Human Gene Expression v3 microarray. Rank feature selection was performed to identify a subset of features that can optimally differentiate BD and controls.\nAt time of import, last updated (by provider) on: Feb 19 2023\n\nContributors: ; [Carlos A Pasqualucci, Claudia K Suemoto, Ricardo Nitrini, Fernanda B Bertonha, Paula V Nunes, Katia C De Oliveira, Carlos M Filho, Helena K Kim, Helena Brentani, Lea T Grinberg, Beny Lafer, André Barbosa, Camila Nascimento, Renata P Leite, Wilson Jacob-Filho]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Analysis of gene-expression changes in depressed subjects with bipolar disorder compared to healthy controls. Results provide information on pathways that may be involved in the pathogenesis of bipolar depression.\nLast Updated (by provider): Aug 27 2010\nContributors:  Robert D Beech</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             The goal of this experiment was to define gene expression patterns of thirteen mouse retinal neuron subsets, labeled by expression of fluorescent proteins in transgenic mice.\nLast Updated (by provider): Sep 23 2013\nContributors:  Jeremy N Kay Joshua R Sanes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            We used RNA sequencing to identify candidate regulators of interactions between photoreceptor axons and bipolar cell (BCs) dendrites in developing mouse retina. We chose three time points: P7, just after the OPL forms and synaptogenesis with BCs begins; P13, as synaptogenesis nears completion and sublamination begins; and P30, when the OPL is mature. We purified cone and rod photoreceptors separately by fluorescence activated cell sorting (FACS) using transgene markers: Rhoicre;Ai9 for rods and HRGPcre;Ai9 for cones. We purified ON BCs, which include ON cone bipolars plus rod bipolars using Grm6:GFP. As appropriate transgenic lines to separate RBCs from CBCs were not available, we performed RNAseq on cells from Grm6:GFP mice that were fixed and immunostained prior to FACS, allowing us to purify RBCs (GFP+PKC+) and CBCs (GFP+PKC-) from the same retinas. As PKC is not highly expressed at P7, profiling of developing rod bipolars separate from developing ON cone bipolars was restricted to P13.\nLast Updated (by provider): Jun 25 2018\nContributors:  Elizabeth Z Sanchez Lawrence S Zipursky Yerbol Z Kurmangaliyev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Brain circuits are assembled from a large variety of morphologically and functionally diverse cell types. It is not known how the intermingled cell types of individual brain regions differ in their expressed genomes. Here we describe an atlas of cell type transcriptomes of the adult retina. We found that each adult cell type expresses a specific set of genes, including a unique set of transcription factors, forming a “barcode” for cell identity. Cell type transcriptomes carry enough information to categorize cells into corresponding morphological classes and types. Surprisingly, several barcode genes are eye disease-associated genes that we demonstrate to be specifically expressed not only in photoreceptors but also in particular retinal circuit elements such as inhibitory neurons as well as in retinal microglia. Our data suggest that distinct cell types of individual brain regions are characterized by marked differences in their expressed genomes. We assembled a library of 22 transgenic mouse lines in which each line had a group of retinal cells marked with fluorescent proteins. We built up the library with the purpose of having some mouse lines in which single retinal cell types and others in which combinations of types from a single class are labeled. The library had mouse lines with labeled cells representing each of the six retinal cell classes. Retinal cells were characterized by physiological recording and immunohistochemical staining. We isolated 200 fluorescent protein-labeled retinal cells (“cell groups”) from at least three different mice of each mouse line by fluorescence-activated cell sorting. The transcripts of each cell group of these biological triplicates were independently amplified in batches. Each batch contained an internal control cell group from the Arc-line.\nLast Updated (by provider): Jun 10 2014\nContributors:  Botond Roska Erik Cabuy Sandra Siegert</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Bipolar disorder is a highly heritable mental illness, but the relevant genetic variants and molecular mechanisms are largely unknown. Recent GWAS’s have identified an intergenic region associated with both enhanced cognitive performance and bipolar disorder. This region contains dozens of putative fetal brain-specific enhancers and is located ~0.7 Mb upstream of the neuronal transcription factor POU3F2. We identified a candidate causal variant, rs77910749, that falls within a highly conserved putative enhancer, LC1. This human-specific variant is a single-base deletion in a PAX6 binding site and is predicted to be functional. We hypothesized that rs77910749 alters LC1 activity and hence POU3F2 expression during neurodevelopment. Indeed, transgenic reporter mice demonstrated LC1 activity in the developing cerebral cortex and amygdala. Furthermore, ex vivo reporter assays in embryonic mouse brain and human iPSC-derived cerebral organoids revealed increased enhancer activity conferred by the variant. To probe the in vivo function of LC1, we deleted the orthologous mouse region, which resulted in amygdala-specific changes in Pou3f2 expression. Lastly, ‘humanized’ rs77910749 knock-in mice displayed behavioral defects in sensory gating, an amygdala-dependent endophenotype seen in patients with bipolar disorder. Our study suggests a molecular mechanism underlying the long-speculated link between higher cognition and neuropsychiatric disease.\nAt time of import, last updated (by provider) on: Jul 28 2021\n\nContributors: ; [Susan Q Shen, Joseph C Corbo]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    This experiment was created by Gemma splitting another: \nExpressionExperiment Id=20933 Name=TCF7L2 lncRNA: A Link between Bipolar Disorder and Body Mass Index through Glucocorticoid Signaling [RNA-Seq] (GSE179921) Bipolar disorder (BD) and obesity are highly comorbid. We previously performed a genome-wide association study (GWAS) for BD risk accounting for the  effect of body mass index (BMI) which identified a genome-wide significant single-nucleotide polymorphism (SNP) in the gene encoding the transcription factor 7 like 2 (TCF7L2). However, the molecular function of TCF7L2 in the central nervous system (CNS) and its possible role in BD and BMI interaction remained unclear. In the present study, we demonstrated by studying human induced pluripotent stem cell (hiPSC)-derived astrocytes, cells which highly express TCF7L2 in the CNS, that the BD-BMI GWAS risk SNP is associated with glucocorticoid-dependent repression of the expression of a previously uncharacterized TCF7L2 transcript variant. That transcript is a long non-coding RNA (lncRNA-TCF7L2) that is highly expressed in the CNS but not in peripheral tissues such as the liver and pancreas which are involved in metabolism.  In astrocytes, knock-down of the lncRNA-TCF7L2 resulted in decreased expression of the parent gene, TCF7L2, as well as alterations in the expression of a series of genes involved in insulin signaling and diabetes.  We also studied the function of TCF7L2 in hiPSC-derived astrocytes by integrating RNA sequencing data after TCF7L2 knock-down with TCF7L2 chromatin-immunoprecipitation sequencing (ChIP-seq) data. Those studies showed that TCF7L2 directly regulated a series of BD-risk genes. In summary, these results support the existence of a CNS-based mechanism underlying BD-BMI genetic risk, a mechanism based on a glucocorticoid-dependent expression quantitative trait locus that regulates the expression of a novel TCF7L2 non-coding transcript.\nAt time of import, last updated (by provider) on: Sep 20 2021\n\nContributors: ; [Mark A Frye, Thanh L Nguyen, Tamas Ordog, Brandon Coombes, Richard M Weinshilboum, Huaizhi Huang, Zhenqing Ye, Liewei Wang, Huanyao Gao, Daniel Kim, Jeong-Heon Lee, Brenna Sharp, Duan Liu, Joanna Biernacka]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Hippocampus of schizophrenic, bipolar, and control subjects. Analyzed from CEL files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Background: Psychosis is a defining feature of schizophrenia and highly prevalent in bipolar disorder. Notably, individuals suffering with these illnesses also have major disruptions in sleep and circadian rhythms, and disturbances to sleep and circadian rhythms can precipitate or exacerbate psychotic symptoms. Psychosis is associated with the striatum, though no study to date has directly measured molecular rhythms and determined how they are altered in the striatum of subjects with psychosis. Methods: Here, we perform RNA-sequencing and both differential expression and rhythmicity analyses to investigate diurnal alterations in gene expression in human postmortem striatal subregions (NAc, caudate, and putamen) in subjects with psychosis relative to unaffected comparison subjects. Results: Across regions, we find differential expression of immune-related transcripts and a substantial loss of rhythmicity in core circadian clock genes in subjects with psychosis. In the nucleus accumbens (NAc), mitochondrial-related transcripts have decreased expression in psychosis subjects, but only in those who died at night. Additionally, we find a loss of rhythmicity in small nucleolar RNAs and a gain of rhythmicity in glutamatergic signaling in the NAc of psychosis subjects. Between region comparisons indicate that rhythmicity in the caudate and putamen is far more similar in subjects with psychosis than in matched comparison subjects. Conclusions: Together, these findings reveal differential and rhythmic gene expression differences across the striatum that may contribute to striatal dysfunction and psychosis in psychotic disorders.\nAt time of import, last updated (by provider) on: Aug 31 2022\n\nContributors: ; [George Tseng, Colleen McClung, Wei Zong, Kyle Ketchesin]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Valproate(VPA) has been used in the treatment of bipolar disorder since the 1990s.  However, the therapeutic targetsof VPA have remained elusive. Here we  used RNA sequencing  in human  iPSCs derived from  bipolar  patients  to further identify important molecular targets. Human iPSCs were homogenized and total RNA was isolated using the RNeasy Plus Micro Kit (Qiagen, Hilden, Germany). RNA quantity and quality were assessed using fluorometry (Qubit RNA Broad Range Assay Kit and Fluorometer; Invitrogen, Carlsbad, CA) and chromatography (Bioanalyzer and RNA 6000 Nano Kit; Agilent, Santa Clara, CA), respectively. Libraries were prepared using TruSeq Stranded mRNA (PolyA+) kit (Illumina, San Diego, CA) and sequenced by Illumina NextSeq 500. The read length was 75bp with 30-40M reads per sample. FastQC (v0.11.3) was performed to assess data quality. TopHat2 (v2.0.9) aligned the reads to the mouse reference genome (Mus musculus UCSC mm10) and to the Ensembl human reference genome (GRCh38.p13) using default parameters. Alignments were then converted to expression count data using HTseq (v0.6.1) with default union mode.\nAt time of import, last updated (by provider) on: Dec 31 2020\n\nContributors: ; [George Tseng, Colleen McClung, Wei Zong, Ryan Logan]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Total RNA sequecing for human induced pluripotent derived cerebral organoids from healthy controls and Bipolar disorder\nAt time of import, last updated (by provider) on: Apr 01 2020\n\nContributors: ; [Annie Kathuria, Rakesh Karmacharya]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Analysis of gene-expression changes in treatment responders vs non-responders to two different treatments among subjectrs participating in LiTMUS. Results provide information on pathways that may be involved in the clinical response to Lithium in patients with bipolar disorder.\nLast Updated (by provider): Apr 01 2013\nContributors:  Robert Beech</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                We have previosuly shown that our Polg(D181A) show spontaneous depressive episodes as a result of mtDNA mutations, but we do not know the cellular mechanisms that link mtDNA mutations to behavioural changes. We hypothesized that mtDNA mutation-induced mitochondrial dysfunction in PVT causes a dysregulation of epigenetics, causing a transcriptional response which ffects neuronal function and ultimately causes the depressive phenotype. We assessed this using a combination of RNA-seq, H3K27Ac ChIP-seq, and ATAC-seq and compared our H3K27Ac results to other brain regions.\nAt time of import, last updated (by provider) on: Jun 01 2021\n\nContributors: ; [Tadafumi Kato, Emilie K Bagge]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Human neuronal-like cells (NT2-N) were treated with either lamotrigine (50 µM), lithium (2.5 mM), quetiapine (50 µM), valproate (0.5 mM) or vehicle control for 24 hours. Genome wide mRNA expression was quantified by RNA-sequencing. Results offer insights on the mechanism(s) of action of bipolar disorder drugs at the transcriptional level.\nAt time of import, last updated (by provider) on: Apr 27 2022\n\nContributors: ; [Srisaiyini Kidnapillai, Chiara Bortolasci, Laura Gray, Trang Truong, Bruna Panizzutti, Mark Richardson, Craig Smith, Olivia Dean, Zoe Liu, Briana Spolding, Michael Berk, Jee H Kim, Ken Walder]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Lithium is the gold standard treatment for bipolar disorder. The goal of this study was to identify gene expression networks associated with lithium response. RNAseq data was obtained from IPSC derived neurons from lithium responders and non-responders. Focal adhesion was the network most associated with response.\nAt time of import, last updated (by provider) on: Jun 09 2022\n\nContributors: ; [Vipavee Niemsiri, Fred Gage, John Kelsoe]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Major psychiatric disorders such as schizophrenia (SCZ) and bipolar disorder (BPD) are complex genetic mental illnesses. Their non-Mendelian features such as monozygotic twins discordant for SCZ or BPD are likely complicated by environmental modifiers of genetic effects. 5-hydroxymethylcytosine (5hmC) is an important epigenetic marker in gene regulation and whether its links with genetic variants contribute to the non-Mendelian features remain largely unexplored. Here, we performed hydroxymethylome and genome analyses of blood DNA from psychiatric disorder-discordant monozygotic twins to study how allele-specific hydroxymethylation (AShM) mediates phenotypic variations. We identified thousands of genetic variants with AShM imbalances who exhibit phenotypic variation-associated AShM transition at regulatory loci. These AShMs have plausible causal associations with psychiatric disorders through effects on interactions between transcription factors, DNA methylations, or other epigenomic markers and then contribute to dysregulated gene expression, which eventually increases disease susceptibility. We then validated that competitive binding of POU3F2 on the alternative allele of psyAShM site rs4558409 (G/T) in PLLP can enhance the PLLP expression, while hydroxymethylated alternative allele alleviating the transcription factor binding activity at rs4558409 site might be associated with downregulated PLLP expression observed in BPD or SCZ. Moreover, disruption of rs4558409 induces gain of PLLP function and promotes neural development and vesicle trafficking. Our study provides a powerful strategy for prioritizing regulatory risk variants and contributes to our understanding of the interplay between genetic and epigenetic factors in mediating complex disease susceptibility.\nAt time of import, last updated (by provider) on: Oct 31 2023\n\nContributors: ; [Zhanwang Huang, Junping Ye]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    MicroRNAs have been implicated in the pathology not only of cancer, but also of psychiatric diseases, such as bipolar disorder and schizophrenia. As several psychiatric disorders share the same risk genes, we hypothesized that this microRNA could also be associated with attention-deficit/hyperactivity disorder (ADHD) and that this association to psychiatric disorders might be due to the variable number of tandem repeats (VNTR) polymorphism within the internal miR-137 (Imir137) promoter (PMID 18316599; PMID 25154622). To further understand the role of the microRNA 137 in the brain a knock-down of miR-137 expression in SH-SY5Y neuroblastoma cells was performed followed by expression analysis using a microarray.\nAt time of import, last updated (by provider) on: Aug 08 2019\n\nContributors: ; [Lena Weißflog, Andreas Reif, Stefanie Berger, Heike Weber, Claus J Scholz]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  To identify genes dysregulated in bipolar disorder (BD1) we carried out global gene expression profiling using whole-genome microarrays. To minimize genetic variation in gene expression levels between cases and controls we compared expression profiles in lymphoblastoid cell lines from monozygotic twin pairs discordant for the disease. We identified 82 genes that were differentially expressed by ? 1.3-fold in 3 BD1 cases compared to their co-twins, and which were statistically (p ? 0.05) differentially expressed between the groups of BD1 cases and controls. Using qRT-PCR we confirmed the differential expression of some of these genes, including: KCNK1, MAL, PFN2, TCF7, PGK1, and PI4KCB, in at least 2 of the twin pairs. In contrast to the findings of a previous study by Kakiuchi and colleagues with similar discordant BD1 twin design1 our data do not support the dysregulation of XBP1 and HSPA5. From pathway and gene ontology analysis we identified up-regulation of the WNT signalling pathway and the biological process of apoptosis. The differentially regulated genes and pathways identified in this study may provide insights into the biology of BD1.\nLast Updated (by provider): Jun 20 2007\nContributors:  Louisa Windus Nicholas Matigian Bryan Mowry Cheryl Filippich John McGrath Heather Smith Nicholas Hayward Christos Pantelis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     To define molecular mechanisms underlying rod and cone differentiation, we generated H9 human embryonic stem cell line carrying a GFP reporter that is controlled by the promoter of cone-rod homeobox (CRX) gene, the first known marker of post-mitotic photoreceptor precursors. CRXp-GFP reporter in H9 line replicates endogenous CRX expression when induced to form self-organizing 3-D retina-like tissue. We define temporal transcriptome dynamics of developing photoreceptors during the establishment of cone and rod cell fate. Our studies provide an essential framework for delineating molecules and cellular pathways that guide human photoreceptor development and should assist in chemical screening and cell-based therapies of retinal degeneration.\nLast Updated (by provider): Nov 14 2017\nContributors:  Kohei Homma Rossukon Kaewkhaw Anand Swaroop Koray D Kaya Jizhong Zou Mahendra Rao Matthew Brooks Vijender Chaitankar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               50 samples of individuals from 4 different diagnostic groups: bipolar, schizophrenia, depression and controls. Samples taken from the Cerebellum.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Gene expression profiles of bipolar disorder (BD) patients were assessed during both a manic and a euthymic phase and compared both intra-individually, and with the gene expression profiles of controls.\nLast Updated (by provider): Sep 05 2014\nContributors:  Christian C Witt Benedikt Brors Dilafruz Juraeva Jens Treutlein Carsten Sticht Stephanie H Witt Jana Strohmaier Helene Dukal Josef Frank Franziska Degenhardt Markus M Nöthen Sven Cichon Maren Lang Marcella Rietschel Sandra Meier Manuel Mattheisen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Bipolar Disorder (BD) is a complex neuropsychiatric disorder that is characterized by intermittent episodes of mania and depression and, without treatment, 15% of patients commit suicide1. Hence, among all diseases, BD has been ranked by the WHO as a top disorder of morbidity and lost productivity2. Previous neuropathological studies have revealed a series of alterations in the brains of BD patients or animal models3, such as reduced glial cell number in the patient prefrontal cortex4, up-regulated activities of the PKA/PKC pathways5-7, and changes in dopamine/5-HT/glutamate neurotransmission systems8-11. However, the roles and causation of these changes in BD are too complex to exactly determine the pathology of the disease; none of the current BD animal models can recapitulate both the manic and depressive phenotypes or spontaneous cycling of BD simultaneously12,13. Furthermore, while some patients show remarkable improvement with lithium treatment, for yet unknown reasons, other patients are refractory to lithium treatment. Therefore, developing an accurate and powerful biological model has been a challenge for research into BD. The development of induced pluripotent stem cell (iPSC) technology has provided such a new approach. Here, we developed a human BD iPSC model and investigated the cellular phenotypes of hippocampal dentate gyrus neurons derived from the patient iPSCs. Using patch clamp recording, somatic Ca2+ imaging and RNA-seq techniques, we found that the neurons derived from BD patients exhibited hyperactive action potential (AP) firing, up-regulated expression of PKA/PKC/AP and mitochondria-related genes. Moreover, lithium selectively reversed these alterations in the neurons of patients who responded to lithium treatment. Therefore, hyper-excitability is one endophenotype of BD that is probably achieved through enhancement in the PKA/PKC and Na+ channel signaling systems, and our BD iPSC model can be used to develop new therapies and drugs aimed at clinical treatment of this disease.\nLast Updated (by provider): Jun 11 2018\nContributors:  Son Pham Jun Yao Fred H Gage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           44 samples of individuals from four different diagnostic groups: depression, bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46/10.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               46 samples of individuals from 4 different diagnostic groups: bipolar, schizophrenia, depression and controls. Samples taken from the Cerebellum.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       We performed the oligonucleotide microarray analysis in bipolar disorder, major depression, schizophrenia, and control subjects using postmortem prefrontal cortices provided by the Stanley Foundation Brain Collection. By comparing the gene expression profiles of similar but distinctive mental disorders, we explored the uniqueness of bipolar disorder and its similarity to other mental disorders at the molecular level. Notably, most of the altered gene expressions in each disease were not shared by one another, suggesting the molecular distinctiveness of these mental disorders. We found a tendency of downregulation of the genes encoding receptor, channels or transporters, and upregulation of the genes encoding stress response proteins or molecular chaperons in bipolar disorder. Altered expressions in bipolar disorder shared by other mental disorders mainly consisted of upregulation of the genes encoding proteins for transcription or translation. The genes identified in this study would be useful for the understanding of the pathophysiology of bipolar disorder, as well as the common pathophysiological background in major mental disorders at the molecular level.\nLast Updated (by provider): Mar 15 2009\nContributors:  Tadafumi Kato Kazuya Iwamoto Chihiro Kakiuchi Kazuhiko Ikeda Miki Bundo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Schizophrenia is a complex psychiatric disorder encompassing a range of symptoms and  etiology dependent upon the interaction of genetic and environmental factors.  Several risk genes, such as DISC1, have been associated with schizophrenia as well as bipolar disorder (BPD) and major depressive disorder (MDD), consistent with the hypothesis that a shared genetic architecture could contribute to divergent clinical syndromes.  The present study compared gene expression profiles across three brain regions in post-mortem tissue from matched subjects with schizophrenia, BPD or MDD and unaffected controls.  Post-mortem brain tissue was collected from control subjects and well-matched subjects with schizophrenia, BPD, and MDD (n=19 from each group).  RNA was isolated from hippocampus, Brodmann Area 46, and associative striatum and hybridized to U133_Plus2 Affymetrix chips.  Data were normalized by RMA, subjected to pairwise comparison followed by Benjamini and Hochberg False Discovery Rate correction (FDR).  Samples derived from patients with schizophrenia exhibited  many more changes in gene expression across all brain regions than observed in BPD or MDD.  Several genes showed changes in both schizophrenia and BPD, though the magnitude of change was usually larger in schizophrenia.  Several genes that have variants associated with schizophrenia were found to have altered expression in multiple regions of brains from subjects with schizophrenia.  Continued evaluation of circuit-level alterations in gene expression and gene-network relationships may further our understanding of how genetic variants may be influencing biological processes to contribute to psychiatric disease.\nLast Updated (by provider): May 19 2014\nContributors:  Thomas A Lanz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       We used microarrays to identify the differently expressed genes in disease model for bipolar disorder and schizophrenia.\nAt time of import, last updated (by provider) on: Feb 01 2019\n\nContributors: ; [Takaya Ishii, Hideyuki Okano]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31:  Most neuroscientists would agree that psychiatric illness is unlikely to arise from pathological changes that occur uniformly across all cells in a given brain region. Despite this fact, the majority of transcriptomic analyses of the human brain to date are conducted using macro-dissected tissue due to the difficulty of conducting single-cell level analyses on donated post-mortem brains. To address this issue statistically, we compiled a database of several thousand transcripts that were specifically-enriched in one of 10 primary brain cell types identified in published single cell type transcriptomic experiments. Using this database, we predicted the relative cell type composition for 157 human dorsolateral prefrontal cortex samples using Affymetrix microarray data collected by the Pritzker Neuropsychiatric Consortium, as well as for 841 samples spanning 160 brain regions included in an Agilent microarray dataset collected by the Allen Brain Atlas. These predictions were generated by averaging normalized expression levels across the transcripts specific to each primary cell type to create a “cell type index”. Using this method, we determined that the expression of cell type specific transcripts identified by different experiments, methodologies, and species clustered into three main cell type groups: neurons, oligodendrocytes, and astrocytes/support cells. Overall, the principal components of variation in the data were largely explained by the neuron to glia ratio of the samples. When comparing across brain regions, we were able to easily capture canonical cell type signatures – increased endothelial cells and vasculature in the choroid plexus, oligodendrocytes in the corpus callosum, astrocytes in the central glial substance, neurons and immature cells in the dentate gyrus, and oligodendrocytes and interneurons in the globus pallidus. The relative balance of these cell types was influenced by a variety of demographic, pre- and post-mortem variables. Age and prolonged anaerobic conditions around the time of death were associated with decreased neuronal content and increased astrocytic and endothelial content in the tissue, replicating the known higher vulnerability of neurons to aging and adverse conditions, and illustrating the proliferation of vasculature in a hypoxic environment. We also found that the red blood cell content was reduced in individuals who died in a manner that involved systemic blood loss. Finally, statistically accounting for cell type improved both the sensitivity and interpretability of diagnosis effects within the data. We were able to observe a decrease in astrocytic content in subjects with Major Depressive Disorder, mirroring what had been previously observed morphometrically. By including a set of “cell type indices” in a larger model examining the relationship between gene expression and neuropsychiatric illness, we were able to successfully detect almost twice as many genes with previously identified relationships to bipolar disorder and schizophrenia than using traditional analysis methods.\nAt time of import, last updated (by provider) on: \n\nContributors: ; [Jun Z Li, Cortney A Turner, Megan H Hagenauer, Stanley J Watson, David M Walsh, Alan F Schatzberg, Huda Akil, Richard M Myers, William E Bunney, Jack D Barchas]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        BACKGROUND: Bipolar disorder (BD) is a highly heritable mood disorder with complex genetic architecture and poorly understood etiology. Previous transcriptomic BD studies have had inconsistent findings due to issues such as small sample sizes and difficulty in adequately accounting for confounders like medication use.  METHODS: We performed a differential expression analysis in a well-characterized BD case-control sample (Nsubjects = 480) by RNA sequencing of whole blood. We further performed co-expression network analysis, functional enrichment, and cell type decomposition, and integrated differentially expressed genes with genetic risk.  RESULTS: While we observed widespread differential gene expression patterns between affected and unaffected individuals, these effects were largely linked to lithium treatment at the time of blood draw (FDR &lt; 0.05, Ngenes = 976) rather than BD diagnosis itself (FDR &lt; 0.05, Ngenes = 6). These lithium-associated genes were enriched for cell signaling and immune response functional annotations, among others, and were associated with neutrophil cell-type proportions, which were elevated in lithium users. Neither genes with altered expression in cases nor in lithium users were enriched for BD, schizophrenia, and depression genetic risk based on information from genome-wide association studies, nor was gene expression associated with polygenic risk scores for BD.  CONCLUSIONS: These findings suggest that BD is associated with minimal changes in whole blood gene expression independent of medication use but emphasize the importance of accounting for medication use and cell type heterogeneity in psychiatric transcriptomic studies. The results of this study add to mounting evidence of lithium's cell signaling and immune-related mechanisms.\nAt time of import, last updated (by provider) on: Oct 24 2019\n\nContributors: ; [Catharine E Krebs, Roel A Ophoff, Loes M Loohuis]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          78 samples of individuals from three different diagnostic groups: bipolar, schizophrenia and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               39 samples of individuals from 4 different diagnostic groups: bipolar, schizophrenia, depression and controls. Samples taken from the DLPFC Broadmann area 46/10.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              RNA-seq profiling was conducted on clinically-annotated human post-mortem brain tissues\nLast Updated (by provider): Jun 26 2018\nContributors:  Ryne C Ramaker Kevin M Bowling Sara J Cooper Brittany N Lasseigne Richard M Myers</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               There are currently no biological tests that differentiate patients with bipolar disorder (BPD) from healthy controls. While there is evidence that peripheral gene expression differences between patients and controls can be utilized as biomarkers for psychiatric illness, it is unclear whether current use or residual effects of antipsychotic and mood stabilizer medication drives much of the differential transcription. We therefore tested whether expression changes in first-episode, never-medicated bipolar patients, can contribute to a biological classifier that is less influenced by medication and could potentially form a practicable biomarker assay for BPD. We employed microarray technology to measure global leukocyte gene expression in first-episode (n=3) and currently medicated BPD patients (n=26), and matched healthy controls (n=25). Following an initial feature selection of the microarray data, we developed a cross-validated 10-gene model that was able to correctly predict the diagnostic group of the training sample (26 medicated patients and 12 controls), with 89% sensitivity and 75% specificity (p&lt;0.001). The 10-gene predictor was further explored via testing on an independent test cohort consisting of three pairs of monozygotic twins discordant for BPD, plus the original enrichment sample cohort (the three never-medicated BPD patients and 13 matched control subjects), and a sample of experimental replicates (n=34). 83% of the independent test sample was correctly predicted, with a sensitivity of 67% and specificity of 100% (although this result did not reach statistical significance). Additionally, 88% of sample diagnostic classes were classified correctly for both the enrichment (p=0.015) and the replicate samples (p&lt;0.001).\nLast Updated (by provider): Jun 25 2013\nContributors:  James D Clelland Catherine L Clelland</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37:  Most neuroscientists would agree that psychiatric illness is unlikely to arise from pathological changes that occur uniformly across all cells in a given brain region. Despite this fact, the majority of transcriptomic analyses of the human brain to date are conducted using macro-dissected tissue due to the difficulty of conducting single-cell level analyses on donated post-mortem brains. To address this issue statistically, we compiled a database of several thousand transcripts that were specifically-enriched in one of 10 primary brain cell types identified in published single cell type transcriptomic experiments. Using this database, we predicted the relative cell type composition for 157 human dorsolateral prefrontal cortex samples using Affymetrix microarray data collected by the Pritzker Neuropsychiatric Consortium, as well as for 841 samples spanning 160 brain regions included in an Agilent microarray dataset collected by the Allen Brain Atlas. These predictions were generated by averaging normalized expression levels across the transcripts specific to each primary cell type to create a “cell type index”. Using this method, we determined that the expression of cell type specific transcripts identified by different experiments, methodologies, and species clustered into three main cell type groups: neurons, oligodendrocytes, and astrocytes/support cells. Overall, the principal components of variation in the data were largely explained by the neuron to glia ratio of the samples. When comparing across brain regions, we were able to easily capture canonical cell type signatures – increased endothelial cells and vasculature in the choroid plexus, oligodendrocytes in the corpus callosum, astrocytes in the central glial substance, neurons and immature cells in the dentate gyrus, and oligodendrocytes and interneurons in the globus pallidus. The relative balance of these cell types was influenced by a variety of demographic, pre- and post-mortem variables. Age and prolonged anaerobic conditions around the time of death were associated with decreased neuronal content and increased astrocytic and endothelial content in the tissue, replicating the known higher vulnerability of neurons to aging and adverse conditions, and illustrating the proliferation of vasculature in a hypoxic environment. We also found that the red blood cell content was reduced in individuals who died in a manner that involved systemic blood loss. Finally, statistically accounting for cell type improved both the sensitivity and interpretability of diagnosis effects within the data. We were able to observe a decrease in astrocytic content in subjects with Major Depressive Disorder, mirroring what had been previously observed morphometrically. By including a set of “cell type indices” in a larger model examining the relationship between gene expression and neuropsychiatric illness, we were able to successfully detect almost twice as many genes with previously identified relationships to bipolar disorder and schizophrenia than using traditional analysis methods.\nAt time of import, last updated (by provider) on: \n\nContributors: ; [Jun Z Li, Cortney A Turner, Megan H Hagenauer, Stanley J Watson, David M Walsh, Alan F Schatzberg, Huda Akil, Richard M Myers, William E Bunney, Jack D Barchas]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Schizophrenia (SZ) and bipolar disorder (BD) are severe neuropsychiatric disorders with serious impact on patients, together termed “major psychosis”. Recently, long intergenic non-coding RNAs (lincRNAs) were reported to play important roles in mental diseases. However, little was known about their molecular mechanism in pathogenesis of SZ and BD. Here, we performed RNA sequencing on 82 post-mortem brain tissues from three brain regions (orbitofrontal cortex (BA11), anterior cingulate cortex (BA24) and dorsolateral prefrontal cortex (BA9)) of patients with SZ and BD and control subjects, generating over one billion reads. We characterized lincRNA transcriptome in the three brain regions and identified 20 differentially expressed lincRNAs (DELincRNAs) in BA11 for BD, 34 and 1 in BA24 and BA9 for SZ, respectively. Our results showed that these DELincRNAs exhibited brain region-specific patterns. Applying weighted gene co-expression network analysis, we revealed that DELincRNAs together with other genes can function as modules to perform different functions in different brain regions, such as immune system development in BA24 and oligodendrocyte differentiation in BA9. Additionally, we found that DNA methylation alteration could partly explain the dysregulation of lincRNAs, some of which could function as enhancers in the pathogenesis of major psychosis. Together, we performed systematical characterization of dysfunctional lincRNAs in multiple brain regions of major psychosis, which provided a valuable resource to understand their roles in SZ and BD pathology and helped to discover novel biomarkers.\nLast Updated (by provider): Jun 26 2018\nContributors:  Jing Hu Jinyuan Xu Lin Pang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 47 samples of individuals from 4 different diagnostic groups: bipolar, schizophrenia, depression and controls. Samples taken from the DLPFC Broadmann area 8/9.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      RNA was extracted from peripheral blood mononuclear cells (PBMC) of 24 adult healthy controls, 8 adult patients with bipolar disorder, and 21 adult patients with major depressive disorder to analyze gene expression patterns that identify biomarkers of disease and that may be correlated with fMRI data.\nLast Updated (by provider): Sep 24 2012\nContributors:  T K Teague Jonathan Savitz Wayne C Drevets Julie H Marino Melissa Bebak Bart Frank</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Fibroblasts from patients with Type I bipolar disorder (BPD) and their unaffected siblings were obtained from an Old Order Amish pedigree with a high incidence of BPD and reprogrammed to induced pluripotent stem cells (iPSCs). Established iPSCs were subsequently differentiated into neuroprogenitors (NPs) and then to neurons. Transcriptomic microarray analysis was conducted on RNA samples from iPSCs, NPs and neurons matured in culture for either 2 weeks (termed early neurons, E) or 4 weeks (termed late neurons, L). Global RNA profiling indicated that BPD and control iPSCs differentiated into NPs and neurons at a similar rate, enabling studies of differentially expressed genes in neurons from controls and BPD cases. Significant disease-associated differences in gene expression were observed only in L neurons. Specifically, 328 genes were differentially expressed between BPD and control L neurons including GAD1, glutamate decarboxylase 1 (2.5 fold) and SCN4B, the voltage gated type IV sodium channel beta subunit (-14.6 fold). Quantitative RT-PCR confirmed the up-regulation of GAD1 in BPD compared to control L neurons. Gene Ontology, GeneGo and Ingenuity Pathway Analysis of differentially regulated genes in L neurons suggest that alterations in RNA biosynthesis and metabolism, protein trafficking as well as receptor signaling pathways GSK3? signaling may play an important role in the pathophysiology of BPD.\nLast Updated (by provider): Jun 03 2018\nContributors:  Jiangang Liu Steven M Paul Jeffrey L Dage Janice A Egeland Rachelle J Sells Galvin Kwi H Kim Rosamund C Smith Kalpana M Merchant</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Bipolar disorder (BD) is a highly heritable and heterogeneous mental illness whose manifestations often include impulsive and risk-taking behavior. This particular phenotype suggests that abnormal striatal function could be involved in BD etiology, yet most transcriptomic studies of this disorder have concentrated on cortical brain regions. We report the first transcriptome profiling by RNA-Seq of the human dorsal striatum comparing bipolar and control subjects. Differential expression analysis and functional pathway enrichment analysis were performed to identify changes in gene expression that correlate with BD status. Further co-expression and enrichment analyses were performed to identify sets of correlated genes that show association to BD.\nLast Updated (by provider): May 17 2017\nContributors:  Ronald L Davis Rodrigo Pacifico</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Background: \tSchizophrenia (SCZ) and bipolar disorder (BD) are highly heritable psychiatric disorders. Associated genetic and gene expression changes have been identified, but many have not been replicated and have unknown functions. We identified groups of genes whose expressions varied together, that is co-expression modules, then tested them for association with SCZ. Using weighted gene co-expression network analysis, we show that two modules were differentially expressed in patients versus controls. One, upregulated in cerebral cortex, was enriched with neuron differentiation and neuron development genes, as well as disease genome-wide association study genetic signals; the second, altered in cerebral cortex and cerebellum, was enriched with genes involved in neuron protection functions. The findings were preserved in five expression data sets, including sets from three brain regions, from a different microarray platform, and from BD patients. From those observations, we propose neuron differentiation and development pathways may be involved in etiologies of both SCZ and BD, and neuron protection function participates in pathological process of the diseases.\nLast Updated (by provider):  Jul 26, 2018\nContributors:  Chao Chen Chunyu Liu Lijun Cheng</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Schizophrenia (SZ) and bipolar disorder (BD) are severe psychiatric conditions, with a lifetime prevalence of about 1%. Both disorders have a neurodevelopment component, with onset of symptoms occurring most frequently during late adolescence or early adulthood. Genetic findings indicate the existence of an overlap in genetic susceptibility across the disorders. These gene expression profiles were used to identify the molecular mechanisms that differentiate SZ and BP from healthy controls but also that distinguish both from healthy individuals. They were also used to expand an analysis from an experiment that searched molecular alterations in human induced pluripotent stem cells derived from fibroblasts from control subject and individual with schizophrenia and further differentiated to neuron to identify genes relevant for the development of schizophrenia (GSE62105).\nLast Updated (by provider): Oct 14 2014\nContributors:  Leandro Lima Mariana Maschietto Dirce M Carraro Carlos A Filho Angelica de Baumont Luiz A Barreta Paulo Belmonte-de-Abreu Ana C Tahira Eloisa H Olivieri Joana A Palha Helena Brentani Daniel Mariani Alex Fiorini</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Fibroblasts and lymphoblastoid cells (LCLs) are the most widely used cells in genetic, genomic, and transcriptomic studies in relation to human diseases. Examining the gene expression patterns in these two cell types will provide valuable information regarding the validity of using them to study gene expression related to various human diseases. Fibroblasts and LCLs from four members of the Old Order Amish family 884 were purchased from Coriell cell repositories (Coriell Institute for Medical Research, Camden, NJ). We used microarrays to profile the patterns of gene expression in these eight cell lines. By employing the PennCNV algorithm to the Illumina HumanHap550 SNP genotyping data, we detected 13 Copy Number Variants (CNV) that exist in these four individuals. CNV-expression association analysis revealed that seven of these 13 CNVs were associated with the expression of genes within or near (&lt;2Mb sweep) these CNVs at a nominal regression P value of 0.05.\nLast Updated (by provider): Feb 20 2009\nContributors:  Maja Bucan Shuzhang Yang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            27 samples of individuals from two different diagnostic groups: bipolar, and controls. Samples taken from the DLPFC Brodmann area 6.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 We used laser capture microdissection to isolate both microvascular endothelial cells and neurons from post mortem brain tissue from patients with schizophrenia and bipolar disorder and healthy controls. RNA was isolated from these cell populations, amplified, and analysed using Affymetrix HG133plus2.0 GeneChips. In the first instance, we used the dataset to compare the neuronal and endothelial data, in order to demonstrate that the predicted differences between cell types could be detected using this methodology. \nLast Updated (by provider): Dec 18 2008\nContributors:  Margaret M Ryan Thomas Giger Martin J Lan Matthew T Wayland Mark Kotter Michael L Mimmack Laura W Harris Lan Wang Irene Wuethrich Helen Lockstone Sabine Bahn</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Impairments in certain cognitive processes (e.g., working memory) are typically most pronounced in schizophrenia (SZ), intermediate in bipolar disorder (BP) and least in major depressive disorder (MDD). Given that working memory depends, in part, on neural circuitry that includes pyramidal neurons in layer 3 (L3) and layer 5 (L5) of the dorsolateral prefrontal cortex (DLPFC), we sought to determine if transcriptome alterations in these neurons were shared or distinctive for each diagnosis.\nLast Updated (by provider): Jul 05 2017\nContributors:  Dominique Arion David A Lewis John F Enwright George Tseng Zhiguang Huo John P Corradi</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         98 samples of individuals from three different diagnostic groups: bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Bipolar affective disorder is a severe psychiatric disorder with a strong genetic component but unknown pathophysiology. We used microarray technology (Affymetrix HG-U133A GeneChips) to determine the expression of approximately 22 000 mRNA transcripts in post-mortem brain tissue (orbitofrontal cortex) from patients with bipolar disorder and matched healthy controls. Orbitofrontal cortex tissue from a cohort of 30 subjects was investigated and the final analysis included 10 bipolar and 11 control subjects. Differences between disease and control groups were identified using a rigorous statistical analysis with correction for confounding variables and multiple testing.\nNote: [] samples from this series, which appear in other Expression Experiments in Gemma, were not imported from the GEO source. The following samples were removed: \nLast Updated (by provider): Jul 27 2006\nContributors:  Sabine Bahn Margaret M Ryan Matthew T Wayland Maree J Webster Stephen J Huffaker Helen E Lockstone\nIncludes GDS2191.\n Update date: Aug 28 2006.\n Dataset description GDS2191: Analysis of postmortem orbitofrontal cortex from 10 adults with bipolar disorder. Results provide insight into the pathophysiology of the disease.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    In psychiatric disorders, common and rare genetic variants cause widespread dysfunction of cells and their interactions, especially in the prefrontal cortex, giving rise to psychiatric symptoms. To better understand these processes, we traced the effects of common and rare genetics, and cumulative disease risk scores, to their molecular footprints in human cortical single-cell types. We demonstrated that examining gene expression at single-exon resolution is crucial for understanding the cortical dysregulation associated with diagnosis and genetic risk derived from common variants. We then used disease risk scores to identify a core set of genes that serve as a footprint of common and rare variants in the cortex. Pathways enriched in these genes included dopamine regulation, circadian entrainment, and hormone regulation. Single-nuclei-RNA-sequencing pinpointed these enriched genes to excitatory cortical neurons. This study highlights the importance of studying sub-gene-level genetic architecture to classify psychiatric disorders based on biology rather than symptomatology, to identify novel targets for treatment development.\nAt time of import, last updated (by provider) on: Nov 20 2022\n\nContributors: ; [Franziska Degenhardt, Fabian J Theis, Janine Knauer-Arloth, Elisabeth Scarr, Nikola S Mueller, Nathalie Gerstner, Holger Thiele, Anna C Koller, Brian Dean, Karolina Worf, Marcella Rietschel, Madhara Udawela, Natalie Matosin, Anna S Froehlich]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           A multitude of genes have been associated with bipolar disorder via SNP genotyping studies.  However, many of these associated SNPs are found within intronic or intergenic regions of the human genome.  We were interested in studying transcriptional profiles/splice variation of genes associated with bipolar disorder within the human striatum. Understanding how these associated genes are transcribed in the human brain may help to guide the development of therapeutic agents for the treatment of bipolar disorder and other neuropsychiatric illnesses.\nLast Updated (by provider): Jun 26 2018\nContributors:  Courtney M MacMullen Ronald L Davis Mohammad Fallahi</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Bipolar affective disorder is a severe psychiatric disorder with a strong genetic component but unknown pathophysiology. We used microarray technology (Affymetrix HG-U133A GeneChips) to determine the expression of approximately 22 000 mRNA transcripts in post-mortem brain tissue (dorsolateral prefrontal cortex) from patients with bipolar disorder and matched healthy controls. A cohort of 70 subjects was investigated and the final analysis included 30 bipolar and 31 control subjects. Differences between disease and control groups were identified using a rigorous statistical analysis with correction for confounding variables and multiple testing.\nLast Updated (by provider): Jan 16 2007\nContributors:  Helen E Lockstone Stephen J Huffaker Matthew T Wayland Sabine Bahn Maree J Webster Margaret M Ryan\nIncludes GDS2190.\n Update date: Aug 28 2006.\n Dataset description GDS2190: Analysis of postmortem dorsolateral prefrontal cortex from 30 adults with bipolar disorder. Results provide insight into the pathophysiology of the disease.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            105 samples of individuals from 3 different diagnostic groups: bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            102 samples of individuals from 3 different diagnostic groups: bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Prefrontal cortex of schizophrenic, bipolar, and control subjects. This is the "McLean 66"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         99 samples of individuals from three different diagnostic groups: bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Anorexia nervosa (AN), bulimia nervosa (BN), and obsessive-compulsive disorder (OCD) are complex psychiatric disorders with shared obsessive features, thought to arise from the interaction of multiple genes of small effect with environmental factors.  Potential candidate genes for AN, BN, and OCD have been identified through clinical association and neuroimaging studies; however, recent genome-wide association studies of eating disorders (ED) so far have failed to report significant findings. Additionally, few if any studies have interrogated postmortem brain tissue for evidence of eQTLs associated with candidate genes, which has particular promise as an approach to elucidating molecular mechanisms of association. We therefore selected single nucleotide polymorphisms (SNPs) based on candidate gene studies for AN, BN, and OCD from the literature, and examined the association of these SNPs with gene expression across the lifespan in prefrontal cortex of a non-psychiatric control cohort (N=268).   Several risk-predisposing SNPs were significantly associated with gene expression among control subjects. We then measured gene expression in the prefrontal cortex of cases previously diagnosed with obsessive psychiatric disorders, e.g., eating disorders (ED; N=15), and obsessive-compulsive disorder/obsessive-compulsive personality disorder or tics (OCD/OCPD/Tic; N=16), and non-psychiatric controls (N=102) and identified 6 and 286 genes that were differentially expressed between ED compared to controls and OCD cases compared to controls, respectively (FDR &lt; 5%). However, none of the clinical risk SNPs were among the eQTLs and none were significantly associated with gene expression within the broad obsessive cohort, suggesting larger sample sizes or other brain regions may be required to identify candidate molecular mechanisms of clinical association in postmortem brain datasets.\nLast Updated (by provider): Aug 08 2014\nContributors:  Andrew E Jaffe</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             98 samples of individuals from 3 different diagnostic groups: bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Neurodevelopmental changes and impaired stress resistance have been implicated in the pathogenesis of bipolar disorder (BD), but the underlying regulatory mechanisms are unresolved.  Here we describe a cerebral organoid model of BD that exhibits altered early neural development, elevated neural network activity, and a major shift in the transcriptome. These phenotypic changes were reproduced in cerebral organoids generated from iPS cell lines derived in multiple different laboratories. The BD cerebral organoid transcriptome showed highly significant enrichment for gene targets of the transcriptional repressor REST. This was associated with reduced nuclear REST and REST binding to target gene recognition sites.  Reducing the oxygen concentration in organoid cultures to a physiological range ameliorated the developmental phenotype and restored REST expression.  These effects were mimicked by treatment with lithium.  Reduced nuclear REST and derepression of REST targets genes was also observed in the prefrontal cortex of BD patients. Thus, an impaired cellular stress response in BD cerebral organoids leads to altered neural development and transcriptional dysregulation associated with downregulation of REST. These findings provide a new model and conceptual framework for exploring the molecular basis of BD\nAt time of import, last updated (by provider) on: Nov 06 2023\n\nContributors: ; [King-Hwa Ling, Jenny Tam, Eunjung A Lee, Angeliki Spathopoulou, Liviu Aron, Pei-Ling Yeo, Li-Huei Tsai, Roy H Perlis, Jaejoon Choi, Bruce A Yankner, Derek Drake, Tak Ko, Mariana Garcia-Corral, Katharina Meyer, George Church]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Schizophrenia (SCZ) and bipolar disorder (BPD) are polygenic disorders with many genes contributing to their etiologies. The aim of this investigation was to search for dysregulated molecular and cellular pathways for these disorders as well as psychosis. We conducted a blood-based microarray investigation in two independent samples with SCZ and BPD from San Diego (SCZ=13, BPD=9, control=8) and Taiwan [data not included](SCZ=11, BPD=14, control=16). Diagnostic groups were compared to controls, and subjects with a history of psychosis [PSYCH(+): San Diego (n=6), Taiwan (n=14)] were compared to subjects without such history [PSYCH(-): San Diego (n=11), Taiwan (n=14)]. Analyses of covariance comparing mean expression levels on a gene-by-gene basis were conducted to generate the top 100 significantly dysregulated gene lists for both samples by each diagnostic group. Gene lists were imported into Ingenuity Pathway Analysis (IPA) software. Results showed the ubiquitin proteasome pathway (UPS) was listed in the top ten canonical pathways for BPD and psychosis diagnostic groups across both samples with a considerably low likelihood of a chance occurrence (p = .001). No overlap in dysregulated genes populating these pathways was observed between the two independent samples. Findings provide preliminary evidence of UPS dysregulation in BPD and psychosis as well as support further investigation of the UPS and other molecular and cellular pathways for potential biomarkers for SCZ, BPD, and/or psychosis. The aim of this investigation was to search for dysregulated molecular and cellular pathways for these disorders as well as psychosis.\nLast Updated (by provider): Oct 19 2012\nContributors:  Sharon D Chandler Chad A Bousman Ian P Everall Erick Tatro Stephen J Glatt Ming T Tsuang James Lohr Ginger Lucero Gursharan Chana William Kremen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                We fine-mapped DNA methylation in neuronal nuclei (NeuN+) isolated by flow cytometry from post-mortem frontal cortex of the brain of individuals diagnosed with schizophrenia, bipolar disorder, and controls (n=29, 26, and 28 individuals).\nAt time of import, last updated (by provider) on: May 15 2019\n\nContributors: ; [Shraddha S Pai, Viviane Labrie]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Background: Schizophrenia (SCZ) and bipolar disorder (BD) are highly heritable psychiatric disorders. Associated genetic and gene expression changes have been identified, but many have not been replicated and have unknown functions. We identified groups of genes whose expressions varied together, that is co-expression modules, then tested them for association with SCZ. Using weighted gene co-expression network analysis, we show that two modules were differentially expressed in patients versus controls. One, upregulated in cerebral cortex, was enriched with neuron differentiation and neuron development genes, as well as disease genome-wide association study genetic signals; the second, altered in cerebral cortex and cerebellum, was enriched with genes involved in neuron protection functions. The findings were preserved in five expression data sets, including sets from three brain\nregions, from a different microarray platform, and from BD patients. From those observations, we propose neuron differentiation and development pathways may be involved in etiologies of both SCZ and BD, and neuron protection function participates in pathological process of the diseases.\nLast Updated (by provider): Jul 26, 2018\nContributors:  Chao Chen Chunyu Liu Lijun Cheng</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Accumulating evidence suggests that mitochondrial dysfunction underlies the pathophysiology of bipolar disorder (BD) and schizophrenia (SZ). We performed large-scale DNA microarray analysis of postmortem brains of patients with BD or SZ, and examined expression patterns of mitochondria-related genes. We found a global down-regulation of mitochondrial genes, such as those encoding respiratory chain components, in BD and SZ samples, even after the effect of sample pH was controlled. However, this was likely due to the effects of medication. Medication-free patients with BD showed tendency of up-regulation of subset of mitochondrial genes. Our findings support the mitochondrial dysfunction hypothesis of BD and SZ pathologies. However, it may be the expression changes of a small fraction of mitochondrial genes rather than the global down-regulation of mitochondrial genes. Our findings warrant further study of the molecular mechanisms underlying mitochondrial dysfunction in BD and SZ. \nLast Updated (by provider): Mar 15 2009\nContributors:  Tadafumi Kato Kazuya Iwamoto Miki Bundo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Microarray experiments were performed using FAC-sorted young photoreceptors to analyze their transcriptome in comparison to remaining retinal cells at same developmental stage and retinal progenitors.\nLast Updated (by provider): Apr 02 2018\nContributors:  Marius Ader Kai Postel</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        We determined the transcriptomes of postmitotic cone photoreceptors in the central region of the mouse retina every day between birth (P0) and eye opening (P12). At each postnatal day we isolated GFP-labeled cells from three different Chrnb4-GFP mice (biological triplicates) using fluorescence-activated cell sorting. We then acquired the transcriptomes of the sorted cones using next generation RNA sequencing. Our data set contained 39 transcriptomes.\nLast Updated (by provider): Jul 03 2018\nContributors:  Janine M Daum Michael B Stadler Özkan Keles Botond Roska Hubertus Kohler</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         mRNA cancer cell line profiles\nLast Updated (by provider): Oct 25 2013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  MUC1-C regulates expression of  MICA/B NKG2D LIGAND AND enhance EXOSOME SECRETION IN HUMAN CANCER CELLS\nAt time of import, last updated (by provider) on: Jun 02 2023\n\nContributors: ; [Yoshihiro Morimoto]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Melatonin is a neurohormone that maintains the circadian rhythms of the body. Although we know the pathway of melatonin action in the brain, we lack comprehensive cross-sectional studies on the periphery of depressed patients.\nAt time of import, last updated (by provider) on: \n\nContributors: ; [Monika Dmitrzak-Weglarz, Karolina Bilska, Aleksandra Szczepankiewicz, Pawel Kapelski, Edyta Reszka, Ewa Banach, Ewa Jablonska, Maria Skibinska, Joanna Pawlak, Beata Narozna]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Melatonin is a neurohormone that maintains the circadian rhythms of the body. Although we know the pathway of melatonin action in the brain, we lack comprehensive cross-sectional studies on the periphery of depressed patients.\nAt time of import, last updated (by provider) on: \n\nContributors: ; [Monika Dmitrzak-Weglarz, Karolina Bilska, Aleksandra Szczepankiewicz, Pawel Kapelski, Edyta Reszka, Ewa Banach, Ewa Jablonska, Maria Skibinska, Joanna Pawlak, Beata Narozna]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Bipolar disorder (BPD) is a debilitating heritable psychiatric disorder.  Contemporary models for the manic pole of BPD have primarily utilized either single locus transgenics or treatment with psychostimulants.  Our lab recently characterized a mouse strain, termed Madison (MSN), which naturally displays a manic phenotype, exhibiting elevated locomotor activity, increased sexual behavior, and higher forced swimming relative to control strains.  Lithium chloride and olanzapine treatments attenuate this phenotype.  In this study, we replicated our locomotor activity experiment, showing that MSN mice display generationally-stable mania relative to their outbred ancestral strain, hsd:ICR (ICR).  We then performed a gene expression microarray experiment to compare hippocampus of MSN and ICR mice.  We found dysregulation of multiple transcripts whose human orthologs are associated with BPD and other psychiatric disorders including schizophrenia and ADHD, including: Epor, Smarca4, Cmklr1, Cat, Tac1, Npsr1, Fhit, and P2rx7.  RT-qPCR confirmed dysregulation for all of seven transcripts tested.  Using a network analysis, we found dysregulation of a gene system related to chromatin packaging, a result convergent with recent human findings on BPD.  Using a novel genomic enrichment algorithm, we found enrichment in genome regions homologous to human loci implicated in BPD in replicated linkage studies including homologs of human cytobands 1p36, 3p14, 3q29, 6p21-22, 12q24, 16q24, and 17q25.  Our findings suggest that MSN mice represent a polygenic model for the manic pole of BPD showing much of the genetic systems complexity of the corresponding human disorder.  Further, the high degree of convergence between our findings and the human literature on BPD brings up novel questions about evolution by analogy in mammalian genomes.\nLast Updated (by provider): May 06 2012\nContributors:  Stephen C Gammie Griffin M Gessay Michael C Saul</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Diverse cell types can be reprogrammed into pluripotent stem cells by ectopic expression of Oct4 (Pou5f1), Klf4, Sox3 and Myc. Many of these induced pluripotent stem cells (iPSCs) retain an epigenetic memory of their cellular origins and this in turn may bias their subsequent differentiation. Differentiated neurons are difficult to reprogram and there has not been a systematic side-by-side characterization of reprogramming efficiency or epigenetic memory across different neuronal subtypes. We have recently developed a new method for reprogramming retinal neurons and successfully reprogrammed rod photoreceptors from the murine retina. Here we extended our retinal reprogramming to cone photoreceptors, bipolar neurons, amacrine/horizontal cell interneurons and Müller glia at two different stages of development. We scored the efficiency of reprogramming across all 5 retinal cell types at each developmental stage and we measured retinal differentiation from each iPSC line using a quantitative standardized scoring system called STEM-RET. We discovered that the rod photoreceptors and bipolar neurons had the lowest reprogramming efficiency but iPSCs derived from rods and bipolar cells had the best retinal differentiation. Epigenetic memory was analyzed by characterizing DNA methylation and performing ChIP-seq for 8 histone marks, Brd4, PolII and CTCF. The epigenetic data were integrated with RNA-Seq data from each iPSC line. Retinal cell types with a greater epigenetic barrier to reprogramming (rods and bipolars) are more likely to retain epigenetic memory of their cellular origins. In addition, we identified biomarkers of iPSCs that are predictive of retinal differentiation. This work will have implications for selection of cell populations for cell based therapy and for using reprogramming of purified cell populations to advance our understanding of the role of the epigenome in normal differentiation.\nLast Updated (by provider): Jun 14 2018\nContributors:  Jiakun Zhang Sharon Frase Suresh Thiagarajan Daniel Hiler Michael A Dyer Dianna Johnson Issam Aldiri Xiang Chen Lyra Griffiths Marie-Elizabeth Barabas Andras Sablauer Beisi Xu Lu Wang Marc Valentine Abbas Shirinifard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Schizophrenia is associated with dysfunction of the dorsolateral prefrontal cortex (DLPFC). This dysfunction is manifest as cognitive deficits that appear to arise from disturbances in gamma frequency oscillations. These oscillations are generated in DLPFC layer 3 via reciprocal connections between pyramidal cells and parvalbumin (PV)-containing interneurons. The density of cortical PV neurons is not altered in schizophrenia, but expression levels of several transcripts involved in PV cell function, including PV, are lower in the disease.\nLast Updated (by provider): Jul 03 2018\nContributors:  George Tseng Dominique Arion David A Lewis John F Enwright Zhiguang Huo John P Corradi</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Transcription factor Sp4 controls dendritic patterning during development of cerebellar granule neurons in culture by limiting branch formation and promoting activity-dependent pruning (Ramos et al., 2007). Sp4 is associated with neuropsychiatric disorders such as major depressive disorder, schizophrenia and bipolar disorder.  In order to identify target genes of Sp4, we compared global gene expression in the cerebella of wild type and Sp4 hypomorph mice (Sp4neo-/-; Zhou et al, 2005).  The results identify candidate Sp4 target genes that may contribute to neuronal development and neuropsychiatric disorders.\nLast Updated (by provider): Nov 09 2017\nContributors:  Xinxin Sun Grace Gill</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Genetic analyses for bipolar disorder (BPD) have achieved prominent success in Europeans in recent years, whereas its genetic basis in other populations remains relatively less understood. We herein report that the lead risk locus for BPD in European genome-wide association studies (GWAS), the single nucleotide polymorphism (SNP) rs9834970 near TRANK1 at 3p22 region, is also genome-wide significantly associated with BPD in 5,748 cases and 65,361 controls of East Asian origin. In this study, we performed RAN-seq analysis of cultured rat neurons treated with shRNA knockdown of Trank1.\nAt time of import, last updated (by provider) on: Jul 02 2020\n\nContributors: ; [Ming Li, Huijuan Li, Hong Chang, Xin Cai]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       These are RNA sequencing data from 12 week old orbital frontal cortex from mice who received a shRNA targeting circHomer1 and Homer1b (double knockdown) or control shRNA (scramble)\nAt time of import, last updated (by provider) on: Feb 08 2022\n\nContributors: ; [Mellios Nikolaos]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                CHD8 (chromodomain helicase DNA binding protein 8), which codes for a member of the CHD family of ATP-dependent chromatin-remodeling factors, is the most commonly mutated gene in autism spectrum disorders (ASD) identified in exome-sequencing studies. Loss of function mutations in the gene have also been found in schizophrenia (SZ) and intellectual disabilities, and affects cancer cell proliferation. To better understanding the molecular links between CHD8 functions and ASD, we have applied the CRISPR/Cas9 technology to knockout (KO) one copy of CHD8 in induced pluripotent stem cells (iPSCs) and build cerebral organoids, a model for the developing telencephalon. RNA-seq was carried out on KO organoids (CHD8+/-) and isogenic controls (CHD8+/+). Differentially expressed genes (DEGs) revealed an enrichment of genes involved in neurogenesis, forebrain development, Wnt/?-catenin signaling and axonal guidance. The SZ and bipolar disorder (BD) candidate gene TCF4 was significantly upregulated. Our CHD8 KO DEGs were significantly overlapped with those found in a transcriptome analysis using cerebral organoids derived from a family with idiopathic ASD and another transcriptome study using iPS cell-derived neurons from patients with BD, a condition characterized in a subgroup of patients by dysregulated WNT/?-catenin signaling. Overall, the findings show that distinct ASD, SZ and BD candidate genes converge on common molecular targets - an important consideration for developing novel therapeutics in genetically heterogeneous complex traits.\nLast Updated (by provider): Jun 04 2018\nContributors:  Deyou Zheng Herbert M Lachman Ryan Mokhtari Ping Wang Can Bayrak Erika Pedrosa Michael Kirschenbaum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     To generate an unbiased view of changes to the retinal gene network in Neurog2 retinal mutants, we generated and compared the P2 transcriptomes from control, heterozygote and mutant mice.  A pair of P2 retinas from each biologic replicate were used to produce libraries for high throughput sequencing (n = 5 biologic replicates/genotype).  Reads were aligned with BWA and Bowtie programs to the mm10 genome.  Aligned reads were then analyzed for differentially expressed transcripts using the CuffDiff program in the Galaxy online bioinformatics package (www.usegalaxy.org).\nLast Updated (by provider): Jun 07 2018\nContributors:  Nadean L Brown Angelica M Kowalchuk</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        We used two siRNAs to knock down GNL3 in human neural progenitor cells which were derived from normal human induced pluripotent stem cells (ATCC, ACS-1011). GNL3 knockdown experiments were done in three biological replicates. Total RNA was extracted from GNL3 knockdown and control groups for RNA sequencing (Illumina Hiseq2000, paired-end 100 bp). Genes that affected by both siRNAs were considered differentially expressed genes between GNL3 knockdown and control groups (adjusted P value &lt; 0.05). Using Gene Ontology and KEGG pathway analysis, we found that those differentially exrepssed genes were mainly related to immune response, response to cytokine, cell cycle, and p53 signaling pathway.\nAt time of import, last updated (by provider) on: May 21 2020\n\nContributors: ; [Chuan Jiao, Qingtuan Meng]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          As part of collaboration between the X. William Yang Lab at UCLA and CHDI, a transcriptomic study of normal murine cortex was carried out. Cortex was dissected from 6-month-old wildtype (WT) control mice. Transcriptomic analysis (RNASeq) was performed.\nAt time of import, last updated (by provider) on: Oct 21 2022\n\nContributors: ; [Jeff Aaronson, X W Yang, Jim Rosinski]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           We have identified and characterized an allelic series of spontaneous Rorb mutations in mice We perform RNASeq to identify gene expression changes associated with Rorb mutations in brain and spinal cord from all five mutant strains.  We also perform CNS region-specific RNASeq in the Rorbh5/h5 mutant.\nAt time of import, last updated (by provider) on: Jun 08 2023\n\nContributors: ; [Abigal D Tadenev, Robert W Burgess, George C Murray]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Chaperonin 60 (Cpn60) is a prototypic molecular chaperone essential for cellular function due to its protein folding actions.  However, over the past decade it has been established that Cpn60 can be released by human cells and by certain bacteria to act as an extracellular signalling protein. Mycobacterium tuberculosis produces two Cpn60 proteins: Cpn60.1 and Cpn60.2. We recently generated a M. tuberculosis mutant with an inactivated cpn60.1 gene and demonstrated that granuloma formation was impaired after murine/guinea pig infection.  This finding suggested that Cpn60.1 may interact with the cellular organisation of the host response to M. tuberculosis bacilli.  In this study, we report that recombinant M. tuberculosis Cpn60.1 has both pro- and anti-inflammatory effects on human circulating monocytes. At high concentrations, recombinant Cpn60.1 induces the synthesis of TNF-?, IL6, and IL8, and promotes the phosphorylation of NF-?Bp65, p44/42MAPK and p38 MAPK. At lower concentrations M. tuberculosis Cpn60.1 inhibits lipopolysaccharide-induced release of TNF-?, and monocyte transcriptional activation program. Both effects are abrogated by proteolysis of Cpn60.1 and therefore cannot be attributed to contamination with lipopolysaccharide. Competition with LPS for binding to a common receptor, the release of IL-10 or down-regulation of TLR4 on the cell surface were excluded as explanations for the inhibitory activity of Cpn60.1. We therefore conclude that M. tuberculosis Cpn60.1 is an unusual protein with the ability to induce bipolar effects on human monocytes, which may help explain the pathology of granuloma formation in tuberculosis. We used microarrays to analyse the bipolar effectsof Cpn60.1  on human monocytes.\nLast Updated (by provider): Oct 29 2009\nContributors:  Anthony R Coates Simon J Waddell Brian Henderson Ana Cehovin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 The circadian nature of mood and its dysfunction in affective disorders is well recognized, but the underlying molecular mechanisms are still unclear. We showed that the circadian nuclear receptor REV-ERBa, which is associated with bipolar disorder, impacts midbrain dopamine production and mood-related behavior in mice. Genetic deletion of the Rev-erba gene or pharmacological inhibition of REV-ERBa activity in the ventral midbrain induced mania-like behavior in association with a central hyperdopaminergic state. We used microarrays to identify differentially expressed genes in the ventral midbrains of wild-type (WT) and Rev-erba knock-out (RKO) mice.\nAt time of import, last updated (by provider) on: Mar 04 2019\n\nContributors: ; [Sooyoung Chung, Kyungjin Kim, Gi H Son]\nIncludes GDS5628 (Last updated by provider at import time: Aug 21 2015)\n Dataset description GDS5628: Analysis of ventral midbrain (VMB) from Rev-erb? knock-outs that were entrained under a 12hr light-dark photoperiod for &gt;10 days, kept in darkness for 2 days, and sacrificed on the third day. REV-ERB? is associated with bipolar disorder. Results provide insight into the role of REV-ERB? in VMB.\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      The transition to motherhood involves CNS changes that modify sociability and affective state.  However, these changes also put females at risk for postpartum depression and psychosis, which impairs parenting abilities and adversely affects children.  Thus, changes in expression and interactions in a core subset of genes may be critical for emergence of a healthy maternal phenotype, but inappropriate changes of the same genes could put women at risk for postpartum disorders.  This study evaluated microarray gene expression changes in medial prefrontal cortex (mPFC), a region implicated in both maternal behavior and psychiatric disorders.  Postpartum mice were compared to virgin controls housed with females and isolated for identical durations.  Using the Modular Single-set Enrichment Test (MSET), we found that the genetic landscape of maternal mPFC bears statistical similarity to gene databases associated with schizophrenia (5 of 5 sets) and bipolar disorder (BPD, 3 of 3 sets).  In contrast to previous studies of maternal lateral septum and medial preoptic area, enrichment of autism and depression-linked genes was not significant (2 of 9 sets, 0 of 4 sets).  Among genes linked to multiple disorders were fatty acid binding protein 7 (Fabp7), glutamate metabotropic receptor 3 (Grm3), platelet derived growth factor, beta polypeptide (Pdgfrb), and nuclear receptor subfamily 1, group D, member 1 (Nr1d1).  RT-qPCR confirmed these gene changes as well as FMS-like tyrosine kinase 1 (Flt1) and proenkephalin (Penk).  Systems-level methods revealed involvement of developmental gene networks in establishing the maternal phenotype and indirectly suggested a role for numerous microRNAs and transcription factors in mediating expression changes.  Together, this study suggests that a subset of genes involved in shaping the healthy maternal brain may also be dysregulated in mental health disorders and put females at risk for postpartum psychosis with aspects of schizophrenia and BPD.\nLast Updated (by provider): Feb 21 2018\nContributors:  Terri M Driessen Changjiu Zhao Stephen C Gammie Brian E Eisinger</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      The retina, the accessible part of the central nervous system, has served as a model system to study the relationship between energy utilization and metabolite supply. When the metabolite supply cannot match the energy demand, retinal neurons are at risk of death. As the powerhouse of eukaryotic cells, mitochondria play a pivotal role in generating ATP, produce precursors for macromolecules, maintain the redox homeostasis, and function as waste management centers for various types of metabolic intermediates. Mitochondrial dysfunction has been implicated in the pathologies of a number of degenerative retinal diseases. It is well known that photoreceptors are particularly vulnerable to mutations affecting mitochondrial function due to their high energy demand and susceptibility to oxidative stress. However, it is unclear how defective mitochondria affect other retinal neurons. Nuclear respiratory factor 1 (Nrf1) is the major transcriptional regulator of mitochondrial biogenesis, and loss of Nrf1 leads to defective mitochondria biogenesis and eventually cell death. Here, we investigated how different retinal neurons respond to the loss of Nrf1. We provide in vivo evidence that the disruption of Nrf1-mediated mitochondrial biogenesis results in a slow, progressive degeneration of all retinal cell types examined, although they present different sensitivity to the deletion of Nrf1, which implicates differential energy demand and utilization, as well as tolerance to mitochondria defects in different neuronal cells. Furthermore, transcriptome analysis on rod-specific Nrf1 deletion uncovered a previously unknown role of Nrf1 in maintaining genome stability.\nAt time of import, last updated (by provider) on: Dec 17 2022\n\nContributors: ; [Chai-An Mao, Takae Kiyama]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Mania is a serious neuropsychiatric condition associated with significant morbidity and mortality. Previous studies have suggested that environmental exposures can contribute to mania pathogenesis.  We measured dietary exposures in a cohort of individuals with mania and other psychiatric disorders as well as in control individual without a psychiatric disorder. We found that a history of eating nitrated dry cured meat, but not other meat or fish products, was strongly and independently associated with current mania (adjusted odds ratio 3.49, 95% confidence interval (CI) 2.24-5.45, p&lt;8.97x 10-8). Lower odds of association were found between eating nitrated dry cured meat and other psychiatric disorders. We further found that the feeding of meat preparations with added nitrate to rats resulted in alterations in behavior and changes in intestinal microbiota. Rats fed diets with added nitrate also showed alterations of brain pathways dysregulated in mania. These findings may lead to new methods for preventing mania and for developing novel therapeutic interventions\nLast Updated (by provider): Aug 20 2018\nContributors:  Seva G Khambadkone C C Talbot Jr Robert H Yolken</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Impaired neuronal processes, including dopamine imbalance, are central to the pathogenesis of major psychosis, but the molecular origins are unclear. We report the first multi-omics study of neurons isolated from the prefrontal cortex of individuals with schizophrenia and bipolar disorder, including genome-wide neuronal DNA methylation using Illumina EPIC microarrays, transcriptomes and SNP genotypes (n=55 cases and 27 controls). Epigenetic, transcriptomic, and genetic-epigenetic interactions in disease converged on pathways of neurodevelopment, synaptic activity, and immune functions. Notably, we discovered prominent hypomethylation of an enhancer within the insulin-like growth factor 2 (IGF2) gene in neurons of major psychosis patients. Chromatin conformation analysis revealed that this enhancer targets the nearby tyrosine hydroxylase (TH) gene, which is responsible for dopamine synthesis. IGF2 enhancer hypomethylation was associated with increased TH protein levels in the human brain. The Igf2 enhancer was deleted in mice to explore the transcriptomic and proteomic consequences of this genomic locus in the frontal cortex and striatum. In mice, Igf2 enhancer deletion disrupted levels of TH protein and striatal dopamine, as well as induced transcriptional and proteomic abnormalities affecting development and synaptic function. Epigenetic control of the IGF2 enhancer may regulate dopamine levels and contribute to psychosis risk.\nAt time of import, last updated (by provider) on: May 15 2019\n\nContributors: ; [Shraddha S Pai, Viviane Labrie]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            The goal of this project is to study transcriptome change by knocking down ZNF804A, a schizophrenia and bipolar disorder candidate gene, in early neurons derived from iPSCs.\nLast Updated (by provider): Sep 16 2016\nContributors:  Deyou Zheng Herbert M Lachman</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  The molecular etiology of invididual differences in complex behavior traits and susceptibility to psychiatric illness remains incomplete. Using an unbiased genetic approach in a mouse model, Quantitative Trait Loci (QTL) influencing anxiety-like behaviors and beta-carboline-induced seizure vulnerability have been mapped to the distal portion of mouse chromosome 10 and an interval specific congenic strain (ISCS; A.B6chr10; 66 cM to telomere) was developed. This A.B6chr10 strain facilitated defining the behavioral influences of this region as well as gene expression profiling to identify candidate gene(s) underlying this QTL. By microarray studies, an unsuspected E3 Ubiquitin Ligase, Ring Finger 41 (Rnf41 / Neuregulin Receptor Degrading Protein1; Nrdp1) was differentially expressed in the region of interest, comparing the hippocampi of A/J vs A.B6chr10 mice as well as A/J vs B6 mice. By RT-PCR, Rnf41 expression levels were significantly increased 1.5 and 1.3-fold in the hippocampi of C57BL6/J and A.B6chr10 mice compared to A/J mice, respectively. In addition, protein levels of Rnf41 were increased in hippocampi of B6 mice compared to A/J mice across postnatal development with a 5.5-fold difference at P56. Among LxS recombinant inbred mice (N=33), Rnf41 hippocampal mRNA expression levels were significantly correlated with open field behavior (r= .454, p=.0073). Re-analyzing a microarray database of human post-mortem prefrontal cortex (Brodmann’s Area 46/10), RNF41 mRNA expression levels were reduced significantly in patients with major depression and bipolar disorder compared to unaffected controls. Overall, Rnf41 is a pleiotropic candidate gene for anxiety-like behaviors, depression, and vulnerability to seizures. RNF41 and its binding partners provide novel etiological pathways for influencing behavior, highlighting a potential role for the ubiquitin proteasome system in psychiatric illness.\nLast Updated (by provider): Jan 15 2010\nContributors:  H K Gershenfeld Sanghyeon Kim K Choi R Reister A F Baykiz S Zhang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  The molecular etiology of invididual differences in complex behavior traits and susceptibility to psychiatric illness remains incomplete. Using an unbiased genetic approach in a mouse model, Quantitative Trait Loci (QTL) influencing anxiety-like behaviors and beta-carboline-induced seizure vulnerability have been mapped to the distal portion of mouse chromosome 10 and an interval specific congenic strain (ISCS; A.B6chr10; 66 cM to telomere) was developed. This A.B6chr10 strain facilitated defining the behavioral influences of this region as well as gene expression profiling to identify candidate gene(s) underlying this QTL. By microarray studies, an unsuspected E3 Ubiquitin Ligase, Ring Finger 41 (Rnf41 / Neuregulin Receptor Degrading Protein1; Nrdp1) was differentially expressed in the region of interest, comparing the hippocampi of A/J vs A.B6chr10 mice as well as A/J vs B6 mice. By RT-PCR, Rnf41 expression levels were significantly increased 1.5 and 1.3-fold in the hippocampi of C57BL6/J and A.B6chr10 mice compared to A/J mice, respectively. In addition, protein levels of Rnf41 were increased in hippocampi of B6 mice compared to A/J mice across postnatal development with a 5.5-fold difference at P56. Among LxS recombinant inbred mice (N=33), Rnf41 hippocampal mRNA expression levels were significantly correlated with open field behavior (r= .454, p=.0073). Re-analyzing a microarray database of human post-mortem prefrontal cortex (Brodmann’s Area 46/10), RNF41 mRNA expression levels were reduced significantly in patients with major depression and bipolar disorder compared to unaffected controls. Overall, Rnf41 is a pleiotropic candidate gene for anxiety-like behaviors, depression, and vulnerability to seizures. RNF41 and its binding partners provide novel etiological pathways for influencing behavior, highlighting a potential role for the ubiquitin proteasome system in psychiatric illness.\nLast Updated (by provider): Jan 15 2010\nContributors:  H K Gershenfeld Sanghyeon Kim K Choi R Reister A F Baykiz S Zhang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  The molecular etiology of invididual differences in complex behavior traits and susceptibility to psychiatric illness remains incomplete. Using an unbiased genetic approach in a mouse model, Quantitative Trait Loci (QTL) influencing anxiety-like behaviors and beta-carboline-induced seizure vulnerability have been mapped to the distal portion of mouse chromosome 10 and an interval specific congenic strain (ISCS; A.B6chr10; 66 cM to telomere) was developed. This A.B6chr10 strain facilitated defining the behavioral influences of this region as well as gene expression profiling to identify candidate gene(s) underlying this QTL. By microarray studies, an unsuspected E3 Ubiquitin Ligase, Ring Finger 41 (Rnf41 / Neuregulin Receptor Degrading Protein1; Nrdp1) was differentially expressed in the region of interest, comparing the hippocampi of A/J vs A.B6chr10 mice as well as A/J vs B6 mice. By RT-PCR, Rnf41 expression levels were significantly increased 1.5 and 1.3-fold in the hippocampi of C57BL6/J and A.B6chr10 mice compared to A/J mice, respectively. In addition, protein levels of Rnf41 were increased in hippocampi of B6 mice compared to A/J mice across postnatal development with a 5.5-fold difference at P56. Among LxS recombinant inbred mice (N=33), Rnf41 hippocampal mRNA expression levels were significantly correlated with open field behavior (r= .454, p=.0073). Re-analyzing a microarray database of human post-mortem prefrontal cortex (Brodmann’s Area 46/10), RNF41 mRNA expression levels were reduced significantly in patients with major depression and bipolar disorder compared to unaffected controls. Overall, Rnf41 is a pleiotropic candidate gene for anxiety-like behaviors, depression, and vulnerability to seizures. RNF41 and its binding partners provide novel etiological pathways for influencing behavior, highlighting a potential role for the ubiquitin proteasome system in psychiatric illness.\nLast Updated (by provider): Jan 15 2010\nContributors:  H K Gershenfeld Sanghyeon Kim K Choi R Reister A F Baykiz S Zhang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Loss of the Atrx chromatin remodeling protein causes dysfunction and death of post-mitotic retinal interneurons in mice. Embryonic conditional deletion of Atrx from multipotent retinal progenitor cells results in the selective loss of the retinal inhibitory interneurons, namely amacrine and horizontal cells. The cell death occurs postnatally after the development of these cell types, peaking at postntal day 17 in the mouse retina.  Identification of molecular factors and pathways that mediate the health and survival of these neurons may suggest novel therapeutic strategies for neuroprotection in ATR-X syndrome and other neurodegenerative diseases. We performed gene expression profiling of wildtype and Atrx conditional knockout mouse retina tissues to identify putative targets of Atrx and molecular pathways that underlie the neurodegenerative phenotype.\nLast Updated (by provider): Feb 21 2018\nContributors:  Pamela S Lagali David J Picketts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Diseases and damage to the retina lead to losses in retinal neurons and eventual visual impairment. Although the mammalian retina has no inherent regenerative capabilities, fish have robust regeneration from Müller glia (MG). Recently, we have shown that driving expression of Ascl1 in adult mouse MG stimulates neurogenesis similar to fish regeneration. The regeneration observed in the mouse is limited in the variety of neurons that can be derived from MG; Ascl1-expressing MG primarily generate bipolar cells. To better understand the limits of MG-based regeneration in mouse retinas, we used ATAC- and RNA-seq to compare newborn progenitors with MG. Our analysis demonstrated striking similarities between MG and progenitors, with losses in regulatory motifs for neurogenesis genes. Young MG were found to have intermediate expression profiles and accessible DNA, which is mirrored in the ability of Ascl1 to direct bipolar neurogenesis in young MG. When comparing what makes bipolar and photoreceptor cells distinct from glial cells, we find that bipolar-specific accessible regions are more frequently linked to bHLH motifs and Ascl1 binding, indicating that Ascl1 preferentially binds to bipolar regions. Overall, our analysis indicates a loss of neurogenic gene expression and motif accessibility during glial maturation that may prevent efficient reprogramming.\nAt time of import, last updated (by provider) on: Aug 19 2020\n\nContributors: ; [Wohl Stefanie G, Wilken Matthew S, Thomas Reh, Chipman Laura, VandenBosch Leah S, Kox Kristen]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Lithium is a first-line treatment for bipolar disorder, where it acts as a mood-stabilizing agent. Although its precise mechanism remains unclear, neuroimaging studies have shown that lithium accumulates in the hippocampus and that chronic use amongst bipolar disorder patients is associated with larger hippocampal volumes. Here, we tested the chronic effects of low (0.75 mM) and high (2.25 mM) doses of lithium on human hippocampal progenitor cells and used immunocytochemistry to investigate the effects of lithium on cell parameters implicated in neurogenesis. Corresponding RNA-sequencing and gene-set enrichment analyses were used to evaluate whether genes affected by lithium in our model overlap with those regulating the volume of specific layers of the dentate gyrus. We observed that high-dose lithium treatment in human hippocampal progenitors increased the generation of neuroblasts (P ≤ 0.01), neurons (P ≤ 0.01), and glia (P ≤ 0.001), alongside the expression of genes which regulate the volume of the molecular layer of the dentate gyrus. This study provides empirical support that adult hippocampal neurogenesis and gliogenesis are mechanisms that could contribute to the effects of lithium on human hippocampal volume.\nAt time of import, last updated (by provider) on: Oct 01 2021\n\nContributors: ; [Rodrigo R R Duarte, Alish B Palmos, Douglas F Nixon, Timothy R Powell, Demelza M Smeeth, Erin C Hedges, Sandrine Thuret]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Neuronal plasticity of the inner retina has been observed in response to photoreceptor degeneration. Typically, this phenomenon has been considered maladaptive and may preclude vision restoration in the blind. However, several recent studies utilizing triggered photoreceptor ablation have shown adaptive responses in bipolar cell dendrites expected to support normal vision. Whether such homeostatic plasticity occurs during progressive photoreceptor degenerative disease to help maintain normal visual behavior is unknown. We addressed these issues in an established mouse model of Retinitis Pigmentosa caused by the P23H mutation in rhodopsin. We show robust modulation of the retinal transcriptomic network reminiscent of the neurodevelopmental state as well as potentiation of rod – rod bipolar cell signaling following rod photoreceptor degeneration. Additionally, we found highly sensitive night vision in P23H mice even when more than half of the rod photoreceptors were lost. The results implicate retinal adaptation leading to persistent visual function during photoreceptor degenerative disease.\nAt time of import, last updated (by provider) on: Oct 19 2020\n\nContributors: ; [Frans Vinberg, Henri Leinonen]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Neuronal plasticity of the inner retina has been observed in response to photoreceptor degeneration. Typically, this phenomenon has been considered maladaptive and may preclude vision restoration in the blind. However, several recent studies utilizing triggered photoreceptor ablation have shown adaptive responses in bipolar cell dendrites expected to support normal vision. Whether such homeostatic plasticity occurs during progressive photoreceptor degenerative disease to help maintain normal visual behavior is unknown. We addressed these issues in an established mouse model of Retinitis Pigmentosa caused by the P23H mutation in rhodopsin. We show robust modulation of the retinal transcriptomic network reminiscent of the neurodevelopmental state as well as potentiation of rod – rod bipolar cell signaling following rod photoreceptor degeneration. Additionally, we found highly sensitive night vision in P23H mice even when more than half of the rod photoreceptors were lost. The results implicate retinal adaptation leading to persistent visual function during photoreceptor degenerative disease.\nAt time of import, last updated (by provider) on: Jun 16 2020\n\nContributors: ; [Frans Vinberg, Henri Leinonen]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;         &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated    experiment.batchEffectText</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      &lt;int&gt;                 &lt;POSc&gt;                        &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                     33    2023-12-17 01:36:32                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                     12    2023-12-20 19:03:56                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                     22    2023-12-18 01:58:21          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                     35    2023-12-18 12:17:06                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                     26    2023-12-19 05:21:57                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                     22    2023-12-22 09:47:02                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                     78    2023-12-19 02:38:44                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                     12    2023-12-16 05:34:15                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                      4    2023-12-17 12:01:40          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                     23    2023-09-08 07:38:04                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                     93    2022-08-30 23:36:54                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                    215    2023-12-17 23:27:14                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                     24    2023-12-17 14:05:13                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                     22    2023-12-17 10:03:11                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                    120    2023-12-19 20:24:26                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                     20    2023-12-17 07:25:15          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                     24    2023-12-18 11:53:22                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                     34    2023-12-18 00:19:38       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                      4    2024-01-23 19:56:05          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                      6    2023-12-20 19:14:46          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:                      6    2023-12-20 21:45:32          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:                     23    2023-12-20 20:07:23                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23:                     50    2023-09-07 23:11:52                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24:                     32    2023-12-20 11:14:58                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25:                     18    2024-02-22 08:34:38       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26:                     44    2023-09-07 23:17:50                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27:                     46    2023-09-07 23:15:42                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28:                     50    2023-12-16 09:50:20                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29:                    205    2023-12-20 12:59:34                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30:                     12    2023-12-16 10:59:29       BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31:                    128    2023-12-21 10:56:11                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32:                    480    2024-03-22 23:43:22 UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33:                     78    2023-12-11 18:39:23                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34:                     39    2023-06-09 16:41:44                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35:                    281    2023-12-21 12:53:40                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36:                     88    2023-12-19 22:14:40                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37:                    235    2023-12-21 10:57:20       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38:                     82    2023-12-21 12:36:35                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39:                     47    2023-09-07 23:15:06                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40:                     53    2023-12-19 12:29:00                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41:                     28    2023-12-21 00:16:06          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42:                     36    2023-12-21 04:26:14       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43:                    168    2023-12-19 06:35:04                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44:                     88    2023-12-20 15:25:17                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45:                      8    2023-12-16 05:30:07          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46:                     27    2023-09-07 23:19:35                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47:                     32    2023-12-16 10:02:53                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48:                    286    2023-12-22 08:39:18                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49:                     98    2023-12-06 18:45:01                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50:                     21    2023-12-20 08:00:37       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51:                    169    2023-12-18 01:28:03                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52:                     16    2023-12-21 12:58:54                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53:                     61    2023-12-20 07:59:57       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54:                    105    2023-12-06 22:52:55                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55:                    102    2023-11-30 00:38:00                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56:                     66    2023-09-21 21:43:05                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57:                     99    2023-12-07 20:23:13                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58:                    133    2023-12-20 13:19:18                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59:                     98    2023-06-08 22:47:42                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60:                     16    2024-01-24 09:18:15       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61:                     30    2023-12-17 14:10:06       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62:                     34    2023-12-16 10:28:32                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63:                    144    2023-12-19 06:34:12       BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64:                    102    2023-12-16 09:48:59       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65:                      9    2023-12-18 20:37:33       BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66:                      6    2023-12-17 09:01:07                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67:                      6    2023-12-17 09:06:10                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68:                     39    2023-12-22 09:37:54                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69:                    206    2024-02-21 01:47:58          BATCH_EFFECT_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70:                     18    2024-04-02 22:38:04     SINGLETON_BATCHES_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71:                      8    2023-12-17 07:32:13          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72:                      6    2023-12-17 07:31:47          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73:                     12    2023-12-18 20:54:13          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74:                     28    2023-12-16 01:27:15                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75:                    141    2023-12-21 11:23:50       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76:                      6    2023-12-20 06:58:42                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77:                      9    2023-12-16 23:32:03                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78:                      8    2023-12-17 03:18:18                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79:                      6    2023-12-21 07:36:15          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80:                     15    2023-12-16 09:19:48                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81:                      9    2023-12-17 12:34:58                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82:                    104    2024-03-07 19:21:43 UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83:                     66    2023-12-18 10:07:58          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84:                     21    2023-12-17 16:26:53                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85:                      4    2023-12-20 09:14:41          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86:                     10    2023-12-20 08:34:54          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87:                     12    2023-12-16 21:54:38                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88:                     20    2023-12-16 02:52:32          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89:                     27    2023-12-16 11:24:58                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90:                      8    2023-12-20 08:20:04       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91:                      9    2023-12-21 08:09:04       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92:                      9    2023-12-21 08:08:42       NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93:                     18    2023-12-21 08:09:27                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94:                      6    2023-12-21 03:24:53          SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95:                      4    2023-12-17 10:28:51                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96:                     11    2023-12-18 10:20:10                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97:                      7    2023-12-17 00:58:06                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98:                     14    2023-12-17 12:50:03                 NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated    experiment.batchEffectText</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.batchCorrected experiment.batchConfound experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        &lt;lgcl&gt;                    &lt;int&gt;                  &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98:                        NA                       NA                     NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.batchCorrected experiment.batchConfound experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.rawData geeq.qScore geeq.sScore taxon.name  taxon.scientific</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;int&gt;       &lt;num&gt;       &lt;num&gt;     &lt;char&gt;            &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77:                 NA          NA          NA        rat Rattus norvegicus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88:                 NA          NA          NA        rat Rattus norvegicus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96:                 NA          NA          NA      human      Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98:                 NA          NA          NA      mouse      Mus musculus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.rawData geeq.qScore geeq.sScore taxon.name  taxon.scientific</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.ID taxon.NCBI taxon.database.name taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;int&gt;      &lt;int&gt;              &lt;char&gt;             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 60:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 61:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 62:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 63:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 64:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 65:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 66:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 67:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 68:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 69:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 70:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 71:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 72:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 73:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 74:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 75:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 76:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 77:        3      10116                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 78:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 79:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 80:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 81:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 82:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 83:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 84:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 85:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 86:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 87:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 88:        3      10116                 rn6                86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 89:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 90:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 91:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 92:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 94:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 96:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 97:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 98:        2      10090                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.ID taxon.NCBI taxon.database.name taxon.database.ID</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.8.</p>
</div>

      </footer></div>

  


  

  </body></html>

