<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Search everything in Gemma — search_gemma • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Search everything in Gemma — search_gemma"><meta property="og:description" content="Search everything in Gemma"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Search everything in Gemma</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>search_gemma.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Search everything in Gemma</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">search_gemma</span><span class="op">(</span></span>
<span>  <span class="va">query</span>,</span>
<span>  taxon <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  platform <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  resultType <span class="op">=</span> <span class="st">"experiment"</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>The search query. Either plain text ('traumatic'), or an ontology
term URI ('http://purl.obolibrary.org/obo/UBERON_0002048'). Datasets that
contain the given string in their short of full name will also be matched.
Can be multiple identifiers separated by commas.</p></dd>


<dt>taxon</dt>
<dd><p>A numerical taxon identifier or an ncbi taxon identifier or a taxon identifier that matches either its scientific or common name</p></dd>


<dt>platform</dt>
<dd><p>A platform numerical identifier or a platform short name</p></dd>


<dt>limit</dt>
<dd><p>Optional, defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt>resultType</dt>
<dd><p>The kind of results that should be included in the output. Can be experiment, gene, platform or a long object type name, documented in the API documentation.</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>If <code>raw = FALSE</code> and resultType is experiment, gene or platform,
a data.table containing the search results. If it is any other type, a list
of results. A list with additional details about the search if <code>raw = TRUE</code></p>


    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">search_gemma</span><span class="op">(</span><span class="st">"bipolar"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ShortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             GSE67645</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             GSE35077</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:             GSE33085</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             GSE29318</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:             GSE98838</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:            GSE126942</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:            GSE127771</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:             GSE10843</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:       stanley_dobrin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:             GSE66196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:         stanley_chen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:             GSE23848</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:     stanley_feinberg</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:         stanley_kato</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:            GSE210064</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:       stanley_sklarB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:              byne-cc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:   McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:             GSE46449</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:        stanley_young</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                       experiment.Name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: Transcriptome dynamics of developing photoreceptors in 3-D retina cultures recapitulates temporal sequence of human cone and rod differentiation revealing cell surface markers and gene networks</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                            A gene expression database for retinal neuron subtypes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                Transcriptome analysis of adult retina cell types.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                             Expression profile of FAC-sorted murine retinal cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                        Combining RNA-Seq and Somatic CRISPR Mutagenesis to Study Mouse Neural Development in vivo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                              The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                         The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells [P7]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                    mRNA Cancer Cell Line Profiles</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                           Stanley array collection DLPFC - Dobrin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                Bipolar disorder and lithium-induced gene expression in two peripheral cell models</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                        Stanley consortium collection DLPFC - Chen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                 Peripheral blood gene-expression in depressed subjects with bipolar disorder vs healthy controls.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                               Stanley consortium collection Cerebellum - Feinberg</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                             Stanley array collection DLPFC - Kato</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                               Gene expression alterations in the postmortem hippocampus from older patients with bipolar disorder – a hypothesis generating study</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                 Stanley consortium collection Cerebellum - SklarB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                              Corpus Callosum data from Stanley collection samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                             Expression data from Patients with Bipolar (BP) Disorder and Matched Control Subjects</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                            Stanley array collection DLPFC - Young</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:         11634</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          7608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:          8887</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:         12310</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:         13214</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:         19454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:         19455</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:          7196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:           843</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:         24428</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:           842</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:          1958</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:           831</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:           833</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:         28041</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:           835</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:          4354</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:           670</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:          6583</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:           837</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    experiment.Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           To define molecular mechanisms underlying rod and cone differentiation, we generated H9 human embryonic stem cell line carrying a GFP reporter that is controlled by the promoter of cone-rod homeobox (CRX) gene, the first known marker of post-mitotic photoreceptor precursors. CRXp-GFP reporter in H9 line replicates endogenous CRX expression when induced to form self-organizing 3-D retina-like tissue. We define temporal transcriptome dynamics of developing photoreceptors during the establishment of cone and rod cell fate. Our studies provide an essential framework for delineating molecules and cellular pathways that guide human photoreceptor development and should assist in chemical screening and cell-based therapies of retinal degeneration.\nLast Updated (by provider): Nov 14 2017\nContributors:  Kohei Homma Rossukon Kaewkhaw Anand Swaroop Koray D Kaya Jizhong Zou Mahendra Rao Matthew Brooks Vijender Chaitankar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   The goal of this experiment was to define gene expression patterns of thirteen mouse retinal neuron subsets, labeled by expression of fluorescent proteins in transgenic mice.\nLast Updated (by provider): Sep 23 2013\nContributors:  Jeremy N Kay Joshua R Sanes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: Brain circuits are assembled from a large variety of morphologically and functionally diverse cell types. It is not known how the intermingled cell types of individual brain regions differ in their expressed genomes. Here we describe an atlas of cell type transcriptomes of the adult retina. We found that each adult cell type expresses a specific set of genes, including a unique set of transcription factors, forming a “barcode” for cell identity. Cell type transcriptomes carry enough information to categorize cells into corresponding morphological classes and types. Surprisingly, several barcode genes are eye disease-associated genes that we demonstrate to be specifically expressed not only in photoreceptors but also in particular retinal circuit elements such as inhibitory neurons as well as in retinal microglia. Our data suggest that distinct cell types of individual brain regions are characterized by marked differences in their expressed genomes. We assembled a library of 22 transgenic mouse lines in which each line had a group of retinal cells marked with fluorescent proteins. We built up the library with the purpose of having some mouse lines in which single retinal cell types and others in which combinations of types from a single class are labeled. The library had mouse lines with labeled cells representing each of the six retinal cell classes. Retinal cells were characterized by physiological recording and immunohistochemical staining. We isolated 200 fluorescent protein-labeled retinal cells (“cell groups”) from at least three different mice of each mouse line by fluorescence-activated cell sorting. The transcripts of each cell group of these biological triplicates were independently amplified in batches. Each batch contained an internal control cell group from the Arc-line.\nLast Updated (by provider): Jun 10 2014\nContributors:  Botond Roska Erik Cabuy Sandra Siegert</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Microarray experiments were performed using FAC-sorted young photoreceptors to analyze their transcriptome in comparison to remaining retinal cells at same developmental stage and retinal progenitors.\nLast Updated (by provider): Apr 02 2018\nContributors:  Marius Ader Kai Postel</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  We used RNA sequencing to identify candidate regulators of interactions between photoreceptor axons and bipolar cell (BCs) dendrites in developing mouse retina. We chose three time points: P7, just after the OPL forms and synaptogenesis with BCs begins; P13, as synaptogenesis nears completion and sublamination begins; and P30, when the OPL is mature. We purified cone and rod photoreceptors separately by fluorescence activated cell sorting (FACS) using transgene markers: Rhoicre;Ai9 for rods and HRGPcre;Ai9 for cones. We purified ON BCs, which include ON cone bipolars plus rod bipolars using Grm6:GFP. As appropriate transgenic lines to separate RBCs from CBCs were not available, we performed RNAseq on cells from Grm6:GFP mice that were fixed and immunostained prior to FACS, allowing us to purify RBCs (GFP+PKC+) and CBCs (GFP+PKC-) from the same retinas. As PKC is not highly expressed at P7, profiling of developing rod bipolars separate from developing ON cone bipolars was restricted to P13.\nLast Updated (by provider): Jun 25 2018\nContributors:  Elizabeth Z Sanchez Lawrence S Zipursky Yerbol Z Kurmangaliyev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               mRNA cancer cell line profiles\nLast Updated (by provider): Oct 25 2013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                78 samples of individuals from three different diagnostic groups: bipolar, schizophrenia and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Bipolar disorder is a severe and heritable psychiatric disorder and affects up to 1% of the population worldwide. Lithium is recommended as first-line treatment for the maintenance treatment of bipolar-affective disorder in current guidelines, its molecular modes of action are however poorly understood. Cell models derived from bipolar patients could prove useful to gain more insight in the molecular mechanisms of bipolar disorder and the common pharmacological treatments. As primary neuronal cell lines cannot be easily derived from patients, peripheral cell models should be evaluated in their usefulness to study pathomechanisms and the mode of action of medication as well as in regard to develop biomarkers for diagnosis and treatment response.\nAt time of import, last updated (by provider) on: Mar 25 2019\n\nContributors: ; [Sarah Kittel-Schneider, Max Hilscher, Andreas Reif, Claus J Scholz]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  27 samples of individuals from two different diagnostic groups: bipolar, and controls. Samples taken from the DLPFC Brodmann area 6.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Analysis of gene-expression changes in depressed subjects with bipolar disorder compared to healthy controls. Results provide information on pathways that may be involved in the pathogenesis of bipolar depression.\nLast Updated (by provider): Aug 27 2010\nContributors:  Robert D Beech</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     50 samples of individuals from 4 different diagnostic groups: bipolar, schizophrenia, depression and controls. Samples taken from the Cerebellum.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  102 samples of individuals from 3 different diagnostic groups: bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Gene expression of samples from the postmortem hippocampus of older bipolar disorder subjects and controls. Gene expression data was generated using the SurePrint G3 Human Gene Expression v3 microarray. Rank feature selection was performed to identify a subset of features that can optimally differentiate BD and controls.\nAt time of import, last updated (by provider) on: Feb 19 2023\n\nContributors: ; [Carlos A Pasqualucci, Claudia K Suemoto, Ricardo Nitrini, Fernanda B Bertonha, Paula V Nunes, Katia C De Oliveira, Carlos M Filho, Helena K Kim, Helena Brentani, Lea T Grinberg, Beny Lafer, André Barbosa, Camila Nascimento, Renata P Leite, Wilson Jacob-Filho]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     46 samples of individuals from 4 different diagnostic groups: bipolar, schizophrenia, depression and controls. Samples taken from the Cerebellum.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Hippocampus of schizophrenic, bipolar, and control subjects. Analyzed from CEL files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                     There are currently no biological tests that differentiate patients with bipolar disorder (BPD) from healthy controls. While there is evidence that peripheral gene expression differences between patients and controls can be utilized as biomarkers for psychiatric illness, it is unclear whether current use or residual effects of antipsychotic and mood stabilizer medication drives much of the differential transcription. We therefore tested whether expression changes in first-episode, never-medicated bipolar patients, can contribute to a biological classifier that is less influenced by medication and could potentially form a practicable biomarker assay for BPD. We employed microarray technology to measure global leukocyte gene expression in first-episode (n=3) and currently medicated BPD patients (n=26), and matched healthy controls (n=25). Following an initial feature selection of the microarray data, we developed a cross-validated 10-gene model that was able to correctly predict the diagnostic group of the training sample (26 medicated patients and 12 controls), with 89% sensitivity and 75% specificity (p&lt;0.001). The 10-gene predictor was further explored via testing on an independent test cohort consisting of three pairs of monozygotic twins discordant for BPD, plus the original enrichment sample cohort (the three never-medicated BPD patients and 13 matched control subjects), and a sample of experimental replicates (n=34). 83% of the independent test sample was correctly predicted, with a sensitivity of 67% and specificity of 100% (although this result did not reach statistical significance). Additionally, 88% of sample diagnostic classes were classified correctly for both the enrichment (p=0.015) and the replicate samples (p&lt;0.001).\nLast Updated (by provider): Jun 25 2013\nContributors:  James D Clelland Catherine L Clelland</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  105 samples of individuals from 3 different diagnostic groups: bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.Troubled experiment.Accession experiment.Database experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.SampleCount experiment.LastUpdated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                     23    2022-08-22 07:31:30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                     26    2022-09-09 07:33:10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                     78    2022-09-01 07:31:34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                      9    2022-08-17 07:30:12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                     22    2022-09-06 07:32:36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                      6    2022-08-13 17:05:54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                      6    2022-08-10 01:09:22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                    206    2020-03-28 19:12:15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                     78    2023-09-07 23:20:22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                     12    2022-05-19 22:58:39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                     27    2023-09-07 23:19:35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                     35    2023-04-24 13:58:56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                     50    2023-09-07 23:11:52</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                    102    2023-09-07 23:14:08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                     22    2023-06-30 07:36:09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                     46    2023-09-07 23:15:42</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                     93    2022-08-30 23:36:54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                     23    2023-09-07 22:56:32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                     88    2022-08-20 07:32:52</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                    105    2023-06-09 07:33:11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:  This data set may have a batch artifact (PC 3), p=0.0022570</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3: This data set may have a batch artifact (PC 1), p=8.2617e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                      BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                UNINFORMATIVE_HEADERS_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: This data set may have a batch artifact (PC 1), p=4.7909e-27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                      NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                         SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:  This data set may have a batch artifact (PC 3), p=0.0065166</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                         &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: This data set may have a batch artifact (PC 5), p=2.0241e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.Name taxon.Scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.Database.Name taxon.Database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                hg38                87</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

