<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve datasets associated to an annotation tags search — search_datasets • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve datasets associated to an annotation tags search — search_datasets"><meta property="og:description" content="This function is deprecated in favor of get_datasets"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve datasets associated to an annotation tags search</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/deprecation.R" class="external-link"><code>R/deprecation.R</code></a></small>
    <div class="hidden name"><code>search_datasets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function is deprecated in favor of <code><a href="get_datasets.html">get_datasets</a></code></p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">search_datasets</span><span class="op">(</span></span>
<span>  <span class="va">query</span>,</span>
<span>  taxon <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  offset <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20L</span>,</span>
<span>  sort <span class="op">=</span> <span class="st">"+id"</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.attributes"</span>, <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>The search query. Either plain text ('traumatic'), or an ontology term URI ('http://purl.obolibrary.org/obo/UBERON_0002048'). Datasets that contain the given string in their short
or full name will also be matched. Can be multiple identifiers separated by commas.</p></dd>


<dt>taxon</dt>
<dd><p>Can either be Taxon ID, Taxon NCBI ID, or one of its string identifiers: scientific name, common name.
It is recommended to use Taxon ID for efficiency.
Please note, that not all taxa have all the possible identifiers available.
Use the <code><a href="get_taxa_by_ids.html">get_taxa_by_ids</a></code> function to retrieve the necessary information. For convenience, below is a list of officially supported taxa:</p><table class="table table"><tr><td><strong>ID</strong></td><td><strong>Comm.name</strong></td><td><strong>Scient.name</strong></td><td><strong>NcbiID</strong></td></tr><tr><td>1</td><td>human</td><td>Homo sapiens</td><td>9606</td></tr><tr><td>2</td><td>mouse</td><td>Mus musculus</td><td>10090</td></tr><tr><td>3</td><td>rat</td><td>Rattus norvegicus</td><td>10116</td></tr><tr><td>11</td><td>yeast</td><td>Saccharomyces cerevisiae</td><td>4932</td></tr><tr><td>12</td><td>zebrafish</td><td>Danio rerio</td><td>7955</td></tr><tr><td>13</td><td>fly</td><td>Drosophila melanogaster</td><td>7227</td></tr><tr><td>14</td><td>worm</td><td>Caenorhabditis elegans</td><td>6239</td></tr></table></dd>


<dt>offset</dt>
<dd><p>The offset of the first retrieved result.</p></dd>


<dt>limit</dt>
<dd><p>Optional, defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt>sort</dt>
<dd><p>Order results by the given property and direction. The '+' sign indicate ascending order whereas the '-' indicate descending.</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be a JSON file. Otherwise,
it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>


<dt>attributes</dt>
<dd><p>If <code>TRUE</code> additional information from the call will be added
into the output object's attributes such as offset and available elements.</p></dd>


<dt>...</dt>
<dd><p>Kept for compatibility, ignored.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data table with information about the queried dataset(s). A list if
<code>raw = TRUE</code>. Returns an empty list if no datasets matched.</p>


<p>The fields of the output data.table are:</p>


<ul><li><p><code>experiment.shortName</code>: Shortname given to the dataset within Gemma. Often corresponds to accession ID</p></li>
<li><p><code>experiment.name</code>: Full title of the dataset</p></li>
<li><p><code>experiment.ID</code>: Internal ID of the dataset.</p></li>
<li><p><code>experiment.description</code>: Description of the dataset</p></li>
<li><p><code>experiment.troubled</code>: Did an automatic process within gemma or a curator mark the dataset as "troubled"</p></li>
<li><p><code>experiment.accession</code>: Accession ID of the dataset in the external database it was taken from</p></li>
<li><p><code>experiment.database</code>: The name of the database where the dataset was taken from</p></li>
<li><p><code>experiment.URI</code>: URI of the original database</p></li>
<li><p><code>experiment.sampleCount</code>: Number of samples in the dataset</p></li>
<li><p><code>experiment.batchEffectText</code>: A text field describing whether the dataset has batch effects</p></li>
<li><p><code>experiment.batchCorrected</code>: Whether batch correction has been performed on the dataset.</p></li>
<li><p><code>experiment.batchConfound</code>: 0 if batch info isn't available, -1 if batch counfoud is detected, 1 if batch information is available and no batch confound found</p></li>
<li><p><code>experiment.batchEffect</code>: -1 if batch p value &lt; 0.0001, 1 if batch p value &gt; 0.1, 0 if otherwise and when there is no batch information is available or when the data is confounded with batches.</p></li>
<li><p><code>experiment.rawData</code>: -1 if no raw data available, 1 if raw data was available. When available, Gemma reprocesses raw data to get expression values and batches</p></li>
<li><p><code>geeq.qScore</code>: Data quality score given to the dataset by Gemma.</p></li>
<li><p><code>geeq.sScore</code>: Suitability score given to the dataset by Gemma. Refers to factors like batches, platforms and other aspects of experimental design</p></li>
<li><p><code>taxon.name</code>: Name of the species</p></li>
<li><p><code>taxon.scientific</code>: Scientific name for the taxon</p></li>
<li><p><code>taxon.ID</code>: Internal identifier given to the species by Gemma</p></li>
<li><p><code>taxon.NCBI</code>: NCBI ID of the taxon</p></li>
<li><p><code>taxon.database.name</code>: Underlying database used in Gemma for the taxon</p></li>
<li><p><code>taxon.database.ID</code>: ID of the underyling database used in Gemma for the taxon</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">search_datasets</span><span class="op">(</span><span class="st">"bipolar"</span>, taxon <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>search_datasets is deprecated. please use get_datasets instead</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:   McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:              byne-cc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:             GSE45484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:            GSE134497</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:            GSE160761</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:          GSE179921.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             GSE66433</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            GSE197966</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:            GSE202537</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:            GSE205422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:            GSE246593</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:            GSE157509</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:             GSE66196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:            GSE210064</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:             GSE23848</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:              GSE5389</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:              GSE5388</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:              GSE7036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:           McLean_PFC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:     stanley_feinberg</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                               experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                        &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                        McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                      Corpus Callosum data from Stanley collection samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                Gene-expression differences in peripheral blood between lithium responders and non-responders in the “Lithium Treatment -Moderate dose Use Study” (LiTMUS)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                              Total RNA sequecing for human induced pluripotent derived cerebral organoids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                    RNA sequencing  in human  iPSCs derived from  bipolar  patients  to identify important therapeutic molecular targets of Valproate(VPA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:   Split part 2 of: TCF7L2 lncRNA: A Link between Bipolar Disorder and Body Mass Index through Glucocorticoid Signaling [RNA-Seq] [collection of material = Experiment 1 ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                  Effects of the microRNA 137 and its connection to psychiatric disorders.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                          Transcriptional effects of bipolar disorder drugs on NT2-N cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                            Diurnal alterations in gene expression across striatal subregions in psychosis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                             Network-based integrative analysis of lithium response in bipolar disorder using transcriptomic and GWAS data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: Transition of allele-specific DNA hydroxymethylation at regulatory loci is associated with phenotypic variation in monozygotic twins discordant for psychiatric disorders</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                           Increased IL-6 and altered inflammatory response in bipolar disorder patient-derived astrocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                        Bipolar disorder and lithium-induced gene expression in two peripheral cell models</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                       Gene expression alterations in the postmortem hippocampus from older patients with bipolar disorder – a hypothesis generating study</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                         Peripheral blood gene-expression in depressed subjects with bipolar disorder vs healthy controls.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                             Adult postmortem brain tissue (orbitofrontal cortex) from subjects with bipolar disorder and healthy controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                   Adult postmortem brain tissue (dorsolateral prefrontal cortex) from subjects with bipolar disorder and healthy controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                 Expression profiling in monozygotic twins discordant for bipolar disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                McLean_PFC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                       Stanley consortium collection Cerebellum - Feinberg</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                               experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:           670</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          4354</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:          6145</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:         16450</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:         17943</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:         21159</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:         24427</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:         25070</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:         25749</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:         25972</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:         31703</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:         23425</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:         24428</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:         28041</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:          1958</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:           326</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:           594</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:           660</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:           672</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:           831</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Hippocampus of schizophrenic, bipolar, and control subjects. Analyzed from CEL files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Analysis of gene-expression changes in treatment responders vs non-responders to two different treatments among subjectrs participating in LiTMUS. Results provide information on pathways that may be involved in the clinical response to Lithium in patients with bipolar disorder.\nLast Updated (by provider): Apr 01 2013\nContributors:  Robert Beech</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Total RNA sequecing for human induced pluripotent derived cerebral organoids from healthy controls and Bipolar disorder\nAt time of import, last updated (by provider) on: Apr 01 2020\n\nContributors: ; [Annie Kathuria, Rakesh Karmacharya]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Valproate(VPA) has been used in the treatment of bipolar disorder since the 1990s.  However, the therapeutic targetsof VPA have remained elusive. Here we  used RNA sequencing  in human  iPSCs derived from  bipolar  patients  to further identify important molecular targets. Human iPSCs were homogenized and total RNA was isolated using the RNeasy Plus Micro Kit (Qiagen, Hilden, Germany). RNA quantity and quality were assessed using fluorometry (Qubit RNA Broad Range Assay Kit and Fluorometer; Invitrogen, Carlsbad, CA) and chromatography (Bioanalyzer and RNA 6000 Nano Kit; Agilent, Santa Clara, CA), respectively. Libraries were prepared using TruSeq Stranded mRNA (PolyA+) kit (Illumina, San Diego, CA) and sequenced by Illumina NextSeq 500. The read length was 75bp with 30-40M reads per sample. FastQC (v0.11.3) was performed to assess data quality. TopHat2 (v2.0.9) aligned the reads to the mouse reference genome (Mus musculus UCSC mm10) and to the Ensembl human reference genome (GRCh38.p13) using default parameters. Alignments were then converted to expression count data using HTseq (v0.6.1) with default union mode.\nAt time of import, last updated (by provider) on: Dec 31 2020\n\nContributors: ; [George Tseng, Colleen McClung, Wei Zong, Ryan Logan]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: This experiment was created by Gemma splitting another: \nExpressionExperiment Id=20933 Name=TCF7L2 lncRNA: A Link between Bipolar Disorder and Body Mass Index through Glucocorticoid Signaling [RNA-Seq] (GSE179921) Bipolar disorder (BD) and obesity are highly comorbid. We previously performed a genome-wide association study (GWAS) for BD risk accounting for the  effect of body mass index (BMI) which identified a genome-wide significant single-nucleotide polymorphism (SNP) in the gene encoding the transcription factor 7 like 2 (TCF7L2). However, the molecular function of TCF7L2 in the central nervous system (CNS) and its possible role in BD and BMI interaction remained unclear. In the present study, we demonstrated by studying human induced pluripotent stem cell (hiPSC)-derived astrocytes, cells which highly express TCF7L2 in the CNS, that the BD-BMI GWAS risk SNP is associated with glucocorticoid-dependent repression of the expression of a previously uncharacterized TCF7L2 transcript variant. That transcript is a long non-coding RNA (lncRNA-TCF7L2) that is highly expressed in the CNS but not in peripheral tissues such as the liver and pancreas which are involved in metabolism.  In astrocytes, knock-down of the lncRNA-TCF7L2 resulted in decreased expression of the parent gene, TCF7L2, as well as alterations in the expression of a series of genes involved in insulin signaling and diabetes.  We also studied the function of TCF7L2 in hiPSC-derived astrocytes by integrating RNA sequencing data after TCF7L2 knock-down with TCF7L2 chromatin-immunoprecipitation sequencing (ChIP-seq) data. Those studies showed that TCF7L2 directly regulated a series of BD-risk genes. In summary, these results support the existence of a CNS-based mechanism underlying BD-BMI genetic risk, a mechanism based on a glucocorticoid-dependent expression quantitative trait locus that regulates the expression of a novel TCF7L2 non-coding transcript.\nAt time of import, last updated (by provider) on: Sep 20 2021\n\nContributors: ; [Mark A Frye, Thanh L Nguyen, Tamas Ordog, Brandon Coombes, Richard M Weinshilboum, Huaizhi Huang, Zhenqing Ye, Liewei Wang, Huanyao Gao, Daniel Kim, Jeong-Heon Lee, Brenna Sharp, Duan Liu, Joanna Biernacka]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 MicroRNAs have been implicated in the pathology not only of cancer, but also of psychiatric diseases, such as bipolar disorder and schizophrenia. As several psychiatric disorders share the same risk genes, we hypothesized that this microRNA could also be associated with attention-deficit/hyperactivity disorder (ADHD) and that this association to psychiatric disorders might be due to the variable number of tandem repeats (VNTR) polymorphism within the internal miR-137 (Imir137) promoter (PMID 18316599; PMID 25154622). To further understand the role of the microRNA 137 in the brain a knock-down of miR-137 expression in SH-SY5Y neuroblastoma cells was performed followed by expression analysis using a microarray.\nAt time of import, last updated (by provider) on: Aug 08 2019\n\nContributors: ; [Lena Weißflog, Andreas Reif, Stefanie Berger, Heike Weber, Claus J Scholz]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Human neuronal-like cells (NT2-N) were treated with either lamotrigine (50 µM), lithium (2.5 mM), quetiapine (50 µM), valproate (0.5 mM) or vehicle control for 24 hours. Genome wide mRNA expression was quantified by RNA-sequencing. Results offer insights on the mechanism(s) of action of bipolar disorder drugs at the transcriptional level.\nAt time of import, last updated (by provider) on: Apr 27 2022\n\nContributors: ; [Srisaiyini Kidnapillai, Chiara Bortolasci, Laura Gray, Trang Truong, Bruna Panizzutti, Mark Richardson, Craig Smith, Olivia Dean, Zoe Liu, Briana Spolding, Michael Berk, Jee H Kim, Ken Walder]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Background: Psychosis is a defining feature of schizophrenia and highly prevalent in bipolar disorder. Notably, individuals suffering with these illnesses also have major disruptions in sleep and circadian rhythms, and disturbances to sleep and circadian rhythms can precipitate or exacerbate psychotic symptoms. Psychosis is associated with the striatum, though no study to date has directly measured molecular rhythms and determined how they are altered in the striatum of subjects with psychosis. Methods: Here, we perform RNA-sequencing and both differential expression and rhythmicity analyses to investigate diurnal alterations in gene expression in human postmortem striatal subregions (NAc, caudate, and putamen) in subjects with psychosis relative to unaffected comparison subjects. Results: Across regions, we find differential expression of immune-related transcripts and a substantial loss of rhythmicity in core circadian clock genes in subjects with psychosis. In the nucleus accumbens (NAc), mitochondrial-related transcripts have decreased expression in psychosis subjects, but only in those who died at night. Additionally, we find a loss of rhythmicity in small nucleolar RNAs and a gain of rhythmicity in glutamatergic signaling in the NAc of psychosis subjects. Between region comparisons indicate that rhythmicity in the caudate and putamen is far more similar in subjects with psychosis than in matched comparison subjects. Conclusions: Together, these findings reveal differential and rhythmic gene expression differences across the striatum that may contribute to striatal dysfunction and psychosis in psychotic disorders.\nAt time of import, last updated (by provider) on: Aug 31 2022\n\nContributors: ; [George Tseng, Colleen McClung, Wei Zong, Kyle Ketchesin]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Lithium is the gold standard treatment for bipolar disorder. The goal of this study was to identify gene expression networks associated with lithium response. RNAseq data was obtained from IPSC derived neurons from lithium responders and non-responders. Focal adhesion was the network most associated with response.\nAt time of import, last updated (by provider) on: Jun 09 2022\n\nContributors: ; [Vipavee Niemsiri, Fred Gage, John Kelsoe]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                          Major psychiatric disorders such as schizophrenia (SCZ) and bipolar disorder (BPD) are complex genetic mental illnesses. Their non-Mendelian features such as monozygotic twins discordant for SCZ or BPD are likely complicated by environmental modifiers of genetic effects. 5-hydroxymethylcytosine (5hmC) is an important epigenetic marker in gene regulation and whether its links with genetic variants contribute to the non-Mendelian features remain largely unexplored. Here, we performed hydroxymethylome and genome analyses of blood DNA from psychiatric disorder-discordant monozygotic twins to study how allele-specific hydroxymethylation (AShM) mediates phenotypic variations. We identified thousands of genetic variants with AShM imbalances who exhibit phenotypic variation-associated AShM transition at regulatory loci. These AShMs have plausible causal associations with psychiatric disorders through effects on interactions between transcription factors, DNA methylations, or other epigenomic markers and then contribute to dysregulated gene expression, which eventually increases disease susceptibility. We then validated that competitive binding of POU3F2 on the alternative allele of psyAShM site rs4558409 (G/T) in PLLP can enhance the PLLP expression, while hydroxymethylated alternative allele alleviating the transcription factor binding activity at rs4558409 site might be associated with downregulated PLLP expression observed in BPD or SCZ. Moreover, disruption of rs4558409 induces gain of PLLP function and promotes neural development and vesicle trafficking. Our study provides a powerful strategy for prioritizing regulatory risk variants and contributes to our understanding of the interplay between genetic and epigenetic factors in mediating complex disease susceptibility.\nAt time of import, last updated (by provider) on: Oct 31 2023\n\nContributors: ; [Zhanwang Huang, Junping Ye]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       The goals of this study are to examine responses to inflammation in astrocytes from  induced pluripotent stem cells derived from healthy controls and bipolar disorder patients. We examine the transcriptomic inflmmatory signature of generated astrocytes following Il1Beta exposure in BD vs. control Results: BD-patient astrocytes show a unique inflammatory response with differentially regulated genes.\nAt time of import, last updated (by provider) on: Mar 19 2021\n\nContributors: ; [Maxim N Shokhirev, Fred Gage, Krishna Vadodaria, Carol Marchetto]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Bipolar disorder is a severe and heritable psychiatric disorder and affects up to 1% of the population worldwide. Lithium is recommended as first-line treatment for the maintenance treatment of bipolar-affective disorder in current guidelines, its molecular modes of action are however poorly understood. Cell models derived from bipolar patients could prove useful to gain more insight in the molecular mechanisms of bipolar disorder and the common pharmacological treatments. As primary neuronal cell lines cannot be easily derived from patients, peripheral cell models should be evaluated in their usefulness to study pathomechanisms and the mode of action of medication as well as in regard to develop biomarkers for diagnosis and treatment response.\nAt time of import, last updated (by provider) on: Mar 25 2019\n\nContributors: ; [Sarah Kittel-Schneider, Max Hilscher, Andreas Reif, Claus J Scholz]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Gene expression of samples from the postmortem hippocampus of older bipolar disorder subjects and controls. Gene expression data was generated using the SurePrint G3 Human Gene Expression v3 microarray. Rank feature selection was performed to identify a subset of features that can optimally differentiate BD and controls.\nAt time of import, last updated (by provider) on: Feb 19 2023\n\nContributors: ; [Carlos A Pasqualucci, Claudia K Suemoto, Ricardo Nitrini, Fernanda B Bertonha, Paula V Nunes, Katia C De Oliveira, Carlos M Filho, Helena K Kim, Helena Brentani, Lea T Grinberg, Beny Lafer, André Barbosa, Camila Nascimento, Renata P Leite, Wilson Jacob-Filho]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Analysis of gene-expression changes in depressed subjects with bipolar disorder compared to healthy controls. Results provide information on pathways that may be involved in the pathogenesis of bipolar depression.\nLast Updated (by provider): Aug 27 2010\nContributors:  Robert D Beech</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Bipolar affective disorder is a severe psychiatric disorder with a strong genetic component but unknown pathophysiology. We used microarray technology (Affymetrix HG-U133A GeneChips) to determine the expression of approximately 22 000 mRNA transcripts in post-mortem brain tissue (orbitofrontal cortex) from patients with bipolar disorder and matched healthy controls. Orbitofrontal cortex tissue from a cohort of 30 subjects was investigated and the final analysis included 10 bipolar and 11 control subjects. Differences between disease and control groups were identified using a rigorous statistical analysis with correction for confounding variables and multiple testing.\nNote: [] samples from this series, which appear in other Expression Experiments in Gemma, were not imported from the GEO source. The following samples were removed: \nLast Updated (by provider): Jul 27 2006\nContributors:  Sabine Bahn Margaret M Ryan Matthew T Wayland Maree J Webster Stephen J Huffaker Helen E Lockstone\nIncludes GDS2191.\n Update date: Aug 28 2006.\n Dataset description GDS2191: Analysis of postmortem orbitofrontal cortex from 10 adults with bipolar disorder. Results provide insight into the pathophysiology of the disease.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Bipolar affective disorder is a severe psychiatric disorder with a strong genetic component but unknown pathophysiology. We used microarray technology (Affymetrix HG-U133A GeneChips) to determine the expression of approximately 22 000 mRNA transcripts in post-mortem brain tissue (dorsolateral prefrontal cortex) from patients with bipolar disorder and matched healthy controls. A cohort of 70 subjects was investigated and the final analysis included 30 bipolar and 31 control subjects. Differences between disease and control groups were identified using a rigorous statistical analysis with correction for confounding variables and multiple testing.\nLast Updated (by provider): Jan 16 2007\nContributors:  Helen E Lockstone Stephen J Huffaker Matthew T Wayland Sabine Bahn Maree J Webster Margaret M Ryan\nIncludes GDS2190.\n Update date: Aug 28 2006.\n Dataset description GDS2190: Analysis of postmortem dorsolateral prefrontal cortex from 30 adults with bipolar disorder. Results provide insight into the pathophysiology of the disease.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               To identify genes dysregulated in bipolar disorder (BD1) we carried out global gene expression profiling using whole-genome microarrays. To minimize genetic variation in gene expression levels between cases and controls we compared expression profiles in lymphoblastoid cell lines from monozygotic twin pairs discordant for the disease. We identified 82 genes that were differentially expressed by ? 1.3-fold in 3 BD1 cases compared to their co-twins, and which were statistically (p ? 0.05) differentially expressed between the groups of BD1 cases and controls. Using qRT-PCR we confirmed the differential expression of some of these genes, including: KCNK1, MAL, PFN2, TCF7, PGK1, and PI4KCB, in at least 2 of the twin pairs. In contrast to the findings of a previous study by Kakiuchi and colleagues with similar discordant BD1 twin design1 our data do not support the dysregulation of XBP1 and HSPA5. From pathway and gene ontology analysis we identified up-regulation of the WNT signalling pathway and the biological process of apoptosis. The differentially regulated genes and pathways identified in this study may provide insights into the biology of BD1.\nLast Updated (by provider): Jun 20 2007\nContributors:  Louisa Windus Nicholas Matigian Bryan Mowry Cheryl Filippich John McGrath Heather Smith Nicholas Hayward Christos Pantelis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Prefrontal cortex of schizophrenic, bipolar, and control subjects. This is the "McLean 66"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            50 samples of individuals from 4 different diagnostic groups: bipolar, schizophrenia, depression and controls. Samples taken from the Cerebellum.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE             GSE45484                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE            GSE134497                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE            GSE160761                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE            GSE179921                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE             GSE66433                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE            GSE197966                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE            GSE202537                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE            GSE205422                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE            GSE246593                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE            GSE157509                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE             GSE66196                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE            GSE210064                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE             GSE23848                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE              GSE5389                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE              GSE5388                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE              GSE7036                 GEO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE                 &lt;NA&gt;                &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                   experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                           &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                         &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                         &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE134497</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE160761</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE179921</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE66433</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE197966</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE202537</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE205422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE246593</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE157509</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE66196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE210064</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE23848</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:   https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE5389</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:   https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE5388</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:   https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE7036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                         &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                         &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                   experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated experiment.batchEffectText</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      &lt;int&gt;                 &lt;POSc&gt;                     &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                     23    2023-09-08 07:38:04              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                     93    2022-08-30 23:36:54              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                    120    2023-12-19 20:24:26              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                     22    2023-12-17 10:03:11              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                     24    2023-12-17 14:05:13              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                      4    2023-12-17 12:01:40       SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                      6    2023-12-20 19:14:46       SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                     24    2023-12-18 11:53:22              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                    215    2024-05-08 23:46:31  SINGLETON_BATCHES_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                     34    2023-12-18 00:19:38    NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                      4    2024-01-23 19:56:05       SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                     33    2023-12-17 01:36:32              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                     12    2023-12-20 19:03:56              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                     22    2023-12-18 01:58:21       SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                     35    2023-12-18 12:17:06              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                     21    2023-12-20 08:00:37    NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                     61    2023-12-20 07:59:57    NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                      6    2023-12-20 21:45:32       SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                     66    2023-09-21 21:43:05              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                     50    2023-09-07 23:11:52              NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated experiment.batchEffectText</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.batchCorrected experiment.batchConfound experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        &lt;lgcl&gt;                    &lt;num&gt;                  &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                     FALSE                        1                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                     FALSE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                     FALSE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                     FALSE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                     FALSE                        1                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                     FALSE                       -1                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                     FALSE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                     FALSE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                     FALSE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                     FALSE                        1                      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                     FALSE                        0                      0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.batchCorrected experiment.batchConfound experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.rawData geeq.qScore geeq.sScore taxon.name taxon.scientific</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;num&gt;       &lt;num&gt;       &lt;num&gt;     &lt;char&gt;           &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                 -1   0.2819105      0.2500      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                 -1   0.4224650      0.4375      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                 -1   0.4254127      0.7500      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                  1   0.4175250      0.7500      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                  1   0.4170825      0.7500      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                  1   0.8542963      0.7500      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                 -1   0.8564356     -0.0375      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                  1   0.2824440      1.0000      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                  1   0.4225033      1.0000      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                  1   0.9948398      0.7500      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                  1   0.7105362      0.7500      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                  1   0.4225534      1.0000      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  1   0.2840504      0.8750      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                 -1   0.8411107      0.3125      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                 -1   0.4246487      0.5000      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                  1   0.8517696      0.8750      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                 -1   0.9948828      0.6250      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                  1   0.7121314      0.8375      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                 -1   0.1370016      0.1250      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                 -1   0.1395366      0.0000      human     Homo sapiens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.rawData geeq.qScore geeq.sScore taxon.name taxon.scientific</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.ID taxon.NCBI taxon.database.name taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;int&gt;      &lt;int&gt;              &lt;char&gt;             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:        1       9606                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.ID taxon.NCBI taxon.database.name taxon.database.ID</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

