<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Search everything in Gemma — search_gemma • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Search everything in Gemma — search_gemma"><meta property="og:description" content="Search everything in Gemma"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Search everything in Gemma</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>search_gemma.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Search everything in Gemma</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">search_gemma</span><span class="op">(</span></span>
<span>  <span class="va">query</span>,</span>
<span>  taxon <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  platform <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  resultType <span class="op">=</span> <span class="st">"experiment"</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>The search query. Either plain text ('traumatic'), or an ontology
term URI ('http://purl.obolibrary.org/obo/UBERON_0002048'). Datasets that
contain the given string in their short of full name will also be matched.
Can be multiple identifiers separated by commas.</p></dd>


<dt>taxon</dt>
<dd><p>A numerical taxon identifier or an ncbi taxon identifier or a taxon identifier that matches either its scientific or common name</p></dd>


<dt>platform</dt>
<dd><p>A platform numerical identifier or a platform short name</p></dd>


<dt>limit</dt>
<dd><p>Optional, defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt>resultType</dt>
<dd><p>The kind of results that should be included in the output. Can be experiment, gene, platform or a long object type name, documented in the API documentation.</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>If <code>raw = FALSE</code> and resultType is experiment, gene or platform,
a data.table containing the search results. If it is any other type, a list
of results. A list with additional details about the search if <code>raw = TRUE</code></p>


    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">search_gemma</span><span class="op">(</span><span class="st">"bipolar"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             GSE67645</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:             GSE33085</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:             GSE35077</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             GSE98838</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:             GSE29318</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:             GSE66196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             GSE10843</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:            GSE157509</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:             GSE23848</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:              GSE7036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:             GSE62191</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:            GSE197966</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:             GSE80336</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:             GSE87610</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:             GSE81396</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:            GSE127771</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:            GSE126942</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:             GSE53987</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:             GSE58933</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:            GSE134497</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                       experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: Transcriptome dynamics of developing photoreceptors in 3-D retina cultures recapitulates temporal sequence of human cone and rod differentiation revealing cell surface markers and gene networks</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                Transcriptome analysis of adult retina cell types.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                            A gene expression database for retinal neuron subtypes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                        Combining RNA-Seq and Somatic CRISPR Mutagenesis to Study Mouse Neural Development in vivo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                             Expression profile of FAC-sorted murine retinal cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                Bipolar disorder and lithium-induced gene expression in two peripheral cell models</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                    mRNA Cancer Cell Line Profiles</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                   Increased IL-6 and altered inflammatory response in bipolar disorder patient-derived astrocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                 Peripheral blood gene-expression in depressed subjects with bipolar disorder vs healthy controls.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                         Expression profiling in monozygotic twins discordant for bipolar disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                    Gene expression profiles of patients with schizophrenia, bipolar disorder and healthy controls</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                  Transcriptional effects of bipolar disorder drugs on NT2-N cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                          Transcriptome profiling of the human dorsal striatum in bipolar disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                              Gene expression of L3 and L5 pyramidal neurons in the DLPFC comparing schizophrenia from bipolar major depressive disorders and unaffected subjects.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                     Total RNAseq of human putamen and caudate nucleus tissues in healthy control and Bipolar Disorder individuals</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                         The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells [P7]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                              The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                Microarray profiling of PFC, HPC and STR from subjects with schizophrenia, bipolar, MDD or control</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                       Hyper-excitability of Neurons generated from Patients with Bipolar Disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                      Total RNA sequecing for human induced pluripotent derived cerebral organoids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                       experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:         11634</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          8887</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:          7608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:         13214</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:         12310</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:         24428</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:          7196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:         23425</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:          1958</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:           660</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:          8995</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:         25070</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:         10627</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:         10878</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:         13338</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:         19455</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:         19454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:          8359</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:         12503</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:         16450</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               To define molecular mechanisms underlying rod and cone differentiation, we generated H9 human embryonic stem cell line carrying a GFP reporter that is controlled by the promoter of cone-rod homeobox (CRX) gene, the first known marker of post-mitotic photoreceptor precursors. CRXp-GFP reporter in H9 line replicates endogenous CRX expression when induced to form self-organizing 3-D retina-like tissue. We define temporal transcriptome dynamics of developing photoreceptors during the establishment of cone and rod cell fate. Our studies provide an essential framework for delineating molecules and cellular pathways that guide human photoreceptor development and should assist in chemical screening and cell-based therapies of retinal degeneration.\nLast Updated (by provider): Nov 14 2017\nContributors:  Kohei Homma Rossukon Kaewkhaw Anand Swaroop Koray D Kaya Jizhong Zou Mahendra Rao Matthew Brooks Vijender Chaitankar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                     Brain circuits are assembled from a large variety of morphologically and functionally diverse cell types. It is not known how the intermingled cell types of individual brain regions differ in their expressed genomes. Here we describe an atlas of cell type transcriptomes of the adult retina. We found that each adult cell type expresses a specific set of genes, including a unique set of transcription factors, forming a “barcode” for cell identity. Cell type transcriptomes carry enough information to categorize cells into corresponding morphological classes and types. Surprisingly, several barcode genes are eye disease-associated genes that we demonstrate to be specifically expressed not only in photoreceptors but also in particular retinal circuit elements such as inhibitory neurons as well as in retinal microglia. Our data suggest that distinct cell types of individual brain regions are characterized by marked differences in their expressed genomes. We assembled a library of 22 transgenic mouse lines in which each line had a group of retinal cells marked with fluorescent proteins. We built up the library with the purpose of having some mouse lines in which single retinal cell types and others in which combinations of types from a single class are labeled. The library had mouse lines with labeled cells representing each of the six retinal cell classes. Retinal cells were characterized by physiological recording and immunohistochemical staining. We isolated 200 fluorescent protein-labeled retinal cells (“cell groups”) from at least three different mice of each mouse line by fluorescence-activated cell sorting. The transcripts of each cell group of these biological triplicates were independently amplified in batches. Each batch contained an internal control cell group from the Arc-line.\nLast Updated (by provider): Jun 10 2014\nContributors:  Botond Roska Erik Cabuy Sandra Siegert</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       The goal of this experiment was to define gene expression patterns of thirteen mouse retinal neuron subsets, labeled by expression of fluorescent proteins in transgenic mice.\nLast Updated (by provider): Sep 23 2013\nContributors:  Jeremy N Kay Joshua R Sanes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      We used RNA sequencing to identify candidate regulators of interactions between photoreceptor axons and bipolar cell (BCs) dendrites in developing mouse retina. We chose three time points: P7, just after the OPL forms and synaptogenesis with BCs begins; P13, as synaptogenesis nears completion and sublamination begins; and P30, when the OPL is mature. We purified cone and rod photoreceptors separately by fluorescence activated cell sorting (FACS) using transgene markers: Rhoicre;Ai9 for rods and HRGPcre;Ai9 for cones. We purified ON BCs, which include ON cone bipolars plus rod bipolars using Grm6:GFP. As appropriate transgenic lines to separate RBCs from CBCs were not available, we performed RNAseq on cells from Grm6:GFP mice that were fixed and immunostained prior to FACS, allowing us to purify RBCs (GFP+PKC+) and CBCs (GFP+PKC-) from the same retinas. As PKC is not highly expressed at P7, profiling of developing rod bipolars separate from developing ON cone bipolars was restricted to P13.\nLast Updated (by provider): Jun 25 2018\nContributors:  Elizabeth Z Sanchez Lawrence S Zipursky Yerbol Z Kurmangaliyev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Microarray experiments were performed using FAC-sorted young photoreceptors to analyze their transcriptome in comparison to remaining retinal cells at same developmental stage and retinal progenitors.\nLast Updated (by provider): Apr 02 2018\nContributors:  Marius Ader Kai Postel</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Bipolar disorder is a severe and heritable psychiatric disorder and affects up to 1% of the population worldwide. Lithium is recommended as first-line treatment for the maintenance treatment of bipolar-affective disorder in current guidelines, its molecular modes of action are however poorly understood. Cell models derived from bipolar patients could prove useful to gain more insight in the molecular mechanisms of bipolar disorder and the common pharmacological treatments. As primary neuronal cell lines cannot be easily derived from patients, peripheral cell models should be evaluated in their usefulness to study pathomechanisms and the mode of action of medication as well as in regard to develop biomarkers for diagnosis and treatment response.\nAt time of import, last updated (by provider) on: Mar 25 2019\n\nContributors: ; [Sarah Kittel-Schneider, Max Hilscher, Andreas Reif, Claus J Scholz]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   mRNA cancer cell line profiles\nLast Updated (by provider): Oct 25 2013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    The goals of this study are to examine responses to inflammation in astrocytes from  induced pluripotent stem cells derived from healthy controls and bipolar disorder patients. We examine the transcriptomic inflmmatory signature of generated astrocytes following Il1Beta exposure in BD vs. control Results: BD-patient astrocytes show a unique inflammatory response with differentially regulated genes.\nAt time of import, last updated (by provider) on: Mar 19 2021\n\nContributors: ; [Maxim N Shokhirev, Fred Gage, Krishna Vadodaria, Carol Marchetto]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Analysis of gene-expression changes in depressed subjects with bipolar disorder compared to healthy controls. Results provide information on pathways that may be involved in the pathogenesis of bipolar depression.\nLast Updated (by provider): Aug 27 2010\nContributors:  Robert D Beech</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            To identify genes dysregulated in bipolar disorder (BD1) we carried out global gene expression profiling using whole-genome microarrays. To minimize genetic variation in gene expression levels between cases and controls we compared expression profiles in lymphoblastoid cell lines from monozygotic twin pairs discordant for the disease. We identified 82 genes that were differentially expressed by ? 1.3-fold in 3 BD1 cases compared to their co-twins, and which were statistically (p ? 0.05) differentially expressed between the groups of BD1 cases and controls. Using qRT-PCR we confirmed the differential expression of some of these genes, including: KCNK1, MAL, PFN2, TCF7, PGK1, and PI4KCB, in at least 2 of the twin pairs. In contrast to the findings of a previous study by Kakiuchi and colleagues with similar discordant BD1 twin design1 our data do not support the dysregulation of XBP1 and HSPA5. From pathway and gene ontology analysis we identified up-regulation of the WNT signalling pathway and the biological process of apoptosis. The differentially regulated genes and pathways identified in this study may provide insights into the biology of BD1.\nLast Updated (by provider): Jun 20 2007\nContributors:  Louisa Windus Nicholas Matigian Bryan Mowry Cheryl Filippich John McGrath Heather Smith Nicholas Hayward Christos Pantelis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Schizophrenia (SZ) and bipolar disorder (BD) are severe psychiatric conditions, with a lifetime prevalence of about 1%. Both disorders have a neurodevelopment component, with onset of symptoms occurring most frequently during late adolescence or early adulthood. Genetic findings indicate the existence of an overlap in genetic susceptibility across the disorders. These gene expression profiles were used to identify the molecular mechanisms that differentiate SZ and BP from healthy controls but also that distinguish both from healthy individuals. They were also used to expand an analysis from an experiment that searched molecular alterations in human induced pluripotent stem cells derived from fibroblasts from control subject and individual with schizophrenia and further differentiated to neuron to identify genes relevant for the development of schizophrenia (GSE62105).\nLast Updated (by provider): Oct 14 2014\nContributors:  Leandro Lima Mariana Maschietto Dirce M Carraro Carlos A Filho Angelica de Baumont Luiz A Barreta Paulo Belmonte-de-Abreu Ana C Tahira Eloisa H Olivieri Joana A Palha Helena Brentani Daniel Mariani Alex Fiorini</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Human neuronal-like cells (NT2-N) were treated with either lamotrigine (50 µM), lithium (2.5 mM), quetiapine (50 µM), valproate (0.5 mM) or vehicle control for 24 hours. Genome wide mRNA expression was quantified by RNA-sequencing. Results offer insights on the mechanism(s) of action of bipolar disorder drugs at the transcriptional level.\nAt time of import, last updated (by provider) on: Apr 27 2022\n\nContributors: ; [Srisaiyini Kidnapillai, Chiara Bortolasci, Laura Gray, Trang Truong, Bruna Panizzutti, Mark Richardson, Craig Smith, Olivia Dean, Zoe Liu, Briana Spolding, Michael Berk, Jee H Kim, Ken Walder]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Bipolar disorder (BD) is a highly heritable and heterogeneous mental illness whose manifestations often include impulsive and risk-taking behavior. This particular phenotype suggests that abnormal striatal function could be involved in BD etiology, yet most transcriptomic studies of this disorder have concentrated on cortical brain regions. We report the first transcriptome profiling by RNA-Seq of the human dorsal striatum comparing bipolar and control subjects. Differential expression analysis and functional pathway enrichment analysis were performed to identify changes in gene expression that correlate with BD status. Further co-expression and enrichment analyses were performed to identify sets of correlated genes that show association to BD.\nLast Updated (by provider): May 17 2017\nContributors:  Ronald L Davis Rodrigo Pacifico</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Impairments in certain cognitive processes (e.g., working memory) are typically most pronounced in schizophrenia (SZ), intermediate in bipolar disorder (BP) and least in major depressive disorder (MDD). Given that working memory depends, in part, on neural circuitry that includes pyramidal neurons in layer 3 (L3) and layer 5 (L5) of the dorsolateral prefrontal cortex (DLPFC), we sought to determine if transcriptome alterations in these neurons were shared or distinctive for each diagnosis.\nLast Updated (by provider): Jul 05 2017\nContributors:  Dominique Arion David A Lewis John F Enwright George Tseng Zhiguang Huo John P Corradi</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     A multitude of genes have been associated with bipolar disorder via SNP genotyping studies.  However, many of these associated SNPs are found within intronic or intergenic regions of the human genome.  We were interested in studying transcriptional profiles/splice variation of genes associated with bipolar disorder within the human striatum. Understanding how these associated genes are transcribed in the human brain may help to guide the development of therapeutic agents for the treatment of bipolar disorder and other neuropsychiatric illnesses.\nLast Updated (by provider): Jun 26 2018\nContributors:  Courtney M MacMullen Ronald L Davis Mohammad Fallahi</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                     Schizophrenia is a complex psychiatric disorder encompassing a range of symptoms and  etiology dependent upon the interaction of genetic and environmental factors.  Several risk genes, such as DISC1, have been associated with schizophrenia as well as bipolar disorder (BPD) and major depressive disorder (MDD), consistent with the hypothesis that a shared genetic architecture could contribute to divergent clinical syndromes.  The present study compared gene expression profiles across three brain regions in post-mortem tissue from matched subjects with schizophrenia, BPD or MDD and unaffected controls.  Post-mortem brain tissue was collected from control subjects and well-matched subjects with schizophrenia, BPD, and MDD (n=19 from each group).  RNA was isolated from hippocampus, Brodmann Area 46, and associative striatum and hybridized to U133_Plus2 Affymetrix chips.  Data were normalized by RMA, subjected to pairwise comparison followed by Benjamini and Hochberg False Discovery Rate correction (FDR).  Samples derived from patients with schizophrenia exhibited  many more changes in gene expression across all brain regions than observed in BPD or MDD.  Several genes showed changes in both schizophrenia and BPD, though the magnitude of change was usually larger in schizophrenia.  Several genes that have variants associated with schizophrenia were found to have altered expression in multiple regions of brains from subjects with schizophrenia.  Continued evaluation of circuit-level alterations in gene expression and gene-network relationships may further our understanding of how genetic variants may be influencing biological processes to contribute to psychiatric disease.\nLast Updated (by provider): May 19 2014\nContributors:  Thomas A Lanz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19: Bipolar Disorder (BD) is a complex neuropsychiatric disorder that is characterized by intermittent episodes of mania and depression and, without treatment, 15% of patients commit suicide1. Hence, among all diseases, BD has been ranked by the WHO as a top disorder of morbidity and lost productivity2. Previous neuropathological studies have revealed a series of alterations in the brains of BD patients or animal models3, such as reduced glial cell number in the patient prefrontal cortex4, up-regulated activities of the PKA/PKC pathways5-7, and changes in dopamine/5-HT/glutamate neurotransmission systems8-11. However, the roles and causation of these changes in BD are too complex to exactly determine the pathology of the disease; none of the current BD animal models can recapitulate both the manic and depressive phenotypes or spontaneous cycling of BD simultaneously12,13. Furthermore, while some patients show remarkable improvement with lithium treatment, for yet unknown reasons, other patients are refractory to lithium treatment. Therefore, developing an accurate and powerful biological model has been a challenge for research into BD. The development of induced pluripotent stem cell (iPSC) technology has provided such a new approach. Here, we developed a human BD iPSC model and investigated the cellular phenotypes of hippocampal dentate gyrus neurons derived from the patient iPSCs. Using patch clamp recording, somatic Ca2+ imaging and RNA-seq techniques, we found that the neurons derived from BD patients exhibited hyperactive action potential (AP) firing, up-regulated expression of PKA/PKC/AP and mitochondria-related genes. Moreover, lithium selectively reversed these alterations in the neurons of patients who responded to lithium treatment. Therefore, hyper-excitability is one endophenotype of BD that is probably achieved through enhancement in the PKA/PKC and Na+ channel signaling systems, and our BD iPSC model can be used to develop new therapies and drugs aimed at clinical treatment of this disease.\nLast Updated (by provider): Jun 11 2018\nContributors:  Son Pham Jun Yao Fred H Gage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Total RNA sequecing for human induced pluripotent derived cerebral organoids from healthy controls and Bipolar disorder\nAt time of import, last updated (by provider) on: Apr 01 2020\n\nContributors: ; [Annie Kathuria, Rakesh Karmacharya]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;         &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated  experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      &lt;int&gt;                 &lt;POSc&gt;                  &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                     23    2023-12-20 20:07:23           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                     78    2023-12-19 02:38:44           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                     26    2023-12-19 05:21:57           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                     22    2023-12-22 09:47:02           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                      9    2023-12-18 20:37:33 BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                     12    2023-12-20 19:03:56           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                    206    2023-12-16 09:52:49           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                     33    2023-12-17 01:36:32           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                     35    2023-12-18 12:17:06           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                      6    2023-12-20 21:45:32    SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                     88    2023-12-20 15:25:17           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                     24    2023-12-18 11:53:22           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                     36    2023-12-21 04:26:14 NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                    286    2023-12-22 08:39:18           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                     16    2023-12-21 12:58:54           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                      6    2023-12-17 09:06:10           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                      6    2023-12-17 09:01:07           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                    205    2023-12-20 12:59:34           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                     18    2023-12-20 12:26:21 NO_BATCH_EFFECT_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                     22    2023-12-17 10:03:11           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated  experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;              &lt;int&gt;            &lt;int&gt;        &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.name taxon.scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;num&gt;       &lt;num&gt;     &lt;char&gt;           &lt;char&gt;    &lt;int&gt;      &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.name taxon.scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.name taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;char&gt;             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.name taxon.database.ID</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

