<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Search everything in Gemma — search_gemma • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Search everything in Gemma — search_gemma"><meta property="og:description" content="Search everything in Gemma"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Search everything in Gemma</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>search_gemma.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Search everything in Gemma</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">search_gemma</span><span class="op">(</span></span>
<span>  <span class="va">query</span>,</span>
<span>  taxon <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  platform <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  limit <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  resultType <span class="op">=</span> <span class="st">"experiment"</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>query</dt>
<dd><p>The search query. Either plain text ('traumatic'), or an ontology
term URI ('http://purl.obolibrary.org/obo/UBERON_0002048'). Datasets that
contain the given string in their short of full name will also be matched.
Can be multiple identifiers separated by commas.</p></dd>


<dt>taxon</dt>
<dd><p>A numerical taxon identifier or an ncbi taxon identifier or a taxon identifier that matches either its scientific or common name</p></dd>


<dt>platform</dt>
<dd><p>A platform numerical identifier or a platform short name</p></dd>


<dt>limit</dt>
<dd><p>Optional, defaults to 20. Limits the result to specified amount
of objects. Has a maximum value of 100. Use together with <code>offset</code> and
the <code>totalElements</code> <a href="https://rdrr.io/r/base/attributes.html" class="external-link">attribute</a> in the output to
compile all data if needed.</p></dd>


<dt>resultType</dt>
<dd><p>The kind of results that should be included in the output. Can be experiment, gene, platform or a long object type name, documented in the API documentation.</p></dd>


<dt>raw</dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt>memoised</dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt>file</dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt>overwrite</dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>If <code>raw = FALSE</code> and resultType is experiment, gene or platform,
a data.table containing the search results. If it is any other type, a list
of results. A list with additional details about the search if <code>raw = TRUE</code></p>


    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">search_gemma</span><span class="op">(</span><span class="st">"bipolar"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:             GSE67645</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:              byne-cc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:            GSE160761</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:             GSE45484</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:            GSE134497</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:   McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:             GSE33085</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:             GSE35077</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:             GSE98838</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:             GSE11767</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:       stanley_vawter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:             GSE29318</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:            GSE126942</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:            GSE127771</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:             GSE10843</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:             GSE66196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:             GSE97534</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:            GSE157509</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:             GSE23848</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:              GSE7036</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.shortName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                       experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1: Transcriptome dynamics of developing photoreceptors in 3-D retina cultures recapitulates temporal sequence of human cone and rod differentiation revealing cell surface markers and gene networks</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                              Corpus Callosum data from Stanley collection samples</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                            RNA sequencing  in human  iPSCs derived from  bipolar  patients  to identify important therapeutic molecular targets of Valproate(VPA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                        Gene-expression differences in peripheral blood between lithium responders and non-responders in the “Lithium Treatment -Moderate dose Use Study” (LiTMUS)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                      Total RNA sequecing for human induced pluripotent derived cerebral organoids</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                McLean Hippocampus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                                                                                                                                                Transcriptome analysis of adult retina cell types.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                            A gene expression database for retinal neuron subtypes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                        Combining RNA-Seq and Somatic CRISPR Mutagenesis to Study Mouse Neural Development in vivo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                   Gene expression profiling of fibroblasts and lymphoblastoid cells derived from four individuals</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                           Stanley array collection DLPFC - Vawter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                             Expression profile of FAC-sorted murine retinal cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                              The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                         The LIM-homeodomain transcription factor LHX4 is required for the differentiation of retinal rod bipolar cells and rod-connecting cone bipolar cells [P7]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                    mRNA Cancer Cell Line Profiles</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                Bipolar disorder and lithium-induced gene expression in two peripheral cell models</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                  RNA sequencing of developing cone photoreceptors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                   Increased IL-6 and altered inflammatory response in bipolar disorder patient-derived astrocytes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                 Peripheral blood gene-expression in depressed subjects with bipolar disorder vs healthy controls.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                         Expression profiling in monozygotic twins discordant for bipolar disorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                       experiment.name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:         11634</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          4354</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:         17943</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:          6145</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:         16450</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:           670</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:          8887</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:          7608</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:         13214</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:          1879</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:           836</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:         12310</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:         19454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:         19455</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:          7196</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:         24428</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:         13694</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:         23425</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:          1958</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:           660</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           To define molecular mechanisms underlying rod and cone differentiation, we generated H9 human embryonic stem cell line carrying a GFP reporter that is controlled by the promoter of cone-rod homeobox (CRX) gene, the first known marker of post-mitotic photoreceptor precursors. CRXp-GFP reporter in H9 line replicates endogenous CRX expression when induced to form self-organizing 3-D retina-like tissue. We define temporal transcriptome dynamics of developing photoreceptors during the establishment of cone and rod cell fate. Our studies provide an essential framework for delineating molecules and cellular pathways that guide human photoreceptor development and should assist in chemical screening and cell-based therapies of retinal degeneration.\nLast Updated (by provider): Nov 14 2017\nContributors:  Kohei Homma Rossukon Kaewkhaw Anand Swaroop Koray D Kaya Jizhong Zou Mahendra Rao Matthew Brooks Vijender Chaitankar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Valproate(VPA) has been used in the treatment of bipolar disorder since the 1990s.  However, the therapeutic targetsof VPA have remained elusive. Here we  used RNA sequencing  in human  iPSCs derived from  bipolar  patients  to further identify important molecular targets. Human iPSCs were homogenized and total RNA was isolated using the RNeasy Plus Micro Kit (Qiagen, Hilden, Germany). RNA quantity and quality were assessed using fluorometry (Qubit RNA Broad Range Assay Kit and Fluorometer; Invitrogen, Carlsbad, CA) and chromatography (Bioanalyzer and RNA 6000 Nano Kit; Agilent, Santa Clara, CA), respectively. Libraries were prepared using TruSeq Stranded mRNA (PolyA+) kit (Illumina, San Diego, CA) and sequenced by Illumina NextSeq 500. The read length was 75bp with 30-40M reads per sample. FastQC (v0.11.3) was performed to assess data quality. TopHat2 (v2.0.9) aligned the reads to the mouse reference genome (Mus musculus UCSC mm10) and to the Ensembl human reference genome (GRCh38.p13) using default parameters. Alignments were then converted to expression count data using HTseq (v0.6.1) with default union mode.\nAt time of import, last updated (by provider) on: Dec 31 2020\n\nContributors: ; [George Tseng, Colleen McClung, Wei Zong, Ryan Logan]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Analysis of gene-expression changes in treatment responders vs non-responders to two different treatments among subjectrs participating in LiTMUS. Results provide information on pathways that may be involved in the clinical response to Lithium in patients with bipolar disorder.\nLast Updated (by provider): Apr 01 2013\nContributors:  Robert Beech</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Total RNA sequecing for human induced pluripotent derived cerebral organoids from healthy controls and Bipolar disorder\nAt time of import, last updated (by provider) on: Apr 01 2020\n\nContributors: ; [Annie Kathuria, Rakesh Karmacharya]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Hippocampus of schizophrenic, bipolar, and control subjects. Analyzed from CEL files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7: Brain circuits are assembled from a large variety of morphologically and functionally diverse cell types. It is not known how the intermingled cell types of individual brain regions differ in their expressed genomes. Here we describe an atlas of cell type transcriptomes of the adult retina. We found that each adult cell type expresses a specific set of genes, including a unique set of transcription factors, forming a “barcode” for cell identity. Cell type transcriptomes carry enough information to categorize cells into corresponding morphological classes and types. Surprisingly, several barcode genes are eye disease-associated genes that we demonstrate to be specifically expressed not only in photoreceptors but also in particular retinal circuit elements such as inhibitory neurons as well as in retinal microglia. Our data suggest that distinct cell types of individual brain regions are characterized by marked differences in their expressed genomes. We assembled a library of 22 transgenic mouse lines in which each line had a group of retinal cells marked with fluorescent proteins. We built up the library with the purpose of having some mouse lines in which single retinal cell types and others in which combinations of types from a single class are labeled. The library had mouse lines with labeled cells representing each of the six retinal cell classes. Retinal cells were characterized by physiological recording and immunohistochemical staining. We isolated 200 fluorescent protein-labeled retinal cells (“cell groups”) from at least three different mice of each mouse line by fluorescence-activated cell sorting. The transcripts of each cell group of these biological triplicates were independently amplified in batches. Each batch contained an internal control cell group from the Arc-line.\nLast Updated (by provider): Jun 10 2014\nContributors:  Botond Roska Erik Cabuy Sandra Siegert</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   The goal of this experiment was to define gene expression patterns of thirteen mouse retinal neuron subsets, labeled by expression of fluorescent proteins in transgenic mice.\nLast Updated (by provider): Sep 23 2013\nContributors:  Jeremy N Kay Joshua R Sanes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  We used RNA sequencing to identify candidate regulators of interactions between photoreceptor axons and bipolar cell (BCs) dendrites in developing mouse retina. We chose three time points: P7, just after the OPL forms and synaptogenesis with BCs begins; P13, as synaptogenesis nears completion and sublamination begins; and P30, when the OPL is mature. We purified cone and rod photoreceptors separately by fluorescence activated cell sorting (FACS) using transgene markers: Rhoicre;Ai9 for rods and HRGPcre;Ai9 for cones. We purified ON BCs, which include ON cone bipolars plus rod bipolars using Grm6:GFP. As appropriate transgenic lines to separate RBCs from CBCs were not available, we performed RNAseq on cells from Grm6:GFP mice that were fixed and immunostained prior to FACS, allowing us to purify RBCs (GFP+PKC+) and CBCs (GFP+PKC-) from the same retinas. As PKC is not highly expressed at P7, profiling of developing rod bipolars separate from developing ON cone bipolars was restricted to P13.\nLast Updated (by provider): Jun 25 2018\nContributors:  Elizabeth Z Sanchez Lawrence S Zipursky Yerbol Z Kurmangaliyev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Fibroblasts and lymphoblastoid cells (LCLs) are the most widely used cells in genetic, genomic, and transcriptomic studies in relation to human diseases. Examining the gene expression patterns in these two cell types will provide valuable information regarding the validity of using them to study gene expression related to various human diseases. Fibroblasts and LCLs from four members of the Old Order Amish family 884 were purchased from Coriell cell repositories (Coriell Institute for Medical Research, Camden, NJ). We used microarrays to profile the patterns of gene expression in these eight cell lines. By employing the PennCNV algorithm to the Illumina HumanHap550 SNP genotyping data, we detected 13 Copy Number Variants (CNV) that exist in these four individuals. CNV-expression association analysis revealed that seven of these 13 CNVs were associated with the expression of genes within or near (&lt;2Mb sweep) these CNVs at a nominal regression P value of 0.05.\nLast Updated (by provider): Feb 20 2009\nContributors:  Maja Bucan Shuzhang Yang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   98 samples of individuals from 3 different diagnostic groups: bipolar, schizophrenia, and controls. Samples taken from the DLPFC Broadmann area 46.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Microarray experiments were performed using FAC-sorted young photoreceptors to analyze their transcriptome in comparison to remaining retinal cells at same developmental stage and retinal progenitors.\nLast Updated (by provider): Apr 02 2018\nContributors:  Marius Ader Kai Postel</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Study the role of the LIM-homeodomain transcription factor LHX4 in the development of retinal bipolar cell subtypes\nAt time of import, last updated (by provider) on: Sep 30 2020\n\nContributors: ; [Lin Gan, Xuhui Dong]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               mRNA cancer cell line profiles\nLast Updated (by provider): Oct 25 2013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Bipolar disorder is a severe and heritable psychiatric disorder and affects up to 1% of the population worldwide. Lithium is recommended as first-line treatment for the maintenance treatment of bipolar-affective disorder in current guidelines, its molecular modes of action are however poorly understood. Cell models derived from bipolar patients could prove useful to gain more insight in the molecular mechanisms of bipolar disorder and the common pharmacological treatments. As primary neuronal cell lines cannot be easily derived from patients, peripheral cell models should be evaluated in their usefulness to study pathomechanisms and the mode of action of medication as well as in regard to develop biomarkers for diagnosis and treatment response.\nAt time of import, last updated (by provider) on: Mar 25 2019\n\nContributors: ; [Sarah Kittel-Schneider, Max Hilscher, Andreas Reif, Claus J Scholz]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              We determined the transcriptomes of postmitotic cone photoreceptors in the central region of the mouse retina every day between birth (P0) and eye opening (P12). At each postnatal day we isolated GFP-labeled cells from three different Chrnb4-GFP mice (biological triplicates) using fluorescence-activated cell sorting. We then acquired the transcriptomes of the sorted cones using next generation RNA sequencing. Our data set contained 39 transcriptomes.\nLast Updated (by provider): Jul 03 2018\nContributors:  Janine M Daum Michael B Stadler Özkan Keles Botond Roska Hubertus Kohler</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                The goals of this study are to examine responses to inflammation in astrocytes from  induced pluripotent stem cells derived from healthy controls and bipolar disorder patients. We examine the transcriptomic inflmmatory signature of generated astrocytes following Il1Beta exposure in BD vs. control Results: BD-patient astrocytes show a unique inflammatory response with differentially regulated genes.\nAt time of import, last updated (by provider) on: Mar 19 2021\n\nContributors: ; [Maxim N Shokhirev, Fred Gage, Krishna Vadodaria, Carol Marchetto]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Analysis of gene-expression changes in depressed subjects with bipolar disorder compared to healthy controls. Results provide information on pathways that may be involved in the pathogenesis of bipolar depression.\nLast Updated (by provider): Aug 27 2010\nContributors:  Robert D Beech</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        To identify genes dysregulated in bipolar disorder (BD1) we carried out global gene expression profiling using whole-genome microarrays. To minimize genetic variation in gene expression levels between cases and controls we compared expression profiles in lymphoblastoid cell lines from monozygotic twin pairs discordant for the disease. We identified 82 genes that were differentially expressed by ? 1.3-fold in 3 BD1 cases compared to their co-twins, and which were statistically (p ? 0.05) differentially expressed between the groups of BD1 cases and controls. Using qRT-PCR we confirmed the differential expression of some of these genes, including: KCNK1, MAL, PFN2, TCF7, PGK1, and PI4KCB, in at least 2 of the twin pairs. In contrast to the findings of a previous study by Kakiuchi and colleagues with similar discordant BD1 twin design1 our data do not support the dysregulation of XBP1 and HSPA5. From pathway and gene ontology analysis we identified up-regulation of the WNT signalling pathway and the biological process of apoptosis. The differentially regulated genes and pathways identified in this study may provide insights into the biology of BD1.\nLast Updated (by provider): Jun 20 2007\nContributors:  Louisa Windus Nicholas Matigian Bryan Mowry Cheryl Filippich John McGrath Heather Smith Nicholas Hayward Christos Pantelis</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    experiment.description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;         &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:               FALSE                 &lt;NA&gt;                &lt;NA&gt;           &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.troubled experiment.accession experiment.database experiment.URI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated  experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      &lt;int&gt;                 &lt;POSc&gt;                  &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                     23    2023-12-20 20:07:23           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                     93    2022-08-30 23:36:54           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                     24    2023-12-17 14:05:13           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                    120    2023-12-19 20:24:26           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                     22    2023-12-17 10:03:11           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                     23    2023-09-08 07:38:04           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                     78    2023-12-19 02:38:44           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                     26    2023-12-19 05:21:57           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                     22    2023-12-22 09:47:02           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                      8    2023-12-16 05:30:07    SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                     98    2023-06-08 22:47:42           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                      9    2023-12-18 20:37:33 BATCH_CORRECTED_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                      6    2023-12-17 09:01:07           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                      6    2023-12-17 09:06:10           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                    206    2024-02-21 01:47:58    BATCH_EFFECT_FAILURE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                     12    2023-12-20 19:03:56           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                     39    2023-12-22 09:37:54           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                     33    2023-12-17 01:36:32           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                     35    2023-12-18 12:17:06           NO_BATCH_INFO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                      6    2023-12-20 21:45:32    SINGLE_BATCH_SUCCESS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     experiment.sampleCount experiment.lastUpdated  experiment.batchEffect</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;lgcl&gt;              &lt;int&gt;            &lt;int&gt;        &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                  NA                 NA               NA           NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.batchCorrected geeq.batchConfound geeq.batchEffect geeq.rawData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.name taxon.scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;num&gt;       &lt;num&gt;     &lt;char&gt;           &lt;char&gt;    &lt;int&gt;      &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:          NA          NA      mouse     Mus musculus        2      10090</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:          NA          NA      human     Homo sapiens        1       9606</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     geeq.qScore geeq.sScore taxon.name taxon.scientific taxon.ID taxon.NCBI</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.name taxon.database.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  &lt;char&gt;             &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  1:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  3:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  4:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  5:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  6:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  7:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  8:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  9:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17:                mm10                81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20:                hg38                87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     taxon.database.name taxon.database.ID</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

