<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Accessing curated gene expression data with gemma.R • gemma.R</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Accessing curated gene expression data with gemma.R">
<meta property="og:description" content="gemma.R">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Accessing curated gene expression data with
gemma.R</h1>
                        <h4 data-toc-skip class="author">Javier
Castillo-Arnemann</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia,
Vancouver,
Canada<br><h4 data-toc-skip class="author">Jordan
Sicherman</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia,
Vancouver,
Canada<br><h4 data-toc-skip class="author">B.
Ogan Mancarci</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia,
Vancouver, Canada<br><small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/vignettes/gemma.R.Rmd" class="external-link"><code>vignettes/gemma.R.Rmd</code></a></small>
      <div class="hidden name"><code>gemma.R.Rmd</code></div>

    </address>
</address>
</address>
</div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pavlidislab.github.io/gemma.R/" class="external-link">gemma.R</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggrepel.slowkow.com/" class="external-link">ggrepel</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="about-gemma">About Gemma<a class="anchor" aria-label="anchor" href="#about-gemma"></a>
</h2>
<p><a href="https://gemma.msl.ubc.ca/" class="external-link">Gemma</a> is a web site, database
and a set of tools for the meta-analysis, re-use and sharing of genomics
data, currently primarily targeted at the analysis of gene expression
profiles. Gemma contains data from thousands of public studies,
referencing thousands of published papers. Every dataset in Gemma has
passed a rigorous curation process that re-annotates the expression
platform at the sequence level, which allows for more consistent
cross-platform comparisons and meta-analyses.</p>
<p>For detailed information on the curation process, read this <a href="https://pavlidislab.github.io/Gemma/curation.html" class="external-link">page</a> or the
latest <a href="https://academic.oup.com/database/article/doi/10.1093/database/baab006/6143045" class="external-link">publication</a>.</p>
</div>
<div class="section level2">
<h2 id="package-cheat-sheet">Package cheat sheet<a class="anchor" aria-label="anchor" href="#package-cheat-sheet"></a>
</h2>
<p><img src="cheat_sheet.png" style="width:100.0%"></p>
</div>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<div class="section level3">
<h3 id="bioconductor">Bioconductor<a class="anchor" aria-label="anchor" href="#bioconductor"></a>
</h3>
<p>You can install <code>gemma.R</code> through <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> with the following
code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"gemma.R"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="searching-for-datasets-of-interest-in-gemma">Searching for datasets of interest in Gemma<a class="anchor" aria-label="anchor" href="#searching-for-datasets-of-interest-in-gemma"></a>
</h2>
<p>Using the <a href="https://pavlidislab.github.io/gemma.R/reference/get_datasets.html" class="external-link"><code>get_datasets</code></a>
function, datasets fitting various criteria can be accessed.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># accessing all mouse and human datasets</span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'mouse'</span>,<span class="st">'human'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>   experiment.shortName
                 &lt;char&gt;
1:              GSE2018
2:              GSE4523
3:              GSE4036
4:              GSE4034
5:              GSE2866
6:              GSE3253
                                                                                                            experiment.name
                                                                                                                     &lt;char&gt;
1:                                                                                              Human Lung Transplant - BAL
2:                                                                   Expression Studies of Melanotransferrin in Mouse Brain
3:                                                                                                  perro-affy-human-186940
4:                                                                                                  palme-affy-mouse-198967
5:                                                                                                Donarum-3R01NS040270-03S1
6: Exaggerated neuroinflammation and sickness behavior in aged mice after activation of the peripheral innate immune system
   taxon.name
       &lt;char&gt;
1:      human
2:      mouse
3:      human
4:      mouse
5:      mouse
6:      mouse</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># accessing human datasets with the word "bipolar"</span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>query <span class="op">=</span> <span class="st">'bipolar'</span>,taxa <span class="op">=</span> <span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>     <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>   experiment.shortName
                 &lt;char&gt;
1:             GSE67645
2:   McLean Hippocampus
3:              byne-cc
4:             GSE45484
5:            GSE134497
6:            GSE160761
                                                                                                                                                                                     experiment.name
                                                                                                                                                                                              &lt;char&gt;
1: Transcriptome dynamics of developing photoreceptors in 3-D retina cultures recapitulates temporal sequence of human cone and rod differentiation revealing cell surface markers and gene networks
2:                                                                                                                                                                                McLean Hippocampus
3:                                                                                                                                              Corpus Callosum data from Stanley collection samples
4:                                        Gene-expression differences in peripheral blood between lithium responders and non-responders in the “Lithium Treatment -Moderate dose Use Study” (LiTMUS)
5:                                                                                                                      Total RNA sequecing for human induced pluripotent derived cerebral organoids
6:                                                            RNA sequencing  in human  iPSCs derived from  bipolar  patients  to identify important therapeutic molecular targets of Valproate(VPA)
   taxon.name
       &lt;char&gt;
1:      human
2:      human
3:      human
4:      human
5:      human
6:      human</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># access human datasets that were annotated with the ontology term for the</span></span>
<span><span class="co"># bipolar disorder</span></span>
<span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>taxa <span class="op">=</span><span class="st">'human'</span>, </span>
<span>             uris <span class="op">=</span> <span class="st">'http://purl.obolibrary.org/obo/MONDO_0004985'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>   experiment.shortName
                 &lt;char&gt;
1:              GSE5389
2:              GSE5388
3:              GSE7036
4:   McLean Hippocampus
5:           McLean_PFC
6:     stanley_feinberg
                                                                                                           experiment.name
                                                                                                                    &lt;char&gt;
1:           Adult postmortem brain tissue (orbitofrontal cortex) from subjects with bipolar disorder and healthy controls
2: Adult postmortem brain tissue (dorsolateral prefrontal cortex) from subjects with bipolar disorder and healthy controls
3:                                               Expression profiling in monozygotic twins discordant for bipolar disorder
4:                                                                                                      McLean Hippocampus
5:                                                                                                              McLean_PFC
6:                                                                     Stanley consortium collection Cerebellum - Feinberg
   taxon.name
       &lt;char&gt;
1:      human
2:      human
3:      human
4:      human
5:      human
6:      human</code></pre>
<p>Note that a single call of these functions will only return 20
results by default and a 100 results maximum, controlled by the
<code>limit</code> argument. In order to get all available results, use
<code>get_all_pages</code> function on the output of the function</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>taxa <span class="op">=</span> <span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/get_all_pages.html">get_all_pages</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>   experiment.shortName
                 &lt;char&gt;
1:              GSE2018
2:              GSE4036
3:              GSE3489
4:              GSE1923
5:               GSE361
6:               GSE492
                                                                          experiment.name
                                                                                   &lt;char&gt;
1:                                                            Human Lung Transplant - BAL
2:                                                                perro-affy-human-186940
3: Patterns of gene dysregulation in the frontal cortex of patients with HIV encephalitis
4: Identification of PDGF-dependent patterns of gene expression in U87 glioblastoma cells
5:                                                   Mammary epithelial cell transduction
6:                               Effect of prostaglandin analogs on aqueous humor outflow
   taxon.name
       &lt;char&gt;
1:      human
2:      human
3:      human
4:      human
5:      human
6:      human</code></pre>
<p>See <a href="#larger-queries">larger queries</a> section for more
details. To keep this vignette simpler we will keep using the first 20
results returned by default in examples below.</p>
<p>The <code>filter</code> parameter of get_datasets allows filtering
for datasets with specific properties. A list of such properties can be
accessed using <code>filter_properties</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filter_properties.html">filter_properties</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">dataset</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>                                      properties    type description
                                          &lt;char&gt;  &lt;char&gt;      &lt;char&gt;
1:                           accession.accession  string        &lt;NA&gt;
2:                    accession.accessionVersion  string        &lt;NA&gt;
3:             accession.externalDatabase.ftpUri  string        &lt;NA&gt;
4:                 accession.externalDatabase.id integer        &lt;NA&gt;
5:        accession.externalDatabase.lastUpdated  string        &lt;NA&gt;
6: accession.externalDatabase.localInstallDbName  string        &lt;NA&gt;</code></pre>
<p>These properties can be used together to fine tune your results</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all datasets with more than 4 samples annotated for any disease</span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>filter <span class="op">=</span> <span class="st">'bioAssayCount &gt; 4 and allCharacteristics.category = disease'</span><span class="op">)</span></span></code></pre></div>
<pre><code>    experiment.shortName
                  &lt;char&gt;
 1:              GSE2018
 2:              GSE4036
 3:              GSE2866
 4:              GSE2426
 5:              GSE2867
 6:              GSE3489
 7:              GSE1837
 8:              GSE2178
 9:               GSE857
10:              GSE1294
11:              GSE1070
12:              GSE1069
13:              GSE1068
14:                GSE63
15:               GSE466
16:               GSE485
17:               GSE460
18:               GSE492
19:               GSE422
20:               GSE833
    experiment.shortName
                                                                           experiment.name
                                                                                    &lt;char&gt;
 1:                                                            Human Lung Transplant - BAL
 2:                                                                perro-affy-human-186940
 3:                                                              Donarum-3R01NS040270-03S1
 4:                                                Pre-Neoplastic Stage of Medulloblastoma
 5:                                                                 Zoghbi-5R01NS027699-14
 6: Patterns of gene dysregulation in the frontal cortex of patients with HIV encephalitis
 7:            Comparison of  SNc and VTA dopaminergic neurons. Greene-5P50NS038399-050001
 8:                                                          Mouse Expression DB 2001 CHMC
 9:                                                             Triple treatment R6/2 mice
10:                   Expression profile of genes in normal and Down syndrome mouse brains
11:                                                         2CdA intervention with PK11195
12:                                                                2.5 mg/kg 2CdA exposure
13:                                                            2CdA-induced microphthalmia
14:                                                   Brain after ischemia and antioxidant
15:                               mRNA expression in regenerated mdx mouse skeletal muscle
16:                                     Genetic basis of sensitivity to pulmonary fibrosis
17:                                                               Murine dermal burn wound
18:                               Effect of prostaglandin analogs on aqueous humor outflow
19:                                                          Epithelium transition profile
20:                                                          Amyotrophic lateral sclerosis
                                                                           experiment.name
    experiment.ID
            &lt;int&gt;
 1:             1
 2:             4
 3:             6
 4:            11
 5:            17
 6:            18
 7:            21
 8:            24
 9:            31
10:            32
11:            36
12:            37
13:            38
14:            47
15:            54
16:            58
17:            99
18:           100
19:           101
20:           106
    experiment.ID
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        experiment.description
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        &lt;char&gt;
 1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Bronchoalveolar lavage samples collected from lung transplant recipients.  Numeric portion of sample name is an arbitrary patient ID and AxBx number indicates the perivascular (A) and bronchiolar (B) scores from biopsies collected on the same day as the BAL fluid was collected.  Several patients have more than one sample in this series and can be determined by patient number followed by a lower case letter.  Acute rejection state is determined by the combined A and B score - specifically, a combined AB score of 2 or greater is considered an acute rejection.
 2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Our laboratory has developed the first mouse model overexpressing a RNA-binding protein, the ELAV-like protein HuD, in the CNS under the control of the CaMKinII alpha promoter. Initial behavioral characterization of the mice revealed that they had significant learning deficits together with abnormalities in prepulse inhibition (PPI). At the molecular level, we found that the expression of the growth-associated protein GAP-43, one of the targets of HuD, was increased in the hippocampus of HuD transgenic mice. To characterize these mice further and to evaluate the utility of these animals in understanding human diseases, we propose to use DNA microarray methods. To test our hypothesis we propose 3 specific aims: 1) To characterize the pattern of gene expression in the hippocampus of HuD overexpressor mice 2) To compare the pattern of gene expression in our mouse model with that in the hippocampus of rats prenatally exposed to alcohol (FAS model) and 3) To compare the pattern of gene expression in our mouse model with that shown in post-mortem tissues of patients with schizophrenia. In our previous protocols we examined the pattern of gene expression in our HuD transgenic mice and in rats prenaltally exposed to alcohol. A report by another group (Hakak et al, 2001) showed that three of the HuD targets were upregulated in the prefrontal cortex of patients with schizophrenia. To evaluate whether other target of HuD may be affected in this illness, in the current protocol, we want to compare the pattern of expression in our transgenic mice with in tissue from patients with schizophrenia Based on the behavioral and molecular properties of our HuD transgenic mice we hypothesize that these animals may be good models for the studying the basis of learning disabilities and of diseases that show deficits in PPI such as fetal alcohol syndrome and schizophrenia. All 28 samples are derived from cerebellar tissues for patients with schizophrenia and matched controls.  The specimens were obtained from the Maryland Brain Collection according to NIH guidelines for confidentially and privacy. The protocol used in these studies was reviewed by our HRRC which found that our studies do not fall within the category of protocols monitored by the IRB (see attached letter form the HRRC). Specimens from 14 patients with a diagnosis of schizophrenia performed according to DSM-IV criteria and 14 sex-, age- and PMI-matched controls was included in this study. No differences were found between patients and control subjects in the average age (45±12 versus 43±10 years, p=0.86) or PMI (12±5 versus 16±6 hours, p=0.11). We will provide 28 samples containing 5 ug of RNA each in DEPC water (see validation of the quality of the RNA below). In addition, we include in our study animals treated with haloperidol as control for medication. These samples will be submitted in a separate protocol.\nDate GSE4036 Last Updated: Jan 13 2006\nContributors:  Nora I Perrone-Bizzozero\nIncludes GDS1917.\n Update date: Jun 14 2006.\n Dataset description GDS1917: Analysis of cortical samples corresponding to the crus I/VIIa area of the cerebellum from schizophrenia patients. A study indicates that targets of the RNA-binding ELAV-like protein HuD are overexpressed in the prefrontal cortex of patients with schizophrenia.
 3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Succinate semialdehyde dehydrogenase (SSADH) deficiency is a rare autosomal recessive disorder effecting approximately 350 people around the world. Patients suffering from SSADH deficiency experience language acquisition failure, memory deficiencies, autism, increased aggressive behaviors, and seizures. There is a chemical buildup of both gamma-aminobutyric acid (GABA) and gamma-hydroxybutyric acid (GHB) in the neurological system of these patients. The Aldh5a1-/- knock out mouse model of SSADH deficiency shows the same chemical imbalances as the human disease, with additional fatal tonic-clonic seizures at three weeks of age.  The elucidation of seizure causing pathways will facilitate treatment of seizure phenotypes in diseases with related epilepsy. Gene expression patterns within the hippocampus, cerebellum, and cortex of SSADH deficient mice (Aldh5a1-/- mice) will be compared to wild type mice at a time point immediately prior to fatal seizures. We hypothesis that the SSADH deficient mice experience a dysfunction of glutamate/GABA/ glutamine neurotransmitter cycle linked to astroglial metabolism and/or uptake of neuronally-released glutamate. The increased levels of GHB and GABA lead to down regulation of GABA-B-Receptor leading to seizures. The SSADH deficient phenotype may also be caused by ongoing oxidative damage and the pathological role of succinic semialdehyde. SSADH deficient mice (Aldh5a1-/- knock out) exhibit fatal seizures around three weeks of age. Mutant and wild type mice will be sacrificed between 17 and 19 days of life, and brain sections will be extracted and frozen (using a standard protocol). Hippocampus, cerebellum, and cortex from three mutant mice and three wild type mice will individually be expression profiled on the Affymetrix platform, giving a total of eighteen arrays. Comparative analysis will then be carried out, evaluating the transcript differences between mutant and wild type mice in each brain region.\nDate GSE2866 Last Updated: Jun 08 2006\nContributors:  E A Donarum\nIncludes GDS1745.\n Update date: Jun 08 2006.\n Dataset description GDS1745: Analysis of brain hippocampi, cerebella, and cortices of succinate semialdehyde dehydrogenase (SSADH)-deficient mutants at 3 weeks of age, when fatal seizures occur. Results indicate that SSADH deficiency results in the dysregulation of genes involved in myelin structure and compaction.
 4: SUMMARY  Medulloblastoma is the most common malignant brain tumor in children. It is thought to result from transformation of granule cell precursors (GCPs) in the developing cerebellum, but little is known about the early stages of the disease. Here we identify a pre-neoplastic stage of medulloblastoma in patched heterozygous mice, a model of the human disease. We show that pre-neoplastic cells are present in the majority of patched mutants, although only 16% of these mice develop tumors. Pre-neoplastic cells, like tumor cells, exhibit activation of the Sonic hedgehog pathway and constitutive proliferation. Importantly, they also lack expression of the wild-type patched allele, suggesting that loss of patched is an early event in tumorigenesis. While pre-neoplastic cells resemble GCPs and tumor cells in many respects, they have a distinct molecular signature. Genes that mark the pre-neoplastic stage include regulators of migration, apoptosis and differentiation, processes critical for normal development but previously unrecognized for their role in medulloblastoma. The identification and molecular characterization of pre-neoplastic cells provides insight into the early steps in medulloblastoma formation, and may yield important markers for early detection and therapy of this disease.  ANIMALS  Patched heterozygous mice (Goodrich et al., 1997) were obtained from Matthew Scott?s lab at Stanford, and maintained by breeding with 129X1/SvJ mice from Jackson Laboratories. Math1-GFP transgenic mice (Lumpkin et al., 2003) were provided by Jane Johnson at UT Southwestern Medical Center. Math1-GFP/patched+/- mice were generated by crossing patched heterozygotes with Math1-GFP mice, and then backcrossing to Math1-GFP mice three times before further analysis. All mice were maintained in the Cancer Center Isolation Facility at Duke University Medical Center.  ISOLATION OF CELLS  Granule cell precursors (GCPs) were isolated from 7-day-old (P7) patched+/- mice; preneoplastic cells were obtained from 6 week-old patched mutants; and tumor cells were obtained from 10-25-week old patched mutants displaying physical and behavioral signs of medulloblastoma. Cells were isolated from each source using a protocol described in (Wechsler- Reya and Scott, 1999). Briefly, cerebella were digested in solution containing 10 U/ml papain Oliver et al. Pre-Neoplastic Stage of Medulloblastoma - 8 - (Worthington, Lakewood, NJ) and 250 U/ml DNase (Sigma) and triturated to obtain a cell suspension. This suspension was centrifuged through a step gradient of 35% and 65% Percoll (Amersham Biosciences, Piscataway, NJ), and cells were harvested from the 35%-65% interface. Cells were resuspended in serum-free culture medium consisting of Neurobasal containing B27 supplement, sodium pyruvate, L-glutamine, and penicillin/streptomycin (all from Invitrogen, Carlsbad, California) and counted on a hemacytometer. Cells used for RNA isolation were centrifuged and flash frozen in liquid nitrogen. For proliferation assays or immunostaining, cells were plated on poly-D-lysine (PDL)-coated tissue culture vessels and incubated in serum-free culture medium.  MICROARRAY HYBRIDIZATION AND ANALYSIS  RNA from GCPs, pre-neoplastic cells and tumor cells (isolated as described above, but not FACS-sorted) and from normal adult cerebellum (not dissociated) was converted to cDNA using the Superscript Choice cDNA kit (Invitrogen) and a T7-dT(24) primer (Genset/Proligo, Boulder, CO). cRNA was generated using a T7-transcription/labeling kit from Enzo Life Sciences and hybridized to Affymetrix U74Av2 chips (Affymetrix, Santa Clara, CA). Chips were scanned, and hybridization data were acquired using Affymetrix Suite 5.0 software. Affymetrix CEL files were normalized and quantified using Bioconductor software with the gcRMA model to quantify gene expression levels (Gentleman and Carey, 2002). Unsupervised principal components analysis (PCA) was used to identify the relationships among normal adult cerebellum, GCPs, pre-neoplastic cells and tumor cells based on expression profiles. To identify genes that were differentially expressed among GCPs, pre-neoplastic cells and tumor cells, supervised analysis was carried out. A gene-by-gene ANOVA model with three groups (GCP, pre-neoplastic, tumor) was used to fit the log2-transformed intensities. To correct for multiple comparisons, the nominal p-value was adjusted using the false discovery rate (Benjamini and Hochberg, 1995). Genes were considered to be differentially expressed if they satisfied all of the following criteria: a difference in expression greater than 1.9-fold between any two groups, a maximum absolute intensity difference larger than 32 units, and an adjusted pvalue &lt; 0.01. There were 118 genes that met these criteria. The identities of differentially expressed genes were verified by integrating data from the Affymetrix and Unigene databases. Gene functions were determined using information from Gene Ontology, Unigene, LocusLink and PubMed databases. Clustering was performed with Cluster and Treeview (Eisen et al., 1998). All statistical analysis was performed using R-1.7 software (Dalgaard, 2002). Results were visualized with Spotfire 6.0 (Somerville, MA).  CORRESPONDING AUTHOR  Robert J. Wechsler-Reya* Dept. of Pharmacology &amp; Cancer Biology, Duke University Medical Center Box 3813, Durham, NC 27710 Phone: 919-613-8754. Fax: 919-668-3556. Email: rw.reya@duke.edu.\nDate GSE2426 Last Updated: Oct 28 2005\nContributors:  R J Wechsler-Reya Trang T Huynh Simon M Lin Jonathan F Wells Tracy A Read Anriada Mehmeti Jessica D Kessler Robert J Wechsler-Reya Trudy G Oliver\nIncludes GDS1110.\n Update date: May 09 2005.\n Dataset description GDS1110: Analysis of granule cell precursors, pre-neoplastic cells, and tumor cells obtained from 7 day old, 6 week old, and 10 to 25 week old patched heterozygous animals, respectively. Results provide insight into the mechanisms of pathogenesis of medulloblastoma at the early stage.
 5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        A number of human neurodegenerative diseases result from the expansion of a glutamine repeat within the disease-causing protein. Spinocerebellar ataxia type1 is one such disease, caused by expansion of a polyglutamine tract in the novel protein ataxin-1. To faithfully model SCA1 in the mouse, we generated  knock-in mice carrying 154 CAG repeats in the mouse Sca1 locus. These mice reproduced many aspects of the human disease. Despite ubiquitous expression of the mutant protein, they developed slowly progressive selective neurodegeneration , which is most distinct in Purkinje cells and spinal cord. Alterations in gene expression have been proposed to be involved in the pathogenesis of polyglutamine disease including SCA1, but the changes of gene expression in an authentic disease model have not been characterized. We believe that knowledge of these genes will give insight into the pathophysiology of SCA1 and may ultimately be relevant to the treatment of SCA1. We will determine gene expression patterns in the cerebellum and forebrain at three different time points. We propose that the genes whose expression is affected  by mutant ataxin-1 expression are effectors for neuronal dysfunction and neuronal degeneration in the knock-in mice. We also hypothesize the difference in the expression levels of these genes accounts for selective vulnerability of neurons. We will examine three time points: 4 weeks, 11 weeks, and 20 weeks of age. We have shown that the mice developed motor incoordination as revealed  by rotating rod test as early as 5 weeks of age but it was not until 10 weeks that they start showing clear  neurological  phenotype such as clasping.  At each point, we will compare the pattern between the mutant and wild-type littermate. Animals will be prepared and sacrificed by a standard procedure. Cerebellum and spinal cord are the most affected areas whereas forebrain shows less neurodegeneration. Tissue will be rapidly dissected from cerebellum and forebrain, frozen in liquid nitrogen, and stored at ?80 C until analysis. Tissues will be sent to the centers (as opposed to RNA). We will be providing 9 tissue samples from three litters for each time point to mitigate any expression differences resulting from subtle differences of procedure or environment where the mice grow.\nDate GSE2867 Last Updated: Jul 06 2005\nContributors:  H Zoghbi\nIncludes GDS1756.\n Update date: May 02 2006.\n Dataset description GDS1756: Analysis of cerebellum and forebrain tissue of knock-in mice carrying 154 CAG repeats in the spinocerebellar ataxia type 1 (SCA1) locus at 4 and 12 weeks of age. Results provide insight into the pathophysiology of the neurodegenerative disease SCA1.
 6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             The neurodegenerative process in HIV encephalitis (HIVE) is associated with extensive damage to the dendritic and synaptic structure that often leads to cognitive impairment. Several mechanisms might be at play, including release of neurotoxins, oxidative stress and decreased activity of neurotrophic factors. Furthermore, HIV-mediated dysregulation of genes involved in neuronal maintenance might play an important role. For this purpose, cRNA was prepared from the brains of 17 AIDS patients for analysis with the Affymetrix Human U95Av2 GeneChip and analyzed with the GeneSpring Expression Analysis Software. Out of 12,625 genes analyzed, 74 were downregulated and 59 were upregulated compared to controls. Initial alternative analysis of RNA was performed by ribonuclease protection assay (RPA). In cases with HIVE, downregulated genes included neuronal molecules involved in synaptic plasticity and transmission (ion channels, synaptogyrin, synapsin II), cell cycle (p35, p39, CDC-L2, CDC42, PAK1) and signaling molecules (PI3K, Ras-Raf-MEK1), transcription factors and cytoskeletal components (MAP-1B, MAP-2, tubulin, adducin-2). Upregulated genes included those involved in neuroimmune (IgG, MHC, ?2microglobulin) and anti-viral responses (interferon-inducible molecules), transcription (STAT1, OLIG2, Pax-6) and signaling modulation (MEK3, EphB1) of the cytoskeleton (myosin, aduccin-3, radixin, dystrobrevin). Taken together, this study suggests that HIV proteins released from infected macrophages might not only induce a neuroinflammatory response, but also may promote neurodegeneration by interfering with neuronal transcription of genes involved in regulating signaling and cytoskeletal molecules important in maintaining synapto-dendritic functioning and integrity.\nDate GSE3489 Last Updated: Mar 07 2006\nContributors:  Eleanor S Roberts Dianne Langford Anthony Adame Edward Rockenstein Leslie Crews Howard S Fox Eliezer Masiliah Ian Everall\nIncludes GDS1726.\n Update date: May 30 2006.\n Dataset description GDS1726: Analysis of brain frontal cortex of HIV-seropositive patients with HIV encephalitis (HIVE). HIVE affects &gt;40% of AIDS patients, promoting neurodegeneration and cognitive impairment. Results suggest HIV-mediated dysregulation of genes involved in neuronal maintenance might play an important role.
 7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     The cardinal clinical features of Parkinson's disease (PD) (rigidity, rest tremor, bradykinesia, and postural instability) result from selective loss of midbrain dopaminergic neurons.  More specifically, dopaminergic neurons in the substantia nigra pars compacta (SNc) are much more susceptible to damage than the adjacent dopaminergic neurons in the ventral tegmental area (VTA).  This dichotomy is not only seen in human Parkinsons disease, but also in many animal models of PD, including administration of the mitochondrial toxin rotenone to rats, which replicates many of the behavioral and neuropathological features of PD.  The factors underlying this selective vulnerability are unknown, but could be related to differences in neuronal circuitry, differences in glial support, or intrinsic differences between the neuronal populations of the two regions.  Elucidation of these factors may lead to a greater understanding of the pathogenesis and treatment of Parkinson's disease. We will determine gene expression profiles of untreated rat SNc and VTA dopaminergic neurons using laser capture microscopy to obtain region-specific neuronal mRNA. There are intrinsic differences in gene expression between dopaminergic neurons in the rat SNc and VTA that result in greater susceptibility of SNc neurons to degeneration in experimental parkinsonism.  These differences may be related to dopamine metabolism, oxidative metabolism and stress, protein aggregation, or other unforseen pathways. We will compare gene expression profiles between SNc and VTA dopaminergic neurons in normal rats.  No treatment or time points will be studied in this experiment.  Animals will be anesthetized, sacrificed by decapitation, and brains frozen on dry ice.  Frozen sections will be collected onto glass microscope slides and rapidly immunostained for tyrosine hydroxylase to identify dopaminergic neurons.  SNc and VTA neurons (approx. 200 per sample) will be isolated using laser capture microscopy.  Total RNA will be extracted and poly-A RNA will be amplified using a modified Eberwine protocol.  aRNA will be sent to the centers for labeling and hybridization to Affymetrix rat U34A arrays.  We have confirmed with the center that our aRNA protocol is compatible with the centers amplification protocols; in fact, it is essentially identical.  We will be providing a two-round amplification product to the center for labeling and hybridization.  We recognize that using RNA after three rounds of amplification  may decrease sensitivity for low copy number transcripts, but favor this approach versus pooling our samples (which are inherently paired) at this point.  We have discussed this point in detail with the center. SNc and VTA samples from eight animals (16 samples total) will be provided to mitigate differences specific to individual animals.  With the assisatnce of the center, paired t-tests will be used to determine differential expression between the two regions.  Permutational t-test analysis and/or Benjamini and Hochberg analysis of expression ratios will be used to protect against multiple comparisons.  Selected differentially expressed genes will be validated on separate tissue samples using quantitative RT-PCR or in situ hybridization.\nDate GSE1837 Last Updated: Feb 02 2006\nContributors:  James G Greene\nIncludes GDS1641.\n Update date: Jun 08 2006.\n Dataset description GDS1641: Analysis of dopaminergic neurons of the substantia nigra pars compacta (SNc) and the ventral tegmental area (VTA) from normal animals. SNc dopaminergic neurons are more susceptible to degeneration in Parkinson's disease than VTA dopaminergic neurons.
 8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Expression profiles of different mouse tissue samples profiles.\nDate GSE2178 Last Updated: May 29 2005\nContributors:  Michael Mucenski Bruce J Aronow Belinda Peach Mitchell Cohen Amy Moseley Susan E Waltz John Maggio Sandra Degen Steve Potter Thomas Doetschman Chris Erwin Joanna A Groden James Lessard Linda Parysek R Hirsch MaryBeth Genter Jianhua Zhang Kathleen Anderson Jonathan D Katz John Dedman Michael Lehman Jorge A Bezerra Michael D Bates Ming Xu Dan Wiginton Jeff A Molkentin\nIncludes GDS1322.\n Update date: Nov 09 2005.\n Dataset description GDS1322: Gene expression profiles across a wide variety of tissue samples. The tissues examined include nervous system, skeletal muscle, male and female reproductive organs, gastrointestinal tract, as well as organs in a developmental context including lung, liver, kidney, and heart.
 9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        Huntington?s disease is a genetic disease caused by a single mutation. It is characterised by progressive movement, emotional and cognitive deficits. R6/2 transgenic mice carrying the Huntington?s disease mutation have a progressive neurological phenotype, including deterioration in cognitive function. The mechanism underlying the cognitive deficits in R6/2 mice is unknown, but dysregulated gene expression, reduced neurotransmitter levels and abnormal synaptic function are present before the cognitive decline becomes pronounced. Our goal here was to ameliorate the cognitive phenotype in R6/2 mice using a combination drug therapy (tacrine, moclobemide and creatine) aimed boosting neurotransmitter levels in the brain. Treatment from 5 weeks of age prevented deterioration in two different cognitive tasks until at least 12 weeks. However, motor deterioration continued unabated. Microarray analysis of global gene expression revealed that many genes significantly up- or down-regulated in untreated R6/2 mice had returned towards normal levels after treatment. Thus dysregulated gene expression was reversed by the combination treatment in the R6/2 mice and probably underlies the observed improvements in cognitive function. Our study shows that cognitive decline caused by a genetic mutation can be slowed by a combination drug treatment, and gives hope that cognitive symptoms in HD can be treated.\nDate GSE857 Last Updated: Dec 21 2005\nContributors:  Angela K Hodges Lesley Jones Paul D Lewis A J Morton Stephen B Dunnett Mark J Hunt\nIncludes GDS717.\n Update date: Sep 09 2004.\n Dataset description GDS717: Whole brain hemispheres from Huntingtons disease model R6/2 transgenic mice treated with creatine, tacrine and moclobemide from 5 weeks of age. Drug therapy was aimed at boosting neurotransmitter levels and improve cognitive functions.
10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Analyses of six Ts1Cje (Down syndrome) and six normal littermate (2N) mouse brains at postnatal day 0.\nDate GSE1294 Last Updated: May 29 2005\nContributors:  Kazuhiro Yamakawa\nIncludes GDS682.\n Update date: Nov 22 2004.\n Dataset description GDS682: Analyses of six Ts1Cje (Down syndrome) and six normal brains at postnatal day 0. Dosage-dependent over-expression of genes in trisomic region of Ts1Cje observed.\nIncludes GDS681.\n Update date: Nov 22 2004.\n Dataset description GDS681: Analyses of six Ts1Cje (Down syndrome) and six normal brains at postnatal day 0. Dosage-dependent over-expression of genes in trisomic region of Ts1Cje observed.
11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        In this time course pregnant CD-1 (outbred) dams were exposed to the 2-chloro analogue of 2'-deoxyadenosine (a metabolic toxin) on 8 d.p.c. The test dose of 2.5 mg/kg 2CdA was modeled for a 5% increased risk of microphthalmia in term fetuses. Exposed embryos were co-treated with PK11195, a nontoxic ligand of the mitochondrial peripheral-type benzodiazepine receptor (Bzrp) that effectively blocks p53 protein induction and developmental toxicity induced with 2CdA. Sampling intervals were anchored to the conditional biomarker (p53 protein induction) expressed in 2CdA-treated embryos between 3.0- and 4.5h post-exposure. Since p53 protein induction has been defined as a critical event in 2CdA-induced microphthalmia, these sampling intervals represent times before (3.0h), during (4.5h), and after (6.0h) this critical event. No p53 induction is observed in embryos co-treated with 2CdA and PK11195. All measurements were on RNA from the embryonic headfold.\nDate GSE1070 Last Updated: May 29 2005\nContributors:  Jeffrey H Charlap Thomas B Knudsen Amar V Singh Kimberly R Nemeth\nIncludes GDS632.\n Update date: Jul 14 2004.\n Dataset description GDS632: Temporal analysis of ocular teratogen 2-chloro-2'deoxyadenosine (2CdA) exposure in developing embryos, intervention with PK11195, and induction of microphthalmia (small eye syndrome). Embryonic headfold examined at 3, 4.5, and 6 hours.
12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                In this time course pregnant CD-1 (outbred) dams were exposed to the 2-chloro analogue of 2'-deoxyadenosine (a metabolic toxin) on 8 d.p.c. The test dose of 2.5 mg/kg 2CdA was modeled for a 5% increased risk of microphthalmia in term fetuses. Sampling intervals were anchored to the conditional biomarker (p53 protein induction) expressed in 2CdA-treated embryos between 3.0- and 4.5h post-exposure. Since p53 protein induction has been defined as a critical event in 2CdA-induced micropthalmia, these sampling intervals represent times before (3.0h), during (4.5h), and after (6.0h) this critical event. All measurements were on RNA from the embryonic headfold.\nDate GSE1069 Last Updated: May 29 2005\nContributors:  Kimberly R Nemeth Jeffrey H Charlap Amar V Singh Thomas B Knudsen\nIncludes GDS631.\n Update date: Jul 14 2004.\n Dataset description GDS631: Temporal analysis of ocular teratogen 2-chloro-2'deoxyadenosine (2CdA) exposure and induction of microphthalmia (small eye syndrome). Pregnant 8 days post coitum CD-1 dams exposed to 2.5 mg/kg 2CdA, and embryonic headfold examined at 3, 4.5, and 6 hours.
13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  In this dose series pregnant CD-1 (outbred) dams were exposed to the 2-chloro analogue of 2'-deoxyadenosine (a metabolic toxin) on 8 d.p.c. Test doses were developed from a teratological series for 2CdA-induced microphthalmia using BMDS modeling software from the US EPA. Benchmark doses for microarray analysis were: 5.0 mg/kg (25% risk), 2.5 mg/kg (5% risk), 1.25 mg/kg (NOAEL), and 0.625 mg/kg (subNOAEL). All measurements were on RNA from the embryo proper at 3.0h post-treatment.\nDate GSE1068 Last Updated: May 29 2005\nContributors:  Jeffrey H Charlap Kimberly R Nemeth Amar V Singh Thomas B Knudsen\nIncludes GDS630.\n Update date: Jul 13 2004.\n Dataset description GDS630: Dose response analysis of ocular teratogen 2-chloro-2'deoxyadenosine (2CdA) exposure and induction of microphthalmia (small eye syndrome). Pregnant 8 days post coitum CD-1 dams exposed to 5, 2.5, 1.25 or 0.625 mg/kg 2CdA, and embryos examined at 3 hours.
14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                A catalytic antioxidant (AEOL 10150) attenuates expression of inflammatory genes in stroke.  White and grey matter in the distribution on the middle cerebral artery 6 hours after administration of the saline/catalytic antioxidant AEOL 10150 and ischemia/reperfusion of the MCAO; pooled mRNA from 6 brains.\nDate GSE63 Last Updated: Jul 18 2005\nContributors:  David S Warner James D Crapo Robert D Pearlstein Huaxin Sheng Russell P Bowler Jan J Enghild\nIncludes GDS55.\n Update date: Apr 09 2003.\n Dataset description GDS55: C57BL/6J mice subjected to 60 min of right middle cerebral artery occlusion (MCAO). Catalytic antioxidant AEOL 10150 given 5 m after onset of reperfusion. After 6 h, MCA parenchyma harvested. Found AEOL 10150 attenuates immune response.
15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          Despite over 3,000 articles published on dystrophin in the last 15 years, the reasons underlying the progression of the human disease, differential muscle involvement, and disparate phenotypes in different species are not understood. The present experiment employed a screen of 12,488 mRNAs in 16-wk-old mouse mdx muscle at a time when the skeletal muscle is avoiding severe dystrophic pathophysiology, despite the absence of a functional dystrophin protein. A number of transcripts whose levels differed between the mdx and human Duchenne muscular dystrophy were noted. A fourfold decrease in myostatin mRNA in the mdx muscle was noted. Differential upregulation of actin-related protein 2/3 (subunit 4), beta-thymosin, calponin, mast cell chymase, and guanidinoacetate methyltransferase mRNA in the more benign mdx was also observed. Transcripts for oxidative and glycolytic enzymes in mdx muscle were not downregulated. These discrepancies could provide candidates for salvage pathways that maintain skeletal muscle integrity in the absence of a functional dystrophin protein in mdx skeletal muscle.\nDate GSE466 Last Updated: Jun 15 2005\nContributors:  P Zhao E P Hoffman B S Tseng J A Granchelli S E Gordon L C Folk F W Booth J S Pattison R W Madsen\nIncludes GDS236.\n Update date: Jun 26 2003.\n Dataset description GDS236: Examination of mdx mouse, Duchenne muscular dystrophy model. 16 week regenerating mdx muscle analyzed in search for salvage pathways that maintain skeletal muscle integrity in the absence of functional dystrophin protein.
16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         The murine pulmonary response is strain specific.  Susceptible strains such as C57BL6/J experience an inflammatory response followed by progressive lung disease.  Resistant strains such as Balb/c do not experience significant degrees of inflammation or fibrosis.  This study aims to determine the genetic basis of sensitivity differences between strains.\nDate GSE485 Last Updated: Jul 27 2006\nContributors:  David   Moller Y W Chen\nIncludes GDS251.\n Update date: Jun 27 2003.\n Dataset description GDS251: Determination of genetic basis of sensitivity to pulmonary fibrosis induced by bleomycin by comparing susceptible (C57BL6/J) and resistant (Balb/c) strains. C57BL6/J experience inflammatory response then progressive lung disease, Balb/c do not.
17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Investigation into murine dermal burn wound. Mouse thermal injury induced, and skin excised at 0 hours, 2 hours, 3 days and 14 days post-injury.\nDate GSE460 Last Updated: Jun 15 2005\nContributors:  David W Mozingo Robert J Feezor Lyle L Moldawer Heather N Paddock Juan C Varela Gregory S Schultz Henry V Baker\nIncludes GDS353.\n Update date: Jun 23 2003.\n Dataset description GDS353: Investigation into dermal burn wound healing. Thermal dermal injury induced. Skin excised at 0 hours, 2 hours, 3 days and 14 days post-injury.
18:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    The purpose of this study is to discover genes that might increase aqueous humor outflow when human ciliary muscle or human trabecular meshwork cells are treated with the prostaglandin analogues latanoprost free acid or prostaglandin F2alpha. Five tissue donors were pooled on each chip.\nDate GSE492 Last Updated: Jul 27 2006\nContributors:  Paul   Russell\nIncludes GDS359.\n Update date: Apr 14 2004.\n Dataset description GDS359: Glaucoma study investigating molecular basis of increase in aqueous humor outflow when human ciliary muscle or human trabecular meshwork cells are treated with prostaglandin analogs latanoprost free acid or prostaglandin F2alpha.
19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Transcriptional Profiling of the Transition from Normal Intestinal Epithelia to Adenomas and Carcinomas in the APC(Min/+) Mouse.  Samples used in analysis: * GSM6191-GSM6196 (WT): Ilea epithelial cells from C57/BL6 wild-type samples * GSM6197-GSM6201 (Adenoma): Epithelial cells from crypts of adenomas of APC(Min/+) mice * GSM6202-GSM6206 (Carcinoma): Epithelial cells from crypts of carcinomas of APC(Min/+) mice  Using a PixCell IIe instrument (Arcturus), ~30,000 laser firings per sample were used to collect cells of interest. RNA was extracted from captured cells by PicoPure (Arcturus) technique, followed by 2 rounds of RiboAmp amplification (Arcturus), with incorporation of biotinylated nucleotides (Enzo) in the IVT of round 2. All protocols were as per manufacturers instructions. 15ug of labeled cRNA from individual samples were hybridized to respective MG_U74Av2 chips (Affymetrix) and washed/stained using the standard EukGEWS2v4 protocol (Affymetrix). Chips were scanned and analyzed using MAS 5 (Affymetrix) and data scaled to TI=500.  Pairwise comparisons using the Mann-Whitney test were performed in DMT 3 (Affymetrix), comparing: * WT vs Adenoma (n=6 x n=5; 83.3% concordance) * WT vs Carcinoma (n=5 x n=5; 84% concordance) * Adenoma vs Carcinoma (n=5 x n=5; 84% concordance)  This resulted in the identification of differentially expressed transcripts. Fold changes were calculated from signal-log-ratios. Identified transcripts were clustered based on functional information which was publicly available at time of analysis, obtained through the NetAffx web portal (Affymetrix).\nDate GSE422 Last Updated: Aug 29 2005\nContributors:  Matthew W Feldman Linda S Gutierrez Nicholas F Paoni Francis J Castellino Victoria A Ploplis\nIncludes GDS389.\n Update date: Sep 03 2003.\n Dataset description GDS389: Examination of transition from normal intestinal epithelia to adenomas and carcinomas in the multiple intestinal neoplasia adenomatous polyposis coli mutant mouse, APC(Min/+).
20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Identification of amyotrophic lateral sclerosis (ALS) associated genes. Post mortem spinal cord grey matter from sporadic and familial ALS patients compared with controls.\nDate GSE833 Last Updated: Jun 28 2005\nContributors:  Steven R Gullans Robert H Brown Fernando Dangond\nIncludes GDS412.\n Update date: Nov 24 2003.\n Dataset description GDS412: Identification of amyotrophic lateral sclerosis (ALS) associated genes. Post mortem spinal cord grey matter from sporadic and familial ALS patients compared with controls. Attempt to identify mechanisms by which ALS destroys motor neurons.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        experiment.description
    experiment.troubled experiment.accession experiment.database
                 &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;
 1:               FALSE              GSE2018                 GEO
 2:               FALSE              GSE4036                 GEO
 3:               FALSE              GSE2866                 GEO
 4:               FALSE              GSE2426                 GEO
 5:               FALSE              GSE2867                 GEO
 6:               FALSE              GSE3489                 GEO
 7:               FALSE              GSE1837                 GEO
 8:               FALSE              GSE2178                 GEO
 9:               FALSE               GSE857                 GEO
10:               FALSE              GSE1294                 GEO
11:               FALSE              GSE1070                 GEO
12:               FALSE              GSE1069                 GEO
13:               FALSE              GSE1068                 GEO
14:               FALSE                GSE63                 GEO
15:               FALSE               GSE466                 GEO
16:               FALSE               GSE485                 GEO
17:               FALSE               GSE460                 GEO
18:               FALSE               GSE492                 GEO
19:               FALSE               GSE422                 GEO
20:               FALSE               GSE833                 GEO
    experiment.troubled experiment.accession experiment.database
                      experiment.URI experiment.sampleCount
                              &lt;char&gt;                  &lt;int&gt;
 1: http://www.ncbi.nlm.nih.gov/geo/                     34
 2: http://www.ncbi.nlm.nih.gov/geo/                     28
 3: http://www.ncbi.nlm.nih.gov/geo/                     18
 4: http://www.ncbi.nlm.nih.gov/geo/                     14
 5: http://www.ncbi.nlm.nih.gov/geo/                     20
 6: http://www.ncbi.nlm.nih.gov/geo/                     28
 7: http://www.ncbi.nlm.nih.gov/geo/                     15
 8: http://www.ncbi.nlm.nih.gov/geo/                    161
 9: http://www.ncbi.nlm.nih.gov/geo/                     13
10: http://www.ncbi.nlm.nih.gov/geo/                     24
11: http://www.ncbi.nlm.nih.gov/geo/                      6
12: http://www.ncbi.nlm.nih.gov/geo/                      6
13: http://www.ncbi.nlm.nih.gov/geo/                      8
14: http://www.ncbi.nlm.nih.gov/geo/                     17
15: http://www.ncbi.nlm.nih.gov/geo/                     10
16: http://www.ncbi.nlm.nih.gov/geo/                     24
17: http://www.ncbi.nlm.nih.gov/geo/                     12
18: http://www.ncbi.nlm.nih.gov/geo/                     12
19: http://www.ncbi.nlm.nih.gov/geo/                     16
20: http://www.ncbi.nlm.nih.gov/geo/                     11
                      experiment.URI experiment.sampleCount
    experiment.lastUpdated  experiment.batchEffect geeq.batchCorrected
                    &lt;POSc&gt;                  &lt;char&gt;              &lt;lgcl&gt;
 1:    2024-02-10 08:31:41 NO_BATCH_EFFECT_SUCCESS               FALSE
 2:    2023-12-19 13:29:05    SINGLE_BATCH_SUCCESS               FALSE
 3:    2023-12-18 19:37:20           NO_BATCH_INFO               FALSE
 4:    2023-12-18 12:59:41 NO_BATCH_EFFECT_SUCCESS               FALSE
 5:    2023-12-18 19:37:54    SINGLE_BATCH_SUCCESS               FALSE
 6:    2023-12-19 05:05:46           NO_BATCH_INFO               FALSE
 7:    2023-12-17 14:25:42 BATCH_CORRECTED_SUCCESS                TRUE
 8:    2023-12-18 06:10:33           NO_BATCH_INFO               FALSE
 9:    2023-12-21 07:45:02           NO_BATCH_INFO               FALSE
10:    2023-01-11 22:51:30 NO_BATCH_EFFECT_SUCCESS               FALSE
11:    2023-12-15 23:44:56           NO_BATCH_INFO               FALSE
12:    2023-12-15 23:41:43           NO_BATCH_INFO               FALSE
13:    2023-12-15 23:39:00           NO_BATCH_INFO               FALSE
14:    2023-12-20 16:00:48           NO_BATCH_INFO               FALSE
15:    2023-12-19 22:29:39    SINGLE_BATCH_SUCCESS               FALSE
16:    2023-12-20 01:32:02           NO_BATCH_INFO               FALSE
17:    2023-12-19 21:33:27           NO_BATCH_INFO               FALSE
18:    2023-12-20 02:28:26 BATCH_CORRECTED_SUCCESS                TRUE
19:    2023-12-19 16:09:04           NO_BATCH_INFO               FALSE
20:    2023-12-21 05:48:43           NO_BATCH_INFO               FALSE
    experiment.lastUpdated  experiment.batchEffect geeq.batchCorrected
    geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore
                 &lt;num&gt;            &lt;num&gt;        &lt;num&gt;       &lt;num&gt;       &lt;num&gt;
 1:                  1                1            1   0.9976341      0.8750
 2:                  1                1            1   0.9971246      0.7500
 3:                 -1                0            1   0.4273933      0.6875
 4:                  1                1            1   0.8558619      0.6875
 5:                  1                1            1   0.8469065      0.6250
 6:                  0                0           -1   0.4098798      0.5000
 7:                  1                1            1   0.9979230      0.3125
 8:                  0                0           -1  -0.2259327      0.0625
 9:                  0                0           -1   0.2789180      0.1875
10:                  1                0           -1   0.8548596      0.3125
11:                  0                0           -1  -0.1476548     -0.1625
12:                  0                0           -1  -0.1473691     -0.1625
13:                  0                0           -1  -0.1431355     -0.1625
14:                  0                0           -1   0.2694997      0.1875
15:                  1                1            1   0.7132509      0.6875
16:                 -1                0            1   0.4256551      0.5625
17:                  0                0           -1   0.2689840      0.4375
18:                  1                1            1   0.8561848      0.3750
19:                  0                0           -1   0.4071109      0.4375
20:                  0                0           -1  -0.0616038      0.0000
    geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore
    taxon.name  taxon.scientific taxon.ID taxon.NCBI taxon.database.name
        &lt;char&gt;            &lt;char&gt;    &lt;int&gt;      &lt;int&gt;              &lt;char&gt;
 1:      human      Homo sapiens        1       9606                hg38
 2:      human      Homo sapiens        1       9606                hg38
 3:      mouse      Mus musculus        2      10090                mm10
 4:      mouse      Mus musculus        2      10090                mm10
 5:      mouse      Mus musculus        2      10090                mm10
 6:      human      Homo sapiens        1       9606                hg38
 7:        rat Rattus norvegicus        3      10116                 rn6
 8:      mouse      Mus musculus        2      10090                mm10
 9:      mouse      Mus musculus        2      10090                mm10
10:      mouse      Mus musculus        2      10090                mm10
11:      mouse      Mus musculus        2      10090                mm10
12:      mouse      Mus musculus        2      10090                mm10
13:      mouse      Mus musculus        2      10090                mm10
14:      mouse      Mus musculus        2      10090                mm10
15:      mouse      Mus musculus        2      10090                mm10
16:      mouse      Mus musculus        2      10090                mm10
17:      mouse      Mus musculus        2      10090                mm10
18:      human      Homo sapiens        1       9606                hg38
19:      mouse      Mus musculus        2      10090                mm10
20:      human      Homo sapiens        1       9606                hg38
    taxon.name  taxon.scientific taxon.ID taxon.NCBI taxon.database.name
    taxon.database.ID
                &lt;int&gt;
 1:                87
 2:                87
 3:                81
 4:                81
 5:                81
 6:                87
 7:                86
 8:                81
 9:                81
10:                81
11:                81
12:                81
13:                81
14:                81
15:                81
16:                81
17:                81
18:                87
19:                81
20:                87
    taxon.database.ID</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all datasets with ontology terms for Alzheimer's disease and Parkinson's disease</span></span>
<span><span class="co"># this is equivalent to using the uris parameter</span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>filter <span class="op">=</span> <span class="st">'allCharacteristics.valueUri in (http://purl.obolibrary.org/obo/MONDO_0004975,http://purl.obolibrary.org/obo/MONDO_0005180</span></span>
<span><span class="st">)'</span><span class="op">)</span></span></code></pre></div>
<pre><code>    experiment.shortName
                  &lt;char&gt;
 1:              GSE1556
 2:              GSE1555
 3:              GSE4757
 4:              GSE6613
 5:             GSE12685
 6:             GSE14499
 7:             GSE10908
 8:              GSE1297
 9:          E-MEXP-2280
10:             GSE20219
11:             GSE26927
12:             GSE27888
13:             GSE28146
14:             GSE29680
15:             GSE32536
16:             GSE32645
17:             GSE29317
18:             GSE15222
19:             GSE34879
20:             GSE37263
    experiment.shortName
                                                                                                                                     experiment.name
                                                                                                                                              &lt;char&gt;
 1:                                                                                                          Control vs. APPSw mice gene expressions
 2:                                                                                         Vehicle and sAPPalpha-treated hippocampal slice cultures
 3:                                                                                                                         Rogers-3U24NS043571-01S1
 4:                                                                                                    Parkinson's disease vs. controls, whole blood
 5:                                                             Expression of mRNAs Regulating Synaptic Function and Neuroplasticity in Incipient AD
 6:                                                                          Effect of BDNF on the APP transgenic mouse model of Alzheimer's disease
 7:                                                                           Differential gene expression in ADAM10 over-expressing transgenic mice
 8:                                                                                   Incipient Alzheimer's Disease: Microarray Correlation Analyses
 9:                                       Comprehensive mRNA expression profiling distinguishes tauopathies and identifies shared molecular pathways
10:                                                           Effects of Long-term Pioglitazone Treatment on Peripheral and Central Markers of Aging
11:                                                                                 Common neuroinflammatory pathways in neurodegenerative diseases.
12:                                                                               Comparative transcriptome analysis of APPs?-DM and APLP2-KO brains
13: Microarray analyses of laser-captured hippocampus reveal distinct gray and white matter signatures associated with incipient Alzheimer’s disease
14:                          Effects of DHA-rich n-3 fatty acid supplementation on gene expression in blood mononuclear leukocytes: the OmegAD study
15:                                              Long-term pioglitazone improves learning and attenuates pathological markers in a mouse model of AD
16:                                                                   The unique molecular signature of cortical tissue injury in Multiple Sclerosis
17:                                   Adult and neonatal astrocytes exhibit diverse gene expression profiles during exposure to beta amyloid ex vivo
18:                                                                     Genetic control of human brain transcript expression in Alzheimer’s disease.
19:                                                          Probing sporadic and familial Alzheimer?s disease using induced pluripotent stem cells.
20:                                            Genome-wide profiling of altered gene expression in the neocortex of Alzheimer's disease (gene level)
                                                                                                                                     experiment.name
    experiment.ID
            &lt;int&gt;
 1:            27
 2:           250
 3:           275
 4:           529
 5:           910
 6:           924
 7:           926
 8:          1565
 9:          1901
10:          1911
11:          2684
12:          2727
13:          3336
14:          3374
15:          4699
16:          5512
17:          5528
18:          5643
19:          5722
20:          5811
    experiment.ID
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      experiment.description
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      &lt;char&gt;
 1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Gene expression changes in mice overexpressing amyloid precursor protein with the Swedish mutation (Tg2576; APPSw).  The gene expression levels in male 12 month-old APPSw mice (GSM26708, GSM26709) were compared to those in male nontransgenic littermate control mice (GSM26706, GSM26707).  Data were analyzed with MAS 5.0 and scaled to 2500.  See Stein, TD, Anders, NJ, DeCarli, C, Chan, SL, Mattson, MP, and Johnson JA.  Neutralization of transthyretin reverses the neuroprotective effects of secreted APP in APPSw mice resulting in tau phosphorylation and loss of hippocampal neurons: support for the amyloid hypothesis. J Neurosci. in press.\nDate GSE1556 Last Updated: May 29 2005\nContributors:  Thor D Stein Jeffrey A Johnson\nIncludes GDS811.\n Update date: Nov 15 2004.\n Dataset description GDS811: Expression profiling of brain left hippocampi from 12 month old Tg2576 transgenics overexpressing the amyloid precursor protein with the Swedish mutation (APPSw).  APPSw linked to the early-onset Alzheimer's disease.  Tg2576 express neuroprotective genes that prevent neuronal cell loss.
 2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Gene expression changes induced by alpha-secretase cleaved amyloid precursor protein (sAPPalpha) in organotypic hippocampal slice cultures of male, postnatal day 15 mice (C57B6/SJL).  Hippocampal slice cultures were treated with phosphate buffered saline (GSM26700, GSM26701, GSM26702) or 1 nM sAPPalpha (GSM26703, GSM26704, GSM26705) for 24 h.  Each sample consists of total RNA isolated from 8-12 slices from 4 mice.  Data were analyzed with MAS 5.0 and scaled to 2500.  sAPPalpha induces the amyloid sequestration protein transthyretin, insulin-like growth factor 2, insulin-like growth factor binding protein 2, and other genes involved in protective pathways such as apoptosis inhibition, detoxification, and retinol transport.  See Stein, TD, Anders, NJ, DeCarli, C, Chan, SL, Mattson, MP, and Johnson JA.  Neutralization of transthyretin reverses the neuroprotective effects of secreted APP in APPSw mice resulting in tau phosphorylation and loss of hippocampal neurons: support for the amyloid hypothesis. J Neurosci. in press.\nLast Updated (by provider): Oct 28 2005\nContributors:  Jeffrey A Johnson Thor D Stein\nIncludes GDS809.\n Update date: Nov 15 2004.\n Dataset description GDS809: Expression profiling of hippocampal slice cultures treated with 1 nM  alpha-secretase cleaved amyloid precursor protein (sAPPalpha) for 24 hours.  Hippocampi obtained from 15 day postnatal C57B6/SJL males.  Results provide insight into the neuroprotective activity of sAPPalpha.
 3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Alzheimer's Disease (AD) is a devastating neurodegenerative disorder affecting approximately 4 million people in the U.S. alone.  AD is characterized by the presence of senile plaques and neurofibrillary tangles in cortical regions of the brain.  These pathological markers are thought to be responsible for the massive cortical neurodegeneration and concomitant loss of memory, reasoning, and often aberrant behaviors that are seen in patients with AD.  Understanding the molecular mechanisms whereby these histopathological markers develop will greatly enhance our understanding of AD development and progression.  A clearer understanding of the mechanisms underlying neurofibrillary tangle formation specifically may help to clarify the basis for dementia of AD as well as the dementias associated with other diseases that are collectively referred to as "tauopathies." To expression profile both neurons containing neurofibrillary tangles and normal neurons from the entorhinal cortex of 10 mid-stage AD cases. The gene expression profile of neurons that contain neurofibrillary tangles will differ from the expression profile of histopathologically normal neurons from the same patient and from the same brain region.  Some of these differences will be informative as to the mechanisms of tangle formation. First use laser capture microdissection to select 1000 neurons bearing neurofibrillary tangles and 1000 normal neurons from the layer 2 stellate island neurons of the entorhinal cortex of 10 mid-stage AD cases.  Second, to isolate the RNA from these captured neurons, perform double round linear amplification and hybridize the labeled cRNA to affymetrix U133A arrays.  Third, to use paired, permutational t-tests to analyze the microarray data to select tangle-specific differences in gene expression.  For the purposes of submitting this proposal, a value of 1 ug of RNA was entered due to web site constraints on values placed in that field.  However, all samples are from LCM captured cells and the actual RNA yield is likely closer to 100 pg.\nLast Updated (by provider): May 01 2006\nContributors:  Joseph Rogers
 4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Parkinson?s disease (PD) progresses relentlessly and affects five million people worldwide. Laboratory tests for PD are critically needed for developing treatments designed to slow or prevent progression of the disease. We performed a transcriptome-wide scan in 105 individuals to interrogate the molecular processes perturbed in cellular blood of patients with early-stage PD. The molecular marker here identified is strongly associated with risk of PD in 66 samples of the training set (third tertile cross-validated odds ratio of 5.7 {P for trend 0.005}). It is further validated in 39 independent test samples (third tertile odds ratio of 5.1 {P for trend 0.04}). The genes differentially expressed in patients with PD, or Alzheimer?s or progressive supranuclear palsy offer unique insights into disease-linked processes detectable in peripheral blood. Combining gene expression scans in blood and linked clinical data will facilitate the rapid characterization of candidate biomarkers as demonstrated here with respect to PD.\nLast Updated (by provider): Jan 23 2007\nContributors:  Clemens R Scherzer
 5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            In Alzheimer’s disease (AD), early deficits in learning and memory are a consequence of synaptic modification which are likely induced by toxic beta-amyloid oligomers (oA?).  To identify molecular targets downstream of oA? binding  we prepared synaptoneurosomes from frontal cortex of control and IAD patients, and isolated mRNAs for comparison of gene expression.  This approach elevated synaptic mRNAs above the threshold necessary for expression changes to be discriminated and also reduced other cellular mRNAs.  In patients with minimal cognitive impairment (MCI) termed incipient AD (IAD) global measures of cognition declined with increasing levels of dimeric A? (dA?).  These patients also showed increased expression of neuroplasticity related genes, many encoding 3' UTR  consensus sequences that regulate local translation in the synapse.  One such gene, GluR2, displayed elevated mRNA and protein expression in IAD.  Other neurotransmitter-related genes were also upregulated.  Overexpressed genes may induce compensatory as well as negative effects on cognition and provide targets for intervention to moderate the response to dA?. \nLast Updated (by provider): Sep 09 2008\nContributors:  Ruty Mehrian Shai Ya-Hsuan Hsu Celia Williams Yongchun Wu Bryan M Spann Yi Mo Traci Sitzer Carol A McCleary Carol A Miller
 6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     We examined transgenic (TG) mice expressing human APP695 bearing the double Swedish (671KM&gt;NL) and Indiana (717V&gt;F) amyloid precursor protein (APP) mutations. Lentiviral vectors constitutively expressing BDNF-GFP under control of the CMV/ß-actin hybrid promoter or GFP alone were injected into the entorhinal cortices of TG mice bilaterally at age 6 months, a time point by which neuropathological degeneration and cell loss are established. Age-matched wild-type littermates underwent sham surgery or injection of lentivirus expressing GFP into the entorhinal cortices bilaterally. \nLast Updated (by provider): Jan 22 2009\nContributors:  D H Geschwind B E Schroeder E Rockenstein M V Chao G Coppola A Blesch A H Nagahara J M Conner A A Chiba M H Tuszynski L Wang A Kim S Tsukada G M Shaked E Masliah D A Merrill E Koo
 7:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              In a transgenic mouse model of Alzheimer disease (AD), cleavage of the amyloid precursor protein (APP) by the ?-secretase ADAM10 prevented amyloid plaque formation and alleviated cognitive deficits. Furthermore, there was a positive influence of ADAM10 over-expression on neurotransmitter-specific formation of synapses and on synaptic plasticity. To assess the influence of ADAM10 on the gene expression profile in the brain we performed microarray analysis using RNA isolated from brains of five month old mice over-expressing either the ?-secretase ADAM10 or a dominant-negative mutant (dn) of this enzyme. As compared to non-transgenic wild-type mice, 355 genes were found to be differentially expressed in ADAM10 transgenic mice and 143 genes in dnADAM10 mice. A higher number of genes was found to be differentially regulated in double-transgenic mouse strains additionally expressing the human APP V717I mutant (APP[V717I]). Thus, ?-secretase cleavage of over-expressed APP[V717I] alters CNS gene expression additionally.\nLast Updated (by provider): Jan 22 2009\nContributors:  Dietrich Trümbach Wolfgang Wurst Falk Fahrenholz Kristina Endres Rolf Postina Claudia Prinzen
 8:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      For these data, we analyzed hippocampal gene expression of nine control and 22 AD subjects of varying severity on 31 separate microarrays. We then tested the correlation of each gene's expression with MiniMental Status Examination (MMSE) and neurofibrillary tangle (NFT) scores across all 31 subjects regardless of diagnosis. These tests revealed a major transcriptional response comprising thousands of genes significantly correlated with AD markers. Several hundred of these genes were also correlated with AD markers across only control and incipient AD subjects (MMSE &gt; 20).\nLast Updated (by provider): May 29 2005\nContributors:  Kuey-Chu Chen Philip W Landfield William R Markesbery Nada M Porter James W Geddes Eric M Blalock\nIncludes GDS810.\n Update date: Nov 01 2004.\n Dataset description GDS810: Expression profiling of brain hippocampi from 22 postmortem subjects with Alzheimer's disease (AD) at various stages of severity.  7, 8, and 7 subjects diagnosed with incipient, moderate, and severe AD respectively.  Results provide insight into mechanisms underlying the early pathogenesis of AD.
 9:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 We have constructed genome wide expression profiles from snap frozen post-mortem tissue from the medial temporal lobe of patients with four neurodegenerative disorders (5 AD, 5 PSP, 5 PiD and 5 FTD patients) and 5 control subjects. All patients were matched for age, gender, ApoE-? and MAPT (tau) haplotype. From all groups a total of 790 probes were shown to be differently expressed when compared to control individuals. The results from these experiments were then used to investigate the correlations between clinical, pathological and molecular findings. From the 790 identified probes we extracted a gene set of 166 probes whose expression could discriminate between these disorders and normal ageing. This gene set can be further developed into an accurate microarray-based classification test. Furthermore, from this dataset we extracted a disease specific set of genes and identified two aging related transcription factors (FOXO1A and FOXO3A) as possible drug targets related to neurodegenerative disease. 
10:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Background: Thiazolidinediones (TZDs) activate peroxisome proliferator-activated receptor gamma (PPARgamma) and are used clinically to help restore peripheral insulin sensitivity in Type 2 diabetes (T2DM). Interestingly, long-term treatment of mouse models of Alzheimer’s disease (AD) with TZDs also has been shown to reduce several well-established brain biomarkers of AD including inflammation, oxidative stress and Abeta accumulation. While some of the TZD actions are becoming clear in AD models and may mediate their reported beneficial impact in AD patients, little is known about the functional consequences of TZDs in animal models of normal aging. Because aging is a common risk factor for both AD and T2DM, we investigated whether the TZD, pioglitazone could alter brain aging under non-pathological conditions.  Findings: The TZD pioglitazone (PIO) was incorporated into the diet to yield a final dose of approximately 2.3 mg/kg body weight/day. PIO reduced insulin levels irrespective of age. Interestingly, a significant reduction in the Ca2+-dependent afterhyperpolarization was seen in the aged animals with no significant change in LTP maintenance or learning and memory performance. Finally, a combination of microarray analyses on hippocampal tissue and serum-based multiplex cytokine assays revealed that age-dependent inflammatory changes in brain and periphery were not reversed by PIO. Conclusions:  While current research efforts continue to address the underlying processes responsible for the progressive decline in cognitive function seen during aging, available medical treatments are still very limited. Our study, therefore, was aimed at elucidating potentially novel actions of TZDs in the aging brain. Using a clinically-relevant dose and delivery method, PIO had no detectable impact on several indices of brain aging and failed to alter both peripheral and central age-related increases in inflammatory signaling. \nLast Updated (by provider): May 05 2010\nContributors:  Kuey-Chu Chen Olivier Thibault Jeremiah T Phelps Nada M Porter Katie Anderson Tristano Pancani Donald A Cohen James L Searcy Eric M Blalock Jelena Popovic James C Gant Margarita Avdiushko
11:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Neurodegenerative diseases of the central nervous system are characterised by pathogenetic cellular and molecular changes in specific areas of the brain that lead to the dysfunction and/or loss of explicit neuronal populations. Despite exhibiting different clinical profiles and selective neuronal loss, common features such as abnormal protein deposition, dysfunctional cellular transport, mitochondrial deficits, glutamate excitotoxicity and inflammation are observed in most, if not all, neurodegenerative disorders, suggesting converging pathways of neurodegeneration. We have generated comparative genome-wide gene expression data for Alzheimer’s disease, amyotrophic lateral sclerosis, Huntington’s disease, multiple sclerosis, Parkinson’s disease and schizophrenia using an extensive cohort of well characterised post-mortem CNS tissues. The analysis of whole genome expression patterns across these major disorders offers an outstanding opportunity not only to look into exclusive disease specific changes, but more importantly to uncover potential common molecular pathogenic mechanisms that could be targeted for therapeutic gain. Surprisingly, no dysregulated gene that passed our selection criteria was found in common across all 6 diseases using our primary method of analysis. However, 61 dysregulated genes were shared when comparing five and four diseases. Our analysis indicates firstly the involvement of common neuronal homeostatic, survival and synaptic plasticity pathways. Secondly, we report changes to immunoregulatory and immunomodulatory pathways in all diseases. Our secondary method of analysis confirmed significant up-regulation of a number of genes in diseases presenting degeneration and showed that somatostatin was downregulated in all 6 diseases. The latter is supportive of a general role for neuroinflammation in the pathogenesis and/or response to neurodegeneration. Unravelling the detailed nature of the molecular changes regulating inflammation in the CNS is key to the development of novel therapeutic approaches for these chronic conditions.\nLast Updated (by provider): Jan 29 2011\nContributors:  Andrea Schmitt Shama Fernando Isidro Ferrer Tim P Bonnert Edna Grünblatt Peter J Gebicke-Haerter David T Dexter Miklos Palkovits Hans Kretzschmar Danielle Seilhean Richard Reynolds Pascal F Durrenberger N B Oumesmar Thomas Arzberger Peter Falkai Samira N Kashefi
12:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Despite its key role in Alzheimer pathogenesis, the physiological function(s) of the amyloid precursor protein (APP) and of its proteolytic fragments are still poorly understood. The secreted APPs? ectodomain has been shown to be involved in neuroprotection and synaptic plasticity. The ?-secretase generated APP intracellular domain, AICD, functions as a transcriptional regulator in heterologous reporter assays although its role for endogenous gene regulation has remained controversial. Previously, we have generated APPs? knockin (KI) mice expressing solely the secreted ectodomain APPs?. Here, we generated double mutants (APPs?-DM) by crossing APPs?-KI mice onto an APLP2-deficient background and show that APPs? rescues the postnatal lethality of the majority of APP/APLP2 double knockout mice. Despite normal CNS morphology and unaltered basal synaptic transmission, young APPs?-DM mice already showed pronounced hippocampal dysfunction, impaired spatial learning and a deficit in LTP. To gain further mechanistic insight into which domains/proteolytic fragments are crucial for hippocampal APP/APLP2 mediated functions, we performed a DNA microarray transcriptome profiling of prefrontal cortex and hippocampus of adult APLP2-KO (APLP2-/-) and APPs?-DM mice (APP?/?APLP2-/- mice).Interestingly, this analysis failed to reveal major genotype-related transcriptional differences. Expression differences between cortex and hippocampus were, however, readily detectable.\nLast Updated (by provider): May 10 2011\nContributors:  Benedikt Brors Ulrike C Müller Mikhail Filippov Dorothee Aydin Sascha Weyer Norbert Gretz
13:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Alzheimer's disease (AD) is a devastating neurodegenerative disorder that threatens to reach epidemic proportions as our population ages. Although much research has examined molecular pathways associated with AD, relatively few studies have focused on critical early stages. Our prior microarray study correlated gene expression in human hippocampus with AD markers. Results suggested a new model of early-stage AD in which pathology spreads along myelinated axons, orchestrated by upregulated transcription and epigenetic factors related to growth and tumor suppression (Blalock et al., 2004). However, the microarray analyses were performed on RNA from fresh frozen hippocampal tissue blocks containing both gray and white matter, potentially obscuring region-specific changes. In the present study, we used laser capture microdissection to exclude major white matter tracts and selectively collect CA1 hippocampal gray matter from formalin-fixed, paraffin-embedded (FFPE) hippoc ampal sections of the same subjects assessed in our prior study. Microarray analyses of this gray matter-enriched tissue revealed many correlations similar to those seen in our prior study, particularly for neuron-related genes. Nonetheless, in the laser-captured tissue, we found a striking paucity of the AD-associated epigenetic and transcription factor genes that had been strongly overrepresented in the prior tissue block study. In addition, we identified novel pathway alterations that may have considerable mechanistic implications, including downregulation of genes stabilizing ryanodine receptor Ca2+ release and upregulation of vascular development genes. We conclude that FFPE tissue can be a reliable resource for microarray studies, that upregulation of growth-related epigenetic/ transcription factors with incipient AD is predominantly localized to white matter, further supporting our prior findings and model, and that alterations in vascular and ryanodine receptor-relat ed pathways in gray matter are closely associated with incipient AD.\nLast Updated (by provider): Aug 18 2011\nContributors:  Philip W Landfield Heather M Buechel James W Geddes Eric M Blalock Jelena Popovic
14:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        The hypothesis was that 6 months of dietary n-3 FA supplementation to humans affected expression of genes that might be of significance for inflammatory processes and for Alzheimer´s disease.\nLast Updated (by provider): Jun 27 2011\nContributors:  Inger Vedin David Brodin
15:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Thiazolidinediones (TZDs) are agonists at peroxisome proliferator-activated gamma-type (PPAR-y) receptors and are used clinically for the treatment of type 2 diabetes where they have been shown to reestablish insulin sensitivity, improve lipids profile, and reduce inflammation. Recent work also suggests that TZDs may be beneficial in Alzheimer's disease (AD), ameliorating cognitive decline early in the disease process. However, there have been only a few studies identifying mechanisms through which cognitive benefits may be exerted. Starting at 10 months of age, the triple transgenic mouse model of AD (3xTg-AD) with accelerated amyloid-B (AB) deposition and tau pathology was treated with the TZD pioglitazone (PIO- Actos) at 18 mg/Kg body weight/day. After four months, PIO-treated animals showed multiple beneficial effects, including improved learning on the active avoidance task, reduced serum cholesterol, decreased hippocampal AB deposits, and enhanced short- and long-term plasticity. Baseline electrophysiological membrane properties and blood glucose levels were unchanged by PIO treatment. Gene microarray analyses of hippocampal tissue identified predicted transcriptional responses following TZD treatment as well as potentially novel targets of TZDs, including facilitation of estrogenic processes, and decreases in glutamatergic and ketone metabolic/ cholesterol dependent processes. Taken together, these results confirm prior animal studies showing that TZDs can ameliorate cognitive deficits associated with AD-related pathology, but also extend these findings by pointing to novel molecular targets in the brain.\nLast Updated (by provider): Apr 15 2012\nContributors:  Kuey-Chu Chen Tristano Pancani James L Searcy Eric M Blalock Jelena Popovic Tina L Beckett Philip W Landfield Olivier Thibault Jeremiah T Phelps Inga Kadish Nada M Porter Michael P Murphy Katie L Anderson
16:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           In the present study we addressed several questions related to the mechanisms of cortical injury. We analyzed genome wide gene expression by microarrays, comparing active multiple sclerosis lesions with highly inflammatory lesions of chronic tuberculous meningitis, with neurodegenerative lesions of Alzheimer’s disease and with normal cortex of age matched controls.\nLast Updated (by provider): Jun 13 2012\nContributors:  Marie-Therese Fischer
17:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Research regarding the role of astrocytes as ?-amyloid (A?) degrading cells has broadened our view about the mechanisms how these common glia cells function in Alzheimer’s disease (AD). Based on previous studies adult mouse astrocytes are able to degrade A? deposits from AD mouse model and human brain tissue sections ex vivo, contrary to neonatal astrocytes. In this study, we studied the possible altered gene expression profiles of adult and neonatal astrocytes cultured for 22 h on top of the A? burdened tg APdE9 or wild-type mouse brain sections using whole genome microarrays. Quantitative RT-PCR analysis confirmed the significant up-regulation of HtrA serine peptidase 1 (Htra1), metallopeptidase 9 (Mmp9), phosphate regulating gene with homologies to endopeptidases on the X chromosome (Phex) and scavenger receptor class A, member 5 (Scara5) in adult astrocytes, whereas neonatal astrocytes up-regulated Mmp9 and down-regulated genes related to cholesterol transport and synthesis: apolipoprotein E (Apoe), 24-dehydrocholesterol reductase (Dhcr24) and 3-hydroxy-3-methylglutaryl-CoA synthase 1 (Hmgcs1). These findings brought out novel genes which expression is altered during A? clearance in adult and neonatal astrocytes ex vivo.\nLast Updated (by provider): Jul 14 2012\nContributors:  Jari Koistinaho Rea Pihlaja Eveliina Pollari Antti Kurronen Markus Storvik Katja Kanninen Milla Koistinaho Garry Wong
18: We recently surveyed the relationship between the human brain transcriptome and genome in a series of neuropathologically normal postmortem samples. We now have analyzed additional samples with a confirmed pathologic diagnosis of late onset Alzheimer's disease (LOAD, final n=187 controls, 176 cases). Nine percent of the cortical transcripts we analyzed had expression profiles correlated with their genotypes in the combined cohort and approximately 5% of transcripts had SNP-transcript relationships that could distinguish LOAD samples. Two of these transcripts have been previously implicated in LOAD candidate gene SNP-expression screens. This study shows how the relationship between common inherited genetic variants and brain transcript expression can be used in the study of human brain disorders. We suggest that studying the transcriptome as a quantitative endo-phenotype has greater power to find risk SNPs influencing expression than the use of discrete diagnostic categories such as presence or absence of disease. see DOI:10.1016/j.ajhg.2009.03.011 for further details and complete author list.\nNote: 187 samples from this series, which appear in other Expression Experiments in Gemma, were not imported from the GEO source. The following samples were removed: GSM226150 on GPL2700,GSM226050 on GPL2700,GSM226051 on GPL2700,GSM226158 on GPL2700,GSM225675 on GPL2700,GSM226052 on GPL2700,GSM226157 on GPL2700,GSM225676 on GPL2700,GSM226053 on GPL2700,GSM226156 on GPL2700,GSM225677 on GPL2700,GSM226155 on GPL2700,GSM225678 on GPL2700,GSM226055 on GPL2700,GSM226154 on GPL2700,GSM225679 on GPL2700,GSM226056 on GPL2700,GSM226151 on GPL2700,GSM226049 on GPL2700,GSM225681 on GPL2700,GSM225680 on GPL2700,GSM225683 on GPL2700,GSM225682 on GPL2700,GSM225685 on GPL2700,GSM226148 on GPL2700,GSM225684 on GPL2700,GSM226149 on GPL2700,GSM225965 on GPL2700,GSM225962 on GPL2700,GSM225961 on GPL2700,GSM226040 on GPL2700,GSM225964 on GPL2700,GSM225963 on GPL2700,GSM225666 on GPL2700,GSM226145 on GPL2700,GSM225667 on GPL2700,GSM226044 on GPL2700,GSM225664 on GPL2700,GSM226147 on GPL2700,GSM226041 on GPL2700,GSM226146 on GPL2700,GSM225665 on GPL2700,GSM226042 on GPL2700,GSM226047 on GPL2700,GSM226048 on GPL2700,GSM225668 on GPL2700,GSM226045 on GPL2700,GSM225669 on GPL2700,GSM226046 on GPL2700,GSM225670 on GPL2700,GSM226039 on GPL2700,GSM226038 on GPL2700,GSM225674 on GPL2700,GSM225673 on GPL2700,GSM225672 on GPL2700,GSM225671 on GPL2700,GSM225764 on GPL2700,GSM225763 on GPL2700,GSM225953 on GPL2700,GSM225952 on GPL2700,GSM225951 on GPL2700,GSM225950 on GPL2700,GSM225957 on GPL2700,GSM225956 on GPL2700,GSM225760 on GPL2700,GSM225955 on GPL2700,GSM225761 on GPL2700,GSM225954 on GPL2700,GSM225762 on GPL2700,GSM225959 on GPL2700,GSM225958 on GPL2700,GSM225697 on GPL2700,GSM225698 on GPL2700,GSM225699 on GPL2700,GSM225753 on GPL2700,GSM225752 on GPL2700,GSM225755 on GPL2700,GSM225754 on GPL2700,GSM225757 on GPL2700,GSM225756 on GPL2700,GSM225759 on GPL2700,GSM225758 on GPL2700,GSM225940 on GPL2700,GSM225942 on GPL2700,GSM225941 on GPL2700,GSM225944 on GPL2700,GSM225943 on GPL2700,GSM225751 on GPL2700,GSM225946 on GPL2700,GSM225945 on GPL2700,GSM225948 on GPL2700,GSM225947 on GPL2700,GSM225949 on GPL2700,GSM225688 on GPL2700,GSM225689 on GPL2700,GSM225686 on GPL2700,GSM225687 on GPL2700,GSM225696 on GPL2700,GSM225695 on GPL2700,GSM225749 on GPL2700,GSM225693 on GPL2700,GSM225692 on GPL2700,GSM225691 on GPL2700,GSM225690 on GPL2700,GSM225744 on GPL2700,GSM225743 on GPL2700,GSM225742 on GPL2700,GSM225741 on GPL2700,GSM225748 on GPL2700,GSM225747 on GPL2700,GSM225746 on GPL2700,GSM225745 on GPL2700,GSM225736 on GPL2700,GSM225737 on GPL2700,GSM225734 on GPL2700,GSM225735 on GPL2700,GSM225938 on GPL2700,GSM225732 on GPL2700,GSM225939 on GPL2700,GSM225733 on GPL2700,GSM225936 on GPL2700,GSM225730 on GPL2700,GSM225937 on GPL2700,GSM225731 on GPL2700,GSM225934 on GPL2700,GSM225935 on GPL2700,GSM225932 on GPL2700,GSM225933 on GPL2700,GSM225930 on GPL2700,GSM225931 on GPL2700,GSM225739 on GPL2700,GSM225929 on GPL2700,GSM225723 on GPL2700,GSM225724 on GPL2700,GSM225725 on GPL2700,GSM225726 on GPL2700,GSM225925 on GPL2700,GSM225926 on GPL2700,GSM225720 on GPL2700,GSM225927 on GPL2700,GSM225721 on GPL2700,GSM225928 on GPL2700,GSM225722 on GPL2700,GSM225921 on GPL2700,GSM225922 on GPL2700,GSM225923 on GPL2700,GSM225924 on GPL2700,GSM225727 on GPL2700,GSM225728 on GPL2700,GSM226082 on GPL2700,GSM225729 on GPL2700,GSM225920 on GPL2700,GSM225916 on GPL2700,GSM225917 on GPL2700,GSM225711 on GPL2700,GSM225915 on GPL2700,GSM225714 on GPL2700,GSM225715 on GPL2700,GSM225918 on GPL2700,GSM225919 on GPL2700,GSM225713 on GPL2700,GSM225718 on GPL2700,GSM225719 on GPL2700,GSM225717 on GPL2700,GSM226037 on GPL2700,GSM226035 on GPL2700,GSM226034 on GPL2700,GSM225662 on GPL2700,GSM225700 on GPL2700,GSM225701 on GPL2700,GSM225702 on GPL2700,GSM225703 on GPL2700,GSM225704 on GPL2700,GSM225705 on GPL2700,GSM225706 on GPL2700,GSM225707 on GPL2700,GSM226159 on GPL2700,GSM225708 on GPL2700,GSM225709 on GPL2700,GSM226164 on GPL2700,GSM226165 on GPL2700,GSM226162 on GPL2700,GSM226163 on GPL2700,GSM226168 on GPL2700,GSM226167 on GPL2700,GSM225652 on GPL2700,GSM226160 on GPL2700\nLast Updated (by provider): Mar 21 2012\nContributors:  Jennifer Webster Amanda Myers Raphael Gibbs
19:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Our understanding of Alzheimer’s disease (AD) pathogenesis is currently limited by difficulties in obtaining live neurons from patients and the inability to model the sporadic form of AD. It may be possible to overcome these challenges by reprogramming primary cells from patients into induced pluripotent stem cells (iPSCs). We reprogrammed primary fibroblasts from two patients with familial AD (both caused by a duplication of APP1, APPDp), two with sporadic AD (sAD1, 2) and two non-demented control individuals (NDCs) into iPSC lines. Neurons from differentiated cultures were FACS-purified and characterized. Purified cultures contained &gt;90% neurons, clustered with fetal brain mRNA samples by microarray criteria, and could form functional synaptic contacts. Virtually all cells exhibited normal electrophysiological activity. Relative to controls, iPSC-derived, purified neurons from the two APPDp patients and patient sAD2 exhibited significantly higher levels of secreted A?1-40, phospho-tauThr231 (pTau) and active GSK3? (aGSK3?). Neurons from APPDp and sAD2 patients also accumulated large Rab5-positive early endosomes compared to controls. Treatment of purified neurons with ?-secretase inhibitors, but not g-secretase inhibitors, caused significant reductions in pTau and aGSK3? levels. These results suggest a direct relationship between APP proteolytic processing, but not A?, in GSK3? activation and tau phosphorylation in human neurons. Additionally, we observed that neurons with the genome of one sAD patient exhibited the phenotypes seen in familial AD samples. More generally, we demonstrate that iPSC technology can be used to observe phenotypes relevant to AD, even though it can take decades for overt disease to manifest in patients.\nLast Updated (by provider): Sep 24 2012\nContributors:  Mason A Israel Sol M Reyna Shauna H Yuan Edward H Koo Martin Marsala Louise C Laurent Yangling Mu Francesca S Boscolo Fred H Gage Cheryl Herrera Michael P Hefferan Anne M Remes Cedric Bardy Sebastiaan Van Gorp Christian T Carson Kristopher L Nazor Lawrence S Goldstein
20:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             We investigated genome-wide gene alterations in the temporal cortex of a well-characterized cohort of Alzheimer’s disease (AD) patients using Affymetrix exon arrays.\nLast Updated (by provider): Oct 19 2012\nContributors:  Michelle G Tan Mitchell K Lai
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      experiment.description
    experiment.troubled experiment.accession experiment.database
                 &lt;lgcl&gt;               &lt;char&gt;              &lt;char&gt;
 1:               FALSE              GSE1556                 GEO
 2:               FALSE              GSE1555                 GEO
 3:               FALSE              GSE4757                 GEO
 4:               FALSE              GSE6613                 GEO
 5:               FALSE             GSE12685                 GEO
 6:               FALSE             GSE14499                 GEO
 7:               FALSE             GSE10908                 GEO
 8:               FALSE              GSE1297                 GEO
 9:               FALSE          E-MEXP-2280        ArrayExpress
10:               FALSE             GSE20219                 GEO
11:               FALSE             GSE26927                 GEO
12:               FALSE             GSE27888                 GEO
13:               FALSE             GSE28146                 GEO
14:               FALSE             GSE29680                 GEO
15:               FALSE             GSE32536                 GEO
16:               FALSE             GSE32645                 GEO
17:               FALSE             GSE29317                 GEO
18:               FALSE             GSE15222                 GEO
19:               FALSE             GSE34879                 GEO
20:               FALSE             GSE37263                 GEO
    experiment.troubled experiment.accession experiment.database
                        experiment.URI experiment.sampleCount
                                &lt;char&gt;                  &lt;int&gt;
 1:   http://www.ncbi.nlm.nih.gov/geo/                      4
 2:   http://www.ncbi.nlm.nih.gov/geo/                      6
 3:   http://www.ncbi.nlm.nih.gov/geo/                     20
 4:   http://www.ncbi.nlm.nih.gov/geo/                    105
 5:   http://www.ncbi.nlm.nih.gov/geo/                     14
 6:   http://www.ncbi.nlm.nih.gov/geo/                     26
 7:   http://www.ncbi.nlm.nih.gov/geo/                     27
 8:   http://www.ncbi.nlm.nih.gov/geo/                     31
 9: http://www.ebi.ac.uk/arrayexpress/                     31
10:   http://www.ncbi.nlm.nih.gov/geo/                     29
11:   http://www.ncbi.nlm.nih.gov/geo/                    118
12:   http://www.ncbi.nlm.nih.gov/geo/                     12
13:   http://www.ncbi.nlm.nih.gov/geo/                     30
14:   http://www.ncbi.nlm.nih.gov/geo/                     31
15:   http://www.ncbi.nlm.nih.gov/geo/                     16
16:   http://www.ncbi.nlm.nih.gov/geo/                     12
17:   http://www.ncbi.nlm.nih.gov/geo/                     12
18:   http://www.ncbi.nlm.nih.gov/geo/                    176
19:   http://www.ncbi.nlm.nih.gov/geo/                     38
20:   http://www.ncbi.nlm.nih.gov/geo/                     16
                        experiment.URI experiment.sampleCount
    experiment.lastUpdated  experiment.batchEffect geeq.batchCorrected
                    &lt;POSc&gt;                  &lt;char&gt;              &lt;lgcl&gt;
 1:    2023-12-17 00:32:19           NO_BATCH_INFO               FALSE
 2:    2023-12-17 00:29:07           NO_BATCH_INFO               FALSE
 3:    2023-12-19 23:39:29 BATCH_CORRECTED_SUCCESS                TRUE
 4:    2023-12-20 18:58:36           NO_BATCH_INFO               FALSE
 5:    2023-12-16 10:05:10 BATCH_CORRECTED_SUCCESS                TRUE
 6:    2023-12-16 19:08:21           NO_BATCH_INFO               FALSE
 7:    2023-12-16 00:59:44           NO_BATCH_INFO               FALSE
 8:    2023-12-16 11:23:40           NO_BATCH_INFO                TRUE
 9:    2023-09-12 20:48:50           NO_BATCH_INFO               FALSE
10:    2023-12-17 23:21:01           NO_BATCH_INFO               FALSE
11:    2023-12-18 17:04:17           NO_BATCH_INFO               FALSE
12:    2023-12-19 10:30:02    SINGLE_BATCH_SUCCESS               FALSE
13:    2023-12-18 18:51:34 BATCH_CORRECTED_SUCCESS                TRUE
14:    2023-12-18 21:18:32           NO_BATCH_INFO               FALSE
15:    2023-12-19 01:46:09 BATCH_CORRECTED_SUCCESS                TRUE
16:    2023-12-19 01:51:57 NO_BATCH_EFFECT_SUCCESS               FALSE
17:    2023-12-18 20:37:12 BATCH_CORRECTED_SUCCESS                TRUE
18:    2023-12-16 22:48:31           NO_BATCH_INFO               FALSE
19:    2023-12-19 05:04:27           NO_BATCH_INFO               FALSE
20:    2023-12-20 08:39:27 NO_BATCH_EFFECT_SUCCESS                TRUE
    experiment.lastUpdated  experiment.batchEffect geeq.batchCorrected
    geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore
                 &lt;num&gt;            &lt;num&gt;        &lt;num&gt;       &lt;num&gt;       &lt;num&gt;
 1:                  0                0           -1   0.2768486      0.3125
 2:                  0                0           -1  -0.1700709      0.4000
 3:                  1                1            1   0.9928338      1.0000
 4:                 -1                0            1   0.4242852      0.8750
 5:                  1                1            1   0.7121066      0.7500
 6:                 -1                0            1   0.4274038      1.0000
 7:                 -1                0            1   0.4276836      1.0000
 8:                  1                0            1   0.5645978      0.8750
 9:                  0                0           -1   0.2683523      0.5000
10:                  1                0            1   0.5707075      0.8750
11:                  0                0           -1   0.1363510      0.2500
12:                  1                1           -1   0.8564026      0.6250
13:                  1                0            1   0.7090478      1.0000
14:                  0                0           -1   0.4130823      0.1875
15:                  1                1            1   0.9994545      0.8750
16:                  1                1           -1   0.8308846      0.6250
17:                  1                0            1   0.7131241      0.6250
18:                  0                0           -1   0.2690545      0.2500
19:                  0                0           -1   0.1386778      0.7500
20:                  1                0            1   0.8488878      0.8750
    geeq.batchConfound geeq.batchEffect geeq.rawData geeq.qScore geeq.sScore
    taxon.name  taxon.scientific taxon.ID taxon.NCBI taxon.database.name
        &lt;char&gt;            &lt;char&gt;    &lt;int&gt;      &lt;int&gt;              &lt;char&gt;
 1:      mouse      Mus musculus        2      10090                mm10
 2:      mouse      Mus musculus        2      10090                mm10
 3:      human      Homo sapiens        1       9606                hg38
 4:      human      Homo sapiens        1       9606                hg38
 5:      human      Homo sapiens        1       9606                hg38
 6:      mouse      Mus musculus        2      10090                mm10
 7:      mouse      Mus musculus        2      10090                mm10
 8:      human      Homo sapiens        1       9606                hg38
 9:      human      Homo sapiens        1       9606                hg38
10:        rat Rattus norvegicus        3      10116                 rn6
11:      human      Homo sapiens        1       9606                hg38
12:      mouse      Mus musculus        2      10090                mm10
13:      human      Homo sapiens        1       9606                hg38
14:      human      Homo sapiens        1       9606                hg38
15:      mouse      Mus musculus        2      10090                mm10
16:      human      Homo sapiens        1       9606                hg38
17:      mouse      Mus musculus        2      10090                mm10
18:      human      Homo sapiens        1       9606                hg38
19:      human      Homo sapiens        1       9606                hg38
20:      human      Homo sapiens        1       9606                hg38
    taxon.name  taxon.scientific taxon.ID taxon.NCBI taxon.database.name
    taxon.database.ID
                &lt;int&gt;
 1:                81
 2:                81
 3:                87
 4:                87
 5:                87
 6:                81
 7:                81
 8:                87
 9:                87
10:                86
11:                87
12:                81
13:                87
14:                87
15:                81
16:                87
17:                81
18:                87
19:                87
20:                87
    taxon.database.ID</code></pre>
<p>Dataset information provided by <code>get_datasets</code> also
includes some quality information that can be used to determine the
suitability of any given experiment. For instance
<code>geeq.batchEffect</code> column will be set to -1 if Gemma’s
preprocessing has detected batch effects that were unable to be resolved
by batch correction. More information about these and other fields can
be found at the function documentation.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>taxa <span class="op">=</span> <span class="st">'human'</span>, filter <span class="op">=</span> <span class="st">'bioAssayCount &gt; 4'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>     <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">geeq.batchEffect</span> <span class="op">!=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>   experiment.shortName
                 &lt;char&gt;
1:              GSE2018
2:              GSE4036
3:              GSE3489
4:              GSE1923
5:               GSE361
6:               GSE492
                                                                          experiment.name
                                                                                   &lt;char&gt;
1:                                                            Human Lung Transplant - BAL
2:                                                                perro-affy-human-186940
3: Patterns of gene dysregulation in the frontal cortex of patients with HIV encephalitis
4: Identification of PDGF-dependent patterns of gene expression in U87 glioblastoma cells
5:                                                   Mammary epithelial cell transduction
6:                               Effect of prostaglandin analogs on aqueous humor outflow
   taxon.name
       &lt;char&gt;
1:      human
2:      human
3:      human
4:      human
5:      human
6:      human</code></pre>
<p>Gemma uses multiple ontologies when annotating datasets and using the
term URIs instead of free text to search can lead to more specific
results. <a href="https://pavlidislab.github.io/gemma.R/reference/search_annotations.html" class="external-link"><code>search_annotations</code></a>
function allows searching for annotation terms that might be relevant to
your query.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/search_annotations.html">search_annotations</a></span><span class="op">(</span><span class="st">'bipolar'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">head</span></span></code></pre></div>
<pre><code>    category.name                         category.URI value.name value.URI
           &lt;char&gt;                               &lt;char&gt;     &lt;char&gt;    &lt;char&gt;
1: disease status                                 &lt;NA&gt;    bipolar      &lt;NA&gt;
2: disease status                                 &lt;NA&gt;    bipolar      &lt;NA&gt;
3:             dx                                 &lt;NA&gt;    Bipolar      &lt;NA&gt;
4: disease status                                 &lt;NA&gt;    bipolar      &lt;NA&gt;
5:       genotype http://www.ebi.ac.uk/efo/EFO_0000513    Bipolar      &lt;NA&gt;
6: disease status                                 &lt;NA&gt;    bipolar      &lt;NA&gt;</code></pre>
</div>
<div class="section level2">
<h2 id="downloading-expression-data">Downloading expression data<a class="anchor" aria-label="anchor" href="#downloading-expression-data"></a>
</h2>
<p>Upon identifying datasets of interest, more information about
specific ones can be requested. In this example we will be using
GSE46416 which includes samples taken from healthy donors along with
manic/euthymic phase bipolar disorder patients.</p>
<p>The data associated with specific experiments can be accessed by
using <a href="https://pavlidislab.github.io/gemma.R/reference/get_datasets_by_ids.html" class="external-link"><code>get_datasets_by_ids</code></a></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_datasets_by_ids.html">get_datasets_by_ids</a></span><span class="op">(</span><span class="st">"GSE46416"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>, <span class="va">experiment.ID</span>, <span class="va">experiment.description</span><span class="op">)</span></span></code></pre></div>
<pre><code>   experiment.shortName
                 &lt;char&gt;
1:             GSE46416
                                                   experiment.name
                                                            &lt;char&gt;
1: State- and trait-specific gene expression in euthymia and mania
   experiment.ID
           &lt;int&gt;
1:          8997
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       experiment.description
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;char&gt;
1: Gene expression profiles of bipolar disorder (BD) patients were assessed during both a manic and a euthymic phase and compared both intra-individually, and with the gene expression profiles of controls.\nLast Updated (by provider): Sep 05 2014\nContributors:  Christian C Witt Benedikt Brors Dilafruz Juraeva Jens Treutlein Carsten Sticht Stephanie H Witt Jana Strohmaier Helene Dukal Josef Frank Franziska Degenhardt Markus M Nöthen Sven Cichon Maren Lang Marcella Rietschel Sandra Meier Manuel Mattheisen</code></pre>
<p>To access the expression data in a convenient form, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/get_dataset_object.html" class="external-link"><code>get_dataset_object</code></a>.
It is a high-level wrapper that combines various endpoint calls to
return lists of annotated <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link"><code>SummarizedExperiment</code></a>
or <a href="https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf" class="external-link"><code>ExpressionSet</code></a>
objects that are compatible with other Bioconductor packages or a <a href="https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html" class="external-link">tidyverse-friendly</a>
long form tibble for downstream analyses. These include the expression
matrix along with the experimental design, and ensure the sample names
match between both when transforming/subsetting data.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset_object.html">get_dataset_object</a></span><span class="op">(</span><span class="st">"GSE46416"</span>,</span>
<span>                          type <span class="op">=</span> <span class="st">'se'</span><span class="op">)</span> <span class="co"># SummarizedExperiment is the default output type</span></span></code></pre></div>
<p>Note that the tidy format is less memory efficient but allows easy
visualization and exploration with <a href="https://ggplot2.tidyverse.org/index.html" class="external-link">ggplot2</a> and the rest
of the <a href="https://www.tidyverse.org/" class="external-link">tidyverse</a>.</p>
<p>To show how subsetting works, we’ll keep the “manic phase” data and
the <code>reference_subject_role</code>s, which refers to the control
samples in Gemma datasets.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check the levels of the disease factor</span></span>
<span><span class="va">dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">disease</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "bipolar disorder has_modifier euthymic phase"
[2] "reference subject role"                      
[3] "bipolar disorder has_modifier manic phase"   </code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Subset patients during manic phase and controls</span></span>
<span><span class="va">manic</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="va">dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">disease</span> <span class="op">==</span> <span class="st">"bipolar disorder,manic phase"</span> <span class="op">|</span> </span>
<span>        <span class="va">dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">disease</span> <span class="op">==</span> <span class="st">"reference subject role"</span><span class="op">]</span></span>
<span><span class="va">manic</span></span></code></pre></div>
<pre><code>class: SummarizedExperiment 
dim: 21407 10 
metadata(8): title abstract ... gemmaSuitabilityScore taxon
assays(1): counts
rownames(21407): 2315430 2315554 ... 7385683 7385696
rowData names(4): Probe GeneSymbol GeneName NCBIid
colnames(10): Control, 15 Control, 8 ... Control, 1_DE50 Control,
  2_DE40
colData names(3): factorValues disease block</code></pre>
<p>Let’s take a look at sample to sample correlation in our subset.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get Expression matrix</span></span>
<span><span class="va">manicExpr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">manic</span>, <span class="st">"counts"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">manicExpr</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">cor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span>col <span class="op">=</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>,border_color <span class="op">=</span> <span class="cn">NA</span>,angle_col <span class="op">=</span> <span class="fl">45</span>,fontsize <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="gemma.R_files/figure-html/boxplot-1.png" alt="Sample to sample correlations of bipolar patients during manic phase and controls." width="700"><p class="caption">
Sample to sample correlations of bipolar patients during manic phase and
controls.
</p>
</div>
<p>You can also use <a href="https://pavlidislab.github.io/gemma.R/reference/get_dataset_expression.html" class="external-link"><code>get_dataset_expression</code></a>
to only get the expression matrix, <a href="https://pavlidislab.github.io/gemma.R/reference/get_dataset_samples.html" class="external-link"><code>get_dataset_samples</code></a>
to get the metadata information. The output of this function can be
simplified to match the version included in the output of
<code>get_dataset_object</code> by using <a href="https://pavlidislab.github.io/gemma.R/reference/make_design" class="external-link"><code>make_design</code></a>
on the output.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_dataset_samples.html">get_dataset_samples</a></span><span class="op">(</span><span class="st">'GSE46416'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/make_design.html">make_design</a></span><span class="op">(</span><span class="st">'text'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>                                            factorValues
Bipolar disorder patient euthymic phase, 17 c("disea....
Bipolar disorder patient euthymic phase, 34 c("disea....
Control, 15                                 c("disea....
Bipolar disorder patient euthymic phase, 32 c("disea....
Control, 8                                  c("disea....
Bipolar disorder patient manic phase, 21    c("block....
                                                                                 disease
Bipolar disorder patient euthymic phase, 17 bipolar disorder has_modifier euthymic phase
Bipolar disorder patient euthymic phase, 34 bipolar disorder has_modifier euthymic phase
Control, 15                                                       reference subject role
Bipolar disorder patient euthymic phase, 32 bipolar disorder has_modifier euthymic phase
Control, 8                                                        reference subject role
Bipolar disorder patient manic phase, 21       bipolar disorder has_modifier manic phase
                                                        block
Bipolar disorder patient euthymic phase, 17 Batch_02_26/11/09
Bipolar disorder patient euthymic phase, 34 Batch_04_02/12/09
Control, 15                                 Batch_02_26/11/09
Bipolar disorder patient euthymic phase, 32 Batch_04_02/12/09
Control, 8                                  Batch_01_25/11/09
Bipolar disorder patient manic phase, 21    Batch_03_27/11/09</code></pre>
</div>
<div class="section level2">
<h2 id="platform-annotations">Platform Annotations<a class="anchor" aria-label="anchor" href="#platform-annotations"></a>
</h2>
<p>Expression data in Gemma comes with annotations for the gene each
expression profile corresponds to. Using the <a href="https://pavlidislab.github.io/gemma.R/reference/get_platform_annotations.html" class="external-link"><code>get_platform_annotations</code></a>
function, these annotations can be retrieved independently of the
expression data, along with additional annotations such as Gene Ontology
terms.</p>
<p>Examples:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platform_annotations.html">get_platform_annotations</a></span><span class="op">(</span><span class="st">'GPL96'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>     ProbeName   GeneSymbols                                     GeneNames
        &lt;char&gt;        &lt;char&gt;                                        &lt;char&gt;
1: 211750_x_at TUBA1C|TUBA1A             tubulin alpha 1c|tubulin alpha 1a
2:   216678_at                                                            
3:   216345_at        ZSWIM8            zinc finger SWIM-type containing 8
4:   207273_at                                                            
5: 216025_x_at        CYP2C9 cytochrome P450 family 2 subfamily C member 9
6: 218191_s_at        LMBRD1                     LMBR1 domain containing 1
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        GOTerms
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         &lt;char&gt;
1: GO:0045121|GO:0005634|GO:0005515|GO:0015630|GO:0000226|GO:0051301|GO:0016787|GO:0005737|GO:0005525|GO:0005874|GO:0030705|GO:0031982|GO:0007017|GO:0046872|GO:0005198|GO:0005200|GO:0000278|GO:0005881|GO:0001764|GO:0051402|GO:0072384|GO:0034612|GO:0044877|GO:0007613|GO:0007098|GO:0007224|GO:0071277|GO:0021542|GO:0009612|GO:0055037|GO:0005886|GO:0000793|GO:0048873|GO:0010001|GO:0042802|GO:0001964|GO:0021987|GO:0036464|GO:0006886|GO:0008542|GO:0050808|GO:0050807|GO:0070062|GO:0061744|GO:0048853|GO:0005879|GO:0008344|GO:0031594|GO:0010467|GO:0140058|GO:0021696|GO:0005829|GO:0046982|GO:0046785|GO:1902065
2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           GO:2000627|GO:0016567|GO:0031462|GO:0031463|GO:1990756|GO:0140958|GO:0043161|GO:0008270|GO:0005829
4:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
5:                                                                                                                                                                                                                                                              GO:0043231|GO:0016098|GO:0005737|GO:0016491|GO:0018676|GO:0008210|GO:0034875|GO:0005506|GO:0008405|GO:0008203|GO:0052741|GO:0006805|GO:0004497|GO:0008404|GO:0005789|GO:0018675|GO:0020037|GO:0019373|GO:0008392|GO:0070989|GO:0042178|GO:0005886|GO:0008202|GO:0043603|GO:0042759|GO:0016712|GO:0097267|GO:0070330|GO:0046456|GO:0032787|GO:0008395|GO:0019627
6:                                                                                                                                                                                                                                                                                                                                                                                                                                   GO:0061462|GO:0005515|GO:0035612|GO:0043231|GO:0005886|GO:0032050|GO:0031419|GO:0005158|GO:0140318|GO:0045334|GO:0005765|GO:0072583|GO:0030136|GO:0007369|GO:0016020|GO:0038016|GO:0005789
        GemmaIDs    NCBIids
          &lt;char&gt;     &lt;char&gt;
1: 360802|172797 84790|7846
2:                         
3:        235733      23053
4:                         
5:         32964       1559
6:        303717      55788</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platform_annotations.html">get_platform_annotations</a></span><span class="op">(</span><span class="st">'Generic_human_ncbiIds'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>   ElementName  GeneSymbols
         &lt;int&gt;       &lt;char&gt;
1:       55236         UBA6
2:       79664         ICE2
3:   100126270     FMR1-AS1
4:   105373684    LINC01818
5:   124900245 LOC124900245
6:   102725051 LOC102725051
                                               GeneNames
                                                  &lt;char&gt;
1:           ubiquitin like modifier activating enzyme 6
2: interactor of little elongation complex ELL subunit 2
3:                                  FMR1 antisense RNA 1
4:           long intergenic non-protein coding RNA 1818
5:                           small nucleolar RNA SNORA40
6:                          uncharacterized LOC102725051
                                                                                                                                                                           GOTerms
                                                                                                                                                                            &lt;char&gt;
1: GO:0005515|GO:0006511|GO:0032446|GO:0021766|GO:0019780|GO:0004792|GO:0007612|GO:0016567|GO:0005737|GO:0060996|GO:0005829|GO:0007626|GO:0005524|GO:0004839|GO:0016779|GO:0021764
2:                                                        GO:0016604|GO:0005515|GO:0008023|GO:0005654|GO:0000791|GO:0045945|GO:0042795|GO:0005829|GO:0042796|GO:0015030|GO:0035363
3:                                                                                                                                                                      GO:0043066
4:                                                                                                                                                                                
5:                                                                                                                                                                                
6:                                                                                                                                                                                
   GemmaIDs   NCBIids
      &lt;int&gt;     &lt;int&gt;
1:   295849     55236
2:   336840     79664
3:  3157248 100126270
4:  9235895 105373684
5: 10578422 124900245
6:  9008981 102725051</code></pre>
<p>If you are interested in a particular gene, you can see which
platforms include it using <a href="https://pavlidislab.github.io/gemma.R/reference/get_gene_probes.html" class="external-link"><code>get_gene_probes</code></a>.
Note that functions to search gene work best with unambigious
identifiers rather than symbols.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># lists genes in gemma matching the symbol or identifier</span></span>
<span><span class="fu"><a href="../reference/get_genes.html">get_genes</a></span><span class="op">(</span><span class="st">'Eno2'</span><span class="op">)</span></span></code></pre></div>
<pre><code>   gene.symbol       gene.ensembl gene.NCBI                      gene.name
        &lt;char&gt;             &lt;char&gt;     &lt;int&gt;                         &lt;char&gt;
1:        ENO2    ENSG00000111674      2026                      enolase 2
2:        Eno2 ENSMUSG00000004267     13807      enolase 2, gamma neuronal
3:        Eno2 ENSRNOG00000013141     24334                      enolase 2
4:        ENO2               &lt;NA&gt;    856579 phosphopyruvate hydratase ENO2
5:        eno2 ENSDARG00000014287    402874                      enolase 2
   gene.MFX.rank taxon.name         taxon.scientific taxon.ID taxon.NCBI
           &lt;num&gt;     &lt;char&gt;                   &lt;char&gt;    &lt;int&gt;      &lt;int&gt;
1:     0.9033096      human             Homo sapiens        1       9606
2:     0.8980022      mouse             Mus musculus        2      10090
3:     0.8822730        rat        Rattus norvegicus        3      10116
4:     0.6279932      yeast Saccharomyces cerevisiae       11       4932
5:     0.8830854  zebrafish              Danio rerio       12       7955
   taxon.database.name taxon.database.ID
                &lt;char&gt;             &lt;int&gt;
1:                hg38                87
2:                mm10                81
3:                 rn6                86
4:                &lt;NA&gt;                NA
5:                &lt;NA&gt;                NA</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ncbi id for human ENO2</span></span>
<span><span class="va">probes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gene_probes.html">get_gene_probes</a></span><span class="op">(</span><span class="fl">2026</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># remove the description for brevity of output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">probes</span><span class="op">[</span>,<span class="va">.SD</span>, .SDcols <span class="op">=</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">probes</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'mapping.Description'</span>,<span class="st">'platform.Description'</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code>   mapping.name          mapping.description platform.shortName
         &lt;char&gt;                       &lt;char&gt;             &lt;char&gt;
1:    201313_at  enolase 2 (gamma, neuronal)               &lt;NA&gt;
2:    201313_at  enolase 2 (gamma, neuronal)               &lt;NA&gt;
3:     40193_at  enolase 2 (gamma, neuronal)               &lt;NA&gt;
4:         1639                         &lt;NA&gt;               &lt;NA&gt;
5:         6621                         &lt;NA&gt;               &lt;NA&gt;
6:          861                         &lt;NA&gt;               &lt;NA&gt;
                                                       platform.name
                                                              &lt;char&gt;
1:          Affymetrix GeneChip Human Genome U133 Array Set HG-U133A
2:              Affymetrix GeneChip Human Genome U133 Plus 2.0 Array
3: Affymetrix GeneChip Human Genome U95 Version [1 or 2] Set HG-U95A
4:                                                        CHUGAI 41K
5:                                                            UP_5Ka
6:                                                            UP_5Ka
   platform.ID platform.type
         &lt;int&gt;        &lt;char&gt;
1:           1      ONECOLOR
2:           4      ONECOLOR
3:           8      ONECOLOR
4:          36      TWOCOLOR
5:          38      TWOCOLOR
6:          38      TWOCOLOR
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             platform.description
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           &lt;char&gt;
1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      The U133 set includes 2 arrays with a total of 44928 entries and was indexed 29-Jan-2002. The set includes over 1,000,000 unique oligonucleotide features covering more than 39,000 transcript variants, which in turn represent greater than 33,000 of the best characterized human genes. Sequences were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from Build 133 of UniGene (April 20, 2001) and refined by analysis and comparison with a number of other publicly available databases including the Washington University EST trace repository and the University of California, Santa Cruz golden-path human genome database (April 2001 release). In addition, ESTs were analyzed for untrimmed low-quality sequence information, correct orientation, false priming, false clustering, alternative splicing and alternative polyadenylation.  Keywords = high density oligonucleotide array\nFrom GPL96\nLast Updated: Mar 09 2006
2:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                Complete coverage of the Human Genome U133 Set plus 6,500 additional genes for analysis of over 47,000 transcripts All probe sets represented on the GeneChip Human Genome U133 Set are identically replicated on the GeneChip Human Genome U133 Plus 2.0 Array. The sequences from which these probe sets were derived were selected from GenBank®, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 133, April 20, 2001) and then refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the University of California, Santa Cruz Golden-Path human genome database (April 2001 release). In addition, there are 9,921 new probe sets representing approximately 6,500 new genes. These gene sequences were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from the UniGene database (Build 159, January 25, 2003) and refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the NCBI human genome assembly (Build 31). \nFrom GPL570\nLast Updated: Mar 09 2006
3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             The Human Genome U95 (HG-U95) Set, consisting of five GeneChip arrays, contains almost 63,000 probe sets interrogating approximately 54,000 clusters derived from the UniGene database (Build 95). Based on this build and associated annotations, the HG-U95Av2 Array represents approximately 10,000 full-length genes, while arrays B through E represent EST clusters.  This platform can also be used to submit data for the HG_U95A [version 1] array.  Keywords = high density oligonucleotide array\nFrom GPL91\nLast Updated: Mar 09 2006
4:  Patients and Sample Preparation. We analyzed NT samples from 38 patients undergoing surgery at the Cancer Institute Hospital, Tokyo, during May 1996 ? May 2001. All samples were collected with informed consent and ethical committee approval and were grossly dissected, snap-frozen in liquid nitrogen within 20 minutes of removal. Initial diagnosis of each sample  from frozen section was later confirmed by detailed analysis of paraffin sections and immunohistochemistry. Following WHO guidelines,1 17 tumours were classified as SCLC including two combined SCLC with adenocarcinoma (SCLC-AD), 8 as LCNEC, including 2 combined LCNEC with AD (LCNEC-AD), 12 as TC and 1 as AC. The prevalence of LCNEC as a percentage all lung cancer resected at the Cancer Institute during the period was 2.7%. Since LCNEC is described as a variant of large cell carcinoma MAGE(LCC), we also analyzed thirteen LCC samples. Twelve well differentiated  AD samples were included to aid interpretation of expression profiles for the SCLC-AD and LCNEC-AD samples, and to allow identification of markers that distinguish NT from non-small cell lung cancer. We included thirty normal lung samples from unrelated individuals. Finally 11 SCLC cell lines were incorporated into the analysis. Full clinical details of all tumour samples and cell lines are provided in supplementary table 1. Reference RNA, a mixture of lung cancer cell lines and normal lung, is exactly as described previously.5 Cell lines were obtained from the American Type Culture Collection. the Japanese Collection of Research Bioresources, or The RIKEN cell bank. cDNA microarray Each microarray contains 40,368 elements, of which 39,936 are derived from IMAGE cDNA clones purchased from Research Genetics (Huntsville, Alabama). This clone set was supplemented with 384 proprietary clones and 48 control elements. According to our latest estimate, the array represents 29,487 unique UNIGENE clusters accounting for 9,085 named genes, 15,276 ESTs and 5126 other un-named genes (for example, clones described as hypothetical proteins). Clones were purchased as sequence verified, but we estimate an error rate of at least 10%. Genes mentioned by name in this paper have been re-sequenced. cDNA PCR products were resuspended in 50% DMSO and printed at a pitch of 150 µm onto CMT-GAPS slides (Corning, Corning, NY) using a MicroGrid II spotting robot fitted with 10K pins (BioRobotics, Cambridge, UK). Relative humidity was controlled within the range 40-50% and temperature between 20-25°C.  cDNA derived from 2µg of cRNA, essentially as described,7 was aminoallyl labeled8 with Cy5 (sample) or Cy3 (reference), hybridized overnight at 42ºC in a buffer containing 50% formamide, 5xSSC, 0.1% SDS, 0.25µg/µl human cot1 DNA and 0.125µg/µl poly dT and washed to a final stringency of 1x SSC at 42°C. After washing, slides were immediately scanned using an Axon GenePix 4000B scanner and quantified using Axon GenePix Pro software (Axon, Union city, CA).  Raw data for all 40,386 elements across all 105 samples assayed here is deposited with\nFrom GPL962\nLast Updated: Sep 08 2006
5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     A cDNA microarray with ~5000 human genes and ESTs.\nFrom GPL230\nLast Updated: May 26 2005
6:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     A cDNA microarray with ~5000 human genes and ESTs.\nFrom GPL230\nLast Updated: May 26 2005
   platform.troubled taxon.name taxon.scientific taxon.ID taxon.NCBI
              &lt;lgcl&gt;     &lt;char&gt;           &lt;char&gt;    &lt;int&gt;      &lt;int&gt;
1:             FALSE      human     Homo sapiens        1       9606
2:             FALSE      human     Homo sapiens        1       9606
3:             FALSE      human     Homo sapiens        1       9606
4:             FALSE      human     Homo sapiens        1       9606
5:             FALSE      human     Homo sapiens        1       9606
6:             FALSE      human     Homo sapiens        1       9606
   taxon.database.name taxon.database.ID
                &lt;char&gt;             &lt;int&gt;
1:                hg38                87
2:                hg38                87
3:                hg38                87
4:                hg38                87
5:                hg38                87
6:                hg38                87</code></pre>
</div>
<div class="section level2">
<h2 id="differential-expression-analyses">Differential expression analyses<a class="anchor" aria-label="anchor" href="#differential-expression-analyses"></a>
</h2>
<p>Gemma contains precomputed differential expression analyses for most
of its datasets. Analyses can involve more than one factor, such as
“sex” as well as “disease”. Some datasets contain more than one analysis
to account for different factors and their interactions. The results are
stored as resultSets, each corresponding to one factor (or their
interaction). You can access them using <a href="https://pavlidislab.github.io/gemma.R/reference/get_differential_expression_values.html" class="external-link"><code>get_differential_expression_values</code></a>.
From here on, we can explore and visualize the data to find the most
differentially-expressed genes</p>
<p>Note that <code>get_differential_expression_values</code> can return
multiple differentials per study if a study has multiple factors to
contrast. Since GSE46416 only has one extracting the first element of
the returned list is all we need.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dif_exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_differential_expression_values.html">get_differential_expression_values</a></span><span class="op">(</span><span class="st">'GSE46416'</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">dif_exp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code>         Probe    NCBIid GeneSymbol
         &lt;int&gt;    &lt;char&gt;     &lt;char&gt;
    1: 2982730      4018        LPA
    2: 2787851    166752      FREM3
    3: 2477558                     
    4: 2910917                     
    5: 3983537    140886     PABPC5
   ---                             
21957: 3301011     64318      NOC3L
21958: 2461654 100130249  LINC02961
21959: 2360346      1141     CHRNB2
21960: 2391172      7293    TNFRSF4
21961: 2525718                     
                                            GeneName pvalue corrected_pvalue
                                              &lt;char&gt;  &lt;num&gt;            &lt;num&gt;
    1:                                lipoprotein(a) 0.9185           0.9521
    2:          FRAS1 related extracellular matrix 3 0.5850           0.7348
    3:                                               0.3965           0.5931
    4:                                               0.6506           0.7815
    5:         poly(A) binding protein cytoplasmic 5 0.1064           0.3227
   ---                                                                      
21957:           NOC3 like DNA replication regulator 0.0636           0.2647
21958:   long intergenic non-protein coding RNA 2961 0.2547           0.4735
21959: cholinergic receptor nicotinic beta 2 subunit 0.0121           0.1288
21960:             TNF receptor superfamily member 4 0.1314           0.3516
21961:                                               0.7526           0.8463
         rank contrast_113004_log2fc contrast_113004_tstat
        &lt;num&gt;                  &lt;num&gt;                 &lt;num&gt;
    1: 0.9647                -0.0247               -0.3587
    2: 0.7960                 0.1895                1.0119
    3: 0.6685                 0.0918                1.1386
    4: 0.8325                 0.1564                0.6739
    5: 0.3299                 0.1845                2.0320
   ---                                                    
21957: 0.2402                 0.1481                0.7178
21958: 0.5380                -0.1763               -1.5357
21959: 0.0942                 0.1029                1.1227
21960: 0.3737                -0.0086               -0.0648
21961: 0.8893                -0.1105               -0.5960
       contrast_113004_pvalue contrast_113005_log2fc contrast_113005_tstat
                        &lt;num&gt;                  &lt;num&gt;                 &lt;num&gt;
    1:                 0.7221                -0.0249               -0.3622
    2:                 0.3191                 0.1403                0.7495
    3:                 0.2633                 0.1016                1.2604
    4:                 0.5052                 0.2096                0.9032
    5:                 0.0505                 0.1603                1.7656
   ---                                                                    
21957:                 0.4781                -0.3343               -1.6208
21958:                 0.1344                -0.0235               -0.2045
21959:                 0.2698                 0.2867                3.1278
21960:                 0.9487                 0.2312                1.7369
21961:                 0.5553                -0.1316               -0.7101
       contrast_113005_pvalue
                        &lt;num&gt;
    1:                 0.7196
    2:                 0.4590
    3:                 0.2166
    4:                 0.3731
    5:                 0.0869
   ---                       
21957:                 0.1148
21958:                 0.8393
21959:                 0.0037
21960:                 0.0920
21961:                 0.4827</code></pre>
<p>By default the columns names of the output correspond to contrast
IDs. To see what conditions these IDs correspond to we can either use
<code>get_dataset_differential_expression_analyses</code> to get the
metadata about differentials of a given dataset, or set
<code>readableContrasts</code> argument of
<code>get_differential_expression_values</code> to <code>TRUE</code>.
The former approach is usually better for a large scale systematic
analysis while the latter is easier to read in an interactive
session.</p>
<p><code>get_dataset_differential_expression_analyses</code> function
returns structured metadata about the differentials.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset_differential_expression_analyses.html">get_dataset_differential_expression_analyses</a></span><span class="op">(</span><span class="st">'GSE46416'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>   result.ID contrast.ID experiment.ID factor.category
       &lt;int&gt;       &lt;int&gt;         &lt;int&gt;          &lt;char&gt;
1:    550248      113004          8997         disease
2:    550248      113005          8997         disease
                    factor.category.URI factor.ID baseline.factors
                                 &lt;char&gt;     &lt;int&gt;           &lt;list&gt;
1: http://www.ebi.ac.uk/efo/EFO_0000408     19134     disease,....
2: http://www.ebi.ac.uk/efo/EFO_0000408     19134     disease,....
   experimental.factors subsetFactor.subset subsetFactor probes.analyzed
                 &lt;list&gt;              &lt;lgcl&gt;       &lt;list&gt;           &lt;int&gt;
1:         disease,....               FALSE logical(....           21961
2:         disease,....               FALSE logical(....           21961
   genes.analyzed
            &lt;int&gt;
1:          18959
2:          18959</code></pre>
<p><code>contrast.ID</code> column corresponds to the column names in
the output of <code>get_differential_expression_values</code> while
<code>result.ID</code> corresponds to the name of the differential in
the output object. Using them together will let one to access
differentially expressed gene counts for each condition contrast</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># using result.ID and contrast.ID of the output above, we can access specific</span></span>
<span><span class="co"># results. Note that one study may have multiple contrast objects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">contrasts</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">result_set</span> <span class="op">=</span> <span class="va">dif_exp</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">contrasts</span><span class="op">[</span><span class="va">i</span>,<span class="op">]</span><span class="op">$</span><span class="va">result.ID</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">p_values</span> <span class="op">=</span> <span class="va">result_set</span><span class="op">[[</span><span class="fu">glue</span><span class="fu">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html" class="external-link">glue</a></span><span class="op">(</span><span class="st">"contrast_{contrasts[i,]$contrast.ID}_pvalue"</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span>    </span>
<span>    <span class="co"># multiple testing correction</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">p_values</span>,method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">dif_exp_genes</span></span>
<span></span>
<span><span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span>result.ID <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">result.ID</span>,</span>
<span>                        contrast.id <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">contrast.ID</span>,</span>
<span>                        baseline.factorValue <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">baseline.factors</span>,</span>
<span>                        experimental.factorValue <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">experimental.factors</span>,</span>
<span>                        n_diff <span class="op">=</span> <span class="va">dif_exp_genes</span><span class="op">)</span></span>
<span></span>
<span><span class="va">contrasts</span></span></code></pre></div>
<pre><code>   result.ID contrast.id baseline.factorValue experimental.factorValue n_diff
       &lt;int&gt;       &lt;int&gt;               &lt;list&gt;                   &lt;list&gt;  &lt;int&gt;
1:    550248      113004         disease,....             disease,....      3
2:    550248      113005         disease,....             disease,....   1410</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrasts</span><span class="op">$</span><span class="va">baseline.factorValue</span></span></code></pre></div>
<pre><code>[[1]]
   category                         category.URI                  value
     &lt;char&gt;                               &lt;char&gt;                 &lt;char&gt;
1:  disease http://www.ebi.ac.uk/efo/EFO_0000408 reference subject role
                                    value.URI predicate predicate.URI object
                                       &lt;char&gt;    &lt;lgcl&gt;        &lt;lgcl&gt; &lt;lgcl&gt;
1: http://purl.obolibrary.org/obo/OBI_0000220        NA            NA     NA
   object.URI                summary     ID factor.ID factor.category
       &lt;lgcl&gt;                 &lt;char&gt;  &lt;int&gt;     &lt;int&gt;          &lt;char&gt;
1:         NA reference subject role 113006     19134         disease
                    factor.category.URI
                                 &lt;char&gt;
1: http://www.ebi.ac.uk/efo/EFO_0000408

[[2]]
   category                         category.URI                  value
     &lt;char&gt;                               &lt;char&gt;                 &lt;char&gt;
1:  disease http://www.ebi.ac.uk/efo/EFO_0000408 reference subject role
                                    value.URI predicate predicate.URI object
                                       &lt;char&gt;    &lt;lgcl&gt;        &lt;lgcl&gt; &lt;lgcl&gt;
1: http://purl.obolibrary.org/obo/OBI_0000220        NA            NA     NA
   object.URI                summary     ID factor.ID factor.category
       &lt;lgcl&gt;                 &lt;char&gt;  &lt;int&gt;     &lt;int&gt;          &lt;char&gt;
1:         NA reference subject role 113006     19134         disease
                    factor.category.URI
                                 &lt;char&gt;
1: http://www.ebi.ac.uk/efo/EFO_0000408</code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrasts</span><span class="op">$</span><span class="va">experimental.factorValue</span></span></code></pre></div>
<pre><code>[[1]]
   category                         category.URI            value
     &lt;char&gt;                               &lt;char&gt;           &lt;char&gt;
1:  disease http://www.ebi.ac.uk/efo/EFO_0000408 bipolar disorder
                                      value.URI    predicate
                                         &lt;char&gt;       &lt;char&gt;
1: http://purl.obolibrary.org/obo/MONDO_0004985 has_modifier
                               predicate.URI      object object.URI
                                      &lt;char&gt;      &lt;char&gt;     &lt;char&gt;
1: http://purl.obolibrary.org/obo/RO_0002573 manic phase       &lt;NA&gt;
                                     summary     ID factor.ID factor.category
                                      &lt;char&gt;  &lt;int&gt;     &lt;int&gt;          &lt;char&gt;
1: bipolar disorder has_modifier manic phase 113004     19134         disease
                    factor.category.URI
                                 &lt;char&gt;
1: http://www.ebi.ac.uk/efo/EFO_0000408

[[2]]
   category                         category.URI            value
     &lt;char&gt;                               &lt;char&gt;           &lt;char&gt;
1:  disease http://www.ebi.ac.uk/efo/EFO_0000408 bipolar disorder
                                      value.URI    predicate
                                         &lt;char&gt;       &lt;char&gt;
1: http://purl.obolibrary.org/obo/MONDO_0004985 has_modifier
                               predicate.URI         object object.URI
                                      &lt;char&gt;         &lt;char&gt;     &lt;char&gt;
1: http://purl.obolibrary.org/obo/RO_0002573 euthymic phase       &lt;NA&gt;
                                        summary     ID factor.ID
                                         &lt;char&gt;  &lt;int&gt;     &lt;int&gt;
1: bipolar disorder has_modifier euthymic phase 113005     19134
   factor.category                  factor.category.URI
            &lt;char&gt;                               &lt;char&gt;
1:         disease http://www.ebi.ac.uk/efo/EFO_0000408</code></pre>
<p>Alternatively we, since we are only looking at one dataset and one
contrast manually, we can simply use <code>readableContrasts</code>.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">de</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_differential_expression_values.html">get_differential_expression_values</a></span><span class="op">(</span><span class="st">"GSE46416"</span>,readableContrasts <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code>         Probe    NCBIid GeneSymbol
         &lt;int&gt;    &lt;char&gt;     &lt;char&gt;
    1: 2982730      4018        LPA
    2: 2787851    166752      FREM3
    3: 2477558                     
    4: 2910917                     
    5: 3983537    140886     PABPC5
   ---                             
21957: 3301011     64318      NOC3L
21958: 2461654 100130249  LINC02961
21959: 2360346      1141     CHRNB2
21960: 2391172      7293    TNFRSF4
21961: 2525718                     
                                            GeneName pvalue corrected_pvalue
                                              &lt;char&gt;  &lt;num&gt;            &lt;num&gt;
    1:                                lipoprotein(a) 0.9185           0.9521
    2:          FRAS1 related extracellular matrix 3 0.5850           0.7348
    3:                                               0.3965           0.5931
    4:                                               0.6506           0.7815
    5:         poly(A) binding protein cytoplasmic 5 0.1064           0.3227
   ---                                                                      
21957:           NOC3 like DNA replication regulator 0.0636           0.2647
21958:   long intergenic non-protein coding RNA 2961 0.2547           0.4735
21959: cholinergic receptor nicotinic beta 2 subunit 0.0121           0.1288
21960:             TNF receptor superfamily member 4 0.1314           0.3516
21961:                                               0.7526           0.8463
         rank contrast_bipolar disorder has_modifier manic phase_logFoldChange
        &lt;num&gt;                                                            &lt;num&gt;
    1: 0.9647                                                          -0.0247
    2: 0.7960                                                           0.1895
    3: 0.6685                                                           0.0918
    4: 0.8325                                                           0.1564
    5: 0.3299                                                           0.1845
   ---                                                                        
21957: 0.2402                                                           0.1481
21958: 0.5380                                                          -0.1763
21959: 0.0942                                                           0.1029
21960: 0.3737                                                          -0.0086
21961: 0.8893                                                          -0.1105
       contrast_bipolar disorder has_modifier manic phase_tstat
                                                          &lt;num&gt;
    1:                                                  -0.3587
    2:                                                   1.0119
    3:                                                   1.1386
    4:                                                   0.6739
    5:                                                   2.0320
   ---                                                         
21957:                                                   0.7178
21958:                                                  -1.5357
21959:                                                   1.1227
21960:                                                  -0.0648
21961:                                                  -0.5960
       contrast_bipolar disorder has_modifier manic phase_pvalue
                                                           &lt;num&gt;
    1:                                                    0.7221
    2:                                                    0.3191
    3:                                                    0.2633
    4:                                                    0.5052
    5:                                                    0.0505
   ---                                                          
21957:                                                    0.4781
21958:                                                    0.1344
21959:                                                    0.2698
21960:                                                    0.9487
21961:                                                    0.5553
       contrast_bipolar disorder has_modifier euthymic phase_logFoldChange
                                                                     &lt;num&gt;
    1:                                                             -0.0249
    2:                                                              0.1403
    3:                                                              0.1016
    4:                                                              0.2096
    5:                                                              0.1603
   ---                                                                    
21957:                                                             -0.3343
21958:                                                             -0.0235
21959:                                                              0.2867
21960:                                                              0.2312
21961:                                                             -0.1316
       contrast_bipolar disorder has_modifier euthymic phase_tstat
                                                             &lt;num&gt;
    1:                                                     -0.3622
    2:                                                      0.7495
    3:                                                      1.2604
    4:                                                      0.9032
    5:                                                      1.7656
   ---                                                            
21957:                                                     -1.6208
21958:                                                     -0.2045
21959:                                                      3.1278
21960:                                                      1.7369
21961:                                                     -0.7101
       contrast_bipolar disorder has_modifier euthymic phase_pvalue
                                                              &lt;num&gt;
    1:                                                       0.7196
    2:                                                       0.4590
    3:                                                       0.2166
    4:                                                       0.3731
    5:                                                       0.0869
   ---                                                             
21957:                                                       0.1148
21958:                                                       0.8393
21959:                                                       0.0037
21960:                                                       0.0920
21961:                                                       0.4827</code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Classify probes for plotting</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">&lt;-</span> <span class="st">"No"</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span> <span class="op">&gt;</span> <span class="fl">1.0</span> <span class="op">&amp;</span> </span>
<span>        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Up"</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">1.0</span> <span class="op">&amp;</span> </span>
<span>        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Down"</span></span>
<span></span>
<span><span class="co"># Upregulated probes</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Up"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span>, </span>
<span>        <span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code>      Probe GeneSymbol
      &lt;int&gt;     &lt;char&gt;
 1: 2319550       RBP7
 2: 2548699     CYP1B1
 3: 3907190       SLPI
 4: 3629103      PCLAF
 5: 3545525      SLIRP
 6: 3146433      COX6C
 7: 2899102       H3C3
 8: 2538349           
 9: 3635198     BCL2A1
10: 2633191      GPR15
    contrast_bipolar disorder has_modifier manic phase_pvalue
                                                        &lt;num&gt;
 1:                                                 8.612e-05
 2:                                                 1.000e-04
 3:                                                 3.000e-04
 4:                                                 5.000e-04
 5:                                                 6.000e-04
 6:                                                 9.000e-04
 7:                                                 1.300e-03
 8:                                                 1.300e-03
 9:                                                 1.800e-03
10:                                                 2.400e-03
    contrast_bipolar disorder has_modifier manic phase_logFoldChange
                                                               &lt;num&gt;
 1:                                                           1.0740
 2:                                                           1.3225
 3:                                                           1.0558
 4:                                                           1.2783
 5:                                                           1.3490
 6:                                                           1.4670
 7:                                                           1.0260
 8:                                                           1.0731
 9:                                                           1.0798
10:                                                           1.2046</code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Downregulated probes</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Down"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span>, </span>
<span>        <span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code>      Probe GeneSymbol
      &lt;int&gt;     &lt;char&gt;
 1: 2775390           
 2: 3760268           
 3: 3124344           
 4: 3673179           
 5: 3245871      WDFY4
 6: 3022689   SND1-IT1
 7: 2679014           
 8: 4019758           
 9: 3336402      RBM14
10: 3930525  RUNX1-IT1
    contrast_bipolar disorder has_modifier manic phase_pvalue
                                                        &lt;num&gt;
 1:                                                 2.095e-06
 2:                                                 1.153e-05
 3:                                                 1.000e-04
 4:                                                 2.000e-04
 5:                                                 2.000e-04
 6:                                                 2.000e-04
 7:                                                 3.000e-04
 8:                                                 4.000e-04
 9:                                                 4.000e-04
10:                                                 4.000e-04
    contrast_bipolar disorder has_modifier manic phase_logFoldChange
                                                               &lt;num&gt;
 1:                                                          -1.5558
 2:                                                          -1.8506
 3:                                                          -1.0370
 4:                                                          -1.0340
 5:                                                          -1.1569
 6:                                                          -1.2199
 7:                                                          -1.1752
 8:                                                          -1.4049
 9:                                                          -1.0711
10:                                                          -1.5169</code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add gene symbols as labels to DE genes</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">delabel</span> <span class="op">&lt;-</span> <span class="st">""</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">delabel</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">de</span><span class="op">$</span><span class="va">GeneSymbol</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Volcano plot for bipolar patients vs controls</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">de</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        x <span class="op">=</span> <span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span>,</span>
<span>        y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span><span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="va">diffexpr</span>,</span>
<span>        label <span class="op">=</span> <span class="va">delabel</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1.0</span>, <span class="fl">1.0</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"log2(FoldChange)"</span>, y <span class="op">=</span> <span class="st">"-log10(p-value)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggrepel.slowkow.com/reference/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="gemma.R_files/figure-html/diffExpr-1.png" alt="Differentially-expressed genes in bipolar patients during manic phase versus controls." width="700"><p class="caption">
Differentially-expressed genes in bipolar patients during manic phase
versus controls.
</p>
</div>
</div>
<div class="section level2">
<h2 id="larger-queries">Larger queries<a class="anchor" aria-label="anchor" href="#larger-queries"></a>
</h2>
<p>To query large amounts of data, the API has a pagination system which
uses the <code>limit</code> and <code>offset</code> parameters. To avoid
overloading the server, calls are limited to a maximum of 100 entries,
so the offset allows you to get the next batch of entries in the next
call(s).</p>
<p>To simplify the process of accessing all available data, gemma.R
includes the <a href="https://pavlidislab.github.io/gemma.R/reference/get_all_pages.html" class="external-link"><code>get_all_pages</code></a>
function which can use the output from one page to make all the follow
up requests</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_platforms_by_ids.html">get_platforms_by_ids</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/get_all_pages.html">get_all_pages</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>     platform.ID platform.shortName
           &lt;int&gt;             &lt;char&gt;
  1:           1              GPL96
  2:           2            GPL1355
  3:           3            GPL1261
  4:           4             GPL570
  5:           5              GPL81
 ---                               
666:        1325            GPL5621
667:        1326           GPL10732
668:        1327           GPL11044
669:        1329           GPL24973
670:        1330           GPL33408
                                                                 platform.name
                                                                        &lt;char&gt;
  1:                  Affymetrix GeneChip Human Genome U133 Array Set HG-U133A
  2:                              Affymetrix GeneChip Rat Genome 230 2.0 Array
  3:                            Affymetrix GeneChip Mouse Genome 430 2.0 Array
  4:                      Affymetrix GeneChip Human Genome U133 Plus 2.0 Array
  5:               Affymetrix GeneChip Murine Genome U74 Version 2 Set MG-U74A
 ---                                                                          
666:                                                SFGF HEEBO Human Set v1.00
667:                                               Mouse-Agilent-22K-D20030711
668: Affymetrix GeneChip Mouse Genome 430 2.0 Array [CDF: Mouse4302_Mm_REFSEQ]
669:                                                     MinION (Mus musculus)
670:                                          NextSeq 1000 (Rattus norvegicus)
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         platform.description
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       &lt;char&gt;
  1:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                The U133 set includes 2 arrays with a total of 44928 entries and was indexed 29-Jan-2002. The set includes over 1,000,000 unique oligonucleotide features covering more than 39,000 transcript variants, which in turn represent greater than 33,000 of the best characterized human genes. Sequences were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from Build 133 of UniGene (April 20, 2001) and refined by analysis and comparison with a number of other publicly available databases including the Washington University EST trace repository and the University of California, Santa Cruz golden-path human genome database (April 2001 release). In addition, ESTs were analyzed for untrimmed low-quality sequence information, correct orientation, false priming, false clustering, alternative splicing and alternative polyadenylation.  Keywords = high density oligonucleotide array\nFrom GPL96\nLast Updated: Mar 09 2006
  2:  The GeneChip Rat Genome 230 2.0 Array is a powerful tool for toxicology, neurobiology, and other applications using rat as a model organism.  - Provides comprehensive coverage of the transcribed rat genome on a single array - Comprised of more than 31,000 probe sets, analyzing over 30,000 transcripts and variants from over 28,000 well-substantiated rat genes - The publicly available draft of the rat genome and leading public rat databases were used to refine sequences and provide a higher quality of data output  All probe sets represented on the GeneChip Rat Expression Set 230 are included on the GeneChip Rat Genome 230 2.0 Array. Sequences used in the design of the GeneChip Rat Genome 230 2.0 Array were selected from GenBank, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 99, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the rat genome from the Baylor College of Medicine Human Genome Sequencing Center (June 2002).  The GeneChip Rat Genome 230 2.0 Array includes representation of the RefSeq database sequences and probe sets related to sequences and refined EST clusters previously represented on the GeneChip Rat Genome U34 Set.  Oligonucleotide probes complementary to each corresponding sequence are synthesized in situ on the arrays. Eleven pairs of oligonucleotide probes are used to measure the level of transcription of each sequence represented on the GeneChip Rat Genome 230 2.0 Array.  Annotations derived from Affymetrix CSV file dated 6/23/2004\nFrom GPL1355\nLast Updated: May 31 2005
  3:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        All probe sets represented on the GeneChip Mouse Expression Set 430 are included on the GeneChip Mouse Genome 430 2.0 Array. The sequences from which these probe sets were derived were selected from GenBank«, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 107, June 2002) and then refined by analysis and comparison with the publicly available draft assembly of the mouse genome from the Whitehead Institute for Genome Research (MGSC, April 2002). \nFrom GPL1261\nLast Updated: Oct 28 2005
  4:                                                                                                                                                                                                                                                                                                                                                                                                                                          Complete coverage of the Human Genome U133 Set plus 6,500 additional genes for analysis of over 47,000 transcripts All probe sets represented on the GeneChip Human Genome U133 Set are identically replicated on the GeneChip Human Genome U133 Plus 2.0 Array. The sequences from which these probe sets were derived were selected from GenBank®, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 133, April 20, 2001) and then refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the University of California, Santa Cruz Golden-Path human genome database (April 2001 release). In addition, there are 9,921 new probe sets representing approximately 6,500 new genes. These gene sequences were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from the UniGene database (Build 159, January 25, 2003) and refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the NCBI human genome assembly (Build 31). \nFrom GPL570\nLast Updated: Mar 09 2006
  5:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            The MG-U74 set includes 3 arrays with a total of 36899 entries and was indexed 29-Jan-2002. The set represents ~36,000 full length genes and EST clusters derived from sequence clusters in Build 74 of the Mouse Unigene Database. MG-U74A represents all sequences (~6,000) in the Mouse UniGene database (Build 74) that have been functionally characterized, as well as ~6,000 EST clusters.  This platform can also be used to submit data for the MG-U74A [version 1] array.  Keywords = high density oligonucleotide array\r\nFrom GPL81\r\nLast Updated: Nov 17 2005
 ---                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
666:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            Probes: Constitutive Exonic Probes (30,718): that will recognize all known transcripts of a gene. Alternatively Spliced / Skipped Exonic Probes (8,441): will recognize exons that are present in some, but not all transcripts of a gene. Non Coding RNA Probes (196): recogning non-protein coding transcripts (ribosomal RNAs, miRNAs). BCR / TCR Genic / Regional Probes (372): recognizing transcripts from genes that undergo somatic rearrangement. Other Probes (843): recognizing human mitochondrion derived DNA sequences.   Controls: A total of 4,189 controls are included in the set, including Negative Controls (398 and 194 buffer only): Buffer only and 410 random sequences are positioned throughout the set to assist in determining background. Positive Controls (864): recognizing a small subset of human transcripts. Doped Controls (1,384): recognizing non-human sequences that can be spiked into RNA samples
667:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
668:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               This identical to GPL1261 but a custom CDF environment was used to extract data.  The .cdf and mapping files, as well as an R package are provided as supplementary files.
669:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
670:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
     platform.troubled platform.experimentCount platform.type taxon.name
                &lt;lgcl&gt;                    &lt;int&gt;        &lt;char&gt;     &lt;char&gt;
  1:             FALSE                      401      ONECOLOR      human
  2:             FALSE                      295      ONECOLOR        rat
  3:             FALSE                     1308      ONECOLOR      mouse
  4:             FALSE                     1499      ONECOLOR      human
  5:             FALSE                      201      ONECOLOR      mouse
 ---                                                                    
666:             FALSE                        0      TWOCOLOR      human
667:             FALSE                        0      ONECOLOR      mouse
668:             FALSE                        0      ONECOLOR      mouse
669:             FALSE                        0    SEQUENCING      mouse
670:             FALSE                        0    SEQUENCING        rat
      taxon.scientific taxon.ID taxon.NCBI taxon.database.name
                &lt;char&gt;    &lt;int&gt;      &lt;int&gt;              &lt;char&gt;
  1:      Homo sapiens        1       9606                hg38
  2: Rattus norvegicus        3      10116                 rn6
  3:      Mus musculus        2      10090                mm10
  4:      Homo sapiens        1       9606                hg38
  5:      Mus musculus        2      10090                mm10
 ---                                                          
666:      Homo sapiens        1       9606                hg38
667:      Mus musculus        2      10090                mm10
668:      Mus musculus        2      10090                mm10
669:      Mus musculus        2      10090                mm10
670: Rattus norvegicus        3      10116                 rn6
     taxon.database.ID
                 &lt;int&gt;
  1:                87
  2:                86
  3:                81
  4:                87
  5:                81
 ---                  
666:                87
667:                81
668:                81
669:                81
670:                86</code></pre>
<p>Alternative way to access all pages is to do so manually. To see how
many available results are there, you can look at the attributes of the
output objects where additional information from the API response is
appended.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">platform_count</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platforms_by_ids.html">get_platforms_by_ids</a></span><span class="op">(</span>limit <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">totalElements</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">platform_count</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 670</code></pre>
<p>After which you can use <code>offset</code> to access all available
platforms.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">platform_count</span>,<span class="fl">100</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">offset</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/get_platforms_by_ids.html">get_platforms_by_ids</a></span><span class="op">(</span>limit <span class="op">=</span> <span class="fl">100</span>, offset <span class="op">=</span> <span class="va">offset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">platform.ID</span>, <span class="va">platform.shortName</span>, <span class="va">taxon.name</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>,<span class="va">.</span><span class="op">)</span></span></code></pre></div>
<pre><code>     platform.ID platform.shortName taxon.name
           &lt;int&gt;             &lt;char&gt;     &lt;char&gt;
  1:           1              GPL96      human
  2:           2            GPL1355        rat
  3:           3            GPL1261      mouse
  4:           4             GPL570      human
  5:           5              GPL81      mouse
 ---                                          
666:        1325            GPL5621      human
667:        1326           GPL10732      mouse
668:        1327           GPL11044      mouse
669:        1329           GPL24973      mouse
670:        1330           GPL33408        rat</code></pre>
<p>Many endpoints only support a single identifier:</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_dataset_annotations.html">get_dataset_annotations</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE46416"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>Error: Please specify one valid identifier for dataset.</code></pre>
<p>In these cases, you will have to loop over all the identifiers you
wish to query and send separate requests.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE12649"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/get_dataset_annotations.html">get_dataset_annotations</a></span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>experiment.shortName <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">class.name</span>, <span class="va">term.name</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>,<span class="va">.</span><span class="op">)</span></span></code></pre></div>
<pre><code>   experiment.shortName     class.name           term.name
                 &lt;char&gt;         &lt;char&gt;              &lt;char&gt;
1:             GSE35974 biological sex              female
2:             GSE35974        disease depressive disorder
3:             GSE35974        disease    bipolar disorder
4:             GSE35974 biological sex                male
5:             GSE35974        disease       schizophrenia
6:             GSE35974  organism part          cerebellum
7:             GSE12649        disease       schizophrenia
8:             GSE12649        disease    bipolar disorder
9:             GSE12649  organism part   prefrontal cortex</code></pre>
</div>
<div class="section level2">
<h2 id="output-options">Output options<a class="anchor" aria-label="anchor" href="#output-options"></a>
</h2>
<div class="section level3">
<h3 id="raw-data">Raw data<a class="anchor" aria-label="anchor" href="#raw-data"></a>
</h3>
<p>By default, Gemma API does some parsing on the raw API results to
make it easier to work with inside of R. In the process, it drops some
typically unused values. If you wish to fetch everything, use
<code>raw = TRUE</code>. Instead of a data table, you’ll usually be
served a list that represents the underlying JSON response.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_gene_locations.html">get_gene_locations</a></span><span class="op">(</span><span class="st">"DYRK1A"</span><span class="op">)</span></span></code></pre></div>
<pre><code>   chromosome strand nucleotide length taxon.name  taxon.scientific taxon.ID
       &lt;char&gt; &lt;char&gt;      &lt;int&gt;  &lt;int&gt;     &lt;char&gt;            &lt;char&gt;    &lt;int&gt;
1:         11      +   33890705 118714        rat Rattus norvegicus        3
2:         21      +   37365572 160785      human      Homo sapiens        1
3:         16      +   94370769 125608      mouse      Mus musculus        2
   taxon.NCBI taxon.database.name taxon.database.ID
        &lt;int&gt;              &lt;char&gt;             &lt;int&gt;
1:      10116                 rn6                86
2:       9606                hg38                87
3:      10090                mm10                81</code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_gene_locations.html">get_gene_locations</a></span><span class="op">(</span><span class="st">"DYRK1A"</span>, raw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>[[1]]
[[1]]$id
[1] 120910616

[[1]]$nucleotide
[1] 33890705

[[1]]$nucleotideLength
[1] 118714

[[1]]$strand
[1] "+"

[[1]]$bin
[1] 105

[[1]]$chromosome
[1] "11"

[[1]]$taxon
[[1]]$taxon$id
[1] 3

[[1]]$taxon$scientificName
[1] "Rattus norvegicus"

[[1]]$taxon$commonName
[1] "rat"

[[1]]$taxon$ncbiId
[1] 10116

[[1]]$taxon$externalDatabase
[[1]]$taxon$externalDatabase$id
[1] 86

[[1]]$taxon$externalDatabase$name
[1] "rn6"

[[1]]$taxon$externalDatabase$description
[1] "RGSC Rnor_6.0"

[[1]]$taxon$externalDatabase$uri
[1] "https://genome.ucsc.edu/cgi-bin/hgTracks?db=rn6"

[[1]]$taxon$externalDatabase$releaseVersion
NULL

[[1]]$taxon$externalDatabase$releaseUrl
NULL

[[1]]$taxon$externalDatabase$lastUpdated
NULL

[[1]]$taxon$externalDatabase$externalDatabases
[[1]]$taxon$externalDatabase$externalDatabases[[1]]
[[1]]$taxon$externalDatabase$externalDatabases[[1]]$id
[1] 116

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$name
[1] "rn6 RNA-Seq annotations"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$description
[1] "Annotations provided by NCBI Genome and used by the RNA-Seq pipeline for rat data."

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$uri
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$releaseVersion
[1] "106"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Rattus_norvegicus/106"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$lastUpdated
[1] "2022-10-24T07:00:00.000+00:00"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$externalDatabases
list()


[[1]]$taxon$externalDatabase$externalDatabases[[2]]
[[1]]$taxon$externalDatabase$externalDatabases[[2]]$id
[1] 100

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$name
[1] "rn4 annotations"

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$description
NULL

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$uri
[1] "https://hgdownload.cse.ucsc.edu/goldenpath/rn4/database/"

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$releaseVersion
NULL

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$releaseUrl
NULL

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$lastUpdated
NULL

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$externalDatabases
list()






[[2]]
[[2]]$id
[1] 120419022

[[2]]$nucleotide
[1] 37365572

[[2]]$nucleotideLength
[1] 160785

[[2]]$strand
[1] "+"

[[2]]$bin
[1] 108

[[2]]$chromosome
[1] "21"

[[2]]$taxon
[[2]]$taxon$id
[1] 1

[[2]]$taxon$scientificName
[1] "Homo sapiens"

[[2]]$taxon$commonName
[1] "human"

[[2]]$taxon$ncbiId
[1] 9606

[[2]]$taxon$externalDatabase
[[2]]$taxon$externalDatabase$id
[1] 87

[[2]]$taxon$externalDatabase$name
[1] "hg38"

[[2]]$taxon$externalDatabase$description
[1] "Genome Reference Consortium Human GRCh38.p13 (GCA_000001405.28)"

[[2]]$taxon$externalDatabase$uri
[1] "https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38"

[[2]]$taxon$externalDatabase$releaseVersion
[1] "GRCh38.p13"

[[2]]$taxon$externalDatabase$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_000001405.28/"

[[2]]$taxon$externalDatabase$lastUpdated
[1] "2022-06-30T07:00:00.000+00:00"

[[2]]$taxon$externalDatabase$externalDatabases
[[2]]$taxon$externalDatabase$externalDatabases[[1]]
[[2]]$taxon$externalDatabase$externalDatabases[[1]]$id
[1] 94

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$name
[1] "hg38 annotations"

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$description
NULL

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$uri
[1] "https://hgdownload.cse.ucsc.edu/goldenpath/hg38/database/"

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$releaseVersion
[1] "GRCh38.p13"

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_000001405.28/"

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$lastUpdated
[1] "2022-06-30T07:00:00.000+00:00"

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$externalDatabases
list()


[[2]]$taxon$externalDatabase$externalDatabases[[2]]
[[2]]$taxon$externalDatabase$externalDatabases[[2]]$id
[1] 124

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$name
[1] "hg38 RNA-Seq annotations"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$description
[1] "Annotations provided by NCBI Genome and used by the RNA-Seq pipeline for human data."

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$uri
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$releaseVersion
[1] "110"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Homo_sapiens/110/"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$lastUpdated
[1] "2023-01-17T20:27:55.059+00:00"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$externalDatabases
list()






[[3]]
[[3]]$id
[1] 120734638

[[3]]$nucleotide
[1] 94370769

[[3]]$nucleotideLength
[1] 125608

[[3]]$strand
[1] "+"

[[3]]$bin
[1] 20

[[3]]$chromosome
[1] "16"

[[3]]$taxon
[[3]]$taxon$id
[1] 2

[[3]]$taxon$scientificName
[1] "Mus musculus"

[[3]]$taxon$commonName
[1] "mouse"

[[3]]$taxon$ncbiId
[1] 10090

[[3]]$taxon$externalDatabase
[[3]]$taxon$externalDatabase$id
[1] 81

[[3]]$taxon$externalDatabase$name
[1] "mm10"

[[3]]$taxon$externalDatabase$description
[1] "Genome Reference Consortium GRCm38, which includes approximately 2.6 Gb of sequence, is considered to be \"essentially complete\". The assembly includes chromosomes 1-19, X, Y, M (mitochondrial DNA) and chr*_random (unlocalized) and chrUn_* (unplaced clone contigs)."

[[3]]$taxon$externalDatabase$uri
[1] "https://genome.ucsc.edu/cgi-bin/hgTracks?db=mm10"

[[3]]$taxon$externalDatabase$releaseVersion
[1] "GCA_000001635.2"

[[3]]$taxon$externalDatabase$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001635.2/"

[[3]]$taxon$externalDatabase$lastUpdated
[1] "2022-06-30T07:00:00.000+00:00"

[[3]]$taxon$externalDatabase$externalDatabases
[[3]]$taxon$externalDatabase$externalDatabases[[1]]
[[3]]$taxon$externalDatabase$externalDatabases[[1]]$id
[1] 97

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$name
[1] "mm10 annotations"

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$description
NULL

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$uri
[1] "https://hgdownload.cse.ucsc.edu/goldenpath/mm10/database/"

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$releaseVersion
[1] "GCA_000001635.2"

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001635.2/"

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$lastUpdated
[1] "2022-06-30T07:00:00.000+00:00"

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$externalDatabases
list()


[[3]]$taxon$externalDatabase$externalDatabases[[2]]
[[3]]$taxon$externalDatabase$externalDatabases[[2]]$id
[1] 115

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$name
[1] "mm10 RNA-Seq annotations"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$description
[1] "Annotations provided by NCBI Genome and used by the RNA-Seq pipeline for mouse data."

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$uri
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$releaseVersion
[1] "108"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Mus_musculus/108/"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$lastUpdated
[1] "2019-10-24T07:00:00.219+00:00"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$externalDatabases
list()






attr(,"call")
[1] "https://gemma.msl.ubc.ca/rest/v2/genes/DYRK1A/locations"
attr(,"env")
&lt;environment: 0x5601470439d8&gt;</code></pre>
</div>
<div class="section level3">
<h3 id="file-outputs">File outputs<a class="anchor" aria-label="anchor" href="#file-outputs"></a>
</h3>
<p>Sometimes, you may wish to save results to a file for future
inspection. You can do this simply by providing a filename to the
<code>file</code> parameter. The extension for this file will be one of
three options:</p>
<ol style="list-style-type: decimal">
<li>
<code>.json</code>, if you requested results with
<code>raw=TRUE</code>
</li>
<li>
<code>.csv</code> if the results have no nested data tables</li>
<li>
<code>.rds</code> otherwise</li>
</ol>
<p>You can also specify whether or not the new fetched results are
allowed to overwrite an existing file by specifying the
<code>overwrite = TRUE</code> parameter.</p>
</div>
<div class="section level3">
<h3 id="memoise-data">Memoise data<a class="anchor" aria-label="anchor" href="#memoise-data"></a>
</h3>
<p>To speed up results, you can remember past results so future queries
can proceed virtually instantly. This is enabled through the <a href="https://github.com/r-lib/memoise" class="external-link"><code>memoise</code></a>
package. To enable memoisation, simply set <code>memoised = TRUE</code>
in the function call whenever you want to refer to the cache, both to
save data for future calls and use the saved data for repeated
calls.</p>
<p>By default this will create a cache in your local filesystem.</p>
<p>You can see the path to your cache by using
<code>rappdirs::user_cache_dir(appname = "gemmaR")</code>. If you want
to use a different directory, use
<code>options(gemma.cache = 'path')</code>.</p>
<p>If you’re done with your fetching and want to ensure no space is
being used for cached results, or if you just want to ensure you’re
getting up-to-date data from Gemma, you can clear the cache using <a href="https://pavlidislab.github.io/gemma.R/reference/forget_gemma_memoised.html" class="external-link"><code>forget_gemma_memoised</code></a>.</p>
</div>
<div class="section level3">
<h3 id="changing-defaults">Changing defaults<a class="anchor" aria-label="anchor" href="#changing-defaults"></a>
</h3>
<p>We’ve seen how to change <code>raw = TRUE</code>,
<code>overwrite = TRUE</code> and <code>memoised = TRUE</code> in
individual function calls. It’s possible that you want to always use the
functions these ways without specifying the option every time. You can
do this by simply changing the default, which is visible in the function
definition. See below for examples.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.memoised <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always refer to cache</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always overwrite when saving files</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.raw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always receive results as-is from Gemma</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] viridis_0.6.5               viridisLite_0.4.2          
 [3] pheatmap_1.0.12             SummarizedExperiment_1.32.0
 [5] Biobase_2.62.0              GenomicRanges_1.54.1       
 [7] GenomeInfoDb_1.38.6         IRanges_2.36.0             
 [9] S4Vectors_0.40.2            BiocGenerics_0.48.1        
[11] MatrixGenerics_1.14.0       matrixStats_1.2.0          
[13] ggrepel_0.9.5               ggplot2_3.4.4              
[15] dplyr_1.1.4                 data.table_1.15.0          
[17] gemma.R_2.9.0               BiocStyle_2.30.0           

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0        farver_2.1.1            bitops_1.0-7           
 [4] fastmap_1.1.1           RCurl_1.98-1.14         digest_0.6.34          
 [7] timechange_0.3.0        lifecycle_1.0.4         magrittr_2.0.3         
[10] compiler_4.3.2          rlang_1.1.3             sass_0.4.8             
[13] tools_4.3.2             utf8_1.2.4              yaml_2.3.8             
[16] knitr_1.45              labeling_0.4.3          S4Arrays_1.2.0         
[19] bit_4.0.5               curl_5.2.0              DelayedArray_0.28.0    
[22] RColorBrewer_1.1-3      abind_1.4-5             withr_3.0.0            
[25] purrr_1.0.2             desc_1.4.3              grid_4.3.2             
[28] fansi_1.0.6             colorspace_2.1-0        scales_1.3.0           
[31] cli_3.6.2               rmarkdown_2.25          crayon_1.5.2           
[34] ragg_1.2.7              generics_0.1.3          httr_1.4.7             
[37] cachem_1.0.8            stringr_1.5.1           zlibbioc_1.48.0        
[40] assertthat_0.2.1        BiocManager_1.30.22     XVector_0.42.0         
[43] vctrs_0.6.5             Matrix_1.6-5            jsonlite_1.8.8         
[46] bookdown_0.37           bit64_4.0.5             systemfonts_1.0.5      
[49] jquerylib_0.1.4         glue_1.7.0              pkgdown_2.0.7          
[52] lubridate_1.9.3         stringi_1.8.3           gtable_0.3.4           
[55] munsell_0.5.0           tibble_3.2.1            pillar_1.9.0           
[58] rappdirs_0.3.3          htmltools_0.5.7         GenomeInfoDbData_1.2.11
[61] R6_2.5.1                textshaping_0.3.7       evaluate_0.23          
[64] lattice_0.22-5          highr_0.10              memoise_2.0.1          
[67] bslib_0.6.1             Rcpp_1.0.12             gridExtra_2.3          
[70] SparseArray_1.2.4       xfun_0.42               fs_1.6.3               
[73] pkgconfig_2.0.3        </code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
