<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Accessing curated gene expression data with gemma.R • gemma.R</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Accessing curated gene expression data with gemma.R">
<meta property="og:description" content="gemma.R">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="gemma.R_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Accessing curated gene expression data with gemma.R</h1>
                        <h4 data-toc-skip class="author">Javier Castillo-Arnemann</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia, Vancouver, Canada<br><h4 data-toc-skip class="author">Jordan Sicherman</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia, Vancouver, Canada<br><h4 data-toc-skip class="author">B. Ogan Mancarci</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia, Vancouver, Canada<br><small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/vignettes/gemma.R.Rmd" class="external-link"><code>vignettes/gemma.R.Rmd</code></a></small>
      <div class="hidden name"><code>gemma.R.Rmd</code></div>

    </address>
</address>
</address>
</div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pavlidislab.github.io/gemma.R/" class="external-link">gemma.R</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel" class="external-link">ggrepel</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="about-gemma">About Gemma<a class="anchor" aria-label="anchor" href="#about-gemma"></a>
</h2>
<p><a href="https://gemma.msl.ubc.ca/" class="external-link">Gemma</a> is a web site, database and a set of tools for the meta-analysis, re-use and sharing of genomics data, currently primarily targeted at the analysis of gene expression profiles. Gemma contains data from thousands of public studies, referencing thousands of published papers. Every dataset in Gemma has passed a rigorous curation process that re-annotates the expression platform at the sequence level, which allows for more consistent cross-platform comparisons and meta-analyses.</p>
<p>For detailed information on the curation process, read this <a href="https://pavlidislab.github.io/Gemma/curation.html" class="external-link">page</a> or the latest <a href="https://academic.oup.com/database/article/doi/10.1093/database/baab006/6143045" class="external-link">publication</a>.</p>
</div>
<div class="section level2">
<h2 id="package-cheat-sheet">Package cheat sheet<a class="anchor" aria-label="anchor" href="#package-cheat-sheet"></a>
</h2>
<p><img src="cheat_sheet.png" style="width:100.0%"></p>
</div>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<div class="section level3">
<h3 id="bioconductor">Bioconductor<a class="anchor" aria-label="anchor" href="#bioconductor"></a>
</h3>
<p>You can install <code>gemma.R</code> through <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> with the following code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"gemma.R"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="searching-for-datasets-of-interest-in-gemma">Searching for datasets of interest in Gemma<a class="anchor" aria-label="anchor" href="#searching-for-datasets-of-interest-in-gemma"></a>
</h2>
<p>The package includes various functions to search for datasets fitting desired criteria.</p>
<p>All datasets belonging to a taxon of interest can be accessed by using <a href="https://pavlidislab.github.io/gemma.R/reference/get_taxon_datasets.html" class="external-link"><code>get_taxon_datasets</code></a> while the function <a href="https://pavlidislab.github.io/gemma.R/reference/search_datasets.html" class="external-link"><code>search_datasets</code></a> can be used to further limit the results by a specified query containing key words, experiment names or ontology term URIs</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_taxon_datasets.html">get_taxon_datasets</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  <span class="co"># all human datasets </span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.ShortName</span>, <span class="va">experiment.Name</span>,<span class="va">taxon.Name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>   experiment.ShortName
1:              GSE2018
2:              GSE4036
3:              GSE3489
4:              GSE1923
5:               GSE361
6:               GSE492
                                                                          experiment.Name
1:                                                            Human Lung Transplant - BAL
2:                                                                perro-affy-human-186940
3: Patterns of gene dysregulation in the frontal cortex of patients with HIV encephalitis
4: Identification of PDGF-dependent patterns of gene expression in U87 glioblastoma cells
5:                                                   Mammary epithelial cell transduction
6:                               Effect of prostaglandin analogs on aqueous humor outflow
   taxon.Name
1:      human
2:      human
3:      human
4:      human
5:      human
6:      human</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/search_datasets.html">search_datasets</a></span><span class="op">(</span><span class="st">'bipolar'</span>,taxon <span class="op">=</span> <span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># human datasets mentioning bipolar</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.ShortName</span>, <span class="va">experiment.Name</span>,<span class="va">taxon.Name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>   experiment.ShortName
1:              GSE5389
2:              GSE4030
3:              GSE5388
4:              GSE7036
5:   McLean Hippocampus
6:           McLean_PFC
                                                                                                           experiment.Name
1:           Adult postmortem brain tissue (orbitofrontal cortex) from subjects with bipolar disorder and healthy controls
2:                                                                                                 bunge-affy-arabi-162779
3: Adult postmortem brain tissue (dorsolateral prefrontal cortex) from subjects with bipolar disorder and healthy controls
4:                                               Expression profiling in monozygotic twins discordant for bipolar disorder
5:                                                                                                      McLean Hippocampus
6:                                                                                                              McLean_PFC
   taxon.Name
1:      human
2:      human
3:      human
4:      human
5:      human
6:      human</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/search_datasets.html">search_datasets</a></span><span class="op">(</span><span class="st">'http://purl.obolibrary.org/obo/DOID_3312'</span>, <span class="co"># ontology term URI for the bipolar disorder</span></span>
<span>                taxon <span class="op">=</span> <span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.ShortName</span>, <span class="va">experiment.Name</span>,<span class="va">taxon.Name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>   experiment.ShortName
1:              GSE5389
2:              GSE5388
3:              GSE7036
4:   McLean Hippocampus
5:           McLean_PFC
6:     stanley_feinberg
                                                                                                           experiment.Name
1:           Adult postmortem brain tissue (orbitofrontal cortex) from subjects with bipolar disorder and healthy controls
2: Adult postmortem brain tissue (dorsolateral prefrontal cortex) from subjects with bipolar disorder and healthy controls
3:                                               Expression profiling in monozygotic twins discordant for bipolar disorder
4:                                                                                                      McLean Hippocampus
5:                                                                                                              McLean_PFC
6:                                                                     Stanley consortium collection Cerebellum - Feinberg
   taxon.Name
1:      human
2:      human
3:      human
4:      human
5:      human
6:      human</code></pre>
<p>Note that a single call of these functions will only return 20 results by default and a 100 results maximum, controlled by the <code>limit</code> argument. In order to get all available results, <code>offset</code> argument should be used to make multiple calls.</p>
<p>To see how many available results are there, you can look at the attributes of the output objects where additional information from the API response is appended.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># a quick call with a limit of 1 to get the result count</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_taxon_datasets.html">get_taxon_datasets</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="st">'human'</span>,limit <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span><span class="op">$</span><span class="va">totalElements</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 5896</code></pre>
<p>Since the maximum limit is 100 getting all results available will require multiple calls.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span><span class="op">$</span><span class="va">totalElements</span></span>
<span><span class="va">all_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">count</span>, <span class="fl">100</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">offset</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/get_taxon_datasets.html">get_taxon_datasets</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="st">'human'</span>,limit <span class="op">=</span> <span class="fl">100</span>, offset <span class="op">=</span> <span class="va">offset</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>,<span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.ShortName</span>, <span class="va">experiment.Name</span>,<span class="va">taxon.Name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>   experiment.ShortName
1:              GSE2018
2:              GSE4036
3:              GSE3489
4:              GSE1923
5:               GSE361
6:               GSE492
                                                                          experiment.Name
1:                                                            Human Lung Transplant - BAL
2:                                                                perro-affy-human-186940
3: Patterns of gene dysregulation in the frontal cortex of patients with HIV encephalitis
4: Identification of PDGF-dependent patterns of gene expression in U87 glioblastoma cells
5:                                                   Mammary epithelial cell transduction
6:                               Effect of prostaglandin analogs on aqueous humor outflow
   taxon.Name
1:      human
2:      human
3:      human
4:      human
5:      human
6:      human</code></pre>
<p>See <a href="#larger-queries">larger queries</a> section for more details. To keep this vignette simpler we will keep using the first 20 results returned by default in examples below.</p>
<p>Information provided about the datasets by these functions include details about the quality and design of the study that can be used to judge if it is suitable for your use case. For instance <code>geeq.batchEffect</code> column will be set to -1 if Gemma’s preprocessing has detected batch effects that were unable to be resolved by batch correction and <code>experiment.SampleCount</code> will include the number of samples used in the experiment. More information about these and other fields can be found at the function documentation.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_taxon_datasets.html">get_taxon_datasets</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># get human datasets</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">geeq.batchEffect</span> <span class="op">!=</span><span class="op">-</span><span class="fl">1</span> <span class="op">&amp;</span> <span class="va">experiment.SampleCount</span> <span class="op">&gt;</span> <span class="fl">4</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># filter for datasets without batch effects and with a sample count more than 4</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.ShortName</span>, <span class="va">experiment.Name</span>,<span class="va">taxon.Name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>   experiment.ShortName
1:              GSE2018
2:              GSE4036
3:              GSE3489
4:              GSE1923
5:               GSE361
6:               GSE492
                                                                          experiment.Name
1:                                                            Human Lung Transplant - BAL
2:                                                                perro-affy-human-186940
3: Patterns of gene dysregulation in the frontal cortex of patients with HIV encephalitis
4: Identification of PDGF-dependent patterns of gene expression in U87 glioblastoma cells
5:                                                   Mammary epithelial cell transduction
6:                               Effect of prostaglandin analogs on aqueous humor outflow
   taxon.Name
1:      human
2:      human
3:      human
4:      human
5:      human
6:      human</code></pre>
<p>Gemma uses multiple ontologies when annotating datasets and using the term URIs instead of free text to search can lead to more specific results. <a href="https://pavlidislab.github.io/gemma.R/reference/search_annotations.html" class="external-link"><code>search_annotations</code></a> function allows searching for annotation terms that might be relevant to your query.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/search_annotations.html">search_annotations</a></span><span class="op">(</span><span class="st">'bipolar'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">head</span></span></code></pre></div>
<pre><code>   category.Name                         category.URI         value.Name
1:          &lt;NA&gt;                                 &lt;NA&gt;            Bipolar
2:       disease http://www.ebi.ac.uk/efo/EFO_0000408 bipolar I disorder
3:       disease http://www.ebi.ac.uk/efo/EFO_0000408   Bipolar Disorder
4:       disease http://www.ebi.ac.uk/efo/EFO_0000408 bipolar II disoder
5:       disease http://www.ebi.ac.uk/efo/EFO_0000408  Bipolar depressed
6:          &lt;NA&gt;                                 &lt;NA&gt;   Bipolar Disorder
                                   value.URI
1:                                      &lt;NA&gt;
2: http://purl.obolibrary.org/obo/DOID_14042
3:  http://purl.obolibrary.org/obo/DOID_3312
4:      http://www.ebi.ac.uk/efo/EFO_0009964
5:                                      &lt;NA&gt;
6:                                      &lt;NA&gt;</code></pre>
</div>
<div class="section level2">
<h2 id="downloading-expression-data">Downloading expression data<a class="anchor" aria-label="anchor" href="#downloading-expression-data"></a>
</h2>
<p>Upon identifying datasets of interest, more information about specific ones can be requested. In this example we will be using GSE46416 which includes samples taken from healthy donors along with manic/euthymic phase bipolar disorder patients.</p>
<p>The data associated with specific experiments can be accessed by using <a href="https://pavlidislab.github.io/gemma.R/reference/get_datasets_by_ids.html" class="external-link"><code>get_datasets_by_ids</code></a></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_datasets_by_ids.html">get_datasets_by_ids</a></span><span class="op">(</span><span class="st">"GSE46416"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.ShortName</span>, <span class="va">experiment.Name</span>, <span class="va">experiment.ID</span>, <span class="va">experiment.Description</span><span class="op">)</span></span></code></pre></div>
<pre><code>   experiment.ShortName
1:             GSE46416
                                                   experiment.Name
1: State- and trait-specific gene expression in euthymia and mania
   experiment.ID
1:          8997
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       experiment.Description
1: Gene expression profiles of bipolar disorder (BD) patients were assessed during both a manic and a euthymic phase and compared both intra-individually, and with the gene expression profiles of controls.\nLast Updated (by provider): Sep 05 2014\nContributors:  Christian C Witt Benedikt Brors Dilafruz Juraeva Jens Treutlein Carsten Sticht Stephanie H Witt Jana Strohmaier Helene Dukal Josef Frank Franziska Degenhardt Markus M Nöthen Sven Cichon Maren Lang Marcella Rietschel Sandra Meier Manuel Mattheisen</code></pre>
<p>To access the expression data in a convenient form, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/get_dataset_object.html" class="external-link"><code>get_dataset_object</code></a>. It is a high-level wrapper that combines various endpoint calls to return an annotated <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link"><code>SummarizedExperiment</code></a> or <a href="https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf" class="external-link"><code>ExpressionSet</code></a> that is compatible with other Bioconductor packages or a <a href="https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html" class="external-link">tidyverse-friendly</a> long form tibble for downstream analyses. They include the expression matrix along with the experimental design, and ensure the sample names match between both when transforming/subsetting data.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset_object.html">get_dataset_object</a></span><span class="op">(</span><span class="st">"GSE46416"</span>,</span>
<span>                          type <span class="op">=</span> <span class="st">'se'</span><span class="op">)</span> <span class="co"># SummarizedExperiment is the default output type</span></span></code></pre></div>
<p>Note that the tidy format is less memory efficient but allows easy visualization and exploration with <a href="https://ggplot2.tidyverse.org/index.html" class="external-link">ggplot2</a> and the rest of the <a href="https://www.tidyverse.org/" class="external-link">tidyverse</a>.</p>
<p>To show how subsetting works, we’ll keep the “manic phase” data and the <code>reference_subject_role</code>s, which refers to the control samples in Gemma datasets.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check the levels of the disease factor</span></span>
<span><span class="va">dat</span><span class="op">$</span><span class="va">disease</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "reference_subject_role"              "euthymic_phase_|_Bipolar_Disorder_|"
[3] "bipolar_disorder_|_manic_phase_|"   </code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Subset patients during manic phase and controls</span></span>
<span><span class="va">manic</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">[</span>, <span class="va">dat</span><span class="op">$</span><span class="va">disease</span> <span class="op">==</span> <span class="st">"bipolar_disorder_|_manic_phase_|"</span> <span class="op">|</span> </span>
<span>        <span class="va">dat</span><span class="op">$</span><span class="va">disease</span> <span class="op">==</span> <span class="st">"reference_subject_role"</span><span class="op">]</span></span>
<span><span class="va">manic</span></span></code></pre></div>
<pre><code>class: SummarizedExperiment 
dim: 21986 21 
metadata(8): title abstract ... GemmaSuitabilityScore taxon
assays(1): counts
rownames(21986): 2315430 2315554 ... 7385683 7385696
rowData names(4): Probe GeneSymbol GeneName NCBIid
colnames(21): Control,1_DE50 Control,12 ...
  Bipolardisorderpatientmanicphase,16
  Bipolardisorderpatientmanicphase,21
colData names(2): batch disease</code></pre>
<p>Let’s take a look at sample to sample correlation in our subset.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get Expression matrix</span></span>
<span><span class="va">manicExpr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">manic</span>, <span class="st">"counts"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">manicExpr</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">cor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span>col <span class="op">=</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>,border_color <span class="op">=</span> <span class="cn">NA</span>,angle_col <span class="op">=</span> <span class="fl">45</span>,fontsize <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="gemma.R_files/figure-html/boxplot-1.png" alt="Sample to sample correlations of bipolar patients during manic phase and controls." width="700"><p class="caption">
Sample to sample correlations of bipolar patients during manic phase and controls.
</p>
</div>
<p>You can also use <a href="https://pavlidislab.github.io/gemma.R/reference/get_dataset_expression.html" class="external-link"><code>get_dataset_expression</code></a> to only get the expression matrix, and <a href="https://pavlidislab.github.io/gemma.R/reference/get_dataset_design.html" class="external-link"><code>get_dataset_design</code></a> to get the experimental design matrix.</p>
</div>
<div class="section level2">
<h2 id="platform-annotations">Platform Annotations<a class="anchor" aria-label="anchor" href="#platform-annotations"></a>
</h2>
<p>Expression data in Gemma comes with annotations for the gene each expression profile corresponds to. Using the <a href="https://pavlidislab.github.io/gemma.R/reference/get_platform_annotations.html" class="external-link"><code>get_platform_annotations</code></a> function, these annotations can be retrieved independently of the expression data, along with additional annotations such as Gene Ontology terms.</p>
<p>Examples:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platform_annotations.html">get_platform_annotations</a></span><span class="op">(</span><span class="st">'GPL96'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>     ProbeName   GeneSymbols                                     GeneNames
1: 211750_x_at TUBA1C|TUBA1A             tubulin alpha 1c|tubulin alpha 1a
2:   216678_at                                                            
3:   216345_at        ZSWIM8            zinc finger SWIM-type containing 8
4:   207273_at                                                            
5: 216025_x_at        CYP2C9 cytochrome P450 family 2 subfamily C member 9
6: 218191_s_at        LMBRD1                     LMBR1 domain containing 1
                                                                                                                                                                                                                                                                                                                                                                      GOTerms
1:                                                                                                                          GO:0005737|GO:0005525|GO:0051301|GO:0005515|GO:0000226|GO:0005634|GO:0007017|GO:0000278|GO:0005874|GO:0005200|GO:0005198|GO:0005881|GO:0030705|GO:0015630|GO:0031982|GO:0070062|GO:0050807|GO:0036464|GO:0042802|GO:0055037|GO:0005829|GO:0031594
2:                                                                                                                                                                                                                                                                                                                                                                           
3:                                                                                                                                                                                                                                                                                    GO:0008270|GO:0043161|GO:0016567|GO:0031463|GO:0005829|GO:0031462|GO:1990756|GO:2000627
4:                                                                                                                                                                                                                                                                                                                                                                           
5: GO:0005737|GO:0052741|GO:0008405|GO:0008404|GO:0020037|GO:0008203|GO:0008202|GO:0043231|GO:0008392|GO:0034875|GO:0070330|GO:0019373|GO:0006805|GO:0005506|GO:0005789|GO:0097267|GO:0005886|GO:0032787|GO:0042178|GO:0004497|GO:0046456|GO:0008210|GO:0008395|GO:0016712|GO:0019627|GO:0042759|GO:0006082|GO:0070989|GO:0043603|GO:0018676|GO:0018675|GO:0016491|GO:0016098
6:                                                                                                                                                GO:0031419|GO:0072583|GO:0005789|GO:0005515|GO:0005886|GO:0005765|GO:0030136|GO:0005774|GO:0005158|GO:0007369|GO:0045334|GO:0061462|GO:0140318|GO:0043231|GO:0038016|GO:0032050|GO:0035612|GO:0072665|GO:0016020|GO:0016021
        GemmaIDs    NCBIids
1: 360802|172797 84790|7846
2:                         
3:        235733      23053
4:                         
5:         32964       1559
6:        303717      55788</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platform_annotations.html">get_platform_annotations</a></span><span class="op">(</span><span class="st">'Generic_human'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>      ProbeName  GeneSymbols
1:        LCN10        LCN10
2:     STAG3L5P     STAG3L5P
3: LOC101059976 LOC101059976
4:         GAB3         GAB3
5: LOC100287155 LOC100287155
6:       RASSF2       RASSF2
                                                                 GeneNames
1:                                                            lipocalin 10
2:                                     stromal antigen 3-like 5 pseudogene
3: arf-GAP with GTPase, ANK repeat and PH domain-containing protein 2-like
4:                                       GRB2 associated binding protein 3
5:                                       hypothetical protein LOC100287155
6:                                  Ras association domain family member 2
                                                                                                                                                                                                                                                                                                    GOTerms
1:                                                                                                                                                                                                                                                                                    GO:0005576|GO:0036094
2:                                                                                                                                                                                                                                                                                                         
3:                                                                                                                                                                                                                                                                                                         
4:                                                                                                                                                                                                                                                              GO:0005737|GO:0030225|GO:0035591|GO:0007165
5:                                                                                                                                                                                                                                                                                                         
6: GO:0045670|GO:0005737|GO:0005515|GO:0005634|GO:0050821|GO:0005654|GO:0005794|GO:1901222|GO:1901223|GO:0046849|GO:0032991|GO:0048872|GO:0046330|GO:0001501|GO:0000776|GO:0005886|GO:0031954|GO:0007049|GO:0045667|GO:0004672|GO:0043065|GO:0045860|GO:0007165|GO:0033137|GO:0001503|GO:0005829|GO:0038168
   GemmaIDs   NCBIids
1:   441399    414332
2:  8799043 101735302
3:  8779607 101059976
4:   389635    139716
5:  8090381 100287155
6:   201914      9770</code></pre>
<p>If you are interested in a particular gene, you can see which platforms include it using <a href="https://pavlidislab.github.io/gemma.R/reference/get_gene_probes.html" class="external-link"><code>get_gene_probes</code></a>. Note that functions to search gene work best with unambigious identifiers rather than symbols.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># lists genes in gemma matching the symbol or identifier</span></span>
<span><span class="fu"><a href="../reference/get_genes.html">get_genes</a></span><span class="op">(</span><span class="st">'Eno2'</span><span class="op">)</span></span></code></pre></div>
<pre><code>   gene.Symbol       gene.Ensembl gene.NCBI                      gene.Name
1:        ENO2    ENSG00000111674      2026                      enolase 2
2:        Eno2 ENSMUSG00000004267     13807      enolase 2, gamma neuronal
3:        Eno2 ENSRNOG00000013141     24334                      enolase 2
4:        ENO2               &lt;NA&gt;    856579 phosphopyruvate hydratase ENO2
5:        eno2 ENSDARG00000014287    402874                      enolase 2
   gene.MFX.Rank taxon.Name         taxon.Scientific taxon.ID taxon.NCBI
1:     0.9284287      human             Homo sapiens        1       9606
2:     0.8634338      mouse             Mus musculus        2      10090
3:     0.8698729        rat        Rattus norvegicus        3      10116
4:     0.6226795      yeast Saccharomyces cerevisiae       11       4932
5:     0.8956117  zebrafish              Danio rerio       12       7955
   taxon.Database.Name taxon.Database.ID
1:                hg38                87
2:                mm10                81
3:                 rn6                86
4:                &lt;NA&gt;                NA
5:                &lt;NA&gt;                NA</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ncbi id for human ENO2</span></span>
<span><span class="va">probes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gene_probes.html">get_gene_probes</a></span><span class="op">(</span><span class="fl">2026</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># remove the description for brevity of output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">probes</span><span class="op">[</span>,<span class="va">.SD</span>, .SDcols <span class="op">=</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">probes</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'mapping.Description'</span>,<span class="st">'platform.Description'</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code>   mapping.Name platform.ShortName
1:        20016            GPL3093
2:        20024            GPL3092
3:        20024       lymphochip-2
4:         1639             GPL962
5:        35850         NHGRI-6.5k
6:    201313_at              GPL96
                                              platform.Name platform.ID
1:                                                    LC-25         211
2:                                                    LC-19         212
3:                                           Lymphochip 37k         229
4:                                               CHUGAI 41K          36
5:                                               NHGRI-6.5k         150
6: Affymetrix GeneChip Human Genome U133 Array Set HG-U133A           1
   platform.Type platform.Troubled taxon.Name taxon.Scientific taxon.ID
1:      TWOCOLOR             FALSE      human     Homo sapiens        1
2:      TWOCOLOR             FALSE      human     Homo sapiens        1
3:      TWOCOLOR             FALSE      human     Homo sapiens        1
4:      TWOCOLOR             FALSE      human     Homo sapiens        1
5:      TWOCOLOR             FALSE      human     Homo sapiens        1
6:      ONECOLOR             FALSE      human     Homo sapiens        1
   taxon.NCBI taxon.Database.Name taxon.Database.ID
1:       9606                hg38                87
2:       9606                hg38                87
3:       9606                hg38                87
4:       9606                hg38                87
5:       9606                hg38                87
6:       9606                hg38                87</code></pre>
</div>
<div class="section level2">
<h2 id="differential-expression-analyses">Differential expression analyses<a class="anchor" aria-label="anchor" href="#differential-expression-analyses"></a>
</h2>
<p>Gemma contains precomputed differential expression analyses for most of its datasets. Analyses can involve more than one factor, such as “sex” as well as “disease”. Some datasets contain more than one analysis to account for different factors and their interactions. The results are stored as resultSets, each corresponding to one factor (or their interaction). You can access them using <a href="https://pavlidislab.github.io/gemma.R/reference/get_differential_expression_values.html" class="external-link"><code>get_differential_expression_values</code></a>. From here on, we can explore and visualize the data to find the most differentially-expressed genes</p>
<p>Note that <code>get_differential_expression_values</code> can return multiple differentials per study if a study has multiple factors to contrast. Since GSE46416 only has one extracting the first element of the returned list is all we need.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dif_exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_differential_expression_values.html">get_differential_expression_values</a></span><span class="op">(</span><span class="st">'GSE46416'</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">dif_exp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code>         Probe    NCBIid GeneSymbol
    1: 2982730      4018        LPA
    2: 2787851    166752      FREM3
    3: 2477558                     
    4: 2910917                     
    5: 3983537    140886     PABPC5
   ---                             
21957: 3301011     64318      NOC3L
21958: 2461654 100130249  LINC02961
21959: 2360346      1141     CHRNB2
21960: 2391172      7293    TNFRSF4
21961: 2525718                     
                                            GeneName  pvalue corrected_pvalue
    1:                                lipoprotein(a) 0.91850           0.9521
    2:          FRAS1 related extracellular matrix 3 0.58500           0.7348
    3:                                               0.39650           0.5931
    4:                                               0.65060           0.7815
    5:         poly(A) binding protein cytoplasmic 5 0.10640           0.3227
   ---                                                                       
21957:           NOC3 like DNA replication regulator 0.06358           0.2647
21958:   long intergenic non-protein coding RNA 2961 0.25470           0.4735
21959: cholinergic receptor nicotinic beta 2 subunit 0.01213           0.1288
21960:             TNF receptor superfamily member 4 0.13140           0.3516
21961:                                               0.75260           0.8463
          rank contrast_113004_log2fc contrast_113004_tstat
    1: 0.96470               -0.02471              -0.35870
    2: 0.79600                0.18950               1.01200
    3: 0.66850                0.09179               1.13900
    4: 0.83250                0.15640               0.67390
    5: 0.32990                0.18450               2.03200
   ---                                                     
21957: 0.24020                0.14810               0.71780
21958: 0.53800               -0.17630              -1.53600
21959: 0.09417                0.10290               1.12300
21960: 0.37370               -0.00863              -0.06482
21961: 0.88930               -0.11050              -0.59600
       contrast_113004_pvalue contrast_113005_log2fc contrast_113005_tstat
    1:                0.72210               -0.02495               -0.3622
    2:                0.31910                0.14030                0.7495
    3:                0.26330                0.10160                1.2600
    4:                0.50520                0.20960                0.9032
    5:                0.05047                0.16030                1.7660
   ---                                                                    
21957:                0.47810               -0.33430               -1.6210
21958:                0.13440               -0.02347               -0.2045
21959:                0.26980                0.28670                3.1280
21960:                0.94870                0.23120                1.7370
21961:                0.55530               -0.13160               -0.7101
       contrast_113005_pvalue
    1:               0.719600
    2:               0.459000
    3:               0.216600
    4:               0.373100
    5:               0.086940
   ---                       
21957:               0.114800
21958:               0.839300
21959:               0.003721
21960:               0.091960
21961:               0.482700</code></pre>
<p>By default the columns names of the output correspond to contrast IDs. To see what conditions these IDs correspond to we can either use <code>get_dataset_differential_expression_analyses</code> to get the metadata about differentials of a given dataset, or set<code>readableContrasts</code> argument of <code>get_differential_expression_values</code> to <code>TRUE</code>. The former approach is usually better for a large scale systematic analysis while the latter is easier to read in an interactive session.</p>
<p><code>get_dataset_differential_expression_analyses</code> function returns structured metadata about the differentials.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset_differential_expression_analyses.html">get_dataset_differential_expression_analyses</a></span><span class="op">(</span><span class="st">'GSE46416'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>   result.ID contrast.id experiment.ID baseline.category
1:    550248      113004          8997           disease
2:    550248      113005          8997           disease
                   baseline.categoryURI   baseline.factorValue
1: http://www.ebi.ac.uk/efo/EFO_0000408 reference subject role
2: http://www.ebi.ac.uk/efo/EFO_0000408 reference subject role
                      baseline.factorValueURI         experimental.factorValue
1: http://purl.obolibrary.org/obo/OBI_0000220    bipolar disorder, manic phase
2: http://purl.obolibrary.org/obo/OBI_0000220 euthymic phase, Bipolar Disorder
                experimental.factorValueURI subsetFactor.subset
1: http://purl.obolibrary.org/obo/DOID_3312               FALSE
2:                                     &lt;NA&gt;               FALSE
   subsetFactor.factorValue subsetFactor.factorValueURI
1:                     &lt;NA&gt;                        &lt;NA&gt;
2:                     &lt;NA&gt;                        &lt;NA&gt;
   subsetFactor.description subsetFactor.category subsetFactor.categoryURI
1:                     &lt;NA&gt;                    NA                       NA
2:                     &lt;NA&gt;                    NA                       NA
   subsetFactor.measurement subsetFactor.type probes.Analyzed genes.Analyzed
1:                       NA                NA           21961          18959
2:                       NA                NA           21961          18959</code></pre>
<p><code>contrast.id</code> column corresponds to the column names in the output of <code>get_differential_expression_values</code> while <code>result.ID</code> corresponds to the name of the differential in the output object. Using them together will let one to access differentially expressed gene counts for each condition contrast</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># using result.ID and contrast.ID of the output above, we can access specific</span></span>
<span><span class="co"># results. Note that one study may have multiple contrast objects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">contrasts</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">result_set</span> <span class="op">=</span> <span class="va">dif_exp</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">contrasts</span><span class="op">[</span><span class="va">i</span>,<span class="op">]</span><span class="op">$</span><span class="va">result.ID</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">p_values</span> <span class="op">=</span> <span class="va">result_set</span><span class="op">[[</span><span class="fu">glue</span><span class="fu">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html" class="external-link">glue</a></span><span class="op">(</span><span class="st">"contrast_{contrasts[i,]$contrast.id}_pvalue"</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span>    </span>
<span>    <span class="co"># multiple testing correction</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">p_values</span>,method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">dif_exp_genes</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>result.ID <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">result.ID</span>,</span>
<span>           contrast.id <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">contrast.id</span>,</span>
<span>           baseline.factorValue <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">baseline.factorValue</span>,</span>
<span>           experimental.factorValue <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">experimental.factorValue</span>,</span>
<span>           n_diff <span class="op">=</span> <span class="va">dif_exp_genes</span><span class="op">)</span></span></code></pre></div>
<pre><code>  result.ID contrast.id   baseline.factorValue         experimental.factorValue
1    550248      113004 reference subject role    bipolar disorder, manic phase
2    550248      113005 reference subject role euthymic phase, Bipolar Disorder
  n_diff
1      3
2   1389</code></pre>
<p>Alternatively we, since we are only looking at one dataset and one contrast manually, we can simply use <code>readableContrasts</code>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">de</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_differential_expression_values.html">get_differential_expression_values</a></span><span class="op">(</span><span class="st">"GSE46416"</span>,readableContrasts <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code>         Probe    NCBIid GeneSymbol
    1: 2982730      4018        LPA
    2: 2787851    166752      FREM3
    3: 2477558                     
    4: 2910917                     
    5: 3983537    140886     PABPC5
   ---                             
21957: 3301011     64318      NOC3L
21958: 2461654 100130249  LINC02961
21959: 2360346      1141     CHRNB2
21960: 2391172      7293    TNFRSF4
21961: 2525718                     
                                            GeneName  pvalue corrected_pvalue
    1:                                lipoprotein(a) 0.91850           0.9521
    2:          FRAS1 related extracellular matrix 3 0.58500           0.7348
    3:                                               0.39650           0.5931
    4:                                               0.65060           0.7815
    5:         poly(A) binding protein cytoplasmic 5 0.10640           0.3227
   ---                                                                       
21957:           NOC3 like DNA replication regulator 0.06358           0.2647
21958:   long intergenic non-protein coding RNA 2961 0.25470           0.4735
21959: cholinergic receptor nicotinic beta 2 subunit 0.01213           0.1288
21960:             TNF receptor superfamily member 4 0.13140           0.3516
21961:                                               0.75260           0.8463
          rank contrast_bipolar disorder, manic phase_logFoldChange
    1: 0.96470                                             -0.02471
    2: 0.79600                                              0.18950
    3: 0.66850                                              0.09179
    4: 0.83250                                              0.15640
    5: 0.32990                                              0.18450
   ---                                                             
21957: 0.24020                                              0.14810
21958: 0.53800                                             -0.17630
21959: 0.09417                                              0.10290
21960: 0.37370                                             -0.00863
21961: 0.88930                                             -0.11050
       contrast_bipolar disorder, manic phase_tstat
    1:                                     -0.35870
    2:                                      1.01200
    3:                                      1.13900
    4:                                      0.67390
    5:                                      2.03200
   ---                                             
21957:                                      0.71780
21958:                                     -1.53600
21959:                                      1.12300
21960:                                     -0.06482
21961:                                     -0.59600
       contrast_bipolar disorder, manic phase_pvalue
    1:                                       0.72210
    2:                                       0.31910
    3:                                       0.26330
    4:                                       0.50520
    5:                                       0.05047
   ---                                              
21957:                                       0.47810
21958:                                       0.13440
21959:                                       0.26980
21960:                                       0.94870
21961:                                       0.55530
       contrast_euthymic phase, Bipolar Disorder_logFoldChange
    1:                                                -0.02495
    2:                                                 0.14030
    3:                                                 0.10160
    4:                                                 0.20960
    5:                                                 0.16030
   ---                                                        
21957:                                                -0.33430
21958:                                                -0.02347
21959:                                                 0.28670
21960:                                                 0.23120
21961:                                                -0.13160
       contrast_euthymic phase, Bipolar Disorder_tstat
    1:                                         -0.3622
    2:                                          0.7495
    3:                                          1.2600
    4:                                          0.9032
    5:                                          1.7660
   ---                                                
21957:                                         -1.6210
21958:                                         -0.2045
21959:                                          3.1280
21960:                                          1.7370
21961:                                         -0.7101
       contrast_euthymic phase, Bipolar Disorder_pvalue
    1:                                         0.719600
    2:                                         0.459000
    3:                                         0.216600
    4:                                         0.373100
    5:                                         0.086940
   ---                                                 
21957:                                         0.114800
21958:                                         0.839300
21959:                                         0.003721
21960:                                         0.091960
21961:                                         0.482700</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Classify probes for plotting</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">&lt;-</span> <span class="st">"No"</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span> <span class="op">&gt;</span> <span class="fl">1.0</span> <span class="op">&amp;</span> </span>
<span>        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Up"</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">1.0</span> <span class="op">&amp;</span> </span>
<span>        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Down"</span></span>
<span></span>
<span><span class="co"># Upregulated probes</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Up"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span>, </span>
<span>        <span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code>      Probe GeneSymbol contrast_bipolar disorder, manic phase_pvalue
 1: 2319550       RBP7                                     8.612e-05
 2: 2548699     CYP1B1                                     1.027e-04
 3: 3907190       SLPI                                     3.326e-04
 4: 3629103      PCLAF                                     5.183e-04
 5: 3545525      SLIRP                                     5.646e-04
 6: 3146433      COX6C                                     9.204e-04
 7: 2538349                                                1.253e-03
 8: 2899102       H3C3                                     1.269e-03
 9: 3635198     BCL2A1                                     1.800e-03
10: 2633191      GPR15                                     2.410e-03
    contrast_bipolar disorder, manic phase_logFoldChange
 1:                                                1.074
 2:                                                1.322
 3:                                                1.056
 4:                                                1.278
 5:                                                1.349
 6:                                                1.467
 7:                                                1.073
 8:                                                1.026
 9:                                                1.080
10:                                                1.205</code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Downregulated probes</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Down"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span>, </span>
<span>        <span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code>      Probe GeneSymbol contrast_bipolar disorder, manic phase_pvalue
 1: 2775390                                                2.095e-06
 2: 3760268                                                1.153e-05
 3: 3124344                                                1.389e-04
 4: 3673179                                                1.581e-04
 5: 3245871      WDFY4                                     1.681e-04
 6: 3022689   SND1-IT1                                     2.267e-04
 7: 2679014                                                2.982e-04
 8: 4019758                                                3.553e-04
 9: 3336402      RBM14                                     3.606e-04
10: 2880955                                                3.740e-04
    contrast_bipolar disorder, manic phase_logFoldChange
 1:                                               -1.556
 2:                                               -1.851
 3:                                               -1.037
 4:                                               -1.034
 5:                                               -1.157
 6:                                               -1.220
 7:                                               -1.175
 8:                                               -1.405
 9:                                               -1.071
10:                                               -1.522</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add gene symbols as labels to DE genes</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">delabel</span> <span class="op">&lt;-</span> <span class="st">""</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">delabel</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">de</span><span class="op">$</span><span class="va">GeneSymbol</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Volcano plot for bipolar patients vs controls</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">de</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        x <span class="op">=</span> <span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span>,</span>
<span>        y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span><span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="va">diffexpr</span>,</span>
<span>        label <span class="op">=</span> <span class="va">delabel</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1.0</span>, <span class="fl">1.0</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"log2(FoldChange)"</span>, y <span class="op">=</span> <span class="st">"-log10(p-value)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="gemma.R_files/figure-html/diffExpr-1.png" alt="Differentially-expressed genes in bipolar patients during manic phase versus controls." width="700"><p class="caption">
Differentially-expressed genes in bipolar patients during manic phase versus controls.
</p>
</div>
</div>
<div class="section level2">
<h2 id="larger-queries">Larger queries<a class="anchor" aria-label="anchor" href="#larger-queries"></a>
</h2>
<p>To query large amounts of data, the API has a pagination system which uses the <code>limit</code> and <code>offset</code> parameters. To avoid overloading the server, calls are limited to a maximum of 100 entries, so the offset allows you to get the next batch of entries in the next call(s).</p>
<p>To see how many available results are there, you can look at the attributes of the output objects where additional information from the API response is appended.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">platform_count</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platforms_by_ids.html">get_platforms_by_ids</a></span><span class="op">(</span>limit <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">totalElements</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">platform_count</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 681</code></pre>
<p>After which you can use <code>offset</code> to access all available platforms.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">platform_count</span>,<span class="fl">100</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">offset</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/get_platforms_by_ids.html">get_platforms_by_ids</a></span><span class="op">(</span>limit <span class="op">=</span> <span class="fl">100</span>, offset <span class="op">=</span> <span class="va">offset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">platform.ID</span>, <span class="va">platform.ShortName</span>, <span class="va">taxon.Name</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>,<span class="va">.</span><span class="op">)</span></span></code></pre></div>
<pre><code>     platform.ID   platform.ShortName taxon.Name
  1:           1                GPL96      human
  2:           2              GPL1355        rat
  3:           3              GPL1261      mouse
  4:           4               GPL570      human
  5:           5                GPL81      mouse
 ---                                            
677:        1314       Rosetta_Merged      human
678:        1315    RG-U34_ABC_Merged        rat
679:        1316             RAE230AB        rat
680:        1317             GPL30172      mouse
681:        1318 Agilent_8x60K_Merged      mouse</code></pre>
<p>Many endpoints only support a single identifier:</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_dataset_annotations.html">get_dataset_annotations</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE46416"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>Error: Please specify one valid identifier for dataset.</code></pre>
<p>In these cases, you will have to loop over all the identifiers you wish to query and send separate requests.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE12649"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/get_dataset_annotations.html">get_dataset_annotations</a></span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>experiment.shortName <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">class.Name</span>, <span class="va">term.Name</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>,<span class="va">.</span><span class="op">)</span></span></code></pre></div>
<pre><code>    experiment.shortName     class.Name              term.Name
 1:             GSE35974        disease       Bipolar Disorder
 2:             GSE35974 biological sex                 female
 3:             GSE35974  organism part             cerebellum
 4:             GSE35974        disease          schizophrenia
 5:             GSE35974 biological sex                   male
 6:             GSE35974        disease      mental depression
 7:             GSE12649        disease       Bipolar Disorder
 8:             GSE12649        disease          schizophrenia
 9:             GSE12649  organism part reference subject role
10:             GSE12649  organism part      prefrontal cortex</code></pre>
</div>
<div class="section level2">
<h2 id="output-options">Output options<a class="anchor" aria-label="anchor" href="#output-options"></a>
</h2>
<div class="section level3">
<h3 id="raw-data">Raw data<a class="anchor" aria-label="anchor" href="#raw-data"></a>
</h3>
<p>By default, Gemma API does some parsing on the raw API results to make it easier to work with inside of R. In the process, it drops some typically unused values. If you wish to fetch everything, use <code>raw = TRUE</code>. Instead of a data table, you’ll usually be served a list that represents the underlying JSON response.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_gene_locations.html">get_gene_locations</a></span><span class="op">(</span><span class="st">"DYRK1A"</span><span class="op">)</span></span></code></pre></div>
<pre><code>   chromosome strand nucleotide length taxon.Name  taxon.Scientific taxon.ID
1:         11      +   33890705 118714        rat Rattus norvegicus        3
2:         21      +   37365572 160785      human      Homo sapiens        1
3:         16      +   94370636 125741      mouse      Mus musculus        2
   taxon.NCBI taxon.Database.Name taxon.Database.ID
1:      10116                 rn6                86
2:       9606                hg38                87
3:      10090                mm10                81</code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_gene_locations.html">get_gene_locations</a></span><span class="op">(</span><span class="st">"DYRK1A"</span>, raw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>[[1]]
[[1]]$id
[1] 84782783

[[1]]$nucleotide
[1] 33890705

[[1]]$nucleotideLength
[1] 118714

[[1]]$strand
[1] "+"

[[1]]$bin
[1] 105

[[1]]$chromosome
[1] "11"

[[1]]$taxon
[[1]]$taxon$id
[1] 3

[[1]]$taxon$scientificName
[1] "Rattus norvegicus"

[[1]]$taxon$commonName
[1] "rat"

[[1]]$taxon$ncbiId
[1] 10116

[[1]]$taxon$externalDatabase
[[1]]$taxon$externalDatabase$id
[1] 86

[[1]]$taxon$externalDatabase$name
[1] "rn6"

[[1]]$taxon$externalDatabase$description
[1] "RGSC Rnor_6.0"

[[1]]$taxon$externalDatabase$uri
[1] "https://genome.ucsc.edu/cgi-bin/hgTracks?db=rn6"

[[1]]$taxon$externalDatabase$releaseVersion
NULL

[[1]]$taxon$externalDatabase$releaseUrl
NULL

[[1]]$taxon$externalDatabase$lastUpdated
NULL

[[1]]$taxon$externalDatabase$externalDatabases
[[1]]$taxon$externalDatabase$externalDatabases[[1]]
[[1]]$taxon$externalDatabase$externalDatabases[[1]]$id
[1] 116

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$name
[1] "rn6 RNA-Seq annotations"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$description
[1] "Annotations provided by NCBI Genome and used by the RNA-Seq pipeline for rat data."

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$uri
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$releaseVersion
[1] "106"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Rattus_norvegicus/106"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$lastUpdated
[1] "2022-10-24T07:00:00.000+00:00"

[[1]]$taxon$externalDatabase$externalDatabases[[1]]$externalDatabases
list()


[[1]]$taxon$externalDatabase$externalDatabases[[2]]
[[1]]$taxon$externalDatabase$externalDatabases[[2]]$id
[1] 100

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$name
[1] "rn4 annotations"

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$description
NULL

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$uri
[1] "https://hgdownload.cse.ucsc.edu/goldenpath/rn4/database/"

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$releaseVersion
NULL

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$releaseUrl
NULL

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$lastUpdated
NULL

[[1]]$taxon$externalDatabase$externalDatabases[[2]]$externalDatabases
list()






[[2]]
[[2]]$id
[1] 84362978

[[2]]$nucleotide
[1] 37365572

[[2]]$nucleotideLength
[1] 160785

[[2]]$strand
[1] "+"

[[2]]$bin
[1] 108

[[2]]$chromosome
[1] "21"

[[2]]$taxon
[[2]]$taxon$id
[1] 1

[[2]]$taxon$scientificName
[1] "Homo sapiens"

[[2]]$taxon$commonName
[1] "human"

[[2]]$taxon$ncbiId
[1] 9606

[[2]]$taxon$externalDatabase
[[2]]$taxon$externalDatabase$id
[1] 87

[[2]]$taxon$externalDatabase$name
[1] "hg38"

[[2]]$taxon$externalDatabase$description
[1] "Genome Reference Consortium Human GRCh38.p13 (GCA_000001405.28)"

[[2]]$taxon$externalDatabase$uri
[1] "https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38"

[[2]]$taxon$externalDatabase$releaseVersion
[1] "GRCh38.p13"

[[2]]$taxon$externalDatabase$releaseUrl
NULL

[[2]]$taxon$externalDatabase$lastUpdated
[1] "2022-06-30T07:00:00.000+00:00"

[[2]]$taxon$externalDatabase$externalDatabases
[[2]]$taxon$externalDatabase$externalDatabases[[1]]
[[2]]$taxon$externalDatabase$externalDatabases[[1]]$id
[1] 94

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$name
[1] "hg38 annotations"

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$description
NULL

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$uri
[1] "https://hgdownload.cse.ucsc.edu/goldenpath/hg38/database/"

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$releaseVersion
[1] "GRCh38.p13"

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$releaseUrl
NULL

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$lastUpdated
[1] "2022-06-30T07:00:00.000+00:00"

[[2]]$taxon$externalDatabase$externalDatabases[[1]]$externalDatabases
list()


[[2]]$taxon$externalDatabase$externalDatabases[[2]]
[[2]]$taxon$externalDatabase$externalDatabases[[2]]$id
[1] 124

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$name
[1] "hg38 RNA-Seq annotations"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$description
[1] "Annotations provided by NCBI Genome and used by the RNA-Seq pipeline for human data."

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$uri
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$releaseVersion
[1] "110"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Homo_sapiens/110/"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$lastUpdated
[1] "2023-01-17T20:27:55.059+00:00"

[[2]]$taxon$externalDatabase$externalDatabases[[2]]$externalDatabases
list()






[[3]]
[[3]]$id
[1] 84610778

[[3]]$nucleotide
[1] 94370636

[[3]]$nucleotideLength
[1] 125741

[[3]]$strand
[1] "+"

[[3]]$bin
[1] 20

[[3]]$chromosome
[1] "16"

[[3]]$taxon
[[3]]$taxon$id
[1] 2

[[3]]$taxon$scientificName
[1] "Mus musculus"

[[3]]$taxon$commonName
[1] "mouse"

[[3]]$taxon$ncbiId
[1] 10090

[[3]]$taxon$externalDatabase
[[3]]$taxon$externalDatabase$id
[1] 81

[[3]]$taxon$externalDatabase$name
[1] "mm10"

[[3]]$taxon$externalDatabase$description
[1] "Genome Reference Consortium GRCm38, which includes approximately 2.6 Gb of sequence, is considered to be \"essentially complete\". The assembly includes chromosomes 1-19, X, Y, M (mitochondrial DNA) and chr*_random (unlocalized) and chrUn_* (unplaced clone contigs)."

[[3]]$taxon$externalDatabase$uri
[1] "https://genome.ucsc.edu/cgi-bin/hgTracks?db=mm10"

[[3]]$taxon$externalDatabase$releaseVersion
NULL

[[3]]$taxon$externalDatabase$releaseUrl
NULL

[[3]]$taxon$externalDatabase$lastUpdated
[1] "2022-06-30T07:00:00.000+00:00"

[[3]]$taxon$externalDatabase$externalDatabases
[[3]]$taxon$externalDatabase$externalDatabases[[1]]
[[3]]$taxon$externalDatabase$externalDatabases[[1]]$id
[1] 97

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$name
[1] "mm10 annotations"

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$description
NULL

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$uri
[1] "https://hgdownload.cse.ucsc.edu/goldenpath/mm10/database/"

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$releaseVersion
NULL

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$releaseUrl
NULL

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$lastUpdated
[1] "2022-06-30T07:00:00.000+00:00"

[[3]]$taxon$externalDatabase$externalDatabases[[1]]$externalDatabases
list()


[[3]]$taxon$externalDatabase$externalDatabases[[2]]
[[3]]$taxon$externalDatabase$externalDatabases[[2]]$id
[1] 115

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$name
[1] "mm10 RNA-Seq annotations"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$description
[1] "Annotations provided by NCBI Genome and used by the RNA-Seq pipeline for mouse data."

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$uri
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$releaseVersion
[1] "109"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$releaseUrl
[1] "https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Mus_musculus/109/"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$lastUpdated
[1] "2023-01-17T20:30:25.219+00:00"

[[3]]$taxon$externalDatabase$externalDatabases[[2]]$externalDatabases
list()






attr(,"call")
[1] "https://gemma.msl.ubc.ca/rest/v2/genes/DYRK1A/locations"</code></pre>
</div>
<div class="section level3">
<h3 id="file-outputs">File outputs<a class="anchor" aria-label="anchor" href="#file-outputs"></a>
</h3>
<p>Sometimes, you may wish to save results to a file for future inspection. You can do this simply by providing a filename to the <code>file</code> parameter. The extension for this file will be one of three options:</p>
<ol style="list-style-type: decimal">
<li>
<code>.json</code>, if you requested results with <code>raw=TRUE</code>
</li>
<li>
<code>.csv</code> if the results have no nested data tables</li>
<li>
<code>.rds</code> otherwise</li>
</ol>
<p>You can also specify whether or not the new fetched results are allowed to overwrite an existing file by specifying the <code>overwrite = TRUE</code> parameter.</p>
</div>
<div class="section level3">
<h3 id="memoise-data">Memoise data<a class="anchor" aria-label="anchor" href="#memoise-data"></a>
</h3>
<p>To speed up results, you can remember past results so future queries can proceed virtually instantly. This is enabled through the <a href="https://github.com/r-lib/memoise" class="external-link"><code>memoise</code></a> package. To enable memoisation, simply set <code>memoised = TRUE</code> in the function call whenever you want to refer to the cache, both to save data for future calls and use the saved data for repeated calls.</p>
<p>By default this will create a cache in your local filesystem.</p>
<p>You can see the path to your cache by using <code>rappdirs::user_cache_dir(appname = "gemmaR")</code>. If you want to use a different directory, use <code>options(gemma.cache = 'path')</code>.</p>
<p>If you’re done with your fetching and want to ensure no space is being used for cached results, or if you just want to ensure you’re getting up-to-date data from Gemma, you can clear the cache using <a href="https://pavlidislab.github.io/gemma.R/reference/forget_gemma_memoised.html" class="external-link"><code>forget_gemma_memoised</code></a>.</p>
</div>
<div class="section level3">
<h3 id="changing-defaults">Changing defaults<a class="anchor" aria-label="anchor" href="#changing-defaults"></a>
</h3>
<p>We’ve seen how to change <code>raw = TRUE</code>, <code>overwrite = TRUE</code> and <code>memoised = TRUE</code> in individual function calls. It’s possible that you want to always use the functions these ways without specifying the option every time. You can do this by simply changing the default, which is visible in the function definition. See below for examples.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.memoised <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always refer to cache</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always overwrite when saving files</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.raw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always receive results as-is from Gemma</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] viridis_0.6.2               viridisLite_0.4.1          
 [3] pheatmap_1.0.12             SummarizedExperiment_1.28.0
 [5] Biobase_2.58.0              GenomicRanges_1.50.2       
 [7] GenomeInfoDb_1.34.6         IRanges_2.32.0             
 [9] S4Vectors_0.36.1            BiocGenerics_0.44.0        
[11] MatrixGenerics_1.10.0       matrixStats_0.63.0         
[13] ggrepel_0.9.2               ggplot2_3.4.0              
[15] dplyr_1.0.10                gemma.R_1.1.1              
[17] BiocStyle_2.26.0           

loaded via a namespace (and not attached):
 [1] httr_1.4.4             sass_0.4.5             bit64_4.0.5           
 [4] jsonlite_1.8.4         bslib_0.4.2            highr_0.10            
 [7] BiocManager_1.30.19    GenomeInfoDbData_1.2.9 yaml_2.3.7            
[10] pillar_1.8.1           lattice_0.20-45        glue_1.6.2            
[13] digest_0.6.31          RColorBrewer_1.1-3     XVector_0.38.0        
[16] colorspace_2.1-0       htmltools_0.5.4        Matrix_1.5-1          
[19] pkgconfig_2.0.3        bookdown_0.32          zlibbioc_1.44.0       
[22] purrr_1.0.1            scales_1.2.1           timechange_0.2.0      
[25] tibble_3.1.8           farver_2.1.1           generics_0.1.3        
[28] cachem_1.0.6           withr_2.5.0            cli_3.6.0             
[31] magrittr_2.0.3         memoise_2.0.1          evaluate_0.20         
[34] fs_1.6.0               fansi_1.0.4            textshaping_0.3.6     
[37] tools_4.2.2            data.table_1.14.6      lifecycle_1.0.3       
[40] stringr_1.5.0          munsell_0.5.0          DelayedArray_0.24.0   
[43] compiler_4.2.2         pkgdown_2.0.7          jquerylib_0.1.4       
[46] systemfonts_1.0.4      rlang_1.0.6            grid_4.2.2            
[49] RCurl_1.98-1.9         rappdirs_0.3.3         labeling_0.4.2        
[52] bitops_1.0-7           rmarkdown_2.20         gtable_0.3.1          
[55] curl_5.0.0             R6_2.5.1               lubridate_1.9.1       
[58] gridExtra_2.3          knitr_1.41             fastmap_1.1.0         
[61] bit_4.0.5              utf8_1.2.2             rprojroot_2.0.3       
[64] ragg_1.2.5             desc_1.4.2             stringi_1.7.12        
[67] Rcpp_1.0.10            vctrs_0.5.2            tidyselect_1.2.0      
[70] xfun_0.36             </code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
