<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Accessing curated gene expression data with gemma.R • gemma.R</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Accessing curated gene expression data with gemma.R">
<meta property="og:description" content="gemma.R">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="gemma.R_files/kePrint-0.0.1/kePrint.js"></script><link href="gemma.R_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<link href="gemma.R_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="gemma.R_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="gemma.R_files/jsoneditor-0.18.7/jsoneditor.min.js"></script><script src="gemma.R_files/jsonedit-binding-4.0.0/jsonedit.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Accessing curated gene expression data with
gemma.R</h1>
                        <h4 data-toc-skip class="author">Javier
Castillo-Arnemann</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia,
Vancouver,
Canada<br><h4 data-toc-skip class="author">Jordan
Sicherman</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia,
Vancouver,
Canada<br><h4 data-toc-skip class="author">B.
Ogan Mancarci</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia,
Vancouver, Canada<br><small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/vignettes/gemma.R.Rmd" class="external-link"><code>vignettes/gemma.R.Rmd</code></a></small>
      <div class="hidden name"><code>gemma.R.Rmd</code></div>

    </address>
</address>
</address>
</div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pavlidislab.github.io/gemma.R/" class="external-link">gemma.R</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggrepel.slowkow.com/" class="external-link">ggrepel</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/timelyportfolio/listviewer" class="external-link">listviewer</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="about-gemma">About Gemma<a class="anchor" aria-label="anchor" href="#about-gemma"></a>
</h2>
<p><a href="https://gemma.msl.ubc.ca/" class="external-link">Gemma</a> is a web site, database
and a set of tools for the meta-analysis, re-use and sharing of genomics
data, currently primarily targeted at the analysis of gene expression
profiles. Gemma contains data from thousands of public studies,
referencing thousands of published papers. Every dataset in Gemma has
passed a rigorous curation process that re-annotates the expression
platform at the sequence level, which allows for more consistent
cross-platform comparisons and meta-analyses.</p>
<p>For detailed information on the curation process, read this <a href="https://pavlidislab.github.io/Gemma/curation.html" class="external-link">page</a> or the
latest <a href="https://academic.oup.com/database/article/doi/10.1093/database/baab006/6143045" class="external-link">publication</a>.</p>
</div>
<div class="section level2">
<h2 id="package-cheat-sheet">Package cheat sheet<a class="anchor" aria-label="anchor" href="#package-cheat-sheet"></a>
</h2>
<p><img src="cheat_sheet.png" style="width:100.0%"></p>
</div>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<div class="section level3">
<h3 id="bioconductor">Bioconductor<a class="anchor" aria-label="anchor" href="#bioconductor"></a>
</h3>
<p>You can install <code>gemma.R</code> through <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> with the following
code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"gemma.R"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="searching-for-datasets-of-interest-in-gemma">Searching for datasets of interest in Gemma<a class="anchor" aria-label="anchor" href="#searching-for-datasets-of-interest-in-gemma"></a>
</h2>
<p>Using the <a href="https://pavlidislab.github.io/gemma.R/reference/get_datasets.html" class="external-link"><code>get_datasets</code></a>
function, datasets fitting various criteria can be accessed.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># accessing all mouse and human datasets</span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'mouse'</span>,<span class="st">'human'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>, </span>
<span>           <span class="va">experiment.description</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
experiment.shortName
</th>
<th style="text-align:left;">
experiment.name
</th>
<th style="text-align:left;">
experiment.description
</th>
<th style="text-align:left;">
taxon.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GSE2018
</td>
<td style="text-align:left;">
Human Lung Transplant - BAL
</td>
<td style="text-align:left;">
Bronchoalveolar lavage samp…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE4523
</td>
<td style="text-align:left;">
Expression Studies of Melan…
</td>
<td style="text-align:left;">
Melanotransferrin (MTf) or …
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE4036
</td>
<td style="text-align:left;">
perro-affy-human-186940
</td>
<td style="text-align:left;">
Our laboratory has develope…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE4034
</td>
<td style="text-align:left;">
palme-affy-mouse-198967
</td>
<td style="text-align:left;">
Fear conditioning (FC) is a…
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE2866
</td>
<td style="text-align:left;">
Donarum-3R01NS040270-03S1
</td>
<td style="text-align:left;">
Succinate semialdehyde dehy…
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE3253
</td>
<td style="text-align:left;">
Exaggerated neuroinflammati…
</td>
<td style="text-align:left;">
Acute cognitive impairment …
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># accessing human datasets with the word "bipolar"</span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>query <span class="op">=</span> <span class="st">'bipolar'</span>,taxa <span class="op">=</span> <span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>, </span>
<span>           <span class="va">experiment.description</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
experiment.shortName
</th>
<th style="text-align:left;">
experiment.name
</th>
<th style="text-align:left;">
experiment.description
</th>
<th style="text-align:left;">
taxon.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GSE157509
</td>
<td style="text-align:left;">
Increased IL-6 and altered …
</td>
<td style="text-align:left;">
The goals of this study ar…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE66196
</td>
<td style="text-align:left;">
Bipolar disorder and lithiu…
</td>
<td style="text-align:left;">
Bipolar disorder is a seve…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE210064
</td>
<td style="text-align:left;">
Gene expression alterations…
</td>
<td style="text-align:left;">
Gene expression of samples…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE23848
</td>
<td style="text-align:left;">
Peripheral blood gene-expre…
</td>
<td style="text-align:left;">
Analysis of gene-expression…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
McLean Hippocampus
</td>
<td style="text-align:left;">
McLean Hippocampus
</td>
<td style="text-align:left;">
Hippocampus of schizophreni…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
byne-cc
</td>
<td style="text-align:left;">
Corpus Callosum data from S…
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
human
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># access human datasets that were annotated with the ontology term for the</span></span>
<span><span class="co"># bipolar disorder</span></span>
<span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>taxa <span class="op">=</span><span class="st">'human'</span>, </span>
<span>             uris <span class="op">=</span> <span class="st">'http://purl.obolibrary.org/obo/MONDO_0004985'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>, </span>
<span>           <span class="va">experiment.description</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
experiment.shortName
</th>
<th style="text-align:left;">
experiment.name
</th>
<th style="text-align:left;">
experiment.description
</th>
<th style="text-align:left;">
taxon.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GSE5389
</td>
<td style="text-align:left;">
Adult postmortem brain tiss…
</td>
<td style="text-align:left;">
Bipolar affective disorder …
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE5388
</td>
<td style="text-align:left;">
Adult postmortem brain tiss…
</td>
<td style="text-align:left;">
Bipolar affective disorder …
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE7036
</td>
<td style="text-align:left;">
Expression profiling in mon…
</td>
<td style="text-align:left;">
To identify genes dysregula…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
McLean Hippocampus
</td>
<td style="text-align:left;">
McLean Hippocampus
</td>
<td style="text-align:left;">
Hippocampus of schizophreni…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
McLean_PFC
</td>
<td style="text-align:left;">
McLean_PFC
</td>
<td style="text-align:left;">
Prefrontal cortex of schizo…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
stanley_feinberg
</td>
<td style="text-align:left;">
Stanley consortium collecti…
</td>
<td style="text-align:left;">
50 samples of individuals f…
</td>
<td style="text-align:left;">
human
</td>
</tr>
</tbody>
</table>
<p>Note that a single call of these functions will only return 20
results by default and a 100 results maximum, controlled by the
<code>limit</code> argument. In order to get all available results, use
<code>get_all_pages</code> function on the output of the function</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>taxa <span class="op">=</span> <span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/get_all_pages.html">get_all_pages</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>, </span>
<span>           <span class="va">experiment.description</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
experiment.shortName
</th>
<th style="text-align:left;">
experiment.name
</th>
<th style="text-align:left;">
experiment.description
</th>
<th style="text-align:left;">
taxon.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GSE2018
</td>
<td style="text-align:left;">
Human Lung Transplant - BAL
</td>
<td style="text-align:left;">
Bronchoalveolar lavage samp…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE4036
</td>
<td style="text-align:left;">
perro-affy-human-186940
</td>
<td style="text-align:left;">
Our laboratory has develope…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE3489
</td>
<td style="text-align:left;">
Patterns of gene dysregulat…
</td>
<td style="text-align:left;">
The neurodegenerative proce…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE1923
</td>
<td style="text-align:left;">
Identification of PDGF-depe…
</td>
<td style="text-align:left;">
Overall study: Identificati…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE361
</td>
<td style="text-align:left;">
Mammary epithelial cell tra…
</td>
<td style="text-align:left;">
Analysis of gene expression…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE492
</td>
<td style="text-align:left;">
Effect of prostaglandin ana…
</td>
<td style="text-align:left;">
The purpose of this study i…
</td>
<td style="text-align:left;">
human
</td>
</tr>
</tbody>
</table>
<p>See <a href="#larger-queries">larger queries</a> section for more
details. To keep this vignette simpler we will keep using the first 20
results returned by default in examples below.</p>
<p>The <code>filter</code> parameter of get_datasets allows filtering
for datasets with specific properties. A list of such properties can be
accessed using <code>filter_properties</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filter_properties.html">filter_properties</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">dataset</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
properties
</th>
<th style="text-align:left;">
type
</th>
<th style="text-align:left;">
description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
accession.accession
</td>
<td style="text-align:left;">
string
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
accession.accessionVersion
</td>
<td style="text-align:left;">
string
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
accession.externalDatabase….
</td>
<td style="text-align:left;">
string
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
accession.externalDatabase.id
</td>
<td style="text-align:left;">
integer
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
accession.externalDatabase….
</td>
<td style="text-align:left;">
string
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
accession.externalDatabase….
</td>
<td style="text-align:left;">
string
</td>
<td style="text-align:left;">
NA
</td>
</tr>
</tbody>
</table>
<p>These properties can be used together to fine tune your results</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all datasets with more than 4 samples annotated for any disease</span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>filter <span class="op">=</span> <span class="st">'bioAssayCount &gt; 4 and allCharacteristics.category = disease'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>, </span>
<span>           <span class="va">experiment.description</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
experiment.shortName
</th>
<th style="text-align:left;">
experiment.name
</th>
<th style="text-align:left;">
experiment.description
</th>
<th style="text-align:left;">
taxon.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GSE2018
</td>
<td style="text-align:left;">
Human Lung Transplant - BAL
</td>
<td style="text-align:left;">
Bronchoalveolar lavage samp…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE4036
</td>
<td style="text-align:left;">
perro-affy-human-186940
</td>
<td style="text-align:left;">
Our laboratory has develope…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE2866
</td>
<td style="text-align:left;">
Donarum-3R01NS040270-03S1
</td>
<td style="text-align:left;">
Succinate semialdehyde dehy…
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE2426
</td>
<td style="text-align:left;">
Pre-Neoplastic Stage of Med…
</td>
<td style="text-align:left;">
SUMMARY Medulloblastoma is…
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE2867
</td>
<td style="text-align:left;">
Zoghbi-5R01NS027699-14
</td>
<td style="text-align:left;">
A number of human neurodege…
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE3489
</td>
<td style="text-align:left;">
Patterns of gene dysregulat…
</td>
<td style="text-align:left;">
The neurodegenerative proce…
</td>
<td style="text-align:left;">
human
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># all datasets with ontology terms for Alzheimer's disease and Parkinson's disease</span></span>
<span><span class="co"># this is equivalent to using the uris parameter</span></span>
<span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>filter <span class="op">=</span> <span class="st">'allCharacteristics.valueUri in (http://purl.obolibrary.org/obo/MONDO_0004975,http://purl.obolibrary.org/obo/MONDO_0005180</span></span>
<span><span class="st">)'</span><span class="op">)</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>, </span>
<span>           <span class="va">experiment.description</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
experiment.shortName
</th>
<th style="text-align:left;">
experiment.name
</th>
<th style="text-align:left;">
experiment.description
</th>
<th style="text-align:left;">
taxon.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GSE1555
</td>
<td style="text-align:left;">
Vehicle and sAPPalpha-treat…
</td>
<td style="text-align:left;">
Gene expression changes ind…
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE4757
</td>
<td style="text-align:left;">
Rogers-3U24NS043571-01S1
</td>
<td style="text-align:left;">
Alzheimer’s Disease (AD) is…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE6613
</td>
<td style="text-align:left;">
Parkinson’s disease vs. con…
</td>
<td style="text-align:left;">
Parkinson?s disease (PD) pr…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE12685
</td>
<td style="text-align:left;">
Expression of mRNAs Regulat…
</td>
<td style="text-align:left;">
In Alzheimer’s disease (AD)…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE14499
</td>
<td style="text-align:left;">
Effect of BDNF on the APP t…
</td>
<td style="text-align:left;">
We examined transgenic (TG)…
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE10908
</td>
<td style="text-align:left;">
Differential gene expressio…
</td>
<td style="text-align:left;">
In a transgenic mouse model…
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
</tbody>
</table>
<p>Dataset information provided by <code>get_datasets</code> also
includes some quality information that can be used to determine the
suitability of any given experiment. For instance
<code>experiment.batchEffect</code> column will be set to -1 if Gemma’s
preprocessing has detected batch effects that were unable to be resolved
by batch correction. More information about these and other fields can
be found at the function documentation.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>taxa <span class="op">=</span> <span class="st">'human'</span>, filter <span class="op">=</span> <span class="st">'bioAssayCount &gt; 4'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>     <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">experiment.batchEffect</span> <span class="op">!=</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>, </span>
<span>           <span class="va">experiment.description</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
experiment.shortName
</th>
<th style="text-align:left;">
experiment.name
</th>
<th style="text-align:left;">
experiment.description
</th>
<th style="text-align:left;">
taxon.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GSE2018
</td>
<td style="text-align:left;">
Human Lung Transplant - BAL
</td>
<td style="text-align:left;">
Bronchoalveolar lavage samp…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE4036
</td>
<td style="text-align:left;">
perro-affy-human-186940
</td>
<td style="text-align:left;">
Our laboratory has develope…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE3489
</td>
<td style="text-align:left;">
Patterns of gene dysregulat…
</td>
<td style="text-align:left;">
The neurodegenerative proce…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE1923
</td>
<td style="text-align:left;">
Identification of PDGF-depe…
</td>
<td style="text-align:left;">
Overall study: Identificati…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE361
</td>
<td style="text-align:left;">
Mammary epithelial cell tra…
</td>
<td style="text-align:left;">
Analysis of gene expression…
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE492
</td>
<td style="text-align:left;">
Effect of prostaglandin ana…
</td>
<td style="text-align:left;">
The purpose of this study i…
</td>
<td style="text-align:left;">
human
</td>
</tr>
</tbody>
</table>
<p>Gemma uses multiple ontologies when annotating datasets and using the
term URIs instead of free text to search can lead to more specific
results. <a href="https://pavlidislab.github.io/gemma.R/reference/search_annotations.html" class="external-link"><code>search_annotations</code></a>
function allows searching for annotation terms that might be relevant to
your query.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/search_annotations.html">search_annotations</a></span><span class="op">(</span><span class="st">'bipolar'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
category.name
</th>
<th style="text-align:left;">
category.URI
</th>
<th style="text-align:left;">
value.name
</th>
<th style="text-align:left;">
value.URI
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
bipolar i disorder
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0009963
</td>
</tr>
<tr>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
bipolar ii disorder
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0009964
</td>
</tr>
<tr>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
bipolar disorder
</td>
<td style="text-align:left;">
<a href="http://purl.obolibrary" class="external-link uri">http://purl.obolibrary</a>…/MONDO_0004985
</td>
</tr>
<tr>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
bipolar affective disorder
</td>
<td style="text-align:left;">
<a href="http://purl.obolibrary" class="external-link uri">http://purl.obolibrary</a>…/HP_0007302
</td>
</tr>
<tr>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
bipolar disorder
</td>
<td style="text-align:left;">
<a href="http://purl.obolibrary" class="external-link uri">http://purl.obolibrary</a>…/NBO_0000258
</td>
</tr>
<tr>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
bipolar neuron
</td>
<td style="text-align:left;">
<a href="http://purl.obolibrary" class="external-link uri">http://purl.obolibrary</a>…/CL_0000103
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="downloading-expression-data">Downloading expression data<a class="anchor" aria-label="anchor" href="#downloading-expression-data"></a>
</h2>
<p>Upon identifying datasets of interest, more information about
specific ones can be requested. In this example we will be using
GSE46416 which includes samples taken from healthy donors along with
manic/euthymic phase bipolar disorder patients.</p>
<p>The data associated with specific experiments can be accessed by
using <a href="https://pavlidislab.github.io/gemma.R/reference/get_datasets_by_ids.html" class="external-link"><code>get_datasets_by_ids</code></a></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_datasets_by_ids.html">get_datasets_by_ids</a></span><span class="op">(</span><span class="st">"GSE46416"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">experiment.name</span>, </span>
<span>           <span class="va">experiment.description</span>,<span class="va">taxon.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
experiment.shortName
</th>
<th style="text-align:left;">
experiment.name
</th>
<th style="text-align:left;">
experiment.description
</th>
<th style="text-align:left;">
taxon.name
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
GSE46416
</td>
<td style="text-align:left;">
State- and trait-specific g…
</td>
<td style="text-align:left;">
Gene expression profiles of…
</td>
<td style="text-align:left;">
human
</td>
</tr></tbody>
</table>
<p>To access the expression data in a convenient form, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/get_dataset_object.html" class="external-link"><code>get_dataset_object</code></a>.
It is a high-level wrapper that combines various endpoint calls to
return lists of annotated <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link"><code>SummarizedExperiment</code></a>
or <a href="https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf" class="external-link"><code>ExpressionSet</code></a>
objects that are compatible with other Bioconductor packages or a <a href="https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html" class="external-link">tidyverse-friendly</a>
long form tibble for downstream analyses. These include the expression
matrix along with the experimental design, and ensure the sample names
match between both when transforming/subsetting data.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset_object.html">get_dataset_object</a></span><span class="op">(</span><span class="st">"GSE46416"</span>,</span>
<span>                          type <span class="op">=</span> <span class="st">'se'</span><span class="op">)</span> <span class="co"># SummarizedExperiment is the default output type</span></span></code></pre></div>
<p>Note that the tidy format is less memory efficient but allows easy
visualization and exploration with <a href="https://ggplot2.tidyverse.org/index.html" class="external-link">ggplot2</a> and the rest
of the <a href="https://www.tidyverse.org/" class="external-link">tidyverse</a>.</p>
<p>To show how subsetting works, we’ll keep the “manic phase” data and
the <code>reference_subject_role</code>s, which refers to the control
samples in Gemma datasets.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check the levels of the disease factor</span></span>
<span><span class="va">dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">disease</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "bipolar disorder has_modifier euthymic phase"
[2] "reference subject role"                      
[3] "bipolar disorder has_modifier manic phase"   </code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Subset patients during manic phase and controls</span></span>
<span><span class="va">manic</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="va">dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">disease</span> <span class="op">==</span> <span class="st">"bipolar disorder,manic phase"</span> <span class="op">|</span> </span>
<span>        <span class="va">dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">disease</span> <span class="op">==</span> <span class="st">"reference subject role"</span><span class="op">]</span></span>
<span><span class="va">manic</span></span></code></pre></div>
<pre><code>class: SummarizedExperiment 
dim: 18758 10 
metadata(8): title abstract ... gemmaSuitabilityScore taxon
assays(1): counts
rownames(18758): 2315430 2315554 ... 7385683 7385696
rowData names(4): Probe GeneSymbol GeneName NCBIid
colnames(10): Control, 15 Control, 8 ... Control, 1_DE50 Control,
  2_DE40
colData names(3): factorValues disease block</code></pre>
<p>Let’s take a look at sample to sample correlation in our subset.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get Expression matrix</span></span>
<span><span class="va">manicExpr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">manic</span>, <span class="st">"counts"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">manicExpr</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">cor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span>col <span class="op">=</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>,border_color <span class="op">=</span> <span class="cn">NA</span>,angle_col <span class="op">=</span> <span class="fl">45</span>,fontsize <span class="op">=</span> <span class="fl">7</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="gemma.R_files/figure-html/boxplot-1.png" alt="Sample to sample correlations of bipolar patients during manic phase and controls." width="700"><p class="caption">
Sample to sample correlations of bipolar patients during manic phase and
controls.
</p>
</div>
<p>You can also use <a href="https://pavlidislab.github.io/gemma.R/reference/get_dataset_expression.html" class="external-link"><code>get_dataset_expression</code></a>
to only get the expression matrix, <a href="https://pavlidislab.github.io/gemma.R/reference/get_dataset_samples.html" class="external-link"><code>get_dataset_samples</code></a>
to get the metadata information. The output of this function can be
simplified to match the version included in the output of
<code>get_dataset_object</code> by using <a href="https://pavlidislab.github.io/gemma.R/reference/make_design" class="external-link"><code>make_design</code></a>
on the output.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_dataset_samples.html">get_dataset_samples</a></span><span class="op">(</span><span class="st">'GSE46416'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/make_design.html">make_design</a></span><span class="op">(</span><span class="st">'text'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
factorValues
</th>
<th style="text-align:left;">
disease
</th>
<th style="text-align:left;">
block
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
Bipolar disorder patient euthymic phase, 17
</td>
<td style="text-align:left;">
c(“disea….
</td>
<td style="text-align:left;">
bipolar disorder has_modifi…
</td>
<td style="text-align:left;">
Batch_02_26/11/09
</td>
</tr>
<tr>
<td style="text-align:left;">
Bipolar disorder patient euthymic phase, 34
</td>
<td style="text-align:left;">
c(“disea….
</td>
<td style="text-align:left;">
bipolar disorder has_modifi…
</td>
<td style="text-align:left;">
Batch_04_02/12/09
</td>
</tr>
<tr>
<td style="text-align:left;">
Control, 15
</td>
<td style="text-align:left;">
c(“disea….
</td>
<td style="text-align:left;">
reference subject role
</td>
<td style="text-align:left;">
Batch_02_26/11/09
</td>
</tr>
<tr>
<td style="text-align:left;">
Bipolar disorder patient euthymic phase, 32
</td>
<td style="text-align:left;">
c(“disea….
</td>
<td style="text-align:left;">
bipolar disorder has_modifi…
</td>
<td style="text-align:left;">
Batch_04_02/12/09
</td>
</tr>
<tr>
<td style="text-align:left;">
Control, 8
</td>
<td style="text-align:left;">
c(“disea….
</td>
<td style="text-align:left;">
reference subject role
</td>
<td style="text-align:left;">
Batch_01_25/11/09
</td>
</tr>
<tr>
<td style="text-align:left;">
Bipolar disorder patient manic phase, 21
</td>
<td style="text-align:left;">
c(“block….
</td>
<td style="text-align:left;">
bipolar disorder has_modifi…
</td>
<td style="text-align:left;">
Batch_03_27/11/09
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="platform-annotations">Platform Annotations<a class="anchor" aria-label="anchor" href="#platform-annotations"></a>
</h2>
<p>Expression data in Gemma comes with annotations for the gene each
expression profile corresponds to. Using the <a href="https://pavlidislab.github.io/gemma.R/reference/get_platform_annotations.html" class="external-link"><code>get_platform_annotations</code></a>
function, these annotations can be retrieved independently of the
expression data, along with additional annotations such as Gene Ontology
terms.</p>
<p>Examples:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platform_annotations.html">get_platform_annotations</a></span><span class="op">(</span><span class="st">'GPL96'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">GOTerms</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>     ProbeName   GeneSymbols                                     GeneNames
        &lt;char&gt;        &lt;char&gt;                                        &lt;char&gt;
1: 211750_x_at TUBA1C|TUBA1A             tubulin alpha 1c|tubulin alpha 1a
2:   216678_at                                                            
3:   216345_at        ZSWIM8            zinc finger SWIM-type containing 8
4:   207273_at                                                            
5: 216025_x_at        CYP2C9 cytochrome P450 family 2 subfamily C member 9
6: 218191_s_at        LMBRD1                     LMBR1 domain containing 1
        GemmaIDs    NCBIids
          &lt;char&gt;     &lt;char&gt;
1: 360802|172797 84790|7846
2:                         
3:        235733      23053
4:                         
5:         32964       1559
6:        303717      55788</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platform_annotations.html">get_platform_annotations</a></span><span class="op">(</span><span class="st">'Generic_human_ncbiIds'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">GOTerms</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>   ElementName  GeneSymbols
         &lt;int&gt;       &lt;char&gt;
1:       55236         UBA6
2:       79664         ICE2
3:   100126270     FMR1-AS1
4:   105373684    LINC01818
5:   124900245 LOC124900245
6:   102725051 LOC102725051
                                               GeneNames GemmaIDs   NCBIids
                                                  &lt;char&gt;    &lt;int&gt;     &lt;int&gt;
1:           ubiquitin like modifier activating enzyme 6   295849     55236
2: interactor of little elongation complex ELL subunit 2   336840     79664
3:                                  FMR1 antisense RNA 1  3157248 100126270
4:           long intergenic non-protein coding RNA 1818  9235895 105373684
5:                           small nucleolar RNA SNORA40 10578422 124900245
6:                          uncharacterized LOC102725051  9008981 102725051</code></pre>
<p>If you are interested in a particular gene, you can see which
platforms include it using <a href="https://pavlidislab.github.io/gemma.R/reference/get_gene_probes.html" class="external-link"><code>get_gene_probes</code></a>.
Note that functions to search gene work best with unambigious
identifiers rather than symbols.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># lists genes in gemma matching the symbol or identifier</span></span>
<span><span class="fu"><a href="../reference/get_genes.html">get_genes</a></span><span class="op">(</span><span class="st">'Eno2'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
gene.symbol
</th>
<th style="text-align:left;">
gene.ensembl
</th>
<th style="text-align:right;">
gene.NCBI
</th>
<th style="text-align:left;">
gene.name
</th>
<th style="text-align:left;">
gene.aliases
</th>
<th style="text-align:right;">
gene.MFX.rank
</th>
<th style="text-align:left;">
taxon.name
</th>
<th style="text-align:left;">
taxon.scientific
</th>
<th style="text-align:right;">
taxon.ID
</th>
<th style="text-align:right;">
taxon.NCBI
</th>
<th style="text-align:left;">
taxon.database.name
</th>
<th style="text-align:right;">
taxon.database.ID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ENO2
</td>
<td style="text-align:left;">
ENSG00000111674
</td>
<td style="text-align:right;">
2026
</td>
<td style="text-align:left;">
enolase 2
</td>
<td style="text-align:left;">
HEL-S-27….
</td>
<td style="text-align:right;">
0.9033096
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
Eno2
</td>
<td style="text-align:left;">
ENSMUSG00000004267
</td>
<td style="text-align:right;">
13807
</td>
<td style="text-align:left;">
enolase 2, gamma neuronal
</td>
<td style="text-align:left;">
D6Ertd37….
</td>
<td style="text-align:right;">
0.8980022
</td>
<td style="text-align:left;">
mouse
</td>
<td style="text-align:left;">
Mus musculus
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
10090
</td>
<td style="text-align:left;">
mm10
</td>
<td style="text-align:right;">
81
</td>
</tr>
<tr>
<td style="text-align:left;">
Eno2
</td>
<td style="text-align:left;">
ENSRNOG00000013141
</td>
<td style="text-align:right;">
24334
</td>
<td style="text-align:left;">
enolase 2
</td>
<td style="text-align:left;">
NSE, RNEN3
</td>
<td style="text-align:right;">
0.8822730
</td>
<td style="text-align:left;">
rat
</td>
<td style="text-align:left;">
Rattus norvegicus
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
10116
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:right;">
86
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ncbi id for human ENO2</span></span>
<span><span class="va">probes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_gene_probes.html">get_gene_probes</a></span><span class="op">(</span><span class="fl">2026</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># remove the description for brevity of output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">probes</span><span class="op">[</span>,<span class="va">.SD</span>, .SDcols <span class="op">=</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">probes</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'mapping.Description'</span>,<span class="st">'platform.Description'</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
element.name
</th>
<th style="text-align:left;">
element.description
</th>
<th style="text-align:left;">
platform.shortName
</th>
<th style="text-align:left;">
platform.name
</th>
<th style="text-align:right;">
platform.ID
</th>
<th style="text-align:left;">
platform.type
</th>
<th style="text-align:left;">
platform.description
</th>
<th style="text-align:left;">
platform.troubled
</th>
<th style="text-align:left;">
taxon.name
</th>
<th style="text-align:left;">
taxon.scientific
</th>
<th style="text-align:right;">
taxon.ID
</th>
<th style="text-align:right;">
taxon.NCBI
</th>
<th style="text-align:left;">
taxon.database.name
</th>
<th style="text-align:right;">
taxon.database.ID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
201313_at
</td>
<td style="text-align:left;">
enolase 2 (gamma, neuronal)
</td>
<td style="text-align:left;">
GPL96
</td>
<td style="text-align:left;">
Affymetrix GeneChip Human G…
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
ONECOLOR
</td>
<td style="text-align:left;">
The U133 set includes 2 ar…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
201313_at
</td>
<td style="text-align:left;">
enolase 2 (gamma, neuronal)
</td>
<td style="text-align:left;">
GPL570
</td>
<td style="text-align:left;">
Affymetrix GeneChip Human G…
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
ONECOLOR
</td>
<td style="text-align:left;">
Complete coverage of the H…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
40193_at
</td>
<td style="text-align:left;">
enolase 2 (gamma, neuronal)
</td>
<td style="text-align:left;">
GPL91
</td>
<td style="text-align:left;">
Affymetrix GeneChip Human G…
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
ONECOLOR
</td>
<td style="text-align:left;">
The Human Genome U95 (HG-U…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
1639
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
GPL962
</td>
<td style="text-align:left;">
CHUGAI 41K
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:left;">
TWOCOLOR
</td>
<td style="text-align:left;">
Patients and Sample Prepar…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
6621
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
GPL230
</td>
<td style="text-align:left;">
UP_5Ka
</td>
<td style="text-align:right;">
38
</td>
<td style="text-align:left;">
TWOCOLOR
</td>
<td style="text-align:left;">
A cDNA microarray with ~50…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
861
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
GPL230
</td>
<td style="text-align:left;">
UP_5Ka
</td>
<td style="text-align:right;">
38
</td>
<td style="text-align:left;">
TWOCOLOR
</td>
<td style="text-align:left;">
A cDNA microarray with ~50…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="differential-expression-analyses">Differential expression analyses<a class="anchor" aria-label="anchor" href="#differential-expression-analyses"></a>
</h2>
<p>Gemma contains precomputed differential expression analyses for most
of its datasets. Analyses can involve more than one factor, such as
“sex” as well as “disease”. Some datasets contain more than one analysis
to account for different factors and their interactions. The results are
stored as resultSets, each corresponding to one factor (or their
interaction). You can access them using <a href="https://pavlidislab.github.io/gemma.R/reference/get_differential_expression_values.html" class="external-link"><code>get_differential_expression_values</code></a>.
From here on, we can explore and visualize the data to find the most
differentially-expressed genes</p>
<p>Note that <code>get_differential_expression_values</code> can return
multiple differentials per study if a study has multiple factors to
contrast. Since GSE46416 only has one extracting the first element of
the returned list is all we need.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dif_exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_differential_expression_values.html">get_differential_expression_values</a></span><span class="op">(</span><span class="st">'GSE46416'</span><span class="op">)</span></span>
<span><span class="va">dif_exp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
Probe
</th>
<th style="text-align:left;">
NCBIid
</th>
<th style="text-align:left;">
GeneSymbol
</th>
<th style="text-align:left;">
GeneName
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
corrected_pvalue
</th>
<th style="text-align:right;">
rank
</th>
<th style="text-align:right;">
contrast_113004_log2fc
</th>
<th style="text-align:right;">
contrast_113004_tstat
</th>
<th style="text-align:right;">
contrast_113004_pvalue
</th>
<th style="text-align:right;">
contrast_113005_log2fc
</th>
<th style="text-align:right;">
contrast_113005_tstat
</th>
<th style="text-align:right;">
contrast_113005_pvalue
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2982730
</td>
<td style="text-align:left;">
4018
</td>
<td style="text-align:left;">
LPA
</td>
<td style="text-align:left;">
lipoprotein(a)
</td>
<td style="text-align:right;">
0.9185
</td>
<td style="text-align:right;">
0.9521
</td>
<td style="text-align:right;">
0.9647
</td>
<td style="text-align:right;">
-0.0247
</td>
<td style="text-align:right;">
-0.3587
</td>
<td style="text-align:right;">
0.7221
</td>
<td style="text-align:right;">
-0.0249
</td>
<td style="text-align:right;">
-0.3622
</td>
<td style="text-align:right;">
0.7196
</td>
</tr>
<tr>
<td style="text-align:right;">
2787851
</td>
<td style="text-align:left;">
166752
</td>
<td style="text-align:left;">
FREM3
</td>
<td style="text-align:left;">
FRAS1 related extracellular…
</td>
<td style="text-align:right;">
0.5850
</td>
<td style="text-align:right;">
0.7348
</td>
<td style="text-align:right;">
0.7960
</td>
<td style="text-align:right;">
0.1895
</td>
<td style="text-align:right;">
1.0119
</td>
<td style="text-align:right;">
0.3191
</td>
<td style="text-align:right;">
0.1403
</td>
<td style="text-align:right;">
0.7495
</td>
<td style="text-align:right;">
0.4590
</td>
</tr>
<tr>
<td style="text-align:right;">
2477558
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
0.3965
</td>
<td style="text-align:right;">
0.5931
</td>
<td style="text-align:right;">
0.6685
</td>
<td style="text-align:right;">
0.0918
</td>
<td style="text-align:right;">
1.1386
</td>
<td style="text-align:right;">
0.2633
</td>
<td style="text-align:right;">
0.1016
</td>
<td style="text-align:right;">
1.2604
</td>
<td style="text-align:right;">
0.2166
</td>
</tr>
<tr>
<td style="text-align:right;">
2910917
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
0.6506
</td>
<td style="text-align:right;">
0.7815
</td>
<td style="text-align:right;">
0.8325
</td>
<td style="text-align:right;">
0.1564
</td>
<td style="text-align:right;">
0.6739
</td>
<td style="text-align:right;">
0.5052
</td>
<td style="text-align:right;">
0.2096
</td>
<td style="text-align:right;">
0.9032
</td>
<td style="text-align:right;">
0.3731
</td>
</tr>
<tr>
<td style="text-align:right;">
3983537
</td>
<td style="text-align:left;">
140886
</td>
<td style="text-align:left;">
PABPC5
</td>
<td style="text-align:left;">
poly(A) binding protein cyt…
</td>
<td style="text-align:right;">
0.1064
</td>
<td style="text-align:right;">
0.3227
</td>
<td style="text-align:right;">
0.3299
</td>
<td style="text-align:right;">
0.1845
</td>
<td style="text-align:right;">
2.0320
</td>
<td style="text-align:right;">
0.0505
</td>
<td style="text-align:right;">
0.1603
</td>
<td style="text-align:right;">
1.7656
</td>
<td style="text-align:right;">
0.0869
</td>
</tr>
<tr>
<td style="text-align:right;">
3538213
</td>
<td style="text-align:left;">
23002
</td>
<td style="text-align:left;">
DAAM1
</td>
<td style="text-align:left;">
dishevelled associated acti…
</td>
<td style="text-align:right;">
0.2271
</td>
<td style="text-align:right;">
0.4477
</td>
<td style="text-align:right;">
0.5073
</td>
<td style="text-align:right;">
0.3398
</td>
<td style="text-align:right;">
1.7279
</td>
<td style="text-align:right;">
0.0936
</td>
<td style="text-align:right;">
0.1200
</td>
<td style="text-align:right;">
0.6104
</td>
<td style="text-align:right;">
0.5459
</td>
</tr>
</tbody>
</table>
<p>By default the columns names of the output correspond to contrast
IDs. To see what conditions these IDs correspond to we can either use
<code>get_dataset_differential_expression_analyses</code> to get the
metadata about differentials of a given dataset, or set
<code>readableContrasts</code> argument of
<code>get_differential_expression_values</code> to <code>TRUE</code>.
The former approach is usually better for a large scale systematic
analysis while the latter is easier to read in an interactive
session.</p>
<p><code>get_dataset_differential_expression_analyses</code> function
returns structured metadata about the differentials.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset_differential_expression_analyses.html">get_dataset_differential_expression_analyses</a></span><span class="op">(</span><span class="st">'GSE46416'</span><span class="op">)</span></span>
<span><span class="va">contrasts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
result.ID
</th>
<th style="text-align:right;">
contrast.ID
</th>
<th style="text-align:right;">
experiment.ID
</th>
<th style="text-align:left;">
factor.category
</th>
<th style="text-align:left;">
factor.category.URI
</th>
<th style="text-align:left;">
factor.ID
</th>
<th style="text-align:left;">
baseline.factors
</th>
<th style="text-align:left;">
experimental.factors
</th>
<th style="text-align:left;">
isSubset
</th>
<th style="text-align:left;">
subsetFactor
</th>
<th style="text-align:right;">
probes.analyzed
</th>
<th style="text-align:right;">
genes.analyzed
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
550248
</td>
<td style="text-align:right;">
113004
</td>
<td style="text-align:right;">
8997
</td>
<td style="text-align:left;">
disease
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000408
</td>
<td style="text-align:left;">
19134
</td>
<td style="text-align:left;">
disease,….
</td>
<td style="text-align:left;">
disease,….
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
logical(….
</td>
<td style="text-align:right;">
21961
</td>
<td style="text-align:right;">
18959
</td>
</tr>
<tr>
<td style="text-align:right;">
550248
</td>
<td style="text-align:right;">
113005
</td>
<td style="text-align:right;">
8997
</td>
<td style="text-align:left;">
disease
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000408
</td>
<td style="text-align:left;">
19134
</td>
<td style="text-align:left;">
disease,….
</td>
<td style="text-align:left;">
disease,….
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
logical(….
</td>
<td style="text-align:right;">
21961
</td>
<td style="text-align:right;">
18959
</td>
</tr>
</tbody>
</table>
<p><code>contrast.ID</code> column corresponds to the column names in
the output of <code>get_differential_expression_values</code> while
<code>result.ID</code> corresponds to the name of the differential in
the output object. Using them together will let one to access
differentially expressed gene counts for each condition contrast</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># using result.ID and contrast.ID of the output above, we can access specific</span></span>
<span><span class="co"># results. Note that one study may have multiple contrast objects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">contrasts</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">result_set</span> <span class="op">=</span> <span class="va">dif_exp</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">contrasts</span><span class="op">[</span><span class="va">i</span>,<span class="op">]</span><span class="op">$</span><span class="va">result.ID</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">p_values</span> <span class="op">=</span> <span class="va">result_set</span><span class="op">[[</span><span class="fu">glue</span><span class="fu">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html" class="external-link">glue</a></span><span class="op">(</span><span class="st">"contrast_{contrasts[i,]$contrast.ID}_pvalue"</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span>    </span>
<span>    <span class="co"># multiple testing correction</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">p_values</span>,method <span class="op">=</span> <span class="st">'BH'</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">dif_exp_genes</span></span>
<span></span>
<span><span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span>result.ID <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">result.ID</span>,</span>
<span>                        contrast.id <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">contrast.ID</span>,</span>
<span>                        baseline.factorValue <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">baseline.factors</span>,</span>
<span>                        experimental.factorValue <span class="op">=</span> <span class="va">contrasts</span><span class="op">$</span><span class="va">experimental.factors</span>,</span>
<span>                        n_diff <span class="op">=</span> <span class="va">dif_exp_genes</span><span class="op">)</span></span>
<span></span>
<span><span class="va">contrasts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
result.ID
</th>
<th style="text-align:right;">
contrast.id
</th>
<th style="text-align:left;">
baseline.factorValue
</th>
<th style="text-align:left;">
experimental.factorValue
</th>
<th style="text-align:right;">
n_diff
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
550248
</td>
<td style="text-align:right;">
113004
</td>
<td style="text-align:left;">
disease,….
</td>
<td style="text-align:left;">
disease,….
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:right;">
550248
</td>
<td style="text-align:right;">
113005
</td>
<td style="text-align:left;">
disease,….
</td>
<td style="text-align:left;">
disease,….
</td>
<td style="text-align:right;">
1410
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrasts</span><span class="op">$</span><span class="va">baseline.factorValue</span></span></code></pre></div>
<pre><code>[[1]]
   category                         category.URI                  value
     &lt;char&gt;                               &lt;char&gt;                 &lt;char&gt;
1:  disease http://www.ebi.ac.uk/efo/EFO_0000408 reference subject role
                                    value.URI predicate predicate.URI object
                                       &lt;char&gt;    &lt;lgcl&gt;        &lt;lgcl&gt; &lt;lgcl&gt;
1: http://purl.obolibrary.org/obo/OBI_0000220        NA            NA     NA
   object.URI                summary     ID factor.ID factor.category
       &lt;lgcl&gt;                 &lt;char&gt;  &lt;int&gt;     &lt;int&gt;          &lt;char&gt;
1:         NA reference subject role 113006     19134         disease
                    factor.category.URI
                                 &lt;char&gt;
1: http://www.ebi.ac.uk/efo/EFO_0000408

[[2]]
   category                         category.URI                  value
     &lt;char&gt;                               &lt;char&gt;                 &lt;char&gt;
1:  disease http://www.ebi.ac.uk/efo/EFO_0000408 reference subject role
                                    value.URI predicate predicate.URI object
                                       &lt;char&gt;    &lt;lgcl&gt;        &lt;lgcl&gt; &lt;lgcl&gt;
1: http://purl.obolibrary.org/obo/OBI_0000220        NA            NA     NA
   object.URI                summary     ID factor.ID factor.category
       &lt;lgcl&gt;                 &lt;char&gt;  &lt;int&gt;     &lt;int&gt;          &lt;char&gt;
1:         NA reference subject role 113006     19134         disease
                    factor.category.URI
                                 &lt;char&gt;
1: http://www.ebi.ac.uk/efo/EFO_0000408</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrasts</span><span class="op">$</span><span class="va">experimental.factorValue</span></span></code></pre></div>
<pre><code>[[1]]
   category                         category.URI            value
     &lt;char&gt;                               &lt;char&gt;           &lt;char&gt;
1:  disease http://www.ebi.ac.uk/efo/EFO_0000408 bipolar disorder
                                      value.URI    predicate
                                         &lt;char&gt;       &lt;char&gt;
1: http://purl.obolibrary.org/obo/MONDO_0004985 has_modifier
                               predicate.URI      object object.URI
                                      &lt;char&gt;      &lt;char&gt;     &lt;char&gt;
1: http://purl.obolibrary.org/obo/RO_0002573 manic phase       &lt;NA&gt;
                                     summary     ID factor.ID factor.category
                                      &lt;char&gt;  &lt;int&gt;     &lt;int&gt;          &lt;char&gt;
1: bipolar disorder has_modifier manic phase 113004     19134         disease
                    factor.category.URI
                                 &lt;char&gt;
1: http://www.ebi.ac.uk/efo/EFO_0000408

[[2]]
   category                         category.URI            value
     &lt;char&gt;                               &lt;char&gt;           &lt;char&gt;
1:  disease http://www.ebi.ac.uk/efo/EFO_0000408 bipolar disorder
                                      value.URI    predicate
                                         &lt;char&gt;       &lt;char&gt;
1: http://purl.obolibrary.org/obo/MONDO_0004985 has_modifier
                               predicate.URI         object object.URI
                                      &lt;char&gt;         &lt;char&gt;     &lt;char&gt;
1: http://purl.obolibrary.org/obo/RO_0002573 euthymic phase       &lt;NA&gt;
                                        summary     ID factor.ID
                                         &lt;char&gt;  &lt;int&gt;     &lt;int&gt;
1: bipolar disorder has_modifier euthymic phase 113005     19134
   factor.category                  factor.category.URI
            &lt;char&gt;                               &lt;char&gt;
1:         disease http://www.ebi.ac.uk/efo/EFO_0000408</code></pre>
<p>Alternatively we, since we are only looking at one dataset and one
contrast manually, we can simply use <code>readableContrasts</code>.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_differential_expression_values.html">get_differential_expression_values</a></span><span class="op">(</span><span class="st">"GSE46416"</span>,readableContrasts <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">de</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
Probe
</th>
<th style="text-align:left;">
NCBIid
</th>
<th style="text-align:left;">
GeneSymbol
</th>
<th style="text-align:left;">
GeneName
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
corrected_pvalue
</th>
<th style="text-align:right;">
rank
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier manic phase_logFoldChange
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier manic phase_tstat
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier manic phase_pvalue
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier euthymic phase_logFoldChange
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier euthymic phase_tstat
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier euthymic phase_pvalue
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2982730
</td>
<td style="text-align:left;">
4018
</td>
<td style="text-align:left;">
LPA
</td>
<td style="text-align:left;">
lipoprotein(a)
</td>
<td style="text-align:right;">
0.9185
</td>
<td style="text-align:right;">
0.9521
</td>
<td style="text-align:right;">
0.9647
</td>
<td style="text-align:right;">
-0.0247
</td>
<td style="text-align:right;">
-0.3587
</td>
<td style="text-align:right;">
0.7221
</td>
<td style="text-align:right;">
-0.0249
</td>
<td style="text-align:right;">
-0.3622
</td>
<td style="text-align:right;">
0.7196
</td>
</tr>
<tr>
<td style="text-align:right;">
2787851
</td>
<td style="text-align:left;">
166752
</td>
<td style="text-align:left;">
FREM3
</td>
<td style="text-align:left;">
FRAS1 related extracellular…
</td>
<td style="text-align:right;">
0.5850
</td>
<td style="text-align:right;">
0.7348
</td>
<td style="text-align:right;">
0.7960
</td>
<td style="text-align:right;">
0.1895
</td>
<td style="text-align:right;">
1.0119
</td>
<td style="text-align:right;">
0.3191
</td>
<td style="text-align:right;">
0.1403
</td>
<td style="text-align:right;">
0.7495
</td>
<td style="text-align:right;">
0.4590
</td>
</tr>
<tr>
<td style="text-align:right;">
2477558
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
0.3965
</td>
<td style="text-align:right;">
0.5931
</td>
<td style="text-align:right;">
0.6685
</td>
<td style="text-align:right;">
0.0918
</td>
<td style="text-align:right;">
1.1386
</td>
<td style="text-align:right;">
0.2633
</td>
<td style="text-align:right;">
0.1016
</td>
<td style="text-align:right;">
1.2604
</td>
<td style="text-align:right;">
0.2166
</td>
</tr>
<tr>
<td style="text-align:right;">
2910917
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
0.6506
</td>
<td style="text-align:right;">
0.7815
</td>
<td style="text-align:right;">
0.8325
</td>
<td style="text-align:right;">
0.1564
</td>
<td style="text-align:right;">
0.6739
</td>
<td style="text-align:right;">
0.5052
</td>
<td style="text-align:right;">
0.2096
</td>
<td style="text-align:right;">
0.9032
</td>
<td style="text-align:right;">
0.3731
</td>
</tr>
<tr>
<td style="text-align:right;">
3983537
</td>
<td style="text-align:left;">
140886
</td>
<td style="text-align:left;">
PABPC5
</td>
<td style="text-align:left;">
poly(A) binding protein cyt…
</td>
<td style="text-align:right;">
0.1064
</td>
<td style="text-align:right;">
0.3227
</td>
<td style="text-align:right;">
0.3299
</td>
<td style="text-align:right;">
0.1845
</td>
<td style="text-align:right;">
2.0320
</td>
<td style="text-align:right;">
0.0505
</td>
<td style="text-align:right;">
0.1603
</td>
<td style="text-align:right;">
1.7656
</td>
<td style="text-align:right;">
0.0869
</td>
</tr>
<tr>
<td style="text-align:right;">
3538213
</td>
<td style="text-align:left;">
23002
</td>
<td style="text-align:left;">
DAAM1
</td>
<td style="text-align:left;">
dishevelled associated acti…
</td>
<td style="text-align:right;">
0.2271
</td>
<td style="text-align:right;">
0.4477
</td>
<td style="text-align:right;">
0.5073
</td>
<td style="text-align:right;">
0.3398
</td>
<td style="text-align:right;">
1.7279
</td>
<td style="text-align:right;">
0.0936
</td>
<td style="text-align:right;">
0.1200
</td>
<td style="text-align:right;">
0.6104
</td>
<td style="text-align:right;">
0.5459
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Classify probes for plotting</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">&lt;-</span> <span class="st">"No"</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span> <span class="op">&gt;</span> <span class="fl">1.0</span> <span class="op">&amp;</span> </span>
<span>        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Up"</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">1.0</span> <span class="op">&amp;</span> </span>
<span>        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Down"</span></span>
<span></span>
<span><span class="co"># Upregulated probes</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Up"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span>, </span>
<span>        <span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
Probe
</th>
<th style="text-align:left;">
GeneSymbol
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier manic phase_pvalue
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier manic phase_logFoldChange
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2319550
</td>
<td style="text-align:left;">
RBP7
</td>
<td style="text-align:right;">
8.61e-05
</td>
<td style="text-align:right;">
1.0740
</td>
</tr>
<tr>
<td style="text-align:right;">
2548699
</td>
<td style="text-align:left;">
CYP1B1
</td>
<td style="text-align:right;">
1.00e-04
</td>
<td style="text-align:right;">
1.3225
</td>
</tr>
<tr>
<td style="text-align:right;">
3907190
</td>
<td style="text-align:left;">
SLPI
</td>
<td style="text-align:right;">
3.00e-04
</td>
<td style="text-align:right;">
1.0558
</td>
</tr>
<tr>
<td style="text-align:right;">
3629103
</td>
<td style="text-align:left;">
PCLAF
</td>
<td style="text-align:right;">
5.00e-04
</td>
<td style="text-align:right;">
1.2783
</td>
</tr>
<tr>
<td style="text-align:right;">
3545525
</td>
<td style="text-align:left;">
SLIRP
</td>
<td style="text-align:right;">
6.00e-04
</td>
<td style="text-align:right;">
1.3490
</td>
</tr>
<tr>
<td style="text-align:right;">
3146433
</td>
<td style="text-align:left;">
COX6C
</td>
<td style="text-align:right;">
9.00e-04
</td>
<td style="text-align:right;">
1.4670
</td>
</tr>
<tr>
<td style="text-align:right;">
2899102
</td>
<td style="text-align:left;">
H3C3
</td>
<td style="text-align:right;">
1.30e-03
</td>
<td style="text-align:right;">
1.0260
</td>
</tr>
<tr>
<td style="text-align:right;">
2538349
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
1.30e-03
</td>
<td style="text-align:right;">
1.0731
</td>
</tr>
<tr>
<td style="text-align:right;">
3635198
</td>
<td style="text-align:left;">
BCL2A1
</td>
<td style="text-align:right;">
1.80e-03
</td>
<td style="text-align:right;">
1.0798
</td>
</tr>
<tr>
<td style="text-align:right;">
2633191
</td>
<td style="text-align:left;">
GPR15
</td>
<td style="text-align:right;">
2.40e-03
</td>
<td style="text-align:right;">
1.2046
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Downregulated probes</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Down"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span>, </span>
<span>        <span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
Probe
</th>
<th style="text-align:left;">
GeneSymbol
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier manic phase_pvalue
</th>
<th style="text-align:right;">
contrast_bipolar disorder has_modifier manic phase_logFoldChange
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2775390
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
2.10e-06
</td>
<td style="text-align:right;">
-1.5558
</td>
</tr>
<tr>
<td style="text-align:right;">
3760268
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
1.15e-05
</td>
<td style="text-align:right;">
-1.8506
</td>
</tr>
<tr>
<td style="text-align:right;">
3124344
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
1.00e-04
</td>
<td style="text-align:right;">
-1.0370
</td>
</tr>
<tr>
<td style="text-align:right;">
3673179
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
2.00e-04
</td>
<td style="text-align:right;">
-1.0340
</td>
</tr>
<tr>
<td style="text-align:right;">
3245871
</td>
<td style="text-align:left;">
WDFY4
</td>
<td style="text-align:right;">
2.00e-04
</td>
<td style="text-align:right;">
-1.1569
</td>
</tr>
<tr>
<td style="text-align:right;">
3022689
</td>
<td style="text-align:left;">
SND1-IT1
</td>
<td style="text-align:right;">
2.00e-04
</td>
<td style="text-align:right;">
-1.2199
</td>
</tr>
<tr>
<td style="text-align:right;">
2679014
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
3.00e-04
</td>
<td style="text-align:right;">
-1.1752
</td>
</tr>
<tr>
<td style="text-align:right;">
4019758
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
4.00e-04
</td>
<td style="text-align:right;">
-1.4049
</td>
</tr>
<tr>
<td style="text-align:right;">
3336402
</td>
<td style="text-align:left;">
RBM14
</td>
<td style="text-align:right;">
4.00e-04
</td>
<td style="text-align:right;">
-1.0711
</td>
</tr>
<tr>
<td style="text-align:right;">
3930525
</td>
<td style="text-align:left;">
RUNX1-IT1
</td>
<td style="text-align:right;">
4.00e-04
</td>
<td style="text-align:right;">
-1.5169
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add gene symbols as labels to DE genes</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">delabel</span> <span class="op">&lt;-</span> <span class="st">""</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">delabel</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">de</span><span class="op">$</span><span class="va">GeneSymbol</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Volcano plot for bipolar patients vs controls</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">de</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        x <span class="op">=</span> <span class="va">`contrast_bipolar disorder has_modifier manic phase_logFoldChange`</span>,</span>
<span>        y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder has_modifier manic phase_pvalue`</span><span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="va">diffexpr</span>,</span>
<span>        label <span class="op">=</span> <span class="va">delabel</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1.0</span>, <span class="fl">1.0</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"log2(FoldChange)"</span>, y <span class="op">=</span> <span class="st">"-log10(p-value)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggrepel.slowkow.com/reference/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="gemma.R_files/figure-html/diffExpr-1.png" alt="Differentially-expressed genes in bipolar patients during manic phase versus controls." width="700"><p class="caption">
Differentially-expressed genes in bipolar patients during manic phase
versus controls.
</p>
</div>
</div>
<div class="section level2">
<h2 id="larger-queries">Larger queries<a class="anchor" aria-label="anchor" href="#larger-queries"></a>
</h2>
<p>To query large amounts of data, the API has a pagination system which
uses the <code>limit</code> and <code>offset</code> parameters. To avoid
overloading the server, calls are limited to a maximum of 100 entries,
so the offset allows you to get the next batch of entries in the next
call(s).</p>
<p>To simplify the process of accessing all available data, gemma.R
includes the <a href="https://pavlidislab.github.io/gemma.R/reference/get_all_pages.html" class="external-link"><code>get_all_pages</code></a>
function which can use the output from one page to make all the follow
up requests</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_platforms_by_ids.html">get_platforms_by_ids</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/get_all_pages.html">get_all_pages</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
platform.ID
</th>
<th style="text-align:left;">
platform.shortName
</th>
<th style="text-align:left;">
platform.name
</th>
<th style="text-align:left;">
platform.description
</th>
<th style="text-align:left;">
platform.troubled
</th>
<th style="text-align:right;">
platform.experimentCount
</th>
<th style="text-align:left;">
platform.type
</th>
<th style="text-align:left;">
taxon.name
</th>
<th style="text-align:left;">
taxon.scientific
</th>
<th style="text-align:right;">
taxon.ID
</th>
<th style="text-align:right;">
taxon.NCBI
</th>
<th style="text-align:left;">
taxon.database.name
</th>
<th style="text-align:right;">
taxon.database.ID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
GPL96
</td>
<td style="text-align:left;">
Affymetrix GeneChip Human G…
</td>
<td style="text-align:left;">
The U133 set includes 2 ar…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
387
</td>
<td style="text-align:left;">
ONECOLOR
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
GPL1355
</td>
<td style="text-align:left;">
Affymetrix GeneChip Rat Gen…
</td>
<td style="text-align:left;">
The GeneChip Rat Genome 23…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
294
</td>
<td style="text-align:left;">
ONECOLOR
</td>
<td style="text-align:left;">
rat
</td>
<td style="text-align:left;">
Rattus norvegicus
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
10116
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:right;">
86
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
GPL1261
</td>
<td style="text-align:left;">
Affymetrix GeneChip Mouse G…
</td>
<td style="text-align:left;">
All probe sets represented…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
1284
</td>
<td style="text-align:left;">
ONECOLOR
</td>
<td style="text-align:left;">
mouse
</td>
<td style="text-align:left;">
Mus musculus
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
10090
</td>
<td style="text-align:left;">
mm10
</td>
<td style="text-align:right;">
81
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
GPL570
</td>
<td style="text-align:left;">
Affymetrix GeneChip Human G…
</td>
<td style="text-align:left;">
Complete coverage of the H…
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
1495
</td>
<td style="text-align:left;">
ONECOLOR
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
GPL81
</td>
<td style="text-align:left;">
Affymetrix GeneChip Murine …
</td>
<td style="text-align:left;">
The MG-U74 set includes 3 …
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
195
</td>
<td style="text-align:left;">
ONECOLOR
</td>
<td style="text-align:left;">
mouse
</td>
<td style="text-align:left;">
Mus musculus
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
10090
</td>
<td style="text-align:left;">
mm10
</td>
<td style="text-align:right;">
81
</td>
</tr>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
GPL85
</td>
<td style="text-align:left;">
Affymetrix GeneChip Rat Gen…
</td>
<td style="text-align:left;">
The RG-U34 set includes 3 …
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
85
</td>
<td style="text-align:left;">
ONECOLOR
</td>
<td style="text-align:left;">
rat
</td>
<td style="text-align:left;">
Rattus norvegicus
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
10116
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:right;">
86
</td>
</tr>
</tbody>
</table>
<p>Alternative way to access all pages is to do so manually. To see how
many available results are there, you can look at the attributes of the
output objects where additional information from the API response is
appended.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">platform_count</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_platforms_by_ids.html">get_platforms_by_ids</a></span><span class="op">(</span>limit <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">totalElements</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">platform_count</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 637</code></pre>
<p>After which you can use <code>offset</code> to access all available
platforms.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="va">platform_count</span>,<span class="fl">100</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">offset</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/get_platforms_by_ids.html">get_platforms_by_ids</a></span><span class="op">(</span>limit <span class="op">=</span> <span class="fl">100</span>, offset <span class="op">=</span> <span class="va">offset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">platform.ID</span>, <span class="va">platform.shortName</span>, <span class="va">taxon.name</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>,<span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
platform.ID
</th>
<th style="text-align:left;">
platform.shortName
</th>
<th style="text-align:left;">
taxon.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
GPL96
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
GPL1355
</td>
<td style="text-align:left;">
rat
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
GPL1261
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
GPL570
</td>
<td style="text-align:left;">
human
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
GPL81
</td>
<td style="text-align:left;">
mouse
</td>
</tr>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
GPL85
</td>
<td style="text-align:left;">
rat
</td>
</tr>
</tbody>
</table>
<p>Many endpoints only support a single identifier:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_dataset_annotations.html">get_dataset_annotations</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE46416"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>Error: Please specify one valid identifier for dataset.</code></pre>
<p>In these cases, you will have to loop over all the identifiers you
wish to query and send separate requests.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE12649"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/get_dataset_annotations.html">get_dataset_annotations</a></span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>experiment.shortName <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>, <span class="va">class.name</span>, <span class="va">term.name</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>,<span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
experiment.shortName
</th>
<th style="text-align:left;">
class.name
</th>
<th style="text-align:left;">
term.name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GSE35974
</td>
<td style="text-align:left;">
biological sex
</td>
<td style="text-align:left;">
female
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE35974
</td>
<td style="text-align:left;">
disease
</td>
<td style="text-align:left;">
depressive disorder
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE35974
</td>
<td style="text-align:left;">
disease
</td>
<td style="text-align:left;">
bipolar disorder
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE35974
</td>
<td style="text-align:left;">
biological sex
</td>
<td style="text-align:left;">
male
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE35974
</td>
<td style="text-align:left;">
labelling
</td>
<td style="text-align:left;">
biotin
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE35974
</td>
<td style="text-align:left;">
disease
</td>
<td style="text-align:left;">
schizophrenia
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE35974
</td>
<td style="text-align:left;">
organism part
</td>
<td style="text-align:left;">
cerebellum
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE12649
</td>
<td style="text-align:left;">
disease
</td>
<td style="text-align:left;">
schizophrenia
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE12649
</td>
<td style="text-align:left;">
labelling
</td>
<td style="text-align:left;">
biotin
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE12649
</td>
<td style="text-align:left;">
disease
</td>
<td style="text-align:left;">
bipolar disorder
</td>
</tr>
<tr>
<td style="text-align:left;">
GSE12649
</td>
<td style="text-align:left;">
organism part
</td>
<td style="text-align:left;">
prefrontal cortex
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="output-options">Output options<a class="anchor" aria-label="anchor" href="#output-options"></a>
</h2>
<div class="section level3">
<h3 id="raw-data">Raw data<a class="anchor" aria-label="anchor" href="#raw-data"></a>
</h3>
<p>By default, Gemma API does some parsing on the raw API results to
make it easier to work with inside of R. In the process, it drops some
typically unused values. If you wish to fetch everything, use
<code>raw = TRUE</code>. Instead of a data table, you’ll usually be
served a list that represents the underlying JSON response.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_gene_locations.html">get_gene_locations</a></span><span class="op">(</span><span class="st">"DYRK1A"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
chromosome
</th>
<th style="text-align:left;">
strand
</th>
<th style="text-align:right;">
nucleotide
</th>
<th style="text-align:right;">
length
</th>
<th style="text-align:left;">
taxon.name
</th>
<th style="text-align:left;">
taxon.scientific
</th>
<th style="text-align:right;">
taxon.ID
</th>
<th style="text-align:right;">
taxon.NCBI
</th>
<th style="text-align:left;">
taxon.database.name
</th>
<th style="text-align:right;">
taxon.database.ID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:right;">
33890705
</td>
<td style="text-align:right;">
118714
</td>
<td style="text-align:left;">
rat
</td>
<td style="text-align:left;">
Rattus norvegicus
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
10116
</td>
<td style="text-align:left;">
rn6
</td>
<td style="text-align:right;">
86
</td>
</tr>
<tr>
<td style="text-align:left;">
21
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:right;">
37365572
</td>
<td style="text-align:right;">
160785
</td>
<td style="text-align:left;">
human
</td>
<td style="text-align:left;">
Homo sapiens
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
9606
</td>
<td style="text-align:left;">
hg38
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
<ul><li></ul>
</td>
<td style="text-align:right;">
94370769
</td>
<td style="text-align:right;">
125608
</td>
<td style="text-align:left;">
mouse
</td>
<td style="text-align:left;">
Mus musculus
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
10090
</td>
<td style="text-align:left;">
mm10
</td>
<td style="text-align:right;">
81
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_gene_locations.html">get_gene_locations</a></span><span class="op">(</span><span class="st">"DYRK1A"</span>, raw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/listviewer/man/jsonedit.html" class="external-link">jsonedit</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="jsonedit html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"data":[{"id":120910616,"nucleotide":33890705,"nucleotideLength":118714,"strand":"+","bin":105,"chromosome":"11","taxon":{"id":3,"scientificName":"Rattus norvegicus","commonName":"rat","ncbiId":10116,"externalDatabase":{"id":86,"name":"rn6","description":"RGSC Rnor_6.0","uri":"https://genome.ucsc.edu/cgi-bin/hgTracks?db=rn6","releaseVersion":[],"releaseUrl":[],"lastUpdated":[],"externalDatabases":[{"id":116,"name":"rn6 RNA-Seq annotations","description":"Annotations provided by NCBI Genome and used by the RNA-Seq pipeline for rat data.","uri":"https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001895.5/","releaseVersion":"106","releaseUrl":"https://ftp.ncbi.nlm.nih.gov/genomes/all/annotation_releases/10116/106/","lastUpdated":"2022-10-24T07:00:00.000+00:00","externalDatabases":[]},{"id":100,"name":"rn4 annotations","description":[],"uri":"https://hgdownload.cse.ucsc.edu/goldenpath/rn4/database/","releaseVersion":[],"releaseUrl":[],"lastUpdated":[],"externalDatabases":[]}]}}},{"id":120419022,"nucleotide":37365572,"nucleotideLength":160785,"strand":"+","bin":108,"chromosome":"21","taxon":{"id":1,"scientificName":"Homo sapiens","commonName":"human","ncbiId":9606,"externalDatabase":{"id":87,"name":"hg38","description":"Genome Reference Consortium Human GRCh38.p13 (GCA_000001405.28)","uri":"https://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38","releaseVersion":"GRCh38.p13","releaseUrl":"https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_000001405.28/","lastUpdated":"2022-06-30T07:00:00.000+00:00","externalDatabases":[{"id":94,"name":"hg38 annotations","description":[],"uri":"https://hgdownload.cse.ucsc.edu/goldenpath/hg38/database/","releaseVersion":"GRCh38.p13","releaseUrl":"https://www.ncbi.nlm.nih.gov/datasets/genome/GCA_000001405.28/","lastUpdated":"2022-06-30T07:00:00.000+00:00","externalDatabases":[]},{"id":124,"name":"hg38 RNA-Seq annotations","description":"Annotations provided by NCBI Genome and used by the RNA-Seq pipeline for human data.","uri":"https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001405.40/","releaseVersion":"110","releaseUrl":"https://ftp.ncbi.nlm.nih.gov/genomes/all/annotation_releases/9606/110/","lastUpdated":"2023-01-17T20:27:55.059+00:00","externalDatabases":[]}]}}},{"id":120734638,"nucleotide":94370769,"nucleotideLength":125608,"strand":"+","bin":20,"chromosome":"16","taxon":{"id":2,"scientificName":"Mus musculus","commonName":"mouse","ncbiId":10090,"externalDatabase":{"id":81,"name":"mm10","description":"Genome Reference Consortium GRCm38, which includes approximately 2.6 Gb of sequence, is considered to be \"essentially complete\". The assembly includes chromosomes 1-19, X, Y, M (mitochondrial DNA) and chr*_random (unlocalized) and chrUn_* (unplaced clone contigs).","uri":"https://genome.ucsc.edu/cgi-bin/hgTracks?db=mm10","releaseVersion":"GCA_000001635.2","releaseUrl":"https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001635.2/","lastUpdated":"2022-06-30T07:00:00.000+00:00","externalDatabases":[{"id":97,"name":"mm10 annotations","description":[],"uri":"https://hgdownload.cse.ucsc.edu/goldenpath/mm10/database/","releaseVersion":"GCA_000001635.2","releaseUrl":"https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001635.2/","lastUpdated":"2022-06-30T07:00:00.000+00:00","externalDatabases":[]},{"id":115,"name":"mm10 RNA-Seq annotations","description":"Annotations provided by NCBI Genome and used by the RNA-Seq pipeline for mouse data.","uri":"https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001635.26/","releaseVersion":"108","releaseUrl":"https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Mus_musculus/108/","lastUpdated":"2019-10-24T07:00:00.219+00:00","externalDatabases":[]}]}}}],"options":{"mode":"tree","modes":["text","tree","table"]}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level3">
<h3 id="file-outputs">File outputs<a class="anchor" aria-label="anchor" href="#file-outputs"></a>
</h3>
<p>Sometimes, you may wish to save results to a file for future
inspection. You can do this simply by providing a filename to the
<code>file</code> parameter. The extension for this file will be one of
three options:</p>
<ol style="list-style-type: decimal">
<li>
<code>.json</code>, if you requested results with
<code>raw=TRUE</code>
</li>
<li>
<code>.csv</code> if the results have no nested data tables</li>
<li>
<code>.rds</code> otherwise</li>
</ol>
<p>You can also specify whether or not the new fetched results are
allowed to overwrite an existing file by specifying the
<code>overwrite = TRUE</code> parameter.</p>
</div>
<div class="section level3">
<h3 id="memoise-data">Memoise data<a class="anchor" aria-label="anchor" href="#memoise-data"></a>
</h3>
<p>To speed up results, you can remember past results so future queries
can proceed virtually instantly. This is enabled through the <a href="https://github.com/r-lib/memoise" class="external-link"><code>memoise</code></a>
package. To enable memoisation, simply set <code>memoised = TRUE</code>
in the function call whenever you want to refer to the cache, both to
save data for future calls and use the saved data for repeated
calls.</p>
<p>By default this will create a cache in your local filesystem.</p>
<p>You can see the path to your cache by using
<code>rappdirs::user_cache_dir(appname = "gemmaR")</code>. If you want
to use a different directory, use
<code>options(gemma.cache = 'path')</code>.</p>
<p>If you’re done with your fetching and want to ensure no space is
being used for cached results, or if you just want to ensure you’re
getting up-to-date data from Gemma, you can clear the cache using <a href="https://pavlidislab.github.io/gemma.R/reference/forget_gemma_memoised.html" class="external-link"><code>forget_gemma_memoised</code></a>.</p>
</div>
<div class="section level3">
<h3 id="changing-defaults">Changing defaults<a class="anchor" aria-label="anchor" href="#changing-defaults"></a>
</h3>
<p>We’ve seen how to change <code>raw = TRUE</code>,
<code>overwrite = TRUE</code> and <code>memoised = TRUE</code> in
individual function calls. It’s possible that you want to always use the
functions these ways without specifying the option every time. You can
do this by simply changing the default, which is visible in the function
definition. See below for examples.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.memoised <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always refer to cache</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always overwrite when saving files</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.raw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always receive results as-is from Gemma</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] listviewer_4.0.0            viridis_0.6.5              
 [3] viridisLite_0.4.2           pheatmap_1.0.12            
 [5] SummarizedExperiment_1.32.0 Biobase_2.62.0             
 [7] GenomicRanges_1.54.1        GenomeInfoDb_1.38.8        
 [9] IRanges_2.36.0              S4Vectors_0.40.2           
[11] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      
[13] matrixStats_1.3.0           ggrepel_0.9.5              
[15] ggplot2_3.5.0               dplyr_1.1.4                
[17] data.table_1.15.4           gemma.R_2.99.4             
[19] BiocStyle_2.30.0           

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1        farver_2.1.1            bitops_1.0-7           
 [4] fastmap_1.1.1           RCurl_1.98-1.14         digest_0.6.35          
 [7] timechange_0.3.0        lifecycle_1.0.4         magrittr_2.0.3         
[10] compiler_4.3.3          rlang_1.1.3             sass_0.4.9             
[13] tools_4.3.3             utf8_1.2.4              yaml_2.3.8             
[16] knitr_1.46              labeling_0.4.3          S4Arrays_1.2.1         
[19] htmlwidgets_1.6.4       bit_4.0.5               curl_5.2.1             
[22] DelayedArray_0.28.0     xml2_1.3.6              RColorBrewer_1.1-3     
[25] abind_1.4-5             withr_3.0.0             purrr_1.0.2            
[28] desc_1.4.3              grid_4.3.3              fansi_1.0.6            
[31] colorspace_2.1-0        scales_1.3.0            cli_3.6.2              
[34] rmarkdown_2.26          crayon_1.5.2            ragg_1.3.0             
[37] generics_0.1.3          rstudioapi_0.16.0       httr_1.4.7             
[40] cachem_1.0.8            stringr_1.5.1           zlibbioc_1.48.2        
[43] assertthat_0.2.1        BiocManager_1.30.22     XVector_0.42.0         
[46] vctrs_0.6.5             Matrix_1.6-5            jsonlite_1.8.8         
[49] bookdown_0.39           bit64_4.0.5             systemfonts_1.0.6      
[52] jquerylib_0.1.4         glue_1.7.0              pkgdown_2.0.9          
[55] lubridate_1.9.3         stringi_1.8.3           gtable_0.3.5           
[58] munsell_0.5.1           tibble_3.2.1            pillar_1.9.0           
[61] rappdirs_0.3.3          htmltools_0.5.8.1       GenomeInfoDbData_1.2.11
[64] R6_2.5.1                textshaping_0.3.7       kableExtra_1.4.0       
[67] evaluate_0.23           lattice_0.22-5          highr_0.10             
[70] memoise_2.0.1           bslib_0.7.0             Rcpp_1.0.12            
[73] svglite_2.1.3           gridExtra_2.3           SparseArray_1.2.4      
[76] xfun_0.43               fs_1.6.3                pkgconfig_2.0.3        </code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
