<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>gemma.R: an R Wrapper for the Gemma API • gemma.R</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="gemma.R: an R Wrapper for the Gemma API">
<meta property="og:description" content="gemma.R">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gemma.R.html">gemma.R: an R Wrapper for the Gemma API</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PavlidisLab/gemma.R/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="gemma.R_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>gemma.R: an R Wrapper for the Gemma API</h1>
                        <h4 class="author">Javier Castillo-Arnemann</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/master/vignettes/gemma.R.Rmd"><code>vignettes/gemma.R.Rmd</code></a></small>
      <div class="hidden name"><code>gemma.R.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://pavlidislab.github.io/gemma.R/">gemma.R</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biobase">Biobase</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com">data.table</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://memoise.r-lib.org">memoise</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joshuaulrich/microbenchmark/">microbenchmark</a></span><span class="op">)</span></code></pre></div>
<div id="about-gemma" class="section level2">
<h2 class="hasAnchor">
<a href="#about-gemma" class="anchor"></a>About Gemma</h2>
<p><a href="https://gemma.ubc.msl.ca/">Gemma</a> is a web site, database and a set of tools for the meta-analysis, re-use and sharing of genomics data, currently primarily targeted at the analysis of gene expression profiles. Gemma contains data from thousands of public studies, referencing thousands of published papers. Every dataset in Gemma has passed a rigorous curation process that re-annotates the expression platform at the sequence level, which allows for more consistent cross-platform comparisons and meta-analyses.</p>
<p>For detailed information on the curation process, check this <a href="https://pavlidislab.github.io/Gemma/curation.html">page</a> or the latest <a href="https://academic.oup.com/database/article/doi/10.1093/database/baab006/6143045">publication</a>.</p>
</div>
<div id="using-gemma-r" class="section level2">
<h2 class="hasAnchor">
<a href="#using-gemma-r" class="anchor"></a>Using gemma.R</h2>
<p>The main goal of this wrapper is to give easy access to Gemma’s curated datasets for downstream analyses or meta-analyses combining multiple datasets. In this example, we want to find datasets that are associated with bipolar disorder, and we are only interested in human data. In addition, we’ll subset our results to datasets that have been batch corrected. To do this, we use the <a href="https://pavlidislab.github.io/gemma.R/reference/searchDatasets.html"><code>searchDatasets()</code></a> endpoint:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/searchDatasets.html">searchDatasets</a></span><span class="op">(</span><span class="st">'bipolar'</span>, limit <span class="op">=</span> <span class="fl">100</span>, taxon <span class="op">=</span> <span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">geeq.batchCorrected</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">ee.ShortName</span>, <span class="va">ee.Name</span>, <span class="va">ee.Samples</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt;    ee.ShortName                                                         ee.Name
#&gt; 1:     GSE35974                 Expression data from the human cerebellum brain
#&gt; 2:     GSE46416 State- and trait-specific gene expression in euthymia and mania
#&gt;    ee.Samples
#&gt; 1:        144
#&gt; 2:         32</code></pre>
<p>We are left with two datasets, and we’ll pick <a href="https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?id=5939">GSE35974</a> since it has the larger number of samples. Now that we have the ID for our experiment, we can fetch the data associated with it.</p>
<p>To get the basic information of a dataset, we use the <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasets.html"><code>getDatasets()</code></a> endpoint.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDatasets.html">getDatasets</a></span><span class="op">(</span><span class="st">'GSE35974'</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt;    ee.ShortName                                         ee.Name ee.ID ee.Public
#&gt; 1:     GSE35974 Expression data from the human cerebellum brain  5939      TRUE
#&gt;    ee.Troubled ee.Accession ee.Database                           ee.URL
#&gt; 1:       FALSE     GSE35974         GEO http://www.ncbi.nlm.nih.gov/geo/
#&gt;    ee.Samples      ee.LastUpdated ee.Batch.P ee.Batch.PC geeq.batchCorrected
#&gt; 1:        144 2020-03-28 11:23:35         NA          NA                TRUE
#&gt;    geeq.qScore geeq.sScore taxon.Name taxon.ID technology.Type
#&gt; 1:   0.9975938           1      human        1        ONECOLOR</code></pre>
<p>To access the actual expression data, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/getBioc.html"><code>getBioc()</code></a>. It is a high-level wrapper that combines various endpoints to return an annotated <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html"><code>SummarizedExperiment</code></a> or <a href="https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf"><code>ExpressionSet</code></a> of the queried dataset that is compatible with other Bioconductor packages for downstream analyses. They include the expression matrix along with the experimental design, and ensure the sample names match between both when transforming/subsetting data.</p>
<p>Alternatively, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetData.html"><code>getDatasetData()</code></a> to get the expression matrix directly, and <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetDesign.html"><code>getDatasetDesign()</code></a> to get the matrix with the experimental design.</p>
<p>Since this dataset also contains patients with schizophrenia and depression, and we’re only interested in bipolar patients, we can remove those samples. We’re also keeping the <code>reference_subject_role</code>s, which refers to the control samples in Gemma datasets.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create ExpressionSet</span>
<span class="va">eset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getBioc.html">getBioc</a></span><span class="op">(</span><span class="st">'ExpressionSet'</span>, <span class="st">'GSE35974'</span><span class="op">)</span>
<span class="va">eset</span>

<span class="co"># Check the levels of the DiseaseState cofactor</span>
<span class="va">eset</span><span class="op">$</span><span class="va">DiseaseState</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Subset patients with bipolar disorder and controls.</span>
<span class="va">bipolar</span> <span class="op">&lt;-</span> <span class="va">eset</span><span class="op">[</span> , <span class="va">eset</span><span class="op">$</span><span class="va">DiseaseState</span> <span class="op">==</span> <span class="st">'Bipolar_Disorder'</span> <span class="op">|</span> <span class="va">eset</span><span class="op">$</span><span class="va">DiseaseState</span> <span class="op">==</span> <span class="st">'reference_subject_role'</span><span class="op">]</span>
<span class="va">bipolar</span></code></pre></div>
<pre><code>#&gt; ExpressionSet (storageMode: lockedEnvironment)
#&gt; assayData: 21939 features, 144 samples 
#&gt;   element names: exprs 
#&gt; protocolData: none
#&gt; phenoData
#&gt;   sampleNames: DE59_111109_rep1 DE60_100909 ... AC101_100909 (144
#&gt;     total)
#&gt;   varLabels: DiseaseState batch
#&gt;   varMetadata: labelDescription
#&gt; featureData: none
#&gt; experimentData: use 'experimentData(object)'
#&gt; Annotation: GPL6244</code></pre>
<pre><code>#&gt; [1] "reference_subject_role" "mental_depression"      "Bipolar_Disorder"      
#&gt; [4] "schizophrenia"</code></pre>
<pre><code>#&gt; ExpressionSet (storageMode: lockedEnvironment)
#&gt; assayData: 21939 features, 87 samples 
#&gt;   element names: exprs 
#&gt; protocolData: none
#&gt; phenoData
#&gt;   sampleNames: DE59_111109_rep1 DE7_111109 ... AC1_100909 (87 total)
#&gt;   varLabels: DiseaseState batch
#&gt;   varMetadata: labelDescription
#&gt; featureData: none
#&gt; experimentData: use 'experimentData(object)'
#&gt; Annotation: GPL6244</code></pre>
<p>Let’s check the expression for every sample to make sure they look OK:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get Expression matrix</span>
<span class="va">bipExpr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span><span class="op">(</span><span class="va">bipolar</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">bipExpr</span>, pch<span class="op">=</span><span class="st">'.'</span>, xaxt<span class="op">=</span><span class="st">'n'</span>, xlab<span class="op">=</span><span class="st">'Sample'</span>, ylab<span class="op">=</span><span class="st">'Expression'</span><span class="op">)</span></code></pre></div>
<p><img src="gemma.R_files/figure-html/base-boxplot-1.png" width="700"></p>
<p>We can quickly see three samples among bipolar patients and controls are empty. This is because during the curation process Gemma flags potential outlier samples automatically, which are then manually reviewed. Samples confirmed to be outliers are represented as missing data to ensure transparency.</p>
<p>You can also use <a href="https://pavlidislab.github.io/gemma.R/reference/getTidyDataset.html"><code>getTidyDataset()</code></a> to fetch data in a long, ‘tidy’ format that combines the expression matrix with the metadata. This is less memory-efficient than the two alternatives above, but allows quick and easy visualization and exploration of the data with <a href="https://ggplot2.tidyverse.org/"><code>ggplot2</code></a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tidyData</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTidyDataset.html">getTidyDataset</a></span><span class="op">(</span><span class="st">'GSE35974'</span><span class="op">)</span></code></pre></div>
<div id="differential-expression-example-todo" class="section level3">
<h3 class="hasAnchor">
<a href="#differential-expression-example-todo" class="anchor"></a>Differential Expression Example (TODO)</h3>
<p>Gemma also contains differential expression results for most of its datasets, and some datasets contain more than one analysis to account for different factors in the experimental design. These tables are stored as resultSets, and you can check the available resultSets for a dataset with <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetsResultsets.html"><code>getDatasetResultsets()</code></a>, then using <a href="https://pavlidislab.github.io/gemma.R/reference/getResultsets.html"><code>getResultsets()</code></a> to fetch the desired table:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDatasetResultSets.html">getDatasetResultSets</a></span><span class="op">(</span><span class="st">"GSE2872"</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt;    resultSet.id analysis.id factor.category              factor.levels
#&gt; 1:       500185      189752       treatment low potassium (LK), GW5074
#&gt; 2:       500186      189752       timepoint                   3 h, 6 h
#&gt; 3:       500187      189752       treatment low potassium (LK), GW5074
#&gt; 4:       500187      189752       timepoint                   3 h, 6 h</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getResultSets.html">getResultSets</a></span><span class="op">(</span><span class="fl">500185</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; Warning in require_bit64_if_needed(ans): Some columns are type 'integer64'
#&gt; but package bit64 is not installed. Those columns will print as strange
#&gt; looking floating point data. There is no need to reload the data. Simply
#&gt; install.packages('bit64') to obtain the integer64 print method and print the
#&gt; data again.</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rs</span></code></pre></div>
<pre><code>#&gt; Warning in require_bit64_if_needed(x): Some columns are type 'integer64'
#&gt; but package bit64 is not installed. Those columns will print as strange
#&gt; looking floating point data. There is no need to reload the data. Simply
#&gt; install.packages('bit64') to obtain the integer64 print method and print the
#&gt; data again.</code></pre>
<pre><code>#&gt;                   id probe_id   probe_name gene_id gene_name gene_ncbi_id
#&gt;     1: 1.231266e-314    47068   1373766_at                               
#&gt;     2: 1.231266e-314    35199   1385651_at                               
#&gt;     3: 1.231266e-314    25756   1395741_at                               
#&gt;     4: 1.231266e-314    35857 1384991_a_at 1040274    Hdac10       362981
#&gt;     5: 1.231266e-314    32978   1388337_at  988737       Pnp       290029
#&gt;    ---                                                                   
#&gt; 31078: 1.231261e-314    52752   1368082_at  940614    Slc4a2        24780
#&gt; 31079: 1.231261e-314    26677   1394820_at                               
#&gt; 31080: 1.231261e-314    29649   1391666_at                               
#&gt; 31081: 1.231261e-314    26678   1394819_at 1007143     Sp110       301570
#&gt; 31082: 1.231261e-314    46669   1374165_at                               
#&gt;        gene_official_symbol  pvalue corrected_pvalue    rank
#&gt;     1:                      0.89790                1 0.87660
#&gt;     2:                      0.02789                1 0.01515
#&gt;     3:                      0.28530                1 0.21280
#&gt;     4:               Hdac10 0.29140                1 0.21860
#&gt;     5:                  Pnp 0.72850                1 0.67500
#&gt;    ---                                                      
#&gt; 31078:               Slc4a2 0.31470                1 0.23800
#&gt; 31079:                      0.65950                1 0.59540
#&gt; 31080:                      0.46580                1 0.38400
#&gt; 31081:                Sp110 0.30950                1 0.23330
#&gt; 31082:                      0.86310                1 0.83320
#&gt;        contrast_134441_log2fc contrast_134441_tstat contrast_134441_pvalue
#&gt;     1:              -0.066240              -0.19810                 0.8479
#&gt;     2:               0.014290               0.30850                 0.7656
#&gt;     3:               0.160400               0.86730                 0.4110
#&gt;     4:               0.094980               0.89850                 0.3952
#&gt;     5:              -0.202600              -0.68020                 0.5156
#&gt;    ---                                                                    
#&gt; 31078:               0.007386               0.04921                 0.9620
#&gt; 31079:               0.003892               0.02543                 0.9803
#&gt; 31080:              -0.010570              -0.10600                 0.9182
#&gt; 31081:               0.131500               1.09300                 0.3062
#&gt; 31082:               0.017110               0.07563                 0.9416</code></pre>
<p>From here on, it’s easy to visualize/explore the data and find the most differentially-expressed genes:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">corrected_pvalue</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; Warning in require_bit64_if_needed(x): Some columns are type 'integer64'
#&gt; but package bit64 is not installed. Those columns will print as strange
#&gt; looking floating point data. There is no need to reload the data. Simply
#&gt; install.packages('bit64') to obtain the integer64 print method and print the
#&gt; data again.</code></pre>
<pre><code>#&gt;                   id probe_id   probe_name gene_id gene_name gene_ncbi_id
#&gt;     1: 1.231266e-314    47068   1373766_at                               
#&gt;     2: 1.231266e-314    35199   1385651_at                               
#&gt;     3: 1.231266e-314    25756   1395741_at                               
#&gt;     4: 1.231266e-314    35857 1384991_a_at 1040274    Hdac10       362981
#&gt;     5: 1.231266e-314    32978   1388337_at  988737       Pnp       290029
#&gt;    ---                                                                   
#&gt; 31078: 1.231262e-314    46430   1374404_at  938767       Jun        24516
#&gt; 31079: 1.231263e-314    47135   1373699_at                               
#&gt; 31080: 1.231262e-314    31787 1389528_s_at  938767       Jun        24516
#&gt; 31081: 1.231260e-314    50519 1370315_a_at  962863     Stmn4        79423
#&gt; 31082: 1.231271e-314    42190   1378645_at                               
#&gt;        gene_official_symbol    pvalue corrected_pvalue      rank
#&gt;     1:                      8.979e-01          1.00000 8.766e-01
#&gt;     2:                      2.789e-02          1.00000 1.515e-02
#&gt;     3:                      2.853e-01          1.00000 2.128e-01
#&gt;     4:               Hdac10 2.914e-01          1.00000 2.186e-01
#&gt;     5:                  Pnp 7.285e-01          1.00000 6.750e-01
#&gt;    ---                                                          
#&gt; 31078:                  Jun 1.770e-04          0.93000 1.287e-04
#&gt; 31079:                      2.586e-04          0.93000 2.574e-04
#&gt; 31080:                  Jun 6.551e-05          0.93000 6.435e-05
#&gt; 31081:                Stmn4 2.275e-04          0.93000 1.930e-04
#&gt; 31082:                      3.328e-07          0.01034 3.217e-05
#&gt;        contrast_134441_log2fc contrast_134441_tstat contrast_134441_pvalue
#&gt;     1:              -0.066240               -0.1981               0.847900
#&gt;     2:               0.014290                0.3085               0.765600
#&gt;     3:               0.160400                0.8673               0.411000
#&gt;     4:               0.094980                0.8985               0.395200
#&gt;     5:              -0.202600               -0.6802               0.515600
#&gt;    ---                                                                    
#&gt; 31078:               0.997100                3.7980               0.005249
#&gt; 31079:              -0.455400               -4.0770               0.003550
#&gt; 31080:               1.251000                4.5160               0.001961
#&gt; 31081:               0.563200                2.5640               0.033420
#&gt; 31082:              -0.009721               -0.1428               0.890000</code></pre>
</div>
<div id="longer-queries" class="section level3">
<h3 class="hasAnchor">
<a href="#longer-queries" class="anchor"></a>Longer queries</h3>
<p>Many endpoints also allow you to query multiple objects at the same time. For example, getting information on 3 datasets at the same time.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDatasets.html">getDatasets</a></span><span class="op">(</span>datasets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'GSE35974'</span>, <span class="st">'GSE12649'</span>, <span class="st">'GSE99349'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; Rows: 3
#&gt; Columns: 18
#&gt; $ ee.ShortName        &lt;chr&gt; "GSE12649", "GSE35974", "GSE99349"
#&gt; $ ee.Name             &lt;chr&gt; "Gene expression from human prefrontal cortex (BA4…
#&gt; $ ee.ID               &lt;int&gt; 1205, 5939, 11690
#&gt; $ ee.Public           &lt;lgl&gt; TRUE, TRUE, TRUE
#&gt; $ ee.Troubled         &lt;lgl&gt; FALSE, FALSE, FALSE
#&gt; $ ee.Accession        &lt;chr&gt; "GSE12649", "GSE35974", "GSE99349"
#&gt; $ ee.Database         &lt;chr&gt; "GEO", "GEO", "GEO"
#&gt; $ ee.URL              &lt;chr&gt; "http://www.ncbi.nlm.nih.gov/geo/", "http://www.nc…
#&gt; $ ee.Samples          &lt;int&gt; 102, 144, 36
#&gt; $ ee.LastUpdated      &lt;dttm&gt; 2021-09-18 07:35:38, 2020-03-28 11:23:35, 2020-03-…
#&gt; $ ee.Batch.P          &lt;dbl&gt; NA, NA, NA
#&gt; $ ee.Batch.PC         &lt;dbl&gt; NA, NA, NA
#&gt; $ geeq.batchCorrected &lt;lgl&gt; FALSE, TRUE, FALSE
#&gt; $ geeq.qScore         &lt;dbl&gt; 0.9972712, 0.9975938, 0.9879754
#&gt; $ geeq.sScore         &lt;dbl&gt; 0.875, 1.000, 1.000
#&gt; $ taxon.Name          &lt;chr&gt; "human", "human", "human"
#&gt; $ taxon.ID            &lt;int&gt; 1, 1, 1
#&gt; $ technology.Type     &lt;chr&gt; "ONECOLOR", "ONECOLOR", "GENELIST"</code></pre>
<p>You’ll be able to tell which endpoints accept multiple inputs from the function examples or simply because the parameter name is plural in the documentation. The rest of the endpoints only support a single identifier:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDatasetAnnotations.html">getDatasetAnnotations</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'GSE35974'</span>, <span class="st">'GSE12649'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; Error: Please specify one valid identifier for dataset.</code></pre>
<p>In these cases, you will have to loop over all the identifiers you wish to query and send separate requests.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'GSE35974'</span>, <span class="st">'GSE12649'</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/getDatasetAnnotations.html">getDatasetAnnotations</a></span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>shortName <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/rbindlist.html">rbindlist</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt;        class.Type     class.Name                                   class.URL
#&gt;  1:   FactorValue        disease        http://www.ebi.ac.uk/efo/EFO_0000408
#&gt;  2:   BioMaterial biological sex http://purl.obolibrary.org/obo/PATO_0000047
#&gt;  3:   BioMaterial biological sex http://purl.obolibrary.org/obo/PATO_0000047
#&gt;  4:   FactorValue        disease        http://www.ebi.ac.uk/efo/EFO_0000408
#&gt;  5: ExperimentTag  organism part        http://www.ebi.ac.uk/efo/EFO_0000635
#&gt;  6:   FactorValue        disease        http://www.ebi.ac.uk/efo/EFO_0000408
#&gt;  7:   BioMaterial  organism part        http://www.ebi.ac.uk/efo/EFO_0000635
#&gt;  8:   BioMaterial        disease        http://www.ebi.ac.uk/efo/EFO_0000408
#&gt;  9: ExperimentTag  organism part        http://www.ebi.ac.uk/efo/EFO_0000635
#&gt; 10:   BioMaterial        disease        http://www.ebi.ac.uk/efo/EFO_0000408
#&gt;     evidence.Code              term.Name
#&gt;  1:            IC       Bipolar Disorder
#&gt;  2:           IIA                   male
#&gt;  3:           IIA                 female
#&gt;  4:            IC          schizophrenia
#&gt;  5:            IC             cerebellum
#&gt;  6:            IC      mental depression
#&gt;  7:           IEA reference subject role
#&gt;  8:           IEA       Bipolar Disorder
#&gt;  9:            IC      prefrontal cortex
#&gt; 10:           IEA          schizophrenia
#&gt;                                          term.URL shortName
#&gt;  1:      http://purl.obolibrary.org/obo/DOID_3312  GSE35974
#&gt;  2:   http://purl.obolibrary.org/obo/PATO_0000384  GSE35974
#&gt;  3:   http://purl.obolibrary.org/obo/PATO_0000383  GSE35974
#&gt;  4:      http://purl.obolibrary.org/obo/DOID_5419  GSE35974
#&gt;  5: http://purl.obolibrary.org/obo/UBERON_0002037  GSE35974
#&gt;  6:      http://purl.obolibrary.org/obo/DOID_1596  GSE35974
#&gt;  7:    http://purl.obolibrary.org/obo/OBI_0000220  GSE12649
#&gt;  8:      http://purl.obolibrary.org/obo/DOID_3312  GSE12649
#&gt;  9: http://purl.obolibrary.org/obo/UBERON_0000451  GSE12649
#&gt; 10:      http://purl.obolibrary.org/obo/DOID_5419  GSE12649</code></pre>
</div>
<div id="memoise-data" class="section level3">
<h3 class="hasAnchor">
<a href="#memoise-data" class="anchor"></a>Memoise data</h3>
<p>To speed up results, you can remember past results so future queries can proceed virtually instantly. This is enabled through the <a href="https://github.com/r-lib/memoise"><code>memoise</code></a> package. To enable memoisation, simply set <code>memoised = T</code>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html">microbenchmark</a></span><span class="op">(</span><span class="fu"><a href="../reference/getDatasetData.html">getDatasetData</a></span><span class="op">(</span><span class="st">'GSE35974'</span>, memoised <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, times <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, unit <span class="op">=</span> <span class="st">'ms'</span><span class="op">)</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html">microbenchmark</a></span><span class="op">(</span><span class="fu"><a href="../reference/getDatasetData.html">getDatasetData</a></span><span class="op">(</span><span class="st">'GSE35974'</span>, memoised <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, times <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, unit <span class="op">=</span> <span class="st">'ms'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>If you’re done with your fetching and want to ensure no space is being used for cached results, or if you just want to ensure you’re getting up-to-date data from Gemma, you can clear the cache using <code>forgetGemmaMemoised</code>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/forgetGemmaMemoised.html">forgetGemmaMemoised</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; [1] TRUE</code></pre>
</div>
</div>
<div id="output-options" class="section level2">
<h2 class="hasAnchor">
<a href="#output-options" class="anchor"></a>Output options</h2>
<div id="raw-data" class="section level3">
<h3 class="hasAnchor">
<a href="#raw-data" class="anchor"></a>Raw data</h3>
<p>By default, Gemma API does some parsing on the raw API results to make it easier to work with inside of R. In the process, it drops some typically unused values. If you wish to fetch everything, use <code>raw = T</code>. Instead of a <code>data.table</code>, you’ll be served (typically) a list that represents the underlying JSON response.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDatasets.html">getDatasets</a></span><span class="op">(</span><span class="st">'GSE35974'</span>, raw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>#&gt; Rows: 1
#&gt; Columns: 36
#&gt; $ id                             &lt;int&gt; 5939
#&gt; $ lastUpdated                    &lt;dbl&gt; 1.585395e+12
#&gt; $ troubled                       &lt;lgl&gt; FALSE
#&gt; $ lastTroubledEvent              &lt;lgl&gt; NA
#&gt; $ needsAttention                 &lt;lgl&gt; FALSE
#&gt; $ lastNeedsAttentionEvent        &lt;lgl&gt; NA
#&gt; $ curationNote                   &lt;lgl&gt; NA
#&gt; $ lastNoteUpdateEvent            &lt;lgl&gt; NA
#&gt; $ bioAssayCount                  &lt;int&gt; 144
#&gt; $ description                    &lt;chr&gt; "Background: Schizophrenia (SCZ) and bi…
#&gt; $ name                           &lt;chr&gt; "Expression data from the human cerebel…
#&gt; $ accession                      &lt;chr&gt; "GSE35974"
#&gt; $ arrayDesignCount               &lt;int&gt; 1
#&gt; $ batchConfound                  &lt;lgl&gt; NA
#&gt; $ batchEffect                    &lt;chr&gt; "Data has been batch-corrected"
#&gt; $ bioMaterialCount               &lt;lgl&gt; NA
#&gt; $ currentUserHasWritePermission  &lt;lgl&gt; FALSE
#&gt; $ currentUserIsOwner             &lt;lgl&gt; FALSE
#&gt; $ experimentalDesign             &lt;int&gt; 5260
#&gt; $ externalDatabase               &lt;chr&gt; "GEO"
#&gt; $ externalUri                    &lt;chr&gt; "http://www.ncbi.nlm.nih.gov/geo/"
#&gt; $ geeq                           &lt;df[,24]&gt; &lt;data.frame[1 x 24]&gt;
#&gt; $ isPublic                       &lt;lgl&gt; TRUE
#&gt; $ isShared                       &lt;lgl&gt; FALSE
#&gt; $ metadata                       &lt;lgl&gt; NA
#&gt; $ processedExpressionVectorCount &lt;int&gt; 33252
#&gt; $ shortName                      &lt;chr&gt; "GSE35974"
#&gt; $ source                         &lt;lgl&gt; NA
#&gt; $ suitableForDEA                 &lt;lgl&gt; TRUE
#&gt; $ taxon                          &lt;chr&gt; "human"
#&gt; $ taxonId                        &lt;int&gt; 1
#&gt; $ technologyType                 &lt;chr&gt; "ONECOLOR"
#&gt; $ userOwned                      &lt;lgl&gt; FALSE
#&gt; $ userCanWrite                   &lt;lgl&gt; FALSE
#&gt; $ securableClass                 &lt;chr&gt; "ubic.gemma.model.expression.experiment…
#&gt; $ troubleDetails                 &lt;chr&gt; "No trouble details provided."</code></pre>
</div>
<div id="file-outputs" class="section level3">
<h3 class="hasAnchor">
<a href="#file-outputs" class="anchor"></a>File outputs</h3>
<p>Sometimes, you may wish to save results to a file for future inspection. You can do this simply by providing a filename to <code>file</code>. The extension for this file will be one of three options:</p>
<ol style="list-style-type: decimal">
<li>
<code>.json</code>, if you requested results with <code>raw = TRUE</code>
</li>
<li>
<code>.csv</code> if the results have no nested <code>data.table</code>s</li>
<li>
<code>.rds</code> otherwise</li>
</ol>
<p>You can also specify whether or not the new fetched results are allowed to overwrite an existing file by specifying the <code>overwrite</code> parameter.</p>
</div>
<div id="changing-defaults" class="section level3">
<h3 class="hasAnchor">
<a href="#changing-defaults" class="anchor"></a>Changing defaults</h3>
<p>We’ve seen how to change <code>memoised = T</code>, <code>overwrite = T</code> and <code>raw = T</code>… It’s possible that you want to always use the functions these ways without specifying the option every time. You can do this by simply changing the default, which is visible in the function definition. For example, if you want to memoise results by default, you can use:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>gemma.memoise <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jordan Sicherman, Ogan Mancarci, Javier Castillo-Arnemann.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
