<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A meta analysis on effects of Parkinson's Disease using Gemma.R • gemma.R</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="A meta analysis on effects of Parkinson's Disease using Gemma.R">
<meta property="og:description" content="gemma.R">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>A meta analysis on effects of Parkinson’s
Disease using Gemma.R</h1>
                        <h4 data-toc-skip class="author">B. Ogan
Mancarci</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia,
Vancouver, Canada<br><small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/vignettes/metanalysis.Rmd" class="external-link"><code>vignettes/metanalysis.Rmd</code></a></small>
      <div class="hidden name"><code>metanalysis.Rmd</code></div>

    </address>
</div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pavlidislab.github.io/gemma.R/" class="external-link">gemma.R</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">poolr</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span></code></pre></div>
<p>Gemma.R contains a large number of datasets representing a wide array
of conditions. These datasets are manually annotated by our curators to
facilitate discovery and categorization making it a useful tool for
meta-analysis.</p>
<p>In this example we search Gemma for datasets comparing healthy
controls and patients with Parkinson’s Disease. Our curators use
ontology terms to annotate datasets when possible. This allows us to use
the MONDO Disease Ontology term “MONDO:0005180” to search for datasets
and samples annotated with the term to make sure we are accessing
datasets relevant for our purposes with minimal effort.</p>
<div class="section level2">
<h2 id="querying-datasets-of-interest">Querying datasets of interest<a class="anchor" aria-label="anchor" href="#querying-datasets-of-interest"></a>
</h2>
<p>The <code>get_datasets</code> function can be used to find datasets
of interest, either using ontology terms or plain text. Since
Parkinson’s Disease has a unambiguous ontology term in the Disease
Ontology, we will be using it to avoid acquiring datasets that
tangentially mention the disease in their descriptions. Here we also
limit our results to only include human samples since we are not
interested in models from other species and samples that are explicitly
from the brain using the ontology term for brain. See documentation for
the function for a detailed explanation of available options</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># getting all resulting datasets using limit and offset arguments</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_datasets.html">get_datasets</a></span><span class="op">(</span>filter <span class="op">=</span> <span class="st">"allCharacteristics.valueUri = http://purl.obolibrary.org/obo/MONDO_0005180 and allCharacteristics.valueUri = http://purl.obolibrary.org/obo/UBERON_0000955"</span>,</span>
<span>                       taxa <span class="op">=</span><span class="st">'human'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/get_all_pages.html">get_all_pages</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="va">results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">experiment.shortName</span>,<span class="va">experiment.name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>   experiment.shortName
                 &lt;char&gt;
1:              GSE7621
2:              GSE7307
3:              GSE8397
4:             GSE20168
5:             GSE20333
6:             GSE20146
                                                                                        experiment.name
                                                                                                 &lt;char&gt;
1: Expression data of substantia nigra from postmortem human brain of Parkinson's disease patients (PD)
2:                                                         Human body index - transcriptional profiling
3:                                                       Expression profiling of the Parkinsonian Brain
4:                                 Transcriptional analysis of prefrontal area 9 in Parkinson's disease
5:                                           Gene expression profiling of parkinsonian substantia nigra
6:                                          Expression analysis of dissected GPi in Parkinson's disease</code></pre>
</div>
<div class="section level2">
<h2 id="filtering-the-datasets-for-suitability">Filtering the datasets for suitability<a class="anchor" aria-label="anchor" href="#filtering-the-datasets-for-suitability"></a>
</h2>
<p>While we know that all the resulting datasets were annotated for the
term for Parkinson’s Disease, we currently do not know how many of them
are comparisons of healthy controls and patients with Parkinson’s
Disease. We also do not know if the datasets have batch effects that may
affect our findings.</p>
<p>For this example, we decided we should not include data sets that
have a batch confound (though that is up to the user). Gemma internally
handles batch correction if batch information is available for the
dataset. We will be looking at <code>experiment.batchEffect</code>
column. As explained in the <code>get_datasets</code> documentation,
this column will be set to -1 for datasets where batch confound is
detected, 0 for datasets without available batch information and to 1 if
the data is clear of batch confounds.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="va">results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">experiment.batchEffect</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>We now want to ensure that the differential expressions we analyze
compare control and Parkinson’s Disease patients. This information is
available via <code>get_dataset_differential_expression_analyses</code>
which returns the experimental groups for differential expression
analyses performed for the dataset. The columns we are primarily
interested in are <code>baseline.factors</code> which typically records
the control group of the differential expression analysis and
<code>experimental.factors</code> which typically records the test
case</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">experiment_contrasts</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">experiment.shortName</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="va">out</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_dataset_differential_expression_analyses.html">get_dataset_differential_expression_analyses</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>        <span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>,<span class="va">.</span><span class="op">)</span></span></code></pre></div>
<p>The contrasts we are interested in should have a
<code>factor.category</code> of “disease” and it should have “<a href="http://purl.obolibrary.org/obo/MONDO_0005180" class="external-link uri">http://purl.obolibrary.org/obo/MONDO_0005180</a>” in it’s
experimental.factors column as the URI for a factor.</p>
<p>The factor we are interested in is Parkinson’s Disease in
<code>experimental.factorValue</code> or “<a href="http://purl.obolibrary.org/obo/MONDO_0005180" class="external-link uri">http://purl.obolibrary.org/obo/MONDO_0005180</a>” in
<code>experimental.factors</code>’s <code>value.URI</code>. We also need
to make sure that the baseline that the sample is compared against is a
control experiment, samples annotated with “reference subject role” or
“<a href="http://purl.obolibrary.org/obo/OBI_0000220" class="external-link uri">http://purl.obolibrary.org/obo/OBI_0000220</a>”</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parkin_contrasts</span> <span class="op">&lt;-</span> <span class="va">experiment_contrasts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">factor.category</span> <span class="op">==</span> <span class="st">'disease'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">experimental.factors</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="st">"http://purl.obolibrary.org/obo/MONDO_0005180"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">x</span><span class="op">$</span><span class="va">value.URI</span></span>
<span>        <span class="op">}</span><span class="op">)</span> <span class="op">&amp;</span></span>
<span>           <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">baseline.factors</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>               <span class="st">"http://purl.obolibrary.org/obo/OBI_0000220"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">x</span><span class="op">$</span><span class="va">value.URI</span></span>
<span>               <span class="op">}</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>In the next steps we will be downloading expression data in bulk. If
you are here to just try the code out, you can speed the process a bit
by limiting the number of experiments to deal with</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># arbitrarily select a few datasets</span></span>
<span><span class="co"># not executed in this example</span></span>
<span><span class="va">parkin_contrasts</span> <span class="op">&lt;-</span> <span class="va">parkin_contrasts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<!-- In the next steps we will be downloading differential expression data in bulk to -->
<!-- perform our meta-analysis. Since downloading and analyzing all these datasets -->
<!-- takes quite some time, in this example we will limit ourselves to a smaller selection. -->
<!--

```r
# We are using the first 5 contrasts to keep this analysis short
# sanity_sets = c('GSE8397','GSE7621','GSE20168','GSE20291','GSE20292')
# sanity_ids = results %>% filter(experiment.Accession %in% sanity_sets) %>%  .$experiment.ID

# parkin_contrasts <- parkin_contrasts %>% filter(experiment.ID %in% sanity_ids)
# parkin_contrasts <- parkin_contrasts[1:10,]
```
-->
<p>Now that we have our relevant contrasts, we can download them using
<code>get_differential_expression_values</code>. This function can be
used to download differential expression fold change and p values,
either using the experiment name/ids or more specifically using the
result.IDs</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">differentials</span> <span class="op">&lt;-</span> <span class="va">parkin_contrasts</span><span class="op">$</span><span class="va">result.ID</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="co"># take the first and only element of the output. the function returns a list </span></span>
<span>    <span class="co"># because single experiments may have multiple resultSets. Here we use the </span></span>
<span>    <span class="co"># resultSet argument to directly access the results we need</span></span>
<span>    <span class="fu"><a href="../reference/get_differential_expression_values.html">get_differential_expression_values</a></span><span class="op">(</span>resultSets <span class="op">=</span> <span class="va">x</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># some datasets might not have all the advertised differential expression results</span></span>
<span><span class="co"># calculated due to a variety of factors. here we remove the empty differentials</span></span>
<span><span class="va">missing_contrasts</span> <span class="op">&lt;-</span> <span class="va">differentials</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">nrow</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="op">{</span><span class="va">.</span><span class="op">==</span><span class="fl">0</span><span class="op">}</span></span>
<span><span class="va">differentials</span> <span class="op">&lt;-</span> <span class="va">differentials</span><span class="op">[</span><span class="op">!</span><span class="va">missing_contrasts</span><span class="op">]</span></span>
<span><span class="va">parkin_contrasts</span> <span class="op">&lt;-</span> <span class="va">parkin_contrasts</span><span class="op">[</span><span class="op">!</span><span class="va">missing_contrasts</span>,<span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="getting-the-p-values-for-the-condition-comparison">Getting the p-values for the condition comparison<a class="anchor" aria-label="anchor" href="#getting-the-p-values-for-the-condition-comparison"></a>
</h2>
<p><code>differentials</code> is now a list of data frames containing
the differential expression information. To run a simple meta-analysis,
we need the p values for the genes from the relevant contrasts.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">condition_diffs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">differentials</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="co"># iterate over the differentials</span></span>
<span>    <span class="va">diff</span> <span class="op">=</span> <span class="va">differentials</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="co"># get the contrast information about the differential</span></span>
<span>    <span class="va">contrast</span> <span class="op">=</span> <span class="va">parkin_contrasts</span><span class="op">[</span><span class="va">i</span>,<span class="op">]</span></span>
<span>    <span class="va">p_vals</span> <span class="op">=</span> <span class="va">diff</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'contrast_'</span>,<span class="va">contrast</span><span class="op">$</span><span class="va">contrast.ID</span>,<span class="st">"_pvalue"</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">log2fc</span> <span class="op">=</span> <span class="va">diff</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'contrast_'</span>,<span class="va">contrast</span><span class="op">$</span><span class="va">contrast.ID</span>,<span class="st">"_log2fc"</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">genes</span> <span class="op">=</span> <span class="va">diff</span><span class="op">$</span><span class="va">GeneSymbol</span></span>
<span>    </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">genes</span>,<span class="va">p_vals</span>,<span class="va">log2fc</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># we can use result.IDs and contrast.IDs to uniquely name this. </span></span>
<span><span class="co"># we add the experiment.id for readability</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">condition_diffs</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">parkin_contrasts</span><span class="op">$</span><span class="va">experiment.ID</span>,<span class="st">'.'</span>,</span>
<span>                                <span class="va">parkin_contrasts</span><span class="op">$</span><span class="va">result.ID</span>,<span class="st">'.'</span>,</span>
<span>                                <span class="va">parkin_contrasts</span><span class="op">$</span><span class="va">contrast.ID</span><span class="op">)</span></span>
<span></span>
<span><span class="va">condition_diffs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>      genes    p_vals  log2fc
1     ALMS1 9.500e-03  0.2934
2    PRSS36 3.924e-05 -0.5074
3    SLC5A3 4.354e-06  1.8432
4    AGPAT4 6.912e-01 -0.1074
5 NUDT16-DT 5.978e-01  0.0418
6       DR1 1.389e-01  0.2875</code></pre>
</div>
<div class="section level2">
<h2 id="combining-the-acquired-p-values">Combining the acquired p values<a class="anchor" aria-label="anchor" href="#combining-the-acquired-p-values"></a>
</h2>
<p>Now that we have acquired the values we need for a meta-analysis from
Gemma, we can proceed with any methodology we deem suitable for our
analysis.</p>
<p>In this example we will use a very simple approach, <a href="https://en.wikipedia.org/wiki/Fisher%27s_method" class="external-link">Fisher’s combined
p-value test</a>. This is implemented in the <code>fisher</code>
function from <code>poolr</code> package. Fisher’s method has the
advantage that it operates on the p-values, which are in the processed
differential expression results in Gemma. It has some disadvantages like
ignoring the direction of the expression change (Gemma’s p-values are
two-tailed) and is very sensitive to a single ‘outlier’ p-value, but for
this demonstration we’ll go with it.</p>
<p>The first step is to identify which genes are available in our
results:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="va">condition_diffs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">x</span><span class="op">$</span><span class="va">genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">unique</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">unlist</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">table</span></span>
<span></span>
<span><span class="co"># we will remove any gene that doesn't appear in all of the results</span></span>
<span><span class="co"># while this criteria is too strict, it does help this example to run </span></span>
<span><span class="co"># considerably faster</span></span>
<span><span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="va">all_genes</span><span class="op">[</span><span class="va">all_genes</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">all_genes</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># remove any probesets matching multiple genes. gemma separates these by using "|"</span></span>
<span><span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="va">all_genes</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"|"</span>,<span class="va">all_genes</span>,fixed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># remove the "". This comes from probesets not aligned to any genes</span></span>
<span><span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="va">all_genes</span><span class="op">[</span><span class="va">all_genes</span> <span class="op">!=</span> <span class="st">""</span><span class="op">]</span></span>
<span><span class="va">all_genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span></span></code></pre></div>
<pre><code>[1] "A2M"   "AAAS"  "AACS"  "AAGAB" "AAK1"  "AAMP" </code></pre>
<p>Now we can run the test on every gene, followed by a multiple testing
correction.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fisher_results</span> <span class="op">&lt;-</span> <span class="va">all_genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">p_vals</span> <span class="op">&lt;-</span> <span class="va">condition_diffs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="co"># we will resolve multiple probesets by taking the minimum p value for</span></span>
<span>        <span class="co"># this example</span></span>
<span>        <span class="va">out</span> <span class="op">=</span> <span class="va">y</span><span class="op">[</span><span class="va">y</span><span class="op">$</span><span class="va">genes</span> <span class="op">==</span> <span class="va">x</span>,<span class="op">]</span><span class="op">$</span><span class="va">p_vals</span></span>
<span>        <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span> <span class="op">||</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>            <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NA</span><span class="op">)</span></span>
<span>        <span class="op">}</span> <span class="kw">else</span><span class="op">{</span></span>
<span>            <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span><span class="op">)</span></span>
<span>        <span class="op">}</span></span>
<span>    <span class="op">}</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="va">fold_changes</span> <span class="op">&lt;-</span> <span class="va">condition_diffs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="va">pv</span> <span class="op">=</span> <span class="va">y</span><span class="op">[</span><span class="va">y</span><span class="op">$</span><span class="va">genes</span> <span class="op">==</span> <span class="va">x</span>,<span class="op">]</span><span class="op">$</span><span class="va">p_vals</span></span>
<span>        <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">pv</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span> <span class="op">||</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">pv</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>            <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NA</span><span class="op">)</span></span>
<span>        <span class="op">}</span> <span class="kw">else</span><span class="op">{</span></span>
<span>            <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">y</span><span class="op">[</span><span class="va">y</span><span class="op">$</span><span class="va">genes</span> <span class="op">==</span> <span class="va">x</span>,<span class="op">]</span><span class="op">$</span><span class="va">log2fc</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.min.html" class="external-link">which.min</a></span><span class="op">(</span><span class="va">pv</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span>        <span class="op">}</span></span>
<span>    <span class="op">}</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="va">median_fc</span> <span class="op">=</span> <span class="va">fold_changes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">median</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">median_fc</span><span class="op">)</span> <span class="op">=</span> <span class="st">'Median FC'</span></span>
<span>    </span>
<span>    <span class="va">combined</span> <span class="op">=</span> <span class="va">p_vals</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/poolr/man/fisher.html" class="external-link">fisher</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="op">{</span><span class="va">.</span><span class="op">$</span><span class="va">p</span><span class="op">}</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">combined</span><span class="op">)</span> <span class="op">=</span> <span class="st">'Combined'</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">p_vals</span>,<span class="va">combined</span>,<span class="va">median_fc</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>,<span class="va">.</span><span class="op">)</span></span>
<span><span class="va">fisher_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">fisher_results</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">fisher_results</span><span class="op">)</span> <span class="op">=</span> <span class="va">all_genes</span></span>
<span></span>
<span><span class="va">fisher_results</span><span class="op">[</span>,<span class="st">'Adjusted'</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">fisher_results</span><span class="op">[</span>,<span class="st">'Combined'</span><span class="op">]</span>,</span>
<span>                                        method <span class="op">=</span> <span class="st">'fdr'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fisher_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Adjusted</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Combined</span>,<span class="va">Adjusted</span>,<span class="va">`Median FC`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">head</span></span></code></pre></div>
<p>We end up with quite a few differentially expressed genes in the
meta-analysis (Fisher’s method is very sensitive)</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">fisher_results</span><span class="op">$</span><span class="va">Adjusted</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span> <span class="co"># FDR&lt;0.05</span></span></code></pre></div>
<pre><code>[1] 5316</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">fisher_results</span><span class="op">)</span> <span class="co"># number of all genes</span></span></code></pre></div>
<pre><code>[1] 6235</code></pre>
<p>Next we look at markers of dopaminergic cell types and how they rank
compared to other genes. Parkinson’s Disease is a neurodegenerative
disorder, leading to death of dopaminergic cells. We should expect them
to show up in our results.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># markers are taken from https://www.eneuro.org/content/4/6/ENEURO.0212-17.2017</span></span>
<span><span class="va">dopa_markers</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ADCYAP1"</span>, <span class="st">"ATP2B2"</span>, <span class="st">"CACNA2D2"</span>, </span>
<span><span class="st">"CADPS2"</span>, <span class="st">"CALB2"</span>, <span class="st">"CD200"</span>, <span class="st">"CDK5R2"</span>, <span class="st">"CELF4"</span>, <span class="st">"CHGA"</span>, <span class="st">"CHGB"</span>, </span>
<span><span class="st">"CHRNA6"</span>, <span class="st">"CLSTN2"</span>, <span class="st">"CNTNAP2"</span>, <span class="st">"CPLX1"</span>, <span class="st">"CYB561"</span>, <span class="st">"DLK1"</span>, <span class="st">"DPP6"</span>, </span>
<span><span class="st">"ELAVL2"</span>, <span class="st">"ENO2"</span>, <span class="st">"GABRG2"</span>, <span class="st">"GRB10"</span>, <span class="st">"GRIA3"</span>, <span class="st">"KCNAB2"</span>, <span class="st">"KLHL1"</span>, </span>
<span><span class="st">"LIN7B"</span>, <span class="st">"MAPK8IP2"</span>, <span class="st">"NAPB"</span>, <span class="st">"NR4A2"</span>, <span class="st">"NRIP3"</span>, <span class="st">"HMP19"</span>, <span class="st">"NTNG1"</span>, </span>
<span><span class="st">"PCBP3"</span>, <span class="st">"PCSK1"</span>, <span class="st">"PRKCG"</span>, <span class="st">"RESP18"</span>, <span class="st">"RET"</span>, <span class="st">"RGS8"</span>, <span class="st">"RNF157"</span>, </span>
<span><span class="st">"SCG2"</span>, <span class="st">"SCN1A"</span>, <span class="st">"SLC12A5"</span>, <span class="st">"SLC4A10"</span>, <span class="st">"SLC6A17"</span>, <span class="st">"SLC6A3"</span>, <span class="st">"SMS"</span>, </span>
<span><span class="st">"SNCG"</span>, <span class="st">"SPINT2"</span>, <span class="st">"SPOCK1"</span>, <span class="st">"SYP"</span>, <span class="st">"SYT4"</span>, <span class="st">"TACR3"</span>, <span class="st">"TENM1"</span>, </span>
<span><span class="st">"TH"</span>, <span class="st">"USP29"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fisher_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Combined</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">rownames</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="op">{</span><span class="va">.</span><span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">dopa_markers</span><span class="op">}</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">which</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span>breaks<span class="op">=</span><span class="fl">20</span>, main <span class="op">=</span> <span class="st">'Rank distribution of dopaminergic markers'</span><span class="op">)</span></span></code></pre></div>
<p><img src="metanalysis_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>In agreement with our hypothesis, we can see that the dopaminergic
markers tend to have high ranks in our results.</p>
</div>
<div class="section level2">
<h2 id="acquiring-the-expression-data-for-a-top-gene">Acquiring the expression data for a top gene<a class="anchor" aria-label="anchor" href="#acquiring-the-expression-data-for-a-top-gene"></a>
</h2>
<p>Now that we have our results, we can take a look at how the
expression of one of our top genes in these experiments. To do this we
will use the <code>get_dataset_expression_for_genes</code> function to
get the expression data. Once we get the expression data for our genes
of interest, we will use the <code>get_dataset_samples</code> function
to identify which samples belongs to which experimental group.</p>
<blockquote>
<p>Note: as of this writing (early 2023), the Gemma.R methods used in
this section are not yet released in Bioconductor; install via devtools
to try it out.</p>
</blockquote>
<p>For starters, lets look at our top pick and the its p values in the
individual datasets, on a log scale:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># the top gene from our results</span></span>
<span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="va">fisher_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Adjusted</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">.</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">rownames</span></span>
<span></span>
<span><span class="va">p_values</span> <span class="op">&lt;-</span> <span class="va">fisher_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Adjusted</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">.</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">Combined</span>,<span class="op">-</span><span class="va">`Median FC`</span>,<span class="op">-</span><span class="va">Adjusted</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">unlist</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#sum(p_values&lt;0.05)</span></span>
<span><span class="co">#length(p_values)</span></span>
<span></span>
<span><span class="co"># p values of the result in individual studies</span></span>
<span><span class="va">p_values</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>`log p value` <span class="op">=</span> <span class="va">.</span>,dataset <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>,check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">`log p value`</span>,x <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,color <span class="op">=</span> <span class="st">'firebrick'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, x <span class="op">=</span> <span class="fl">0</span>, label <span class="op">=</span> <span class="st">'p &lt; 0.05'</span>,vjust <span class="op">=</span><span class="op">-</span><span class="fl">1</span>,hjust <span class="op">=</span> <span class="op">-</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"P-values for top gene ("</span>, <span class="va">gene</span>, <span class="st">") in the data sets"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="metanalysis_files/figure-html/unnamed-chunk-14-1.png" width="700">
This shows that the gene is nominally significant in some but not all of
the data sets.</p>
<p>To examine this further, our next step is to acquire gene expression
data from the relevant datasets using <code>get_dataset_object</code>.
We will use both dataset ids and resultSet ids when using this function
since some of the returned analysis are only performed on a subset of
the data. Providing resultSet ids allows us to harmonize the
differential expression results with expression data by returning the
relevant subset.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># we need the NCBI id of the gene in question, lets get that from the original</span></span>
<span><span class="co"># results</span></span>
<span><span class="va">NCBIid</span> <span class="op">&lt;-</span> <span class="va">differentials</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">GeneSymbol</span> <span class="op">==</span> <span class="va">gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">.</span><span class="op">$</span><span class="va">NCBIid</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">unique</span></span>
<span><span class="co">#NCBIid</span></span>
<span></span>
<span></span>
<span><span class="va">expression_frame</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset_object.html">get_dataset_object</a></span><span class="op">(</span>datasets <span class="op">=</span> <span class="va">parkin_contrasts</span><span class="op">$</span><span class="va">experiment.ID</span>,</span>
<span>                                       resultSets <span class="op">=</span> <span class="va">parkin_contrasts</span><span class="op">$</span><span class="va">result.ID</span>,</span>
<span>                                       contrasts <span class="op">=</span> <span class="va">parkin_contrasts</span><span class="op">$</span><span class="va">contrast.ID</span>,</span>
<span>                                       genes <span class="op">=</span> <span class="va">NCBIid</span>,type <span class="op">=</span> <span class="st">'tidy'</span>,consolidate <span class="op">=</span> <span class="st">'pickvar'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get the contrast names for significance markers </span></span>
<span><span class="va">signif_contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">p_values</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">names</span></span></code></pre></div>
<p>Finally, we’ll make a plot showing the expression of the gene of
interest in all the data sets. This helps us see the extent to which
there is evidence of consistent differential expression.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expression_frame</span> <span class="op">&lt;-</span> <span class="va">expression_frame</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="co"># add a column to represent the contrast </span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">experiment.ID</span>,<span class="st">'.'</span>,</span>
<span>                                    <span class="va">result.ID</span>,<span class="st">'.'</span>,</span>
<span>                                    <span class="va">contrast.ID</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># simplify the labels</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>disease <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">==</span> <span class="st">'reference subject role'</span>,<span class="st">'Control'</span>,<span class="st">'PD'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># for adding human readable labels on the plot</span></span>
<span><span class="va">labeller</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">x</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>contrasts <span class="op">=</span> <span class="va">contrasts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="st">'.'</span>,fixed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                     <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_chr</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                     <span class="op">{</span><span class="va">expression_frame</span><span class="op">$</span><span class="va">experiment.shortName</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">.</span>,<span class="va">expression_frame</span><span class="op">$</span><span class="va">experiment.ID</span><span class="op">)</span><span class="op">]</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># pass it all to ggplot</span></span>
<span><span class="va">expression_frame</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">disease</span>,y <span class="op">=</span> <span class="va">expression</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">contrasts</span>,scales <span class="op">=</span> <span class="st">'free'</span>,labeller <span class="op">=</span> <span class="va">labeller</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Expression of"</span>, <span class="va">gene</span>, <span class="st">" per study"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>contrasts <span class="op">=</span> <span class="va">signif_contrasts</span>,</span>
<span>                             <span class="va">expression_frame</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                             <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">contrasts</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                             <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                             <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">signif_contrasts</span>,<span class="va">.</span><span class="op">$</span><span class="va">contrasts</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span>,</span>
<span>              x <span class="op">=</span> <span class="fl">1.5</span>, label <span class="op">=</span> <span class="st">'*'</span>,size<span class="op">=</span><span class="fl">5</span>,vjust<span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="metanalysis_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] stringr_1.5.1    ggplot2_3.5.0    poolr_1.1-1      dplyr_1.1.4     
[5] gemma.R_2.99.2   BiocStyle_2.30.0

loaded via a namespace (and not attached):
 [1] tidyr_1.3.1         sass_0.4.9          utf8_1.2.4         
 [4] generics_0.1.3      stringi_1.8.3       digest_0.6.35      
 [7] magrittr_2.0.3      timechange_0.3.0    evaluate_0.23      
[10] grid_4.3.3          bookdown_0.38       fastmap_1.1.1      
[13] jsonlite_1.8.8      BiocManager_1.30.22 httr_1.4.7         
[16] purrr_1.0.2         fansi_1.0.6         scales_1.3.0       
[19] textshaping_0.3.7   jquerylib_0.1.4     cli_3.6.2          
[22] rlang_1.1.3         bit64_4.0.5         munsell_0.5.1      
[25] withr_3.0.0         cachem_1.0.8        yaml_2.3.8         
[28] tools_4.3.3         memoise_2.0.1       colorspace_2.1-0   
[31] mathjaxr_1.6-0      curl_5.2.1          assertthat_0.2.1   
[34] vctrs_0.6.5         R6_2.5.1            lubridate_1.9.3    
[37] lifecycle_1.0.4     fs_1.6.3            htmlwidgets_1.6.4  
[40] bit_4.0.5           ragg_1.3.0          pkgconfig_2.0.3    
[43] desc_1.4.3          pkgdown_2.0.8       pillar_1.9.0       
[46] bslib_0.7.0         gtable_0.3.4        data.table_1.15.4  
[49] glue_1.7.0          systemfonts_1.0.6   highr_0.10         
[52] xfun_0.43           tibble_3.2.1        tidyselect_1.2.1   
[55] knitr_1.46          farver_2.1.1        htmltools_0.5.8.1  
[58] labeling_0.4.3      rmarkdown_2.26      compiler_4.3.3     </code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.8.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
