#' Gemma RESTful API
#'
#' This website documents the usage of the [Gemma REST API](https://gemma.msl.ubc.ca/rest/v2/). Here you can find example script usage of the API, as well as graphical interface for each endpoint, with description of its parameters and the endpoint URL.  The documentation of the underlying java code can be found [here](https://gemma.msl.ubc.ca/resources/apidocs/ubic/gemma/web/services/rest/package-summary.html). See the [links section](https://gemma.msl.ubc.ca/resources/restapidocs/#footer) in the footer of this page for other relevant links.  Use of this webpage and Gemma web services, including the REST API, is subject to [these terms and conditions](https://pavlidislab.github.io/Gemma/terms.html). Please read these in full before continuing to use this webpage or any other part of the Gemma system. 
#'
#' The version of the OpenAPI document: 2.4.1
#' Contact: pavlab-support@msl.ubc.ca
#' Generated by: https://openapi-generator.tech
#'
#' @docType class
#' @title Default operations
#' @description 
#' @format An \code{R6Class} generator object
#' @field api_client Handles the client-server communication.
#'
#' @section Methods:
#' \describe{
#' \strong{ get_api_info } \emph{ Retrieve an object with basic API information }
#'
#' \itemize{
#' \item \emph{ @returnType } \link{ResponseDataObjectApiInfoValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectApiInfoValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_annotations } \emph{ Retrieve the annotations analysis of a dataset }
#'
#' \itemize{
#' \item \emph{ @param } dataset \link{GetResultSetsDatasetsParameterInner}
#' \item \emph{ @returnType } \link{ResponseDataObjectSetAnnotationValueObject} \cr
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : ResponseDataObjectSetAnnotationValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The dataset does not exist.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_design } \emph{ Retrieve the design of a dataset }
#'
#' \itemize{
#' \item \emph{ @param } dataset \link{GetResultSetsDatasetsParameterInner}
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : data.frame
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The dataset does not exist.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_differential_expression } \emph{ Retrieve the expression levels of a set of datasets subject to a threshold on their differential expressions }
#'
#' \itemize{
#' \item \emph{ @param } datasets list( \link{GetResultSetsDatasetsParameterInner} )
#' \item \emph{ @param } diff_ex_set integer
#' \item \emph{ @param } threshold numeric
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } keep_non_specific character
#' \item \emph{ @param } consolidate Enum < [pickmax, pickvar, average] >
#' \item \emph{ @returnType } \link{ResponseDataObjectListExperimentExpressionLevelsValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListExperimentExpressionLevelsValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_differential_expression_analyses } \emph{ Retrieve the differential analyses of a dataset }
#'
#' \itemize{
#' \item \emph{ @param } dataset \link{GetResultSetsDatasetsParameterInner}
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @returnType } \link{ResponseDataObjectListDifferentialExpressionAnalysisValueObject} \cr
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : ResponseDataObjectListDifferentialExpressionAnalysisValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The dataset does not exist.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_differential_expression_analyses_result_sets } \emph{ Retrieve the result sets of all differential analyses of a dataset }
#'
#' \itemize{
#' \item \emph{ @param } dataset \link{GetResultSetsDatasetsParameterInner}
#'
#'
#' \item status code : 302 | If the dataset is found, a redirection to the corresponding getResultSets operation.
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The dataset does not exist.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_expression } \emph{ Retrieve the expression data of a dataset }
#'
#' \itemize{
#' \item \emph{ @param } dataset \link{GetResultSetsDatasetsParameterInner}
#' \item \emph{ @param } filter character
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : data.frame
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 204 | The dataset expression matrix is empty.
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The dataset does not exist.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_expression_for_genes } \emph{ Retrieve the expression data matrix of a set of datasets and genes }
#'
#' \itemize{
#' \item \emph{ @param } datasets list( \link{GetResultSetsDatasetsParameterInner} )
#' \item \emph{ @param } genes list( \link{GetDatasetExpressionForGenesGenesParameterInner} )
#' \item \emph{ @param } keep_non_specific character
#' \item \emph{ @param } consolidate Enum < [pickmax, pickvar, average] >
#' \item \emph{ @returnType } \link{ResponseDataObjectListExperimentExpressionLevelsValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListExperimentExpressionLevelsValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_expression_pca } \emph{ Retrieve the principal components (PCA) of a set of datasets }
#'
#' \itemize{
#' \item \emph{ @param } datasets list( \link{GetResultSetsDatasetsParameterInner} )
#' \item \emph{ @param } component integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } keep_non_specific character
#' \item \emph{ @param } consolidate Enum < [pickmax, pickvar, average] >
#' \item \emph{ @returnType } \link{ResponseDataObjectListExperimentExpressionLevelsValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListExperimentExpressionLevelsValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_platforms } \emph{ Retrieve the platform of a dataset }
#'
#' \itemize{
#' \item \emph{ @param } dataset \link{GetResultSetsDatasetsParameterInner}
#' \item \emph{ @returnType } \link{ResponseDataObjectListArrayDesignValueObject} \cr
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : ResponseDataObjectListArrayDesignValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The dataset does not exist.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_raw_expression } \emph{ Retrieve raw expression data of a dataset }
#'
#' \itemize{
#' \item \emph{ @param } dataset \link{GetResultSetsDatasetsParameterInner}
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : data.frame
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The dataset does not exist.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_samples } \emph{ Retrieve the samples of a dataset }
#'
#' \itemize{
#' \item \emph{ @param } dataset \link{GetResultSetsDatasetsParameterInner}
#' \item \emph{ @returnType } \link{ResponseDataObjectListBioAssayValueObject} \cr
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : ResponseDataObjectListBioAssayValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The dataset does not exist.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_dataset_svd } \emph{ Retrieve the singular value decomposition (SVD) of a dataset expression data }
#'
#' \itemize{
#' \item \emph{ @param } dataset \link{GetResultSetsDatasetsParameterInner}
#' \item \emph{ @returnType } \link{ResponseDataObjectSimpleSVDValueObject} \cr
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : ResponseDataObjectSimpleSVDValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The dataset does not exist.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_datasets } \emph{ Retrieve all datasets }
#'
#' \itemize{
#' \item \emph{ @param } filter character
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } sort character
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectExpressionExperimentValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectExpressionExperimentValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_datasets_by_ids } \emph{ Retrieve datasets by their identifiers }
#'
#' \itemize{
#' \item \emph{ @param } dataset list( \link{GetResultSetsDatasetsParameterInner} )
#' \item \emph{ @param } filter character
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } sort character
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectExpressionExperimentValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectExpressionExperimentValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_gene_gene_coexpression } \emph{ Retrieve the coexpression of two given genes }
#'
#' \itemize{
#' \item \emph{ @param } gene \link{GetDatasetExpressionForGenesGenesParameterInner}
#' \item \emph{ @param } with \link{GetDatasetExpressionForGenesGenesParameterInner}
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } stringency integer
#' \item \emph{ @returnType } \link{ResponseDataObjectListCoexpressionValueObjectExt} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListCoexpressionValueObjectExt
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_gene_go_terms } \emph{ Retrieve the GO terms associated to a gene }
#'
#' \itemize{
#' \item \emph{ @param } gene \link{GetDatasetExpressionForGenesGenesParameterInner}
#' \item \emph{ @returnType } \link{ResponseDataObjectListGeneOntologyTermValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListGeneOntologyTermValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_gene_locations } \emph{ Retrieve the physical locations of a given gene }
#'
#' \itemize{
#' \item \emph{ @param } gene \link{GetDatasetExpressionForGenesGenesParameterInner}
#' \item \emph{ @returnType } \link{ResponseDataObjectListPhysicalLocationValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListPhysicalLocationValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_gene_locations_in_taxon } \emph{ Retrieve physical locations for a given gene and taxon }
#'
#' \itemize{
#' \item \emph{ @param } taxon \link{SearchTaxonDatasetsTaxonParameter}
#' \item \emph{ @param } gene \link{GetDatasetExpressionForGenesGenesParameterInner}
#' \item \emph{ @returnType } \link{ResponseDataObjectListPhysicalLocationValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListPhysicalLocationValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_gene_probes } \emph{ Retrieve the probes associated to a genes }
#'
#' \itemize{
#' \item \emph{ @param } gene \link{GetDatasetExpressionForGenesGenesParameterInner}
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectCompositeSequenceValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectCompositeSequenceValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_genes } \emph{ Retrieve genes matching a gene identifier }
#'
#' \itemize{
#' \item \emph{ @param } genes list( \link{GetDatasetExpressionForGenesGenesParameterInner} )
#' \item \emph{ @returnType } \link{ResponseDataObjectListGeneValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListGeneValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_platform_annotations } \emph{ Retrieve the annotations of a given platform }
#'
#' \itemize{
#' \item \emph{ @param } platform \link{GetPlatformAnnotationsPlatformParameter}
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : data.frame
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_platform_datasets } \emph{ Retrieve all experiments within a given platform }
#'
#' \itemize{
#' \item \emph{ @param } platform \link{GetPlatformAnnotationsPlatformParameter}
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectExpressionExperimentValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectExpressionExperimentValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_platform_element } \emph{ Retrieve the selected composite sequences for a given platform }
#'
#' \itemize{
#' \item \emph{ @param } platform \link{GetPlatformAnnotationsPlatformParameter}
#' \item \emph{ @param } probes list( \link{GetPlatformElementProbesParameterInner} )
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectCompositeSequenceValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectCompositeSequenceValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_platform_element_genes } \emph{ Retrieve the genes associated to a probe in a given platform }
#'
#' \itemize{
#' \item \emph{ @param } platform \link{GetPlatformAnnotationsPlatformParameter}
#' \item \emph{ @param } probe \link{GetPlatformElementProbesParameterInner}
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectGeneValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectGeneValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_platform_elements } \emph{ Retrieve the composite sequences for a given platform }
#'
#' \itemize{
#' \item \emph{ @param } platform \link{GetPlatformAnnotationsPlatformParameter}
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectCompositeSequenceValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectCompositeSequenceValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_platforms } \emph{ Retrieve all platforms }
#'
#' \itemize{
#' \item \emph{ @param } filter character
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } sort character
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectArrayDesignValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectArrayDesignValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_platforms_by_ids } \emph{ Retrieve all platforms matching a set of platform identifiers }
#'
#' \itemize{
#' \item \emph{ @param } platform list( \link{GetPlatformAnnotationsPlatformParameter} )
#' \item \emph{ @param } filter character
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } sort character
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectArrayDesignValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectArrayDesignValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_result_set } \emph{ Retrieve a single analysis result set by its identifier }
#'
#' \itemize{
#' \item \emph{ @param } result_set integer
#' \item \emph{ @returnType } \link{ResponseDataObjectExpressionAnalysisResultSetValueObject} \cr
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : ResponseDataObjectExpressionAnalysisResultSetValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The analysis result set could not be found.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_result_set_as_tsv } \emph{ Retrieve a single analysis result set by its identifier as a tab-separated values }
#'
#' \itemize{
#' \item \emph{ @param } result_set_ integer
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : data.frame
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | The analysis result set could not be found.
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_result_sets } \emph{ Retrieve all result sets matching the provided criteria }
#'
#' \itemize{
#' \item \emph{ @param } datasets list( \link{GetResultSetsDatasetsParameterInner} )
#' \item \emph{ @param } database_entries list( \link{GetResultSetsDatabaseEntriesParameterInner} )
#' \item \emph{ @param } filter character
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } sort character
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectExpressionAnalysisResultSetValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectExpressionAnalysisResultSetValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_taxa } \emph{ Retrieve all available taxa }
#'
#' \itemize{
#' \item \emph{ @returnType } \link{ResponseDataObjectListTaxonValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListTaxonValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_taxa_by_ids } \emph{ Retrieve taxa by their identifiers }
#'
#' \itemize{
#' \item \emph{ @param } taxa list( \link{SearchTaxonDatasetsTaxonParameter} )
#' \item \emph{ @returnType } \link{ResponseDataObjectListTaxonValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListTaxonValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_taxon_datasets } \emph{ Retrieve the datasets for a given taxon }
#'
#' \itemize{
#' \item \emph{ @param } taxon \link{SearchTaxonDatasetsTaxonParameter}
#' \item \emph{ @param } filter character
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } sort character
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectExpressionExperimentValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectExpressionExperimentValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_taxon_genes } \emph{ Retrieve all genes in a given taxon }
#'
#' \itemize{
#' \item \emph{ @param } taxon \link{SearchTaxonDatasetsTaxonParameter}
#' \item \emph{ @param } gene \link{GetDatasetExpressionForGenesGenesParameterInner}
#' \item \emph{ @returnType } \link{ResponseDataObjectListGeneValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListGeneValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ get_taxon_genes_overlapping_chromosome } \emph{ Retrieve genes overlapping a given region in a taxon }
#'
#' \itemize{
#' \item \emph{ @param } taxon \link{SearchTaxonDatasetsTaxonParameter}
#' \item \emph{ @param } chromosome character
#' \item \emph{ @param } strand character
#' \item \emph{ @param } start integer
#' \item \emph{ @param } size integer
#' \item \emph{ @returnType } \link{ResponseDataObjectListGeneValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : ResponseDataObjectListGeneValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ search } \emph{ Search everything in Gemma. }
#'
#' \itemize{
#' \item \emph{ @param } query character
#' \item \emph{ @param } taxon \link{SearchTaxonDatasetsTaxonParameter}
#' \item \emph{ @param } platform \link{GetPlatformAnnotationsPlatformParameter}
#' \item \emph{ @param } result_types Enum < [ubic.gemma.model.common.description.BibliographicReference, ubic.gemma.model.analysis.expression.ExpressionExperimentSet, ubic.gemma.model.genome.gene.GeneSet, ubic.gemma.model.genome.Gene, ubic.gemma.model.expression.experiment.ExpressionExperiment, ubic.gemma.model.genome.biosequence.BioSequence, ubic.gemma.model.association.phenotype.PhenotypeAssociation, ubic.gemma.model.expression.arrayDesign.ArrayDesign, ubic.gemma.model.expression.designElement.CompositeSequence] >
#' \item \emph{ @param } limit integer
#' \item \emph{ @returnType } \link{SearchResultsResponseDataObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : SearchResultsResponseDataObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ search_annotations } \emph{ Search for annotation tags }
#'
#' \itemize{
#' \item \emph{ @param } query list( character )
#' \item \emph{ @returnType } \link{ResponseDataObjectListAnnotationSearchResultValueObject} \cr
#'
#'
#' \item status code : 200 | 
#'
#' \item return type : ResponseDataObjectListAnnotationSearchResultValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | The search query is empty.
#'
#' \item return type : ResponseErrorObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ search_datasets } \emph{ Retrieve datasets associated to an annotation tags search }
#'
#' \itemize{
#' \item \emph{ @param } query list( character )
#' \item \emph{ @param } filter character
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } sort character
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectExpressionExperimentValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectExpressionExperimentValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ search_taxon_datasets } \emph{ Retrieve datasets within a given taxa associated to an annotation tags search }
#'
#' \itemize{
#' \item \emph{ @param } taxon \link{SearchTaxonDatasetsTaxonParameter}
#' \item \emph{ @param } query list( character )
#' \item \emph{ @param } filter character
#' \item \emph{ @param } offset integer
#' \item \emph{ @param } limit integer
#' \item \emph{ @param } sort character
#' \item \emph{ @returnType } \link{PaginatedResponseDataObjectExpressionExperimentValueObject} \cr
#'
#'
#' \item status code : 0 | default response
#'
#' \item return type : PaginatedResponseDataObjectExpressionExperimentValueObject
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' }
#'
#'
#' @examples
#' \dontrun{
#' ####################  get_api_info  ####################
#'
#' library(gemma.R)
#'
#' #Retrieve an object with basic API information
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_api_info(data_file = "result.txt")
#' result <- api_instance$get_api_info()
#' dput(result)
#'
#'
#' ####################  get_dataset_annotations  ####################
#'
#' library(gemma.R)
#' var_dataset <- GetResultSetsDatasetsParameterInner$new() # GetResultSetsDatasetsParameterInner | 
#'
#' #Retrieve the annotations analysis of a dataset
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_annotations(var_datasetdata_file = "result.txt")
#' result <- api_instance$get_dataset_annotations(var_dataset)
#' dput(result)
#'
#'
#' ####################  get_dataset_design  ####################
#'
#' library(gemma.R)
#' var_dataset <- GetResultSetsDatasetsParameterInner$new() # GetResultSetsDatasetsParameterInner | 
#'
#' #Retrieve the design of a dataset
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_design(var_datasetdata_file = "result.txt")
#' result <- api_instance$get_dataset_design(var_dataset)
#' dput(result)
#'
#'
#' ####################  get_dataset_differential_expression  ####################
#'
#' library(gemma.R)
#' var_datasets <- [GetResultSetsDatasetsParameterInner$new()] # array[GetResultSetsDatasetsParameterInner] | 
#' var_diff_ex_set <- 56 # integer | 
#' var_threshold <- 1.0 # numeric | 
#' var_limit <- 100 # integer | 
#' var_keep_non_specific <- FALSE # character | 
#' var_consolidate <- "consolidate_example" # character | 
#'
#' #Retrieve the expression levels of a set of datasets subject to a threshold on their differential expressions
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_differential_expression(var_datasets, diff_ex_set = var_diff_ex_set, threshold = var_threshold, limit = var_limit, keep_non_specific = var_keep_non_specific, consolidate = var_consolidatedata_file = "result.txt")
#' result <- api_instance$get_dataset_differential_expression(var_datasets, diff_ex_set = var_diff_ex_set, threshold = var_threshold, limit = var_limit, keep_non_specific = var_keep_non_specific, consolidate = var_consolidate)
#' dput(result)
#'
#'
#' ####################  get_dataset_differential_expression_analyses  ####################
#'
#' library(gemma.R)
#' var_dataset <- GetResultSetsDatasetsParameterInner$new() # GetResultSetsDatasetsParameterInner | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#'
#' #Retrieve the differential analyses of a dataset
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_differential_expression_analyses(var_dataset, offset = var_offset, limit = var_limitdata_file = "result.txt")
#' result <- api_instance$get_dataset_differential_expression_analyses(var_dataset, offset = var_offset, limit = var_limit)
#' dput(result)
#'
#'
#' ####################  get_dataset_differential_expression_analyses_result_sets  ####################
#'
#' library(gemma.R)
#' var_dataset <- GetResultSetsDatasetsParameterInner$new() # GetResultSetsDatasetsParameterInner | 
#'
#' #Retrieve the result sets of all differential analyses of a dataset
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' api_instance$get_dataset_differential_expression_analyses_result_sets(var_dataset)
#'
#'
#' ####################  get_dataset_expression  ####################
#'
#' library(gemma.R)
#' var_dataset <- GetResultSetsDatasetsParameterInner$new() # GetResultSetsDatasetsParameterInner | 
#' var_filter <- FALSE # character | 
#'
#' #Retrieve the expression data of a dataset
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_expression(var_dataset, filter = var_filterdata_file = "result.txt")
#' result <- api_instance$get_dataset_expression(var_dataset, filter = var_filter)
#' dput(result)
#'
#'
#' ####################  get_dataset_expression_for_genes  ####################
#'
#' library(gemma.R)
#' var_datasets <- [GetResultSetsDatasetsParameterInner$new()] # array[GetResultSetsDatasetsParameterInner] | 
#' var_genes <- [GetDatasetExpressionForGenesGenesParameterInner$new()] # array[GetDatasetExpressionForGenesGenesParameterInner] | 
#' var_keep_non_specific <- FALSE # character | 
#' var_consolidate <- "consolidate_example" # character | 
#'
#' #Retrieve the expression data matrix of a set of datasets and genes
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_expression_for_genes(var_datasets, var_genes, keep_non_specific = var_keep_non_specific, consolidate = var_consolidatedata_file = "result.txt")
#' result <- api_instance$get_dataset_expression_for_genes(var_datasets, var_genes, keep_non_specific = var_keep_non_specific, consolidate = var_consolidate)
#' dput(result)
#'
#'
#' ####################  get_dataset_expression_pca  ####################
#'
#' library(gemma.R)
#' var_datasets <- [GetResultSetsDatasetsParameterInner$new()] # array[GetResultSetsDatasetsParameterInner] | 
#' var_component <- 1 # integer | 
#' var_limit <- 100 # integer | 
#' var_keep_non_specific <- FALSE # character | 
#' var_consolidate <- "consolidate_example" # character | 
#'
#' #Retrieve the principal components (PCA) of a set of datasets
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_expression_pca(var_datasets, component = var_component, limit = var_limit, keep_non_specific = var_keep_non_specific, consolidate = var_consolidatedata_file = "result.txt")
#' result <- api_instance$get_dataset_expression_pca(var_datasets, component = var_component, limit = var_limit, keep_non_specific = var_keep_non_specific, consolidate = var_consolidate)
#' dput(result)
#'
#'
#' ####################  get_dataset_platforms  ####################
#'
#' library(gemma.R)
#' var_dataset <- GetResultSetsDatasetsParameterInner$new() # GetResultSetsDatasetsParameterInner | 
#'
#' #Retrieve the platform of a dataset
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_platforms(var_datasetdata_file = "result.txt")
#' result <- api_instance$get_dataset_platforms(var_dataset)
#' dput(result)
#'
#'
#' ####################  get_dataset_raw_expression  ####################
#'
#' library(gemma.R)
#' var_dataset <- GetResultSetsDatasetsParameterInner$new() # GetResultSetsDatasetsParameterInner | 
#'
#' #Retrieve raw expression data of a dataset
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_raw_expression(var_datasetdata_file = "result.txt")
#' result <- api_instance$get_dataset_raw_expression(var_dataset)
#' dput(result)
#'
#'
#' ####################  get_dataset_samples  ####################
#'
#' library(gemma.R)
#' var_dataset <- GetResultSetsDatasetsParameterInner$new() # GetResultSetsDatasetsParameterInner | 
#'
#' #Retrieve the samples of a dataset
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_samples(var_datasetdata_file = "result.txt")
#' result <- api_instance$get_dataset_samples(var_dataset)
#' dput(result)
#'
#'
#' ####################  get_dataset_svd  ####################
#'
#' library(gemma.R)
#' var_dataset <- GetResultSetsDatasetsParameterInner$new() # GetResultSetsDatasetsParameterInner | 
#'
#' #Retrieve the singular value decomposition (SVD) of a dataset expression data
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_dataset_svd(var_datasetdata_file = "result.txt")
#' result <- api_instance$get_dataset_svd(var_dataset)
#' dput(result)
#'
#'
#' ####################  get_datasets  ####################
#'
#' library(gemma.R)
#' var_filter <- "" # character | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#' var_sort <- "+id" # character | 
#'
#' #Retrieve all datasets
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_datasets(filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sortdata_file = "result.txt")
#' result <- api_instance$get_datasets(filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sort)
#' dput(result)
#'
#'
#' ####################  get_datasets_by_ids  ####################
#'
#' library(gemma.R)
#' var_dataset <- [GetResultSetsDatasetsParameterInner$new()] # array[GetResultSetsDatasetsParameterInner] | 
#' var_filter <- "" # character | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#' var_sort <- "+id" # character | 
#'
#' #Retrieve datasets by their identifiers
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_datasets_by_ids(var_dataset, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sortdata_file = "result.txt")
#' result <- api_instance$get_datasets_by_ids(var_dataset, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sort)
#' dput(result)
#'
#'
#' ####################  get_gene_gene_coexpression  ####################
#'
#' library(gemma.R)
#' var_gene <- GetDatasetExpressionForGenesGenesParameterInner$new() # GetDatasetExpressionForGenesGenesParameterInner | 
#' var_with <- GetDatasetExpressionForGenesGenesParameterInner$new() # GetDatasetExpressionForGenesGenesParameterInner | 
#' var_limit <- 100 # integer | 
#' var_stringency <- 1 # integer | 
#'
#' #Retrieve the coexpression of two given genes
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_gene_gene_coexpression(var_gene, with = var_with, limit = var_limit, stringency = var_stringencydata_file = "result.txt")
#' result <- api_instance$get_gene_gene_coexpression(var_gene, with = var_with, limit = var_limit, stringency = var_stringency)
#' dput(result)
#'
#'
#' ####################  get_gene_go_terms  ####################
#'
#' library(gemma.R)
#' var_gene <- GetDatasetExpressionForGenesGenesParameterInner$new() # GetDatasetExpressionForGenesGenesParameterInner | 
#'
#' #Retrieve the GO terms associated to a gene
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_gene_go_terms(var_genedata_file = "result.txt")
#' result <- api_instance$get_gene_go_terms(var_gene)
#' dput(result)
#'
#'
#' ####################  get_gene_locations  ####################
#'
#' library(gemma.R)
#' var_gene <- GetDatasetExpressionForGenesGenesParameterInner$new() # GetDatasetExpressionForGenesGenesParameterInner | 
#'
#' #Retrieve the physical locations of a given gene
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_gene_locations(var_genedata_file = "result.txt")
#' result <- api_instance$get_gene_locations(var_gene)
#' dput(result)
#'
#'
#' ####################  get_gene_locations_in_taxon  ####################
#'
#' library(gemma.R)
#' var_taxon <- SearchTaxonDatasetsTaxonParameter$new() # SearchTaxonDatasetsTaxonParameter | 
#' var_gene <- GetDatasetExpressionForGenesGenesParameterInner$new() # GetDatasetExpressionForGenesGenesParameterInner | 
#'
#' #Retrieve physical locations for a given gene and taxon
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_gene_locations_in_taxon(var_taxon, var_genedata_file = "result.txt")
#' result <- api_instance$get_gene_locations_in_taxon(var_taxon, var_gene)
#' dput(result)
#'
#'
#' ####################  get_gene_probes  ####################
#'
#' library(gemma.R)
#' var_gene <- GetDatasetExpressionForGenesGenesParameterInner$new() # GetDatasetExpressionForGenesGenesParameterInner | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#'
#' #Retrieve the probes associated to a genes
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_gene_probes(var_gene, offset = var_offset, limit = var_limitdata_file = "result.txt")
#' result <- api_instance$get_gene_probes(var_gene, offset = var_offset, limit = var_limit)
#' dput(result)
#'
#'
#' ####################  get_genes  ####################
#'
#' library(gemma.R)
#' var_genes <- [GetDatasetExpressionForGenesGenesParameterInner$new()] # array[GetDatasetExpressionForGenesGenesParameterInner] | 
#'
#' #Retrieve genes matching a gene identifier
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_genes(var_genesdata_file = "result.txt")
#' result <- api_instance$get_genes(var_genes)
#' dput(result)
#'
#'
#' ####################  get_platform_annotations  ####################
#'
#' library(gemma.R)
#' var_platform <- GetPlatformAnnotationsPlatformParameter$new() # GetPlatformAnnotationsPlatformParameter | 
#'
#' #Retrieve the annotations of a given platform
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_platform_annotations(var_platformdata_file = "result.txt")
#' result <- api_instance$get_platform_annotations(var_platform)
#' dput(result)
#'
#'
#' ####################  get_platform_datasets  ####################
#'
#' library(gemma.R)
#' var_platform <- GetPlatformAnnotationsPlatformParameter$new() # GetPlatformAnnotationsPlatformParameter | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#'
#' #Retrieve all experiments within a given platform
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_platform_datasets(var_platform, offset = var_offset, limit = var_limitdata_file = "result.txt")
#' result <- api_instance$get_platform_datasets(var_platform, offset = var_offset, limit = var_limit)
#' dput(result)
#'
#'
#' ####################  get_platform_element  ####################
#'
#' library(gemma.R)
#' var_platform <- GetPlatformAnnotationsPlatformParameter$new() # GetPlatformAnnotationsPlatformParameter | 
#' var_probes <- [GetPlatformElementProbesParameterInner$new()] # array[GetPlatformElementProbesParameterInner] | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#'
#' #Retrieve the selected composite sequences for a given platform
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_platform_element(var_platform, var_probes, offset = var_offset, limit = var_limitdata_file = "result.txt")
#' result <- api_instance$get_platform_element(var_platform, var_probes, offset = var_offset, limit = var_limit)
#' dput(result)
#'
#'
#' ####################  get_platform_element_genes  ####################
#'
#' library(gemma.R)
#' var_platform <- GetPlatformAnnotationsPlatformParameter$new() # GetPlatformAnnotationsPlatformParameter | 
#' var_probe <- GetPlatformElementProbesParameterInner$new() # GetPlatformElementProbesParameterInner | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#'
#' #Retrieve the genes associated to a probe in a given platform
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_platform_element_genes(var_platform, var_probe, offset = var_offset, limit = var_limitdata_file = "result.txt")
#' result <- api_instance$get_platform_element_genes(var_platform, var_probe, offset = var_offset, limit = var_limit)
#' dput(result)
#'
#'
#' ####################  get_platform_elements  ####################
#'
#' library(gemma.R)
#' var_platform <- GetPlatformAnnotationsPlatformParameter$new() # GetPlatformAnnotationsPlatformParameter | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#'
#' #Retrieve the composite sequences for a given platform
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_platform_elements(var_platform, offset = var_offset, limit = var_limitdata_file = "result.txt")
#' result <- api_instance$get_platform_elements(var_platform, offset = var_offset, limit = var_limit)
#' dput(result)
#'
#'
#' ####################  get_platforms  ####################
#'
#' library(gemma.R)
#' var_filter <- "" # character | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#' var_sort <- "+id" # character | 
#'
#' #Retrieve all platforms
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_platforms(filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sortdata_file = "result.txt")
#' result <- api_instance$get_platforms(filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sort)
#' dput(result)
#'
#'
#' ####################  get_platforms_by_ids  ####################
#'
#' library(gemma.R)
#' var_platform <- [GetPlatformAnnotationsPlatformParameter$new()] # array[GetPlatformAnnotationsPlatformParameter] | 
#' var_filter <- "" # character | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#' var_sort <- "+id" # character | 
#'
#' #Retrieve all platforms matching a set of platform identifiers
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_platforms_by_ids(var_platform, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sortdata_file = "result.txt")
#' result <- api_instance$get_platforms_by_ids(var_platform, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sort)
#' dput(result)
#'
#'
#' ####################  get_result_set  ####################
#'
#' library(gemma.R)
#' var_result_set <- 56 # integer | 
#'
#' #Retrieve a single analysis result set by its identifier
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_result_set(var_result_setdata_file = "result.txt")
#' result <- api_instance$get_result_set(var_result_set)
#' dput(result)
#'
#'
#' ####################  get_result_set_as_tsv  ####################
#'
#' library(gemma.R)
#' var_result_set_ <- 56 # integer | 
#'
#' #Retrieve a single analysis result set by its identifier as a tab-separated values
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_result_set_as_tsv(var_result_set_data_file = "result.txt")
#' result <- api_instance$get_result_set_as_tsv(var_result_set_)
#' dput(result)
#'
#'
#' ####################  get_result_sets  ####################
#'
#' library(gemma.R)
#' var_datasets <- [GetResultSetsDatasetsParameterInner$new()] # array[GetResultSetsDatasetsParameterInner] | 
#' var_database_entries <- [GetResultSetsDatabaseEntriesParameterInner$new()] # array[GetResultSetsDatabaseEntriesParameterInner] | 
#' var_filter <- "" # character | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#' var_sort <- "+id" # character | 
#'
#' #Retrieve all result sets matching the provided criteria
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_result_sets(datasets = var_datasets, database_entries = var_database_entries, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sortdata_file = "result.txt")
#' result <- api_instance$get_result_sets(datasets = var_datasets, database_entries = var_database_entries, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sort)
#' dput(result)
#'
#'
#' ####################  get_taxa  ####################
#'
#' library(gemma.R)
#'
#' #Retrieve all available taxa
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_taxa(data_file = "result.txt")
#' result <- api_instance$get_taxa()
#' dput(result)
#'
#'
#' ####################  get_taxa_by_ids  ####################
#'
#' library(gemma.R)
#' var_taxa <- [SearchTaxonDatasetsTaxonParameter$new()] # array[SearchTaxonDatasetsTaxonParameter] | 
#'
#' #Retrieve taxa by their identifiers
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_taxa_by_ids(var_taxadata_file = "result.txt")
#' result <- api_instance$get_taxa_by_ids(var_taxa)
#' dput(result)
#'
#'
#' ####################  get_taxon_datasets  ####################
#'
#' library(gemma.R)
#' var_taxon <- SearchTaxonDatasetsTaxonParameter$new() # SearchTaxonDatasetsTaxonParameter | 
#' var_filter <- "" # character | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#' var_sort <- "+id" # character | 
#'
#' #Retrieve the datasets for a given taxon
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_taxon_datasets(var_taxon, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sortdata_file = "result.txt")
#' result <- api_instance$get_taxon_datasets(var_taxon, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sort)
#' dput(result)
#'
#'
#' ####################  get_taxon_genes  ####################
#'
#' library(gemma.R)
#' var_taxon <- SearchTaxonDatasetsTaxonParameter$new() # SearchTaxonDatasetsTaxonParameter | 
#' var_gene <- GetDatasetExpressionForGenesGenesParameterInner$new() # GetDatasetExpressionForGenesGenesParameterInner | 
#'
#' #Retrieve all genes in a given taxon
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_taxon_genes(var_taxon, var_genedata_file = "result.txt")
#' result <- api_instance$get_taxon_genes(var_taxon, var_gene)
#' dput(result)
#'
#'
#' ####################  get_taxon_genes_overlapping_chromosome  ####################
#'
#' library(gemma.R)
#' var_taxon <- SearchTaxonDatasetsTaxonParameter$new() # SearchTaxonDatasetsTaxonParameter | 
#' var_chromosome <- "chromosome_example" # character | 
#' var_strand <- "+" # character | 
#' var_start <- 56 # integer | 
#' var_size <- 56 # integer | 
#'
#' #Retrieve genes overlapping a given region in a taxon
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$get_taxon_genes_overlapping_chromosome(var_taxon, var_chromosome, strand = var_strand, start = var_start, size = var_sizedata_file = "result.txt")
#' result <- api_instance$get_taxon_genes_overlapping_chromosome(var_taxon, var_chromosome, strand = var_strand, start = var_start, size = var_size)
#' dput(result)
#'
#'
#' ####################  search  ####################
#'
#' library(gemma.R)
#' var_query <- "query_example" # character | 
#' var_taxon <- SearchTaxonDatasetsTaxonParameter$new() # SearchTaxonDatasetsTaxonParameter | 
#' var_platform <- GetPlatformAnnotationsPlatformParameter$new() # GetPlatformAnnotationsPlatformParameter | 
#' var_result_types <- ["result_types_example"] # array[character] | 
#' var_limit <- 20 # integer | 
#'
#' #Search everything in Gemma.
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$search(query = var_query, taxon = var_taxon, platform = var_platform, result_types = var_result_types, limit = var_limitdata_file = "result.txt")
#' result <- api_instance$search(query = var_query, taxon = var_taxon, platform = var_platform, result_types = var_result_types, limit = var_limit)
#' dput(result)
#'
#'
#' ####################  search_annotations  ####################
#'
#' library(gemma.R)
#' var_query <- ["query_example"] # array[character] | 
#'
#' #Search for annotation tags
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$search_annotations(var_querydata_file = "result.txt")
#' result <- api_instance$search_annotations(var_query)
#' dput(result)
#'
#'
#' ####################  search_datasets  ####################
#'
#' library(gemma.R)
#' var_query <- ["query_example"] # array[character] | 
#' var_filter <- "" # character | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#' var_sort <- "+id" # character | 
#'
#' #Retrieve datasets associated to an annotation tags search
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$search_datasets(var_query, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sortdata_file = "result.txt")
#' result <- api_instance$search_datasets(var_query, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sort)
#' dput(result)
#'
#'
#' ####################  search_taxon_datasets  ####################
#'
#' library(gemma.R)
#' var_taxon <- SearchTaxonDatasetsTaxonParameter$new() # SearchTaxonDatasetsTaxonParameter | 
#' var_query <- ["query_example"] # array[character] | 
#' var_filter <- "" # character | 
#' var_offset <- 0 # integer | 
#' var_limit <- 20 # integer | 
#' var_sort <- "+id" # character | 
#'
#' #Retrieve datasets within a given taxa associated to an annotation tags search
#' api_instance <- DefaultApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # Configure API key authorization: cookieAuth
#' api_instance$api_client$api_keys["JSESSIONID"] <- Sys.getenv("API_KEY")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$search_taxon_datasets(var_taxon, var_query, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sortdata_file = "result.txt")
#' result <- api_instance$search_taxon_datasets(var_taxon, var_query, filter = var_filter, offset = var_offset, limit = var_limit, sort = var_sort)
#' dput(result)
#'
#'
#' }
#' @importFrom R6 R6Class
#' @importFrom base64enc base64encode
#' @keywords internal
DefaultApi <- R6::R6Class(
  "DefaultApi",
  public = list(
    api_client = NULL,
    #' Initialize a new DefaultApi.
    #'
    #' @description
    #' Initialize a new DefaultApi.
    #'
    #' @param api_client An instance of API client.
    #' @keywords internal
    initialize = function(api_client) {
      if (!missing(api_client)) {
        self$api_client <- api_client
      } else {
        self$api_client <- ApiClient$new()
      }
    },
    #' Retrieve an object with basic API information
    #'
    #' @description
    #' Retrieve an object with basic API information
    #'
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectApiInfoValueObject
    #' @keywords internal
    get_api_info = function(data_file = NULL, ...) {
      local_var_response <- self$get_api_info_with_http_info(data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve an object with basic API information
    #'
    #' @description
    #' Retrieve an object with basic API information
    #'
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectApiInfoValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_api_info_with_http_info = function(data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      local_var_url_path <- "/"
      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_api_info`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectApiInfoValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the annotations analysis of a dataset
    #'
    #' @description
    #' Retrieve the annotations analysis of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectSetAnnotationValueObject
    #' @keywords internal
    get_dataset_annotations = function(dataset, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_annotations_with_http_info(dataset, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the annotations analysis of a dataset
    #'
    #' @description
    #' Retrieve the annotations analysis of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectSetAnnotationValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_annotations_with_http_info = function(dataset, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }


      local_var_url_path <- "/datasets/{dataset}/annotations"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_annotations`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectSetAnnotationValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the design of a dataset
    #'
    #' @description
    #' Retrieve the design of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return data.frame
    #' @keywords internal
    get_dataset_design = function(dataset, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_design_with_http_info(dataset, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the design of a dataset
    #'
    #' @description
    #' Retrieve the design of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (data.frame) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_design_with_http_info = function(dataset, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }


      local_var_url_path <- "/datasets/{dataset}/design"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_design`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("text/tab-separated-values; charset=UTF-8", "application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "data.frame", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the expression levels of a set of datasets subject to a threshold on their differential expressions
    #'
    #' @description
    #' Retrieve the expression levels of a set of datasets subject to a threshold on their differential expressions
    #'
    #' @param datasets 
    #' @param diff_ex_set (optional) No description
    #' @param threshold (optional) No description (default value: 1.0)
    #' @param limit (optional) No description (default value: 100)
    #' @param keep_non_specific (optional) No description (default value: FALSE)
    #' @param consolidate (optional) No description
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListExperimentExpressionLevelsValueObject
    #' @keywords internal
    get_dataset_differential_expression = function(datasets, diff_ex_set = NULL, threshold = 1.0, limit = 100, keep_non_specific = FALSE, consolidate = NULL, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_differential_expression_with_http_info(datasets, diff_ex_set, threshold, limit, keep_non_specific, consolidate, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the expression levels of a set of datasets subject to a threshold on their differential expressions
    #'
    #' @description
    #' Retrieve the expression levels of a set of datasets subject to a threshold on their differential expressions
    #'
    #' @param datasets 
    #' @param diff_ex_set (optional) No description
    #' @param threshold (optional) No description (default value: 1.0)
    #' @param limit (optional) No description (default value: 100)
    #' @param keep_non_specific (optional) No description (default value: FALSE)
    #' @param consolidate (optional) No description
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListExperimentExpressionLevelsValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_differential_expression_with_http_info = function(datasets, diff_ex_set = NULL, threshold = 1.0, limit = 100, keep_non_specific = FALSE, consolidate = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`datasets`)) {
        stop("Missing required parameter `datasets`.")
      }




      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_dataset_differential_expression, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_dataset_differential_expression, must be bigger than or equal to 1.")
      }



      query_params["diffExSet"] <- `diff_ex_set`

      query_params["threshold"] <- `threshold`

      query_params["limit"] <- `limit`

      query_params["keepNonSpecific"] <- `keep_non_specific`

      query_params["consolidate"] <- `consolidate`

      local_var_url_path <- "/datasets/{datasets}/expressions/differential"
      if (!missing(`datasets`)) {
        local_var_url_path <- gsub(paste0("\\{", "datasets", "\\}"), URLencode(as.character(`datasets`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_differential_expression`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListExperimentExpressionLevelsValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the differential analyses of a dataset
    #'
    #' @description
    #' Retrieve the differential analyses of a dataset
    #'
    #' @param dataset 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListDifferentialExpressionAnalysisValueObject
    #' @keywords internal
    get_dataset_differential_expression_analyses = function(dataset, offset = 0, limit = 20, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_differential_expression_analyses_with_http_info(dataset, offset, limit, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the differential analyses of a dataset
    #'
    #' @description
    #' Retrieve the differential analyses of a dataset
    #'
    #' @param dataset 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListDifferentialExpressionAnalysisValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_differential_expression_analyses_with_http_info = function(dataset, offset = 0, limit = 20, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }


      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_dataset_differential_expression_analyses, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_dataset_differential_expression_analyses, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_dataset_differential_expression_analyses, must be bigger than or equal to 1.")
      }

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      local_var_url_path <- "/datasets/{dataset}/analyses/differential"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_differential_expression_analyses`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListDifferentialExpressionAnalysisValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the result sets of all differential analyses of a dataset
    #'
    #' @description
    #' Retrieve the result sets of all differential analyses of a dataset
    #'
    #' @param dataset 
    #' @param ... Other optional arguments
    #' @return void
    #' @keywords internal
    get_dataset_differential_expression_analyses_result_sets = function(dataset, ...) {
      local_var_response <- self$get_dataset_differential_expression_analyses_result_sets_with_http_info(dataset, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the result sets of all differential analyses of a dataset
    #'
    #' @description
    #' Retrieve the result sets of all differential analyses of a dataset
    #'
    #' @param dataset 
    #' @param ... Other optional arguments
    #' @return API response (void) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_differential_expression_analyses_result_sets_with_http_info = function(dataset, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }


      local_var_url_path <- "/datasets/{dataset}/analyses/differential/resultSets"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_differential_expression_analyses_result_sets`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("*/*")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        local_var_resp$content <- NULL
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the expression data of a dataset
    #'
    #' @description
    #' Retrieve the expression data of a dataset
    #'
    #' @param dataset 
    #' @param filter (optional) No description (default value: FALSE)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return data.frame
    #' @keywords internal
    get_dataset_expression = function(dataset, filter = FALSE, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_expression_with_http_info(dataset, filter, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the expression data of a dataset
    #'
    #' @description
    #' Retrieve the expression data of a dataset
    #'
    #' @param dataset 
    #' @param filter (optional) No description (default value: FALSE)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (data.frame) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_expression_with_http_info = function(dataset, filter = FALSE, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }



      query_params["filter"] <- `filter`

      local_var_url_path <- "/datasets/{dataset}/data"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_expression`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("text/tab-separated-values; charset=UTF-8", "application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "data.frame", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the expression data matrix of a set of datasets and genes
    #'
    #' @description
    #' Retrieve the expression data matrix of a set of datasets and genes
    #'
    #' @param datasets 
    #' @param genes 
    #' @param keep_non_specific (optional) No description (default value: FALSE)
    #' @param consolidate (optional) No description
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListExperimentExpressionLevelsValueObject
    #' @keywords internal
    get_dataset_expression_for_genes = function(datasets, genes, keep_non_specific = FALSE, consolidate = NULL, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_expression_for_genes_with_http_info(datasets, genes, keep_non_specific, consolidate, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the expression data matrix of a set of datasets and genes
    #'
    #' @description
    #' Retrieve the expression data matrix of a set of datasets and genes
    #'
    #' @param datasets 
    #' @param genes 
    #' @param keep_non_specific (optional) No description (default value: FALSE)
    #' @param consolidate (optional) No description
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListExperimentExpressionLevelsValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_expression_for_genes_with_http_info = function(datasets, genes, keep_non_specific = FALSE, consolidate = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`datasets`)) {
        stop("Missing required parameter `datasets`.")
      }

      if (missing(`genes`)) {
        stop("Missing required parameter `genes`.")
      }





      query_params["keepNonSpecific"] <- `keep_non_specific`

      query_params["consolidate"] <- `consolidate`

      local_var_url_path <- "/datasets/{datasets}/expressions/genes/{genes}"
      if (!missing(`datasets`)) {
        local_var_url_path <- gsub(paste0("\\{", "datasets", "\\}"), URLencode(as.character(`datasets`), reserved = TRUE), local_var_url_path)
      }

      if (!missing(`genes`)) {
        local_var_url_path <- gsub(paste0("\\{", "genes", "\\}"), URLencode(as.character(`genes`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_expression_for_genes`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListExperimentExpressionLevelsValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the principal components (PCA) of a set of datasets
    #'
    #' @description
    #' Retrieve the principal components (PCA) of a set of datasets
    #'
    #' @param datasets 
    #' @param component (optional) No description (default value: 1)
    #' @param limit (optional) No description (default value: 100)
    #' @param keep_non_specific (optional) No description (default value: FALSE)
    #' @param consolidate (optional) No description
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListExperimentExpressionLevelsValueObject
    #' @keywords internal
    get_dataset_expression_pca = function(datasets, component = 1, limit = 100, keep_non_specific = FALSE, consolidate = NULL, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_expression_pca_with_http_info(datasets, component, limit, keep_non_specific, consolidate, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the principal components (PCA) of a set of datasets
    #'
    #' @description
    #' Retrieve the principal components (PCA) of a set of datasets
    #'
    #' @param datasets 
    #' @param component (optional) No description (default value: 1)
    #' @param limit (optional) No description (default value: 100)
    #' @param keep_non_specific (optional) No description (default value: FALSE)
    #' @param consolidate (optional) No description
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListExperimentExpressionLevelsValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_expression_pca_with_http_info = function(datasets, component = 1, limit = 100, keep_non_specific = FALSE, consolidate = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`datasets`)) {
        stop("Missing required parameter `datasets`.")
      }



      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_dataset_expression_pca, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_dataset_expression_pca, must be bigger than or equal to 1.")
      }



      query_params["component"] <- `component`

      query_params["limit"] <- `limit`

      query_params["keepNonSpecific"] <- `keep_non_specific`

      query_params["consolidate"] <- `consolidate`

      local_var_url_path <- "/datasets/{datasets}/expressions/pca"
      if (!missing(`datasets`)) {
        local_var_url_path <- gsub(paste0("\\{", "datasets", "\\}"), URLencode(as.character(`datasets`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_expression_pca`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListExperimentExpressionLevelsValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the platform of a dataset
    #'
    #' @description
    #' Retrieve the platform of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListArrayDesignValueObject
    #' @keywords internal
    get_dataset_platforms = function(dataset, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_platforms_with_http_info(dataset, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the platform of a dataset
    #'
    #' @description
    #' Retrieve the platform of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListArrayDesignValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_platforms_with_http_info = function(dataset, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }


      local_var_url_path <- "/datasets/{dataset}/platforms"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_platforms`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListArrayDesignValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve raw expression data of a dataset
    #'
    #' @description
    #' Retrieve raw expression data of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return data.frame
    #' @keywords internal
    get_dataset_raw_expression = function(dataset, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_raw_expression_with_http_info(dataset, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve raw expression data of a dataset
    #'
    #' @description
    #' Retrieve raw expression data of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (data.frame) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_raw_expression_with_http_info = function(dataset, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }


      local_var_url_path <- "/datasets/{dataset}/data/raw"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_raw_expression`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("text/tab-separated-values; charset=UTF-8", "application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "data.frame", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the samples of a dataset
    #'
    #' @description
    #' Retrieve the samples of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListBioAssayValueObject
    #' @keywords internal
    get_dataset_samples = function(dataset, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_samples_with_http_info(dataset, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the samples of a dataset
    #'
    #' @description
    #' Retrieve the samples of a dataset
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListBioAssayValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_samples_with_http_info = function(dataset, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }


      local_var_url_path <- "/datasets/{dataset}/samples"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_samples`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListBioAssayValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the singular value decomposition (SVD) of a dataset expression data
    #'
    #' @description
    #' Retrieve the singular value decomposition (SVD) of a dataset expression data
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectSimpleSVDValueObject
    #' @keywords internal
    get_dataset_svd = function(dataset, data_file = NULL, ...) {
      local_var_response <- self$get_dataset_svd_with_http_info(dataset, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the singular value decomposition (SVD) of a dataset expression data
    #'
    #' @description
    #' Retrieve the singular value decomposition (SVD) of a dataset expression data
    #'
    #' @param dataset 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectSimpleSVDValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_dataset_svd_with_http_info = function(dataset, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }


      local_var_url_path <- "/datasets/{dataset}/svd"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_dataset_svd`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectSimpleSVDValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve all datasets
    #'
    #' @description
    #' Retrieve all datasets
    #'
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectExpressionExperimentValueObject
    #' @keywords internal
    get_datasets = function(filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      local_var_response <- self$get_datasets_with_http_info(filter, offset, limit, sort, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve all datasets
    #'
    #' @description
    #' Retrieve all datasets
    #'
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectExpressionExperimentValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_datasets_with_http_info = function(filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE


      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_datasets, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_datasets, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_datasets, must be bigger than or equal to 1.")
      }

      if (!str_detect(`sort`, "^(\\+|-?)(\\w+)$")) {
        stop("Invalid value for `sort` when calling DefaultApi$get_datasets, must conform to the pattern ^(\\+|-?)(\\w+)$.")
      }

      query_params["filter"] <- `filter`

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      query_params["sort"] <- `sort`

      local_var_url_path <- "/datasets"
      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_datasets`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectExpressionExperimentValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve datasets by their identifiers
    #'
    #' @description
    #' Retrieve datasets by their identifiers
    #'
    #' @param dataset 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectExpressionExperimentValueObject
    #' @keywords internal
    get_datasets_by_ids = function(dataset, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      local_var_response <- self$get_datasets_by_ids_with_http_info(dataset, filter, offset, limit, sort, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve datasets by their identifiers
    #'
    #' @description
    #' Retrieve datasets by their identifiers
    #'
    #' @param dataset 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectExpressionExperimentValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_datasets_by_ids_with_http_info = function(dataset, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`dataset`)) {
        stop("Missing required parameter `dataset`.")
      }



      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_datasets_by_ids, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_datasets_by_ids, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_datasets_by_ids, must be bigger than or equal to 1.")
      }

      if (!str_detect(`sort`, "^(\\+|-?)(\\w+)$")) {
        stop("Invalid value for `sort` when calling DefaultApi$get_datasets_by_ids, must conform to the pattern ^(\\+|-?)(\\w+)$.")
      }

      query_params["filter"] <- `filter`

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      query_params["sort"] <- `sort`

      local_var_url_path <- "/datasets/{dataset}"
      if (!missing(`dataset`)) {
        local_var_url_path <- gsub(paste0("\\{", "dataset", "\\}"), URLencode(as.character(`dataset`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_datasets_by_ids`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectExpressionExperimentValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the coexpression of two given genes
    #'
    #' @description
    #' Retrieve the coexpression of two given genes
    #'
    #' @param gene 
    #' @param with (optional) No description
    #' @param limit (optional) No description (default value: 100)
    #' @param stringency (optional) No description (default value: 1)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListCoexpressionValueObjectExt
    #' @keywords internal
    get_gene_gene_coexpression = function(gene, with = NULL, limit = 100, stringency = 1, data_file = NULL, ...) {
      local_var_response <- self$get_gene_gene_coexpression_with_http_info(gene, with, limit, stringency, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the coexpression of two given genes
    #'
    #' @description
    #' Retrieve the coexpression of two given genes
    #'
    #' @param gene 
    #' @param with (optional) No description
    #' @param limit (optional) No description (default value: 100)
    #' @param stringency (optional) No description (default value: 1)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListCoexpressionValueObjectExt) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_gene_gene_coexpression_with_http_info = function(gene, with = NULL, limit = 100, stringency = 1, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`gene`)) {
        stop("Missing required parameter `gene`.")
      }



      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_gene_gene_coexpression, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_gene_gene_coexpression, must be bigger than or equal to 1.")
      }


      query_params["with"] <- `with`

      query_params["limit"] <- `limit`

      query_params["stringency"] <- `stringency`

      local_var_url_path <- "/genes/{gene}/coexpression"
      if (!missing(`gene`)) {
        local_var_url_path <- gsub(paste0("\\{", "gene", "\\}"), URLencode(as.character(`gene`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_gene_gene_coexpression`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListCoexpressionValueObjectExt", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the GO terms associated to a gene
    #'
    #' @description
    #' Retrieve the GO terms associated to a gene
    #'
    #' @param gene 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListGeneOntologyTermValueObject
    #' @keywords internal
    get_gene_go_terms = function(gene, data_file = NULL, ...) {
      local_var_response <- self$get_gene_go_terms_with_http_info(gene, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the GO terms associated to a gene
    #'
    #' @description
    #' Retrieve the GO terms associated to a gene
    #'
    #' @param gene 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListGeneOntologyTermValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_gene_go_terms_with_http_info = function(gene, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`gene`)) {
        stop("Missing required parameter `gene`.")
      }


      local_var_url_path <- "/genes/{gene}/goTerms"
      if (!missing(`gene`)) {
        local_var_url_path <- gsub(paste0("\\{", "gene", "\\}"), URLencode(as.character(`gene`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_gene_go_terms`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListGeneOntologyTermValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the physical locations of a given gene
    #'
    #' @description
    #' Retrieve the physical locations of a given gene
    #'
    #' @param gene 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListPhysicalLocationValueObject
    #' @keywords internal
    get_gene_locations = function(gene, data_file = NULL, ...) {
      local_var_response <- self$get_gene_locations_with_http_info(gene, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the physical locations of a given gene
    #'
    #' @description
    #' Retrieve the physical locations of a given gene
    #'
    #' @param gene 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListPhysicalLocationValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_gene_locations_with_http_info = function(gene, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`gene`)) {
        stop("Missing required parameter `gene`.")
      }


      local_var_url_path <- "/genes/{gene}/locations"
      if (!missing(`gene`)) {
        local_var_url_path <- gsub(paste0("\\{", "gene", "\\}"), URLencode(as.character(`gene`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_gene_locations`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListPhysicalLocationValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve physical locations for a given gene and taxon
    #'
    #' @description
    #' Retrieve physical locations for a given gene and taxon
    #'
    #' @param taxon 
    #' @param gene 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListPhysicalLocationValueObject
    #' @keywords internal
    get_gene_locations_in_taxon = function(taxon, gene, data_file = NULL, ...) {
      local_var_response <- self$get_gene_locations_in_taxon_with_http_info(taxon, gene, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve physical locations for a given gene and taxon
    #'
    #' @description
    #' Retrieve physical locations for a given gene and taxon
    #'
    #' @param taxon 
    #' @param gene 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListPhysicalLocationValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_gene_locations_in_taxon_with_http_info = function(taxon, gene, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`taxon`)) {
        stop("Missing required parameter `taxon`.")
      }

      if (missing(`gene`)) {
        stop("Missing required parameter `gene`.")
      }



      local_var_url_path <- "/taxa/{taxon}/genes/{gene}/locations"
      if (!missing(`taxon`)) {
        local_var_url_path <- gsub(paste0("\\{", "taxon", "\\}"), URLencode(as.character(`taxon`), reserved = TRUE), local_var_url_path)
      }

      if (!missing(`gene`)) {
        local_var_url_path <- gsub(paste0("\\{", "gene", "\\}"), URLencode(as.character(`gene`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_gene_locations_in_taxon`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListPhysicalLocationValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the probes associated to a genes
    #'
    #' @description
    #' Retrieve the probes associated to a genes
    #'
    #' @param gene 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectCompositeSequenceValueObject
    #' @keywords internal
    get_gene_probes = function(gene, offset = 0, limit = 20, data_file = NULL, ...) {
      local_var_response <- self$get_gene_probes_with_http_info(gene, offset, limit, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the probes associated to a genes
    #'
    #' @description
    #' Retrieve the probes associated to a genes
    #'
    #' @param gene 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectCompositeSequenceValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_gene_probes_with_http_info = function(gene, offset = 0, limit = 20, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`gene`)) {
        stop("Missing required parameter `gene`.")
      }


      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_gene_probes, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_gene_probes, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_gene_probes, must be bigger than or equal to 1.")
      }

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      local_var_url_path <- "/genes/{gene}/probes"
      if (!missing(`gene`)) {
        local_var_url_path <- gsub(paste0("\\{", "gene", "\\}"), URLencode(as.character(`gene`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_gene_probes`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectCompositeSequenceValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve genes matching a gene identifier
    #'
    #' @description
    #' Retrieve genes matching a gene identifier
    #'
    #' @param genes 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListGeneValueObject
    #' @keywords internal
    get_genes = function(genes, data_file = NULL, ...) {
      local_var_response <- self$get_genes_with_http_info(genes, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve genes matching a gene identifier
    #'
    #' @description
    #' Retrieve genes matching a gene identifier
    #'
    #' @param genes 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListGeneValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_genes_with_http_info = function(genes, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`genes`)) {
        stop("Missing required parameter `genes`.")
      }


      local_var_url_path <- "/genes/{genes}"
      if (!missing(`genes`)) {
        local_var_url_path <- gsub(paste0("\\{", "genes", "\\}"), URLencode(as.character(`genes`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_genes`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListGeneValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the annotations of a given platform
    #'
    #' @description
    #' Retrieve the annotations of a given platform
    #'
    #' @param platform 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return data.frame
    #' @keywords internal
    get_platform_annotations = function(platform, data_file = NULL, ...) {
      local_var_response <- self$get_platform_annotations_with_http_info(platform, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the annotations of a given platform
    #'
    #' @description
    #' Retrieve the annotations of a given platform
    #'
    #' @param platform 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (data.frame) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_platform_annotations_with_http_info = function(platform, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`platform`)) {
        stop("Missing required parameter `platform`.")
      }


      local_var_url_path <- "/platforms/{platform}/annotations"
      if (!missing(`platform`)) {
        local_var_url_path <- gsub(paste0("\\{", "platform", "\\}"), URLencode(as.character(`platform`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_platform_annotations`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("text/tab-separated-values; charset=UTF-8")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "data.frame", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve all experiments within a given platform
    #'
    #' @description
    #' Retrieve all experiments within a given platform
    #'
    #' @param platform 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectExpressionExperimentValueObject
    #' @keywords internal
    get_platform_datasets = function(platform, offset = 0, limit = 20, data_file = NULL, ...) {
      local_var_response <- self$get_platform_datasets_with_http_info(platform, offset, limit, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve all experiments within a given platform
    #'
    #' @description
    #' Retrieve all experiments within a given platform
    #'
    #' @param platform 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectExpressionExperimentValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_platform_datasets_with_http_info = function(platform, offset = 0, limit = 20, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`platform`)) {
        stop("Missing required parameter `platform`.")
      }


      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_platform_datasets, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platform_datasets, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platform_datasets, must be bigger than or equal to 1.")
      }

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      local_var_url_path <- "/platforms/{platform}/datasets"
      if (!missing(`platform`)) {
        local_var_url_path <- gsub(paste0("\\{", "platform", "\\}"), URLencode(as.character(`platform`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_platform_datasets`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectExpressionExperimentValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the selected composite sequences for a given platform
    #'
    #' @description
    #' Retrieve the selected composite sequences for a given platform
    #'
    #' @param platform 
    #' @param probes 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectCompositeSequenceValueObject
    #' @keywords internal
    get_platform_element = function(platform, probes, offset = 0, limit = 20, data_file = NULL, ...) {
      local_var_response <- self$get_platform_element_with_http_info(platform, probes, offset, limit, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the selected composite sequences for a given platform
    #'
    #' @description
    #' Retrieve the selected composite sequences for a given platform
    #'
    #' @param platform 
    #' @param probes 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectCompositeSequenceValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_platform_element_with_http_info = function(platform, probes, offset = 0, limit = 20, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`platform`)) {
        stop("Missing required parameter `platform`.")
      }

      if (missing(`probes`)) {
        stop("Missing required parameter `probes`.")
      }



      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_platform_element, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platform_element, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platform_element, must be bigger than or equal to 1.")
      }

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      local_var_url_path <- "/platforms/{platform}/elements/{probes}"
      if (!missing(`platform`)) {
        local_var_url_path <- gsub(paste0("\\{", "platform", "\\}"), URLencode(as.character(`platform`), reserved = TRUE), local_var_url_path)
      }

      if (!missing(`probes`)) {
        local_var_url_path <- gsub(paste0("\\{", "probes", "\\}"), URLencode(as.character(`probes`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_platform_element`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectCompositeSequenceValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the genes associated to a probe in a given platform
    #'
    #' @description
    #' Retrieve the genes associated to a probe in a given platform
    #'
    #' @param platform 
    #' @param probe 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectGeneValueObject
    #' @keywords internal
    get_platform_element_genes = function(platform, probe, offset = 0, limit = 20, data_file = NULL, ...) {
      local_var_response <- self$get_platform_element_genes_with_http_info(platform, probe, offset, limit, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the genes associated to a probe in a given platform
    #'
    #' @description
    #' Retrieve the genes associated to a probe in a given platform
    #'
    #' @param platform 
    #' @param probe 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectGeneValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_platform_element_genes_with_http_info = function(platform, probe, offset = 0, limit = 20, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`platform`)) {
        stop("Missing required parameter `platform`.")
      }

      if (missing(`probe`)) {
        stop("Missing required parameter `probe`.")
      }



      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_platform_element_genes, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platform_element_genes, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platform_element_genes, must be bigger than or equal to 1.")
      }

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      local_var_url_path <- "/platforms/{platform}/elements/{probe}/genes"
      if (!missing(`platform`)) {
        local_var_url_path <- gsub(paste0("\\{", "platform", "\\}"), URLencode(as.character(`platform`), reserved = TRUE), local_var_url_path)
      }

      if (!missing(`probe`)) {
        local_var_url_path <- gsub(paste0("\\{", "probe", "\\}"), URLencode(as.character(`probe`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_platform_element_genes`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectGeneValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the composite sequences for a given platform
    #'
    #' @description
    #' Retrieve the composite sequences for a given platform
    #'
    #' @param platform 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectCompositeSequenceValueObject
    #' @keywords internal
    get_platform_elements = function(platform, offset = 0, limit = 20, data_file = NULL, ...) {
      local_var_response <- self$get_platform_elements_with_http_info(platform, offset, limit, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the composite sequences for a given platform
    #'
    #' @description
    #' Retrieve the composite sequences for a given platform
    #'
    #' @param platform 
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectCompositeSequenceValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_platform_elements_with_http_info = function(platform, offset = 0, limit = 20, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`platform`)) {
        stop("Missing required parameter `platform`.")
      }


      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_platform_elements, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platform_elements, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platform_elements, must be bigger than or equal to 1.")
      }

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      local_var_url_path <- "/platforms/{platform}/elements"
      if (!missing(`platform`)) {
        local_var_url_path <- gsub(paste0("\\{", "platform", "\\}"), URLencode(as.character(`platform`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_platform_elements`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectCompositeSequenceValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve all platforms
    #'
    #' @description
    #' Retrieve all platforms
    #'
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectArrayDesignValueObject
    #' @keywords internal
    get_platforms = function(filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      local_var_response <- self$get_platforms_with_http_info(filter, offset, limit, sort, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve all platforms
    #'
    #' @description
    #' Retrieve all platforms
    #'
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectArrayDesignValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_platforms_with_http_info = function(filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE


      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_platforms, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platforms, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platforms, must be bigger than or equal to 1.")
      }

      if (!str_detect(`sort`, "^(\\+|-?)(\\w+)$")) {
        stop("Invalid value for `sort` when calling DefaultApi$get_platforms, must conform to the pattern ^(\\+|-?)(\\w+)$.")
      }

      query_params["filter"] <- `filter`

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      query_params["sort"] <- `sort`

      local_var_url_path <- "/platforms"
      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_platforms`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectArrayDesignValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve all platforms matching a set of platform identifiers
    #'
    #' @description
    #' Retrieve all platforms matching a set of platform identifiers
    #'
    #' @param platform 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectArrayDesignValueObject
    #' @keywords internal
    get_platforms_by_ids = function(platform, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      local_var_response <- self$get_platforms_by_ids_with_http_info(platform, filter, offset, limit, sort, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve all platforms matching a set of platform identifiers
    #'
    #' @description
    #' Retrieve all platforms matching a set of platform identifiers
    #'
    #' @param platform 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectArrayDesignValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_platforms_by_ids_with_http_info = function(platform, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`platform`)) {
        stop("Missing required parameter `platform`.")
      }



      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_platforms_by_ids, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platforms_by_ids, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_platforms_by_ids, must be bigger than or equal to 1.")
      }

      if (!str_detect(`sort`, "^(\\+|-?)(\\w+)$")) {
        stop("Invalid value for `sort` when calling DefaultApi$get_platforms_by_ids, must conform to the pattern ^(\\+|-?)(\\w+)$.")
      }

      query_params["filter"] <- `filter`

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      query_params["sort"] <- `sort`

      local_var_url_path <- "/platforms/{platform}"
      if (!missing(`platform`)) {
        local_var_url_path <- gsub(paste0("\\{", "platform", "\\}"), URLencode(as.character(`platform`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_platforms_by_ids`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectArrayDesignValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve a single analysis result set by its identifier
    #'
    #' @description
    #' Retrieve a single analysis result set by its identifier
    #'
    #' @param result_set 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectExpressionAnalysisResultSetValueObject
    #' @keywords internal
    get_result_set = function(result_set, data_file = NULL, ...) {
      local_var_response <- self$get_result_set_with_http_info(result_set, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve a single analysis result set by its identifier
    #'
    #' @description
    #' Retrieve a single analysis result set by its identifier
    #'
    #' @param result_set 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectExpressionAnalysisResultSetValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_result_set_with_http_info = function(result_set, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`result_set`)) {
        stop("Missing required parameter `result_set`.")
      }


      local_var_url_path <- "/resultSets/{resultSet}"
      if (!missing(`result_set`)) {
        local_var_url_path <- gsub(paste0("\\{", "resultSet", "\\}"), URLencode(as.character(`result_set`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_result_set`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectExpressionAnalysisResultSetValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve a single analysis result set by its identifier as a tab-separated values
    #'
    #' @description
    #' Retrieve a single analysis result set by its identifier as a tab-separated values
    #'
    #' @param result_set_ 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return data.frame
    #' @keywords internal
    get_result_set_as_tsv = function(result_set_, data_file = NULL, ...) {
      local_var_response <- self$get_result_set_as_tsv_with_http_info(result_set_, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve a single analysis result set by its identifier as a tab-separated values
    #'
    #' @description
    #' Retrieve a single analysis result set by its identifier as a tab-separated values
    #'
    #' @param result_set_ 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (data.frame) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_result_set_as_tsv_with_http_info = function(result_set_, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`result_set_`)) {
        stop("Missing required parameter `result_set_`.")
      }


      local_var_url_path <- "/resultSets/{resultSet_}"
      if (!missing(`result_set_`)) {
        local_var_url_path <- gsub(paste0("\\{", "resultSet_", "\\}"), URLencode(as.character(`result_set_`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_result_set_as_tsv`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("text/tab-separated-values; charset=UTF-8")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "data.frame", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve all result sets matching the provided criteria
    #'
    #' @description
    #' Retrieve all result sets matching the provided criteria
    #'
    #' @param datasets (optional) No description
    #' @param database_entries (optional) No description
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectExpressionAnalysisResultSetValueObject
    #' @keywords internal
    get_result_sets = function(datasets = NULL, database_entries = NULL, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      local_var_response <- self$get_result_sets_with_http_info(datasets, database_entries, filter, offset, limit, sort, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve all result sets matching the provided criteria
    #'
    #' @description
    #' Retrieve all result sets matching the provided criteria
    #'
    #' @param datasets (optional) No description
    #' @param database_entries (optional) No description
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectExpressionAnalysisResultSetValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_result_sets_with_http_info = function(datasets = NULL, database_entries = NULL, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE




      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_result_sets, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_result_sets, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_result_sets, must be bigger than or equal to 1.")
      }

      if (!str_detect(`sort`, "^(\\+|-?)(\\w+)$")) {
        stop("Invalid value for `sort` when calling DefaultApi$get_result_sets, must conform to the pattern ^(\\+|-?)(\\w+)$.")
      }

      query_params["datasets"] <- `datasets`

      query_params["databaseEntries"] <- `database_entries`

      query_params["filter"] <- `filter`

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      query_params["sort"] <- `sort`

      local_var_url_path <- "/resultSets"
      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_result_sets`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectExpressionAnalysisResultSetValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve all available taxa
    #'
    #' @description
    #' Retrieve all available taxa
    #'
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListTaxonValueObject
    #' @keywords internal
    get_taxa = function(data_file = NULL, ...) {
      local_var_response <- self$get_taxa_with_http_info(data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve all available taxa
    #'
    #' @description
    #' Retrieve all available taxa
    #'
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListTaxonValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_taxa_with_http_info = function(data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      local_var_url_path <- "/taxa"
      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_taxa`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListTaxonValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve taxa by their identifiers
    #'
    #' @description
    #' Retrieve taxa by their identifiers
    #'
    #' @param taxa 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListTaxonValueObject
    #' @keywords internal
    get_taxa_by_ids = function(taxa, data_file = NULL, ...) {
      local_var_response <- self$get_taxa_by_ids_with_http_info(taxa, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve taxa by their identifiers
    #'
    #' @description
    #' Retrieve taxa by their identifiers
    #'
    #' @param taxa 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListTaxonValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_taxa_by_ids_with_http_info = function(taxa, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`taxa`)) {
        stop("Missing required parameter `taxa`.")
      }


      local_var_url_path <- "/taxa/{taxa}"
      if (!missing(`taxa`)) {
        local_var_url_path <- gsub(paste0("\\{", "taxa", "\\}"), URLencode(as.character(`taxa`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_taxa_by_ids`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListTaxonValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve the datasets for a given taxon
    #'
    #' @description
    #' Retrieve the datasets for a given taxon
    #'
    #' @param taxon 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectExpressionExperimentValueObject
    #' @keywords internal
    get_taxon_datasets = function(taxon, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      local_var_response <- self$get_taxon_datasets_with_http_info(taxon, filter, offset, limit, sort, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve the datasets for a given taxon
    #'
    #' @description
    #' Retrieve the datasets for a given taxon
    #'
    #' @param taxon 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectExpressionExperimentValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_taxon_datasets_with_http_info = function(taxon, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`taxon`)) {
        stop("Missing required parameter `taxon`.")
      }



      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$get_taxon_datasets, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$get_taxon_datasets, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$get_taxon_datasets, must be bigger than or equal to 1.")
      }

      if (!str_detect(`sort`, "^(\\+|-?)(\\w+)$")) {
        stop("Invalid value for `sort` when calling DefaultApi$get_taxon_datasets, must conform to the pattern ^(\\+|-?)(\\w+)$.")
      }

      query_params["filter"] <- `filter`

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      query_params["sort"] <- `sort`

      local_var_url_path <- "/taxa/{taxon}/datasets"
      if (!missing(`taxon`)) {
        local_var_url_path <- gsub(paste0("\\{", "taxon", "\\}"), URLencode(as.character(`taxon`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_taxon_datasets`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectExpressionExperimentValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve all genes in a given taxon
    #'
    #' @description
    #' Retrieve all genes in a given taxon
    #'
    #' @param taxon 
    #' @param gene 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListGeneValueObject
    #' @keywords internal
    get_taxon_genes = function(taxon, gene, data_file = NULL, ...) {
      local_var_response <- self$get_taxon_genes_with_http_info(taxon, gene, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve all genes in a given taxon
    #'
    #' @description
    #' Retrieve all genes in a given taxon
    #'
    #' @param taxon 
    #' @param gene 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListGeneValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_taxon_genes_with_http_info = function(taxon, gene, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`taxon`)) {
        stop("Missing required parameter `taxon`.")
      }

      if (missing(`gene`)) {
        stop("Missing required parameter `gene`.")
      }



      local_var_url_path <- "/taxa/{taxon}/genes/{gene}"
      if (!missing(`taxon`)) {
        local_var_url_path <- gsub(paste0("\\{", "taxon", "\\}"), URLencode(as.character(`taxon`), reserved = TRUE), local_var_url_path)
      }

      if (!missing(`gene`)) {
        local_var_url_path <- gsub(paste0("\\{", "gene", "\\}"), URLencode(as.character(`gene`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_taxon_genes`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListGeneValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve genes overlapping a given region in a taxon
    #'
    #' @description
    #' Retrieve genes overlapping a given region in a taxon
    #'
    #' @param taxon 
    #' @param chromosome 
    #' @param strand (optional) No description (default value: "+")
    #' @param start (optional) No description
    #' @param size (optional) No description
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListGeneValueObject
    #' @keywords internal
    get_taxon_genes_overlapping_chromosome = function(taxon, chromosome, strand = "+", start = NULL, size = NULL, data_file = NULL, ...) {
      local_var_response <- self$get_taxon_genes_overlapping_chromosome_with_http_info(taxon, chromosome, strand, start, size, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve genes overlapping a given region in a taxon
    #'
    #' @description
    #' Retrieve genes overlapping a given region in a taxon
    #'
    #' @param taxon 
    #' @param chromosome 
    #' @param strand (optional) No description (default value: "+")
    #' @param start (optional) No description
    #' @param size (optional) No description
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListGeneValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    get_taxon_genes_overlapping_chromosome_with_http_info = function(taxon, chromosome, strand = "+", start = NULL, size = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`taxon`)) {
        stop("Missing required parameter `taxon`.")
      }

      if (missing(`chromosome`)) {
        stop("Missing required parameter `chromosome`.")
      }






      query_params["strand"] <- `strand`

      query_params["start"] <- `start`

      query_params["size"] <- `size`

      local_var_url_path <- "/taxa/{taxon}/chromosomes/{chromosome}/genes"
      if (!missing(`taxon`)) {
        local_var_url_path <- gsub(paste0("\\{", "taxon", "\\}"), URLencode(as.character(`taxon`), reserved = TRUE), local_var_url_path)
      }

      if (!missing(`chromosome`)) {
        local_var_url_path <- gsub(paste0("\\{", "chromosome", "\\}"), URLencode(as.character(`chromosome`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `get_taxon_genes_overlapping_chromosome`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListGeneValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Search everything in Gemma.
    #'
    #' @description
    #' Search everything in Gemma.
    #'
    #' @param query (optional) No description
    #' @param taxon (optional) No description
    #' @param platform (optional) No description
    #' @param result_types (optional) No description
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return SearchResultsResponseDataObject
    #' @keywords internal
    search = function(query = NULL, taxon = NULL, platform = NULL, result_types = NULL, limit = 20, data_file = NULL, ...) {
      local_var_response <- self$search_with_http_info(query, taxon, platform, result_types, limit, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Search everything in Gemma.
    #'
    #' @description
    #' Search everything in Gemma.
    #'
    #' @param query (optional) No description
    #' @param taxon (optional) No description
    #' @param platform (optional) No description
    #' @param result_types (optional) No description
    #' @param limit (optional) No description (default value: 20)
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (SearchResultsResponseDataObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    search_with_http_info = function(query = NULL, taxon = NULL, platform = NULL, result_types = NULL, limit = 20, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE





      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$search, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$search, must be bigger than or equal to 1.")
      }

      query_params["query"] <- `query`

      query_params["taxon"] <- `taxon`

      query_params["platform"] <- `platform`

      query_params["resultTypes"] <- `result_types`

      query_params["limit"] <- `limit`

      local_var_url_path <- "/search"
      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `search`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "SearchResultsResponseDataObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Search for annotation tags
    #'
    #' @description
    #' Search for annotation tags
    #'
    #' @param query 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ResponseDataObjectListAnnotationSearchResultValueObject
    #' @keywords internal
    search_annotations = function(query, data_file = NULL, ...) {
      local_var_response <- self$search_annotations_with_http_info(query, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Search for annotation tags
    #'
    #' @description
    #' Search for annotation tags
    #'
    #' @param query 
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ResponseDataObjectListAnnotationSearchResultValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    search_annotations_with_http_info = function(query, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`query`)) {
        stop("Missing required parameter `query`.")
      }


      local_var_url_path <- "/annotations/search/{query}"
      if (!missing(`query`)) {
        local_var_url_path <- gsub(paste0("\\{", "query", "\\}"), URLencode(as.character(`query`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `search_annotations`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "ResponseDataObjectListAnnotationSearchResultValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve datasets associated to an annotation tags search
    #'
    #' @description
    #' Retrieve datasets associated to an annotation tags search
    #'
    #' @param query 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectExpressionExperimentValueObject
    #' @keywords internal
    search_datasets = function(query, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      local_var_response <- self$search_datasets_with_http_info(query, filter, offset, limit, sort, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve datasets associated to an annotation tags search
    #'
    #' @description
    #' Retrieve datasets associated to an annotation tags search
    #'
    #' @param query 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectExpressionExperimentValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    search_datasets_with_http_info = function(query, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`query`)) {
        stop("Missing required parameter `query`.")
      }



      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$search_datasets, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$search_datasets, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$search_datasets, must be bigger than or equal to 1.")
      }

      if (!str_detect(`sort`, "^(\\+|-?)(\\w+)$")) {
        stop("Invalid value for `sort` when calling DefaultApi$search_datasets, must conform to the pattern ^(\\+|-?)(\\w+)$.")
      }

      query_params["filter"] <- `filter`

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      query_params["sort"] <- `sort`

      local_var_url_path <- "/annotations/search/{query}/datasets"
      if (!missing(`query`)) {
        local_var_url_path <- gsub(paste0("\\{", "query", "\\}"), URLencode(as.character(`query`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `search_datasets`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectExpressionExperimentValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve datasets within a given taxa associated to an annotation tags search
    #'
    #' @description
    #' Retrieve datasets within a given taxa associated to an annotation tags search
    #'
    #' @param taxon 
    #' @param query 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PaginatedResponseDataObjectExpressionExperimentValueObject
    #' @keywords internal
    search_taxon_datasets = function(taxon, query, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      local_var_response <- self$search_taxon_datasets_with_http_info(taxon, query, filter, offset, limit, sort, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve datasets within a given taxa associated to an annotation tags search
    #'
    #' @description
    #' Retrieve datasets within a given taxa associated to an annotation tags search
    #'
    #' @param taxon 
    #' @param query 
    #' @param filter (optional) No description (default value: "")
    #' @param offset (optional) No description (default value: 0)
    #' @param limit (optional) No description (default value: 20)
    #' @param sort (optional) No description (default value: "+id")
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PaginatedResponseDataObjectExpressionExperimentValueObject) with additional information such as HTTP status code, headers
    #' @keywords internal
    search_taxon_datasets_with_http_info = function(taxon, query, filter = "", offset = 0, limit = 20, sort = "+id", data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`taxon`)) {
        stop("Missing required parameter `taxon`.")
      }

      if (missing(`query`)) {
        stop("Missing required parameter `query`.")
      }




      if (`offset` < 0) {
        stop("Invalid value for `offset` when calling DefaultApi$search_taxon_datasets, must be bigger than or equal to 0.")
      }

      if (`limit` > 100) {
        stop("Invalid value for `limit` when calling DefaultApi$search_taxon_datasets, must be smaller than or equal to 100.")
      }
      if (`limit` < 1) {
        stop("Invalid value for `limit` when calling DefaultApi$search_taxon_datasets, must be bigger than or equal to 1.")
      }

      if (!str_detect(`sort`, "^(\\+|-?)(\\w+)$")) {
        stop("Invalid value for `sort` when calling DefaultApi$search_taxon_datasets, must conform to the pattern ^(\\+|-?)(\\w+)$.")
      }

      query_params["filter"] <- `filter`

      query_params["offset"] <- `offset`

      query_params["limit"] <- `limit`

      query_params["sort"] <- `sort`

      local_var_url_path <- "/annotations/{taxon}/search/{query}/datasets"
      if (!missing(`taxon`)) {
        local_var_url_path <- gsub(paste0("\\{", "taxon", "\\}"), URLencode(as.character(`taxon`), reserved = TRUE), local_var_url_path)
      }

      if (!missing(`query`)) {
        local_var_url_path <- gsub(paste0("\\{", "query", "\\}"), URLencode(as.character(`query`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (is.null(self$api_client$username) || is.null(self$api_client$password)) {
        stop("username, password in `api_client` must be set for authentication in the endpoint `search_taxon_datasets`.")
      }
      header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      # API key authentication

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
        # save response in a file
        if (!is.null(data_file)) {
            write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response, "PaginatedResponseDataObjectExpressionExperimentValueObject", loadNamespace("gemma.R")),
          error = function(e) {
             stop("Failed to deserialize response")
          }
        )
        local_var_resp$content <- deserialized_resp_obj
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    }
  )
)
